{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:24:51.672960Z",
     "start_time": "2018-11-26T20:24:51.374631Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Parameters specific to where your folders are and your data\n",
    "parameter_file = 'params/params.yaml'\n",
    "import yaml\n",
    "import sys\n",
    "\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "#p = dic2obj(**doc)\n",
    "\n",
    "data_folder = doc['data_folder']\n",
    "tissues = doc['tissues'].split(',')\n",
    "sys.path.append(doc['pipeline_path'])\n",
    "ref_fa = doc['ref_fa']\n",
    "annotation=doc['annotation']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:24:51.679157Z",
     "start_time": "2018-11-26T20:24:51.675822Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss_annotation = doc['tss_annotation']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:05.846176Z",
     "start_time": "2018-11-26T20:24:51.681931Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Number of tissues: ', 15)\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import pickle\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "from matplotlib import pyplot as plt\n",
    "from itertools import product\n",
    "import glob\n",
    "import re\n",
    "from matplotlib_venn import venn2\n",
    "from matplotlib import rcParams\n",
    "import inspect\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "sys.setrecursionlimit(3000)\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "rcParams['figure.figsize'] = 8, 6\n",
    "import tqdm\n",
    "### notebook specific configuration ###\n",
    "from os.path import basename\n",
    "#mpl.style.use('ggplot')\n",
    "mpl.style.use('fivethirtyeight')\n",
    "from cycler import cycler\n",
    "mpl.rcParams['axes.prop_cycle'] = cycler(color='bgrcmyk')\n",
    "\n",
    "from Homer import *\n",
    "\n",
    "print('Number of tissues: ',len(tissues))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:06.196373Z",
     "start_time": "2018-11-26T20:25:05.848480Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Name</th>\n",
       "      <th>Full</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDM1hKLA_ATAC1</td>\n",
       "      <td>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDM1hKLA_ATAC2</td>\n",
       "      <td>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>GRO</td>\n",
       "      <td>BMDM1hKLA_GRO1</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>BMDM1hKLA_GROCap1</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDMwt</td>\n",
       "      <td>GRO</td>\n",
       "      <td>BMDMwt_GRO1</td>\n",
       "      <td>CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Tissue Experiment  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  BMDM1hKLA       ATAC   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  BMDM1hKLA       ATAC   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  BMDM1hKLA        GRO   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA     GROCap   \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....     BMDMwt        GRO   \n",
       "\n",
       "                                                                 Name  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...    BMDM1hKLA_ATAC1   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...    BMDM1hKLA_ATAC2   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...     BMDM1hKLA_GRO1   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA_GROCap1   \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....        BMDMwt_GRO1   \n",
       "\n",
       "                                                                                                 Full  \n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...  \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...  \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...  \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...  \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....  CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_samples = pd.read_csv('../raw_metaSamples.tsv',sep=\"\\t\",index_col=0)\n",
    "meta_samples.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load anno_peaks. \n",
    "Later will filter based on Distance to TSS\n",
    "\n",
    "## Load peaks_tissues matrix\n",
    "Later will be used to merge two data types in a tissue and for more confident peaks if they have more than 1 tissues present\n",
    "\n",
    "## Load tss\n",
    "\n",
    "## Load txn_df\n",
    "\n",
    "## Load geneID_df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:27.308761Z",
     "start_time": "2018-11-26T20:25:06.198390Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "      <th>Nearest gene</th>\n",
       "      <th>Nearest gene_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>picr_9</td>\n",
       "      <td>30423275</td>\n",
       "      <td>30423564</td>\n",
       "      <td>+</td>\n",
       "      <td>10.506897</td>\n",
       "      <td>exon (asmbl_83977, exon 1 of 1)</td>\n",
       "      <td>asmbl_83977</td>\n",
       "      <td>4098.0</td>\n",
       "      <td>False</td>\n",
       "      <td>SALL1</td>\n",
       "      <td>gene_29627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>picr_27</td>\n",
       "      <td>14937806</td>\n",
       "      <td>14938107</td>\n",
       "      <td>-</td>\n",
       "      <td>22.268621</td>\n",
       "      <td>promoter-TSS (asmbl_40109)</td>\n",
       "      <td>asmbl_40105</td>\n",
       "      <td>-159.0</td>\n",
       "      <td>False</td>\n",
       "      <td>RBM25</td>\n",
       "      <td>gene_13991_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>picr_2</td>\n",
       "      <td>5127066</td>\n",
       "      <td>5127335</td>\n",
       "      <td>+</td>\n",
       "      <td>1.732333</td>\n",
       "      <td>intron (asmbl_29716, intron 9 of 9)</td>\n",
       "      <td>asmbl_29719</td>\n",
       "      <td>-8640.0</td>\n",
       "      <td>True</td>\n",
       "      <td>FAM83C</td>\n",
       "      <td>gene_10160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>picr_15</td>\n",
       "      <td>26110487</td>\n",
       "      <td>26110670</td>\n",
       "      <td>+</td>\n",
       "      <td>1.659667</td>\n",
       "      <td>intron (asmbl_22566, intron 1 of 4)</td>\n",
       "      <td>asmbl_22566</td>\n",
       "      <td>24071.0</td>\n",
       "      <td>True</td>\n",
       "      <td>ZNRF1</td>\n",
       "      <td>gene_7657_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>73670402</td>\n",
       "      <td>73670628</td>\n",
       "      <td>+</td>\n",
       "      <td>1.694250</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>asmbl_3271</td>\n",
       "      <td>-74096.0</td>\n",
       "      <td>True</td>\n",
       "      <td>MAN2A1</td>\n",
       "      <td>gene_975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr     Start       End Strand       Stat  \\\n",
       "ID                                                  \n",
       "0    picr_9  30423275  30423564      +  10.506897   \n",
       "1   picr_27  14937806  14938107      -  22.268621   \n",
       "2    picr_2   5127066   5127335      +   1.732333   \n",
       "3   picr_15  26110487  26110670      +   1.659667   \n",
       "4    picr_0  73670402  73670628      +   1.694250   \n",
       "\n",
       "                             Annotation  Nearest TSS  Distance to TSS  \\\n",
       "ID                                                                      \n",
       "0       exon (asmbl_83977, exon 1 of 1)  asmbl_83977           4098.0   \n",
       "1            promoter-TSS (asmbl_40109)  asmbl_40105           -159.0   \n",
       "2   intron (asmbl_29716, intron 9 of 9)  asmbl_29719          -8640.0   \n",
       "3   intron (asmbl_22566, intron 1 of 4)  asmbl_22566          24071.0   \n",
       "4                            Intergenic   asmbl_3271         -74096.0   \n",
       "\n",
       "    isSameStrand Nearest gene Nearest gene_id  \n",
       "ID                                             \n",
       "0          False        SALL1      gene_29627  \n",
       "1          False        RBM25    gene_13991_3  \n",
       "2           True       FAM83C      gene_10160  \n",
       "3           True        ZNRF1     gene_7657_3  \n",
       "4           True       MAN2A1        gene_975  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno_peaks = pd.read_csv('Results/merged/peaks_with_tss_distances.tsv',sep='\\t',index_col=0)\n",
    "anno_peaks.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:29.329061Z",
     "start_time": "2018-11-26T20:25:27.311011Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f04_peaks/Brain_START2_and_Brain_STARTinput1</th>\n",
       "      <th>f04_peaks/Brain_GROCap1_and_Brain_GRO1</th>\n",
       "      <th>f04_peaks/FemaleReproductive_START1_and_FemaleReproductive_STARTinput1</th>\n",
       "      <th>f04_peaks/Kidney_START1_and_Kidney_STARTinput1</th>\n",
       "      <th>f04_peaks/Muscle_START2_and_Muscle_STARTinput1</th>\n",
       "      <th>f04_peaks/Heart_START1_and_Heart_STARTinput1</th>\n",
       "      <th>f04_peaks/Lung_GROCap1_and_Lung_GRO1</th>\n",
       "      <th>f04_peaks/Liver_START2_and_Liver_STARTinput1</th>\n",
       "      <th>f04_peaks/MiscOrgans_START2_and_MiscOrgans_STARTinput1</th>\n",
       "      <th>f04_peaks/Muscle_START1_and_Muscle_STARTinput1</th>\n",
       "      <th>...</th>\n",
       "      <th>f04_peaks/Brain_START1_and_Brain_STARTinput2</th>\n",
       "      <th>f04_peaks/FemaleReproductive_START2_and_FemaleReproductive_STARTinput1</th>\n",
       "      <th>f04_peaks/Pancreas_START2_and_Pancreas_STARTinput1</th>\n",
       "      <th>f04_peaks/Brain_START3_and_Brain_STARTinput1</th>\n",
       "      <th>f04_peaks/Kidney_GROCap1_and_Kidney_GRO1</th>\n",
       "      <th>f04_peaks/Brain_START1_and_Brain_STARTinput1</th>\n",
       "      <th>f04_peaks/CHO_GROCap1_and_CHO_GRO1</th>\n",
       "      <th>f04_peaks/BMDMwt_GROCap1_and_BMDMwt_GRO1</th>\n",
       "      <th>f04_peaks/Heart_START2_and_Heart_STARTinput1</th>\n",
       "      <th>f04_peaks/Lung_START1_and_Lung_STARTinput1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15.5</td>\n",
       "      <td>5.12</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.50</td>\n",
       "      <td>22.50</td>\n",
       "      <td>2.34</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.37</td>\n",
       "      <td>1.93</td>\n",
       "      <td>...</td>\n",
       "      <td>23.00</td>\n",
       "      <td>16.10</td>\n",
       "      <td>10.10</td>\n",
       "      <td>11.60</td>\n",
       "      <td>1.05</td>\n",
       "      <td>23.00</td>\n",
       "      <td>3.12</td>\n",
       "      <td>4.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9.6</td>\n",
       "      <td>50.80</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.09</td>\n",
       "      <td>7.78</td>\n",
       "      <td>3.68</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.30</td>\n",
       "      <td>5.78</td>\n",
       "      <td>...</td>\n",
       "      <td>5.25</td>\n",
       "      <td>5.85</td>\n",
       "      <td>4.77</td>\n",
       "      <td>2.48</td>\n",
       "      <td>103.00</td>\n",
       "      <td>5.25</td>\n",
       "      <td>46.00</td>\n",
       "      <td>65.6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.15</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.5</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   f04_peaks/Brain_START2_and_Brain_STARTinput1  \\\n",
       "0                                          15.5   \n",
       "1                                           9.6   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           0.0   \n",
       "\n",
       "   f04_peaks/Brain_GROCap1_and_Brain_GRO1  \\\n",
       "0                                    5.12   \n",
       "1                                   50.80   \n",
       "2                                    0.00   \n",
       "3                                    0.00   \n",
       "4                                    0.00   \n",
       "\n",
       "   f04_peaks/FemaleReproductive_START1_and_FemaleReproductive_STARTinput1  \\\n",
       "0                                                0.0                        \n",
       "1                                                0.0                        \n",
       "2                                                0.0                        \n",
       "3                                                0.0                        \n",
       "4                                                0.0                        \n",
       "\n",
       "   f04_peaks/Kidney_START1_and_Kidney_STARTinput1  \\\n",
       "0                                           26.50   \n",
       "1                                            4.09   \n",
       "2                                            0.00   \n",
       "3                                            0.00   \n",
       "4                                            0.00   \n",
       "\n",
       "   f04_peaks/Muscle_START2_and_Muscle_STARTinput1  \\\n",
       "0                                           22.50   \n",
       "1                                            7.78   \n",
       "2                                            0.00   \n",
       "3                                            0.00   \n",
       "4                                            0.00   \n",
       "\n",
       "   f04_peaks/Heart_START1_and_Heart_STARTinput1  \\\n",
       "0                                          2.34   \n",
       "1                                          3.68   \n",
       "2                                          0.00   \n",
       "3                                          0.00   \n",
       "4                                          0.00   \n",
       "\n",
       "   f04_peaks/Lung_GROCap1_and_Lung_GRO1  \\\n",
       "0                                   0.0   \n",
       "1                                   0.0   \n",
       "2                                   0.0   \n",
       "3                                   0.0   \n",
       "4                                   0.0   \n",
       "\n",
       "   f04_peaks/Liver_START2_and_Liver_STARTinput1  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           2.5   \n",
       "\n",
       "   f04_peaks/MiscOrgans_START2_and_MiscOrgans_STARTinput1  \\\n",
       "0                                               9.37        \n",
       "1                                               6.30        \n",
       "2                                               0.00        \n",
       "3                                               0.00        \n",
       "4                                               0.00        \n",
       "\n",
       "   f04_peaks/Muscle_START1_and_Muscle_STARTinput1  \\\n",
       "0                                            1.93   \n",
       "1                                            5.78   \n",
       "2                                            0.00   \n",
       "3                                            0.00   \n",
       "4                                            0.00   \n",
       "\n",
       "                      ...                      \\\n",
       "0                     ...                       \n",
       "1                     ...                       \n",
       "2                     ...                       \n",
       "3                     ...                       \n",
       "4                     ...                       \n",
       "\n",
       "   f04_peaks/Brain_START1_and_Brain_STARTinput2  \\\n",
       "0                                         23.00   \n",
       "1                                          5.25   \n",
       "2                                          0.00   \n",
       "3                                          0.00   \n",
       "4                                          0.00   \n",
       "\n",
       "   f04_peaks/FemaleReproductive_START2_and_FemaleReproductive_STARTinput1  \\\n",
       "0                                              16.10                        \n",
       "1                                               5.85                        \n",
       "2                                               0.00                        \n",
       "3                                               0.00                        \n",
       "4                                               0.00                        \n",
       "\n",
       "   f04_peaks/Pancreas_START2_and_Pancreas_STARTinput1  \\\n",
       "0                                              10.10    \n",
       "1                                               4.77    \n",
       "2                                               0.00    \n",
       "3                                               0.00    \n",
       "4                                               0.00    \n",
       "\n",
       "   f04_peaks/Brain_START3_and_Brain_STARTinput1  \\\n",
       "0                                         11.60   \n",
       "1                                          2.48   \n",
       "2                                          0.00   \n",
       "3                                          0.00   \n",
       "4                                          0.00   \n",
       "\n",
       "   f04_peaks/Kidney_GROCap1_and_Kidney_GRO1  \\\n",
       "0                                      1.05   \n",
       "1                                    103.00   \n",
       "2                                      1.15   \n",
       "3                                      1.94   \n",
       "4                                      0.00   \n",
       "\n",
       "   f04_peaks/Brain_START1_and_Brain_STARTinput1  \\\n",
       "0                                         23.00   \n",
       "1                                          5.25   \n",
       "2                                          0.00   \n",
       "3                                          0.00   \n",
       "4                                          0.00   \n",
       "\n",
       "   f04_peaks/CHO_GROCap1_and_CHO_GRO1  \\\n",
       "0                                3.12   \n",
       "1                               46.00   \n",
       "2                                0.00   \n",
       "3                                0.00   \n",
       "4                                0.79   \n",
       "\n",
       "   f04_peaks/BMDMwt_GROCap1_and_BMDMwt_GRO1  \\\n",
       "0                                       4.7   \n",
       "1                                      65.6   \n",
       "2                                       0.0   \n",
       "3                                       0.0   \n",
       "4                                       0.0   \n",
       "\n",
       "   f04_peaks/Heart_START2_and_Heart_STARTinput1  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           0.0   \n",
       "\n",
       "   f04_peaks/Lung_START1_and_Lung_STARTinput1  \n",
       "0                                        14.4  \n",
       "1                                        10.7  \n",
       "2                                         0.0  \n",
       "3                                         0.0  \n",
       "4                                         0.0  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peaks_tissue = pd.read_csv(\"Results/merged/samples.merge.peaksexpression\",index_col=0,sep=\"\\t\")\n",
    "peaks_tissue.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Read in tss and txn_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:29.761882Z",
     "start_time": "2018-11-26T20:25:29.332423Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>cs</th>\n",
       "      <th>Length</th>\n",
       "      <th>actual_start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gene=SERHL;transcript_id=asmbl_2;cs=2</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>176</td>\n",
       "      <td>38694</td>\n",
       "      <td>+</td>\n",
       "      <td>asmbl_2</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>SERHL</td>\n",
       "      <td>asmbl_2</td>\n",
       "      <td>2</td>\n",
       "      <td>38518</td>\n",
       "      <td>176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_4;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7715</td>\n",
       "      <td>46630</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>1</td>\n",
       "      <td>38915</td>\n",
       "      <td>46630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_7;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7759</td>\n",
       "      <td>44597</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_7</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_7</td>\n",
       "      <td>1</td>\n",
       "      <td>36838</td>\n",
       "      <td>44597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_8;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7767</td>\n",
       "      <td>44597</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_8</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_8</td>\n",
       "      <td>1</td>\n",
       "      <td>36830</td>\n",
       "      <td>44597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_9;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7789</td>\n",
       "      <td>44568</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_9</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_9</td>\n",
       "      <td>1</td>\n",
       "      <td>36779</td>\n",
       "      <td>44568</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Chr  Start    End  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  picr_0    176  38694   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7715  46630   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7759  44597   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7767  44597   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7789  44568   \n",
       "\n",
       "                                                   Strand       ID    Parent  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...      +  asmbl_2  gene_1_2   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_4    gene_2   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_7    gene_2   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_8    gene_2   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_9    gene_2   \n",
       "\n",
       "                                                     gene_id   gene  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  gene_1_2  SERHL   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "\n",
       "                                                   transcript_id  cs  Length  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...       asmbl_2   2   38518   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_4   1   38915   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_7   1   36838   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_8   1   36830   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_9   1   36779   \n",
       "\n",
       "                                                    actual_start  \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...           176  \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...         46630  \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...         44597  \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...         44597  \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...         44568  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tss = pd.read_csv(tss_annotation,sep='\\t',index_col=0)\n",
    "\n",
    "tss.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:35.400215Z",
     "start_time": "2018-11-26T20:25:29.764838Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>peaks</th>\n",
       "      <th>closest_to_X_numberOfPeaks</th>\n",
       "      <th>bin_-1000_1000</th>\n",
       "      <th>sameStrand_bin_-1000_1000</th>\n",
       "      <th>samples</th>\n",
       "      <th>minDistance</th>\n",
       "      <th>minDistancePeak</th>\n",
       "      <th>maxValue</th>\n",
       "      <th>maxPeakId</th>\n",
       "      <th>maxSample</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>maxSamples</th>\n",
       "      <th>hasGene</th>\n",
       "      <th>Number of SS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>asmbl_2</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>inf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_4</th>\n",
       "      <td>[215685, 249167, 398989, 483229]</td>\n",
       "      <td>4</td>\n",
       "      <td>[215685]</td>\n",
       "      <td>[215685]</td>\n",
       "      <td>[CHO_START1_and_CHO_STARTinput1]</td>\n",
       "      <td>706.000000</td>\n",
       "      <td>215685</td>\n",
       "      <td>3.04</td>\n",
       "      <td>215685</td>\n",
       "      <td>NaN</td>\n",
       "      <td>picr_0</td>\n",
       "      <td>45849</td>\n",
       "      <td>45999</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>[CHO_START1_and_CHO_STARTinput1]</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_7</th>\n",
       "      <td>[104402]</td>\n",
       "      <td>1</td>\n",
       "      <td>[104402]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>inf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_8</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>inf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_9</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>inf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    peaks  closest_to_X_numberOfPeaks  \\\n",
       "asmbl_2                                []                           0   \n",
       "asmbl_4  [215685, 249167, 398989, 483229]                           4   \n",
       "asmbl_7                          [104402]                           1   \n",
       "asmbl_8                                []                           0   \n",
       "asmbl_9                                []                           0   \n",
       "\n",
       "        bin_-1000_1000 sameStrand_bin_-1000_1000  \\\n",
       "asmbl_2             []                        []   \n",
       "asmbl_4       [215685]                  [215685]   \n",
       "asmbl_7       [104402]                        []   \n",
       "asmbl_8             []                        []   \n",
       "asmbl_9             []                        []   \n",
       "\n",
       "                                  samples  minDistance minDistancePeak  \\\n",
       "asmbl_2                                []          inf             NaN   \n",
       "asmbl_4  [CHO_START1_and_CHO_STARTinput1]   706.000000          215685   \n",
       "asmbl_7                                []          inf             NaN   \n",
       "asmbl_8                                []          inf             NaN   \n",
       "asmbl_9                                []          inf             NaN   \n",
       "\n",
       "        maxValue maxPeakId maxSample     Chr  Start    End Strand gene_id  \\\n",
       "asmbl_2      NaN       NaN       NaN     NaN    NaN    NaN    NaN     NaN   \n",
       "asmbl_4     3.04    215685       NaN  picr_0  45849  45999      -  gene_2   \n",
       "asmbl_7      NaN       NaN       NaN     NaN    NaN    NaN    NaN     NaN   \n",
       "asmbl_8      NaN       NaN       NaN     NaN    NaN    NaN    NaN     NaN   \n",
       "asmbl_9      NaN       NaN       NaN     NaN    NaN    NaN    NaN     NaN   \n",
       "\n",
       "          gene transcript                        maxSamples  hasGene  \\\n",
       "asmbl_2    NaN        NaN                                []    False   \n",
       "asmbl_4  RRP7A    asmbl_4  [CHO_START1_and_CHO_STARTinput1]     True   \n",
       "asmbl_7    NaN        NaN                                []    False   \n",
       "asmbl_8    NaN        NaN                                []    False   \n",
       "asmbl_9    NaN        NaN                                []    False   \n",
       "\n",
       "         Number of SS  \n",
       "asmbl_2             0  \n",
       "asmbl_4             1  \n",
       "asmbl_7             0  \n",
       "asmbl_8             0  \n",
       "asmbl_9             0  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "txn_df = pickle.load(open('Results/tss_annotation/txn_df_02.p','rb'))# pd.read_csv('Results/txn_df.csv',index_col=0)\n",
    "txn_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:39.608807Z",
     "start_time": "2018-11-26T20:25:35.409826Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "geneID_df = pickle.load(open('Results/tss_annotation/geneID_df_02.p','rb'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge tissue columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:39.642741Z",
     "start_time": "2018-11-26T20:25:39.611525Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def merge_two_columns(df, mapping_dict):\n",
    "    '''Function that merges columns by taking the mean of them.\n",
    "    Merges based on if they have the same element in the meta_samples meta_column.\n",
    "    Returns:\n",
    "        new_df: Dataframe but with the columns of interest merged. Also column names are now based\n",
    "                on the unique meta_samples[meta_column].\n",
    "    '''\n",
    "\n",
    "    vals = mapping_dict.keys()#np.unique(meta_samples[meta_column].values)\n",
    "    new_df = pd.DataFrame(index=df.index, columns=vals)\n",
    "    for i in vals:\n",
    "        if not mapping_dict[i] == []:#(meta_samples[meta_column] == i).any():\n",
    "            new_col = (df.loc[:, mapping_dict[i]])\n",
    "            new_col = new_col.mean(axis=1)\n",
    "            new_df.loc[:, i] = new_col\n",
    "    new_df = new_df.loc[:, ~(new_df.isnull().all())]\n",
    "    return new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:39.695314Z",
     "start_time": "2018-11-26T20:25:39.645863Z"
    }
   },
   "outputs": [],
   "source": [
    "tissue_to_peaks = defaultdict(list)\n",
    "for val in peaks_tissue.columns.values:\n",
    "    count = 0\n",
    "    for t in np.unique(meta_samples[\"Tissue\"].values):\n",
    "        if t in val:\n",
    "            count += 1\n",
    "            tissue_to_peaks[t].append(val)\n",
    "            if count > 1:\n",
    "                print(\"This sample has multiple tissues associated with it\")\n",
    "                print(\"Sample: \",val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:41.850448Z",
     "start_time": "2018-11-26T20:25:39.698890Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "peaks_tissue_merged = merge_two_columns(peaks_tissue,mapping_dict=tissue_to_peaks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:41.943979Z",
     "start_time": "2018-11-26T20:25:41.857314Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Heart</th>\n",
       "      <th>BMDMwt</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>FemaleReproductive</th>\n",
       "      <th>CHO</th>\n",
       "      <th>Brain</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Liver</th>\n",
       "      <th>MiscOrgans</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Kidney</th>\n",
       "      <th>Pancreas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.17</td>\n",
       "      <td>4.7</td>\n",
       "      <td>3.455</td>\n",
       "      <td>8.050</td>\n",
       "      <td>4.1820</td>\n",
       "      <td>15.045714</td>\n",
       "      <td>4.800000</td>\n",
       "      <td>14.380000</td>\n",
       "      <td>10.485</td>\n",
       "      <td>12.215</td>\n",
       "      <td>10.576667</td>\n",
       "      <td>6.875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.84</td>\n",
       "      <td>65.6</td>\n",
       "      <td>5.195</td>\n",
       "      <td>2.925</td>\n",
       "      <td>48.2000</td>\n",
       "      <td>12.208571</td>\n",
       "      <td>3.566667</td>\n",
       "      <td>25.080000</td>\n",
       "      <td>7.100</td>\n",
       "      <td>6.780</td>\n",
       "      <td>37.786667</td>\n",
       "      <td>3.375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.1494</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.100000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.383333</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0798</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.880000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.646667</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.3274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.713333</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Heart  BMDMwt  Spleen  FemaleReproductive      CHO      Brain      Lung  \\\n",
       "0   1.17     4.7   3.455               8.050   4.1820  15.045714  4.800000   \n",
       "1   1.84    65.6   5.195               2.925  48.2000  12.208571  3.566667   \n",
       "2   0.00     0.0   0.000               0.000   0.1494   0.000000  0.000000   \n",
       "3   0.00     0.0   0.000               0.000   0.0798   0.000000  0.000000   \n",
       "4   0.00     0.0   0.000               0.000   0.3274   0.000000  0.000000   \n",
       "\n",
       "       Liver  MiscOrgans  Muscle     Kidney  Pancreas  \n",
       "0  14.380000      10.485  12.215  10.576667     6.875  \n",
       "1  25.080000       7.100   6.780  37.786667     3.375  \n",
       "2   1.100000       0.000   0.000   0.383333     0.000  \n",
       "3   0.880000       0.000   0.000   0.646667     0.000  \n",
       "4   1.713333       0.000   0.000   0.000000     0.000  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peaks_tissue_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create directory of different peak groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:35:53.590351Z",
     "start_time": "2018-11-26T20:35:53.515052Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "if not os.path.exists(\"Results/peak_groups\"):\n",
    "    os.mkdir(\"Results/peak_groups\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:41.999462Z",
     "start_time": "2018-11-26T20:25:41.967788Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def save_peak_as_bed(peaks_df,f_save, name='index',score_col = 'Peak Score',is_sort = True):\n",
    "    ## Assumes there is Chr,Start,End,Name is index\n",
    "    ## Peak file is 1-index, bed files are 0, so need to shift everything by one\n",
    "    \n",
    "    if name == 'index':\n",
    "        peaks_df.loc[:,'Name'] = peaks_df.index\n",
    "    else:\n",
    "        peaks_df.loc[:,'Name'] = peaks_df[name]\n",
    "    \n",
    "    if score_col == '': #If no scoring col, set to 0\n",
    "        peaks_df.loc[:,'Score'] = 0\n",
    "        score_col = 'Score'\n",
    "        \n",
    "    peaks_df.loc[:,'Start'] = peaks_df['Start'] - 1\n",
    "    peaks_df.loc[:,'End'] = peaks_df['End'] - 1\n",
    "    \n",
    "    peaks_df = peaks_df[['Chr','Start','End','Name',score_col,'Strand']]\n",
    "    \n",
    "    if is_sort:\n",
    "        peaks_df.sort_values(['Chr','Start'],inplace=True)\n",
    "        \n",
    "    if not f_save.endswith('.bed'):\n",
    "        f_save = f_save + '.bed'\n",
    "    peaks_df.to_csv(f_save ,sep='\\t',index=False,header=None)\n",
    "    return\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## a) Separate confident promoters and confident ncRNA \n",
    "Define by distance to RefSeq being 0 and ncRNA far >40kb from a gene and in at least two tissues\n",
    "\n",
    "### c) Predicted promoters vs RefSeq promoters\n",
    "\n",
    "### d) Promoters organized based on distance to refseq (so heatmap is not sorted by heat but by distance)\n",
    "\n",
    "### e) Only look at peaks specific for each tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:42.029291Z",
     "start_time": "2018-11-26T20:25:42.001522Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def filter_only_with_divergent(df,divergent_threshold = 500 ):\n",
    "    inds_to_keep = []\n",
    "    for ind,val in df[df['isSameStrand']].iterrows():\n",
    "        curr = df[(val['Chr'] == df['Chr']) & (df['isSameStrand'] == False)]\n",
    "        if val['Strand'] == '-':\n",
    "            curr = curr[(val['End'] - curr['Start'] < divergent_threshold) & \n",
    "                       (val['End'] - curr['Start'] > 0)]\n",
    "        else: \n",
    "            curr = curr[(val['Start'] - curr['End'] < divergent_threshold) &\n",
    "                       (val['Start'] - curr['End'] > 0 ) ]\n",
    "\n",
    "        ## If there are divergent peaks, add this peak\n",
    "        if len(curr) > 0: \n",
    "            inds_to_keep.append(ind)\n",
    "    \n",
    "    df = df.loc[inds_to_keep]\n",
    "    return df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:39:46.132121Z",
     "start_time": "2018-11-26T20:39:46.039082Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def tissue_specific_promoter(t,\n",
    "                             anno_peaks,\n",
    "                             peaks_tissue_merged,\n",
    "                             txn_df,tss,\n",
    "                             peak_count_thresh = 1, \n",
    "                             prom_bin=[-1000, 1000],\n",
    "                             enhancer_d=40000,\n",
    "                             divergent_thr=500,\n",
    "                             outfolder='Results/peak_groups/'):\n",
    "\n",
    "    ''' Function to create bed files relevant for a specific tissue\n",
    "        Will save:\n",
    "        a) All peaks that were found in tissue\n",
    "        b) Peaks within prom_bin distance from reference & has divergent peak nearby (divergent_thr distance)\n",
    "        c) Same as b, but no divergent restriction\n",
    "        d) Same as c, but the peak center is now the annnotated start site instead of experimental\n",
    "        e) ncRNA, possibly eRNA that are greater than enhancer_d from any annotated start sites.\n",
    "        f) no_peaks: genes with no peaks nearby, even divergent peaks'''\n",
    "    \n",
    "    \n",
    "    \n",
    "    t_peaks = peaks_tissue_merged[peaks_tissue_merged.loc[:, t] > peak_count_thresh].index #Peaks in that tissue\n",
    "    t_peaks = anno_peaks.loc[t_peaks]\n",
    "    t_peaks = t_peaks[(~(t_peaks['Distance to TSS'] == np.infty))] ## Remove peaks that are on a scaffold with no gene\n",
    "\n",
    "    ### Absence of peaks for that tissue.\n",
    "    no_t_peaks = txn_df[txn_df['samples'].apply(lambda x: np.array(map(lambda y: t not in y,x)).any())]\n",
    "    no_t_peaks = tss[tss['transcript_id'].apply(lambda x: x in no_t_peaks.index)]\n",
    "    #no_t_peaks = peaks_tissue_merged[peaks_tissue_merged.loc[:, t] <= peak_count_thresh].index #Peaks not in the tissue\n",
    "    #no_t_peaks = anno_peaks.loc[no_t_peaks]\n",
    "    \n",
    "\n",
    "\n",
    "    t_prom = t_peaks[(t_peaks['Distance to TSS'] > prom_bin[0])\n",
    "                     & (t_peaks['Distance to TSS'] < prom_bin[1])]\n",
    "    t_prom_isSame = t_prom[t_prom['isSameStrand']]\n",
    "    \n",
    "    \n",
    "    t_enh = t_peaks[np.abs(t_peaks['Distance to TSS']) > enhancer_d]\n",
    "    t_prom_div = filter_only_with_divergent(\n",
    "        t_prom, divergent_threshold=divergent_thr)\n",
    "    t_enh_div = filter_only_with_divergent(\n",
    "        t_enh, divergent_threshold=divergent_thr)\n",
    "\n",
    "    t_prom_div = t_prom_div.sort_values(['Distance to TSS'])\n",
    "    \n",
    "    #Look at RefSeq vs experimental TSS\n",
    "    tss_withPeaks = t_prom_isSame['Nearest TSS'].unique()\n",
    "    refSeq = tss[tss['transcript_id'].apply(lambda x: x in tss_withPeaks)] \n",
    "    # Dont want the whole gene length, just the start of the peak \n",
    "    refSeq['Start'] = refSeq['actual_start'] #actual_start takes care of +/- strand\n",
    "    refSeq['End'] = refSeq['actual_start']\n",
    "    \n",
    "    ## Saving bed files\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    if not os.path.exists(out_dir):\n",
    "        os.mkdir(out_dir)\n",
    "\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_peaks,\n",
    "        os.path.join(out_dir, 'all_peaks'),\n",
    "        name='index',\n",
    "        score_col='Stat')    \n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_prom_div,\n",
    "        os.path.join(out_dir, 'confident_promoters'),\n",
    "        name='index',\n",
    "        score_col='Stat')\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_prom_isSame,\n",
    "        os.path.join(out_dir, 'confident_promoters_noDiv'),\n",
    "        name='index',\n",
    "        score_col='Stat')  \n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        refSeq,\n",
    "        os.path.join(out_dir, 'confident_promoters_noDiv_PICR'),\n",
    "        name='index',\n",
    "        score_col='cs')\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_enh_div,\n",
    "        os.path.join(out_dir, 'confident_enhancers'),\n",
    "        name='index',\n",
    "        score_col='Stat')\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        no_t_peaks,\n",
    "        os.path.join(out_dir, 'no_peaks'),\n",
    "        name='index',\n",
    "        score_col='cs')\n",
    "    \n",
    "    \n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:42.128263Z",
     "start_time": "2018-11-26T20:25:42.108731Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tissues_with_atac = ['CHO','Brain','Liver','Lung','Kidney','Spleen','BMDMwt']#,'BMDM1hKLA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:53:55.342260Z",
     "start_time": "2018-11-26T20:39:50.297480Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:51: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:52: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/indexing.py:362: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/indexing.py:543: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self.obj[item] = s\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Brain\n",
      "Liver\n",
      "Lung\n",
      "Kidney\n",
      "Spleen\n",
      "BMDMwt\n",
      "CPU times: user 14min 5s, sys: 4.23 s, total: 14min 9s\n",
      "Wall time: 14min 5s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    tissue_specific_promoter(\n",
    "        t=t,\n",
    "        anno_peaks=anno_peaks,\n",
    "        peaks_tissue_merged=peaks_tissue_merged,\n",
    "        txn_df=txn_df,\n",
    "        tss=tss)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TSS peaks near ATAC peaks. \n",
    "Using Bedtools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sort ATAC bed files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:28:35.020014Z",
     "start_time": "2018-11-26T23:28:27.716471Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/CHO/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.gz > Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort\n",
      "Brain\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Brain/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "Liver\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Liver/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "Lung\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Lung/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "Kidney\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Kidney/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "Spleen\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Spleen/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "BMDMwt\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/BMDMwt/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.gz > Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort\n"
     ]
    }
   ],
   "source": [
    "## Sort ATAC bed files\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    atac_f = glob.glob(os.path.join(data_folder, t, 'ATAC/out/peak/macs2/overlap/optimal_set/*narrowPeak.gz'))[0]\n",
    "    atac_f_out = atac_f.replace('.gz','') + '.sort'\n",
    "    atac_f_out = os.path.join(\"Results/ATAC_results/\",t,os.path.basename(atac_f).replace('.gz','.sort'))\n",
    "    if not os.path.exists(os.path.join(\"Results/ATAC_results/\",t)):\n",
    "        os.mkdir(os.path.join(\"Results/ATAC_results/\",t))\n",
    "    \n",
    "    cmd = 'bedtools sort -i %s > %s' % (atac_f,atac_f_out)   \n",
    "    print(cmd)\n",
    "    os.system(cmd)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get peaks close to TSS within distance of 500 bp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:20:11.904803Z",
     "start_time": "2018-11-26T23:20:11.843946Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def peaks_closest(f_a, f_b, out_f, dist_thresh):\n",
    "    '''Function that calls bedtools closest to find the closest peak of two bed files, and awk to filter those within distance dist_thresh'''\n",
    "\n",
    "    cmd = \"bedtools closest -d -t first -a %s -b %s | awk '{if ($NF < %d) print }' > %s\" % (\n",
    "       f_a, f_b, dist_thresh,out_f) #t is to only report one of the atac peaks\n",
    "    \n",
    "#     cmd = \"bedtools closest -d -t first -a %s -b %s  > %s\" % (\n",
    "#         f_a, f_b,out_f)\n",
    "    \n",
    "    print(cmd)\n",
    "    os.system(cmd)\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:29:22.829551Z",
     "start_time": "2018-11-26T23:29:07.791014Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/all_peaks.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/CHO/all_peaks.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_promoters.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/CHO/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_enhancers.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/CHO/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_promoters_noDiv.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/CHO/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/no_peaks.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/CHO/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Brain\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/all_peaks.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Brain/all_peaks.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_promoters.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Brain/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_enhancers.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Brain/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_promoters_noDiv.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Brain/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/no_peaks.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Brain/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Liver\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/all_peaks.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Liver/all_peaks.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_promoters.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Liver/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_enhancers.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Liver/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_promoters_noDiv.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Liver/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/no_peaks.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Liver/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Lung\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/all_peaks.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Lung/all_peaks.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_promoters.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Lung/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_enhancers.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Lung/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_promoters_noDiv.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Lung/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/no_peaks.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Lung/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Kidney\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/all_peaks.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Kidney/all_peaks.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_promoters.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Kidney/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_enhancers.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Kidney/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_promoters_noDiv.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Kidney/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/no_peaks.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Kidney/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Spleen\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/all_peaks.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Spleen/all_peaks.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_promoters.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Spleen/confident_promoters.bed_atac_overlap_500bp.bed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_enhancers.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Spleen/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_promoters_noDiv.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Spleen/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/no_peaks.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/Spleen/no_peaks.bed_atac_overlap_500bp.bed\n",
      "BMDMwt\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/all_peaks.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/BMDMwt/all_peaks.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_promoters.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/BMDMwt/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_enhancers.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/BMDMwt/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/no_peaks.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 500) print }' > Results/peak_groups/BMDMwt/no_peaks.bed_atac_overlap_500bp.bed\n"
     ]
    }
   ],
   "source": [
    "# Intersect ATAC and TSS\n",
    "dist_thresh = 500 #500 basepairs left or right\n",
    "outfolder = 'Results/peak_groups/'\n",
    "f_names = [\n",
    "    'all_peaks','confident_promoters', 'confident_enhancers', 'confident_promoters_noDiv',\n",
    "    'confident_promoters_noDiv_PICR', 'no_peaks'\n",
    "]\n",
    "\n",
    "\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    ## Outdirectory to save in\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    \n",
    "    #ATAC peak file\n",
    "    atac_out = glob.glob(os.path.join(\"Results/ATAC_results\", t , '*sort'))[0]\n",
    "    \n",
    "    ## The different TSS files\n",
    "    for f in f_names:\n",
    "        f_in = os.path.join(out_dir, f) + '.bed'\n",
    "        out_f = '%s_atac_overlap_%dbp.bed' % (f_in,dist_thresh)\n",
    "        peaks_closest(f_in,atac_out,out_f,dist_thresh=dist_thresh)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:33:32.610050Z",
     "start_time": "2018-11-26T23:33:32.384247Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all_peaks.bed\r\n",
      "all_peaks.bed_atac_overlap_500bp.bed\r\n",
      "confident_enhancers.bed\r\n",
      "confident_enhancers.bed_atac_overlap_500bp.bed\r\n",
      "confident_promoters.bed\r\n",
      "confident_promoters.bed_atac_overlap_500bp.bed\r\n",
      "confident_promoters_noDiv.bed\r\n",
      "confident_promoters_noDiv.bed_atac_overlap_500bp.bed\r\n",
      "confident_promoters_noDiv_PICR.bed\r\n",
      "confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\r\n",
      "no_peaks.bed\r\n",
      "no_peaks.bed_atac_overlap_500bp.bed\r\n"
     ]
    }
   ],
   "source": [
    "ls Results/peak_groups/Brain/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:34:28.450371Z",
     "start_time": "2018-11-26T23:34:14.062872Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/all_peaks.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_promoters.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_enhancers.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_promoters_noDiv.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/no_peaks.bed -b Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Brain\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/all_peaks.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_promoters.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_enhancers.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_promoters_noDiv.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/no_peaks.bed -b Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Liver\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/all_peaks.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_promoters.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_enhancers.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_promoters_noDiv.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/no_peaks.bed -b Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Lung\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/all_peaks.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_promoters.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_enhancers.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_promoters_noDiv.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/no_peaks.bed -b Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Kidney\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/all_peaks.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_promoters.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_enhancers.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_promoters_noDiv.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/no_peaks.bed -b Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Spleen\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/all_peaks.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_promoters.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/confident_promoters.bed_atac_overlap_100bp.bed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_enhancers.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_promoters_noDiv.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/no_peaks.bed -b Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/no_peaks.bed_atac_overlap_100bp.bed\n",
      "BMDMwt\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/all_peaks.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_promoters.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_enhancers.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/no_peaks.bed -b Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/no_peaks.bed_atac_overlap_100bp.bed\n"
     ]
    }
   ],
   "source": [
    "# Intersect ATAC and TSS\n",
    "dist_thresh = 100 #500 basepairs left or right\n",
    "outfolder = 'Results/peak_groups/'\n",
    "f_names = [\n",
    "    'all_peaks','confident_promoters', 'confident_enhancers', 'confident_promoters_noDiv',\n",
    "    'confident_promoters_noDiv_PICR', 'no_peaks'\n",
    "]\n",
    "\n",
    "\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    ## Outdirectory to save in\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    \n",
    "    #ATAC peak file\n",
    "    atac_out = glob.glob(os.path.join(\"Results/ATAC_results\", t , '*sort'))[0]\n",
    "    \n",
    "    ## The different TSS files\n",
    "    for f in f_names:\n",
    "        f_in = os.path.join(out_dir, f) + '.bed'\n",
    "        out_f = '%s_atac_overlap_%dbp.bed' % (f_in,dist_thresh)\n",
    "        peaks_closest(f_in,atac_out,out_f,dist_thresh=dist_thresh)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Use wc -l to determine the number of peaks that are near ATAC, compared to total\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:37:32.071720Z",
     "start_time": "2018-11-26T23:37:32.014335Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from subprocess import check_output\n",
    "\n",
    "def wc(filename):\n",
    "    return int(check_output([\"wc\", \"-l\", filename]).split()[0])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:41:49.785098Z",
     "start_time": "2018-11-26T23:41:42.380304Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n",
      "Results/peak_groups/CHO/all_peaks.bed\n",
      "Results/peak_groups/CHO/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/all_peaks.bed\n",
      "Results/peak_groups/CHO/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters.bed\n",
      "Results/peak_groups/CHO/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters.bed\n",
      "Results/peak_groups/CHO/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/confident_enhancers.bed\n",
      "Results/peak_groups/CHO/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/confident_enhancers.bed\n",
      "Results/peak_groups/CHO/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/no_peaks.bed\n",
      "Results/peak_groups/CHO/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/no_peaks.bed\n",
      "Results/peak_groups/CHO/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Brain\n",
      "Results/peak_groups/Brain/all_peaks.bed\n",
      "Results/peak_groups/Brain/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/all_peaks.bed\n",
      "Results/peak_groups/Brain/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters.bed\n",
      "Results/peak_groups/Brain/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters.bed\n",
      "Results/peak_groups/Brain/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/confident_enhancers.bed\n",
      "Results/peak_groups/Brain/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/confident_enhancers.bed\n",
      "Results/peak_groups/Brain/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/no_peaks.bed\n",
      "Results/peak_groups/Brain/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/no_peaks.bed\n",
      "Results/peak_groups/Brain/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Liver\n",
      "Results/peak_groups/Liver/all_peaks.bed\n",
      "Results/peak_groups/Liver/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/all_peaks.bed\n",
      "Results/peak_groups/Liver/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters.bed\n",
      "Results/peak_groups/Liver/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters.bed\n",
      "Results/peak_groups/Liver/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/confident_enhancers.bed\n",
      "Results/peak_groups/Liver/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/confident_enhancers.bed\n",
      "Results/peak_groups/Liver/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/no_peaks.bed\n",
      "Results/peak_groups/Liver/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/no_peaks.bed\n",
      "Results/peak_groups/Liver/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Lung\n",
      "Results/peak_groups/Lung/all_peaks.bed\n",
      "Results/peak_groups/Lung/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/all_peaks.bed\n",
      "Results/peak_groups/Lung/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters.bed\n",
      "Results/peak_groups/Lung/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters.bed\n",
      "Results/peak_groups/Lung/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/confident_enhancers.bed\n",
      "Results/peak_groups/Lung/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/confident_enhancers.bed\n",
      "Results/peak_groups/Lung/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/no_peaks.bed\n",
      "Results/peak_groups/Lung/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/no_peaks.bed\n",
      "Results/peak_groups/Lung/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Kidney\n",
      "Results/peak_groups/Kidney/all_peaks.bed\n",
      "Results/peak_groups/Kidney/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/all_peaks.bed\n",
      "Results/peak_groups/Kidney/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters.bed\n",
      "Results/peak_groups/Kidney/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters.bed\n",
      "Results/peak_groups/Kidney/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/confident_enhancers.bed\n",
      "Results/peak_groups/Kidney/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/confident_enhancers.bed\n",
      "Results/peak_groups/Kidney/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/no_peaks.bed\n",
      "Results/peak_groups/Kidney/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/no_peaks.bed\n",
      "Results/peak_groups/Kidney/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Spleen\n",
      "Results/peak_groups/Spleen/all_peaks.bed\n",
      "Results/peak_groups/Spleen/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/all_peaks.bed\n",
      "Results/peak_groups/Spleen/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters.bed\n",
      "Results/peak_groups/Spleen/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters.bed\n",
      "Results/peak_groups/Spleen/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Spleen/confident_enhancers.bed\n",
      "Results/peak_groups/Spleen/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/confident_enhancers.bed\n",
      "Results/peak_groups/Spleen/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results/peak_groups/Spleen/no_peaks.bed\n",
      "Results/peak_groups/Spleen/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/no_peaks.bed\n",
      "Results/peak_groups/Spleen/no_peaks.bed_atac_overlap_500bp.bed\n",
      "BMDMwt\n",
      "Results/peak_groups/BMDMwt/all_peaks.bed\n",
      "Results/peak_groups/BMDMwt/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/all_peaks.bed\n",
      "Results/peak_groups/BMDMwt/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_enhancers.bed\n",
      "Results/peak_groups/BMDMwt/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_enhancers.bed\n",
      "Results/peak_groups/BMDMwt/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/no_peaks.bed\n",
      "Results/peak_groups/BMDMwt/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/no_peaks.bed\n",
      "Results/peak_groups/BMDMwt/no_peaks.bed_atac_overlap_500bp.bed\n"
     ]
    }
   ],
   "source": [
    "dist_thresholds = [100,500]\n",
    "atac_tss_df = pd.DataFrame()\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    ## The different TSS files\n",
    "    for f in f_names:\n",
    "        f_in = os.path.join(out_dir, f) + '.bed'\n",
    "        \n",
    "        for d in dist_thresholds:\n",
    "            out_f = '%s_atac_overlap_%dbp.bed' % (f_in,d)\n",
    "            print(f_in)\n",
    "            pre_filt = wc(f_in)\n",
    "            print(out_f)\n",
    "            post_filt = wc(out_f)\n",
    "            name = os.path.basename(f_in)\n",
    "            frac = 1.0*post_filt/pre_filt\n",
    "            atac_tss_df.at[t,name] = pre_filt\n",
    "            name_atac = '_with_ATAC_%d' % d\n",
    "            ## Add to dataframe\n",
    "            \n",
    "            atac_tss_df.at[t,name + name_atac] = post_filt\n",
    "            atac_tss_df.at[t,name + name_atac + '_fraction'] = frac\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:41:49.867954Z",
     "start_time": "2018-11-26T23:41:49.787389Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all_peaks.bed</th>\n",
       "      <th>all_peaks.bed_with_ATAC_100</th>\n",
       "      <th>all_peaks.bed_with_ATAC_100_fraction</th>\n",
       "      <th>all_peaks.bed_with_ATAC_500</th>\n",
       "      <th>all_peaks.bed_with_ATAC_500_fraction</th>\n",
       "      <th>confident_promoters.bed</th>\n",
       "      <th>confident_promoters.bed_with_ATAC_100</th>\n",
       "      <th>confident_promoters.bed_with_ATAC_100_fraction</th>\n",
       "      <th>confident_promoters.bed_with_ATAC_500</th>\n",
       "      <th>confident_promoters.bed_with_ATAC_500_fraction</th>\n",
       "      <th>...</th>\n",
       "      <th>confident_promoters_noDiv_PICR.bed</th>\n",
       "      <th>confident_promoters_noDiv_PICR.bed_with_ATAC_100</th>\n",
       "      <th>confident_promoters_noDiv_PICR.bed_with_ATAC_100_fraction</th>\n",
       "      <th>confident_promoters_noDiv_PICR.bed_with_ATAC_500</th>\n",
       "      <th>confident_promoters_noDiv_PICR.bed_with_ATAC_500_fraction</th>\n",
       "      <th>no_peaks.bed</th>\n",
       "      <th>no_peaks.bed_with_ATAC_100</th>\n",
       "      <th>no_peaks.bed_with_ATAC_100_fraction</th>\n",
       "      <th>no_peaks.bed_with_ATAC_500</th>\n",
       "      <th>no_peaks.bed_with_ATAC_500_fraction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CHO</th>\n",
       "      <td>54931.0</td>\n",
       "      <td>46443.0</td>\n",
       "      <td>0.845479</td>\n",
       "      <td>49900.0</td>\n",
       "      <td>0.908412</td>\n",
       "      <td>5271.0</td>\n",
       "      <td>4878.0</td>\n",
       "      <td>0.925441</td>\n",
       "      <td>5124.0</td>\n",
       "      <td>0.972112</td>\n",
       "      <td>...</td>\n",
       "      <td>11117.0</td>\n",
       "      <td>9096.0</td>\n",
       "      <td>0.818206</td>\n",
       "      <td>10300.0</td>\n",
       "      <td>0.926509</td>\n",
       "      <td>23724.0</td>\n",
       "      <td>19227.0</td>\n",
       "      <td>0.810445</td>\n",
       "      <td>19531.0</td>\n",
       "      <td>0.823259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain</th>\n",
       "      <td>37316.0</td>\n",
       "      <td>29313.0</td>\n",
       "      <td>0.785534</td>\n",
       "      <td>31252.0</td>\n",
       "      <td>0.837496</td>\n",
       "      <td>4080.0</td>\n",
       "      <td>3916.0</td>\n",
       "      <td>0.959804</td>\n",
       "      <td>4013.0</td>\n",
       "      <td>0.983578</td>\n",
       "      <td>...</td>\n",
       "      <td>13374.0</td>\n",
       "      <td>10650.0</td>\n",
       "      <td>0.796321</td>\n",
       "      <td>11876.0</td>\n",
       "      <td>0.887992</td>\n",
       "      <td>23792.0</td>\n",
       "      <td>20947.0</td>\n",
       "      <td>0.880422</td>\n",
       "      <td>21249.0</td>\n",
       "      <td>0.893115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Liver</th>\n",
       "      <td>56905.0</td>\n",
       "      <td>8847.0</td>\n",
       "      <td>0.155470</td>\n",
       "      <td>10891.0</td>\n",
       "      <td>0.191389</td>\n",
       "      <td>3426.0</td>\n",
       "      <td>1171.0</td>\n",
       "      <td>0.341798</td>\n",
       "      <td>1371.0</td>\n",
       "      <td>0.400175</td>\n",
       "      <td>...</td>\n",
       "      <td>11272.0</td>\n",
       "      <td>3212.0</td>\n",
       "      <td>0.284954</td>\n",
       "      <td>3993.0</td>\n",
       "      <td>0.354241</td>\n",
       "      <td>24409.0</td>\n",
       "      <td>10256.0</td>\n",
       "      <td>0.420173</td>\n",
       "      <td>10732.0</td>\n",
       "      <td>0.439674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lung</th>\n",
       "      <td>45148.0</td>\n",
       "      <td>25302.0</td>\n",
       "      <td>0.560423</td>\n",
       "      <td>28473.0</td>\n",
       "      <td>0.630659</td>\n",
       "      <td>4036.0</td>\n",
       "      <td>3512.0</td>\n",
       "      <td>0.870168</td>\n",
       "      <td>3738.0</td>\n",
       "      <td>0.926165</td>\n",
       "      <td>...</td>\n",
       "      <td>12684.0</td>\n",
       "      <td>8922.0</td>\n",
       "      <td>0.703406</td>\n",
       "      <td>10120.0</td>\n",
       "      <td>0.797856</td>\n",
       "      <td>24324.0</td>\n",
       "      <td>18765.0</td>\n",
       "      <td>0.771460</td>\n",
       "      <td>19223.0</td>\n",
       "      <td>0.790289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kidney</th>\n",
       "      <td>50389.0</td>\n",
       "      <td>35906.0</td>\n",
       "      <td>0.712576</td>\n",
       "      <td>38668.0</td>\n",
       "      <td>0.767390</td>\n",
       "      <td>3813.0</td>\n",
       "      <td>3604.0</td>\n",
       "      <td>0.945188</td>\n",
       "      <td>3721.0</td>\n",
       "      <td>0.975872</td>\n",
       "      <td>...</td>\n",
       "      <td>12050.0</td>\n",
       "      <td>9710.0</td>\n",
       "      <td>0.805809</td>\n",
       "      <td>10801.0</td>\n",
       "      <td>0.896349</td>\n",
       "      <td>24322.0</td>\n",
       "      <td>21619.0</td>\n",
       "      <td>0.888866</td>\n",
       "      <td>21906.0</td>\n",
       "      <td>0.900666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Spleen</th>\n",
       "      <td>38631.0</td>\n",
       "      <td>21151.0</td>\n",
       "      <td>0.547514</td>\n",
       "      <td>23424.0</td>\n",
       "      <td>0.606352</td>\n",
       "      <td>3117.0</td>\n",
       "      <td>2655.0</td>\n",
       "      <td>0.851781</td>\n",
       "      <td>2792.0</td>\n",
       "      <td>0.895733</td>\n",
       "      <td>...</td>\n",
       "      <td>11313.0</td>\n",
       "      <td>7732.0</td>\n",
       "      <td>0.683462</td>\n",
       "      <td>8728.0</td>\n",
       "      <td>0.771502</td>\n",
       "      <td>24586.0</td>\n",
       "      <td>18175.0</td>\n",
       "      <td>0.739242</td>\n",
       "      <td>18624.0</td>\n",
       "      <td>0.757504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BMDMwt</th>\n",
       "      <td>85558.0</td>\n",
       "      <td>38630.0</td>\n",
       "      <td>0.451507</td>\n",
       "      <td>44369.0</td>\n",
       "      <td>0.518584</td>\n",
       "      <td>3138.0</td>\n",
       "      <td>2998.0</td>\n",
       "      <td>0.955386</td>\n",
       "      <td>3108.0</td>\n",
       "      <td>0.990440</td>\n",
       "      <td>...</td>\n",
       "      <td>10751.0</td>\n",
       "      <td>8516.0</td>\n",
       "      <td>0.792112</td>\n",
       "      <td>9517.0</td>\n",
       "      <td>0.885220</td>\n",
       "      <td>24294.0</td>\n",
       "      <td>21493.0</td>\n",
       "      <td>0.884704</td>\n",
       "      <td>21747.0</td>\n",
       "      <td>0.895159</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7 rows Ã— 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        all_peaks.bed  all_peaks.bed_with_ATAC_100  \\\n",
       "CHO           54931.0                      46443.0   \n",
       "Brain         37316.0                      29313.0   \n",
       "Liver         56905.0                       8847.0   \n",
       "Lung          45148.0                      25302.0   \n",
       "Kidney        50389.0                      35906.0   \n",
       "Spleen        38631.0                      21151.0   \n",
       "BMDMwt        85558.0                      38630.0   \n",
       "\n",
       "        all_peaks.bed_with_ATAC_100_fraction  all_peaks.bed_with_ATAC_500  \\\n",
       "CHO                                 0.845479                      49900.0   \n",
       "Brain                               0.785534                      31252.0   \n",
       "Liver                               0.155470                      10891.0   \n",
       "Lung                                0.560423                      28473.0   \n",
       "Kidney                              0.712576                      38668.0   \n",
       "Spleen                              0.547514                      23424.0   \n",
       "BMDMwt                              0.451507                      44369.0   \n",
       "\n",
       "        all_peaks.bed_with_ATAC_500_fraction  confident_promoters.bed  \\\n",
       "CHO                                 0.908412                   5271.0   \n",
       "Brain                               0.837496                   4080.0   \n",
       "Liver                               0.191389                   3426.0   \n",
       "Lung                                0.630659                   4036.0   \n",
       "Kidney                              0.767390                   3813.0   \n",
       "Spleen                              0.606352                   3117.0   \n",
       "BMDMwt                              0.518584                   3138.0   \n",
       "\n",
       "        confident_promoters.bed_with_ATAC_100  \\\n",
       "CHO                                    4878.0   \n",
       "Brain                                  3916.0   \n",
       "Liver                                  1171.0   \n",
       "Lung                                   3512.0   \n",
       "Kidney                                 3604.0   \n",
       "Spleen                                 2655.0   \n",
       "BMDMwt                                 2998.0   \n",
       "\n",
       "        confident_promoters.bed_with_ATAC_100_fraction  \\\n",
       "CHO                                           0.925441   \n",
       "Brain                                         0.959804   \n",
       "Liver                                         0.341798   \n",
       "Lung                                          0.870168   \n",
       "Kidney                                        0.945188   \n",
       "Spleen                                        0.851781   \n",
       "BMDMwt                                        0.955386   \n",
       "\n",
       "        confident_promoters.bed_with_ATAC_500  \\\n",
       "CHO                                    5124.0   \n",
       "Brain                                  4013.0   \n",
       "Liver                                  1371.0   \n",
       "Lung                                   3738.0   \n",
       "Kidney                                 3721.0   \n",
       "Spleen                                 2792.0   \n",
       "BMDMwt                                 3108.0   \n",
       "\n",
       "        confident_promoters.bed_with_ATAC_500_fraction  \\\n",
       "CHO                                           0.972112   \n",
       "Brain                                         0.983578   \n",
       "Liver                                         0.400175   \n",
       "Lung                                          0.926165   \n",
       "Kidney                                        0.975872   \n",
       "Spleen                                        0.895733   \n",
       "BMDMwt                                        0.990440   \n",
       "\n",
       "                       ...                   \\\n",
       "CHO                    ...                    \n",
       "Brain                  ...                    \n",
       "Liver                  ...                    \n",
       "Lung                   ...                    \n",
       "Kidney                 ...                    \n",
       "Spleen                 ...                    \n",
       "BMDMwt                 ...                    \n",
       "\n",
       "        confident_promoters_noDiv_PICR.bed  \\\n",
       "CHO                                11117.0   \n",
       "Brain                              13374.0   \n",
       "Liver                              11272.0   \n",
       "Lung                               12684.0   \n",
       "Kidney                             12050.0   \n",
       "Spleen                             11313.0   \n",
       "BMDMwt                             10751.0   \n",
       "\n",
       "        confident_promoters_noDiv_PICR.bed_with_ATAC_100  \\\n",
       "CHO                                               9096.0   \n",
       "Brain                                            10650.0   \n",
       "Liver                                             3212.0   \n",
       "Lung                                              8922.0   \n",
       "Kidney                                            9710.0   \n",
       "Spleen                                            7732.0   \n",
       "BMDMwt                                            8516.0   \n",
       "\n",
       "        confident_promoters_noDiv_PICR.bed_with_ATAC_100_fraction  \\\n",
       "CHO                                              0.818206           \n",
       "Brain                                            0.796321           \n",
       "Liver                                            0.284954           \n",
       "Lung                                             0.703406           \n",
       "Kidney                                           0.805809           \n",
       "Spleen                                           0.683462           \n",
       "BMDMwt                                           0.792112           \n",
       "\n",
       "        confident_promoters_noDiv_PICR.bed_with_ATAC_500  \\\n",
       "CHO                                              10300.0   \n",
       "Brain                                            11876.0   \n",
       "Liver                                             3993.0   \n",
       "Lung                                             10120.0   \n",
       "Kidney                                           10801.0   \n",
       "Spleen                                            8728.0   \n",
       "BMDMwt                                            9517.0   \n",
       "\n",
       "        confident_promoters_noDiv_PICR.bed_with_ATAC_500_fraction  \\\n",
       "CHO                                              0.926509           \n",
       "Brain                                            0.887992           \n",
       "Liver                                            0.354241           \n",
       "Lung                                             0.797856           \n",
       "Kidney                                           0.896349           \n",
       "Spleen                                           0.771502           \n",
       "BMDMwt                                           0.885220           \n",
       "\n",
       "        no_peaks.bed  no_peaks.bed_with_ATAC_100  \\\n",
       "CHO          23724.0                     19227.0   \n",
       "Brain        23792.0                     20947.0   \n",
       "Liver        24409.0                     10256.0   \n",
       "Lung         24324.0                     18765.0   \n",
       "Kidney       24322.0                     21619.0   \n",
       "Spleen       24586.0                     18175.0   \n",
       "BMDMwt       24294.0                     21493.0   \n",
       "\n",
       "        no_peaks.bed_with_ATAC_100_fraction  no_peaks.bed_with_ATAC_500  \\\n",
       "CHO                                0.810445                     19531.0   \n",
       "Brain                              0.880422                     21249.0   \n",
       "Liver                              0.420173                     10732.0   \n",
       "Lung                               0.771460                     19223.0   \n",
       "Kidney                             0.888866                     21906.0   \n",
       "Spleen                             0.739242                     18624.0   \n",
       "BMDMwt                             0.884704                     21747.0   \n",
       "\n",
       "        no_peaks.bed_with_ATAC_500_fraction  \n",
       "CHO                                0.823259  \n",
       "Brain                              0.893115  \n",
       "Liver                              0.439674  \n",
       "Lung                               0.790289  \n",
       "Kidney                             0.900666  \n",
       "Spleen                             0.757504  \n",
       "BMDMwt                             0.895159  \n",
       "\n",
       "[7 rows x 30 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atac_tss_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:43:10.047176Z",
     "start_time": "2018-11-26T23:43:09.915252Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all peaks</th>\n",
       "      <th>all peaks with ATAC peak within 100</th>\n",
       "      <th>all peaks with ATAC peak within 100 fraction</th>\n",
       "      <th>all peaks with ATAC peak within 500</th>\n",
       "      <th>all peaks with ATAC peak within 500 fraction</th>\n",
       "      <th>confident promoters</th>\n",
       "      <th>confident promoters with ATAC peak within 100</th>\n",
       "      <th>confident promoters with ATAC peak within 100 fraction</th>\n",
       "      <th>confident promoters with ATAC peak within 500</th>\n",
       "      <th>confident promoters with ATAC peak within 500 fraction</th>\n",
       "      <th>...</th>\n",
       "      <th>confident promoters noDiv PICR</th>\n",
       "      <th>confident promoters noDiv PICR with ATAC peak within 100</th>\n",
       "      <th>confident promoters noDiv PICR with ATAC peak within 100 fraction</th>\n",
       "      <th>confident promoters noDiv PICR with ATAC peak within 500</th>\n",
       "      <th>confident promoters noDiv PICR with ATAC peak within 500 fraction</th>\n",
       "      <th>no peaks</th>\n",
       "      <th>no peaks with ATAC peak within 100</th>\n",
       "      <th>no peaks with ATAC peak within 100 fraction</th>\n",
       "      <th>no peaks with ATAC peak within 500</th>\n",
       "      <th>no peaks with ATAC peak within 500 fraction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CHO</th>\n",
       "      <td>54931.0</td>\n",
       "      <td>46443.0</td>\n",
       "      <td>0.845479</td>\n",
       "      <td>49900.0</td>\n",
       "      <td>0.908412</td>\n",
       "      <td>5271.0</td>\n",
       "      <td>4878.0</td>\n",
       "      <td>0.925441</td>\n",
       "      <td>5124.0</td>\n",
       "      <td>0.972112</td>\n",
       "      <td>...</td>\n",
       "      <td>11117.0</td>\n",
       "      <td>9096.0</td>\n",
       "      <td>0.818206</td>\n",
       "      <td>10300.0</td>\n",
       "      <td>0.926509</td>\n",
       "      <td>23724.0</td>\n",
       "      <td>19227.0</td>\n",
       "      <td>0.810445</td>\n",
       "      <td>19531.0</td>\n",
       "      <td>0.823259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain</th>\n",
       "      <td>37316.0</td>\n",
       "      <td>29313.0</td>\n",
       "      <td>0.785534</td>\n",
       "      <td>31252.0</td>\n",
       "      <td>0.837496</td>\n",
       "      <td>4080.0</td>\n",
       "      <td>3916.0</td>\n",
       "      <td>0.959804</td>\n",
       "      <td>4013.0</td>\n",
       "      <td>0.983578</td>\n",
       "      <td>...</td>\n",
       "      <td>13374.0</td>\n",
       "      <td>10650.0</td>\n",
       "      <td>0.796321</td>\n",
       "      <td>11876.0</td>\n",
       "      <td>0.887992</td>\n",
       "      <td>23792.0</td>\n",
       "      <td>20947.0</td>\n",
       "      <td>0.880422</td>\n",
       "      <td>21249.0</td>\n",
       "      <td>0.893115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Liver</th>\n",
       "      <td>56905.0</td>\n",
       "      <td>8847.0</td>\n",
       "      <td>0.155470</td>\n",
       "      <td>10891.0</td>\n",
       "      <td>0.191389</td>\n",
       "      <td>3426.0</td>\n",
       "      <td>1171.0</td>\n",
       "      <td>0.341798</td>\n",
       "      <td>1371.0</td>\n",
       "      <td>0.400175</td>\n",
       "      <td>...</td>\n",
       "      <td>11272.0</td>\n",
       "      <td>3212.0</td>\n",
       "      <td>0.284954</td>\n",
       "      <td>3993.0</td>\n",
       "      <td>0.354241</td>\n",
       "      <td>24409.0</td>\n",
       "      <td>10256.0</td>\n",
       "      <td>0.420173</td>\n",
       "      <td>10732.0</td>\n",
       "      <td>0.439674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lung</th>\n",
       "      <td>45148.0</td>\n",
       "      <td>25302.0</td>\n",
       "      <td>0.560423</td>\n",
       "      <td>28473.0</td>\n",
       "      <td>0.630659</td>\n",
       "      <td>4036.0</td>\n",
       "      <td>3512.0</td>\n",
       "      <td>0.870168</td>\n",
       "      <td>3738.0</td>\n",
       "      <td>0.926165</td>\n",
       "      <td>...</td>\n",
       "      <td>12684.0</td>\n",
       "      <td>8922.0</td>\n",
       "      <td>0.703406</td>\n",
       "      <td>10120.0</td>\n",
       "      <td>0.797856</td>\n",
       "      <td>24324.0</td>\n",
       "      <td>18765.0</td>\n",
       "      <td>0.771460</td>\n",
       "      <td>19223.0</td>\n",
       "      <td>0.790289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kidney</th>\n",
       "      <td>50389.0</td>\n",
       "      <td>35906.0</td>\n",
       "      <td>0.712576</td>\n",
       "      <td>38668.0</td>\n",
       "      <td>0.767390</td>\n",
       "      <td>3813.0</td>\n",
       "      <td>3604.0</td>\n",
       "      <td>0.945188</td>\n",
       "      <td>3721.0</td>\n",
       "      <td>0.975872</td>\n",
       "      <td>...</td>\n",
       "      <td>12050.0</td>\n",
       "      <td>9710.0</td>\n",
       "      <td>0.805809</td>\n",
       "      <td>10801.0</td>\n",
       "      <td>0.896349</td>\n",
       "      <td>24322.0</td>\n",
       "      <td>21619.0</td>\n",
       "      <td>0.888866</td>\n",
       "      <td>21906.0</td>\n",
       "      <td>0.900666</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Spleen</th>\n",
       "      <td>38631.0</td>\n",
       "      <td>21151.0</td>\n",
       "      <td>0.547514</td>\n",
       "      <td>23424.0</td>\n",
       "      <td>0.606352</td>\n",
       "      <td>3117.0</td>\n",
       "      <td>2655.0</td>\n",
       "      <td>0.851781</td>\n",
       "      <td>2792.0</td>\n",
       "      <td>0.895733</td>\n",
       "      <td>...</td>\n",
       "      <td>11313.0</td>\n",
       "      <td>7732.0</td>\n",
       "      <td>0.683462</td>\n",
       "      <td>8728.0</td>\n",
       "      <td>0.771502</td>\n",
       "      <td>24586.0</td>\n",
       "      <td>18175.0</td>\n",
       "      <td>0.739242</td>\n",
       "      <td>18624.0</td>\n",
       "      <td>0.757504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BMDMwt</th>\n",
       "      <td>85558.0</td>\n",
       "      <td>38630.0</td>\n",
       "      <td>0.451507</td>\n",
       "      <td>44369.0</td>\n",
       "      <td>0.518584</td>\n",
       "      <td>3138.0</td>\n",
       "      <td>2998.0</td>\n",
       "      <td>0.955386</td>\n",
       "      <td>3108.0</td>\n",
       "      <td>0.990440</td>\n",
       "      <td>...</td>\n",
       "      <td>10751.0</td>\n",
       "      <td>8516.0</td>\n",
       "      <td>0.792112</td>\n",
       "      <td>9517.0</td>\n",
       "      <td>0.885220</td>\n",
       "      <td>24294.0</td>\n",
       "      <td>21493.0</td>\n",
       "      <td>0.884704</td>\n",
       "      <td>21747.0</td>\n",
       "      <td>0.895159</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7 rows Ã— 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        all peaks  all peaks with ATAC peak within 100  \\\n",
       "CHO       54931.0                              46443.0   \n",
       "Brain     37316.0                              29313.0   \n",
       "Liver     56905.0                               8847.0   \n",
       "Lung      45148.0                              25302.0   \n",
       "Kidney    50389.0                              35906.0   \n",
       "Spleen    38631.0                              21151.0   \n",
       "BMDMwt    85558.0                              38630.0   \n",
       "\n",
       "        all peaks with ATAC peak within 100 fraction  \\\n",
       "CHO                                         0.845479   \n",
       "Brain                                       0.785534   \n",
       "Liver                                       0.155470   \n",
       "Lung                                        0.560423   \n",
       "Kidney                                      0.712576   \n",
       "Spleen                                      0.547514   \n",
       "BMDMwt                                      0.451507   \n",
       "\n",
       "        all peaks with ATAC peak within 500  \\\n",
       "CHO                                 49900.0   \n",
       "Brain                               31252.0   \n",
       "Liver                               10891.0   \n",
       "Lung                                28473.0   \n",
       "Kidney                              38668.0   \n",
       "Spleen                              23424.0   \n",
       "BMDMwt                              44369.0   \n",
       "\n",
       "        all peaks with ATAC peak within 500 fraction  confident promoters  \\\n",
       "CHO                                         0.908412               5271.0   \n",
       "Brain                                       0.837496               4080.0   \n",
       "Liver                                       0.191389               3426.0   \n",
       "Lung                                        0.630659               4036.0   \n",
       "Kidney                                      0.767390               3813.0   \n",
       "Spleen                                      0.606352               3117.0   \n",
       "BMDMwt                                      0.518584               3138.0   \n",
       "\n",
       "        confident promoters with ATAC peak within 100  \\\n",
       "CHO                                            4878.0   \n",
       "Brain                                          3916.0   \n",
       "Liver                                          1171.0   \n",
       "Lung                                           3512.0   \n",
       "Kidney                                         3604.0   \n",
       "Spleen                                         2655.0   \n",
       "BMDMwt                                         2998.0   \n",
       "\n",
       "        confident promoters with ATAC peak within 100 fraction  \\\n",
       "CHO                                              0.925441        \n",
       "Brain                                            0.959804        \n",
       "Liver                                            0.341798        \n",
       "Lung                                             0.870168        \n",
       "Kidney                                           0.945188        \n",
       "Spleen                                           0.851781        \n",
       "BMDMwt                                           0.955386        \n",
       "\n",
       "        confident promoters with ATAC peak within 500  \\\n",
       "CHO                                            5124.0   \n",
       "Brain                                          4013.0   \n",
       "Liver                                          1371.0   \n",
       "Lung                                           3738.0   \n",
       "Kidney                                         3721.0   \n",
       "Spleen                                         2792.0   \n",
       "BMDMwt                                         3108.0   \n",
       "\n",
       "        confident promoters with ATAC peak within 500 fraction  \\\n",
       "CHO                                              0.972112        \n",
       "Brain                                            0.983578        \n",
       "Liver                                            0.400175        \n",
       "Lung                                             0.926165        \n",
       "Kidney                                           0.975872        \n",
       "Spleen                                           0.895733        \n",
       "BMDMwt                                           0.990440        \n",
       "\n",
       "                           ...                       \\\n",
       "CHO                        ...                        \n",
       "Brain                      ...                        \n",
       "Liver                      ...                        \n",
       "Lung                       ...                        \n",
       "Kidney                     ...                        \n",
       "Spleen                     ...                        \n",
       "BMDMwt                     ...                        \n",
       "\n",
       "        confident promoters noDiv PICR  \\\n",
       "CHO                            11117.0   \n",
       "Brain                          13374.0   \n",
       "Liver                          11272.0   \n",
       "Lung                           12684.0   \n",
       "Kidney                         12050.0   \n",
       "Spleen                         11313.0   \n",
       "BMDMwt                         10751.0   \n",
       "\n",
       "        confident promoters noDiv PICR with ATAC peak within 100  \\\n",
       "CHO                                                9096.0          \n",
       "Brain                                             10650.0          \n",
       "Liver                                              3212.0          \n",
       "Lung                                               8922.0          \n",
       "Kidney                                             9710.0          \n",
       "Spleen                                             7732.0          \n",
       "BMDMwt                                             8516.0          \n",
       "\n",
       "        confident promoters noDiv PICR with ATAC peak within 100 fraction  \\\n",
       "CHO                                              0.818206                   \n",
       "Brain                                            0.796321                   \n",
       "Liver                                            0.284954                   \n",
       "Lung                                             0.703406                   \n",
       "Kidney                                           0.805809                   \n",
       "Spleen                                           0.683462                   \n",
       "BMDMwt                                           0.792112                   \n",
       "\n",
       "        confident promoters noDiv PICR with ATAC peak within 500  \\\n",
       "CHO                                               10300.0          \n",
       "Brain                                             11876.0          \n",
       "Liver                                              3993.0          \n",
       "Lung                                              10120.0          \n",
       "Kidney                                            10801.0          \n",
       "Spleen                                             8728.0          \n",
       "BMDMwt                                             9517.0          \n",
       "\n",
       "        confident promoters noDiv PICR with ATAC peak within 500 fraction  \\\n",
       "CHO                                              0.926509                   \n",
       "Brain                                            0.887992                   \n",
       "Liver                                            0.354241                   \n",
       "Lung                                             0.797856                   \n",
       "Kidney                                           0.896349                   \n",
       "Spleen                                           0.771502                   \n",
       "BMDMwt                                           0.885220                   \n",
       "\n",
       "        no peaks  no peaks with ATAC peak within 100  \\\n",
       "CHO      23724.0                             19227.0   \n",
       "Brain    23792.0                             20947.0   \n",
       "Liver    24409.0                             10256.0   \n",
       "Lung     24324.0                             18765.0   \n",
       "Kidney   24322.0                             21619.0   \n",
       "Spleen   24586.0                             18175.0   \n",
       "BMDMwt   24294.0                             21493.0   \n",
       "\n",
       "        no peaks with ATAC peak within 100 fraction  \\\n",
       "CHO                                        0.810445   \n",
       "Brain                                      0.880422   \n",
       "Liver                                      0.420173   \n",
       "Lung                                       0.771460   \n",
       "Kidney                                     0.888866   \n",
       "Spleen                                     0.739242   \n",
       "BMDMwt                                     0.884704   \n",
       "\n",
       "        no peaks with ATAC peak within 500  \\\n",
       "CHO                                19531.0   \n",
       "Brain                              21249.0   \n",
       "Liver                              10732.0   \n",
       "Lung                               19223.0   \n",
       "Kidney                             21906.0   \n",
       "Spleen                             18624.0   \n",
       "BMDMwt                             21747.0   \n",
       "\n",
       "        no peaks with ATAC peak within 500 fraction  \n",
       "CHO                                        0.823259  \n",
       "Brain                                      0.893115  \n",
       "Liver                                      0.439674  \n",
       "Lung                                       0.790289  \n",
       "Kidney                                     0.900666  \n",
       "Spleen                                     0.757504  \n",
       "BMDMwt                                     0.895159  \n",
       "\n",
       "[7 rows x 30 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_names = np.array(atac_tss_df.columns.values)\n",
    "#to_drop = col_names[map(lambda x: 'fraction' not in x or '100' not in x,atac_tss_df.columns.values)]\n",
    "#output = atac_tss_df.drop(to_drop,axis=1) \n",
    "output = atac_tss_df.copy()\n",
    "col_names = np.array(output.columns.values)\n",
    "for i,name in enumerate((col_names)):\n",
    "    name = name.replace('.bed','')\n",
    "    #name = name.replace('fraction','')\n",
    "    name = name.replace('ATAC','ATAC peak within')\n",
    "    #name = 'Fraction of ' + name\n",
    "    #name = name + 'bp'\n",
    "    col_names[i] = name.replace('_',' ')\n",
    "output.columns = col_names\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:43:22.111851Z",
     "start_time": "2018-11-26T23:43:21.990149Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHO</th>\n",
       "      <th>Brain</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Kidney</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>BMDMwt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>all peaks</th>\n",
       "      <td>54931.000000</td>\n",
       "      <td>37316.000000</td>\n",
       "      <td>56905.000000</td>\n",
       "      <td>45148.000000</td>\n",
       "      <td>50389.000000</td>\n",
       "      <td>38631.000000</td>\n",
       "      <td>85558.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all peaks with ATAC peak within 100</th>\n",
       "      <td>46443.000000</td>\n",
       "      <td>29313.000000</td>\n",
       "      <td>8847.000000</td>\n",
       "      <td>25302.000000</td>\n",
       "      <td>35906.000000</td>\n",
       "      <td>21151.000000</td>\n",
       "      <td>38630.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all peaks with ATAC peak within 100 fraction</th>\n",
       "      <td>0.845479</td>\n",
       "      <td>0.785534</td>\n",
       "      <td>0.155470</td>\n",
       "      <td>0.560423</td>\n",
       "      <td>0.712576</td>\n",
       "      <td>0.547514</td>\n",
       "      <td>0.451507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all peaks with ATAC peak within 500</th>\n",
       "      <td>49900.000000</td>\n",
       "      <td>31252.000000</td>\n",
       "      <td>10891.000000</td>\n",
       "      <td>28473.000000</td>\n",
       "      <td>38668.000000</td>\n",
       "      <td>23424.000000</td>\n",
       "      <td>44369.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all peaks with ATAC peak within 500 fraction</th>\n",
       "      <td>0.908412</td>\n",
       "      <td>0.837496</td>\n",
       "      <td>0.191389</td>\n",
       "      <td>0.630659</td>\n",
       "      <td>0.767390</td>\n",
       "      <td>0.606352</td>\n",
       "      <td>0.518584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters</th>\n",
       "      <td>5271.000000</td>\n",
       "      <td>4080.000000</td>\n",
       "      <td>3426.000000</td>\n",
       "      <td>4036.000000</td>\n",
       "      <td>3813.000000</td>\n",
       "      <td>3117.000000</td>\n",
       "      <td>3138.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters with ATAC peak within 100</th>\n",
       "      <td>4878.000000</td>\n",
       "      <td>3916.000000</td>\n",
       "      <td>1171.000000</td>\n",
       "      <td>3512.000000</td>\n",
       "      <td>3604.000000</td>\n",
       "      <td>2655.000000</td>\n",
       "      <td>2998.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters with ATAC peak within 100 fraction</th>\n",
       "      <td>0.925441</td>\n",
       "      <td>0.959804</td>\n",
       "      <td>0.341798</td>\n",
       "      <td>0.870168</td>\n",
       "      <td>0.945188</td>\n",
       "      <td>0.851781</td>\n",
       "      <td>0.955386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters with ATAC peak within 500</th>\n",
       "      <td>5124.000000</td>\n",
       "      <td>4013.000000</td>\n",
       "      <td>1371.000000</td>\n",
       "      <td>3738.000000</td>\n",
       "      <td>3721.000000</td>\n",
       "      <td>2792.000000</td>\n",
       "      <td>3108.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters with ATAC peak within 500 fraction</th>\n",
       "      <td>0.972112</td>\n",
       "      <td>0.983578</td>\n",
       "      <td>0.400175</td>\n",
       "      <td>0.926165</td>\n",
       "      <td>0.975872</td>\n",
       "      <td>0.895733</td>\n",
       "      <td>0.990440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident enhancers</th>\n",
       "      <td>1302.000000</td>\n",
       "      <td>164.000000</td>\n",
       "      <td>613.000000</td>\n",
       "      <td>230.000000</td>\n",
       "      <td>478.000000</td>\n",
       "      <td>324.000000</td>\n",
       "      <td>575.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident enhancers with ATAC peak within 100</th>\n",
       "      <td>1187.000000</td>\n",
       "      <td>148.000000</td>\n",
       "      <td>63.000000</td>\n",
       "      <td>146.000000</td>\n",
       "      <td>427.000000</td>\n",
       "      <td>150.000000</td>\n",
       "      <td>518.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident enhancers with ATAC peak within 100 fraction</th>\n",
       "      <td>0.911674</td>\n",
       "      <td>0.902439</td>\n",
       "      <td>0.102773</td>\n",
       "      <td>0.634783</td>\n",
       "      <td>0.893305</td>\n",
       "      <td>0.462963</td>\n",
       "      <td>0.900870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident enhancers with ATAC peak within 500</th>\n",
       "      <td>1233.000000</td>\n",
       "      <td>155.000000</td>\n",
       "      <td>77.000000</td>\n",
       "      <td>162.000000</td>\n",
       "      <td>441.000000</td>\n",
       "      <td>167.000000</td>\n",
       "      <td>545.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident enhancers with ATAC peak within 500 fraction</th>\n",
       "      <td>0.947005</td>\n",
       "      <td>0.945122</td>\n",
       "      <td>0.125612</td>\n",
       "      <td>0.704348</td>\n",
       "      <td>0.922594</td>\n",
       "      <td>0.515432</td>\n",
       "      <td>0.947826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv</th>\n",
       "      <td>13044.000000</td>\n",
       "      <td>14974.000000</td>\n",
       "      <td>12863.000000</td>\n",
       "      <td>14468.000000</td>\n",
       "      <td>13651.000000</td>\n",
       "      <td>12435.000000</td>\n",
       "      <td>12285.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv with ATAC peak within 100</th>\n",
       "      <td>11456.000000</td>\n",
       "      <td>12904.000000</td>\n",
       "      <td>3774.000000</td>\n",
       "      <td>10804.000000</td>\n",
       "      <td>11735.000000</td>\n",
       "      <td>9051.000000</td>\n",
       "      <td>10033.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv with ATAC peak within 100 fraction</th>\n",
       "      <td>0.878258</td>\n",
       "      <td>0.861760</td>\n",
       "      <td>0.293400</td>\n",
       "      <td>0.746751</td>\n",
       "      <td>0.859644</td>\n",
       "      <td>0.727865</td>\n",
       "      <td>0.816687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv with ATAC peak within 500</th>\n",
       "      <td>12363.000000</td>\n",
       "      <td>13613.000000</td>\n",
       "      <td>4526.000000</td>\n",
       "      <td>11837.000000</td>\n",
       "      <td>12475.000000</td>\n",
       "      <td>9751.000000</td>\n",
       "      <td>11024.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv with ATAC peak within 500 fraction</th>\n",
       "      <td>0.947792</td>\n",
       "      <td>0.909109</td>\n",
       "      <td>0.351862</td>\n",
       "      <td>0.818150</td>\n",
       "      <td>0.913852</td>\n",
       "      <td>0.784158</td>\n",
       "      <td>0.897354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv PICR</th>\n",
       "      <td>11117.000000</td>\n",
       "      <td>13374.000000</td>\n",
       "      <td>11272.000000</td>\n",
       "      <td>12684.000000</td>\n",
       "      <td>12050.000000</td>\n",
       "      <td>11313.000000</td>\n",
       "      <td>10751.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv PICR with ATAC peak within 100</th>\n",
       "      <td>9096.000000</td>\n",
       "      <td>10650.000000</td>\n",
       "      <td>3212.000000</td>\n",
       "      <td>8922.000000</td>\n",
       "      <td>9710.000000</td>\n",
       "      <td>7732.000000</td>\n",
       "      <td>8516.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv PICR with ATAC peak within 100 fraction</th>\n",
       "      <td>0.818206</td>\n",
       "      <td>0.796321</td>\n",
       "      <td>0.284954</td>\n",
       "      <td>0.703406</td>\n",
       "      <td>0.805809</td>\n",
       "      <td>0.683462</td>\n",
       "      <td>0.792112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv PICR with ATAC peak within 500</th>\n",
       "      <td>10300.000000</td>\n",
       "      <td>11876.000000</td>\n",
       "      <td>3993.000000</td>\n",
       "      <td>10120.000000</td>\n",
       "      <td>10801.000000</td>\n",
       "      <td>8728.000000</td>\n",
       "      <td>9517.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident promoters noDiv PICR with ATAC peak within 500 fraction</th>\n",
       "      <td>0.926509</td>\n",
       "      <td>0.887992</td>\n",
       "      <td>0.354241</td>\n",
       "      <td>0.797856</td>\n",
       "      <td>0.896349</td>\n",
       "      <td>0.771502</td>\n",
       "      <td>0.885220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no peaks</th>\n",
       "      <td>23724.000000</td>\n",
       "      <td>23792.000000</td>\n",
       "      <td>24409.000000</td>\n",
       "      <td>24324.000000</td>\n",
       "      <td>24322.000000</td>\n",
       "      <td>24586.000000</td>\n",
       "      <td>24294.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no peaks with ATAC peak within 100</th>\n",
       "      <td>19227.000000</td>\n",
       "      <td>20947.000000</td>\n",
       "      <td>10256.000000</td>\n",
       "      <td>18765.000000</td>\n",
       "      <td>21619.000000</td>\n",
       "      <td>18175.000000</td>\n",
       "      <td>21493.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no peaks with ATAC peak within 100 fraction</th>\n",
       "      <td>0.810445</td>\n",
       "      <td>0.880422</td>\n",
       "      <td>0.420173</td>\n",
       "      <td>0.771460</td>\n",
       "      <td>0.888866</td>\n",
       "      <td>0.739242</td>\n",
       "      <td>0.884704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no peaks with ATAC peak within 500</th>\n",
       "      <td>19531.000000</td>\n",
       "      <td>21249.000000</td>\n",
       "      <td>10732.000000</td>\n",
       "      <td>19223.000000</td>\n",
       "      <td>21906.000000</td>\n",
       "      <td>18624.000000</td>\n",
       "      <td>21747.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no peaks with ATAC peak within 500 fraction</th>\n",
       "      <td>0.823259</td>\n",
       "      <td>0.893115</td>\n",
       "      <td>0.439674</td>\n",
       "      <td>0.790289</td>\n",
       "      <td>0.900666</td>\n",
       "      <td>0.757504</td>\n",
       "      <td>0.895159</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             CHO  \\\n",
       "all peaks                                           54931.000000   \n",
       "all peaks with ATAC peak within 100                 46443.000000   \n",
       "all peaks with ATAC peak within 100 fraction            0.845479   \n",
       "all peaks with ATAC peak within 500                 49900.000000   \n",
       "all peaks with ATAC peak within 500 fraction            0.908412   \n",
       "confident promoters                                  5271.000000   \n",
       "confident promoters with ATAC peak within 100        4878.000000   \n",
       "confident promoters with ATAC peak within 100 f...      0.925441   \n",
       "confident promoters with ATAC peak within 500        5124.000000   \n",
       "confident promoters with ATAC peak within 500 f...      0.972112   \n",
       "confident enhancers                                  1302.000000   \n",
       "confident enhancers with ATAC peak within 100        1187.000000   \n",
       "confident enhancers with ATAC peak within 100 f...      0.911674   \n",
       "confident enhancers with ATAC peak within 500        1233.000000   \n",
       "confident enhancers with ATAC peak within 500 f...      0.947005   \n",
       "confident promoters noDiv                           13044.000000   \n",
       "confident promoters noDiv with ATAC peak within...  11456.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.878258   \n",
       "confident promoters noDiv with ATAC peak within...  12363.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.947792   \n",
       "confident promoters noDiv PICR                      11117.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...   9096.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.818206   \n",
       "confident promoters noDiv PICR with ATAC peak w...  10300.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.926509   \n",
       "no peaks                                            23724.000000   \n",
       "no peaks with ATAC peak within 100                  19227.000000   \n",
       "no peaks with ATAC peak within 100 fraction             0.810445   \n",
       "no peaks with ATAC peak within 500                  19531.000000   \n",
       "no peaks with ATAC peak within 500 fraction             0.823259   \n",
       "\n",
       "                                                           Brain  \\\n",
       "all peaks                                           37316.000000   \n",
       "all peaks with ATAC peak within 100                 29313.000000   \n",
       "all peaks with ATAC peak within 100 fraction            0.785534   \n",
       "all peaks with ATAC peak within 500                 31252.000000   \n",
       "all peaks with ATAC peak within 500 fraction            0.837496   \n",
       "confident promoters                                  4080.000000   \n",
       "confident promoters with ATAC peak within 100        3916.000000   \n",
       "confident promoters with ATAC peak within 100 f...      0.959804   \n",
       "confident promoters with ATAC peak within 500        4013.000000   \n",
       "confident promoters with ATAC peak within 500 f...      0.983578   \n",
       "confident enhancers                                   164.000000   \n",
       "confident enhancers with ATAC peak within 100         148.000000   \n",
       "confident enhancers with ATAC peak within 100 f...      0.902439   \n",
       "confident enhancers with ATAC peak within 500         155.000000   \n",
       "confident enhancers with ATAC peak within 500 f...      0.945122   \n",
       "confident promoters noDiv                           14974.000000   \n",
       "confident promoters noDiv with ATAC peak within...  12904.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.861760   \n",
       "confident promoters noDiv with ATAC peak within...  13613.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.909109   \n",
       "confident promoters noDiv PICR                      13374.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...  10650.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.796321   \n",
       "confident promoters noDiv PICR with ATAC peak w...  11876.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.887992   \n",
       "no peaks                                            23792.000000   \n",
       "no peaks with ATAC peak within 100                  20947.000000   \n",
       "no peaks with ATAC peak within 100 fraction             0.880422   \n",
       "no peaks with ATAC peak within 500                  21249.000000   \n",
       "no peaks with ATAC peak within 500 fraction             0.893115   \n",
       "\n",
       "                                                           Liver  \\\n",
       "all peaks                                           56905.000000   \n",
       "all peaks with ATAC peak within 100                  8847.000000   \n",
       "all peaks with ATAC peak within 100 fraction            0.155470   \n",
       "all peaks with ATAC peak within 500                 10891.000000   \n",
       "all peaks with ATAC peak within 500 fraction            0.191389   \n",
       "confident promoters                                  3426.000000   \n",
       "confident promoters with ATAC peak within 100        1171.000000   \n",
       "confident promoters with ATAC peak within 100 f...      0.341798   \n",
       "confident promoters with ATAC peak within 500        1371.000000   \n",
       "confident promoters with ATAC peak within 500 f...      0.400175   \n",
       "confident enhancers                                   613.000000   \n",
       "confident enhancers with ATAC peak within 100          63.000000   \n",
       "confident enhancers with ATAC peak within 100 f...      0.102773   \n",
       "confident enhancers with ATAC peak within 500          77.000000   \n",
       "confident enhancers with ATAC peak within 500 f...      0.125612   \n",
       "confident promoters noDiv                           12863.000000   \n",
       "confident promoters noDiv with ATAC peak within...   3774.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.293400   \n",
       "confident promoters noDiv with ATAC peak within...   4526.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.351862   \n",
       "confident promoters noDiv PICR                      11272.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...   3212.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.284954   \n",
       "confident promoters noDiv PICR with ATAC peak w...   3993.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.354241   \n",
       "no peaks                                            24409.000000   \n",
       "no peaks with ATAC peak within 100                  10256.000000   \n",
       "no peaks with ATAC peak within 100 fraction             0.420173   \n",
       "no peaks with ATAC peak within 500                  10732.000000   \n",
       "no peaks with ATAC peak within 500 fraction             0.439674   \n",
       "\n",
       "                                                            Lung  \\\n",
       "all peaks                                           45148.000000   \n",
       "all peaks with ATAC peak within 100                 25302.000000   \n",
       "all peaks with ATAC peak within 100 fraction            0.560423   \n",
       "all peaks with ATAC peak within 500                 28473.000000   \n",
       "all peaks with ATAC peak within 500 fraction            0.630659   \n",
       "confident promoters                                  4036.000000   \n",
       "confident promoters with ATAC peak within 100        3512.000000   \n",
       "confident promoters with ATAC peak within 100 f...      0.870168   \n",
       "confident promoters with ATAC peak within 500        3738.000000   \n",
       "confident promoters with ATAC peak within 500 f...      0.926165   \n",
       "confident enhancers                                   230.000000   \n",
       "confident enhancers with ATAC peak within 100         146.000000   \n",
       "confident enhancers with ATAC peak within 100 f...      0.634783   \n",
       "confident enhancers with ATAC peak within 500         162.000000   \n",
       "confident enhancers with ATAC peak within 500 f...      0.704348   \n",
       "confident promoters noDiv                           14468.000000   \n",
       "confident promoters noDiv with ATAC peak within...  10804.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.746751   \n",
       "confident promoters noDiv with ATAC peak within...  11837.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.818150   \n",
       "confident promoters noDiv PICR                      12684.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...   8922.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.703406   \n",
       "confident promoters noDiv PICR with ATAC peak w...  10120.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.797856   \n",
       "no peaks                                            24324.000000   \n",
       "no peaks with ATAC peak within 100                  18765.000000   \n",
       "no peaks with ATAC peak within 100 fraction             0.771460   \n",
       "no peaks with ATAC peak within 500                  19223.000000   \n",
       "no peaks with ATAC peak within 500 fraction             0.790289   \n",
       "\n",
       "                                                          Kidney  \\\n",
       "all peaks                                           50389.000000   \n",
       "all peaks with ATAC peak within 100                 35906.000000   \n",
       "all peaks with ATAC peak within 100 fraction            0.712576   \n",
       "all peaks with ATAC peak within 500                 38668.000000   \n",
       "all peaks with ATAC peak within 500 fraction            0.767390   \n",
       "confident promoters                                  3813.000000   \n",
       "confident promoters with ATAC peak within 100        3604.000000   \n",
       "confident promoters with ATAC peak within 100 f...      0.945188   \n",
       "confident promoters with ATAC peak within 500        3721.000000   \n",
       "confident promoters with ATAC peak within 500 f...      0.975872   \n",
       "confident enhancers                                   478.000000   \n",
       "confident enhancers with ATAC peak within 100         427.000000   \n",
       "confident enhancers with ATAC peak within 100 f...      0.893305   \n",
       "confident enhancers with ATAC peak within 500         441.000000   \n",
       "confident enhancers with ATAC peak within 500 f...      0.922594   \n",
       "confident promoters noDiv                           13651.000000   \n",
       "confident promoters noDiv with ATAC peak within...  11735.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.859644   \n",
       "confident promoters noDiv with ATAC peak within...  12475.000000   \n",
       "confident promoters noDiv with ATAC peak within...      0.913852   \n",
       "confident promoters noDiv PICR                      12050.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...   9710.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.805809   \n",
       "confident promoters noDiv PICR with ATAC peak w...  10801.000000   \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.896349   \n",
       "no peaks                                            24322.000000   \n",
       "no peaks with ATAC peak within 100                  21619.000000   \n",
       "no peaks with ATAC peak within 100 fraction             0.888866   \n",
       "no peaks with ATAC peak within 500                  21906.000000   \n",
       "no peaks with ATAC peak within 500 fraction             0.900666   \n",
       "\n",
       "                                                          Spleen        BMDMwt  \n",
       "all peaks                                           38631.000000  85558.000000  \n",
       "all peaks with ATAC peak within 100                 21151.000000  38630.000000  \n",
       "all peaks with ATAC peak within 100 fraction            0.547514      0.451507  \n",
       "all peaks with ATAC peak within 500                 23424.000000  44369.000000  \n",
       "all peaks with ATAC peak within 500 fraction            0.606352      0.518584  \n",
       "confident promoters                                  3117.000000   3138.000000  \n",
       "confident promoters with ATAC peak within 100        2655.000000   2998.000000  \n",
       "confident promoters with ATAC peak within 100 f...      0.851781      0.955386  \n",
       "confident promoters with ATAC peak within 500        2792.000000   3108.000000  \n",
       "confident promoters with ATAC peak within 500 f...      0.895733      0.990440  \n",
       "confident enhancers                                   324.000000    575.000000  \n",
       "confident enhancers with ATAC peak within 100         150.000000    518.000000  \n",
       "confident enhancers with ATAC peak within 100 f...      0.462963      0.900870  \n",
       "confident enhancers with ATAC peak within 500         167.000000    545.000000  \n",
       "confident enhancers with ATAC peak within 500 f...      0.515432      0.947826  \n",
       "confident promoters noDiv                           12435.000000  12285.000000  \n",
       "confident promoters noDiv with ATAC peak within...   9051.000000  10033.000000  \n",
       "confident promoters noDiv with ATAC peak within...      0.727865      0.816687  \n",
       "confident promoters noDiv with ATAC peak within...   9751.000000  11024.000000  \n",
       "confident promoters noDiv with ATAC peak within...      0.784158      0.897354  \n",
       "confident promoters noDiv PICR                      11313.000000  10751.000000  \n",
       "confident promoters noDiv PICR with ATAC peak w...   7732.000000   8516.000000  \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.683462      0.792112  \n",
       "confident promoters noDiv PICR with ATAC peak w...   8728.000000   9517.000000  \n",
       "confident promoters noDiv PICR with ATAC peak w...      0.771502      0.885220  \n",
       "no peaks                                            24586.000000  24294.000000  \n",
       "no peaks with ATAC peak within 100                  18175.000000  21493.000000  \n",
       "no peaks with ATAC peak within 100 fraction             0.739242      0.884704  \n",
       "no peaks with ATAC peak within 500                  18624.000000  21747.000000  \n",
       "no peaks with ATAC peak within 500 fraction             0.757504      0.895159  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output.transpose()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---------"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
