{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pickle\n",
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "tss = pd.read_csv('Results/tss_final/tss_peakCenter_1kb_1kb.bed',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_df_1kb_1kb = pickle.load(open(\"Results/gene_df_1kb_1kb.p\",'rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "txn_df_1kb_1kb = pickle.load(open(\"Results/txn_df_1kb_1kb.p\",'rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>peaks</th>\n",
       "      <th>numberOfPeaks</th>\n",
       "      <th>bin_-1000_1000</th>\n",
       "      <th>tss_bin_-1000_1000</th>\n",
       "      <th>tissues</th>\n",
       "      <th>minDistance</th>\n",
       "      <th>minDistancePeak</th>\n",
       "      <th>maxValue</th>\n",
       "      <th>maxPeakId</th>\n",
       "      <th>maxTissues</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>hasGene</th>\n",
       "      <th>Number of SS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>asmbl_2</th>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>inf</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_4</th>\n",
       "      <td>[Merged-picr_0-45924-1, Merged-picr_0-50486-1,...</td>\n",
       "      <td>3</td>\n",
       "      <td>[Merged-picr_0-45924-1]</td>\n",
       "      <td>[Merged-picr_0-45924-1]</td>\n",
       "      <td>[CHO_mSTART]</td>\n",
       "      <td>706.000000</td>\n",
       "      <td>Merged-picr_0-45924-1</td>\n",
       "      <td>3.04</td>\n",
       "      <td>Merged-picr_0-45924-1</td>\n",
       "      <td>[CHO_mSTART]</td>\n",
       "      <td>picr_0</td>\n",
       "      <td>45849</td>\n",
       "      <td>45999</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_7</th>\n",
       "      <td>[Merged-picr_0-44638-7]</td>\n",
       "      <td>1</td>\n",
       "      <td>[Merged-picr_0-44638-7]</td>\n",
       "      <td>[]</td>\n",
       "      <td></td>\n",
       "      <td>inf</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_8</th>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>inf</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_9</th>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>inf</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     peaks  numberOfPeaks  \\\n",
       "asmbl_2                                                                 0   \n",
       "asmbl_4  [Merged-picr_0-45924-1, Merged-picr_0-50486-1,...              3   \n",
       "asmbl_7                            [Merged-picr_0-44638-7]              1   \n",
       "asmbl_8                                                                 0   \n",
       "asmbl_9                                                                 0   \n",
       "\n",
       "                  bin_-1000_1000       tss_bin_-1000_1000       tissues  \\\n",
       "asmbl_2                      NaN                      NaN                 \n",
       "asmbl_4  [Merged-picr_0-45924-1]  [Merged-picr_0-45924-1]  [CHO_mSTART]   \n",
       "asmbl_7  [Merged-picr_0-44638-7]                       []                 \n",
       "asmbl_8                      NaN                      NaN                 \n",
       "asmbl_9                      NaN                      NaN                 \n",
       "\n",
       "         minDistance        minDistancePeak maxValue              maxPeakId  \\\n",
       "asmbl_2          inf                             NaN                    NaN   \n",
       "asmbl_4   706.000000  Merged-picr_0-45924-1     3.04  Merged-picr_0-45924-1   \n",
       "asmbl_7          inf                             NaN                    NaN   \n",
       "asmbl_8          inf                             NaN                    NaN   \n",
       "asmbl_9          inf                             NaN                    NaN   \n",
       "\n",
       "           maxTissues     Chr  Start    End gene_id   gene transcript  \\\n",
       "asmbl_2                   NaN    NaN    NaN     NaN    NaN        NaN   \n",
       "asmbl_4  [CHO_mSTART]  picr_0  45849  45999  gene_2  RRP7A    asmbl_4   \n",
       "asmbl_7                   NaN    NaN    NaN     NaN    NaN        NaN   \n",
       "asmbl_8                   NaN    NaN    NaN     NaN    NaN        NaN   \n",
       "asmbl_9                   NaN    NaN    NaN     NaN    NaN        NaN   \n",
       "\n",
       "         hasGene  Number of SS  \n",
       "asmbl_2    False             0  \n",
       "asmbl_4     True             1  \n",
       "asmbl_7    False             0  \n",
       "asmbl_8    False             0  \n",
       "asmbl_9    False             0  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "txn_df_1kb_1kb.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in gff file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cds = pd.read_csv('/data/isshamie/genome/CDS.gff3',sep='\\t',header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>CDS</td>\n",
       "      <td>10450</td>\n",
       "      <td>10535</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=asmbl_6.cds7;Parent=asmbl_6;gene_id=gene_2;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>CDS</td>\n",
       "      <td>10450</td>\n",
       "      <td>10535</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=asmbl_9.cds7;Parent=asmbl_9;gene_id=gene_2;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>CDS</td>\n",
       "      <td>10450</td>\n",
       "      <td>10535</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>ID=asmbl_8.cds7;Parent=asmbl_8;gene_id=gene_2;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>CDS</td>\n",
       "      <td>11432</td>\n",
       "      <td>11630</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=asmbl_6.cds6;Parent=asmbl_6;gene_id=gene_2;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>CDS</td>\n",
       "      <td>11432</td>\n",
       "      <td>11630</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>ID=asmbl_9.cds6;Parent=asmbl_9;gene_id=gene_2;...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        0      1    2      3      4  5  6  7  \\\n",
       "0  picr_0  merge  CDS  10450  10535  .  -  1   \n",
       "1  picr_0  merge  CDS  10450  10535  .  -  1   \n",
       "2  picr_0  merge  CDS  10450  10535  .  -  1   \n",
       "3  picr_0  merge  CDS  11432  11630  .  -  0   \n",
       "4  picr_0  merge  CDS  11432  11630  .  -  0   \n",
       "\n",
       "                                                   8  \n",
       "0  ID=asmbl_6.cds7;Parent=asmbl_6;gene_id=gene_2;...  \n",
       "1  ID=asmbl_9.cds7;Parent=asmbl_9;gene_id=gene_2;...  \n",
       "2  ID=asmbl_8.cds7;Parent=asmbl_8;gene_id=gene_2;...  \n",
       "3  ID=asmbl_6.cds6;Parent=asmbl_6;gene_id=gene_2;...  \n",
       "4  ID=asmbl_9.cds6;Parent=asmbl_9;gene_id=gene_2;...  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cds.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "anno = pd.read_csv('/data/isshamie/genome/updated_final_sort.gff3',sep='\\t',header=None,index_col=8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID=gene_1_1;gene_id=gene_1_1;gene=gene_1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>gene</td>\n",
       "      <td>10</td>\n",
       "      <td>41421</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_1;Parent=gene_1_1;gene_id=gene_1_1;gene=gene_1;transcript_id=asmbl_1;cs=5</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>10</td>\n",
       "      <td>41421</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_1.exon1;Parent=asmbl_1;gene_id=gene_1_1;gene=gene_1;transcript_id=asmbl_1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>exon</td>\n",
       "      <td>10</td>\n",
       "      <td>4041</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=gene_1_2;gene_id=gene_1_2;gene=SERHL</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>gene</td>\n",
       "      <td>176</td>\n",
       "      <td>38694</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gene=SERHL;transcript_id=asmbl_2;cs=2</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>merge</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>176</td>\n",
       "      <td>38694</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                         0      1     2    3  \\\n",
       "8                                                                              \n",
       "ID=gene_1_1;gene_id=gene_1_1;gene=gene_1            picr_0  merge  gene   10   \n",
       "ID=asmbl_1;Parent=gene_1_1;gene_id=gene_1_1;gen...  picr_0  merge  mRNA   10   \n",
       "ID=asmbl_1.exon1;Parent=asmbl_1;gene_id=gene_1_...  picr_0  merge  exon   10   \n",
       "ID=gene_1_2;gene_id=gene_1_2;gene=SERHL             picr_0  merge  gene  176   \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  picr_0  merge  mRNA  176   \n",
       "\n",
       "                                                        4  5  6  7  \n",
       "8                                                                   \n",
       "ID=gene_1_1;gene_id=gene_1_1;gene=gene_1            41421  .  +  .  \n",
       "ID=asmbl_1;Parent=gene_1_1;gene_id=gene_1_1;gen...  41421  .  +  .  \n",
       "ID=asmbl_1.exon1;Parent=asmbl_1;gene_id=gene_1_...   4041  .  +  .  \n",
       "ID=gene_1_2;gene_id=gene_1_2;gene=SERHL             38694  .  +  .  \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  38694  .  +  .  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:14: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  \n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "cannot convert float NaN to integer",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-27-9c375cdf6636>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mu'time'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34mu''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34mu\"#Munge data for downstream processing\\ncols = anno.columns.values\\ncols = ['Chr','Merge','Type','Start','End','.','Strand','..']\\nanno.columns= cols\\n#Drop the unncesseary columns\\nanno.drop(['Merge','.','..','Type'],axis=1,inplace=True)\\n\\n\\ntmp_anno = anno.copy()\\nfor ind in tmp_anno.index.values:\\n    curr = ind.split(';')\\n    for i in curr:\\n        curr_split = i.split('=')\\n        tmp_anno.set_value(ind,curr_split[0],curr_split[1])\\ntmp_anno['cs'] = tmp_anno['cs'].astype(int)\\ntmp_anno['Length'] = tmp_anno['End'] - tmp_anno['Start']\\ntmp_anno.head()\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.pyc\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2113\u001b[0m             \u001b[0mmagic_arg_s\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvar_expand\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstack_depth\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2114\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2115\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2116\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2117\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<decorator-gen-59>\u001b[0m in \u001b[0;36mtime\u001b[0;34m(self, line, cell, local_ns)\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/IPython/core/magic.pyc\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    187\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 188\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    189\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    190\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/IPython/core/magics/execution.pyc\u001b[0m in \u001b[0;36mtime\u001b[0;34m(self, line, cell, local_ns)\u001b[0m\n\u001b[1;32m   1183\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1184\u001b[0m             \u001b[0mst\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclock2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1185\u001b[0;31m             \u001b[0;32mexec\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mglob\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlocal_ns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1186\u001b[0m             \u001b[0mend\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclock2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1187\u001b[0m             \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/util/_decorators.pyc\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    176\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    177\u001b[0m                     \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnew_arg_name\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnew_arg_value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 178\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    179\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    180\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0m_deprecate_kwarg\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/generic.pyc\u001b[0m in \u001b[0;36mastype\u001b[0;34m(self, dtype, copy, errors, **kwargs)\u001b[0m\n\u001b[1;32m   4995\u001b[0m             \u001b[0;31m# else, only a single dtype is given\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4996\u001b[0m             new_data = self._data.astype(dtype=dtype, copy=copy, errors=errors,\n\u001b[0;32m-> 4997\u001b[0;31m                                          **kwargs)\n\u001b[0m\u001b[1;32m   4998\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_constructor\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnew_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__finalize__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4999\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/internals.pyc\u001b[0m in \u001b[0;36mastype\u001b[0;34m(self, dtype, **kwargs)\u001b[0m\n\u001b[1;32m   3712\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3713\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3714\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'astype'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3715\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3716\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mconvert\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/internals.pyc\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self, f, axes, filter, do_integrity_check, consolidate, **kwargs)\u001b[0m\n\u001b[1;32m   3579\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3580\u001b[0m             \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'mgr'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3581\u001b[0;31m             \u001b[0mapplied\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3582\u001b[0m             \u001b[0mresult_blocks\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_extend_blocks\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mapplied\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresult_blocks\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3583\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/internals.pyc\u001b[0m in \u001b[0;36mastype\u001b[0;34m(self, dtype, copy, errors, values, **kwargs)\u001b[0m\n\u001b[1;32m    573\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mFalse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merrors\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'raise'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    574\u001b[0m         return self._astype(dtype, copy=copy, errors=errors, values=values,\n\u001b[0;32m--> 575\u001b[0;31m                             **kwargs)\n\u001b[0m\u001b[1;32m    576\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    577\u001b[0m     def _astype(self, dtype, copy=False, errors='raise', values=None,\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/internals.pyc\u001b[0m in \u001b[0;36m_astype\u001b[0;34m(self, dtype, copy, errors, values, klass, mgr, **kwargs)\u001b[0m\n\u001b[1;32m    662\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    663\u001b[0m                 \u001b[0;31m# _astype_nansafe works fine with 1-d only\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 664\u001b[0;31m                 \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mastype_nansafe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    665\u001b[0m                 \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreshape\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    666\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/dtypes/cast.pyc\u001b[0m in \u001b[0;36mastype_nansafe\u001b[0;34m(arr, dtype, copy)\u001b[0m\n\u001b[1;32m    707\u001b[0m         \u001b[0;31m# work around NumPy brokenness, #1987\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    708\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0missubdtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minteger\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 709\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype_intsafe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreshape\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    710\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    711\u001b[0m         \u001b[0;31m# if we have a datetime/timedelta array of objects\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/lib.pyx\u001b[0m in \u001b[0;36mpandas._libs.lib.astype_intsafe\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/src/util.pxd\u001b[0m in \u001b[0;36mutil.set_value_at_unsafe\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: cannot convert float NaN to integer"
     ]
    }
   ],
   "source": [
    "%%time \n",
    "#Munge data for downstream processing\n",
    "cols = anno.columns.values\n",
    "cols = ['Chr','Merge','Type','Start','End','.','Strand','..']\n",
    "anno.columns= cols\n",
    "#Drop the unncesseary columns\n",
    "anno.drop(['Merge','.','..','Type'],axis=1,inplace=True)\n",
    "\n",
    "\n",
    "tmp_anno = anno.copy()\n",
    "for ind in tmp_anno.index.values:\n",
    "    curr = ind.split(';')\n",
    "    for i in curr:\n",
    "        curr_split = i.split('=')\n",
    "        tmp_anno.set_value(ind,curr_split[0],curr_split[1])\n",
    "\n",
    "tmp_anno['cs'] = tmp_anno['cs'].fillna(-1).astype(int)\n",
    "tmp_anno['Length'] = tmp_anno['End'] - tmp_anno['Start']\n",
    "tmp_anno.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>ID</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>Parent</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>cs</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID=gene_1_1;gene_id=gene_1_1;gene=gene_1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>10</td>\n",
       "      <td>41421</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_1_1</td>\n",
       "      <td>gene_1_1</td>\n",
       "      <td>gene_1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_1;Parent=gene_1_1;gene_id=gene_1_1;gene=gene_1;transcript_id=asmbl_1;cs=5</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>10</td>\n",
       "      <td>41421</td>\n",
       "      <td>+</td>\n",
       "      <td>asmbl_1</td>\n",
       "      <td>gene_1_1</td>\n",
       "      <td>gene_1</td>\n",
       "      <td>gene_1_1</td>\n",
       "      <td>asmbl_1</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_1.exon1;Parent=asmbl_1;gene_id=gene_1_1;gene=gene_1;transcript_id=asmbl_1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>10</td>\n",
       "      <td>4041</td>\n",
       "      <td>+</td>\n",
       "      <td>asmbl_1.exon1</td>\n",
       "      <td>gene_1_1</td>\n",
       "      <td>gene_1</td>\n",
       "      <td>asmbl_1</td>\n",
       "      <td>asmbl_1</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=gene_1_2;gene_id=gene_1_2;gene=SERHL</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>176</td>\n",
       "      <td>38694</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>SERHL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gene=SERHL;transcript_id=asmbl_2;cs=2</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>176</td>\n",
       "      <td>38694</td>\n",
       "      <td>+</td>\n",
       "      <td>asmbl_2</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>SERHL</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>asmbl_2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38518</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Chr  Start    End  \\\n",
       "8                                                                          \n",
       "ID=gene_1_1;gene_id=gene_1_1;gene=gene_1            picr_0     10  41421   \n",
       "ID=asmbl_1;Parent=gene_1_1;gene_id=gene_1_1;gen...  picr_0     10  41421   \n",
       "ID=asmbl_1.exon1;Parent=asmbl_1;gene_id=gene_1_...  picr_0     10   4041   \n",
       "ID=gene_1_2;gene_id=gene_1_2;gene=SERHL             picr_0    176  38694   \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  picr_0    176  38694   \n",
       "\n",
       "                                                   Strand             ID  \\\n",
       "8                                                                          \n",
       "ID=gene_1_1;gene_id=gene_1_1;gene=gene_1                +       gene_1_1   \n",
       "ID=asmbl_1;Parent=gene_1_1;gene_id=gene_1_1;gen...      +        asmbl_1   \n",
       "ID=asmbl_1.exon1;Parent=asmbl_1;gene_id=gene_1_...      +  asmbl_1.exon1   \n",
       "ID=gene_1_2;gene_id=gene_1_2;gene=SERHL                 +       gene_1_2   \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...      +        asmbl_2   \n",
       "\n",
       "                                                     gene_id    gene  \\\n",
       "8                                                                      \n",
       "ID=gene_1_1;gene_id=gene_1_1;gene=gene_1            gene_1_1  gene_1   \n",
       "ID=asmbl_1;Parent=gene_1_1;gene_id=gene_1_1;gen...  gene_1_1  gene_1   \n",
       "ID=asmbl_1.exon1;Parent=asmbl_1;gene_id=gene_1_...  gene_1_1  gene_1   \n",
       "ID=gene_1_2;gene_id=gene_1_2;gene=SERHL             gene_1_2   SERHL   \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  gene_1_2   SERHL   \n",
       "\n",
       "                                                      Parent transcript_id  \\\n",
       "8                                                                            \n",
       "ID=gene_1_1;gene_id=gene_1_1;gene=gene_1                 NaN           NaN   \n",
       "ID=asmbl_1;Parent=gene_1_1;gene_id=gene_1_1;gen...  gene_1_1       asmbl_1   \n",
       "ID=asmbl_1.exon1;Parent=asmbl_1;gene_id=gene_1_...   asmbl_1       asmbl_1   \n",
       "ID=gene_1_2;gene_id=gene_1_2;gene=SERHL                  NaN           NaN   \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  gene_1_2       asmbl_2   \n",
       "\n",
       "                                                    cs protein_id  Length  \n",
       "8                                                                          \n",
       "ID=gene_1_1;gene_id=gene_1_1;gene=gene_1            -1        NaN   41411  \n",
       "ID=asmbl_1;Parent=gene_1_1;gene_id=gene_1_1;gen...   5        NaN   41411  \n",
       "ID=asmbl_1.exon1;Parent=asmbl_1;gene_id=gene_1_...  -1        NaN    4031  \n",
       "ID=gene_1_2;gene_id=gene_1_2;gene=SERHL             -1        NaN   38518  \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...   2        NaN   38518  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp_anno['cs'] = tmp_anno['cs'].fillna(-1).astype(int)\n",
    "tmp_anno['Length'] = tmp_anno['End'] - tmp_anno['Start']\n",
    "tmp_anno.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loop through each peak and\n",
    "NOTE: Note which strand you are working with  \n",
    "1) Add sequence -2k:+2k  \n",
    "2) Determine if there is a nearby transcript id within 2k bp upstream and if it is part of the same gene_id or different. If it is the same, write where the end of it is.   \n",
    "3) Determine where the coding region of this associated gene is.   \n",
    "4) Determine if any genes downstream within 2kb (Does not matter which strand its in)  \n",
    "\n",
    "Next, loop through each non-experimental transcript, pick the first isoform and extract their sequence and do the same as before . \n",
    "\n",
    "INPUT: \n",
    "1) Annotation file: List of each mRNA, exon, and coding region  \n",
    "2) txn peak file  \n",
    "3) picr.fa genome for extracting sequences  \n",
    "4) txn_df to get transcripts that do not have a peak (May not be as necessary)  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extract sequences from tss output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ref_fa = '/data/isshamie/genome/picr.fa'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### First make flanking -2kb to +2kb for each peak"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss_experiment = 'Results/tss_final/tss_peakCenter_1kb_1kb.bed'\n",
    "out = 'Results/tss_final/tss_peakCenter_1kb_1kb.fa'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cmd = 'homerTools extract {bed_f} {ref_fa} -fa > {out}'.format(bed_f=tss_experiment,ref_fa=ref_fa,out=out)\n",
    "#!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "homerTools extract peaks.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss_seq = pd.DataFrame(index=gene_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "awk '{print $3}' /data/isshamie/genome/updated_final_sort.gff3 | sort | uniq  "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
