{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1405: UserWarning: \n",
      "This call to matplotlib.use() has no effect because the backend has already\n",
      "been chosen; matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import pickle\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "mpl.style.use('fivethirtyeight')\n",
    "from matplotlib import pyplot as plt\n",
    "from itertools import product\n",
    "import glob\n",
    "import re\n",
    "from matplotlib_venn import venn2\n",
    "from matplotlib import rcParams\n",
    "import inspect\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "sys.setrecursionlimit(3000)\n",
    "sns.set_context('notebook')\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "rcParams['figure.figsize'] = 8, 6\n",
    "#rcParams['font.family'] = 'sans-serif'\n",
    "#rcParams['font.sans-serif'] = ['Tahoma']\n",
    "import tqdm\n",
    "### notebook specific configuration ###\n",
    "from os.path import basename\n",
    "from tqdm import tqdm_notebook\n",
    "\n",
    "from Modules.Homer import *\n",
    "\n",
    "sys.path.append('/home/isshamie/software/NGS-Pipeline')\n",
    "\n",
    "data_folder = '/data/isshamie/CH_tissue_TSS/'\n",
    "\n",
    "ref_fa = '/data/genome/hamster/picr/picr.fa'\n",
    "annotation = '/data/genome/hamster/picr/updated_final_sort.gff3'\n",
    "tss_annotation = '/data/isshamie/genome/start_site_mRNA_updated_final_sort.tsv'\n",
    "mrna_peak = '/data/isshamie/TSS_CHO/mRNA.peak'\n",
    "\n",
    "tissues = ['BloondnegSpleen','BMDM1hKLA1','BMDMwt','Brain','Brain7neg1',\n",
    "          'FemaleReproductive','Heart','Intestine','Kidney','Liver','Lung',\n",
    "           'MiscOrgans','Muscle','Pancreas','Skin','Spleen','Total','CHO']\n",
    "\n",
    "len(tissues)\n",
    "\n",
    "### Load in meta_sampe\n",
    "\n",
    "meta_samples = pd.read_csv('Results/meta_samples.csv',index_col=0)\n",
    "\n",
    "long_meta_samples = pd.read_csv('Results/long_meta_samples.csv',index_col=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "txn_df = pickle.load(open('Results/txn_df.p','rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "gene_df = pickle.load(open('Results/gene_df.p','rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "gene_id_df = pickle.load(open('Results/gene_id_df.p','rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Only use genes with a peak\n",
    "gene_df = gene_df[gene_df['hasGene']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load gene_tissue_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BloondnegSpleen/GRO/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>BMDM1hKLA1/GRO/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>BMDMwt/GRO/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Brain/GRO/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Brain/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>FemaleReproductive/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Heart/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Kidney/GRO/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Kidney/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Liver/GRO/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Liver/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Lung/GRO/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Lung/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>MiscOrgans/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Muscle/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Pancreas/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>Spleen/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>CHO/GRO/f04_peaks/merge_bg_2.peak</th>\n",
       "      <th>CHO/mSTART/f04_peaks/merge_bg_2.peak</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SERHL</th>\n",
       "      <td>188.199997</td>\n",
       "      <td>13.300000</td>\n",
       "      <td>90.199997</td>\n",
       "      <td>333.500000</td>\n",
       "      <td>501.799988</td>\n",
       "      <td>1089.349976</td>\n",
       "      <td>777.750000</td>\n",
       "      <td>453.299988</td>\n",
       "      <td>18793.900391</td>\n",
       "      <td>374.600006</td>\n",
       "      <td>337.950012</td>\n",
       "      <td>294.100006</td>\n",
       "      <td>282.199982</td>\n",
       "      <td>697.300049</td>\n",
       "      <td>593.450012</td>\n",
       "      <td>529.299988</td>\n",
       "      <td>201.000000</td>\n",
       "      <td>971.449951</td>\n",
       "      <td>819.599976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RRP7A</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>25.299999</td>\n",
       "      <td>27.600000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>29.650000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>91.099998</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>34.299999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POLDIP3</th>\n",
       "      <td>135.199997</td>\n",
       "      <td>31.799999</td>\n",
       "      <td>97.699997</td>\n",
       "      <td>122.599998</td>\n",
       "      <td>367.649994</td>\n",
       "      <td>148.149994</td>\n",
       "      <td>79.500000</td>\n",
       "      <td>84.199997</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>85.099998</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>132.199997</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>174.800003</td>\n",
       "      <td>105.100006</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>135.500000</td>\n",
       "      <td>317.799988</td>\n",
       "      <td>324.100006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CYB5R3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>91.400002</td>\n",
       "      <td>45.099998</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>14.160000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.850000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>30.799999</td>\n",
       "      <td>324.600006</td>\n",
       "      <td>257.299988</td>\n",
       "      <td>59.900002</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>136.250000</td>\n",
       "      <td>345.149994</td>\n",
       "      <td>85.300003</td>\n",
       "      <td>48.950001</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>95.199997</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>98.949997</td>\n",
       "      <td>291.750000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>37.400002</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         BloondnegSpleen/GRO/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                       188.199997   \n",
       "RRP7A                                         0.000000   \n",
       "POLDIP3                                     135.199997   \n",
       "CYB5R3                                        0.000000   \n",
       "A4GALT                                       30.799999   \n",
       "\n",
       "         BMDM1hKLA1/GRO/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                   13.300000   \n",
       "RRP7A                                    0.000000   \n",
       "POLDIP3                                 31.799999   \n",
       "CYB5R3                                  91.400002   \n",
       "A4GALT                                 324.600006   \n",
       "\n",
       "         BMDMwt/GRO/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                               90.199997   \n",
       "RRP7A                                0.000000   \n",
       "POLDIP3                             97.699997   \n",
       "CYB5R3                              45.099998   \n",
       "A4GALT                             257.299988   \n",
       "\n",
       "         Brain/GRO/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                             333.500000   \n",
       "RRP7A                              25.299999   \n",
       "POLDIP3                           122.599998   \n",
       "CYB5R3                              0.000000   \n",
       "A4GALT                             59.900002   \n",
       "\n",
       "         Brain/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                501.799988   \n",
       "RRP7A                                 27.600000   \n",
       "POLDIP3                              367.649994   \n",
       "CYB5R3                                 0.000000   \n",
       "A4GALT                                 0.000000   \n",
       "\n",
       "         FemaleReproductive/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                          1089.349976     \n",
       "RRP7A                                             0.000000     \n",
       "POLDIP3                                         148.149994     \n",
       "CYB5R3                                           14.160000     \n",
       "A4GALT                                          136.250000     \n",
       "\n",
       "         Heart/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                777.750000   \n",
       "RRP7A                                  0.000000   \n",
       "POLDIP3                               79.500000   \n",
       "CYB5R3                                 0.000000   \n",
       "A4GALT                               345.149994   \n",
       "\n",
       "         Kidney/GRO/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                              453.299988   \n",
       "RRP7A                                0.000000   \n",
       "POLDIP3                             84.199997   \n",
       "CYB5R3                               0.000000   \n",
       "A4GALT                              85.300003   \n",
       "\n",
       "         Kidney/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                               18793.900391   \n",
       "RRP7A                                   0.000000   \n",
       "POLDIP3                                 0.000000   \n",
       "CYB5R3                                  0.000000   \n",
       "A4GALT                                 48.950001   \n",
       "\n",
       "         Liver/GRO/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                             374.600006   \n",
       "RRP7A                               0.000000   \n",
       "POLDIP3                            85.099998   \n",
       "CYB5R3                              0.000000   \n",
       "A4GALT                              0.000000   \n",
       "\n",
       "         Liver/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                337.950012   \n",
       "RRP7A                                  0.000000   \n",
       "POLDIP3                                0.000000   \n",
       "CYB5R3                                 0.000000   \n",
       "A4GALT                                 0.000000   \n",
       "\n",
       "         Lung/GRO/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                            294.100006   \n",
       "RRP7A                              0.000000   \n",
       "POLDIP3                          132.199997   \n",
       "CYB5R3                             0.000000   \n",
       "A4GALT                            95.199997   \n",
       "\n",
       "         Lung/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                               282.199982   \n",
       "RRP7A                                 0.000000   \n",
       "POLDIP3                               0.000000   \n",
       "CYB5R3                                0.000000   \n",
       "A4GALT                                0.000000   \n",
       "\n",
       "         MiscOrgans/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                     697.300049   \n",
       "RRP7A                                      29.650000   \n",
       "POLDIP3                                   174.800003   \n",
       "CYB5R3                                      0.000000   \n",
       "A4GALT                                     98.949997   \n",
       "\n",
       "         Muscle/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                 593.450012   \n",
       "RRP7A                                   0.000000   \n",
       "POLDIP3                               105.100006   \n",
       "CYB5R3                                  0.000000   \n",
       "A4GALT                                291.750000   \n",
       "\n",
       "         Pancreas/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                   529.299988   \n",
       "RRP7A                                     0.000000   \n",
       "POLDIP3                                   0.000000   \n",
       "CYB5R3                                    0.000000   \n",
       "A4GALT                                    0.000000   \n",
       "\n",
       "         Spleen/mSTART/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                                 201.000000   \n",
       "RRP7A                                  91.099998   \n",
       "POLDIP3                               135.500000   \n",
       "CYB5R3                                  0.000000   \n",
       "A4GALT                                 37.400002   \n",
       "\n",
       "         CHO/GRO/f04_peaks/merge_bg_2.peak  \\\n",
       "SERHL                           971.449951   \n",
       "RRP7A                             0.000000   \n",
       "POLDIP3                         317.799988   \n",
       "CYB5R3                            0.000000   \n",
       "A4GALT                            0.000000   \n",
       "\n",
       "         CHO/mSTART/f04_peaks/merge_bg_2.peak  \n",
       "SERHL                              819.599976  \n",
       "RRP7A                               34.299999  \n",
       "POLDIP3                            324.100006  \n",
       "CYB5R3                               7.850000  \n",
       "A4GALT                               0.000000  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_tissue_matrix = pd.read_csv('Results/merged_samples/genes_tissues.csv',index_col=0)\n",
    "gene_tissue_matrix.fillna(0,inplace=True)\n",
    "## Remove the genes with no peaks, if any\n",
    "gene_tissue_matrix = gene_tissue_matrix[~(gene_tissue_matrix==0).all(axis=1)]\n",
    "gene_tissue_matrix.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Different ways to separate in CHO and not in CHO\n",
    "1. Genes expressed in at least one tissue and not in CHO\n",
    "2. Genes that were expressed in every other tissue and not in CHO samples.    \n",
    "3. Genes expressed in >= 50% of the tissues and not in CHO samples.  \n",
    "4. Genes with counts greater than 10 in every other tissue and CHO has counts less than 10.  \n",
    "5. Genes with counts greater than 10 in >= 50% of the tissuea and CHO has counts less than 10.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#######################################\n",
    "## Merge the tissues together to take the max. \n",
    "def merge_tissues(tissue_df,tissues):\n",
    "    ''' Merging samples with the same tissue'''\n",
    "    gene_tissue_df = pd.DataFrame(index=tissue_df.index,columns=tissues)\n",
    "    for t in tissues:\n",
    "        curr_cols = tissue_df.loc[:,tissue_df.columns.str.contains(t)]\n",
    "        gene_tissue_df.at[:,t] = curr_cols.max(axis=1)\n",
    "\n",
    "    gene_tissue_df = gene_tissue_df.loc[:,~((gene_tissue_df.isnull()).all(axis=0))] #if any tissues were not processed\n",
    "    \n",
    "    return gene_tissue_df\n",
    "\n",
    "\n",
    "#######################################\n",
    "def at_least_one_not_cho(tissue_df,cho_df):\n",
    "    '''Parameters:\n",
    "        tissue_df: The gene-by-tissue expression matrix. Sample names should have tissue in name\n",
    "        tissues: List of tissues to consider.\n",
    "        cho_df: Same rows, but columns are the cho samples.\n",
    "        A tissue could have more than one sample, and if at least one is present then it is ok.'''\n",
    "    \n",
    "    in_cho = cho_df[((cho_df>0).any(axis=1))].index.values\n",
    "    not_in_cho = tissue_df[((tissue_df>0).any(axis=1)) & ((cho_df==0).all(axis=1))].index.values\n",
    "    \n",
    "    return in_cho, not_in_cho\n",
    "\n",
    "\n",
    "#######################################\n",
    "def all_but_cho(tissue_df,cho_df):\n",
    "    '''Parameters:\n",
    "        tissue_df: The gene-by-tissue expression matrix. Sample names should have tissue in name\n",
    "        tissues: List of tissues to consider.\n",
    "        cho_df: Same rows, but columns are the cho samples.\n",
    "        A tissue could have more than one sample, and if at least one is present then it is ok.'''\n",
    "    \n",
    "    in_cho = cho_df[((cho_df>0).any(axis=1))].index.values\n",
    "    not_in_cho = tissue_df[((tissue_df>0).all(axis=1)) & ((cho_df==0).all(axis=1))].index.values\n",
    "    \n",
    "    return in_cho, not_in_cho\n",
    "\n",
    "\n",
    "#######################################\n",
    "def half_and_not_cho(tissue_df,cho_df):\n",
    "    '''Parameters:\n",
    "        tissue_df: The genexsample expression matrix. \n",
    "        A tissue could have more than one sample, and if at least one is present then it is ok.\n",
    "    '''\n",
    "    \n",
    "    fifty = len(tissues)/2\n",
    "    in_cho = cho_df[((cho_df>0).any(axis=1))].index.values\n",
    "    not_in_cho = tissue_df[(np.sum(tissue_df>0,axis=1)>fifty) & ((cho_df==0).all(axis=1))].index.values\n",
    "    \n",
    "    return in_cho, not_in_cho\n",
    "\n",
    "\n",
    "\n",
    "#######################################\n",
    "def count_n_all_but_cho(tissue_df,cho_df,n=10):\n",
    "    '''Parameters:\n",
    "        tissue_df: The genexsample expression matrix. \n",
    "        A tissue could have more than one sample, and if at least one is present then it is ok.\n",
    "    '''\n",
    "    \n",
    "    in_cho = cho_df[((cho_df>n).any(axis=1))].index.values\n",
    "    not_in_cho = tissue_df[(np.sum(tissue_df>n,axis=1)==tissue_df.shape[1]) & ((cho_df<=n).all(axis=1))].index.values\n",
    "    \n",
    "    return in_cho, not_in_cho\n",
    "\n",
    "#######################################\n",
    "def count_ten_half_and_not_cho(tissue_df,cho_df,n=10):\n",
    "    '''Parameters:\n",
    "        tissue_df: The genexsample expression matrix. \n",
    "        A tissue could have more than one sample, and if at least one is present then it is ok.\n",
    "    '''\n",
    "    \n",
    "    fifty = len(tissues)/2\n",
    "    in_cho = cho_df[((cho_df>n).any(axis=1))].index.values\n",
    "    not_in_cho = tissue_df[(np.sum(tissue_df>n,axis=1) >fifty) & ((cho_df <= 0).all(axis=1))].index.values\n",
    "    \n",
    "    return in_cho, not_in_cho"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene_tissue_df = merge_tissues(gene_tissue_matrix,tissues)\n",
    "cho = gene_tissue_df.loc[:,'CHO']\n",
    "cho = pd.DataFrame(cho)\n",
    "gene_tissue_df = gene_tissue_df.drop('CHO',axis=1) #Remove CHO columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "7175\n",
      "4744\n"
     ]
    }
   ],
   "source": [
    "in_cho1, not_in_cho1 = at_least_one_not_cho(gene_tissue_df,cho)\n",
    "print('1')\n",
    "print(len(in_cho1))\n",
    "print(len(not_in_cho1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n",
      "7175\n",
      "42\n"
     ]
    }
   ],
   "source": [
    "in_cho2, not_in_cho2 = all_but_cho(gene_tissue_df,cho)\n",
    "print('2')\n",
    "print(len(in_cho2))\n",
    "print(len(not_in_cho2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n",
      "7175\n",
      "318\n"
     ]
    }
   ],
   "source": [
    "in_cho3, not_in_cho3 = half_and_not_cho(gene_tissue_df,cho)\n",
    "print('3')\n",
    "print(len(in_cho3))\n",
    "print(len(not_in_cho3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4\n",
      "6834\n",
      "44\n"
     ]
    }
   ],
   "source": [
    "in_cho4, not_in_cho4 = count_n_all_but_cho(gene_tissue_df,cho)\n",
    "print('4')\n",
    "print(len(in_cho4))\n",
    "print(len(not_in_cho4))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5\n",
      "6834\n",
      "304\n"
     ]
    }
   ],
   "source": [
    "in_cho5, not_in_cho5 = count_ten_half_and_not_cho(gene_tissue_df,cho)\n",
    "print('5')\n",
    "print(len(in_cho5))\n",
    "print(len(not_in_cho5))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Apoptotic genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import xml.etree.ElementTree as et"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['100682525', 'cge']"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'100682525 cge'.split()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## Load in apoptotic genes\n",
    "apop1_f  = '/data/isshamie/CH_tissue_TSS/Supplemental_Data/gene_sets/cge04210.xml'\n",
    "apop2_f = '/data/isshamie/CH_tissue_TSS/Supplemental_Data/gene_sets/cge04215.xml'\n",
    "\n",
    "apop1 = set()\n",
    "tree = et.parse(apop1_f)\n",
    "root = tree.getroot()\n",
    "for child in root:\n",
    "    #print child.tag, child.attrib\n",
    "    if 'name' in child.attrib and 'cge:' in child.attrib['name'] : \n",
    "        apop1.add(child.attrib['name'].split(':')[1].split()[0]) #Take the first cge:number \n",
    "        \n",
    "apop2 = set()\n",
    "tree = et.parse(apop2_f)\n",
    "root = tree.getroot()\n",
    "for child in root:\n",
    "    #print child.tag, child.attrib\n",
    "    if 'name' in child.attrib and 'cge:' in child.attrib['name']: \n",
    "        apop2.add(child.attrib['name'].split(':')[1].split()[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "entry {'type': 'map', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge04115', 'id': '1', 'name': 'path:cge04115'}\n",
      "entry {'type': 'map', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge04668', 'id': '2', 'name': 'path:cge04668'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689371', 'id': '3', 'name': 'cge:100689371'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100762210', 'id': '4', 'name': 'cge:100762210'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100754226', 'id': '5', 'name': 'cge:100754226'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100682525', 'id': '6', 'name': 'cge:100682525'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100770499', 'id': '7', 'name': 'cge:100770499'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100771367', 'id': '8', 'name': 'cge:100771367'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100774873', 'id': '9', 'name': 'cge:100774873'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689366+cge:100689374+cge:100756596+cge:100761700+cge:100769162', 'id': '10', 'name': 'cge:100689366 cge:100689374 cge:100756596 cge:100761700 cge:100769162'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100768492', 'id': '11', 'name': 'cge:100768492'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689370+cge:103160479', 'id': '12', 'name': 'cge:100689370 cge:103160479'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100766857', 'id': '13', 'name': 'cge:100766857'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689061', 'id': '14', 'name': 'cge:100689061'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100760426', 'id': '15', 'name': 'cge:100760426'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100763542+cge:100770607+cge:107976968', 'id': '16', 'name': 'cge:100763542 cge:100770607 cge:107976968'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689366+cge:100689374+cge:100756596+cge:100761700+cge:100769162', 'id': '17', 'name': 'cge:100689366 cge:100689374 cge:100756596 cge:100761700 cge:100769162'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100766458', 'id': '18', 'name': 'cge:100766458'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100754153', 'id': '19', 'name': 'cge:100754153'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100758500', 'id': '20', 'name': 'cge:100758500'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100754043+cge:100757098+cge:100764460+cge:100768478+cge:100773626+cge:103160806+cge:103164112+cge:107977792', 'id': '21', 'name': 'cge:100754043 cge:100757098 cge:100764460 cge:100768478 cge:100773626 cge:103160806 cge:103164112 cge:107977792'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100754606', 'id': '22', 'name': 'cge:100754606'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689369', 'id': '23', 'name': 'cge:100689369'}\n",
      "entry {'type': 'ortholog', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?K04400', 'id': '24', 'name': 'ko:K04400'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100758636+cge:100771239', 'id': '25', 'name': 'cge:100758636 cge:100771239'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100752454+cge:100768351+cge:100769460', 'id': '26', 'name': 'cge:100752454 cge:100768351 cge:100769460'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100766459', 'id': '27', 'name': 'cge:100766459'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100750488+cge:100754432+cge:100759762', 'id': '28', 'name': 'cge:100750488 cge:100754432 cge:100759762'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100757136', 'id': '29', 'name': 'cge:100757136'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689251+cge:100689252+cge:100750761+cge:100769225+cge:100771181+cge:100774067', 'id': '30', 'name': 'cge:100689251 cge:100689252 cge:100750761 cge:100769225 cge:100771181 cge:100774067'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100762099', 'id': '31', 'name': 'cge:100762099'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100761116', 'id': '32', 'name': 'cge:100761116'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100758679', 'id': '33', 'name': 'cge:100758679'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100774599', 'id': '34', 'name': 'cge:100774599'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100758679', 'id': '35', 'name': 'cge:100758679'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100774599', 'id': '36', 'name': 'cge:100774599'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100764512+cge:103158748', 'id': '37', 'name': 'cge:100764512 cge:103158748'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100751271', 'id': '38', 'name': 'cge:100751271'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100765017', 'id': '39', 'name': 'cge:100765017'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100770489', 'id': '40', 'name': 'cge:100770489'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100762537', 'id': '41', 'name': 'cge:100762537'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:103161604', 'id': '42', 'name': 'cge:103161604'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100750778', 'id': '43', 'name': 'cge:100750778'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100764780', 'id': '44', 'name': 'cge:100764780'}\n",
      "entry {'type': 'map', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge04210', 'id': '45', 'name': 'path:cge04210'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100754968', 'id': '46', 'name': 'cge:100754968'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100773293', 'id': '47', 'name': 'cge:100773293'}\n",
      "entry {'type': 'map', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge04064', 'id': '48', 'name': 'path:cge04064'}\n",
      "entry {'type': 'map', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge04151', 'id': '49', 'name': 'path:cge04151'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100763248+cge:100771374', 'id': '50', 'name': 'cge:100763248 cge:100771374'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100757316+cge:100766421+cge:103161362', 'id': '51', 'name': 'cge:100757316 cge:100766421 cge:103161362'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689463+cge:100753655+cge:100764843+cge:100765231', 'id': '52', 'name': 'cge:100689463 cge:100753655 cge:100764843 cge:100765231'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100754770', 'id': '53', 'name': 'cge:100754770'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100753932', 'id': '54', 'name': 'cge:100753932'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100760629', 'id': '55', 'name': 'cge:100760629'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689032', 'id': '56', 'name': 'cge:100689032'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689368', 'id': '57', 'name': 'cge:100689368'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100754606', 'id': '58', 'name': 'cge:100754606'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100755393', 'id': '59', 'name': 'cge:100755393'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:103164214', 'id': '60', 'name': 'cge:103164214'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100758500', 'id': '61', 'name': 'cge:100758500'}\n",
      "entry {'type': 'ortholog', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?K13773', 'id': '62', 'name': 'ko:K13773'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100762537', 'id': '63', 'name': 'cge:100762537'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100770489', 'id': '64', 'name': 'cge:100770489'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100762099', 'id': '65', 'name': 'cge:100762099'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100750680', 'id': '66', 'name': 'cge:100750680'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100759746', 'id': '67', 'name': 'cge:100759746'}\n",
      "entry {'type': 'map', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge04141', 'id': '68', 'name': 'path:cge04141'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100758903', 'id': '69', 'name': 'cge:100758903'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100757136', 'id': '70', 'name': 'cge:100757136'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100759228+cge:100766127+cge:103159266', 'id': '71', 'name': 'cge:100759228 cge:100766127 cge:103159266'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100758620', 'id': '72', 'name': 'cge:100758620'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100756802+cge:100762099', 'id': '73', 'name': 'cge:100756802 cge:100762099'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689032', 'id': '74', 'name': 'cge:100689032'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689368', 'id': '75', 'name': 'cge:100689368'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689371', 'id': '76', 'name': 'cge:100689371'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100774618', 'id': '77', 'name': 'cge:100774618'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100752376', 'id': '78', 'name': 'cge:100752376'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100754361+cge:100764964+cge:100771734', 'id': '79', 'name': 'cge:100754361 cge:100764964 cge:100771734'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100752536', 'id': '80', 'name': 'cge:100752536'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100766306', 'id': '81', 'name': 'cge:100766306'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689418', 'id': '82', 'name': 'cge:100689418'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100682525', 'id': '83', 'name': 'cge:100682525'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100762537', 'id': '84', 'name': 'cge:100762537'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100773293', 'id': '85', 'name': 'cge:100773293'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100771754', 'id': '86', 'name': 'cge:100771754'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:103164214', 'id': '87', 'name': 'cge:103164214'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100755393', 'id': '88', 'name': 'cge:100755393'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689392+cge:100751823+cge:100756494+cge:100756691+cge:100756791+cge:100756958+cge:100756995+cge:100757077+cge:100765210+cge:100765299+cge:100765407+cge:100765592+cge:100766062+cge:100766628+cge:100767305+cge:100767833+cge:100768693+cge:100769144+cge:100772033+cge:100772318+cge:100772611', 'id': '89', 'name': 'cge:100689392 cge:100751823 cge:100756494 cge:100756691 cge:100756791 cge:100756958 cge:100756995 cge:100757077 cge:100765210 cge:100765299 cge:100765407 cge:100765592 cge:100766062 cge:100766628 cge:100767305 cge:100767833 cge:100768693 cge:100769144 cge:100772033 cge:100772318 cge:100772611'}\n",
      "entry {'type': 'compound', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?C00076', 'id': '90', 'name': 'cpd:C00076'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100763514', 'id': '91', 'name': 'cge:100763514'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100765343', 'id': '92', 'name': 'cge:100765343'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100761915', 'id': '93', 'name': 'cge:100761915'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689386', 'id': '94', 'name': 'cge:100689386'}\n",
      "entry {'type': 'map', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge04115', 'id': '95', 'name': 'path:cge04115'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100751775+cge:100758965+cge:100764450', 'id': '96', 'name': 'cge:100751775 cge:100758965 cge:100764450'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100772477', 'id': '97', 'name': 'cge:100772477'}\n",
      "entry {'type': 'compound', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?C05981', 'id': '98', 'name': 'cpd:C05981'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100754606', 'id': '99', 'name': 'cge:100754606'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689445', 'id': '100', 'name': 'cge:100689445'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100753147+cge:107978008', 'id': '101', 'name': 'cge:100753147 cge:107978008'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100753147+cge:107978008', 'id': '102', 'name': 'cge:100753147 cge:107978008'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100751926+cge:100754015+cge:100754310+cge:100754595+cge:100754885+cge:100755174+cge:100755462+cge:100765811', 'id': '103', 'name': 'cge:100751926 cge:100754015 cge:100754310 cge:100754595 cge:100754885 cge:100755174 cge:100755462 cge:100765811'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100761729', 'id': '104', 'name': 'cge:100761729'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689477+cge:100764022+cge:103159101', 'id': '105', 'name': 'cge:100689477 cge:100764022 cge:103159101'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100682526+cge:100682527+cge:100682528+cge:100753526+cge:100755907+cge:100761720+cge:100762888+cge:100765599+cge:100767630+cge:100767686+cge:107977167+cge:107978406+cge:107978411', 'id': '106', 'name': 'cge:100682526 cge:100682527 cge:100682528 cge:100753526 cge:100755907 cge:100761720 cge:100762888 cge:100765599 cge:100767630 cge:100767686 cge:107977167 cge:107978406 cge:107978411'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100764237', 'id': '107', 'name': 'cge:100764237'}\n",
      "entry {'type': 'map', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge04060', 'id': '108', 'name': 'path:cge04060'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100771754', 'id': '109', 'name': 'cge:100771754'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100764237', 'id': '110', 'name': 'cge:100764237'}\n",
      "entry {'type': 'compound', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?C00319', 'id': '111', 'name': 'cpd:C00319'}\n",
      "entry {'type': 'compound', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?C00027', 'id': '113', 'name': 'cpd:C00027'}\n",
      "entry {'type': 'ortholog', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?K02512', 'id': '114', 'name': 'ko:K02512'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100753790+cge:100757614+cge:100763178+cge:100773737+cge:107977156+cge:107978809', 'id': '115', 'name': 'cge:100753790 cge:100757614 cge:100763178 cge:100773737 cge:107977156 cge:107978809'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100769018', 'id': '116', 'name': 'cge:100769018'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100689403+cge:100774204', 'id': '117', 'name': 'cge:100689403 cge:100774204'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100753149+cge:100759514', 'id': '118', 'name': 'cge:100753149 cge:100759514'}\n",
      "entry {'type': 'gene', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge:100774873', 'id': '119', 'name': 'cge:100774873'}\n",
      "entry {'type': 'map', 'link': 'http://www.kegg.jp/dbget-bin/www_bget?cge04010', 'id': '120', 'name': 'path:cge04010'}\n",
      "entry {'type': 'group', 'id': '121', 'name': 'undefined'}\n",
      "entry {'type': 'group', 'id': '122', 'name': 'undefined'}\n",
      "entry {'type': 'group', 'id': '123', 'name': 'undefined'}\n",
      "entry {'type': 'group', 'id': '124', 'name': 'undefined'}\n",
      "relation {'entry2': '40', 'entry1': '46', 'type': 'PPrel'}\n",
      "relation {'entry2': '36', 'entry1': '40', 'type': 'PPrel'}\n",
      "relation {'entry2': '41', 'entry1': '47', 'type': 'PPrel'}\n",
      "relation {'entry2': '36', 'entry1': '41', 'type': 'PPrel'}\n",
      "relation {'entry2': '23', 'entry1': '36', 'type': 'PPrel'}\n",
      "relation {'entry2': '77', 'entry1': '41', 'type': 'PPrel'}\n",
      "relation {'entry2': '78', 'entry1': '77', 'type': 'PPrel'}\n",
      "relation {'entry2': '39', 'entry1': '44', 'type': 'PPrel'}\n",
      "relation {'entry2': '121', 'entry1': '39', 'type': 'PPrel'}\n",
      "relation {'entry2': '122', 'entry1': '39', 'type': 'PPrel'}\n",
      "relation {'entry2': '80', 'entry1': '31', 'type': 'PPrel'}\n",
      "relation {'entry2': '78', 'entry1': '80', 'type': 'PPrel'}\n",
      "relation {'entry2': '27', 'entry1': '31', 'type': 'PPrel'}\n",
      "relation {'entry2': '79', 'entry1': '78', 'type': 'PPrel'}\n",
      "relation {'entry2': '93', 'entry1': '92', 'type': 'PPrel'}\n",
      "relation {'entry2': '94', 'entry1': '93', 'type': 'PPrel'}\n",
      "relation {'entry2': '78', 'entry1': '123', 'type': 'PPrel'}\n",
      "relation {'entry2': '25', 'entry1': '90', 'type': 'PCrel'}\n",
      "relation {'entry2': '67', 'entry1': '25', 'type': 'PPrel'}\n",
      "relation {'entry2': '23', 'entry1': '29', 'type': 'PPrel'}\n",
      "relation {'entry2': '91', 'entry1': '94', 'type': 'GErel'}\n",
      "relation {'entry2': '76', 'entry1': '91', 'type': 'PPrel'}\n",
      "relation {'entry2': '9', 'entry1': '91', 'type': 'PPrel'}\n",
      "relation {'entry2': '76', 'entry1': '89', 'type': 'PPrel'}\n",
      "relation {'entry2': '9', 'entry1': '89', 'type': 'PPrel'}\n",
      "relation {'entry2': '109', 'entry1': '110', 'type': 'PPrel'}\n",
      "relation {'entry2': '76', 'entry1': '109', 'type': 'PPrel'}\n",
      "relation {'entry2': '9', 'entry1': '109', 'type': 'PPrel'}\n",
      "relation {'entry2': '76', 'entry1': '79', 'type': 'PPrel'}\n",
      "relation {'entry2': '9', 'entry1': '79', 'type': 'PPrel'}\n",
      "relation {'entry2': '99', 'entry1': '79', 'type': 'PPrel'}\n",
      "relation {'entry2': '15', 'entry1': '79', 'type': 'PPrel'}\n",
      "relation {'entry2': '28', 'entry1': '27', 'type': 'PPrel'}\n",
      "relation {'entry2': '19', 'entry1': '28', 'type': 'PPrel'}\n",
      "relation {'entry2': '16', 'entry1': '19', 'type': 'PPrel'}\n",
      "relation {'entry2': '38', 'entry1': '43', 'type': 'PPrel'}\n",
      "relation {'entry2': '37', 'entry1': '42', 'type': 'PPrel'}\n",
      "relation {'entry2': '115', 'entry1': '37', 'type': 'PPrel'}\n",
      "relation {'entry2': '116', 'entry1': '115', 'type': 'PPrel'}\n",
      "relation {'entry2': '98', 'entry1': '30', 'type': 'PCrel'}\n",
      "relation {'entry2': '97', 'entry1': '98', 'type': 'PCrel'}\n",
      "relation {'entry2': '26', 'entry1': '97', 'type': 'PPrel'}\n",
      "relation {'entry2': '117', 'entry1': '116', 'type': 'PPrel'}\n",
      "relation {'entry2': '118', 'entry1': '117', 'type': 'PPrel'}\n",
      "relation {'entry2': '30', 'entry1': '38', 'type': 'PPrel'}\n",
      "relation {'entry2': '115', 'entry1': '38', 'type': 'PPrel'}\n",
      "relation {'entry2': '30', 'entry1': '37', 'type': 'PPrel'}\n",
      "relation {'entry2': '15', 'entry1': '26', 'type': 'PPrel'}\n",
      "relation {'entry2': '28', 'entry1': '26', 'type': 'PPrel'}\n",
      "relation {'entry2': '69', 'entry1': '16', 'type': 'GErel'}\n",
      "relation {'entry2': '70', 'entry1': '16', 'type': 'GErel'}\n",
      "relation {'entry2': '10', 'entry1': '16', 'type': 'GErel'}\n",
      "relation {'entry2': '71', 'entry1': '16', 'type': 'GErel'}\n",
      "relation {'entry2': '73', 'entry1': '16', 'type': 'GErel'}\n",
      "relation {'entry2': '3', 'entry1': '16', 'type': 'GErel'}\n",
      "relation {'entry2': '72', 'entry1': '16', 'type': 'GErel'}\n",
      "relation {'entry2': '119', 'entry1': '118', 'type': 'GErel'}\n",
      "relation {'entry2': '81', 'entry1': '79', 'type': 'PPrel'}\n",
      "relation {'entry2': '82', 'entry1': '79', 'type': 'PPrel'}\n",
      "relation {'entry2': '83', 'entry1': '81', 'type': 'GErel'}\n",
      "relation {'entry2': '84', 'entry1': '81', 'type': 'GErel'}\n",
      "relation {'entry2': '85', 'entry1': '81', 'type': 'GErel'}\n",
      "relation {'entry2': '86', 'entry1': '81', 'type': 'GErel'}\n",
      "relation {'entry2': '83', 'entry1': '82', 'type': 'GErel'}\n",
      "relation {'entry2': '84', 'entry1': '82', 'type': 'GErel'}\n",
      "relation {'entry2': '85', 'entry1': '82', 'type': 'GErel'}\n",
      "relation {'entry2': '86', 'entry1': '82', 'type': 'GErel'}\n",
      "relation {'entry2': '9', 'entry1': '88', 'type': 'PPrel'}\n",
      "relation {'entry2': '76', 'entry1': '88', 'type': 'PPrel'}\n",
      "relation {'entry2': '9', 'entry1': '87', 'type': 'PPrel'}\n",
      "relation {'entry2': '76', 'entry1': '87', 'type': 'PPrel'}\n",
      "relation {'entry2': '74', 'entry1': '76', 'type': 'PPrel'}\n",
      "relation {'entry2': '75', 'entry1': '76', 'type': 'PPrel'}\n",
      "relation {'entry2': '74', 'entry1': '9', 'type': 'PPrel'}\n",
      "relation {'entry2': '75', 'entry1': '9', 'type': 'PPrel'}\n",
      "relation {'entry2': '22', 'entry1': '124', 'type': 'PPrel'}\n",
      "relation {'entry2': '17', 'entry1': '53', 'type': 'PPrel'}\n",
      "relation {'entry2': '17', 'entry1': '54', 'type': 'PPrel'}\n",
      "relation {'entry2': '17', 'entry1': '55', 'type': 'PPrel'}\n",
      "relation {'entry2': '17', 'entry1': '89', 'type': 'PPrel'}\n",
      "relation {'entry2': '12', 'entry1': '17', 'type': 'PPrel'}\n",
      "relation {'entry2': '22', 'entry1': '89', 'type': 'PPrel'}\n",
      "relation {'entry2': '12', 'entry1': '21', 'type': 'PPrel'}\n",
      "relation {'entry2': '12', 'entry1': '20', 'type': 'PPrel'}\n",
      "relation {'entry2': '20', 'entry1': '21', 'type': 'PPrel'}\n",
      "relation {'entry2': '6', 'entry1': '5', 'type': 'PPrel'}\n",
      "relation {'entry2': '57', 'entry1': '6', 'type': 'GErel'}\n",
      "relation {'entry2': '56', 'entry1': '6', 'type': 'GErel'}\n",
      "relation {'entry2': '102', 'entry1': '6', 'type': 'GErel'}\n",
      "relation {'entry2': '58', 'entry1': '6', 'type': 'GErel'}\n",
      "relation {'entry2': '61', 'entry1': '6', 'type': 'GErel'}\n",
      "relation {'entry2': '63', 'entry1': '6', 'type': 'GErel'}\n",
      "relation {'entry2': '64', 'entry1': '6', 'type': 'GErel'}\n",
      "relation {'entry2': '60', 'entry1': '6', 'type': 'GErel'}\n",
      "relation {'entry2': '59', 'entry1': '6', 'type': 'GErel'}\n",
      "relation {'entry2': '9', 'entry1': '15', 'type': 'PPrel'}\n",
      "relation {'entry2': '23', 'entry1': '121', 'type': 'PPrel'}\n",
      "relation {'entry2': '106', 'entry1': '103', 'type': 'PPrel'}\n",
      "relation {'entry2': '107', 'entry1': '103', 'type': 'PPrel'}\n",
      "relation {'entry2': '51', 'entry1': '103', 'type': 'PPrel'}\n",
      "relation {'entry2': '52', 'entry1': '103', 'type': 'PPrel'}\n",
      "relation {'entry2': '4', 'entry1': '103', 'type': 'PPrel'}\n",
      "relation {'entry2': '50', 'entry1': '14', 'type': 'PPrel'}\n",
      "relation {'entry2': '51', 'entry1': '18', 'type': 'PPrel'}\n",
      "relation {'entry2': '52', 'entry1': '14', 'type': 'PPrel'}\n",
      "relation {'entry2': '52', 'entry1': '13', 'type': 'PPrel'}\n",
      "relation {'entry2': '52', 'entry1': '12', 'type': 'PPrel'}\n",
      "relation {'entry2': '4', 'entry1': '14', 'type': 'PPrel'}\n",
      "relation {'entry2': '105', 'entry1': '18', 'type': 'PPrel'}\n",
      "relation {'entry2': '105', 'entry1': '14', 'type': 'PPrel'}\n",
      "relation {'entry2': '105', 'entry1': '13', 'type': 'PPrel'}\n",
      "relation {'entry2': '14', 'entry1': '103', 'type': 'PPrel'}\n",
      "relation {'entry2': '13', 'entry1': '103', 'type': 'PPrel'}\n",
      "relation {'entry2': '23', 'entry1': '103', 'type': 'PPrel'}\n",
      "relation {'entry2': '14', 'entry1': '67', 'type': 'PPrel'}\n",
      "relation {'entry2': '13', 'entry1': '67', 'type': 'PPrel'}\n",
      "relation {'entry2': '18', 'entry1': '23', 'type': 'PPrel'}\n",
      "relation {'entry2': '14', 'entry1': '23', 'type': 'PPrel'}\n",
      "relation {'entry2': '13', 'entry1': '23', 'type': 'PPrel'}\n",
      "relation {'entry2': '18', 'entry1': '12', 'type': 'PPrel'}\n",
      "relation {'entry2': '14', 'entry1': '12', 'type': 'PPrel'}\n",
      "relation {'entry2': '13', 'entry1': '12', 'type': 'PPrel'}\n",
      "relation {'entry2': '14', 'entry1': '17', 'type': 'PPrel'}\n",
      "relation {'entry2': '13', 'entry1': '17', 'type': 'PPrel'}\n",
      "relation {'entry2': '22', 'entry1': '103', 'type': 'PPrel'}\n",
      "relation {'entry2': '74', 'entry1': '109', 'type': 'PPrel'}\n"
     ]
    }
   ],
   "source": [
    "tree = et.parse(apop1_f)\n",
    "root = tree.getroot()\n",
    "for child in root:\n",
    "    print child.tag, child.attrib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22\n",
      "81\n"
     ]
    }
   ],
   "source": [
    "print(len(apop2))\n",
    "print(len(apop1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'100689032',\n",
       " '100689061',\n",
       " '100689366',\n",
       " '100689368',\n",
       " '100689369',\n",
       " '100689370',\n",
       " '100689371',\n",
       " '100753932',\n",
       " '100754043',\n",
       " '100754361',\n",
       " '100754606',\n",
       " '100754770',\n",
       " '100758500',\n",
       " '100760629',\n",
       " '100765017',\n",
       " '100766857',\n",
       " '100774599'}"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "apop2.intersection(apop1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Convert from id to symbol"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneID</th>\n",
       "      <th>Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1246500</td>\n",
       "      <td>repA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1246501</td>\n",
       "      <td>repA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1246502</td>\n",
       "      <td>leuA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1246503</td>\n",
       "      <td>leuB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1246504</td>\n",
       "      <td>leuC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    GeneID Symbol\n",
       "0  1246500  repA1\n",
       "1  1246501  repA2\n",
       "2  1246502   leuA\n",
       "3  1246503   leuB\n",
       "4  1246504   leuC"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene2refseq = pd.read_csv('/data/isshamie/CH_tissue_TSS/Supplemental_Data/gene2refseq',sep='\\t',header=0,\n",
    "                         usecols=['GeneID','Symbol'])\n",
    "gene2refseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5122505, 2)"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene2refseq.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100751775\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689386\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100768492\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100759228\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100750778\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100762099\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100762210\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100682526\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100771367\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100758620\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100774599\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100763514\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100753147\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100772477\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689403\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100754226\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100770499\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100760629\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100765017\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100754153\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100682525\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "103161604\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100754361\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100761729\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100774618\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100763248\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100765343\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689392\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689366\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100760426\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100764512\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100756802\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100753149\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100764237\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100769018\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100751926\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100755393\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100766458\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100766459\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100758903\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100773293\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100752376\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689368\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100759746\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100750680\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100771754\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689369\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689418\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100757316\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100750488\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100758636\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689061\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100758500\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100753932\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100758679\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100754968\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100761116\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100766857\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100770489\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100754043\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100754770\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689251\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100757136\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100754606\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100763542\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100762537\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100764780\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100751271\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100766306\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100753790\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689463\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100761915\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100774873\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689477\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689371\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689370\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689445\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100752536\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100752454\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "100689032\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n",
      "103164214\n",
      "Empty DataFrame\n",
      "Columns: [GeneID, Symbol]\n",
      "Index: []\n"
     ]
    }
   ],
   "source": [
    "apop1_symbol = set()\n",
    "for i in apop1:\n",
    "    print(i)\n",
    "    print(gene2refseq[gene2refseq['GeneID'] == int(i)])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get list of SecM genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensgid</th>\n",
       "      <th>Subsystem</th>\n",
       "      <th>category.HPA</th>\n",
       "      <th>tissues.HPA</th>\n",
       "      <th>category.GTEx</th>\n",
       "      <th>tissues.GTEx</th>\n",
       "      <th>fpkm.HPA.adrenal</th>\n",
       "      <th>fpkm.HPA.appendices</th>\n",
       "      <th>fpkm.HPA.bonemarrow</th>\n",
       "      <th>fpkm.HPA.brain</th>\n",
       "      <th>...</th>\n",
       "      <th>fpkm.GTEx.prostate</th>\n",
       "      <th>fpkm.GTEx.skeletalmuscle</th>\n",
       "      <th>fpkm.GTEx.skin</th>\n",
       "      <th>fpkm.GTEx.smallintestine</th>\n",
       "      <th>fpkm.GTEx.spleen</th>\n",
       "      <th>fpkm.GTEx.stomach</th>\n",
       "      <th>fpkm.GTEx.testis</th>\n",
       "      <th>fpkm.GTEx.thyroid</th>\n",
       "      <th>fpkm.GTEx.uterus</th>\n",
       "      <th>fpkm.GTEx.vagina</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>RAB44</th>\n",
       "      <td>ENSG00000255587</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>bonemarrow</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>blood</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.49</td>\n",
       "      <td>27.67</td>\n",
       "      <td>0.01</td>\n",
       "      <td>...</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B4GALNT1</th>\n",
       "      <td>ENSG00000135454</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum,nerve</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.03</td>\n",
       "      <td>12.69</td>\n",
       "      <td>...</td>\n",
       "      <td>1.73</td>\n",
       "      <td>2.63</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.83</td>\n",
       "      <td>1.21</td>\n",
       "      <td>1.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NAPB</th>\n",
       "      <td>ENSG00000125814</td>\n",
       "      <td>COPII</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum</td>\n",
       "      <td>21.39</td>\n",
       "      <td>4.80</td>\n",
       "      <td>5.52</td>\n",
       "      <td>156.18</td>\n",
       "      <td>...</td>\n",
       "      <td>5.64</td>\n",
       "      <td>2.05</td>\n",
       "      <td>4.75</td>\n",
       "      <td>5.05</td>\n",
       "      <td>5.22</td>\n",
       "      <td>3.72</td>\n",
       "      <td>7.49</td>\n",
       "      <td>6.72</td>\n",
       "      <td>6.05</td>\n",
       "      <td>6.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NSF</th>\n",
       "      <td>ENSG00000073969</td>\n",
       "      <td>COPII</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14.46</td>\n",
       "      <td>13.96</td>\n",
       "      <td>4.66</td>\n",
       "      <td>117.52</td>\n",
       "      <td>...</td>\n",
       "      <td>9.25</td>\n",
       "      <td>11.03</td>\n",
       "      <td>9.75</td>\n",
       "      <td>9.38</td>\n",
       "      <td>9.92</td>\n",
       "      <td>8.49</td>\n",
       "      <td>14.16</td>\n",
       "      <td>8.88</td>\n",
       "      <td>6.93</td>\n",
       "      <td>7.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AGAP2</th>\n",
       "      <td>ENSG00000135439</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum</td>\n",
       "      <td>1.47</td>\n",
       "      <td>7.62</td>\n",
       "      <td>7.22</td>\n",
       "      <td>65.72</td>\n",
       "      <td>...</td>\n",
       "      <td>1.59</td>\n",
       "      <td>3.04</td>\n",
       "      <td>1.03</td>\n",
       "      <td>5.51</td>\n",
       "      <td>9.41</td>\n",
       "      <td>1.36</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.96</td>\n",
       "      <td>1.05</td>\n",
       "      <td>0.96</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ensgid               Subsystem     category.HPA  \\\n",
       "gene_name                                                             \n",
       "RAB44      ENSG00000255587  Post-Golgi trafficking  Tissue enriched   \n",
       "B4GALNT1   ENSG00000135454     Golgi glycosylation  Tissue enriched   \n",
       "NAPB       ENSG00000125814                   COPII  Tissue enriched   \n",
       "NSF        ENSG00000073969                   COPII  Tissue enriched   \n",
       "AGAP2      ENSG00000135439  Trafficking regulation  Tissue enriched   \n",
       "\n",
       "          tissues.HPA     category.GTEx  \\\n",
       "gene_name                                 \n",
       "RAB44      bonemarrow          Enhanced   \n",
       "B4GALNT1        brain    Group enriched   \n",
       "NAPB            brain    Group enriched   \n",
       "NSF             brain  Expressed in all   \n",
       "AGAP2           brain    Group enriched   \n",
       "\n",
       "                                    tissues.GTEx  fpkm.HPA.adrenal  \\\n",
       "gene_name                                                            \n",
       "RAB44                                      blood              0.04   \n",
       "B4GALNT1   brain.cerebellum,brain.cerebrum,nerve              0.56   \n",
       "NAPB             brain.cerebellum,brain.cerebrum             21.39   \n",
       "NSF                                          NaN             14.46   \n",
       "AGAP2            brain.cerebellum,brain.cerebrum              1.47   \n",
       "\n",
       "           fpkm.HPA.appendices  fpkm.HPA.bonemarrow  fpkm.HPA.brain  \\\n",
       "gene_name                                                             \n",
       "RAB44                     0.49                27.67            0.01   \n",
       "B4GALNT1                  0.62                 0.03           12.69   \n",
       "NAPB                      4.80                 5.52          156.18   \n",
       "NSF                      13.96                 4.66          117.52   \n",
       "AGAP2                     7.62                 7.22           65.72   \n",
       "\n",
       "                 ...         fpkm.GTEx.prostate  fpkm.GTEx.skeletalmuscle  \\\n",
       "gene_name        ...                                                        \n",
       "RAB44            ...                       0.46                      0.06   \n",
       "B4GALNT1         ...                       1.73                      2.63   \n",
       "NAPB             ...                       5.64                      2.05   \n",
       "NSF              ...                       9.25                     11.03   \n",
       "AGAP2            ...                       1.59                      3.04   \n",
       "\n",
       "           fpkm.GTEx.skin  fpkm.GTEx.smallintestine  fpkm.GTEx.spleen  \\\n",
       "gene_name                                                               \n",
       "RAB44                0.68                      0.56              0.89   \n",
       "B4GALNT1             0.26                      0.95              0.22   \n",
       "NAPB                 4.75                      5.05              5.22   \n",
       "NSF                  9.75                      9.38              9.92   \n",
       "AGAP2                1.03                      5.51              9.41   \n",
       "\n",
       "           fpkm.GTEx.stomach  fpkm.GTEx.testis  fpkm.GTEx.thyroid  \\\n",
       "gene_name                                                           \n",
       "RAB44                   0.37              0.13               0.05   \n",
       "B4GALNT1                0.48              0.56               0.83   \n",
       "NAPB                    3.72              7.49               6.72   \n",
       "NSF                     8.49             14.16               8.88   \n",
       "AGAP2                   1.36              0.95               1.96   \n",
       "\n",
       "           fpkm.GTEx.uterus  fpkm.GTEx.vagina  \n",
       "gene_name                                      \n",
       "RAB44                  0.39              0.54  \n",
       "B4GALNT1               1.21              1.09  \n",
       "NAPB                   6.05              6.50  \n",
       "NSF                    6.93              7.71  \n",
       "AGAP2                  1.05              0.96  \n",
       "\n",
       "[5 rows x 68 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "secM = pd.read_csv('070870-1_SecretoryMachinery.csv',index_col=1,skiprows=4)\n",
    "secM.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### secM silenced in CHO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensgid</th>\n",
       "      <th>Subsystem</th>\n",
       "      <th>category.HPA</th>\n",
       "      <th>tissues.HPA</th>\n",
       "      <th>category.GTEx</th>\n",
       "      <th>tissues.GTEx</th>\n",
       "      <th>fpkm.HPA.adrenal</th>\n",
       "      <th>fpkm.HPA.appendices</th>\n",
       "      <th>fpkm.HPA.bonemarrow</th>\n",
       "      <th>fpkm.HPA.brain</th>\n",
       "      <th>...</th>\n",
       "      <th>fpkm.GTEx.prostate</th>\n",
       "      <th>fpkm.GTEx.skeletalmuscle</th>\n",
       "      <th>fpkm.GTEx.skin</th>\n",
       "      <th>fpkm.GTEx.smallintestine</th>\n",
       "      <th>fpkm.GTEx.spleen</th>\n",
       "      <th>fpkm.GTEx.stomach</th>\n",
       "      <th>fpkm.GTEx.testis</th>\n",
       "      <th>fpkm.GTEx.thyroid</th>\n",
       "      <th>fpkm.GTEx.uterus</th>\n",
       "      <th>fpkm.GTEx.vagina</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>B4GALNT1</th>\n",
       "      <td>ENSG00000135454</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum,nerve</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.03</td>\n",
       "      <td>12.69</td>\n",
       "      <td>...</td>\n",
       "      <td>1.73</td>\n",
       "      <td>2.63</td>\n",
       "      <td>0.26</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.56</td>\n",
       "      <td>0.83</td>\n",
       "      <td>1.21</td>\n",
       "      <td>1.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AGAP2</th>\n",
       "      <td>ENSG00000135439</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum</td>\n",
       "      <td>1.47</td>\n",
       "      <td>7.62</td>\n",
       "      <td>7.22</td>\n",
       "      <td>65.72</td>\n",
       "      <td>...</td>\n",
       "      <td>1.59</td>\n",
       "      <td>3.04</td>\n",
       "      <td>1.03</td>\n",
       "      <td>5.51</td>\n",
       "      <td>9.41</td>\n",
       "      <td>1.36</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.96</td>\n",
       "      <td>1.05</td>\n",
       "      <td>0.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AP3B2</th>\n",
       "      <td>ENSG00000103723</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum,pituitary</td>\n",
       "      <td>2.37</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0.05</td>\n",
       "      <td>17.70</td>\n",
       "      <td>...</td>\n",
       "      <td>1.38</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.74</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.64</td>\n",
       "      <td>4.90</td>\n",
       "      <td>1.22</td>\n",
       "      <td>0.78</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNAJC6</th>\n",
       "      <td>ENSG00000116675</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum,pituitary</td>\n",
       "      <td>10.31</td>\n",
       "      <td>0.83</td>\n",
       "      <td>2.53</td>\n",
       "      <td>83.68</td>\n",
       "      <td>...</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.52</td>\n",
       "      <td>0.74</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB6B</th>\n",
       "      <td>ENSG00000154917</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain.cerebrum</td>\n",
       "      <td>9.89</td>\n",
       "      <td>2.38</td>\n",
       "      <td>6.01</td>\n",
       "      <td>106.17</td>\n",
       "      <td>...</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.53</td>\n",
       "      <td>1.27</td>\n",
       "      <td>2.16</td>\n",
       "      <td>1.72</td>\n",
       "      <td>2.63</td>\n",
       "      <td>2.40</td>\n",
       "      <td>1.54</td>\n",
       "      <td>1.07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNAP25</th>\n",
       "      <td>ENSG00000132639</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum</td>\n",
       "      <td>39.99</td>\n",
       "      <td>3.75</td>\n",
       "      <td>0.03</td>\n",
       "      <td>916.53</td>\n",
       "      <td>...</td>\n",
       "      <td>4.32</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.45</td>\n",
       "      <td>0.05</td>\n",
       "      <td>2.23</td>\n",
       "      <td>3.48</td>\n",
       "      <td>0.70</td>\n",
       "      <td>5.51</td>\n",
       "      <td>1.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRYAA</th>\n",
       "      <td>ENSG00000160202</td>\n",
       "      <td>Protein folding</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>kidney</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>kidney</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HSPA4L</th>\n",
       "      <td>ENSG00000164070</td>\n",
       "      <td>ERAD</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>9.51</td>\n",
       "      <td>0.96</td>\n",
       "      <td>1.03</td>\n",
       "      <td>7.31</td>\n",
       "      <td>...</td>\n",
       "      <td>2.17</td>\n",
       "      <td>0.64</td>\n",
       "      <td>5.85</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.44</td>\n",
       "      <td>57.43</td>\n",
       "      <td>1.62</td>\n",
       "      <td>0.87</td>\n",
       "      <td>4.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABO</th>\n",
       "      <td>ENSG00000175164</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>colon</td>\n",
       "      <td>3.85</td>\n",
       "      <td>4.92</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.26</td>\n",
       "      <td>...</td>\n",
       "      <td>2.02</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.74</td>\n",
       "      <td>7.80</td>\n",
       "      <td>0.39</td>\n",
       "      <td>1.29</td>\n",
       "      <td>1.60</td>\n",
       "      <td>7.94</td>\n",
       "      <td>0.76</td>\n",
       "      <td>4.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSD4</th>\n",
       "      <td>ENSG00000125637</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.73</td>\n",
       "      <td>16.61</td>\n",
       "      <td>7.83</td>\n",
       "      <td>1.77</td>\n",
       "      <td>...</td>\n",
       "      <td>17.57</td>\n",
       "      <td>0.20</td>\n",
       "      <td>10.51</td>\n",
       "      <td>24.51</td>\n",
       "      <td>35.05</td>\n",
       "      <td>13.41</td>\n",
       "      <td>4.87</td>\n",
       "      <td>16.69</td>\n",
       "      <td>1.13</td>\n",
       "      <td>4.91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB9B</th>\n",
       "      <td>ENSG00000123570</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain.cerebellum</td>\n",
       "      <td>2.54</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.07</td>\n",
       "      <td>7.19</td>\n",
       "      <td>...</td>\n",
       "      <td>3.74</td>\n",
       "      <td>2.15</td>\n",
       "      <td>0.40</td>\n",
       "      <td>2.08</td>\n",
       "      <td>0.35</td>\n",
       "      <td>1.92</td>\n",
       "      <td>1.23</td>\n",
       "      <td>1.02</td>\n",
       "      <td>4.72</td>\n",
       "      <td>1.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GXYLT2</th>\n",
       "      <td>ENSG00000172986</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.92</td>\n",
       "      <td>4.00</td>\n",
       "      <td>0.90</td>\n",
       "      <td>1.72</td>\n",
       "      <td>...</td>\n",
       "      <td>1.33</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.66</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.62</td>\n",
       "      <td>2.94</td>\n",
       "      <td>2.59</td>\n",
       "      <td>2.11</td>\n",
       "      <td>2.41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB37</th>\n",
       "      <td>ENSG00000172794</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain.cerebellum</td>\n",
       "      <td>1.37</td>\n",
       "      <td>8.81</td>\n",
       "      <td>11.79</td>\n",
       "      <td>1.64</td>\n",
       "      <td>...</td>\n",
       "      <td>3.27</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.65</td>\n",
       "      <td>7.45</td>\n",
       "      <td>11.54</td>\n",
       "      <td>1.91</td>\n",
       "      <td>1.84</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.61</td>\n",
       "      <td>1.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GALNT18</th>\n",
       "      <td>ENSG00000110328</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.15</td>\n",
       "      <td>3.64</td>\n",
       "      <td>0.01</td>\n",
       "      <td>11.48</td>\n",
       "      <td>...</td>\n",
       "      <td>8.27</td>\n",
       "      <td>3.25</td>\n",
       "      <td>7.25</td>\n",
       "      <td>5.03</td>\n",
       "      <td>2.42</td>\n",
       "      <td>3.29</td>\n",
       "      <td>11.72</td>\n",
       "      <td>27.47</td>\n",
       "      <td>14.61</td>\n",
       "      <td>7.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CYTH4</th>\n",
       "      <td>ENSG00000100055</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>blood,spleen</td>\n",
       "      <td>6.46</td>\n",
       "      <td>45.25</td>\n",
       "      <td>21.59</td>\n",
       "      <td>3.70</td>\n",
       "      <td>...</td>\n",
       "      <td>4.42</td>\n",
       "      <td>0.56</td>\n",
       "      <td>2.19</td>\n",
       "      <td>16.17</td>\n",
       "      <td>58.76</td>\n",
       "      <td>3.36</td>\n",
       "      <td>1.03</td>\n",
       "      <td>2.61</td>\n",
       "      <td>2.14</td>\n",
       "      <td>3.09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PGAP1</th>\n",
       "      <td>ENSG00000197121</td>\n",
       "      <td>GPI biosynthesis</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.36</td>\n",
       "      <td>1.15</td>\n",
       "      <td>1.00</td>\n",
       "      <td>7.35</td>\n",
       "      <td>...</td>\n",
       "      <td>2.65</td>\n",
       "      <td>0.59</td>\n",
       "      <td>4.29</td>\n",
       "      <td>2.07</td>\n",
       "      <td>0.52</td>\n",
       "      <td>1.27</td>\n",
       "      <td>1.68</td>\n",
       "      <td>1.75</td>\n",
       "      <td>2.38</td>\n",
       "      <td>2.97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B3GNTL1</th>\n",
       "      <td>ENSG00000175711</td>\n",
       "      <td>ER glycosylation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.18</td>\n",
       "      <td>3.47</td>\n",
       "      <td>4.25</td>\n",
       "      <td>1.30</td>\n",
       "      <td>...</td>\n",
       "      <td>1.69</td>\n",
       "      <td>0.74</td>\n",
       "      <td>1.61</td>\n",
       "      <td>1.35</td>\n",
       "      <td>2.68</td>\n",
       "      <td>0.92</td>\n",
       "      <td>2.51</td>\n",
       "      <td>1.44</td>\n",
       "      <td>2.18</td>\n",
       "      <td>1.85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TBC1D30</th>\n",
       "      <td>ENSG00000111490</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.82</td>\n",
       "      <td>1.95</td>\n",
       "      <td>0.26</td>\n",
       "      <td>6.27</td>\n",
       "      <td>...</td>\n",
       "      <td>1.91</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.78</td>\n",
       "      <td>1.13</td>\n",
       "      <td>1.88</td>\n",
       "      <td>1.21</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GALNT12</th>\n",
       "      <td>ENSG00000119514</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>colon</td>\n",
       "      <td>1.39</td>\n",
       "      <td>6.82</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.61</td>\n",
       "      <td>...</td>\n",
       "      <td>3.39</td>\n",
       "      <td>0.23</td>\n",
       "      <td>2.90</td>\n",
       "      <td>13.76</td>\n",
       "      <td>1.62</td>\n",
       "      <td>19.78</td>\n",
       "      <td>4.01</td>\n",
       "      <td>10.47</td>\n",
       "      <td>1.70</td>\n",
       "      <td>3.63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MYO5B</th>\n",
       "      <td>ENSG00000167306</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.28</td>\n",
       "      <td>7.79</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.93</td>\n",
       "      <td>...</td>\n",
       "      <td>2.90</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7.82</td>\n",
       "      <td>5.99</td>\n",
       "      <td>0.26</td>\n",
       "      <td>3.37</td>\n",
       "      <td>4.36</td>\n",
       "      <td>6.91</td>\n",
       "      <td>0.22</td>\n",
       "      <td>7.19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGAT3</th>\n",
       "      <td>ENSG00000128268</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain.cerebellum</td>\n",
       "      <td>0.60</td>\n",
       "      <td>3.04</td>\n",
       "      <td>3.50</td>\n",
       "      <td>17.42</td>\n",
       "      <td>...</td>\n",
       "      <td>1.08</td>\n",
       "      <td>0.33</td>\n",
       "      <td>1.83</td>\n",
       "      <td>8.07</td>\n",
       "      <td>4.49</td>\n",
       "      <td>3.23</td>\n",
       "      <td>4.67</td>\n",
       "      <td>1.33</td>\n",
       "      <td>3.16</td>\n",
       "      <td>1.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB40B</th>\n",
       "      <td>ENSG00000141542</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.37</td>\n",
       "      <td>1.85</td>\n",
       "      <td>0.42</td>\n",
       "      <td>26.88</td>\n",
       "      <td>...</td>\n",
       "      <td>5.74</td>\n",
       "      <td>7.39</td>\n",
       "      <td>5.88</td>\n",
       "      <td>2.93</td>\n",
       "      <td>2.49</td>\n",
       "      <td>6.62</td>\n",
       "      <td>3.46</td>\n",
       "      <td>3.70</td>\n",
       "      <td>4.54</td>\n",
       "      <td>4.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB3C</th>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>adrenal,brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum,pituitary</td>\n",
       "      <td>8.34</td>\n",
       "      <td>1.18</td>\n",
       "      <td>0.03</td>\n",
       "      <td>29.50</td>\n",
       "      <td>...</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.40</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERP27</th>\n",
       "      <td>ENSG00000139055</td>\n",
       "      <td>Protein folding</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>fallopiantube,pancreas</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>pancreas</td>\n",
       "      <td>0.31</td>\n",
       "      <td>2.92</td>\n",
       "      <td>1.98</td>\n",
       "      <td>0.09</td>\n",
       "      <td>...</td>\n",
       "      <td>1.37</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.79</td>\n",
       "      <td>1.57</td>\n",
       "      <td>2.48</td>\n",
       "      <td>1.62</td>\n",
       "      <td>0.13</td>\n",
       "      <td>4.44</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB2B</th>\n",
       "      <td>ENSG00000129472</td>\n",
       "      <td>COPI</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11.15</td>\n",
       "      <td>7.29</td>\n",
       "      <td>10.85</td>\n",
       "      <td>10.08</td>\n",
       "      <td>...</td>\n",
       "      <td>12.24</td>\n",
       "      <td>2.40</td>\n",
       "      <td>9.00</td>\n",
       "      <td>8.37</td>\n",
       "      <td>9.58</td>\n",
       "      <td>6.77</td>\n",
       "      <td>11.35</td>\n",
       "      <td>14.07</td>\n",
       "      <td>9.26</td>\n",
       "      <td>8.66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CUL1</th>\n",
       "      <td>ENSG00000055130</td>\n",
       "      <td>ERAD</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23.80</td>\n",
       "      <td>27.77</td>\n",
       "      <td>16.25</td>\n",
       "      <td>23.88</td>\n",
       "      <td>...</td>\n",
       "      <td>22.10</td>\n",
       "      <td>31.45</td>\n",
       "      <td>19.44</td>\n",
       "      <td>20.76</td>\n",
       "      <td>20.51</td>\n",
       "      <td>20.28</td>\n",
       "      <td>75.95</td>\n",
       "      <td>20.87</td>\n",
       "      <td>23.60</td>\n",
       "      <td>20.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIGL</th>\n",
       "      <td>ENSG00000108474</td>\n",
       "      <td>GPI biosynthesis</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.95</td>\n",
       "      <td>4.89</td>\n",
       "      <td>4.01</td>\n",
       "      <td>4.53</td>\n",
       "      <td>...</td>\n",
       "      <td>5.26</td>\n",
       "      <td>0.97</td>\n",
       "      <td>6.34</td>\n",
       "      <td>4.04</td>\n",
       "      <td>3.13</td>\n",
       "      <td>4.44</td>\n",
       "      <td>4.12</td>\n",
       "      <td>5.27</td>\n",
       "      <td>4.31</td>\n",
       "      <td>4.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FBXO6</th>\n",
       "      <td>ENSG00000116663</td>\n",
       "      <td>Protein folding</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.61</td>\n",
       "      <td>9.00</td>\n",
       "      <td>7.78</td>\n",
       "      <td>2.91</td>\n",
       "      <td>...</td>\n",
       "      <td>8.20</td>\n",
       "      <td>6.76</td>\n",
       "      <td>6.98</td>\n",
       "      <td>6.33</td>\n",
       "      <td>12.08</td>\n",
       "      <td>7.41</td>\n",
       "      <td>13.69</td>\n",
       "      <td>5.94</td>\n",
       "      <td>4.76</td>\n",
       "      <td>6.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STT3A</th>\n",
       "      <td>ENSG00000134910</td>\n",
       "      <td>ER glycosylation</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.31</td>\n",
       "      <td>43.18</td>\n",
       "      <td>17.18</td>\n",
       "      <td>17.34</td>\n",
       "      <td>...</td>\n",
       "      <td>16.18</td>\n",
       "      <td>4.42</td>\n",
       "      <td>12.01</td>\n",
       "      <td>15.29</td>\n",
       "      <td>17.05</td>\n",
       "      <td>16.05</td>\n",
       "      <td>14.00</td>\n",
       "      <td>21.62</td>\n",
       "      <td>23.59</td>\n",
       "      <td>20.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNAJB12</th>\n",
       "      <td>ENSG00000148719</td>\n",
       "      <td>Protein folding</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17.34</td>\n",
       "      <td>14.97</td>\n",
       "      <td>12.76</td>\n",
       "      <td>12.41</td>\n",
       "      <td>...</td>\n",
       "      <td>12.95</td>\n",
       "      <td>8.43</td>\n",
       "      <td>12.91</td>\n",
       "      <td>9.65</td>\n",
       "      <td>12.26</td>\n",
       "      <td>9.99</td>\n",
       "      <td>14.35</td>\n",
       "      <td>12.38</td>\n",
       "      <td>11.09</td>\n",
       "      <td>9.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEC23A</th>\n",
       "      <td>ENSG00000100934</td>\n",
       "      <td>COPII</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.57</td>\n",
       "      <td>30.10</td>\n",
       "      <td>5.78</td>\n",
       "      <td>18.83</td>\n",
       "      <td>...</td>\n",
       "      <td>10.30</td>\n",
       "      <td>11.11</td>\n",
       "      <td>5.46</td>\n",
       "      <td>13.95</td>\n",
       "      <td>5.56</td>\n",
       "      <td>9.05</td>\n",
       "      <td>6.72</td>\n",
       "      <td>8.95</td>\n",
       "      <td>16.39</td>\n",
       "      <td>12.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HSP90AA1</th>\n",
       "      <td>ENSG00000080824</td>\n",
       "      <td>Protein folding</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>292.57</td>\n",
       "      <td>216.58</td>\n",
       "      <td>182.86</td>\n",
       "      <td>422.36</td>\n",
       "      <td>...</td>\n",
       "      <td>180.49</td>\n",
       "      <td>157.19</td>\n",
       "      <td>136.32</td>\n",
       "      <td>161.76</td>\n",
       "      <td>173.69</td>\n",
       "      <td>172.62</td>\n",
       "      <td>454.43</td>\n",
       "      <td>193.85</td>\n",
       "      <td>171.67</td>\n",
       "      <td>179.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EEA1</th>\n",
       "      <td>ENSG00000102189</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.05</td>\n",
       "      <td>6.61</td>\n",
       "      <td>1.90</td>\n",
       "      <td>6.35</td>\n",
       "      <td>...</td>\n",
       "      <td>2.23</td>\n",
       "      <td>4.22</td>\n",
       "      <td>3.77</td>\n",
       "      <td>1.83</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.85</td>\n",
       "      <td>2.77</td>\n",
       "      <td>2.66</td>\n",
       "      <td>3.65</td>\n",
       "      <td>4.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATF4</th>\n",
       "      <td>ENSG00000128272</td>\n",
       "      <td>UPR</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>176.34</td>\n",
       "      <td>156.45</td>\n",
       "      <td>413.76</td>\n",
       "      <td>88.06</td>\n",
       "      <td>...</td>\n",
       "      <td>232.84</td>\n",
       "      <td>161.37</td>\n",
       "      <td>152.33</td>\n",
       "      <td>174.90</td>\n",
       "      <td>157.97</td>\n",
       "      <td>219.19</td>\n",
       "      <td>95.17</td>\n",
       "      <td>147.49</td>\n",
       "      <td>185.31</td>\n",
       "      <td>162.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EXOC6B</th>\n",
       "      <td>ENSG00000144036</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11.11</td>\n",
       "      <td>3.84</td>\n",
       "      <td>0.59</td>\n",
       "      <td>17.09</td>\n",
       "      <td>...</td>\n",
       "      <td>3.65</td>\n",
       "      <td>2.19</td>\n",
       "      <td>11.03</td>\n",
       "      <td>3.87</td>\n",
       "      <td>1.99</td>\n",
       "      <td>2.87</td>\n",
       "      <td>3.68</td>\n",
       "      <td>3.51</td>\n",
       "      <td>4.82</td>\n",
       "      <td>5.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERLEC1</th>\n",
       "      <td>ENSG00000068912</td>\n",
       "      <td>Protein folding</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.69</td>\n",
       "      <td>29.09</td>\n",
       "      <td>16.54</td>\n",
       "      <td>38.64</td>\n",
       "      <td>...</td>\n",
       "      <td>27.40</td>\n",
       "      <td>11.73</td>\n",
       "      <td>14.55</td>\n",
       "      <td>20.24</td>\n",
       "      <td>22.70</td>\n",
       "      <td>21.47</td>\n",
       "      <td>30.54</td>\n",
       "      <td>32.03</td>\n",
       "      <td>25.53</td>\n",
       "      <td>22.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VAMP2</th>\n",
       "      <td>ENSG00000220205</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Not detected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>81.54</td>\n",
       "      <td>40.52</td>\n",
       "      <td>81.23</td>\n",
       "      <td>241.65</td>\n",
       "      <td>...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UBE2E2</th>\n",
       "      <td>ENSG00000182247</td>\n",
       "      <td>ERAD</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.23</td>\n",
       "      <td>21.53</td>\n",
       "      <td>10.37</td>\n",
       "      <td>48.61</td>\n",
       "      <td>...</td>\n",
       "      <td>7.07</td>\n",
       "      <td>3.48</td>\n",
       "      <td>7.43</td>\n",
       "      <td>4.05</td>\n",
       "      <td>6.25</td>\n",
       "      <td>4.62</td>\n",
       "      <td>8.75</td>\n",
       "      <td>5.69</td>\n",
       "      <td>7.76</td>\n",
       "      <td>8.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STX11</th>\n",
       "      <td>ENSG00000135604</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>bonemarrow</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>blood</td>\n",
       "      <td>3.08</td>\n",
       "      <td>10.07</td>\n",
       "      <td>80.41</td>\n",
       "      <td>0.55</td>\n",
       "      <td>...</td>\n",
       "      <td>1.09</td>\n",
       "      <td>0.84</td>\n",
       "      <td>4.89</td>\n",
       "      <td>2.40</td>\n",
       "      <td>11.46</td>\n",
       "      <td>1.27</td>\n",
       "      <td>1.26</td>\n",
       "      <td>6.29</td>\n",
       "      <td>1.01</td>\n",
       "      <td>2.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GLT1D1</th>\n",
       "      <td>ENSG00000151948</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>bonemarrow,testis</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>blood</td>\n",
       "      <td>0.22</td>\n",
       "      <td>5.83</td>\n",
       "      <td>13.53</td>\n",
       "      <td>10.92</td>\n",
       "      <td>...</td>\n",
       "      <td>0.40</td>\n",
       "      <td>2.09</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.30</td>\n",
       "      <td>8.74</td>\n",
       "      <td>0.25</td>\n",
       "      <td>16.04</td>\n",
       "      <td>0.52</td>\n",
       "      <td>1.73</td>\n",
       "      <td>0.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSD</th>\n",
       "      <td>ENSG00000059915</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain.cerebrum,pituitary</td>\n",
       "      <td>4.42</td>\n",
       "      <td>1.67</td>\n",
       "      <td>0.14</td>\n",
       "      <td>35.65</td>\n",
       "      <td>...</td>\n",
       "      <td>1.89</td>\n",
       "      <td>0.10</td>\n",
       "      <td>0.82</td>\n",
       "      <td>22.63</td>\n",
       "      <td>0.74</td>\n",
       "      <td>15.76</td>\n",
       "      <td>8.82</td>\n",
       "      <td>0.53</td>\n",
       "      <td>29.05</td>\n",
       "      <td>6.63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB26</th>\n",
       "      <td>ENSG00000167964</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>pituitary,brain.cerebellum</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.13</td>\n",
       "      <td>10.33</td>\n",
       "      <td>...</td>\n",
       "      <td>3.92</td>\n",
       "      <td>0.25</td>\n",
       "      <td>1.31</td>\n",
       "      <td>4.84</td>\n",
       "      <td>0.62</td>\n",
       "      <td>6.80</td>\n",
       "      <td>1.11</td>\n",
       "      <td>1.39</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB39B</th>\n",
       "      <td>ENSG00000155961</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum,pituitary</td>\n",
       "      <td>1.62</td>\n",
       "      <td>2.64</td>\n",
       "      <td>0.43</td>\n",
       "      <td>9.94</td>\n",
       "      <td>...</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.07</td>\n",
       "      <td>1.03</td>\n",
       "      <td>1.49</td>\n",
       "      <td>0.30</td>\n",
       "      <td>0.68</td>\n",
       "      <td>0.62</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IQSEC3</th>\n",
       "      <td>ENSG00000120645</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum,pituitary</td>\n",
       "      <td>0.63</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.37</td>\n",
       "      <td>7.80</td>\n",
       "      <td>...</td>\n",
       "      <td>1.05</td>\n",
       "      <td>0.16</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.41</td>\n",
       "      <td>0.39</td>\n",
       "      <td>2.21</td>\n",
       "      <td>0.50</td>\n",
       "      <td>2.12</td>\n",
       "      <td>1.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB33A</th>\n",
       "      <td>ENSG00000134594</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain.cerebrum,brain.cerebellum</td>\n",
       "      <td>0.72</td>\n",
       "      <td>2.50</td>\n",
       "      <td>2.36</td>\n",
       "      <td>15.60</td>\n",
       "      <td>...</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.85</td>\n",
       "      <td>1.77</td>\n",
       "      <td>2.23</td>\n",
       "      <td>0.36</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.43</td>\n",
       "      <td>1.38</td>\n",
       "      <td>0.73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB11FIP4</th>\n",
       "      <td>ENSG00000131242</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain.cerebrum</td>\n",
       "      <td>0.96</td>\n",
       "      <td>6.84</td>\n",
       "      <td>7.65</td>\n",
       "      <td>33.97</td>\n",
       "      <td>...</td>\n",
       "      <td>1.66</td>\n",
       "      <td>0.06</td>\n",
       "      <td>4.17</td>\n",
       "      <td>5.25</td>\n",
       "      <td>3.74</td>\n",
       "      <td>5.01</td>\n",
       "      <td>12.99</td>\n",
       "      <td>2.82</td>\n",
       "      <td>0.36</td>\n",
       "      <td>1.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SGSM1</th>\n",
       "      <td>ENSG00000167037</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>pituitary,brain.cerebellum</td>\n",
       "      <td>2.86</td>\n",
       "      <td>0.34</td>\n",
       "      <td>0.08</td>\n",
       "      <td>7.60</td>\n",
       "      <td>...</td>\n",
       "      <td>1.34</td>\n",
       "      <td>0.15</td>\n",
       "      <td>1.07</td>\n",
       "      <td>0.76</td>\n",
       "      <td>0.27</td>\n",
       "      <td>1.33</td>\n",
       "      <td>2.15</td>\n",
       "      <td>2.12</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPP6</th>\n",
       "      <td>ENSG00000130226</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain,endometrium</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>uterus,brain.cerebellum</td>\n",
       "      <td>7.94</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.05</td>\n",
       "      <td>54.40</td>\n",
       "      <td>...</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.61</td>\n",
       "      <td>2.80</td>\n",
       "      <td>0.68</td>\n",
       "      <td>3.42</td>\n",
       "      <td>4.09</td>\n",
       "      <td>9.42</td>\n",
       "      <td>27.43</td>\n",
       "      <td>0.81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MGAT4C</th>\n",
       "      <td>ENSG00000182050</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>brain,thyroid</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>thyroid,testis</td>\n",
       "      <td>2.79</td>\n",
       "      <td>0.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>6.53</td>\n",
       "      <td>...</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.04</td>\n",
       "      <td>2.42</td>\n",
       "      <td>2.58</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB25</th>\n",
       "      <td>ENSG00000132698</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>esophagus</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>esophagus</td>\n",
       "      <td>0.05</td>\n",
       "      <td>9.62</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.15</td>\n",
       "      <td>...</td>\n",
       "      <td>23.96</td>\n",
       "      <td>0.05</td>\n",
       "      <td>98.01</td>\n",
       "      <td>20.42</td>\n",
       "      <td>0.19</td>\n",
       "      <td>37.00</td>\n",
       "      <td>0.51</td>\n",
       "      <td>25.19</td>\n",
       "      <td>0.82</td>\n",
       "      <td>56.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB36</th>\n",
       "      <td>ENSG00000100228</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>fallopiantube</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.53</td>\n",
       "      <td>1.19</td>\n",
       "      <td>0.22</td>\n",
       "      <td>4.15</td>\n",
       "      <td>...</td>\n",
       "      <td>3.34</td>\n",
       "      <td>0.08</td>\n",
       "      <td>1.23</td>\n",
       "      <td>1.17</td>\n",
       "      <td>1.31</td>\n",
       "      <td>1.50</td>\n",
       "      <td>6.93</td>\n",
       "      <td>10.49</td>\n",
       "      <td>3.72</td>\n",
       "      <td>1.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HAS2</th>\n",
       "      <td>ENSG00000170961</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>fat</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.34</td>\n",
       "      <td>8.48</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.78</td>\n",
       "      <td>...</td>\n",
       "      <td>1.43</td>\n",
       "      <td>0.25</td>\n",
       "      <td>1.51</td>\n",
       "      <td>1.85</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.75</td>\n",
       "      <td>2.33</td>\n",
       "      <td>0.87</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GALNT15</th>\n",
       "      <td>ENSG00000131386</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>fat,brain</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>adipose</td>\n",
       "      <td>2.98</td>\n",
       "      <td>1.20</td>\n",
       "      <td>0.01</td>\n",
       "      <td>20.70</td>\n",
       "      <td>...</td>\n",
       "      <td>3.70</td>\n",
       "      <td>12.34</td>\n",
       "      <td>23.94</td>\n",
       "      <td>10.75</td>\n",
       "      <td>6.35</td>\n",
       "      <td>7.13</td>\n",
       "      <td>3.12</td>\n",
       "      <td>14.33</td>\n",
       "      <td>7.96</td>\n",
       "      <td>12.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HAS3</th>\n",
       "      <td>ENSG00000103044</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>lung,urinarybladder</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>esophagus</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.69</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.34</td>\n",
       "      <td>...</td>\n",
       "      <td>4.05</td>\n",
       "      <td>0.29</td>\n",
       "      <td>3.93</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.60</td>\n",
       "      <td>2.06</td>\n",
       "      <td>2.56</td>\n",
       "      <td>4.50</td>\n",
       "      <td>0.80</td>\n",
       "      <td>7.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB3B</th>\n",
       "      <td>ENSG00000169213</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>prostate</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebrum,pituitary,prostate</td>\n",
       "      <td>2.61</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.01</td>\n",
       "      <td>4.19</td>\n",
       "      <td>...</td>\n",
       "      <td>5.25</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.72</td>\n",
       "      <td>0.44</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B4GALNT2</th>\n",
       "      <td>ENSG00000167080</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>rectum,colon</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>colon</td>\n",
       "      <td>0.05</td>\n",
       "      <td>6.85</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.37</td>\n",
       "      <td>0.64</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.28</td>\n",
       "      <td>0.23</td>\n",
       "      <td>1.34</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WAS</th>\n",
       "      <td>ENSG00000015285</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>spleen</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>blood,spleen</td>\n",
       "      <td>3.56</td>\n",
       "      <td>38.57</td>\n",
       "      <td>27.91</td>\n",
       "      <td>2.68</td>\n",
       "      <td>...</td>\n",
       "      <td>3.97</td>\n",
       "      <td>0.52</td>\n",
       "      <td>1.90</td>\n",
       "      <td>18.40</td>\n",
       "      <td>70.53</td>\n",
       "      <td>3.43</td>\n",
       "      <td>1.76</td>\n",
       "      <td>2.37</td>\n",
       "      <td>2.00</td>\n",
       "      <td>3.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB27B</th>\n",
       "      <td>ENSG00000041353</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>stomach</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.02</td>\n",
       "      <td>1.07</td>\n",
       "      <td>0.49</td>\n",
       "      <td>12.65</td>\n",
       "      <td>...</td>\n",
       "      <td>5.67</td>\n",
       "      <td>0.02</td>\n",
       "      <td>5.76</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.61</td>\n",
       "      <td>5.68</td>\n",
       "      <td>2.25</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.25</td>\n",
       "      <td>3.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GALNT6</th>\n",
       "      <td>ENSG00000139629</td>\n",
       "      <td>Golgi glycosylation</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>stomach</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.63</td>\n",
       "      <td>7.72</td>\n",
       "      <td>4.53</td>\n",
       "      <td>3.23</td>\n",
       "      <td>...</td>\n",
       "      <td>0.78</td>\n",
       "      <td>0.07</td>\n",
       "      <td>2.79</td>\n",
       "      <td>2.88</td>\n",
       "      <td>3.75</td>\n",
       "      <td>7.73</td>\n",
       "      <td>0.46</td>\n",
       "      <td>1.04</td>\n",
       "      <td>0.28</td>\n",
       "      <td>1.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB43</th>\n",
       "      <td>ENSG00000172780</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Enhanced</td>\n",
       "      <td>thyroid</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.66</td>\n",
       "      <td>2.74</td>\n",
       "      <td>3.38</td>\n",
       "      <td>0.69</td>\n",
       "      <td>...</td>\n",
       "      <td>2.35</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.94</td>\n",
       "      <td>2.05</td>\n",
       "      <td>2.77</td>\n",
       "      <td>1.85</td>\n",
       "      <td>2.92</td>\n",
       "      <td>2.20</td>\n",
       "      <td>2.05</td>\n",
       "      <td>1.43</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>105 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ensgid               Subsystem      category.HPA  \\\n",
       "gene_name                                                              \n",
       "B4GALNT1   ENSG00000135454     Golgi glycosylation   Tissue enriched   \n",
       "AGAP2      ENSG00000135439  Trafficking regulation   Tissue enriched   \n",
       "AP3B2      ENSG00000103723  Post-Golgi trafficking   Tissue enriched   \n",
       "DNAJC6     ENSG00000116675  Post-Golgi trafficking   Tissue enriched   \n",
       "RAB6B      ENSG00000154917  Post-Golgi trafficking   Tissue enriched   \n",
       "SNAP25     ENSG00000132639  Post-Golgi trafficking   Tissue enriched   \n",
       "CRYAA      ENSG00000160202         Protein folding   Tissue enriched   \n",
       "HSPA4L     ENSG00000164070                    ERAD   Tissue enriched   \n",
       "ABO        ENSG00000175164     Golgi glycosylation             Mixed   \n",
       "PSD4       ENSG00000125637  Trafficking regulation             Mixed   \n",
       "RAB9B      ENSG00000123570  Post-Golgi trafficking             Mixed   \n",
       "GXYLT2     ENSG00000172986     Golgi glycosylation             Mixed   \n",
       "RAB37      ENSG00000172794  Post-Golgi trafficking             Mixed   \n",
       "GALNT18    ENSG00000110328     Golgi glycosylation             Mixed   \n",
       "CYTH4      ENSG00000100055  Trafficking regulation             Mixed   \n",
       "PGAP1      ENSG00000197121        GPI biosynthesis             Mixed   \n",
       "B3GNTL1    ENSG00000175711        ER glycosylation             Mixed   \n",
       "TBC1D30    ENSG00000111490  Trafficking regulation             Mixed   \n",
       "GALNT12    ENSG00000119514     Golgi glycosylation             Mixed   \n",
       "MYO5B      ENSG00000167306  Trafficking regulation             Mixed   \n",
       "MGAT3      ENSG00000128268     Golgi glycosylation             Mixed   \n",
       "RAB40B     ENSG00000141542  Post-Golgi trafficking             Mixed   \n",
       "RAB3C      ENSG00000152932  Post-Golgi trafficking    Group enriched   \n",
       "ERP27      ENSG00000139055         Protein folding    Group enriched   \n",
       "RAB2B      ENSG00000129472                    COPI  Expressed in all   \n",
       "CUL1       ENSG00000055130                    ERAD  Expressed in all   \n",
       "PIGL       ENSG00000108474        GPI biosynthesis  Expressed in all   \n",
       "FBXO6      ENSG00000116663         Protein folding  Expressed in all   \n",
       "STT3A      ENSG00000134910        ER glycosylation  Expressed in all   \n",
       "DNAJB12    ENSG00000148719         Protein folding  Expressed in all   \n",
       "...                    ...                     ...               ...   \n",
       "SEC23A     ENSG00000100934                   COPII  Expressed in all   \n",
       "HSP90AA1   ENSG00000080824         Protein folding  Expressed in all   \n",
       "EEA1       ENSG00000102189  Trafficking regulation  Expressed in all   \n",
       "ATF4       ENSG00000128272                     UPR  Expressed in all   \n",
       "EXOC6B     ENSG00000144036  Post-Golgi trafficking  Expressed in all   \n",
       "ERLEC1     ENSG00000068912         Protein folding  Expressed in all   \n",
       "VAMP2      ENSG00000220205  Post-Golgi trafficking  Expressed in all   \n",
       "UBE2E2     ENSG00000182247                    ERAD  Expressed in all   \n",
       "STX11      ENSG00000135604  Post-Golgi trafficking          Enhanced   \n",
       "GLT1D1     ENSG00000151948     Golgi glycosylation          Enhanced   \n",
       "PSD        ENSG00000059915  Trafficking regulation          Enhanced   \n",
       "RAB26      ENSG00000167964  Post-Golgi trafficking          Enhanced   \n",
       "RAB39B     ENSG00000155961  Post-Golgi trafficking          Enhanced   \n",
       "IQSEC3     ENSG00000120645  Trafficking regulation          Enhanced   \n",
       "RAB33A     ENSG00000134594  Post-Golgi trafficking          Enhanced   \n",
       "RAB11FIP4  ENSG00000131242  Trafficking regulation          Enhanced   \n",
       "SGSM1      ENSG00000167037  Trafficking regulation          Enhanced   \n",
       "DPP6       ENSG00000130226     Golgi glycosylation          Enhanced   \n",
       "MGAT4C     ENSG00000182050     Golgi glycosylation          Enhanced   \n",
       "RAB25      ENSG00000132698  Post-Golgi trafficking          Enhanced   \n",
       "RAB36      ENSG00000100228  Post-Golgi trafficking          Enhanced   \n",
       "HAS2       ENSG00000170961     Golgi glycosylation          Enhanced   \n",
       "GALNT15    ENSG00000131386     Golgi glycosylation          Enhanced   \n",
       "HAS3       ENSG00000103044     Golgi glycosylation          Enhanced   \n",
       "RAB3B      ENSG00000169213  Post-Golgi trafficking          Enhanced   \n",
       "B4GALNT2   ENSG00000167080     Golgi glycosylation          Enhanced   \n",
       "WAS        ENSG00000015285  Trafficking regulation          Enhanced   \n",
       "RAB27B     ENSG00000041353  Post-Golgi trafficking          Enhanced   \n",
       "GALNT6     ENSG00000139629     Golgi glycosylation          Enhanced   \n",
       "RAB43      ENSG00000172780  Post-Golgi trafficking          Enhanced   \n",
       "\n",
       "                      tissues.HPA     category.GTEx  \\\n",
       "gene_name                                             \n",
       "B4GALNT1                    brain    Group enriched   \n",
       "AGAP2                       brain    Group enriched   \n",
       "AP3B2                       brain    Group enriched   \n",
       "DNAJC6                      brain    Group enriched   \n",
       "RAB6B                       brain          Enhanced   \n",
       "SNAP25                      brain    Group enriched   \n",
       "CRYAA                      kidney   Tissue enriched   \n",
       "HSPA4L                     testis   Tissue enriched   \n",
       "ABO                           NaN          Enhanced   \n",
       "PSD4                          NaN             Mixed   \n",
       "RAB9B                         NaN          Enhanced   \n",
       "GXYLT2                        NaN             Mixed   \n",
       "RAB37                         NaN          Enhanced   \n",
       "GALNT18                       NaN             Mixed   \n",
       "CYTH4                         NaN    Group enriched   \n",
       "PGAP1                         NaN             Mixed   \n",
       "B3GNTL1                       NaN  Expressed in all   \n",
       "TBC1D30                       NaN             Mixed   \n",
       "GALNT12                       NaN          Enhanced   \n",
       "MYO5B                         NaN             Mixed   \n",
       "MGAT3                         NaN          Enhanced   \n",
       "RAB40B                        NaN             Mixed   \n",
       "RAB3C               adrenal,brain    Group enriched   \n",
       "ERP27      fallopiantube,pancreas   Tissue enriched   \n",
       "RAB2B                         NaN  Expressed in all   \n",
       "CUL1                          NaN  Expressed in all   \n",
       "PIGL                          NaN  Expressed in all   \n",
       "FBXO6                         NaN  Expressed in all   \n",
       "STT3A                         NaN  Expressed in all   \n",
       "DNAJB12                       NaN  Expressed in all   \n",
       "...                           ...               ...   \n",
       "SEC23A                        NaN  Expressed in all   \n",
       "HSP90AA1                      NaN  Expressed in all   \n",
       "EEA1                          NaN             Mixed   \n",
       "ATF4                          NaN  Expressed in all   \n",
       "EXOC6B                        NaN             Mixed   \n",
       "ERLEC1                        NaN  Expressed in all   \n",
       "VAMP2                         NaN      Not detected   \n",
       "UBE2E2                        NaN  Expressed in all   \n",
       "STX11                  bonemarrow          Enhanced   \n",
       "GLT1D1          bonemarrow,testis          Enhanced   \n",
       "PSD                         brain          Enhanced   \n",
       "RAB26                       brain          Enhanced   \n",
       "RAB39B                      brain    Group enriched   \n",
       "IQSEC3                      brain    Group enriched   \n",
       "RAB33A                      brain          Enhanced   \n",
       "RAB11FIP4                   brain          Enhanced   \n",
       "SGSM1                       brain          Enhanced   \n",
       "DPP6            brain,endometrium          Enhanced   \n",
       "MGAT4C              brain,thyroid          Enhanced   \n",
       "RAB25                   esophagus          Enhanced   \n",
       "RAB36               fallopiantube             Mixed   \n",
       "HAS2                          fat             Mixed   \n",
       "GALNT15                 fat,brain          Enhanced   \n",
       "HAS3          lung,urinarybladder          Enhanced   \n",
       "RAB3B                    prostate    Group enriched   \n",
       "B4GALNT2             rectum,colon   Tissue enriched   \n",
       "WAS                        spleen    Group enriched   \n",
       "RAB27B                    stomach             Mixed   \n",
       "GALNT6                    stomach             Mixed   \n",
       "RAB43                     thyroid             Mixed   \n",
       "\n",
       "                                        tissues.GTEx  fpkm.HPA.adrenal  \\\n",
       "gene_name                                                                \n",
       "B4GALNT1       brain.cerebellum,brain.cerebrum,nerve              0.56   \n",
       "AGAP2                brain.cerebellum,brain.cerebrum              1.47   \n",
       "AP3B2      brain.cerebellum,brain.cerebrum,pituitary              2.37   \n",
       "DNAJC6     brain.cerebellum,brain.cerebrum,pituitary             10.31   \n",
       "RAB6B                                 brain.cerebrum              9.89   \n",
       "SNAP25               brain.cerebellum,brain.cerebrum             39.99   \n",
       "CRYAA                                         kidney              0.00   \n",
       "HSPA4L                                        testis              9.51   \n",
       "ABO                                            colon              3.85   \n",
       "PSD4                                             NaN              7.73   \n",
       "RAB9B                               brain.cerebellum              2.54   \n",
       "GXYLT2                                           NaN              2.92   \n",
       "RAB37                               brain.cerebellum              1.37   \n",
       "GALNT18                                          NaN              4.15   \n",
       "CYTH4                                   blood,spleen              6.46   \n",
       "PGAP1                                            NaN              8.36   \n",
       "B3GNTL1                                          NaN              1.18   \n",
       "TBC1D30                                          NaN              0.82   \n",
       "GALNT12                                        colon              1.39   \n",
       "MYO5B                                            NaN              1.28   \n",
       "MGAT3                               brain.cerebellum              0.60   \n",
       "RAB40B                                           NaN              5.37   \n",
       "RAB3C      brain.cerebellum,brain.cerebrum,pituitary              8.34   \n",
       "ERP27                                       pancreas              0.31   \n",
       "RAB2B                                            NaN             11.15   \n",
       "CUL1                                             NaN             23.80   \n",
       "PIGL                                             NaN              4.95   \n",
       "FBXO6                                            NaN              4.61   \n",
       "STT3A                                            NaN             35.31   \n",
       "DNAJB12                                          NaN             17.34   \n",
       "...                                              ...               ...   \n",
       "SEC23A                                           NaN             27.57   \n",
       "HSP90AA1                                         NaN            292.57   \n",
       "EEA1                                             NaN              6.05   \n",
       "ATF4                                             NaN            176.34   \n",
       "EXOC6B                                           NaN             11.11   \n",
       "ERLEC1                                           NaN             44.69   \n",
       "VAMP2                                            NaN             81.54   \n",
       "UBE2E2                                           NaN             22.23   \n",
       "STX11                                          blood              3.08   \n",
       "GLT1D1                                         blood              0.22   \n",
       "PSD                         brain.cerebrum,pituitary              4.42   \n",
       "RAB26                     pituitary,brain.cerebellum              0.48   \n",
       "RAB39B     brain.cerebellum,brain.cerebrum,pituitary              1.62   \n",
       "IQSEC3     brain.cerebellum,brain.cerebrum,pituitary              0.63   \n",
       "RAB33A               brain.cerebrum,brain.cerebellum              0.72   \n",
       "RAB11FIP4                             brain.cerebrum              0.96   \n",
       "SGSM1                     pituitary,brain.cerebellum              2.86   \n",
       "DPP6                         uterus,brain.cerebellum              7.94   \n",
       "MGAT4C                                thyroid,testis              2.79   \n",
       "RAB25                                      esophagus              0.05   \n",
       "RAB36                                            NaN              3.53   \n",
       "HAS2                                             NaN              0.34   \n",
       "GALNT15                                      adipose              2.98   \n",
       "HAS3                                       esophagus              0.25   \n",
       "RAB3B              brain.cerebrum,pituitary,prostate              2.61   \n",
       "B4GALNT2                                       colon              0.05   \n",
       "WAS                                     blood,spleen              3.56   \n",
       "RAB27B                                           NaN              2.02   \n",
       "GALNT6                                           NaN              2.63   \n",
       "RAB43                                            NaN              1.66   \n",
       "\n",
       "           fpkm.HPA.appendices  fpkm.HPA.bonemarrow  fpkm.HPA.brain  \\\n",
       "gene_name                                                             \n",
       "B4GALNT1                  0.62                 0.03           12.69   \n",
       "AGAP2                     7.62                 7.22           65.72   \n",
       "AP3B2                     0.28                 0.05           17.70   \n",
       "DNAJC6                    0.83                 2.53           83.68   \n",
       "RAB6B                     2.38                 6.01          106.17   \n",
       "SNAP25                    3.75                 0.03          916.53   \n",
       "CRYAA                     0.00                 0.00            0.05   \n",
       "HSPA4L                    0.96                 1.03            7.31   \n",
       "ABO                       4.92                 0.40            0.26   \n",
       "PSD4                     16.61                 7.83            1.77   \n",
       "RAB9B                     0.86                 0.07            7.19   \n",
       "GXYLT2                    4.00                 0.90            1.72   \n",
       "RAB37                     8.81                11.79            1.64   \n",
       "GALNT18                   3.64                 0.01           11.48   \n",
       "CYTH4                    45.25                21.59            3.70   \n",
       "PGAP1                     1.15                 1.00            7.35   \n",
       "B3GNTL1                   3.47                 4.25            1.30   \n",
       "TBC1D30                   1.95                 0.26            6.27   \n",
       "GALNT12                   6.82                 0.94            0.61   \n",
       "MYO5B                     7.79                 0.09            0.93   \n",
       "MGAT3                     3.04                 3.50           17.42   \n",
       "RAB40B                    1.85                 0.42           26.88   \n",
       "RAB3C                     1.18                 0.03           29.50   \n",
       "ERP27                     2.92                 1.98            0.09   \n",
       "RAB2B                     7.29                10.85           10.08   \n",
       "CUL1                     27.77                16.25           23.88   \n",
       "PIGL                      4.89                 4.01            4.53   \n",
       "FBXO6                     9.00                 7.78            2.91   \n",
       "STT3A                    43.18                17.18           17.34   \n",
       "DNAJB12                  14.97                12.76           12.41   \n",
       "...                        ...                  ...             ...   \n",
       "SEC23A                   30.10                 5.78           18.83   \n",
       "HSP90AA1                216.58               182.86          422.36   \n",
       "EEA1                      6.61                 1.90            6.35   \n",
       "ATF4                    156.45               413.76           88.06   \n",
       "EXOC6B                    3.84                 0.59           17.09   \n",
       "ERLEC1                   29.09                16.54           38.64   \n",
       "VAMP2                    40.52                81.23          241.65   \n",
       "UBE2E2                   21.53                10.37           48.61   \n",
       "STX11                    10.07                80.41            0.55   \n",
       "GLT1D1                    5.83                13.53           10.92   \n",
       "PSD                       1.67                 0.14           35.65   \n",
       "RAB26                     0.00                 0.13           10.33   \n",
       "RAB39B                    2.64                 0.43            9.94   \n",
       "IQSEC3                    0.43                 0.37            7.80   \n",
       "RAB33A                    2.50                 2.36           15.60   \n",
       "RAB11FIP4                 6.84                 7.65           33.97   \n",
       "SGSM1                     0.34                 0.08            7.60   \n",
       "DPP6                      0.27                 0.05           54.40   \n",
       "MGAT4C                    0.18                 0.00            6.53   \n",
       "RAB25                     9.62                 0.04            0.15   \n",
       "RAB36                     1.19                 0.22            4.15   \n",
       "HAS2                      8.48                 0.05            0.78   \n",
       "GALNT15                   1.20                 0.01           20.70   \n",
       "HAS3                      0.69                 0.43            0.34   \n",
       "RAB3B                     0.50                 0.01            4.19   \n",
       "B4GALNT2                  6.85                 0.00            0.00   \n",
       "WAS                      38.57                27.91            2.68   \n",
       "RAB27B                    1.07                 0.49           12.65   \n",
       "GALNT6                    7.72                 4.53            3.23   \n",
       "RAB43                     2.74                 3.38            0.69   \n",
       "\n",
       "                 ...         fpkm.GTEx.prostate  fpkm.GTEx.skeletalmuscle  \\\n",
       "gene_name        ...                                                        \n",
       "B4GALNT1         ...                       1.73                      2.63   \n",
       "AGAP2            ...                       1.59                      3.04   \n",
       "AP3B2            ...                       1.38                      0.18   \n",
       "DNAJC6           ...                       0.46                      0.16   \n",
       "RAB6B            ...                       0.91                      0.37   \n",
       "SNAP25           ...                       4.32                      0.27   \n",
       "CRYAA            ...                       0.01                      0.00   \n",
       "HSPA4L           ...                       2.17                      0.64   \n",
       "ABO              ...                       2.02                      0.12   \n",
       "PSD4             ...                      17.57                      0.20   \n",
       "RAB9B            ...                       3.74                      2.15   \n",
       "GXYLT2           ...                       1.33                      0.25   \n",
       "RAB37            ...                       3.27                      0.22   \n",
       "GALNT18          ...                       8.27                      3.25   \n",
       "CYTH4            ...                       4.42                      0.56   \n",
       "PGAP1            ...                       2.65                      0.59   \n",
       "B3GNTL1          ...                       1.69                      0.74   \n",
       "TBC1D30          ...                       1.91                      0.09   \n",
       "GALNT12          ...                       3.39                      0.23   \n",
       "MYO5B            ...                       2.90                      0.08   \n",
       "MGAT3            ...                       1.08                      0.33   \n",
       "RAB40B           ...                       5.74                      7.39   \n",
       "RAB3C            ...                       0.08                      0.05   \n",
       "ERP27            ...                       1.37                      0.08   \n",
       "RAB2B            ...                      12.24                      2.40   \n",
       "CUL1             ...                      22.10                     31.45   \n",
       "PIGL             ...                       5.26                      0.97   \n",
       "FBXO6            ...                       8.20                      6.76   \n",
       "STT3A            ...                      16.18                      4.42   \n",
       "DNAJB12          ...                      12.95                      8.43   \n",
       "...              ...                        ...                       ...   \n",
       "SEC23A           ...                      10.30                     11.11   \n",
       "HSP90AA1         ...                     180.49                    157.19   \n",
       "EEA1             ...                       2.23                      4.22   \n",
       "ATF4             ...                     232.84                    161.37   \n",
       "EXOC6B           ...                       3.65                      2.19   \n",
       "ERLEC1           ...                      27.40                     11.73   \n",
       "VAMP2            ...                       0.01                      0.01   \n",
       "UBE2E2           ...                       7.07                      3.48   \n",
       "STX11            ...                       1.09                      0.84   \n",
       "GLT1D1           ...                       0.40                      2.09   \n",
       "PSD              ...                       1.89                      0.10   \n",
       "RAB26            ...                       3.92                      0.25   \n",
       "RAB39B           ...                       0.68                      0.02   \n",
       "IQSEC3           ...                       1.05                      0.16   \n",
       "RAB33A           ...                       0.37                      0.06   \n",
       "RAB11FIP4        ...                       1.66                      0.06   \n",
       "SGSM1            ...                       1.34                      0.15   \n",
       "DPP6             ...                       0.14                      0.18   \n",
       "MGAT4C           ...                       0.07                      0.00   \n",
       "RAB25            ...                      23.96                      0.05   \n",
       "RAB36            ...                       3.34                      0.08   \n",
       "HAS2             ...                       1.43                      0.25   \n",
       "GALNT15          ...                       3.70                     12.34   \n",
       "HAS3             ...                       4.05                      0.29   \n",
       "RAB3B            ...                       5.25                      0.04   \n",
       "B4GALNT2         ...                       0.01                      0.00   \n",
       "WAS              ...                       3.97                      0.52   \n",
       "RAB27B           ...                       5.67                      0.02   \n",
       "GALNT6           ...                       0.78                      0.07   \n",
       "RAB43            ...                       2.35                      0.21   \n",
       "\n",
       "           fpkm.GTEx.skin  fpkm.GTEx.smallintestine  fpkm.GTEx.spleen  \\\n",
       "gene_name                                                               \n",
       "B4GALNT1             0.26                      0.95              0.22   \n",
       "AGAP2                1.03                      5.51              9.41   \n",
       "AP3B2                0.74                      0.63              0.83   \n",
       "DNAJC6               0.66                      0.52              0.74   \n",
       "RAB6B                0.53                      1.27              2.16   \n",
       "SNAP25               0.57                      1.45              0.05   \n",
       "CRYAA                0.02                      0.02              0.03   \n",
       "HSPA4L               5.85                      0.99              0.57   \n",
       "ABO                  0.74                      7.80              0.39   \n",
       "PSD4                10.51                     24.51             35.05   \n",
       "RAB9B                0.40                      2.08              0.35   \n",
       "GXYLT2               0.82                      0.66              0.11   \n",
       "RAB37                0.65                      7.45             11.54   \n",
       "GALNT18              7.25                      5.03              2.42   \n",
       "CYTH4                2.19                     16.17             58.76   \n",
       "PGAP1                4.29                      2.07              0.52   \n",
       "B3GNTL1              1.61                      1.35              2.68   \n",
       "TBC1D30              0.43                      0.78              1.13   \n",
       "GALNT12              2.90                     13.76              1.62   \n",
       "MYO5B                7.82                      5.99              0.26   \n",
       "MGAT3                1.83                      8.07              4.49   \n",
       "RAB40B               5.88                      2.93              2.49   \n",
       "RAB3C                0.10                      0.40              0.02   \n",
       "ERP27                0.79                      1.57              2.48   \n",
       "RAB2B                9.00                      8.37              9.58   \n",
       "CUL1                19.44                     20.76             20.51   \n",
       "PIGL                 6.34                      4.04              3.13   \n",
       "FBXO6                6.98                      6.33             12.08   \n",
       "STT3A               12.01                     15.29             17.05   \n",
       "DNAJB12             12.91                      9.65             12.26   \n",
       "...                   ...                       ...               ...   \n",
       "SEC23A               5.46                     13.95              5.56   \n",
       "HSP90AA1           136.32                    161.76            173.69   \n",
       "EEA1                 3.77                      1.83              0.95   \n",
       "ATF4               152.33                    174.90            157.97   \n",
       "EXOC6B              11.03                      3.87              1.99   \n",
       "ERLEC1              14.55                     20.24             22.70   \n",
       "VAMP2                0.01                      0.02              0.02   \n",
       "UBE2E2               7.43                      4.05              6.25   \n",
       "STX11                4.89                      2.40             11.46   \n",
       "GLT1D1               0.60                      0.30              8.74   \n",
       "PSD                  0.82                     22.63              0.74   \n",
       "RAB26                1.31                      4.84              0.62   \n",
       "RAB39B               0.07                      1.03              1.49   \n",
       "IQSEC3               0.15                      0.46              0.41   \n",
       "RAB33A               0.85                      1.77              2.23   \n",
       "RAB11FIP4            4.17                      5.25              3.74   \n",
       "SGSM1                1.07                      0.76              0.27   \n",
       "DPP6                 0.61                      2.80              0.68   \n",
       "MGAT4C               0.02                      0.05              0.00   \n",
       "RAB25               98.01                     20.42              0.19   \n",
       "RAB36                1.23                      1.17              1.31   \n",
       "HAS2                 1.51                      1.85              0.08   \n",
       "GALNT15             23.94                     10.75              6.35   \n",
       "HAS3                 3.93                      0.88              0.60   \n",
       "RAB3B                0.72                      0.44              0.05   \n",
       "B4GALNT2             0.37                      0.64              0.00   \n",
       "WAS                  1.90                     18.40             70.53   \n",
       "RAB27B               5.76                      0.91              0.61   \n",
       "GALNT6               2.79                      2.88              3.75   \n",
       "RAB43                0.94                      2.05              2.77   \n",
       "\n",
       "           fpkm.GTEx.stomach  fpkm.GTEx.testis  fpkm.GTEx.thyroid  \\\n",
       "gene_name                                                           \n",
       "B4GALNT1                0.48              0.56               0.83   \n",
       "AGAP2                   1.36              0.95               1.96   \n",
       "AP3B2                   0.64              4.90               1.22   \n",
       "DNAJC6                  0.88              0.94               0.61   \n",
       "RAB6B                   1.72              2.63               2.40   \n",
       "SNAP25                  2.23              3.48               0.70   \n",
       "CRYAA                   0.01              0.20               0.01   \n",
       "HSPA4L                  1.44             57.43               1.62   \n",
       "ABO                     1.29              1.60               7.94   \n",
       "PSD4                   13.41              4.87              16.69   \n",
       "RAB9B                   1.92              1.23               1.02   \n",
       "GXYLT2                  0.62              2.94               2.59   \n",
       "RAB37                   1.91              1.84               0.46   \n",
       "GALNT18                 3.29             11.72              27.47   \n",
       "CYTH4                   3.36              1.03               2.61   \n",
       "PGAP1                   1.27              1.68               1.75   \n",
       "B3GNTL1                 0.92              2.51               1.44   \n",
       "TBC1D30                 1.88              1.21               0.91   \n",
       "GALNT12                19.78              4.01              10.47   \n",
       "MYO5B                   3.37              4.36               6.91   \n",
       "MGAT3                   3.23              4.67               1.33   \n",
       "RAB40B                  6.62              3.46               3.70   \n",
       "RAB3C                   0.41              0.16               0.14   \n",
       "ERP27                   1.62              0.13               4.44   \n",
       "RAB2B                   6.77             11.35              14.07   \n",
       "CUL1                   20.28             75.95              20.87   \n",
       "PIGL                    4.44              4.12               5.27   \n",
       "FBXO6                   7.41             13.69               5.94   \n",
       "STT3A                  16.05             14.00              21.62   \n",
       "DNAJB12                 9.99             14.35              12.38   \n",
       "...                      ...               ...                ...   \n",
       "SEC23A                  9.05              6.72               8.95   \n",
       "HSP90AA1              172.62            454.43             193.85   \n",
       "EEA1                    1.85              2.77               2.66   \n",
       "ATF4                  219.19             95.17             147.49   \n",
       "EXOC6B                  2.87              3.68               3.51   \n",
       "ERLEC1                 21.47             30.54              32.03   \n",
       "VAMP2                   0.00              0.05               0.02   \n",
       "UBE2E2                  4.62              8.75               5.69   \n",
       "STX11                   1.27              1.26               6.29   \n",
       "GLT1D1                  0.25             16.04               0.52   \n",
       "PSD                    15.76              8.82               0.53   \n",
       "RAB26                   6.80              1.11               1.39   \n",
       "RAB39B                  0.30              0.68               0.62   \n",
       "IQSEC3                  0.39              2.21               0.50   \n",
       "RAB33A                  0.36              0.61               0.43   \n",
       "RAB11FIP4               5.01             12.99               2.82   \n",
       "SGSM1                   1.33              2.15               2.12   \n",
       "DPP6                    3.42              4.09               9.42   \n",
       "MGAT4C                  0.04              2.42               2.58   \n",
       "RAB25                  37.00              0.51              25.19   \n",
       "RAB36                   1.50              6.93              10.49   \n",
       "HAS2                    0.75              2.33               0.87   \n",
       "GALNT15                 7.13              3.12              14.33   \n",
       "HAS3                    2.06              2.56               4.50   \n",
       "RAB3B                   0.29              0.39               0.06   \n",
       "B4GALNT2                2.28              0.23               1.34   \n",
       "WAS                     3.43              1.76               2.37   \n",
       "RAB27B                  5.68              2.25               0.08   \n",
       "GALNT6                  7.73              0.46               1.04   \n",
       "RAB43                   1.85              2.92               2.20   \n",
       "\n",
       "           fpkm.GTEx.uterus  fpkm.GTEx.vagina  \n",
       "gene_name                                      \n",
       "B4GALNT1               1.21              1.09  \n",
       "AGAP2                  1.05              0.96  \n",
       "AP3B2                  0.78              0.75  \n",
       "DNAJC6                 0.29              0.64  \n",
       "RAB6B                  1.54              1.07  \n",
       "SNAP25                 5.51              1.23  \n",
       "CRYAA                  0.02              0.03  \n",
       "HSPA4L                 0.87              4.79  \n",
       "ABO                    0.76              4.02  \n",
       "PSD4                   1.13              4.91  \n",
       "RAB9B                  4.72              1.48  \n",
       "GXYLT2                 2.11              2.41  \n",
       "RAB37                  0.61              1.59  \n",
       "GALNT18               14.61              7.21  \n",
       "CYTH4                  2.14              3.09  \n",
       "PGAP1                  2.38              2.97  \n",
       "B3GNTL1                2.18              1.85  \n",
       "TBC1D30                0.79              0.80  \n",
       "GALNT12                1.70              3.63  \n",
       "MYO5B                  0.22              7.19  \n",
       "MGAT3                  3.16              1.30  \n",
       "RAB40B                 4.54              4.87  \n",
       "RAB3C                  0.20              0.25  \n",
       "ERP27                  0.31              0.70  \n",
       "RAB2B                  9.26              8.66  \n",
       "CUL1                  23.60             20.12  \n",
       "PIGL                   4.31              4.28  \n",
       "FBXO6                  4.76              6.04  \n",
       "STT3A                 23.59             20.86  \n",
       "DNAJB12               11.09              9.82  \n",
       "...                     ...               ...  \n",
       "SEC23A                16.39             12.22  \n",
       "HSP90AA1             171.67            179.82  \n",
       "EEA1                   3.65              4.47  \n",
       "ATF4                 185.31            162.84  \n",
       "EXOC6B                 4.82              5.55  \n",
       "ERLEC1                25.53             22.53  \n",
       "VAMP2                  0.02              0.02  \n",
       "UBE2E2                 7.76              8.54  \n",
       "STX11                  1.01              2.10  \n",
       "GLT1D1                 1.73              0.54  \n",
       "PSD                   29.05              6.63  \n",
       "RAB26                  0.15              0.51  \n",
       "RAB39B                 0.16              0.25  \n",
       "IQSEC3                 2.12              1.92  \n",
       "RAB33A                 1.38              0.73  \n",
       "RAB11FIP4              0.36              1.03  \n",
       "SGSM1                  0.71              0.99  \n",
       "DPP6                  27.43              0.81  \n",
       "MGAT4C                 0.02              0.06  \n",
       "RAB25                  0.82             56.00  \n",
       "RAB36                  3.72              1.83  \n",
       "HAS2                   0.49              0.66  \n",
       "GALNT15                7.96             12.36  \n",
       "HAS3                   0.80              7.47  \n",
       "RAB3B                  0.01              0.13  \n",
       "B4GALNT2               0.01              0.03  \n",
       "WAS                    2.00              3.27  \n",
       "RAB27B                 0.25              3.67  \n",
       "GALNT6                 0.28              1.01  \n",
       "RAB43                  2.05              1.43  \n",
       "\n",
       "[105 rows x 68 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "secM[secM.index.isin(not_in_cho1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensgid</th>\n",
       "      <th>Subsystem</th>\n",
       "      <th>category.HPA</th>\n",
       "      <th>tissues.HPA</th>\n",
       "      <th>category.GTEx</th>\n",
       "      <th>tissues.GTEx</th>\n",
       "      <th>fpkm.HPA.adrenal</th>\n",
       "      <th>fpkm.HPA.appendices</th>\n",
       "      <th>fpkm.HPA.bonemarrow</th>\n",
       "      <th>fpkm.HPA.brain</th>\n",
       "      <th>...</th>\n",
       "      <th>fpkm.GTEx.prostate</th>\n",
       "      <th>fpkm.GTEx.skeletalmuscle</th>\n",
       "      <th>fpkm.GTEx.skin</th>\n",
       "      <th>fpkm.GTEx.smallintestine</th>\n",
       "      <th>fpkm.GTEx.spleen</th>\n",
       "      <th>fpkm.GTEx.stomach</th>\n",
       "      <th>fpkm.GTEx.testis</th>\n",
       "      <th>fpkm.GTEx.thyroid</th>\n",
       "      <th>fpkm.GTEx.uterus</th>\n",
       "      <th>fpkm.GTEx.vagina</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HSPA4L</th>\n",
       "      <td>ENSG00000164070</td>\n",
       "      <td>ERAD</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>9.51</td>\n",
       "      <td>0.96</td>\n",
       "      <td>1.03</td>\n",
       "      <td>7.31</td>\n",
       "      <td>...</td>\n",
       "      <td>2.17</td>\n",
       "      <td>0.64</td>\n",
       "      <td>5.85</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.44</td>\n",
       "      <td>57.43</td>\n",
       "      <td>1.62</td>\n",
       "      <td>0.87</td>\n",
       "      <td>4.79</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ensgid Subsystem     category.HPA tissues.HPA  \\\n",
       "gene_name                                                           \n",
       "HSPA4L     ENSG00000164070      ERAD  Tissue enriched      testis   \n",
       "\n",
       "             category.GTEx tissues.GTEx  fpkm.HPA.adrenal  \\\n",
       "gene_name                                                   \n",
       "HSPA4L     Tissue enriched       testis              9.51   \n",
       "\n",
       "           fpkm.HPA.appendices  fpkm.HPA.bonemarrow  fpkm.HPA.brain  \\\n",
       "gene_name                                                             \n",
       "HSPA4L                    0.96                 1.03            7.31   \n",
       "\n",
       "                 ...         fpkm.GTEx.prostate  fpkm.GTEx.skeletalmuscle  \\\n",
       "gene_name        ...                                                        \n",
       "HSPA4L           ...                       2.17                      0.64   \n",
       "\n",
       "           fpkm.GTEx.skin  fpkm.GTEx.smallintestine  fpkm.GTEx.spleen  \\\n",
       "gene_name                                                               \n",
       "HSPA4L               5.85                      0.99              0.57   \n",
       "\n",
       "           fpkm.GTEx.stomach  fpkm.GTEx.testis  fpkm.GTEx.thyroid  \\\n",
       "gene_name                                                           \n",
       "HSPA4L                  1.44             57.43               1.62   \n",
       "\n",
       "           fpkm.GTEx.uterus  fpkm.GTEx.vagina  \n",
       "gene_name                                      \n",
       "HSPA4L                 0.87              4.79  \n",
       "\n",
       "[1 rows x 68 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "secM[secM.index.isin(not_in_cho2)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensgid</th>\n",
       "      <th>Subsystem</th>\n",
       "      <th>category.HPA</th>\n",
       "      <th>tissues.HPA</th>\n",
       "      <th>category.GTEx</th>\n",
       "      <th>tissues.GTEx</th>\n",
       "      <th>fpkm.HPA.adrenal</th>\n",
       "      <th>fpkm.HPA.appendices</th>\n",
       "      <th>fpkm.HPA.bonemarrow</th>\n",
       "      <th>fpkm.HPA.brain</th>\n",
       "      <th>...</th>\n",
       "      <th>fpkm.GTEx.prostate</th>\n",
       "      <th>fpkm.GTEx.skeletalmuscle</th>\n",
       "      <th>fpkm.GTEx.skin</th>\n",
       "      <th>fpkm.GTEx.smallintestine</th>\n",
       "      <th>fpkm.GTEx.spleen</th>\n",
       "      <th>fpkm.GTEx.stomach</th>\n",
       "      <th>fpkm.GTEx.testis</th>\n",
       "      <th>fpkm.GTEx.thyroid</th>\n",
       "      <th>fpkm.GTEx.uterus</th>\n",
       "      <th>fpkm.GTEx.vagina</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SNAP25</th>\n",
       "      <td>ENSG00000132639</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum</td>\n",
       "      <td>39.99</td>\n",
       "      <td>3.75</td>\n",
       "      <td>0.03</td>\n",
       "      <td>916.53</td>\n",
       "      <td>...</td>\n",
       "      <td>4.32</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.45</td>\n",
       "      <td>0.05</td>\n",
       "      <td>2.23</td>\n",
       "      <td>3.48</td>\n",
       "      <td>0.70</td>\n",
       "      <td>5.51</td>\n",
       "      <td>1.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HSPA4L</th>\n",
       "      <td>ENSG00000164070</td>\n",
       "      <td>ERAD</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>9.51</td>\n",
       "      <td>0.96</td>\n",
       "      <td>1.03</td>\n",
       "      <td>7.31</td>\n",
       "      <td>...</td>\n",
       "      <td>2.17</td>\n",
       "      <td>0.64</td>\n",
       "      <td>5.85</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.44</td>\n",
       "      <td>57.43</td>\n",
       "      <td>1.62</td>\n",
       "      <td>0.87</td>\n",
       "      <td>4.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB40B</th>\n",
       "      <td>ENSG00000141542</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.37</td>\n",
       "      <td>1.85</td>\n",
       "      <td>0.42</td>\n",
       "      <td>26.88</td>\n",
       "      <td>...</td>\n",
       "      <td>5.74</td>\n",
       "      <td>7.39</td>\n",
       "      <td>5.88</td>\n",
       "      <td>2.93</td>\n",
       "      <td>2.49</td>\n",
       "      <td>6.62</td>\n",
       "      <td>3.46</td>\n",
       "      <td>3.70</td>\n",
       "      <td>4.54</td>\n",
       "      <td>4.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB11FIP1</th>\n",
       "      <td>ENSG00000156675</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.35</td>\n",
       "      <td>16.61</td>\n",
       "      <td>30.77</td>\n",
       "      <td>0.88</td>\n",
       "      <td>...</td>\n",
       "      <td>4.80</td>\n",
       "      <td>1.85</td>\n",
       "      <td>13.69</td>\n",
       "      <td>18.17</td>\n",
       "      <td>7.10</td>\n",
       "      <td>10.22</td>\n",
       "      <td>4.41</td>\n",
       "      <td>6.65</td>\n",
       "      <td>2.25</td>\n",
       "      <td>6.86</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ensgid               Subsystem      category.HPA  \\\n",
       "gene_name                                                              \n",
       "SNAP25     ENSG00000132639  Post-Golgi trafficking   Tissue enriched   \n",
       "HSPA4L     ENSG00000164070                    ERAD   Tissue enriched   \n",
       "RAB40B     ENSG00000141542  Post-Golgi trafficking             Mixed   \n",
       "RAB11FIP1  ENSG00000156675  Trafficking regulation  Expressed in all   \n",
       "\n",
       "          tissues.HPA     category.GTEx                     tissues.GTEx  \\\n",
       "gene_name                                                                  \n",
       "SNAP25          brain    Group enriched  brain.cerebellum,brain.cerebrum   \n",
       "HSPA4L         testis   Tissue enriched                           testis   \n",
       "RAB40B            NaN             Mixed                              NaN   \n",
       "RAB11FIP1         NaN  Expressed in all                              NaN   \n",
       "\n",
       "           fpkm.HPA.adrenal  fpkm.HPA.appendices  fpkm.HPA.bonemarrow  \\\n",
       "gene_name                                                               \n",
       "SNAP25                39.99                 3.75                 0.03   \n",
       "HSPA4L                 9.51                 0.96                 1.03   \n",
       "RAB40B                 5.37                 1.85                 0.42   \n",
       "RAB11FIP1              4.35                16.61                30.77   \n",
       "\n",
       "           fpkm.HPA.brain        ...         fpkm.GTEx.prostate  \\\n",
       "gene_name                        ...                              \n",
       "SNAP25             916.53        ...                       4.32   \n",
       "HSPA4L               7.31        ...                       2.17   \n",
       "RAB40B              26.88        ...                       5.74   \n",
       "RAB11FIP1            0.88        ...                       4.80   \n",
       "\n",
       "           fpkm.GTEx.skeletalmuscle  fpkm.GTEx.skin  fpkm.GTEx.smallintestine  \\\n",
       "gene_name                                                                       \n",
       "SNAP25                         0.27            0.57                      1.45   \n",
       "HSPA4L                         0.64            5.85                      0.99   \n",
       "RAB40B                         7.39            5.88                      2.93   \n",
       "RAB11FIP1                      1.85           13.69                     18.17   \n",
       "\n",
       "           fpkm.GTEx.spleen  fpkm.GTEx.stomach  fpkm.GTEx.testis  \\\n",
       "gene_name                                                          \n",
       "SNAP25                 0.05               2.23              3.48   \n",
       "HSPA4L                 0.57               1.44             57.43   \n",
       "RAB40B                 2.49               6.62              3.46   \n",
       "RAB11FIP1              7.10              10.22              4.41   \n",
       "\n",
       "           fpkm.GTEx.thyroid  fpkm.GTEx.uterus  fpkm.GTEx.vagina  \n",
       "gene_name                                                         \n",
       "SNAP25                  0.70              5.51              1.23  \n",
       "HSPA4L                  1.62              0.87              4.79  \n",
       "RAB40B                  3.70              4.54              4.87  \n",
       "RAB11FIP1               6.65              2.25              6.86  \n",
       "\n",
       "[4 rows x 68 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "secM[secM.index.isin(not_in_cho3)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensgid</th>\n",
       "      <th>Subsystem</th>\n",
       "      <th>category.HPA</th>\n",
       "      <th>tissues.HPA</th>\n",
       "      <th>category.GTEx</th>\n",
       "      <th>tissues.GTEx</th>\n",
       "      <th>fpkm.HPA.adrenal</th>\n",
       "      <th>fpkm.HPA.appendices</th>\n",
       "      <th>fpkm.HPA.bonemarrow</th>\n",
       "      <th>fpkm.HPA.brain</th>\n",
       "      <th>...</th>\n",
       "      <th>fpkm.GTEx.prostate</th>\n",
       "      <th>fpkm.GTEx.skeletalmuscle</th>\n",
       "      <th>fpkm.GTEx.skin</th>\n",
       "      <th>fpkm.GTEx.smallintestine</th>\n",
       "      <th>fpkm.GTEx.spleen</th>\n",
       "      <th>fpkm.GTEx.stomach</th>\n",
       "      <th>fpkm.GTEx.testis</th>\n",
       "      <th>fpkm.GTEx.thyroid</th>\n",
       "      <th>fpkm.GTEx.uterus</th>\n",
       "      <th>fpkm.GTEx.vagina</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HSPA4L</th>\n",
       "      <td>ENSG00000164070</td>\n",
       "      <td>ERAD</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>9.51</td>\n",
       "      <td>0.96</td>\n",
       "      <td>1.03</td>\n",
       "      <td>7.31</td>\n",
       "      <td>...</td>\n",
       "      <td>2.17</td>\n",
       "      <td>0.64</td>\n",
       "      <td>5.85</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.44</td>\n",
       "      <td>57.43</td>\n",
       "      <td>1.62</td>\n",
       "      <td>0.87</td>\n",
       "      <td>4.79</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ensgid Subsystem     category.HPA tissues.HPA  \\\n",
       "gene_name                                                           \n",
       "HSPA4L     ENSG00000164070      ERAD  Tissue enriched      testis   \n",
       "\n",
       "             category.GTEx tissues.GTEx  fpkm.HPA.adrenal  \\\n",
       "gene_name                                                   \n",
       "HSPA4L     Tissue enriched       testis              9.51   \n",
       "\n",
       "           fpkm.HPA.appendices  fpkm.HPA.bonemarrow  fpkm.HPA.brain  \\\n",
       "gene_name                                                             \n",
       "HSPA4L                    0.96                 1.03            7.31   \n",
       "\n",
       "                 ...         fpkm.GTEx.prostate  fpkm.GTEx.skeletalmuscle  \\\n",
       "gene_name        ...                                                        \n",
       "HSPA4L           ...                       2.17                      0.64   \n",
       "\n",
       "           fpkm.GTEx.skin  fpkm.GTEx.smallintestine  fpkm.GTEx.spleen  \\\n",
       "gene_name                                                               \n",
       "HSPA4L               5.85                      0.99              0.57   \n",
       "\n",
       "           fpkm.GTEx.stomach  fpkm.GTEx.testis  fpkm.GTEx.thyroid  \\\n",
       "gene_name                                                           \n",
       "HSPA4L                  1.44             57.43               1.62   \n",
       "\n",
       "           fpkm.GTEx.uterus  fpkm.GTEx.vagina  \n",
       "gene_name                                      \n",
       "HSPA4L                 0.87              4.79  \n",
       "\n",
       "[1 rows x 68 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "secM[secM.index.isin(not_in_cho4)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensgid</th>\n",
       "      <th>Subsystem</th>\n",
       "      <th>category.HPA</th>\n",
       "      <th>tissues.HPA</th>\n",
       "      <th>category.GTEx</th>\n",
       "      <th>tissues.GTEx</th>\n",
       "      <th>fpkm.HPA.adrenal</th>\n",
       "      <th>fpkm.HPA.appendices</th>\n",
       "      <th>fpkm.HPA.bonemarrow</th>\n",
       "      <th>fpkm.HPA.brain</th>\n",
       "      <th>...</th>\n",
       "      <th>fpkm.GTEx.prostate</th>\n",
       "      <th>fpkm.GTEx.skeletalmuscle</th>\n",
       "      <th>fpkm.GTEx.skin</th>\n",
       "      <th>fpkm.GTEx.smallintestine</th>\n",
       "      <th>fpkm.GTEx.spleen</th>\n",
       "      <th>fpkm.GTEx.stomach</th>\n",
       "      <th>fpkm.GTEx.testis</th>\n",
       "      <th>fpkm.GTEx.thyroid</th>\n",
       "      <th>fpkm.GTEx.uterus</th>\n",
       "      <th>fpkm.GTEx.vagina</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SNAP25</th>\n",
       "      <td>ENSG00000132639</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>brain</td>\n",
       "      <td>Group enriched</td>\n",
       "      <td>brain.cerebellum,brain.cerebrum</td>\n",
       "      <td>39.99</td>\n",
       "      <td>3.75</td>\n",
       "      <td>0.03</td>\n",
       "      <td>916.53</td>\n",
       "      <td>...</td>\n",
       "      <td>4.32</td>\n",
       "      <td>0.27</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.45</td>\n",
       "      <td>0.05</td>\n",
       "      <td>2.23</td>\n",
       "      <td>3.48</td>\n",
       "      <td>0.70</td>\n",
       "      <td>5.51</td>\n",
       "      <td>1.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HSPA4L</th>\n",
       "      <td>ENSG00000164070</td>\n",
       "      <td>ERAD</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>9.51</td>\n",
       "      <td>0.96</td>\n",
       "      <td>1.03</td>\n",
       "      <td>7.31</td>\n",
       "      <td>...</td>\n",
       "      <td>2.17</td>\n",
       "      <td>0.64</td>\n",
       "      <td>5.85</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.44</td>\n",
       "      <td>57.43</td>\n",
       "      <td>1.62</td>\n",
       "      <td>0.87</td>\n",
       "      <td>4.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB40B</th>\n",
       "      <td>ENSG00000141542</td>\n",
       "      <td>Post-Golgi trafficking</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mixed</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.37</td>\n",
       "      <td>1.85</td>\n",
       "      <td>0.42</td>\n",
       "      <td>26.88</td>\n",
       "      <td>...</td>\n",
       "      <td>5.74</td>\n",
       "      <td>7.39</td>\n",
       "      <td>5.88</td>\n",
       "      <td>2.93</td>\n",
       "      <td>2.49</td>\n",
       "      <td>6.62</td>\n",
       "      <td>3.46</td>\n",
       "      <td>3.70</td>\n",
       "      <td>4.54</td>\n",
       "      <td>4.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB11FIP1</th>\n",
       "      <td>ENSG00000156675</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.35</td>\n",
       "      <td>16.61</td>\n",
       "      <td>30.77</td>\n",
       "      <td>0.88</td>\n",
       "      <td>...</td>\n",
       "      <td>4.80</td>\n",
       "      <td>1.85</td>\n",
       "      <td>13.69</td>\n",
       "      <td>18.17</td>\n",
       "      <td>7.10</td>\n",
       "      <td>10.22</td>\n",
       "      <td>4.41</td>\n",
       "      <td>6.65</td>\n",
       "      <td>2.25</td>\n",
       "      <td>6.86</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ensgid               Subsystem      category.HPA  \\\n",
       "gene_name                                                              \n",
       "SNAP25     ENSG00000132639  Post-Golgi trafficking   Tissue enriched   \n",
       "HSPA4L     ENSG00000164070                    ERAD   Tissue enriched   \n",
       "RAB40B     ENSG00000141542  Post-Golgi trafficking             Mixed   \n",
       "RAB11FIP1  ENSG00000156675  Trafficking regulation  Expressed in all   \n",
       "\n",
       "          tissues.HPA     category.GTEx                     tissues.GTEx  \\\n",
       "gene_name                                                                  \n",
       "SNAP25          brain    Group enriched  brain.cerebellum,brain.cerebrum   \n",
       "HSPA4L         testis   Tissue enriched                           testis   \n",
       "RAB40B            NaN             Mixed                              NaN   \n",
       "RAB11FIP1         NaN  Expressed in all                              NaN   \n",
       "\n",
       "           fpkm.HPA.adrenal  fpkm.HPA.appendices  fpkm.HPA.bonemarrow  \\\n",
       "gene_name                                                               \n",
       "SNAP25                39.99                 3.75                 0.03   \n",
       "HSPA4L                 9.51                 0.96                 1.03   \n",
       "RAB40B                 5.37                 1.85                 0.42   \n",
       "RAB11FIP1              4.35                16.61                30.77   \n",
       "\n",
       "           fpkm.HPA.brain        ...         fpkm.GTEx.prostate  \\\n",
       "gene_name                        ...                              \n",
       "SNAP25             916.53        ...                       4.32   \n",
       "HSPA4L               7.31        ...                       2.17   \n",
       "RAB40B              26.88        ...                       5.74   \n",
       "RAB11FIP1            0.88        ...                       4.80   \n",
       "\n",
       "           fpkm.GTEx.skeletalmuscle  fpkm.GTEx.skin  fpkm.GTEx.smallintestine  \\\n",
       "gene_name                                                                       \n",
       "SNAP25                         0.27            0.57                      1.45   \n",
       "HSPA4L                         0.64            5.85                      0.99   \n",
       "RAB40B                         7.39            5.88                      2.93   \n",
       "RAB11FIP1                      1.85           13.69                     18.17   \n",
       "\n",
       "           fpkm.GTEx.spleen  fpkm.GTEx.stomach  fpkm.GTEx.testis  \\\n",
       "gene_name                                                          \n",
       "SNAP25                 0.05               2.23              3.48   \n",
       "HSPA4L                 0.57               1.44             57.43   \n",
       "RAB40B                 2.49               6.62              3.46   \n",
       "RAB11FIP1              7.10              10.22              4.41   \n",
       "\n",
       "           fpkm.GTEx.thyroid  fpkm.GTEx.uterus  fpkm.GTEx.vagina  \n",
       "gene_name                                                         \n",
       "SNAP25                  0.70              5.51              1.23  \n",
       "HSPA4L                  1.62              0.87              4.79  \n",
       "RAB40B                  3.70              4.54              4.87  \n",
       "RAB11FIP1               6.65              2.25              6.86  \n",
       "\n",
       "[4 rows x 68 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "secM[secM.index.isin(not_in_cho5)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Golgi glycosylation genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.sum(secM.loc[secM_silenced_in_CHO.index,'Subsystem'] == 'Golgi glycosylation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "318"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### machinery expressed in CHO\n",
    "len(secM_in_CHO.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SecM genes not in CHO but in all other tissues\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensgid</th>\n",
       "      <th>Subsystem</th>\n",
       "      <th>category.HPA</th>\n",
       "      <th>tissues.HPA</th>\n",
       "      <th>category.GTEx</th>\n",
       "      <th>tissues.GTEx</th>\n",
       "      <th>fpkm.HPA.adrenal</th>\n",
       "      <th>fpkm.HPA.appendices</th>\n",
       "      <th>fpkm.HPA.bonemarrow</th>\n",
       "      <th>fpkm.HPA.brain</th>\n",
       "      <th>...</th>\n",
       "      <th>fpkm.GTEx.prostate</th>\n",
       "      <th>fpkm.GTEx.skeletalmuscle</th>\n",
       "      <th>fpkm.GTEx.skin</th>\n",
       "      <th>fpkm.GTEx.smallintestine</th>\n",
       "      <th>fpkm.GTEx.spleen</th>\n",
       "      <th>fpkm.GTEx.stomach</th>\n",
       "      <th>fpkm.GTEx.testis</th>\n",
       "      <th>fpkm.GTEx.thyroid</th>\n",
       "      <th>fpkm.GTEx.uterus</th>\n",
       "      <th>fpkm.GTEx.vagina</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ERLEC1</th>\n",
       "      <td>ENSG00000068912</td>\n",
       "      <td>Protein folding</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.69</td>\n",
       "      <td>29.09</td>\n",
       "      <td>16.54</td>\n",
       "      <td>38.64</td>\n",
       "      <td>...</td>\n",
       "      <td>27.40</td>\n",
       "      <td>11.73</td>\n",
       "      <td>14.55</td>\n",
       "      <td>20.24</td>\n",
       "      <td>22.70</td>\n",
       "      <td>21.47</td>\n",
       "      <td>30.54</td>\n",
       "      <td>32.03</td>\n",
       "      <td>25.53</td>\n",
       "      <td>22.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HSPA4L</th>\n",
       "      <td>ENSG00000164070</td>\n",
       "      <td>ERAD</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>Tissue enriched</td>\n",
       "      <td>testis</td>\n",
       "      <td>9.51</td>\n",
       "      <td>0.96</td>\n",
       "      <td>1.03</td>\n",
       "      <td>7.31</td>\n",
       "      <td>...</td>\n",
       "      <td>2.17</td>\n",
       "      <td>0.64</td>\n",
       "      <td>5.85</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.57</td>\n",
       "      <td>1.44</td>\n",
       "      <td>57.43</td>\n",
       "      <td>1.62</td>\n",
       "      <td>0.87</td>\n",
       "      <td>4.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB11FIP1</th>\n",
       "      <td>ENSG00000156675</td>\n",
       "      <td>Trafficking regulation</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Expressed in all</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.35</td>\n",
       "      <td>16.61</td>\n",
       "      <td>30.77</td>\n",
       "      <td>0.88</td>\n",
       "      <td>...</td>\n",
       "      <td>4.80</td>\n",
       "      <td>1.85</td>\n",
       "      <td>13.69</td>\n",
       "      <td>18.17</td>\n",
       "      <td>7.10</td>\n",
       "      <td>10.22</td>\n",
       "      <td>4.41</td>\n",
       "      <td>6.65</td>\n",
       "      <td>2.25</td>\n",
       "      <td>6.86</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 68 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ensgid               Subsystem      category.HPA  \\\n",
       "ERLEC1     ENSG00000068912         Protein folding  Expressed in all   \n",
       "HSPA4L     ENSG00000164070                    ERAD   Tissue enriched   \n",
       "RAB11FIP1  ENSG00000156675  Trafficking regulation  Expressed in all   \n",
       "\n",
       "          tissues.HPA     category.GTEx tissues.GTEx  fpkm.HPA.adrenal  \\\n",
       "ERLEC1            NaN  Expressed in all          NaN             44.69   \n",
       "HSPA4L         testis   Tissue enriched       testis              9.51   \n",
       "RAB11FIP1         NaN  Expressed in all          NaN              4.35   \n",
       "\n",
       "           fpkm.HPA.appendices  fpkm.HPA.bonemarrow  fpkm.HPA.brain  \\\n",
       "ERLEC1                   29.09                16.54           38.64   \n",
       "HSPA4L                    0.96                 1.03            7.31   \n",
       "RAB11FIP1                16.61                30.77            0.88   \n",
       "\n",
       "                 ...         fpkm.GTEx.prostate  fpkm.GTEx.skeletalmuscle  \\\n",
       "ERLEC1           ...                      27.40                     11.73   \n",
       "HSPA4L           ...                       2.17                      0.64   \n",
       "RAB11FIP1        ...                       4.80                      1.85   \n",
       "\n",
       "           fpkm.GTEx.skin  fpkm.GTEx.smallintestine  fpkm.GTEx.spleen  \\\n",
       "ERLEC1              14.55                     20.24             22.70   \n",
       "HSPA4L               5.85                      0.99              0.57   \n",
       "RAB11FIP1           13.69                     18.17              7.10   \n",
       "\n",
       "           fpkm.GTEx.stomach  fpkm.GTEx.testis  fpkm.GTEx.thyroid  \\\n",
       "ERLEC1                 21.47             30.54              32.03   \n",
       "HSPA4L                  1.44             57.43               1.62   \n",
       "RAB11FIP1              10.22              4.41               6.65   \n",
       "\n",
       "           fpkm.GTEx.uterus  fpkm.GTEx.vagina  \n",
       "ERLEC1                25.53             22.53  \n",
       "HSPA4L                 0.87              4.79  \n",
       "RAB11FIP1              2.25              6.86  \n",
       "\n",
       "[3 rows x 68 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print('SecM genes not in CHO but in all other tissues')\n",
    "secM.loc[only_not_in_cho[only_not_in_cho.index.isin(secM.index)].index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
