{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T20:28:35.814166Z",
     "start_time": "2020-04-19T20:28:34.841025Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/ipykernel_launcher.py:7: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  import sys\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of tissues:  13\n",
      "/data2/resources/software/homer/v4.11_10242019/bin//homer\r\n"
     ]
    }
   ],
   "source": [
    "## Parameters specific to where your folders are and your data\n",
    "parameter_file = '../parameters/params.yaml'\n",
    "import yaml\n",
    "import sys\n",
    "import os\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "#p = dic2obj(**doc)\n",
    "\n",
    "data_folder = doc['data_folder']\n",
    "tissues = doc['tissues'].split(',')\n",
    "sys.path.append(doc['pipeline_path'])\n",
    "ref_fa = doc['ref_fa']\n",
    "#anno_gff=doc['annotation']\n",
    "os.chdir(doc[\"results\"])\n",
    "#mRNA_peak_file = doc[\"mRNA_peak_file\"]\n",
    "#tss_annotation = doc['tss_annotation']\n",
    "\n",
    "anno_gff = doc[\"alt_annotation\"]\n",
    "import sys\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import pickle\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "from matplotlib import pyplot as plt\n",
    "from itertools import product\n",
    "import glob\n",
    "import re\n",
    "from os.path import join\n",
    "from matplotlib_venn import venn2\n",
    "from matplotlib import rcParams\n",
    "import inspect\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "sys.setrecursionlimit(3000)\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "rcParams['figure.figsize'] = 8, 6\n",
    "import tqdm\n",
    "\n",
    "from os.path import basename\n",
    "##mpl.use('Agg')\n",
    "#mpl.style.use('ggplot')\n",
    "#mpl.style.use('fivethirtyeight')\n",
    "from Homer import *\n",
    "import helper\n",
    "import create_output\n",
    "print('Number of tissues: ',len(tissues))\n",
    "\n",
    "homer_path =  \"/data2/resources/software/homer/v4.11_10242019/bin/\"\n",
    "os.environ['PATH'] = f\"{homer_path}:{os.environ['PATH']}\"\n",
    "!which homer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T20:28:35.838609Z",
     "start_time": "2020-04-19T20:28:35.819394Z"
    }
   },
   "outputs": [],
   "source": [
    "from numpanpar import parallel_ar as parar"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Indir of tag files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T20:28:35.865152Z",
     "start_time": "2020-04-19T20:28:35.841245Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{('BMDM1hKLA', 'START'): [],\n",
       " ('BMDM1hKLA',\n",
       "  'GROCap'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1'],\n",
       " ('BMDMwt', 'START'): [],\n",
       " ('BMDMwt',\n",
       "  'GROCap'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1'],\n",
       " ('Brain',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3', '/data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1', '/data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2'],\n",
       " ('Brain',\n",
       "  'GROCap'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1'],\n",
       " ('FemaleReproductive',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2', '/data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1'],\n",
       " ('FemaleReproductive', 'GROCap'): [],\n",
       " ('Heart',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1', '/data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2'],\n",
       " ('Heart', 'GROCap'): [],\n",
       " ('Kidney',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1', '/data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2'],\n",
       " ('Kidney',\n",
       "  'GROCap'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1'],\n",
       " ('Liver',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1', '/data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2'],\n",
       " ('Liver',\n",
       "  'GROCap'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1'],\n",
       " ('Lung',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1', '/data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2'],\n",
       " ('Lung',\n",
       "  'GROCap'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1'],\n",
       " ('MiscOrgans',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1', '/data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2'],\n",
       " ('MiscOrgans', 'GROCap'): [],\n",
       " ('Muscle',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1', '/data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2'],\n",
       " ('Muscle', 'GROCap'): [],\n",
       " ('Pancreas',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2', '/data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1'],\n",
       " ('Pancreas', 'GROCap'): [],\n",
       " ('Spleen',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1', '/data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2'],\n",
       " ('Spleen', 'GROCap'): [],\n",
       " ('CHO',\n",
       "  'START'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1'],\n",
       " ('CHO',\n",
       "  'GROCap'): ['/data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1', '/data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2']}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tag_folders = dict()\n",
    "for t in tissues:\n",
    "    for exp in [\"START\", \"GROCap\"]:  \n",
    "        #print(glob.glob(join(doc[\"data_folder\"], t, ))\n",
    "        files = glob.glob(join(doc[\"data_folder\"],t,exp,'f03_tags','*'+exp+'*'))\n",
    "        tag_folders[(t,exp)] = list(filter(lambda x: 'STARTinput' not in x, files)) #remove input\n",
    "tag_folders"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Old/new peaks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T20:28:35.881091Z",
     "start_time": "2020-04-19T20:28:35.867093Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "## Create RefSeq peak file for this\n",
    "eTSS = \"analysis_ncbi_picr/output/TSS1.exp.bed\"\n",
    "meta_f = \"analysis_ncbi_picr/output/TSS1.exp.meta\"\n",
    "rTSS = \"alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed\"\n",
    "#mrna_filt = \"Results/Figures/Figure3/A.mrna\"\n",
    "#create_output.exp_bed_to_refseq(peak_f,meta_f,refseq_f=mRNA_peak_file,save_f=save_f,is_unique=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T20:28:35.901043Z",
     "start_time": "2020-04-19T20:28:35.882776Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'tags_on_annotation'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "outdir = \"tags_on_annotation\"\n",
    "if not os.path.exists(outdir):\n",
    "    os.mkdir(outdir)\n",
    "    \n",
    "outdir_tis = join(outdir, \"merged_tissue_experiment\")\n",
    "outdir_exp = join(outdir,\"new\")\n",
    "outdir_ref = join(outdir,\"old\")\n",
    "\n",
    "if not os.path.exists(outdir_exp):\n",
    "    os.mkdir(outdir_exp)\n",
    "if not os.path.exists(outdir_ref):\n",
    "    os.mkdir(outdir_ref)\n",
    "outdir\n",
    "\n",
    "#save_f = \"Results/Figures/Figure3/A.TSS1_mrna\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T19:43:41.308954Z",
     "start_time": "2020-04-19T19:39:09.663632Z"
    },
    "deletable": false,
    "editable": false,
    "run_control": {
     "frozen": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BMDM1hKLA START\n",
      "BMDM1hKLA GROCap\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tTotal Tags = 7575061.0\n",
      "\tTotal Positions = 763804\n",
      "\tAverage tag length = 54.6\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 9.814\n",
      "\tFragment Length Estimate: 75\n",
      "\tPeak Width Estimate: 1256\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 11.8\n",
      "\t\tDiff strand fold enrichment: 1.1\n",
      "\t\tSame / Diff fold enrichment: 25.6\n",
      "\n",
      "\t\tGuessing sample is strand specific RNA-Seq\n",
      "\t\tSetting fragment length estimate to 75, edit tagInfo.txt to change\n",
      "\n",
      "BMDMwt START\n",
      "BMDMwt GROCap\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tTotal Tags = 5302781.0\n",
      "\tTotal Positions = 708952\n",
      "\tAverage tag length = 57.0\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 7.444\n",
      "\tFragment Length Estimate: 73\n",
      "\tPeak Width Estimate: 866\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 11.5\n",
      "\t\tDiff strand fold enrichment: 1.4\n",
      "\t\tSame / Diff fold enrichment: 26.6\n",
      "\n",
      "\t\tGuessing sample is strand specific RNA-Seq\n",
      "\t\tSetting fragment length estimate to 75, edit tagInfo.txt to change\n",
      "\n",
      "Brain START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Brain_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Brain_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2322766414\n",
      "\tTotal Tags = 68030044.0\n",
      "\tTotal Positions = 8568080\n",
      "\tAverage tag length = 34.7\n",
      "\tMedian tags per position = 2 (ideal: 1)\n",
      "\tAverage tags per position = 6.484\n",
      "\t\t!! Might have some clonal amplification in this sample if sonication was used\n",
      "\t\tIf using BED files, try using -forceBED to ignore the 5th column\n",
      "\t\tIgnore if analyzing RNA, MNase, etc. data\n",
      "\tFragment Length Estimate: 400\n",
      "\tPeak Width Estimate: 466\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 7.7\n",
      "\t\tDiff strand fold enrichment: 4.4\n",
      "\t\tSame / Diff fold enrichment: 1.8\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\n",
      "Brain GROCap\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Brain_GROCap -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Brain_GROCap and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tTotal Tags = 25213601.0\n",
      "\tTotal Positions = 1566029\n",
      "\tAverage tag length = 54.3\n",
      "\tMedian tags per position = 11 (ideal: 1)\n",
      "\tAverage tags per position = 15.599\n",
      "\t\t!! Might have some clonal amplification in this sample if sonication was used\n",
      "\t\tIf this is ChIP-Seq using sonicated fragments, consider adding the option \"-tbp 1\"\n",
      "\t\tIgnore if analyzing RNA, MNase, etc. data\n",
      "\tFragment Length Estimate: 230\n",
      "\tPeak Width Estimate: 643\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 11.0\n",
      "\t\tDiff strand fold enrichment: 3.8\n",
      "\t\tSame / Diff fold enrichment: 3.2\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\n",
      "FemaleReproductive START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2314594756\n",
      "\tTotal Tags = 10371454.0\n",
      "\tTotal Positions = 2843338\n",
      "\tAverage tag length = 32.6\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 3.402\n",
      "\tFragment Length Estimate: 402\n",
      "\tPeak Width Estimate: 1028\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 7.4\n",
      "\t\tDiff strand fold enrichment: 3.8\n",
      "\t\tSame / Diff fold enrichment: 1.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\t\tMay be strand-unspecific RNA-Seq (i.e. fragmented cDNA lib).  If so,\n",
      "\t\tmanually set the fragment length estimate in tagInfo.txt\n",
      "\n",
      "FemaleReproductive GROCap\n",
      "Heart START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Heart_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Heart_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2312292173\n",
      "\tTotal Tags = 11453438.0\n",
      "\tTotal Positions = 922127\n",
      "\tAverage tag length = 28.1\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 10.760\n",
      "\tFragment Length Estimate: 58\n",
      "\tPeak Width Estimate: 45\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 46.3\n",
      "\t\tDiff strand fold enrichment: 1.0\n",
      "\t\tSame / Diff fold enrichment: 65.4\n",
      "\n",
      "\t\tGuessing sample is strand specific RNA-Seq\n",
      "\t\tSetting fragment length estimate to 75, edit tagInfo.txt to change\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Heart GROCap\n",
      "Kidney START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Kidney_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Kidney_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2299059615\n",
      "\tTotal Tags = 5088225.0\n",
      "\tTotal Positions = 1119645\n",
      "\tAverage tag length = 34.5\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 4.002\n",
      "\tFragment Length Estimate: 574\n",
      "\tPeak Width Estimate: 762\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 8.3\n",
      "\t\tDiff strand fold enrichment: 4.0\n",
      "\t\tSame / Diff fold enrichment: 2.7\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\t\tMay be strand-unspecific RNA-Seq (i.e. fragmented cDNA lib).  If so,\n",
      "\t\tmanually set the fragment length estimate in tagInfo.txt\n",
      "\n",
      "Kidney GROCap\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Kidney_GROCap -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Kidney_GROCap and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tTotal Tags = 19091807.0\n",
      "\tTotal Positions = 1981029\n",
      "\tAverage tag length = 49.8\n",
      "\tMedian tags per position = 6 (ideal: 1)\n",
      "\tAverage tags per position = 9.591\n",
      "\t\t!! Might have some clonal amplification in this sample if sonication was used\n",
      "\t\tIf this is ChIP-Seq using sonicated fragments, consider adding the option \"-tbp 1\"\n",
      "\t\tIgnore if analyzing RNA, MNase, etc. data\n",
      "\tFragment Length Estimate: 223\n",
      "\tPeak Width Estimate: 1065\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 9.6\n",
      "\t\tDiff strand fold enrichment: 3.0\n",
      "\t\tSame / Diff fold enrichment: 3.2\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\n",
      "Liver START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Liver_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Liver_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2299831656\n",
      "\tTotal Tags = 3918429.0\n",
      "\tTotal Positions = 712528\n",
      "\tAverage tag length = 31.0\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 4.338\n",
      "\tFragment Length Estimate: 577\n",
      "\tPeak Width Estimate: 646\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 6.7\n",
      "\t\tDiff strand fold enrichment: 3.3\n",
      "\t\tSame / Diff fold enrichment: 2.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\t\tMay be strand-unspecific RNA-Seq (i.e. fragmented cDNA lib).  If so,\n",
      "\t\tmanually set the fragment length estimate in tagInfo.txt\n",
      "\n",
      "Liver GROCap\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Liver_GROCap -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Liver_GROCap and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tTotal Tags = 1780403.0\n",
      "\tTotal Positions = 559905\n",
      "\tAverage tag length = 41.9\n",
      "\tMedian tags per position = 2 (ideal: 1)\n",
      "\tAverage tags per position = 3.174\n",
      "\t\t!! Might have some clonal amplification in this sample if sonication was used\n",
      "\t\tIf using BED files, try using -forceBED to ignore the 5th column\n",
      "\t\tIgnore if analyzing RNA, MNase, etc. data\n",
      "\tFragment Length Estimate: 801\n",
      "\tPeak Width Estimate: 1043\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 2.2\n",
      "\t\tDiff strand fold enrichment: 2.9\n",
      "\t\tSame / Diff fold enrichment: 2.3\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "Lung START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Lung_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Lung_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2320924835\n",
      "\tTotal Tags = 14136927.0\n",
      "\tTotal Positions = 2078052\n",
      "\tAverage tag length = 28.7\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 5.856\n",
      "\tFragment Length Estimate: 408\n",
      "\tPeak Width Estimate: 1013\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 8.5\n",
      "\t\tDiff strand fold enrichment: 3.8\n",
      "\t\tSame / Diff fold enrichment: 2.5\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\t\tMay be strand-unspecific RNA-Seq (i.e. fragmented cDNA lib).  If so,\n",
      "\t\tmanually set the fragment length estimate in tagInfo.txt\n",
      "\n",
      "Lung GROCap\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Lung_GROCap -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Lung_GROCap and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tTotal Tags = 15298799.0\n",
      "\tTotal Positions = 1562301\n",
      "\tAverage tag length = 55.5\n",
      "\tMedian tags per position = 6 (ideal: 1)\n",
      "\tAverage tags per position = 9.492\n",
      "\t\t!! Might have some clonal amplification in this sample if sonication was used\n",
      "\t\tIf this is ChIP-Seq using sonicated fragments, consider adding the option \"-tbp 1\"\n",
      "\t\tIgnore if analyzing RNA, MNase, etc. data\n",
      "\tFragment Length Estimate: 395\n",
      "\tPeak Width Estimate: 696\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 7.6\n",
      "\t\tDiff strand fold enrichment: 4.3\n",
      "\t\tSame / Diff fold enrichment: 1.9\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MiscOrgans START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/MiscOrgans_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/MiscOrgans_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2314043005\n",
      "\tTotal Tags = 9732143.0\n",
      "\tTotal Positions = 2571732\n",
      "\tAverage tag length = 29.9\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 3.091\n",
      "\tFragment Length Estimate: 397\n",
      "\tPeak Width Estimate: 773\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 8.0\n",
      "\t\tDiff strand fold enrichment: 3.8\n",
      "\t\tSame / Diff fold enrichment: 2.0\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\n",
      "MiscOrgans GROCap\n",
      "Muscle START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Muscle_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Muscle_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2312779702\n",
      "\tTotal Tags = 16938016.0\n",
      "\tTotal Positions = 2552371\n",
      "\tAverage tag length = 31.6\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 5.890\n",
      "\tFragment Length Estimate: 404\n",
      "\tPeak Width Estimate: 709\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 7.1\n",
      "\t\tDiff strand fold enrichment: 3.6\n",
      "\t\tSame / Diff fold enrichment: 2.5\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\t\tMay be strand-unspecific RNA-Seq (i.e. fragmented cDNA lib).  If so,\n",
      "\t\tmanually set the fragment length estimate in tagInfo.txt\n",
      "\n",
      "Muscle GROCap\n",
      "Pancreas START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Pancreas_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Pancreas_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2308808363\n",
      "\tTotal Tags = 11255613.0\n",
      "\tTotal Positions = 845181\n",
      "\tAverage tag length = 26.9\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 7.226\n",
      "\tFragment Length Estimate: 48\n",
      "\tPeak Width Estimate: 46\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 54.3\n",
      "\t\tDiff strand fold enrichment: 1.3\n",
      "\t\tSame / Diff fold enrichment: 61.7\n",
      "\n",
      "\t\tGuessing sample is strand specific RNA-Seq\n",
      "\t\tSetting fragment length estimate to 75, edit tagInfo.txt to change\n",
      "\n",
      "Pancreas GROCap\n",
      "Spleen START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/Spleen_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/Spleen_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2305944238\n",
      "\tTotal Tags = 16806188.0\n",
      "\tTotal Positions = 2034208\n",
      "\tAverage tag length = 32.4\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 5.444\n",
      "\tFragment Length Estimate: 399\n",
      "\tPeak Width Estimate: 905\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 7.5\n",
      "\t\tDiff strand fold enrichment: 3.7\n",
      "\t\tSame / Diff fold enrichment: 2.2\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\n",
      "Spleen GROCap\n",
      "CHO START\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/CHO_START -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/CHO_START and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tTotal Tags = 20082672.0\n",
      "\tTotal Positions = 6452427\n",
      "\tAverage tag length = 35.7\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 2.844\n",
      "\tFragment Length Estimate: 237\n",
      "\tPeak Width Estimate: 696\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 10.4\n",
      "\t\tDiff strand fold enrichment: 3.5\n",
      "\t\tSame / Diff fold enrichment: 2.2\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\n",
      "CHO GROCap\n",
      "makeTagDirectory tags_on_annotation/merged_tissue_experiment/CHO_GROCap -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2\n",
      "\n",
      "\tCreating directory: tags_on_annotation/merged_tissue_experiment/CHO_GROCap and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2332523660\n",
      "\tTotal Tags = 48117814.0\n",
      "\tTotal Positions = 18816347\n",
      "\tAverage tag length = 63.7\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 2.557\n",
      "\tFragment Length Estimate: 231\n",
      "\tPeak Width Estimate: 565\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 5.6\n",
      "\t\tDiff strand fold enrichment: 2.8\n",
      "\t\tSame / Diff fold enrichment: 1.8\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq - uneven enrichment between same strand and\n",
      "\t\tdifferent strands - may have problems such as clonal amplification.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for t,exp in tag_folders:\n",
    "    print(t,exp)\n",
    "    # Merge for each tissue/exp    \n",
    "    tag_out = join(outdir_tis, t+'_'+exp)    \n",
    "    tag_files = tag_folders[(t,exp)]\n",
    "    if len(tag_files) > 0:\n",
    "        cmd = 'makeTagDirectory {out_f} -single -d {files}'.format(out_f=tag_out,\n",
    "                                                                   files = ' '.join(tag_files))\n",
    "        print(cmd)\n",
    "        !{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T20:43:16.842114Z",
     "start_time": "2020-04-19T20:43:16.804694Z"
    }
   },
   "outputs": [],
   "source": [
    "def run_hist_compare_exp_ref(t_exp_names, eTSS, rTSS, anno_gff, ref_fa, outdir_exp, outdir_ref, outdir):\n",
    "    for t,exp in t_exp_names:\n",
    "        print(t,exp)\n",
    "        \n",
    "        # Merge for each tissue/exp    \n",
    "        tag_out = join(outdir_tis, t+'_'+exp)\n",
    "        if len(glob.glob(tag_out)) == 0:\n",
    "            continue\n",
    "\n",
    "        print(tag_out)\n",
    "\n",
    "        # Make for experimental\n",
    "        hist_out_exp = join(outdir_exp, t+'_'+exp+'.tsv')\n",
    "        hist(tag_out,hist_out_exp,ref_fa,anno_gff,mode='peak',peak=eTSS,region=2000,res=1,pc=3)\n",
    "        hist_plot(hist_out_exp)\n",
    "        #heat_df = heat_plot(hist_out_exp+'MatS',save_f = hist_out_exp + '.heat.png' )\n",
    "\n",
    "        hist_out_ref = join(outdir_ref, t+'_'+exp+'.tsv')\n",
    "        \n",
    "        hist(tag_out,hist_out_ref,ref_fa,anno_gff,mode='peak',peak=rTSS,region=2000,res=1,pc=3)\n",
    "        hist_plot(hist_out_ref)\n",
    "        \n",
    "        #heat_df = heat_plot(hist_out_ref+'MatS',save_f = hist_out_ref + '.heat.png' )\n",
    "        \n",
    "        wrap_hist_plot([hist_out_exp,hist_out_ref] ,\n",
    "                        hist_save=join(outdir, t+'_'+exp),\n",
    "                        names=[\"Experimental\",\"RefSeq\"],to_norm=True)\n",
    "    return\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T20:40:36.366345Z",
     "start_time": "2020-04-19T20:28:35.923747Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BMDMwt START\n",
      "BMDM1hKLA START\n",
      "BMDM1hKLA GROCap\n",
      "Brain START\n",
      "Brain GROCap\n",
      "FemaleReproductive GROCap\n",
      "BMDMwt GROCap\n",
      "FemaleReproductive START\n",
      "Heart START\n",
      "tags_on_annotation/merged_tissue_experiment/Brain_START\n",
      "Lung START\n",
      "tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap\n",
      "tags_on_annotation/merged_tissue_experiment/Brain_GROCap\n",
      "MiscOrgans START\n",
      "CHO START\n",
      "Liver GROCap\n",
      "Lung GROCap\n",
      "CHO GROCap\n",
      "tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap\n",
      "tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START\n",
      "MiscOrgans GROCap\n",
      "Pancreas START\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Brain_START -pc 3 > tags_on_annotation/new/Brain_START.tsv\n",
      "tags_on_annotation/merged_tissue_experiment/Heart_START\n",
      "Muscle START\n",
      "Muscle GROCap\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap -pc 3 > tags_on_annotation/new/BMDM1hKLA_GROCap.tsv\n",
      "Spleen START\n",
      "tags_on_annotation/merged_tissue_experiment/Lung_START\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Brain_GROCap -pc 3 > tags_on_annotation/new/Brain_GROCap.tsv\n",
      "tags_on_annotation/merged_tissue_experiment/CHO_START\n",
      "tags_on_annotation/merged_tissue_experiment/MiscOrgans_START\n",
      "Pancreas GROCap\n",
      "Spleen GROCap\n",
      "tags_on_annotation/merged_tissue_experiment/Lung_GROCap\n",
      "tags_on_annotation/merged_tissue_experiment/Liver_GROCap\n",
      "tags_on_annotation/merged_tissue_experiment/CHO_GROCap\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap -pc 3 > tags_on_annotation/new/BMDMwt_GROCap.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START -pc 3 > tags_on_annotation/new/FemaleReproductive_START.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Heart_START -pc 3 > tags_on_annotation/new/Heart_START.tsv\n",
      "tags_on_annotation/merged_tissue_experiment/Pancreas_START\n",
      "tags_on_annotation/merged_tissue_experiment/Muscle_START\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Lung_START -pc 3 > tags_on_annotation/new/Lung_START.tsv\n",
      "tags_on_annotation/merged_tissue_experiment/Spleen_START\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/CHO_START -pc 3 > tags_on_annotation/new/CHO_START.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Liver_GROCap -pc 3 > tags_on_annotation/new/Liver_GROCap.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/CHO_GROCap -pc 3 > tags_on_annotation/new/CHO_GROCap.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Lung_GROCap -pc 3 > tags_on_annotation/new/Lung_GROCap.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/MiscOrgans_START -pc 3 > tags_on_annotation/new/MiscOrgans_START.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Pancreas_START -pc 3 > tags_on_annotation/new/Pancreas_START.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Muscle_START -pc 3 > tags_on_annotation/new/Muscle_START.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Spleen_START -pc 3 > tags_on_annotation/new/Spleen_START.tsv\n",
      "Liver START\n",
      "Kidney GROCap\n",
      "tags_on_annotation/merged_tissue_experiment/Liver_START\n",
      "tags_on_annotation/merged_tissue_experiment/Kidney_GROCap\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Liver_START -pc 3 > tags_on_annotation/new/Liver_START.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Kidney_GROCap -pc 3 > tags_on_annotation/new/Kidney_GROCap.tsv\n",
      "Heart GROCap\n",
      "Kidney START\n",
      "tags_on_annotation/merged_tissue_experiment/Kidney_START\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Kidney_START -pc 3 > tags_on_annotation/new/Kidney_START.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Liver_GROCap -pc 3 > tags_on_annotation/new/Liver_GROCap.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap -pc 3 > tags_on_annotation/new/BMDMwt_GROCap.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap -pc 3 > tags_on_annotation/new/BMDM1hKLA_GROCap.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Liver_START -pc 3 > tags_on_annotation/new/Liver_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Pancreas_START -pc 3 > tags_on_annotation/new/Pancreas_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Heart_START -pc 3 > tags_on_annotation/new/Heart_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Kidney_START -pc 3 > tags_on_annotation/new/Kidney_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Brain_GROCap -pc 3 > tags_on_annotation/new/Brain_GROCap.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Lung_GROCap -pc 3 > tags_on_annotation/new/Lung_GROCap.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Liver_GROCap -pc 3 -strand + > tags_on_annotation/new/Liver_GROCap.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Kidney_GROCap -pc 3 > tags_on_annotation/new/Kidney_GROCap.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap -pc 3 -strand + > tags_on_annotation/new/BMDMwt_GROCap.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Spleen_START -pc 3 > tags_on_annotation/new/Spleen_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap -pc 3 -strand + > tags_on_annotation/new/BMDM1hKLA_GROCap.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Lung_START -pc 3 > tags_on_annotation/new/Lung_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Liver_START -pc 3 -strand + > tags_on_annotation/new/Liver_START.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Muscle_START -pc 3 > tags_on_annotation/new/Muscle_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/MiscOrgans_START -pc 3 > tags_on_annotation/new/MiscOrgans_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Pancreas_START -pc 3 -strand + > tags_on_annotation/new/Pancreas_START.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Heart_START -pc 3 -strand + > tags_on_annotation/new/Heart_START.tsvMatS\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START -pc 3 > tags_on_annotation/new/FemaleReproductive_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Kidney_START -pc 3 -strand + > tags_on_annotation/new/Kidney_START.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Brain_GROCap -pc 3 -strand + > tags_on_annotation/new/Brain_GROCap.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Lung_GROCap -pc 3 -strand + > tags_on_annotation/new/Lung_GROCap.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Kidney_GROCap -pc 3 -strand + > tags_on_annotation/new/Kidney_GROCap.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Spleen_START -pc 3 -strand + > tags_on_annotation/new/Spleen_START.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Lung_START -pc 3 -strand + > tags_on_annotation/new/Lung_START.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Liver_GROCap -pc 3 -strand - > tags_on_annotation/new/Liver_GROCap.tsvMatAS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Muscle_START -pc 3 -strand + > tags_on_annotation/new/Muscle_START.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap -pc 3 -strand - > tags_on_annotation/new/BMDMwt_GROCap.tsvMatAS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/MiscOrgans_START -pc 3 -strand + > tags_on_annotation/new/MiscOrgans_START.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap -pc 3 -strand - > tags_on_annotation/new/BMDM1hKLA_GROCap.tsvMatAS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/CHO_START -pc 3 > tags_on_annotation/new/CHO_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START -pc 3 -strand + > tags_on_annotation/new/FemaleReproductive_START.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Liver_START -pc 3 -strand - > tags_on_annotation/new/Liver_START.tsvMatAS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Pancreas_START -pc 3 -strand - > tags_on_annotation/new/Pancreas_START.tsvMatAS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Brain_START -pc 3 > tags_on_annotation/new/Brain_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Heart_START -pc 3 -strand - > tags_on_annotation/new/Heart_START.tsvMatAS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Kidney_START -pc 3 -strand - > tags_on_annotation/new/Kidney_START.tsvMatAS\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Liver_GROCap -pc 3 > tags_on_annotation/old/Liver_GROCap.tsv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Brain_GROCap -pc 3 -strand - > tags_on_annotation/new/Brain_GROCap.tsvMatAS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Lung_GROCap -pc 3 -strand - > tags_on_annotation/new/Lung_GROCap.tsvMatAS\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap -pc 3 > tags_on_annotation/old/BMDMwt_GROCap.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Kidney_GROCap -pc 3 -strand - > tags_on_annotation/new/Kidney_GROCap.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Liver_GROCap -pc 3 > tags_on_annotation/old/Liver_GROCap.tsvNorm\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap -pc 3 > tags_on_annotation/old/BMDM1hKLA_GROCap.tsv\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap -pc 3 > tags_on_annotation/old/BMDMwt_GROCap.tsvNorm\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Liver_GROCap -pc 3 -strand + > tags_on_annotation/old/Liver_GROCap.tsvMatS\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Pancreas_START -pc 3 > tags_on_annotation/old/Pancreas_START.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Muscle_START -pc 3 -strand - > tags_on_annotation/new/Muscle_START.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap -pc 3 > tags_on_annotation/old/BMDM1hKLA_GROCap.tsvNorm\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Liver_START -pc 3 > tags_on_annotation/old/Liver_START.tsv\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Heart_START -pc 3 > tags_on_annotation/old/Heart_START.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Spleen_START -pc 3 -strand - > tags_on_annotation/new/Spleen_START.tsvMatAS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Lung_START -pc 3 -strand - > tags_on_annotation/new/Lung_START.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap -pc 3 -strand + > tags_on_annotation/old/BMDMwt_GROCap.tsvMatS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Liver_START -pc 3 > tags_on_annotation/old/Liver_START.tsvNorm\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Pancreas_START -pc 3 > tags_on_annotation/old/Pancreas_START.tsvNorm\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap -pc 3 -strand + > tags_on_annotation/old/BMDM1hKLA_GROCap.tsvMatS\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START -pc 3 -strand - > tags_on_annotation/new/FemaleReproductive_START.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Heart_START -pc 3 > tags_on_annotation/old/Heart_START.tsvNorm\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Kidney_START -pc 3 > tags_on_annotation/old/Kidney_START.tsv\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Liver_START -pc 3 -strand + > tags_on_annotation/old/Liver_START.tsvMatS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/MiscOrgans_START -pc 3 -strand - > tags_on_annotation/new/MiscOrgans_START.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Pancreas_START -pc 3 -strand + > tags_on_annotation/old/Pancreas_START.tsvMatS\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Lung_GROCap -pc 3 > tags_on_annotation/old/Lung_GROCap.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/CHO_GROCap -pc 3 > tags_on_annotation/new/CHO_GROCap.tsvNorm\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Brain_GROCap -pc 3 > tags_on_annotation/old/Brain_GROCap.tsv\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/CHO_START -pc 3 -strand + > tags_on_annotation/new/CHO_START.tsvMatS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Liver_GROCap -pc 3 -strand - > tags_on_annotation/old/Liver_GROCap.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Heart_START -pc 3 -strand + > tags_on_annotation/old/Heart_START.tsvMatS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Kidney_START -pc 3 > tags_on_annotation/old/Kidney_START.tsvNorm\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Kidney_GROCap -pc 3 > tags_on_annotation/old/Kidney_GROCap.tsv\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/BMDMwt_GROCap -pc 3 -strand - > tags_on_annotation/old/BMDMwt_GROCap.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Brain_GROCap -pc 3 > tags_on_annotation/old/Brain_GROCap.tsvNorm\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Lung_GROCap -pc 3 > tags_on_annotation/old/Lung_GROCap.tsvNorm\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Muscle_START -pc 3 > tags_on_annotation/old/Muscle_START.tsv\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Kidney_START -pc 3 -strand + > tags_on_annotation/old/Kidney_START.tsvMatS\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/BMDM1hKLA_GROCap -pc 3 -strand - > tags_on_annotation/old/BMDM1hKLA_GROCap.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Kidney_GROCap -pc 3 > tags_on_annotation/old/Kidney_GROCap.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Brain_START -pc 3 -strand + > tags_on_annotation/new/Brain_START.tsvMatS\n",
      "(-500, 500)\n",
      "0.001393984967307135\n",
      "[-0.97180612 71.14123942 71.44915127 71.92751197]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.00278796993461427\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.0120591043307873\n",
      "[-0.63155772  0.63664946]\n",
      "Max value: 0.0241182086615746\n",
      "Full-width a t half-maximum: 1.268207181959823 (nts)\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Lung_START -pc 3 > tags_on_annotation/old/Lung_START.tsv\n",
      "(-500, 500)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Spleen_START -pc 3 > tags_on_annotation/old/Spleen_START.tsv\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Pancreas_START -pc 3 -strand - > tags_on_annotation/old/Pancreas_START.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Liver_START -pc 3 -strand - > tags_on_annotation/old/Liver_START.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Brain_GROCap -pc 3 -strand + > tags_on_annotation/old/Brain_GROCap.tsvMatS\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START -pc 3 > tags_on_annotation/old/FemaleReproductive_START.tsv\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Lung_GROCap -pc 3 -strand + > tags_on_annotation/old/Lung_GROCap.tsvMatS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Muscle_START -pc 3 > tags_on_annotation/old/Muscle_START.tsvNorm\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Heart_START -pc 3 -strand - > tags_on_annotation/old/Heart_START.tsvMatAS\n",
      "(-500, 500)\n",
      "0.00098638825189162\n",
      "[-3.43609367 -2.21510579 -1.00457247 67.59860975 69.17927604 70.42206641]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.00197277650378324\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.00830485525859585\n",
      "[-0.63565133  0.64304359]\n",
      "Max value: 0.0166097105171917\n",
      "Full-width a t half-maximum: 1.2786949232860405 (nts)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Kidney_GROCap -pc 3 -strand + > tags_on_annotation/old/Kidney_GROCap.tsvMatS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Spleen_START -pc 3 > tags_on_annotation/old/Spleen_START.tsvNorm\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Lung_START -pc 3 > tags_on_annotation/old/Lung_START.tsvNorm\n",
      "(-500, 500)\n",
      "0.00077030373685723\n",
      "[-0.72238153 71.43580688 72.89254763 74.103925  ]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.00154060747371446\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.00665039558024225\n",
      "[-0.64057421  0.65155803]\n",
      "Max value: 0.0133007911604845\n",
      "Full-width a t half-maximum: 1.2921322343865556 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/MiscOrgans_START -pc 3 > tags_on_annotation/old/MiscOrgans_START.tsv\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Kidney_START -pc 3 -strand - > tags_on_annotation/old/Kidney_START.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START -pc 3 > tags_on_annotation/old/FemaleReproductive_START.tsvNorm\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Muscle_START -pc 3 -strand + > tags_on_annotation/old/Muscle_START.tsvMatS\n",
      "0.000622595242628675\n",
      "[-3.79967115 -2.00508596 -0.46860884 71.19216186]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.00124519048525735\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.0036317028421957546\n",
      "[-0.70151133  0.75133036]\n",
      "Max value: 0.007263405684391509\n",
      "Full-width a t half-maximum: 1.4528416889778568 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Spleen_START -pc 3 -strand + > tags_on_annotation/old/Spleen_START.tsvMatS\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Lung_START -pc 3 -strand + > tags_on_annotation/old/Lung_START.tsvMatS\n",
      "0.001159905961108095\n",
      "[-3.45295781 -2.59757135  0.45256972 69.9024194 ]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.00231981192221619\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.0087323263184674\n",
      "[-0.67553276  0.68558605]\n",
      "Max value: 0.0174646526369348\n",
      "Full-width a t half-maximum: 1.3611188087734951 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Brain_GROCap -pc 3 -strand - > tags_on_annotation/old/Brain_GROCap.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Lung_GROCap -pc 3 -strand - > tags_on_annotation/old/Lung_GROCap.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/MiscOrgans_START -pc 3 > tags_on_annotation/old/MiscOrgans_START.tsvNorm\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/CHO_START -pc 3 -strand - > tags_on_annotation/new/CHO_START.tsvMatAS\n",
      "(-500, 500)\n",
      "0.000835059403821345\n",
      "[-7.52417584 -6.82850201 -4.04919327 69.71445363]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.00167011880764269\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.00428211415837431\n",
      "[-0.72630354  0.73346126]\n",
      "Max value: 0.00856422831674862\n",
      "Full-width a t half-maximum: 1.459764802234597 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Kidney_GROCap -pc 3 -strand - > tags_on_annotation/old/Kidney_GROCap.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START -pc 3 -strand + > tags_on_annotation/old/FemaleReproductive_START.tsvMatS\n",
      "(-500, 500)\n",
      "0.001326883168901005\n",
      "[-3.99794839 70.14946122]\n",
      "Max value: 0.00265376633780201\n",
      "Full-width a t half-maximum: 74.14740961027941 (nts)\n",
      "0.0082604893463419\n",
      "[-0.69973128  0.6937384 ]\n",
      "Max value: 0.0165209786926838\n",
      "Full-width a t half-maximum: 1.3934696812741614 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/MiscOrgans_START -pc 3 -strand + > tags_on_annotation/old/MiscOrgans_START.tsvMatS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Muscle_START -pc 3 -strand - > tags_on_annotation/old/Muscle_START.tsvMatAS\n",
      "(-500, 500)\n",
      "0.00112154435284239\n",
      "[-14.15380852  70.6141716 ]\n",
      "Max value: 0.00224308870568478\n",
      "Full-width a t half-maximum: 84.76798012384573 (nts)\n",
      "0.00572768997843175\n",
      "[-0.67531309  0.71001535]\n",
      "Max value: 0.0114553799568635\n",
      "Full-width a t half-maximum: 1.385328436816312 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(-500, 500)\n",
      "0.0016200983854155401\n",
      "[-12.09504737 -10.26057851  -9.31375252  69.79932899]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.0032401967708310803\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.00827769348243805\n",
      "[-0.67921281  0.71820754]\n",
      "Max value: 0.0165553869648761\n",
      "Full-width a t half-maximum: 1.3974203558499079 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Spleen_START -pc 3 -strand - > tags_on_annotation/old/Spleen_START.tsvMatAS\n",
      "(-500, 500)\n",
      "0.00132741147147925\n",
      "[-12.08336598 -10.20464644  -8.9870531   -5.31233977  -4.80837558\n",
      "  73.09250239]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.0026548229429585\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.00701617270928045\n",
      "[-0.67897457  0.71218015]\n",
      "Max value: 0.0140323454185609\n",
      "Full-width a t half-maximum: 1.3911547166923588 (nts)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Lung_START -pc 3 -strand - > tags_on_annotation/old/Lung_START.tsvMatAS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Brain_START -pc 3 -strand - > tags_on_annotation/new/Brain_START.tsvMatAS\n",
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/CHO_GROCap -pc 3 -strand + > tags_on_annotation/new/CHO_GROCap.tsvMatS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/FemaleReproductive_START -pc 3 -strand - > tags_on_annotation/old/FemaleReproductive_START.tsvMatAS\n",
      "(-500, 500)\n",
      "0.000862404132150405\n",
      "[-9.34126538 -8.36000447 -7.62025751 77.07041945]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.00172480826430081\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.003828436800090355\n",
      "[-0.78101074  0.76957649]\n",
      "Max value: 0.00765687360018071\n",
      "Full-width a t half-maximum: 1.5505872387349542 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/CHO_START -pc 3 > tags_on_annotation/old/CHO_START.tsv\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/MiscOrgans_START -pc 3 -strand - > tags_on_annotation/old/MiscOrgans_START.tsvMatAS\n",
      "(-500, 500)\n",
      "0.0008303218523672299\n",
      "[-8.58433944 75.22920552 76.75574496 77.16310946]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.0016606437047344599\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "(-500, 500)\n",
      "0.003603396282520045\n",
      "[-0.77619222  0.776711  ]\n",
      "Max value: 0.00720679256504009\n",
      "Full-width a t half-maximum: 1.552903215662924 (nts)\n",
      "0.0008245989381644901\n",
      "[-9.31202844 70.18070176 71.70084951 72.14098617]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.0016491978763289801\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.003945224714002625\n",
      "[-0.75544324  0.76163487]\n",
      "Max value: 0.00789044942800525\n",
      "Full-width a t half-maximum: 1.5170781135106919 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(-500, 500)\n",
      "0.00135826493295295\n",
      "[-9.12297517 -8.18720233 -7.97891275 70.65395198]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.0027165298659059\n",
      "Full-width a t half-maximum: -1 (nts)\n",
      "0.00573067248616525\n",
      "[-0.75722871  0.75905241]\n",
      "Max value: 0.0114613449723305\n",
      "Full-width a t half-maximum: 1.5162811216443883 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/CHO_START -pc 3 > tags_on_annotation/old/CHO_START.tsvNorm\n",
      "(-500, 500)\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/Brain_START -pc 3 > tags_on_annotation/old/Brain_START.tsv\n",
      "0.001292632832240595\n",
      "[-7.62861112 74.65803498]\n",
      "Max value: 0.00258526566448119\n",
      "Full-width a t half-maximum: 82.28664610136205 (nts)\n",
      "0.00599597198733765\n",
      "[-0.76119678  0.77101011]\n",
      "Max value: 0.0119919439746753\n",
      "Full-width a t half-maximum: 1.5322068894321497 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl analysis_ncbi_picr/output/TSS1.exp.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/CHO_GROCap -pc 3 -strand - > tags_on_annotation/new/CHO_GROCap.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/CHO_START -pc 3 -strand + > tags_on_annotation/old/CHO_START.tsvMatS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/Brain_START -pc 3 > tags_on_annotation/old/Brain_START.tsvNorm\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/CHO_START -pc 3 -strand - > tags_on_annotation/old/CHO_START.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Brain_START -pc 3 -strand + > tags_on_annotation/old/Brain_START.tsvMatS\n",
      "(-500, 500)\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -d tags_on_annotation/merged_tissue_experiment/CHO_GROCap -pc 3 > tags_on_annotation/old/CHO_GROCap.tsv\n",
      "(-500, 500)\n",
      "0.001227713293128735\n",
      "[-8.98948569 77.71856969]\n",
      "Max value: 0.00245542658625747\n",
      "Full-width a t half-maximum: 86.7080553793685 (nts)\n",
      "0.0032205206354168053\n",
      "[-2.67664306 -1.80288403 -1.17343637  3.52899918]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.0064410412708336105\n",
      "Full-width a t half-maximum: -1 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/Brain_START -pc 3 -strand - > tags_on_annotation/old/Brain_START.tsvMatAS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -histNorm 100 -d tags_on_annotation/merged_tissue_experiment/CHO_GROCap -pc 3 > tags_on_annotation/old/CHO_GROCap.tsvNorm\n",
      "(-500, 500)\n",
      "0.00059021818497608\n",
      "[-20.73443312  82.08581333]\n",
      "Max value: 0.00118043636995216\n",
      "Full-width a t half-maximum: 102.82024645597049 (nts)\n",
      "0.001572937613833185\n",
      "[-0.99346142  0.98133806  1.83376201  3.17312685  4.34974563  5.41219116]\n",
      "Multiple Peaks are found\n",
      "Max value: 0.00314587522766637\n",
      "Full-width a t half-maximum: -1 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/CHO_GROCap -pc 3 -strand + > tags_on_annotation/old/CHO_GROCap.tsvMatS\n",
      "annotatePeaks.pl alt_anno/Figure4/motifs_from_meme/TSS1.exp_refseq_centered.bed /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3 -fragLength 1 -size 2000 -hist 1 -ghist -d tags_on_annotation/merged_tissue_experiment/CHO_GROCap -pc 3 -strand - > tags_on_annotation/old/CHO_GROCap.tsvMatAS\n",
      "(-500, 500)\n",
      "0.00070835804532686\n",
      "[-17.41441188  85.37610815]\n",
      "Max value: 0.00141671609065372\n",
      "Full-width a t half-maximum: 102.79052003190252 (nts)\n",
      "0.0015678521051521802\n",
      "[-5.50346874  5.81929799]\n",
      "Max value: 0.0031357042103043603\n",
      "Full-width a t half-maximum: 11.322766724455693 (nts)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n",
      "The PostScript backend does not support transparency; partially transparent artists will be rendered opaque.\n"
     ]
    }
   ],
   "source": [
    "parar(np.array(list(tag_folders.keys())), run_hist_compare_exp_ref,\n",
    "      func_args=(eTSS, rTSS, anno_gff, ref_fa, outdir_exp, outdir_ref, outdir), num_processes=24)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
