{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Note this is using the Analysis_0330 csRNA peaks (which shouldnt matter)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Load the csRNA merged peaks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Load the findPeaks peaks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Merge the two"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:15.382132Z",
     "start_time": "2020-09-01T17:24:12.745147Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/ipykernel_launcher.py:7: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  import sys\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data2/resources/software/homer/v4.11_10242019/bin//homer\r\n"
     ]
    }
   ],
   "source": [
    "## Parameters specific to where your folders are and your data\n",
    "parameter_file = '../parameters/params.yaml'\n",
    "import yaml\n",
    "import sys\n",
    "import os\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "#p = dic2obj(**doc)\n",
    "\n",
    "#data_folder = doc['data_folder']\n",
    "#tissues = doc['tissues'].split(',')\n",
    "#sys.path.append(doc['pipeline_path'])\n",
    "ref_fa = doc['GENOME_FA']\n",
    "anno_gff=doc['ANNOTATION']\n",
    "#mRNA_peak_file = doc[\"mRNA_peak_file\"]\n",
    "#tss_annotation = doc['tss_annotation']\n",
    "\n",
    "\n",
    "import sys\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import pickle\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "from matplotlib import pyplot as plt\n",
    "from itertools import product\n",
    "import glob\n",
    "import re\n",
    "from os.path import join\n",
    "from matplotlib_venn import venn2\n",
    "from matplotlib import rcParams\n",
    "import inspect\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "sys.setrecursionlimit(3000)\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "rcParams['figure.figsize'] = 8, 6\n",
    "import tqdm\n",
    "\n",
    "from os.path import basename\n",
    "##mpl.use('Agg')\n",
    "#mpl.style.use('ggplot')\n",
    "#mpl.style.use('fivethirtyeight')\n",
    "#from Homer import *\n",
    "\n",
    "homer_path =  \"/data2/resources/software/homer/v4.11_10242019/bin/\"\n",
    "os.environ['PATH'] = f\"{homer_path}:{os.environ['PATH']}\"\n",
    "!which homer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:16.729877Z",
     "start_time": "2020-09-01T17:24:15.384687Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "params {'PREFIX': 'ncbi_pice_103', 'ROOT_DIR': '/data/isshamie/TSS/', 'DATA_DIR': '/data/isshamie/TSS/NCBI_PICR_103_Processed/', 'DATA_PROCESSED_DIR': '/data/isshamie/TSS/TSS/data/processed/', 'SUPPLEMENTAL_DIR': '/data/isshamie/TSS/Analysis/supplemental/', 'PIPELINE': '/data/isshamie/TSS/tss', 'GENOME_DIR': '/data/isshamie/genome', 'GENOME_FA': '/data/isshamie/genome/picr.fa', 'GENOME_GFF3': '/data/isshamie/genome/updated_final_sort.gff3', 'GENOME_GTF': '/data/isshamie/genome/updated_final_sort.gtf', 'TISSUES': 'BMDM1hKLA,BMDMwt,Brain,FemaleReproductive,Heart,Intestine,', 'META_FILE': '/data/isshamie/TSS/TSS/parameters/raw_metaSamples.tsv'}\n",
      "fig_utils from mplh\n",
      "/data2/resources/software/homer/v4.11_10242019/bin//\n"
     ]
    }
   ],
   "source": [
    "from tss.data import annotation\n",
    "from numpanpar import parallel_df as pardf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:16.750225Z",
     "start_time": "2020-09-01T17:24:16.732306Z"
    }
   },
   "outputs": [],
   "source": [
    "os.chdir(doc[\"results\"])\n",
    "genome=\"GCF\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Output files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:16.771715Z",
     "start_time": "2020-09-01T17:24:16.752372Z"
    }
   },
   "outputs": [],
   "source": [
    "save_dir = join(\"stable_rna\", genome)\n",
    "out_csrna_merged = join(save_dir, \"merged_stable_peaks.tsv\")\n",
    "out_all_csrna_merged = join(save_dir, \"all_merged_stable_peaks.tsv\")\n",
    "out_eTSS_csrna_merged = join(save_dir, \"eTSS_merged_stable_peaks.tsv.merge\")\n",
    "out_stable_minus_eTSS = join(save_dir, \"stable_minus_eTSS.tsv\")\n",
    "if not os.path.exists(save_dir):\n",
    "    os.makedirs(save_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### input files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:16.792634Z",
     "start_time": "2020-09-01T17:24:16.773562Z"
    }
   },
   "outputs": [],
   "source": [
    "anno_gff = '/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gff',\n",
    "indir = \"stableRNA/Analysis_03302020/csRNA_peaks_tissues.merged/\"\n",
    "merge_f = join(indir, \"csRNA_peaks.tissues.merge\")\n",
    "anno_f = join(indir, \"csRNA_peaks.tissues.merge.anno\")\n",
    "eTSS_bed_f = join(genome, \"eTSS/TSS.bed\")\n",
    "all_merged_f = join(genome, \"merged/samples.merge\" )\n",
    "all_merged_anno_f = join(genome,\"merged/samples.merge.anno\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:17.242683Z",
     "start_time": "2020-09-01T17:24:16.794439Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>actual_start</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Dbxref</th>\n",
       "      <th>Name</th>\n",
       "      <th>gbkey</th>\n",
       "      <th>...</th>\n",
       "      <th>model_evidence</th>\n",
       "      <th>product</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>Note</th>\n",
       "      <th>exception</th>\n",
       "      <th>end_range</th>\n",
       "      <th>partial</th>\n",
       "      <th>start_range</th>\n",
       "      <th>inference</th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>XM_027386529.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>722308</td>\n",
       "      <td>722308</td>\n",
       "      <td>-</td>\n",
       "      <td>722308</td>\n",
       "      <td>rna-XM_027386529.1</td>\n",
       "      <td>gene-Bmp4</td>\n",
       "      <td>GeneID:100754303,Genbank:XM_027386529.1</td>\n",
       "      <td>XM_027386529.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 29...</td>\n",
       "      <td>bone morphogenetic protein 4%2C transcript var...</td>\n",
       "      <td>XM_027386529.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027386528.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>724610</td>\n",
       "      <td>724610</td>\n",
       "      <td>-</td>\n",
       "      <td>724610</td>\n",
       "      <td>rna-XM_027386528.1</td>\n",
       "      <td>gene-Bmp4</td>\n",
       "      <td>GeneID:100754303,Genbank:XM_027386528.1</td>\n",
       "      <td>XM_027386528.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 29...</td>\n",
       "      <td>bone morphogenetic protein 4%2C transcript var...</td>\n",
       "      <td>XM_027386528.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027386531.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>1085435</td>\n",
       "      <td>1085435</td>\n",
       "      <td>+</td>\n",
       "      <td>1085435</td>\n",
       "      <td>rna-XM_027386531.1</td>\n",
       "      <td>gene-Cdkn3</td>\n",
       "      <td>GeneID:100752216,Genbank:XM_027386531.1</td>\n",
       "      <td>XM_027386531.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 5 ...</td>\n",
       "      <td>cyclin dependent kinase inhibitor 3%2C transcr...</td>\n",
       "      <td>XM_027386531.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027386533.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>1085744</td>\n",
       "      <td>1085744</td>\n",
       "      <td>+</td>\n",
       "      <td>1085744</td>\n",
       "      <td>rna-XM_027386533.1</td>\n",
       "      <td>gene-Cdkn3</td>\n",
       "      <td>GeneID:100752216,Genbank:XM_027386533.1</td>\n",
       "      <td>XM_027386533.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 1 ...</td>\n",
       "      <td>cyclin dependent kinase inhibitor 3%2C transcr...</td>\n",
       "      <td>XM_027386533.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027386534.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>1114413</td>\n",
       "      <td>1114413</td>\n",
       "      <td>-</td>\n",
       "      <td>1114413</td>\n",
       "      <td>rna-XM_027386534.1</td>\n",
       "      <td>gene-Cnih1</td>\n",
       "      <td>GeneID:103160710,Genbank:XM_027386534.1</td>\n",
       "      <td>XM_027386534.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 10...</td>\n",
       "      <td>cornichon family AMPA receptor auxiliary prote...</td>\n",
       "      <td>XM_027386534.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434589.1</td>\n",
       "      <td>NW_020824185.1</td>\n",
       "      <td>3893</td>\n",
       "      <td>3893</td>\n",
       "      <td>+</td>\n",
       "      <td>3893</td>\n",
       "      <td>rna-XM_027434589.1</td>\n",
       "      <td>gene-LOC113839146</td>\n",
       "      <td>GeneID:113839146,Genbank:XM_027434589.1</td>\n",
       "      <td>XM_027434589.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 3 ...</td>\n",
       "      <td>mothers against decapentaplegic homolog 1-like</td>\n",
       "      <td>XM_027434589.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>.,3893</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434591.1</td>\n",
       "      <td>NW_020824186.1</td>\n",
       "      <td>17914</td>\n",
       "      <td>17914</td>\n",
       "      <td>+</td>\n",
       "      <td>17914</td>\n",
       "      <td>rna-XM_027434591.1</td>\n",
       "      <td>gene-LOC113839147</td>\n",
       "      <td>GeneID:113839147,Genbank:XM_027434591.1</td>\n",
       "      <td>XM_027434591.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 6 ...</td>\n",
       "      <td>PRAME family member 12-like%2C transcript vari...</td>\n",
       "      <td>XM_027434591.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434593.1</td>\n",
       "      <td>NW_020824186.1</td>\n",
       "      <td>29432</td>\n",
       "      <td>29432</td>\n",
       "      <td>+</td>\n",
       "      <td>29432</td>\n",
       "      <td>rna-XM_027434593.1</td>\n",
       "      <td>gene-LOC103158952</td>\n",
       "      <td>GeneID:103158952,Genbank:XM_027434593.1</td>\n",
       "      <td>XM_027434593.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 2 ...</td>\n",
       "      <td>PRAME family member 12-like%2C transcript vari...</td>\n",
       "      <td>XM_027434593.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434595.1</td>\n",
       "      <td>NW_020824188.1</td>\n",
       "      <td>2147</td>\n",
       "      <td>2147</td>\n",
       "      <td>+</td>\n",
       "      <td>2147</td>\n",
       "      <td>rna-XM_027434595.1</td>\n",
       "      <td>gene-LOC113839149</td>\n",
       "      <td>GeneID:113839149,Genbank:XM_027434595.1</td>\n",
       "      <td>XM_027434595.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 25...</td>\n",
       "      <td>zinc finger protein 431-like%2C transcript var...</td>\n",
       "      <td>XM_027434595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434597.1</td>\n",
       "      <td>NW_020824192.1</td>\n",
       "      <td>6650</td>\n",
       "      <td>6650</td>\n",
       "      <td>-</td>\n",
       "      <td>6650</td>\n",
       "      <td>rna-XM_027434597.1</td>\n",
       "      <td>gene-LOC113839150</td>\n",
       "      <td>GeneID:113839150,Genbank:XM_027434597.1</td>\n",
       "      <td>XM_027434597.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>...</td>\n",
       "      <td>Supporting evidence includes similarity to: 8 ...</td>\n",
       "      <td>mothers against decapentaplegic homolog 1-like</td>\n",
       "      <td>XM_027434597.1</td>\n",
       "      <td>The sequence of the model RefSeq transcript wa...</td>\n",
       "      <td>unclassified transcription discrepancy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35679 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Chr    Start      End Strand  actual_start  \\\n",
       "XM_027386529.1  NW_020822366.1   722308   722308      -        722308   \n",
       "XM_027386528.1  NW_020822366.1   724610   724610      -        724610   \n",
       "XM_027386531.1  NW_020822366.1  1085435  1085435      +       1085435   \n",
       "XM_027386533.1  NW_020822366.1  1085744  1085744      +       1085744   \n",
       "XM_027386534.1  NW_020822366.1  1114413  1114413      -       1114413   \n",
       "...                        ...      ...      ...    ...           ...   \n",
       "XM_027434589.1  NW_020824185.1     3893     3893      +          3893   \n",
       "XM_027434591.1  NW_020824186.1    17914    17914      +         17914   \n",
       "XM_027434593.1  NW_020824186.1    29432    29432      +         29432   \n",
       "XM_027434595.1  NW_020824188.1     2147     2147      +          2147   \n",
       "XM_027434597.1  NW_020824192.1     6650     6650      -          6650   \n",
       "\n",
       "                                ID             Parent  \\\n",
       "XM_027386529.1  rna-XM_027386529.1          gene-Bmp4   \n",
       "XM_027386528.1  rna-XM_027386528.1          gene-Bmp4   \n",
       "XM_027386531.1  rna-XM_027386531.1         gene-Cdkn3   \n",
       "XM_027386533.1  rna-XM_027386533.1         gene-Cdkn3   \n",
       "XM_027386534.1  rna-XM_027386534.1         gene-Cnih1   \n",
       "...                            ...                ...   \n",
       "XM_027434589.1  rna-XM_027434589.1  gene-LOC113839146   \n",
       "XM_027434591.1  rna-XM_027434591.1  gene-LOC113839147   \n",
       "XM_027434593.1  rna-XM_027434593.1  gene-LOC103158952   \n",
       "XM_027434595.1  rna-XM_027434595.1  gene-LOC113839149   \n",
       "XM_027434597.1  rna-XM_027434597.1  gene-LOC113839150   \n",
       "\n",
       "                                                 Dbxref            Name gbkey  \\\n",
       "XM_027386529.1  GeneID:100754303,Genbank:XM_027386529.1  XM_027386529.1  mRNA   \n",
       "XM_027386528.1  GeneID:100754303,Genbank:XM_027386528.1  XM_027386528.1  mRNA   \n",
       "XM_027386531.1  GeneID:100752216,Genbank:XM_027386531.1  XM_027386531.1  mRNA   \n",
       "XM_027386533.1  GeneID:100752216,Genbank:XM_027386533.1  XM_027386533.1  mRNA   \n",
       "XM_027386534.1  GeneID:103160710,Genbank:XM_027386534.1  XM_027386534.1  mRNA   \n",
       "...                                                 ...             ...   ...   \n",
       "XM_027434589.1  GeneID:113839146,Genbank:XM_027434589.1  XM_027434589.1  mRNA   \n",
       "XM_027434591.1  GeneID:113839147,Genbank:XM_027434591.1  XM_027434591.1  mRNA   \n",
       "XM_027434593.1  GeneID:103158952,Genbank:XM_027434593.1  XM_027434593.1  mRNA   \n",
       "XM_027434595.1  GeneID:113839149,Genbank:XM_027434595.1  XM_027434595.1  mRNA   \n",
       "XM_027434597.1  GeneID:113839150,Genbank:XM_027434597.1  XM_027434597.1  mRNA   \n",
       "\n",
       "                ...                                     model_evidence  \\\n",
       "XM_027386529.1  ...  Supporting evidence includes similarity to: 29...   \n",
       "XM_027386528.1  ...  Supporting evidence includes similarity to: 29...   \n",
       "XM_027386531.1  ...  Supporting evidence includes similarity to: 5 ...   \n",
       "XM_027386533.1  ...  Supporting evidence includes similarity to: 1 ...   \n",
       "XM_027386534.1  ...  Supporting evidence includes similarity to: 10...   \n",
       "...             ...                                                ...   \n",
       "XM_027434589.1  ...  Supporting evidence includes similarity to: 3 ...   \n",
       "XM_027434591.1  ...  Supporting evidence includes similarity to: 6 ...   \n",
       "XM_027434593.1  ...  Supporting evidence includes similarity to: 2 ...   \n",
       "XM_027434595.1  ...  Supporting evidence includes similarity to: 25...   \n",
       "XM_027434597.1  ...  Supporting evidence includes similarity to: 8 ...   \n",
       "\n",
       "                                                          product  \\\n",
       "XM_027386529.1  bone morphogenetic protein 4%2C transcript var...   \n",
       "XM_027386528.1  bone morphogenetic protein 4%2C transcript var...   \n",
       "XM_027386531.1  cyclin dependent kinase inhibitor 3%2C transcr...   \n",
       "XM_027386533.1  cyclin dependent kinase inhibitor 3%2C transcr...   \n",
       "XM_027386534.1  cornichon family AMPA receptor auxiliary prote...   \n",
       "...                                                           ...   \n",
       "XM_027434589.1     mothers against decapentaplegic homolog 1-like   \n",
       "XM_027434591.1  PRAME family member 12-like%2C transcript vari...   \n",
       "XM_027434593.1  PRAME family member 12-like%2C transcript vari...   \n",
       "XM_027434595.1  zinc finger protein 431-like%2C transcript var...   \n",
       "XM_027434597.1     mothers against decapentaplegic homolog 1-like   \n",
       "\n",
       "                 transcript_id  \\\n",
       "XM_027386529.1  XM_027386529.1   \n",
       "XM_027386528.1  XM_027386528.1   \n",
       "XM_027386531.1  XM_027386531.1   \n",
       "XM_027386533.1  XM_027386533.1   \n",
       "XM_027386534.1  XM_027386534.1   \n",
       "...                        ...   \n",
       "XM_027434589.1  XM_027434589.1   \n",
       "XM_027434591.1  XM_027434591.1   \n",
       "XM_027434593.1  XM_027434593.1   \n",
       "XM_027434595.1  XM_027434595.1   \n",
       "XM_027434597.1  XM_027434597.1   \n",
       "\n",
       "                                                             Note  \\\n",
       "XM_027386529.1                                                NaN   \n",
       "XM_027386528.1                                                NaN   \n",
       "XM_027386531.1                                                NaN   \n",
       "XM_027386533.1                                                NaN   \n",
       "XM_027386534.1                                                NaN   \n",
       "...                                                           ...   \n",
       "XM_027434589.1                                                NaN   \n",
       "XM_027434591.1                                                NaN   \n",
       "XM_027434593.1                                                NaN   \n",
       "XM_027434595.1                                                NaN   \n",
       "XM_027434597.1  The sequence of the model RefSeq transcript wa...   \n",
       "\n",
       "                                             exception end_range partial  \\\n",
       "XM_027386529.1                                     NaN       NaN     NaN   \n",
       "XM_027386528.1                                     NaN       NaN     NaN   \n",
       "XM_027386531.1                                     NaN       NaN     NaN   \n",
       "XM_027386533.1                                     NaN       NaN     NaN   \n",
       "XM_027386534.1                                     NaN       NaN     NaN   \n",
       "...                                                ...       ...     ...   \n",
       "XM_027434589.1                                     NaN       NaN    True   \n",
       "XM_027434591.1                                     NaN       NaN     NaN   \n",
       "XM_027434593.1                                     NaN       NaN     NaN   \n",
       "XM_027434595.1                                     NaN       NaN     NaN   \n",
       "XM_027434597.1  unclassified transcription discrepancy       NaN     NaN   \n",
       "\n",
       "               start_range inference Length  \n",
       "XM_027386529.1         NaN       NaN      1  \n",
       "XM_027386528.1         NaN       NaN      1  \n",
       "XM_027386531.1         NaN       NaN      1  \n",
       "XM_027386533.1         NaN       NaN      1  \n",
       "XM_027386534.1         NaN       NaN      1  \n",
       "...                    ...       ...    ...  \n",
       "XM_027434589.1      .,3893       NaN      1  \n",
       "XM_027434591.1         NaN       NaN      1  \n",
       "XM_027434593.1         NaN       NaN      1  \n",
       "XM_027434595.1         NaN       NaN      1  \n",
       "XM_027434597.1         NaN       NaN      1  \n",
       "\n",
       "[35679 rows x 21 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mrna_peak = pd.read_csv(join(genome, \"genome\", \"start_site_mRNA.tsv\"), sep=\"\\t\", index_col=0)\n",
    "mrna_peak"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:20.191608Z",
     "start_time": "2020-09-01T17:24:17.244433Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\tConverted 223560 peaks total\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "#!module load homer_v\n",
    "!pos2bed.pl {merge_f} > {merge_f}.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:20.224947Z",
     "start_time": "2020-09-01T17:24:20.193878Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['stableRNA/Analysis_03302020/csRNA_peaks/Brain_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START2_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START3_STARTinput1_RNASeq2.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Kidney_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START2_STARTinput1_RNASeq2.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Spleen_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/FemaleReproductive_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Muscle_START2_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Spleen_START2_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/MiscOrgans_START2_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START2_STARTinput2_RNASeq2.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Heart_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Heart_START2_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START1_STARTinput2_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/FemaleReproductive_START2_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START1_STARTinput1_RNASeq2.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Pancreas_START2_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Liver_START2_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START3_STARTinput2_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Liver_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START2_STARTinput2_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Pancreas_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/MiscOrgans_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Lung_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START3_STARTinput2_RNASeq2.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START3_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Brain_START1_STARTinput2_RNASeq2.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Lung_START2_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Muscle_START1_STARTinput1_RNASeq1.tss.txt',\n",
       " 'stableRNA/Analysis_03302020/csRNA_peaks/Kidney_START2_STARTinput1_RNASeq1.tss.txt']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_stable_files = glob.glob(\"stableRNA/Analysis_03302020/csRNA_peaks/*.tss.txt\")\n",
    "merged_stable_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:20.991180Z",
     "start_time": "2020-09-01T17:24:20.228302Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3051: DtypeWarning: Columns (8,9,10,11,12,13,14,15,16,17) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/FemaleReproductive.tss.txt.merge</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.txt.merge</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/Muscle.tss.txt.merge</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/Spleen.tss.txt.merge</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/Lung.tss.txt.merge</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/Kidney.tss.txt.merge</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/Brain.tss.txt.merge</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/Liver.tss.txt.merge</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans.tss.txt.merge</th>\n",
       "      <th>csRNA_pipeline/csRNA_peaks_tissues/Pancreas.tss.txt.merge</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>33967773</td>\n",
       "      <td>33967923</td>\n",
       "      <td>+</td>\n",
       "      <td>40.900002</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21852</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NW_020822455.1</td>\n",
       "      <td>6758790</td>\n",
       "      <td>6758940</td>\n",
       "      <td>+</td>\n",
       "      <td>30.000000</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21853</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>NW_020822461.1</td>\n",
       "      <td>34406811</td>\n",
       "      <td>34406961</td>\n",
       "      <td>-</td>\n",
       "      <td>36.799999</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21854</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NW_020822687.1</td>\n",
       "      <td>15499622</td>\n",
       "      <td>15499772</td>\n",
       "      <td>-</td>\n",
       "      <td>39.599998</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21855</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>NW_020822632.1</td>\n",
       "      <td>9912312</td>\n",
       "      <td>9912462</td>\n",
       "      <td>+</td>\n",
       "      <td>36.799999</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21857</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>223555</td>\n",
       "      <td>NW_020822406.1</td>\n",
       "      <td>14607236</td>\n",
       "      <td>14607386</td>\n",
       "      <td>-</td>\n",
       "      <td>18.049999</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44239</td>\n",
       "      <td>32282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>223556</td>\n",
       "      <td>NW_020822558.1</td>\n",
       "      <td>1002132</td>\n",
       "      <td>1002282</td>\n",
       "      <td>+</td>\n",
       "      <td>10.850000</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32227</td>\n",
       "      <td>22615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>223557</td>\n",
       "      <td>NW_020822467.1</td>\n",
       "      <td>3885361</td>\n",
       "      <td>3885511</td>\n",
       "      <td>-</td>\n",
       "      <td>22.250000</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33834</td>\n",
       "      <td>23909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>223558</td>\n",
       "      <td>NW_020822616.1</td>\n",
       "      <td>4092026</td>\n",
       "      <td>4092176</td>\n",
       "      <td>-</td>\n",
       "      <td>24.350000</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44274</td>\n",
       "      <td>32317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>223559</td>\n",
       "      <td>NW_020822427.1</td>\n",
       "      <td>801340</td>\n",
       "      <td>801499</td>\n",
       "      <td>+</td>\n",
       "      <td>36.700001</td>\n",
       "      <td>csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21849</td>\n",
       "      <td>24167</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>223560 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Chr     Start       End Strand       Stat  \\\n",
       "ID                                                             \n",
       "0       NW_020822601.1  33967773  33967923      +  40.900002   \n",
       "1       NW_020822455.1   6758790   6758940      +  30.000000   \n",
       "2       NW_020822461.1  34406811  34406961      -  36.799999   \n",
       "3       NW_020822687.1  15499622  15499772      -  39.599998   \n",
       "4       NW_020822632.1   9912312   9912462      +  36.799999   \n",
       "...                ...       ...       ...    ...        ...   \n",
       "223555  NW_020822406.1  14607236  14607386      -  18.049999   \n",
       "223556  NW_020822558.1   1002132   1002282      +  10.850000   \n",
       "223557  NW_020822467.1   3885361   3885511      -  22.250000   \n",
       "223558  NW_020822616.1   4092026   4092176      -  24.350000   \n",
       "223559  NW_020822427.1    801340    801499      +  36.700001   \n",
       "\n",
       "                                             Parent files  Total subpeaks  \\\n",
       "ID                                                                          \n",
       "0       csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...               1   \n",
       "1       csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...               1   \n",
       "2       csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...               1   \n",
       "3       csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...               1   \n",
       "4       csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.t...               1   \n",
       "...                                                   ...             ...   \n",
       "223555  csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....               2   \n",
       "223556  csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....               2   \n",
       "223557  csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....               2   \n",
       "223558  csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....               2   \n",
       "223559  csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans....               2   \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/FemaleReproductive.tss.txt.merge  \\\n",
       "ID                                                                           \n",
       "0                                                     NaN                    \n",
       "1                                                     NaN                    \n",
       "2                                                     NaN                    \n",
       "3                                                     NaN                    \n",
       "4                                                     NaN                    \n",
       "...                                                   ...                    \n",
       "223555                                                NaN                    \n",
       "223556                                                NaN                    \n",
       "223557                                                NaN                    \n",
       "223558                                                NaN                    \n",
       "223559                                                NaN                    \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.txt.merge  \\\n",
       "ID                                                              \n",
       "0                                                   21852       \n",
       "1                                                   21853       \n",
       "2                                                   21854       \n",
       "3                                                   21855       \n",
       "4                                                   21857       \n",
       "...                                                   ...       \n",
       "223555                                                NaN       \n",
       "223556                                                NaN       \n",
       "223557                                                NaN       \n",
       "223558                                                NaN       \n",
       "223559                                                NaN       \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/Muscle.tss.txt.merge  \\\n",
       "ID                                                               \n",
       "0                                                     NaN        \n",
       "1                                                     NaN        \n",
       "2                                                     NaN        \n",
       "3                                                     NaN        \n",
       "4                                                     NaN        \n",
       "...                                                   ...        \n",
       "223555                                                NaN        \n",
       "223556                                                NaN        \n",
       "223557                                                NaN        \n",
       "223558                                                NaN        \n",
       "223559                                                NaN        \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/Spleen.tss.txt.merge  \\\n",
       "ID                                                               \n",
       "0                                                     NaN        \n",
       "1                                                     NaN        \n",
       "2                                                     NaN        \n",
       "3                                                     NaN        \n",
       "4                                                     NaN        \n",
       "...                                                   ...        \n",
       "223555                                                NaN        \n",
       "223556                                                NaN        \n",
       "223557                                                NaN        \n",
       "223558                                                NaN        \n",
       "223559                                                NaN        \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/Lung.tss.txt.merge  \\\n",
       "ID                                                             \n",
       "0                                                     NaN      \n",
       "1                                                     NaN      \n",
       "2                                                     NaN      \n",
       "3                                                     NaN      \n",
       "4                                                     NaN      \n",
       "...                                                   ...      \n",
       "223555                                                NaN      \n",
       "223556                                                NaN      \n",
       "223557                                                NaN      \n",
       "223558                                                NaN      \n",
       "223559                                                NaN      \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/Kidney.tss.txt.merge  \\\n",
       "ID                                                               \n",
       "0                                                     NaN        \n",
       "1                                                     NaN        \n",
       "2                                                     NaN        \n",
       "3                                                     NaN        \n",
       "4                                                     NaN        \n",
       "...                                                   ...        \n",
       "223555                                                NaN        \n",
       "223556                                                NaN        \n",
       "223557                                                NaN        \n",
       "223558                                                NaN        \n",
       "223559                                                NaN        \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/Brain.tss.txt.merge  \\\n",
       "ID                                                              \n",
       "0                                                     NaN       \n",
       "1                                                     NaN       \n",
       "2                                                     NaN       \n",
       "3                                                     NaN       \n",
       "4                                                     NaN       \n",
       "...                                                   ...       \n",
       "223555                                                NaN       \n",
       "223556                                                NaN       \n",
       "223557                                                NaN       \n",
       "223558                                                NaN       \n",
       "223559                                                NaN       \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/Liver.tss.txt.merge  \\\n",
       "ID                                                              \n",
       "0                                                     NaN       \n",
       "1                                                     NaN       \n",
       "2                                                     NaN       \n",
       "3                                                     NaN       \n",
       "4                                                     NaN       \n",
       "...                                                   ...       \n",
       "223555                                                NaN       \n",
       "223556                                                NaN       \n",
       "223557                                                NaN       \n",
       "223558                                                NaN       \n",
       "223559                                                NaN       \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans.tss.txt.merge  \\\n",
       "ID                                                                   \n",
       "0                                                     NaN            \n",
       "1                                                     NaN            \n",
       "2                                                     NaN            \n",
       "3                                                     NaN            \n",
       "4                                                     NaN            \n",
       "...                                                   ...            \n",
       "223555                                              44239            \n",
       "223556                                              32227            \n",
       "223557                                              33834            \n",
       "223558                                              44274            \n",
       "223559                                              21849            \n",
       "\n",
       "       csRNA_pipeline/csRNA_peaks_tissues/Pancreas.tss.txt.merge  \n",
       "ID                                                                \n",
       "0                                                     NaN         \n",
       "1                                                     NaN         \n",
       "2                                                     NaN         \n",
       "3                                                     NaN         \n",
       "4                                                     NaN         \n",
       "...                                                   ...         \n",
       "223555                                              32282         \n",
       "223556                                              22615         \n",
       "223557                                              23909         \n",
       "223558                                              32317         \n",
       "223559                                              24167         \n",
       "\n",
       "[223560 rows x 17 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_df = pd.read_csv(merge_f, sep=\"\\t\", index_col=0)\n",
    "merge_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save each stable RNA file, then merge those files. \n",
    "Then for our merged peak file we use in beginning, merge with these combined stable rnas and take only the overlaps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:24:59.413267Z",
     "start_time": "2020-09-01T17:24:20.993933Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3051: DtypeWarning: Columns (24) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n",
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3051: DtypeWarning: Columns (16,17,18,19) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mergePeaks -d given -strand stable_rna/GCF/stable_Brain_START1_STARTinput2_RNASeq2.tss.txt stable_rna/GCF/stable_Brain_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Liver_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Pancreas_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Brain_START2_STARTinput2_RNASeq1.tss.txt stable_rna/GCF/stable_Liver_START2_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_MiscOrgans_START2_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Kidney_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Brain_START1_STARTinput2_RNASeq1.tss.txt stable_rna/GCF/stable_Muscle_START2_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Brain_START2_STARTinput1_RNASeq2.tss.txt stable_rna/GCF/stable_Kidney_START2_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Lung_START2_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Brain_START3_STARTinput1_RNASeq2.tss.txt stable_rna/GCF/stable_Spleen_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Brain_START3_STARTinput2_RNASeq2.tss.txt stable_rna/GCF/stable_Brain_START2_STARTinput2_RNASeq2.tss.txt stable_rna/GCF/stable_Lung_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Heart_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_FemaleReproductive_START2_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Brain_START2_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Brain_START1_STARTinput1_RNASeq2.tss.txt stable_rna/GCF/stable_Muscle_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Spleen_START2_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_FemaleReproductive_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Brain_START3_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Pancreas_START2_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Brain_START3_STARTinput2_RNASeq1.tss.txt stable_rna/GCF/stable_MiscOrgans_START1_STARTinput1_RNASeq1.tss.txt stable_rna/GCF/stable_Heart_START2_STARTinput1_RNASeq1.tss.txt > stable_rna/GCF/merged_stable_peaks.tsv\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples_stable = dict()\n",
    "for i in merged_stable_files:\n",
    "    curr = pd.read_csv(i, sep=\"\\t\", index_col=0)\n",
    "    curr = curr[curr[\"Stable/Unstable\"] == \"stable\"]\n",
    "    samples_stable[join(save_dir, \"stable_\" + os.path.basename(i))] = curr\n",
    "    curr.to_csv(join(save_dir, \"stable_\" + os.path.basename(i)), sep='\\t')\n",
    "\n",
    "\n",
    "\n",
    "#cmd = f\"mergePeaks {' '.split(glob.glob(join(save_dir, 'stable_*')))}\n",
    "dist = 'given'\n",
    "input_files = glob.glob(join(save_dir, 'stable_*'))\n",
    "cmd = f\"mergePeaks -d {dist} -strand {' '.join(input_files)} > {out_csrna_merged}\"\n",
    "print(cmd)\n",
    "os.system(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:00.324402Z",
     "start_time": "2020-09-01T17:24:59.415565Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x7f33d5ae35c0>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sns.countplot(pd.read_csv(out_csrna_merged, sep='\\t', index_col=0)[\"Total subpeaks\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge with eTSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:00.344840Z",
     "start_time": "2020-09-01T17:25:00.326340Z"
    },
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "from tss.utils import Homer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:03.033990Z",
     "start_time": "2020-09-01T17:25:00.346686Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mergePeaks -d 151 -venn stable_rna/GCF/eTSS_merged_stable_peaks.tsv.merge_venn -matrix stable_rna/GCF/eTSS_merged_stable_peaks.tsv.merge_mat -strand GCF/eTSS/TSS.bed stable_rna/GCF/merged_stable_peaks.tsv > stable_rna/GCF/eTSS_merged_stable_peaks.tsv.merge\n"
     ]
    }
   ],
   "source": [
    "Homer.merge_peaks([eTSS_bed_f, out_csrna_merged], out_eTSS_csrna_merged, dist=151, type_merge='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:03.065638Z",
     "start_time": "2020-09-01T17:25:03.036251Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GCF/eTSS/TSS.bed</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "      <th>Total</th>\n",
       "      <th>Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>46818</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>X</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21350</td>\n",
       "      <td>GCF/eTSS/TSS.bed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>X</td>\n",
       "      <td>X</td>\n",
       "      <td>19097</td>\n",
       "      <td>GCF/eTSS/TSS.bed|stable_rna/GCF/merged_stable_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GCF/eTSS/TSS.bed stable_rna/GCF/merged_stable_peaks.tsv  Total  \\\n",
       "0              NaN                                      X  46818   \n",
       "1                X                                    NaN  21350   \n",
       "2                X                                      X  19097   \n",
       "\n",
       "                                                Name  \n",
       "0             stable_rna/GCF/merged_stable_peaks.tsv  \n",
       "1                                   GCF/eTSS/TSS.bed  \n",
       "2  GCF/eTSS/TSS.bed|stable_rna/GCF/merged_stable_...  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(out_eTSS_csrna_merged + \"_venn\",sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:03.269547Z",
     "start_time": "2020-09-01T17:25:03.067560Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>GCF/eTSS/TSS.bed</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NW_020822381.1</td>\n",
       "      <td>3321150</td>\n",
       "      <td>3321300</td>\n",
       "      <td>-</td>\n",
       "      <td>12.100000</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822381.1-3321225-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NW_020822410.1</td>\n",
       "      <td>3731831</td>\n",
       "      <td>3731981</td>\n",
       "      <td>+</td>\n",
       "      <td>50.799999</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822410.1-3731906-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>67126802</td>\n",
       "      <td>67126952</td>\n",
       "      <td>-</td>\n",
       "      <td>57.299999</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822601.1-67126877-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NW_020822505.1</td>\n",
       "      <td>1552632</td>\n",
       "      <td>1552782</td>\n",
       "      <td>+</td>\n",
       "      <td>11.900000</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822505.1-1552707-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>NW_020822616.1</td>\n",
       "      <td>6038910</td>\n",
       "      <td>6039060</td>\n",
       "      <td>+</td>\n",
       "      <td>1627.942871</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822616.1-6038973-14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Chr     Start       End Strand         Stat  \\\n",
       "ID                                                           \n",
       "0   NW_020822381.1   3321150   3321300      -    12.100000   \n",
       "1   NW_020822410.1   3731831   3731981      +    50.799999   \n",
       "2   NW_020822601.1  67126802  67126952      -    57.299999   \n",
       "3   NW_020822505.1   1552632   1552782      +    11.900000   \n",
       "4   NW_020822616.1   6038910   6039060      +  1627.942871   \n",
       "\n",
       "                              Parent files  Total subpeaks GCF/eTSS/TSS.bed  \\\n",
       "ID                                                                            \n",
       "0   stable_rna/GCF/merged_stable_peaks.tsv               1              NaN   \n",
       "1   stable_rna/GCF/merged_stable_peaks.tsv               1              NaN   \n",
       "2   stable_rna/GCF/merged_stable_peaks.tsv               1              NaN   \n",
       "3   stable_rna/GCF/merged_stable_peaks.tsv               1              NaN   \n",
       "4   stable_rna/GCF/merged_stable_peaks.tsv               1              NaN   \n",
       "\n",
       "   stable_rna/GCF/merged_stable_peaks.tsv  \n",
       "ID                                         \n",
       "0         Merged-NW_020822381.1-3321225-2  \n",
       "1         Merged-NW_020822410.1-3731906-1  \n",
       "2        Merged-NW_020822601.1-67126877-1  \n",
       "3         Merged-NW_020822505.1-1552707-1  \n",
       "4        Merged-NW_020822616.1-6038973-14  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eTSS_csrna_merged = pd.read_csv(out_eTSS_csrna_merged,sep=\"\\t\",index_col=0)\n",
    "eTSS_csrna_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filter for peaks not seen in eTSS but is stable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:03.318430Z",
     "start_time": "2020-09-01T17:25:03.271363Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>GCF/eTSS/TSS.bed</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NW_020822381.1</td>\n",
       "      <td>3321150</td>\n",
       "      <td>3321300</td>\n",
       "      <td>-</td>\n",
       "      <td>12.100000</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822381.1-3321225-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NW_020822410.1</td>\n",
       "      <td>3731831</td>\n",
       "      <td>3731981</td>\n",
       "      <td>+</td>\n",
       "      <td>50.799999</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822410.1-3731906-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>67126802</td>\n",
       "      <td>67126952</td>\n",
       "      <td>-</td>\n",
       "      <td>57.299999</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822601.1-67126877-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NW_020822505.1</td>\n",
       "      <td>1552632</td>\n",
       "      <td>1552782</td>\n",
       "      <td>+</td>\n",
       "      <td>11.900000</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822505.1-1552707-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>NW_020822616.1</td>\n",
       "      <td>6038910</td>\n",
       "      <td>6039060</td>\n",
       "      <td>+</td>\n",
       "      <td>1627.942871</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822616.1-6038973-14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46813</td>\n",
       "      <td>NW_020822438.1</td>\n",
       "      <td>15115356</td>\n",
       "      <td>15115506</td>\n",
       "      <td>+</td>\n",
       "      <td>74.199997</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822438.1-15115431-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46814</td>\n",
       "      <td>NW_020822461.1</td>\n",
       "      <td>33025279</td>\n",
       "      <td>33025429</td>\n",
       "      <td>+</td>\n",
       "      <td>17.799999</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822461.1-33025354-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46815</td>\n",
       "      <td>NW_020822644.1</td>\n",
       "      <td>132585</td>\n",
       "      <td>132735</td>\n",
       "      <td>-</td>\n",
       "      <td>48.625000</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822644.1-132643-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46816</td>\n",
       "      <td>NW_020822439.1</td>\n",
       "      <td>72613039</td>\n",
       "      <td>72613189</td>\n",
       "      <td>-</td>\n",
       "      <td>7.700000</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822439.1-72613114-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>46817</td>\n",
       "      <td>NW_020822468.1</td>\n",
       "      <td>5143469</td>\n",
       "      <td>5143619</td>\n",
       "      <td>-</td>\n",
       "      <td>25.529999</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-NW_020822468.1-5143545-10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>46818 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Chr     Start       End Strand         Stat  \\\n",
       "ID                                                              \n",
       "0      NW_020822381.1   3321150   3321300      -    12.100000   \n",
       "1      NW_020822410.1   3731831   3731981      +    50.799999   \n",
       "2      NW_020822601.1  67126802  67126952      -    57.299999   \n",
       "3      NW_020822505.1   1552632   1552782      +    11.900000   \n",
       "4      NW_020822616.1   6038910   6039060      +  1627.942871   \n",
       "...               ...       ...       ...    ...          ...   \n",
       "46813  NW_020822438.1  15115356  15115506      +    74.199997   \n",
       "46814  NW_020822461.1  33025279  33025429      +    17.799999   \n",
       "46815  NW_020822644.1    132585    132735      -    48.625000   \n",
       "46816  NW_020822439.1  72613039  72613189      -     7.700000   \n",
       "46817  NW_020822468.1   5143469   5143619      -    25.529999   \n",
       "\n",
       "                                 Parent files  Total subpeaks  \\\n",
       "ID                                                              \n",
       "0      stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "1      stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "2      stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "3      stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "4      stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "...                                       ...             ...   \n",
       "46813  stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "46814  stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "46815  stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "46816  stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "46817  stable_rna/GCF/merged_stable_peaks.tsv               1   \n",
       "\n",
       "      GCF/eTSS/TSS.bed stable_rna/GCF/merged_stable_peaks.tsv  \n",
       "ID                                                             \n",
       "0                  NaN        Merged-NW_020822381.1-3321225-2  \n",
       "1                  NaN        Merged-NW_020822410.1-3731906-1  \n",
       "2                  NaN       Merged-NW_020822601.1-67126877-1  \n",
       "3                  NaN        Merged-NW_020822505.1-1552707-1  \n",
       "4                  NaN       Merged-NW_020822616.1-6038973-14  \n",
       "...                ...                                    ...  \n",
       "46813              NaN       Merged-NW_020822438.1-15115431-1  \n",
       "46814              NaN       Merged-NW_020822461.1-33025354-1  \n",
       "46815              NaN         Merged-NW_020822644.1-132643-4  \n",
       "46816              NaN       Merged-NW_020822439.1-72613114-1  \n",
       "46817              NaN       Merged-NW_020822468.1-5143545-10  \n",
       "\n",
       "[46818 rows x 9 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stable_minus_eTSS = eTSS_csrna_merged[eTSS_csrna_merged[\"Parent files\"] == out_csrna_merged]\n",
    "stable_minus_eTSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge with our mergePeaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:13.433554Z",
     "start_time": "2020-09-01T17:25:03.320212Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mergePeaks -d given -strand GCF/merged/samples.merge stable_rna/GCF/merged_stable_peaks.tsv -venn stable_rna/GCF/merged_stable_peaks.venn > stable_rna/GCF/all_merged_stable_peaks.tsv \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3051: DtypeWarning: Columns (8,9) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "cmd = f\"mergePeaks -d {dist} -strand {all_merged_f} {out_csrna_merged} -venn {join(save_dir, 'merged_stable_peaks.venn')} > {out_all_csrna_merged} \"\n",
    "print(cmd)\n",
    "os.system(cmd)\n",
    "#all_merged_f \n",
    "\n",
    "### For overlapping TSS \n",
    "merged_df = pd.read_csv(out_all_csrna_merged, sep='\\t', index_col=0)\n",
    "merged_df = merged_df[~(merged_df.isnull().any(axis=1))]\n",
    "merged_df.to_csv(out_all_csrna_merged, sep='\\t')\n",
    "\n",
    "#Homer.merge_peaks(input_files, output_file, dist='given', type_merge=''):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:13.467263Z",
     "start_time": "2020-09-01T17:25:13.436087Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GCF/merged/samples.merge</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "      <th>Total</th>\n",
       "      <th>Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>6347</td>\n",
       "      <td>stable_rna/GCF/merged_stable_peaks.tsv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>X</td>\n",
       "      <td>NaN</td>\n",
       "      <td>481087</td>\n",
       "      <td>GCF/merged/samples.merge</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>X</td>\n",
       "      <td>X</td>\n",
       "      <td>58808</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GCF/merged/samples.merge stable_rna/GCF/merged_stable_peaks.tsv   Total  \\\n",
       "0                      NaN                                      X    6347   \n",
       "1                        X                                    NaN  481087   \n",
       "2                        X                                      X   58808   \n",
       "\n",
       "                                                Name  \n",
       "0             stable_rna/GCF/merged_stable_peaks.tsv  \n",
       "1                           GCF/merged/samples.merge  \n",
       "2  GCF/merged/samples.merge|stable_rna/GCF/merged...  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(\"stable_rna/GCF/merged_stable_peaks.venn\",sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Take the peaks that are not seen in eTSS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### From stable_minus_eTSS,  extract the IDs and take those that overlap with the merged all peaks."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:13.711483Z",
     "start_time": "2020-09-01T17:25:13.469209Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>GCF/merged/samples.merge</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#name (cmd = mergePeaks -d given -strand GCF/merged/samples.merge stable_rna/GCF/merged_stable_peaks.tsv -venn stable_rna/GCF/merged_stable_peaks.venn)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822429.1-6852555-2</td>\n",
       "      <td>NW_020822429.1</td>\n",
       "      <td>6852464</td>\n",
       "      <td>6852668</td>\n",
       "      <td>-</td>\n",
       "      <td>32.527054</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>2799</td>\n",
       "      <td>Merged-NW_020822429.1-6852544-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822429.1-8601620-2</td>\n",
       "      <td>NW_020822429.1</td>\n",
       "      <td>8601541</td>\n",
       "      <td>8601697</td>\n",
       "      <td>+</td>\n",
       "      <td>27.842501</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>537618</td>\n",
       "      <td>Merged-NW_020822429.1-8601622-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822519.1-2588673-2</td>\n",
       "      <td>NW_020822519.1</td>\n",
       "      <td>2588568</td>\n",
       "      <td>2588831</td>\n",
       "      <td>-</td>\n",
       "      <td>41.549290</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>21856</td>\n",
       "      <td>Merged-NW_020822519.1-2588648-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822407.1-5171143-3</td>\n",
       "      <td>NW_020822407.1</td>\n",
       "      <td>5170887</td>\n",
       "      <td>5171328</td>\n",
       "      <td>+</td>\n",
       "      <td>22.681282</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>3</td>\n",
       "      <td>1190</td>\n",
       "      <td>Merged-NW_020822407.1-5171071-2,Merged-NW_0208...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822615.1-22431564-2</td>\n",
       "      <td>NW_020822615.1</td>\n",
       "      <td>22431441</td>\n",
       "      <td>22431672</td>\n",
       "      <td>-</td>\n",
       "      <td>39.881836</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>1192</td>\n",
       "      <td>Merged-NW_020822615.1-22431582-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822454.1-15400704-3</td>\n",
       "      <td>NW_020822454.1</td>\n",
       "      <td>15400490</td>\n",
       "      <td>15400911</td>\n",
       "      <td>-</td>\n",
       "      <td>10.926667</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>3</td>\n",
       "      <td>13857,323573</td>\n",
       "      <td>Merged-NW_020822454.1-15400711-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822370.1-25440839-4</td>\n",
       "      <td>NW_020822370.1</td>\n",
       "      <td>25440552</td>\n",
       "      <td>25441131</td>\n",
       "      <td>+</td>\n",
       "      <td>123.615433</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>4</td>\n",
       "      <td>165447,195267</td>\n",
       "      <td>Merged-NW_020822370.1-25440729-4,Merged-NW_020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822529.1-58424804-3</td>\n",
       "      <td>NW_020822529.1</td>\n",
       "      <td>58424583</td>\n",
       "      <td>58425084</td>\n",
       "      <td>+</td>\n",
       "      <td>4.830000</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>3</td>\n",
       "      <td>81715,532831</td>\n",
       "      <td>Merged-NW_020822529.1-58424805-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822423.1-205758-3</td>\n",
       "      <td>NW_020822423.1</td>\n",
       "      <td>205600</td>\n",
       "      <td>205929</td>\n",
       "      <td>-</td>\n",
       "      <td>3.807500</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>3</td>\n",
       "      <td>105511,287550</td>\n",
       "      <td>Merged-NW_020822423.1-205746-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822615.1-17196339-3</td>\n",
       "      <td>NW_020822615.1</td>\n",
       "      <td>17196157</td>\n",
       "      <td>17196530</td>\n",
       "      <td>-</td>\n",
       "      <td>10.287867</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>3</td>\n",
       "      <td>333179,137500</td>\n",
       "      <td>Merged-NW_020822615.1-17196380-5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58808 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               chr     start  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                             \n",
       "Merged-NW_020822429.1-6852555-2                     NW_020822429.1   6852464   \n",
       "Merged-NW_020822429.1-8601620-2                     NW_020822429.1   8601541   \n",
       "Merged-NW_020822519.1-2588673-2                     NW_020822519.1   2588568   \n",
       "Merged-NW_020822407.1-5171143-3                     NW_020822407.1   5170887   \n",
       "Merged-NW_020822615.1-22431564-2                    NW_020822615.1  22431441   \n",
       "...                                                            ...       ...   \n",
       "Merged-NW_020822454.1-15400704-3                    NW_020822454.1  15400490   \n",
       "Merged-NW_020822370.1-25440839-4                    NW_020822370.1  25440552   \n",
       "Merged-NW_020822529.1-58424804-3                    NW_020822529.1  58424583   \n",
       "Merged-NW_020822423.1-205758-3                      NW_020822423.1    205600   \n",
       "Merged-NW_020822615.1-17196339-3                    NW_020822615.1  17196157   \n",
       "\n",
       "                                                         end strand  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                    \n",
       "Merged-NW_020822429.1-6852555-2                      6852668      -   \n",
       "Merged-NW_020822429.1-8601620-2                      8601697      +   \n",
       "Merged-NW_020822519.1-2588673-2                      2588831      -   \n",
       "Merged-NW_020822407.1-5171143-3                      5171328      +   \n",
       "Merged-NW_020822615.1-22431564-2                    22431672      -   \n",
       "...                                                      ...    ...   \n",
       "Merged-NW_020822454.1-15400704-3                    15400911      -   \n",
       "Merged-NW_020822370.1-25440839-4                    25441131      +   \n",
       "Merged-NW_020822529.1-58424804-3                    58425084      +   \n",
       "Merged-NW_020822423.1-205758-3                        205929      -   \n",
       "Merged-NW_020822615.1-17196339-3                    17196530      -   \n",
       "\n",
       "                                                          Stat  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...               \n",
       "Merged-NW_020822429.1-6852555-2                      32.527054   \n",
       "Merged-NW_020822429.1-8601620-2                      27.842501   \n",
       "Merged-NW_020822519.1-2588673-2                      41.549290   \n",
       "Merged-NW_020822407.1-5171143-3                      22.681282   \n",
       "Merged-NW_020822615.1-22431564-2                     39.881836   \n",
       "...                                                        ...   \n",
       "Merged-NW_020822454.1-15400704-3                     10.926667   \n",
       "Merged-NW_020822370.1-25440839-4                    123.615433   \n",
       "Merged-NW_020822529.1-58424804-3                      4.830000   \n",
       "Merged-NW_020822423.1-205758-3                        3.807500   \n",
       "Merged-NW_020822615.1-17196339-3                     10.287867   \n",
       "\n",
       "                                                                                         Parent files  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                                                      \n",
       "Merged-NW_020822429.1-6852555-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822429.1-8601620-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822519.1-2588673-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822407.1-5171143-3                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822615.1-22431564-2                    GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "...                                                                                               ...   \n",
       "Merged-NW_020822454.1-15400704-3                    GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822370.1-25440839-4                    GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822529.1-58424804-3                    GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822423.1-205758-3                      GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822615.1-17196339-3                    GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "\n",
       "                                                    Total subpeaks  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                   \n",
       "Merged-NW_020822429.1-6852555-2                                  2   \n",
       "Merged-NW_020822429.1-8601620-2                                  2   \n",
       "Merged-NW_020822519.1-2588673-2                                  2   \n",
       "Merged-NW_020822407.1-5171143-3                                  3   \n",
       "Merged-NW_020822615.1-22431564-2                                 2   \n",
       "...                                                            ...   \n",
       "Merged-NW_020822454.1-15400704-3                                 3   \n",
       "Merged-NW_020822370.1-25440839-4                                 4   \n",
       "Merged-NW_020822529.1-58424804-3                                 3   \n",
       "Merged-NW_020822423.1-205758-3                                   3   \n",
       "Merged-NW_020822615.1-17196339-3                                 3   \n",
       "\n",
       "                                                   GCF/merged/samples.merge  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                            \n",
       "Merged-NW_020822429.1-6852555-2                                        2799   \n",
       "Merged-NW_020822429.1-8601620-2                                      537618   \n",
       "Merged-NW_020822519.1-2588673-2                                       21856   \n",
       "Merged-NW_020822407.1-5171143-3                                        1190   \n",
       "Merged-NW_020822615.1-22431564-2                                       1192   \n",
       "...                                                                     ...   \n",
       "Merged-NW_020822454.1-15400704-3                               13857,323573   \n",
       "Merged-NW_020822370.1-25440839-4                              165447,195267   \n",
       "Merged-NW_020822529.1-58424804-3                               81715,532831   \n",
       "Merged-NW_020822423.1-205758-3                                105511,287550   \n",
       "Merged-NW_020822615.1-17196339-3                              333179,137500   \n",
       "\n",
       "                                                               stable_rna/GCF/merged_stable_peaks.tsv  \n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                                                     \n",
       "Merged-NW_020822429.1-6852555-2                                       Merged-NW_020822429.1-6852544-3  \n",
       "Merged-NW_020822429.1-8601620-2                                       Merged-NW_020822429.1-8601622-1  \n",
       "Merged-NW_020822519.1-2588673-2                                       Merged-NW_020822519.1-2588648-3  \n",
       "Merged-NW_020822407.1-5171143-3                     Merged-NW_020822407.1-5171071-2,Merged-NW_0208...  \n",
       "Merged-NW_020822615.1-22431564-2                                     Merged-NW_020822615.1-22431582-3  \n",
       "...                                                                                               ...  \n",
       "Merged-NW_020822454.1-15400704-3                                     Merged-NW_020822454.1-15400711-2  \n",
       "Merged-NW_020822370.1-25440839-4                    Merged-NW_020822370.1-25440729-4,Merged-NW_020...  \n",
       "Merged-NW_020822529.1-58424804-3                                     Merged-NW_020822529.1-58424805-1  \n",
       "Merged-NW_020822423.1-205758-3                                         Merged-NW_020822423.1-205746-1  \n",
       "Merged-NW_020822615.1-17196339-3                                     Merged-NW_020822615.1-17196380-5  \n",
       "\n",
       "[58808 rows x 9 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_csrna_merged = pd.read_csv(out_all_csrna_merged,sep='\\t',index_col=0)\n",
    "all_csrna_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:13.734866Z",
     "start_time": "2020-09-01T17:25:13.713684Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID\n",
       "0         Merged-NW_020822381.1-3321225-2\n",
       "1         Merged-NW_020822410.1-3731906-1\n",
       "2        Merged-NW_020822601.1-67126877-1\n",
       "3         Merged-NW_020822505.1-1552707-1\n",
       "4        Merged-NW_020822616.1-6038973-14\n",
       "                       ...               \n",
       "46813    Merged-NW_020822438.1-15115431-1\n",
       "46814    Merged-NW_020822461.1-33025354-1\n",
       "46815      Merged-NW_020822644.1-132643-4\n",
       "46816    Merged-NW_020822439.1-72613114-1\n",
       "46817    Merged-NW_020822468.1-5143545-10\n",
       "Name: stable_rna/GCF/merged_stable_peaks.tsv, Length: 46818, dtype: object"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stable_minus_eTSS[out_csrna_merged]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:13.783253Z",
     "start_time": "2020-09-01T17:25:13.736648Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "232"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_csrna_merged[all_merged_f].str.contains(',').sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load samples.merge.anno to then extract the stable_minus_eTSS peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:17.079997Z",
     "start_time": "2020-09-01T17:25:13.785070Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3051: DtypeWarning: Columns (8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>MiscOrgans_START1_and_MiscOrgans_startinput1</th>\n",
       "      <th>Liver_START2_and_Liver_startinput1</th>\n",
       "      <th>Brain_GROCap1_and_Brain_GRO1</th>\n",
       "      <th>...</th>\n",
       "      <th>MiscOrgans_START2_and_MiscOrgans_startinput1</th>\n",
       "      <th>Spleen_START1_and_Spleen_startinput1</th>\n",
       "      <th>Lung_START2_and_Lung_startinput1</th>\n",
       "      <th>CHO_START1_and_CHO_startinput1</th>\n",
       "      <th>FemaleReproductive_START1_and_FemaleReproductive_startinput1</th>\n",
       "      <th>Kidney_START2_and_Kidney_startinput1</th>\n",
       "      <th>Kidney_GROCap1_and_Kidney_GRO1</th>\n",
       "      <th>CHO_GROCap1_and_CHO_GRO1</th>\n",
       "      <th>BMDMwt_GROCap1_and_BMDMwt_GRO1</th>\n",
       "      <th>Spleen_START2_and_Spleen_startinput1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NW_020822452.1</td>\n",
       "      <td>16590187</td>\n",
       "      <td>16590584</td>\n",
       "      <td>+</td>\n",
       "      <td>16.227333</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>15</td>\n",
       "      <td>NW_020822452.1-1136</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822452.1-1139</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822452.1-450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822452.1-419</td>\n",
       "      <td>NW_020822452.1-145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NW_020822597.1</td>\n",
       "      <td>23595304</td>\n",
       "      <td>23595489</td>\n",
       "      <td>+</td>\n",
       "      <td>5.740000</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>16</td>\n",
       "      <td>NW_020822597.1-387</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822597.1-324</td>\n",
       "      <td>...</td>\n",
       "      <td>NW_020822597.1-103</td>\n",
       "      <td>NW_020822597.1-248</td>\n",
       "      <td>NW_020822597.1-10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822597.1-101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822597.1-126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>NW_020822680.1</td>\n",
       "      <td>728607</td>\n",
       "      <td>728829</td>\n",
       "      <td>+</td>\n",
       "      <td>7.579579</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>19</td>\n",
       "      <td>NW_020822680.1-92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822680.1-114</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822680.1-37</td>\n",
       "      <td>NW_020822680.1-56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822680.1-67</td>\n",
       "      <td>NW_020822680.1-19</td>\n",
       "      <td>NW_020822680.1-44</td>\n",
       "      <td>NW_020822680.1-72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NW_020822602.1</td>\n",
       "      <td>10883074</td>\n",
       "      <td>10883337</td>\n",
       "      <td>+</td>\n",
       "      <td>3.992000</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>20</td>\n",
       "      <td>NW_020822602.1-204</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NW_020822602.1-254</td>\n",
       "      <td>NW_020822602.1-122</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822602.1-226</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822602.1-303</td>\n",
       "      <td>NW_020822602.1-594</td>\n",
       "      <td>NW_020822602.1-266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>37427816</td>\n",
       "      <td>37427999</td>\n",
       "      <td>+</td>\n",
       "      <td>129.031738</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>23</td>\n",
       "      <td>NW_020822601.1-18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822601.1-1445</td>\n",
       "      <td>...</td>\n",
       "      <td>NW_020822601.1-52</td>\n",
       "      <td>NW_020822601.1-123</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822601.1-41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822601.1-1949</td>\n",
       "      <td>NW_020822601.1-992</td>\n",
       "      <td>NW_020822601.1-2287</td>\n",
       "      <td>NW_020822601.1-12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Chr     Start       End Strand        Stat  \\\n",
       "ID                                                          \n",
       "0   NW_020822452.1  16590187  16590584      +   16.227333   \n",
       "1   NW_020822597.1  23595304  23595489      +    5.740000   \n",
       "2   NW_020822680.1    728607    728829      +    7.579579   \n",
       "3   NW_020822602.1  10883074  10883337      +    3.992000   \n",
       "4   NW_020822601.1  37427816  37427999      +  129.031738   \n",
       "\n",
       "                                         Parent files  Total subpeaks  \\\n",
       "ID                                                                      \n",
       "0   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              15   \n",
       "1   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              16   \n",
       "2   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              19   \n",
       "3   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              20   \n",
       "4   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              23   \n",
       "\n",
       "   MiscOrgans_START1_and_MiscOrgans_startinput1  \\\n",
       "ID                                                \n",
       "0                           NW_020822452.1-1136   \n",
       "1                            NW_020822597.1-387   \n",
       "2                             NW_020822680.1-92   \n",
       "3                            NW_020822602.1-204   \n",
       "4                             NW_020822601.1-18   \n",
       "\n",
       "   Liver_START2_and_Liver_startinput1 Brain_GROCap1_and_Brain_GRO1  ...  \\\n",
       "ID                                                                  ...   \n",
       "0                                 NaN          NW_020822452.1-1139  ...   \n",
       "1                                 NaN           NW_020822597.1-324  ...   \n",
       "2                                 NaN           NW_020822680.1-114  ...   \n",
       "3                                 NaN                          NaN  ...   \n",
       "4                                 NaN          NW_020822601.1-1445  ...   \n",
       "\n",
       "   MiscOrgans_START2_and_MiscOrgans_startinput1  \\\n",
       "ID                                                \n",
       "0                                           NaN   \n",
       "1                            NW_020822597.1-103   \n",
       "2                                           NaN   \n",
       "3                            NW_020822602.1-254   \n",
       "4                             NW_020822601.1-52   \n",
       "\n",
       "   Spleen_START1_and_Spleen_startinput1 Lung_START2_and_Lung_startinput1  \\\n",
       "ID                                                                         \n",
       "0                                   NaN                              NaN   \n",
       "1                    NW_020822597.1-248                NW_020822597.1-10   \n",
       "2                                   NaN                              NaN   \n",
       "3                    NW_020822602.1-122                              NaN   \n",
       "4                    NW_020822601.1-123                              NaN   \n",
       "\n",
       "   CHO_START1_and_CHO_startinput1  \\\n",
       "ID                                  \n",
       "0              NW_020822452.1-450   \n",
       "1                             NaN   \n",
       "2               NW_020822680.1-37   \n",
       "3              NW_020822602.1-226   \n",
       "4               NW_020822601.1-41   \n",
       "\n",
       "   FemaleReproductive_START1_and_FemaleReproductive_startinput1  \\\n",
       "ID                                                                \n",
       "0                                                 NaN             \n",
       "1                                  NW_020822597.1-101             \n",
       "2                                   NW_020822680.1-56             \n",
       "3                                                 NaN             \n",
       "4                                                 NaN             \n",
       "\n",
       "   Kidney_START2_and_Kidney_startinput1 Kidney_GROCap1_and_Kidney_GRO1  \\\n",
       "ID                                                                       \n",
       "0                                   NaN             NW_020822452.1-419   \n",
       "1                                   NaN                            NaN   \n",
       "2                                   NaN              NW_020822680.1-67   \n",
       "3                                   NaN                            NaN   \n",
       "4                                   NaN            NW_020822601.1-1949   \n",
       "\n",
       "   CHO_GROCap1_and_CHO_GRO1 BMDMwt_GROCap1_and_BMDMwt_GRO1  \\\n",
       "ID                                                           \n",
       "0        NW_020822452.1-145                            NaN   \n",
       "1                       NaN                            NaN   \n",
       "2         NW_020822680.1-19              NW_020822680.1-44   \n",
       "3        NW_020822602.1-303             NW_020822602.1-594   \n",
       "4        NW_020822601.1-992            NW_020822601.1-2287   \n",
       "\n",
       "   Spleen_START2_and_Spleen_startinput1  \n",
       "ID                                       \n",
       "0                                   NaN  \n",
       "1                    NW_020822597.1-126  \n",
       "2                     NW_020822680.1-72  \n",
       "3                    NW_020822602.1-266  \n",
       "4                     NW_020822601.1-12  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_df = pd.read_csv(all_merged_f,sep='\\t',index_col=0)\n",
    "merge_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:21.515473Z",
     "start_time": "2020-09-01T17:25:17.081903Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>Nearest PromoterID</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl data/processed/GCF/merged/samples.merge /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gff)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>539323</td>\n",
       "      <td>NW_020822460.1</td>\n",
       "      <td>1084630</td>\n",
       "      <td>1084782</td>\n",
       "      <td>+</td>\n",
       "      <td>66162.929688</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|L...</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003482412.1-1;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003482412.1-1;Parent...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>ID=exon-XR_003482412.1-1;Parent=rna-XR_0034824...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>353477</td>\n",
       "      <td>NW_020822618.1</td>\n",
       "      <td>401845</td>\n",
       "      <td>402000</td>\n",
       "      <td>-</td>\n",
       "      <td>21905.132812</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|L...</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003487470.1-1;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003487470.1-1;Parent...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>ID=exon-XR_003487470.1-1;Parent=rna-XR_0034874...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>145</td>\n",
       "      <td>NW_020822410.1</td>\n",
       "      <td>4827637</td>\n",
       "      <td>4827815</td>\n",
       "      <td>+</td>\n",
       "      <td>20179.468750</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|L...</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027391402.1-1;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XM_027391402.1-1;Parent...</td>\n",
       "      <td>-11.0</td>\n",
       "      <td>ID=exon-XM_027391402.1-1;Parent=rna-XM_0273914...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>349134</td>\n",
       "      <td>NW_020822505.1</td>\n",
       "      <td>689618</td>\n",
       "      <td>689773</td>\n",
       "      <td>-</td>\n",
       "      <td>18150.236328</td>\n",
       "      <td>Brain_GROCap1_and_Brain_GRO1|Heart_START1_and_...</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003483964.1-10;Parent...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003483964.1-10;Paren...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ID=exon-XR_003484049.1-1;Parent=rna-XR_0034840...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>300005</td>\n",
       "      <td>NW_020822602.1</td>\n",
       "      <td>9283364</td>\n",
       "      <td>9283514</td>\n",
       "      <td>+</td>\n",
       "      <td>15297.872070</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027422995.1-6;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XM_027422995.1-6;Parent...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>ID=exon-XR_003486731.1-1;Parent=rna-XR_0034867...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               Chr    Start  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                            \n",
       "539323                                              NW_020822460.1  1084630   \n",
       "353477                                              NW_020822618.1   401845   \n",
       "145                                                 NW_020822410.1  4827637   \n",
       "349134                                              NW_020822505.1   689618   \n",
       "300005                                              NW_020822602.1  9283364   \n",
       "\n",
       "                                                        End Strand  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                   \n",
       "539323                                              1084782      +   \n",
       "353477                                               402000      -   \n",
       "145                                                 4827815      +   \n",
       "349134                                               689773      -   \n",
       "300005                                              9283514      +   \n",
       "\n",
       "                                                      Peak Score  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                 \n",
       "539323                                              66162.929688   \n",
       "353477                                              21905.132812   \n",
       "145                                                 20179.468750   \n",
       "349134                                              18150.236328   \n",
       "300005                                              15297.872070   \n",
       "\n",
       "                                                                              Focus Ratio/Region Size  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                                                      \n",
       "539323                                              MiscOrgans_START1_and_MiscOrgans_startinput1|L...   \n",
       "353477                                              MiscOrgans_START1_and_MiscOrgans_startinput1|L...   \n",
       "145                                                 MiscOrgans_START1_and_MiscOrgans_startinput1|L...   \n",
       "349134                                              Brain_GROCap1_and_Brain_GRO1|Heart_START1_and_...   \n",
       "300005                                              MiscOrgans_START1_and_MiscOrgans_startinput1|B...   \n",
       "\n",
       "                                                                                           Annotation  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                                                      \n",
       "539323                                              promoter-TSS (ID=exon-XR_003482412.1-1;Parent=...   \n",
       "353477                                              promoter-TSS (ID=exon-XR_003487470.1-1;Parent=...   \n",
       "145                                                 promoter-TSS (ID=exon-XM_027391402.1-1;Parent=...   \n",
       "349134                                              promoter-TSS (ID=exon-XR_003483964.1-10;Parent...   \n",
       "300005                                              promoter-TSS (ID=exon-XM_027422995.1-6;Parent=...   \n",
       "\n",
       "                                                                                  Detailed Annotation  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                                                      \n",
       "539323                                              -promoter-TSS (ID=exon-XR_003482412.1-1;Parent...   \n",
       "353477                                              -promoter-TSS (ID=exon-XR_003487470.1-1;Parent...   \n",
       "145                                                 -promoter-TSS (ID=exon-XM_027391402.1-1;Parent...   \n",
       "349134                                              -promoter-TSS (ID=exon-XR_003483964.1-10;Paren...   \n",
       "300005                                              -promoter-TSS (ID=exon-XM_027422995.1-6;Parent...   \n",
       "\n",
       "                                                    Distance to TSS  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                    \n",
       "539323                                                         -1.0   \n",
       "353477                                                          4.0   \n",
       "145                                                           -11.0   \n",
       "349134                                                          3.0   \n",
       "300005                                                          0.0   \n",
       "\n",
       "                                                                                   Nearest PromoterID  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                                                      \n",
       "539323                                              ID=exon-XR_003482412.1-1;Parent=rna-XR_0034824...   \n",
       "353477                                              ID=exon-XR_003487470.1-1;Parent=rna-XR_0034874...   \n",
       "145                                                 ID=exon-XM_027391402.1-1;Parent=rna-XM_0273914...   \n",
       "349134                                              ID=exon-XR_003484049.1-1;Parent=rna-XR_0034840...   \n",
       "300005                                              ID=exon-XR_003486731.1-1;Parent=rna-XR_0034867...   \n",
       "\n",
       "                                                    Entrez ID  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...              \n",
       "539323                                                    NaN   \n",
       "353477                                                    NaN   \n",
       "145                                                       NaN   \n",
       "349134                                                    NaN   \n",
       "300005                                                    NaN   \n",
       "\n",
       "                                                    Nearest Unigene  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                    \n",
       "539323                                                          NaN   \n",
       "353477                                                          NaN   \n",
       "145                                                             NaN   \n",
       "349134                                                          NaN   \n",
       "300005                                                          NaN   \n",
       "\n",
       "                                                    Nearest Refseq  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                   \n",
       "539323                                                         NaN   \n",
       "353477                                                         NaN   \n",
       "145                                                            NaN   \n",
       "349134                                                         NaN   \n",
       "300005                                                         NaN   \n",
       "\n",
       "                                                    Nearest Ensembl  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                    \n",
       "539323                                                          NaN   \n",
       "353477                                                          NaN   \n",
       "145                                                             NaN   \n",
       "349134                                                          NaN   \n",
       "300005                                                          NaN   \n",
       "\n",
       "                                                    Gene Name  Gene Alias  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                          \n",
       "539323                                                    NaN         NaN   \n",
       "353477                                                    NaN         NaN   \n",
       "145                                                       NaN         NaN   \n",
       "349134                                                    NaN         NaN   \n",
       "300005                                                    NaN         NaN   \n",
       "\n",
       "                                                    Gene Description  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                     \n",
       "539323                                                           NaN   \n",
       "353477                                                           NaN   \n",
       "145                                                              NaN   \n",
       "349134                                                           NaN   \n",
       "300005                                                           NaN   \n",
       "\n",
       "                                                    Gene Type  \n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...             \n",
       "539323                                                    NaN  \n",
       "353477                                                    NaN  \n",
       "145                                                       NaN  \n",
       "349134                                                    NaN  \n",
       "300005                                                    NaN  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_merge_anno_df = pd.read_csv(all_merged_anno_f,sep='\\t',index_col=0)\n",
    "all_merge_anno_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:21.944886Z",
     "start_time": "2020-09-01T17:25:21.517286Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Intergenic', nan], dtype=object)"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_merge_anno_df.loc[~(all_merge_anno_df[\"Annotation\"].fillna('').str.contains(\"exon\"))][\"Detailed Annotation\"].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get the non-coding RNAs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.381129Z",
     "start_time": "2020-09-01T17:25:21.946763Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>Nearest PromoterID</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl data/processed/GCF/merged/samples.merge /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gff)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>539323</td>\n",
       "      <td>NW_020822460.1</td>\n",
       "      <td>1084630</td>\n",
       "      <td>1084782</td>\n",
       "      <td>+</td>\n",
       "      <td>66162.929688</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|L...</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003482412.1-1;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003482412.1-1;Parent...</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>ID=exon-XR_003482412.1-1;Parent=rna-XR_0034824...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>353477</td>\n",
       "      <td>NW_020822618.1</td>\n",
       "      <td>401845</td>\n",
       "      <td>402000</td>\n",
       "      <td>-</td>\n",
       "      <td>21905.132812</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|L...</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003487470.1-1;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003487470.1-1;Parent...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>ID=exon-XR_003487470.1-1;Parent=rna-XR_0034874...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>349134</td>\n",
       "      <td>NW_020822505.1</td>\n",
       "      <td>689618</td>\n",
       "      <td>689773</td>\n",
       "      <td>-</td>\n",
       "      <td>18150.236328</td>\n",
       "      <td>Brain_GROCap1_and_Brain_GRO1|Heart_START1_and_...</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003483964.1-10;Parent...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003483964.1-10;Paren...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>ID=exon-XR_003484049.1-1;Parent=rna-XR_0034840...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>509442</td>\n",
       "      <td>NW_020822421.1</td>\n",
       "      <td>11363589</td>\n",
       "      <td>11363741</td>\n",
       "      <td>+</td>\n",
       "      <td>11703.328125</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|L...</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003480724.1-1;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003480724.1-1;Parent...</td>\n",
       "      <td>-2.0</td>\n",
       "      <td>ID=exon-XR_003480724.1-1;Parent=rna-XR_0034807...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>418707</td>\n",
       "      <td>NW_020822519.1</td>\n",
       "      <td>30535313</td>\n",
       "      <td>30535519</td>\n",
       "      <td>+</td>\n",
       "      <td>6618.855469</td>\n",
       "      <td>Brain_GROCap1_and_Brain_GRO1|Liver_START1_and_...</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003484581.1-1;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003484581.1-1;Parent...</td>\n",
       "      <td>28.0</td>\n",
       "      <td>ID=exon-XR_003484611.1-1;Parent=rna-XR_0034846...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>430755</td>\n",
       "      <td>NW_020822439.1</td>\n",
       "      <td>25030902</td>\n",
       "      <td>25031052</td>\n",
       "      <td>+</td>\n",
       "      <td>0.339000</td>\n",
       "      <td>CHO_GROCap1_and_CHO_GRO1</td>\n",
       "      <td>TTS (ID=exon-XR_003481482.1-2;Parent=rna-XR_00...</td>\n",
       "      <td>-TTS (ID=exon-XR_003481482.1-2;Parent=rna-XR_0...</td>\n",
       "      <td>892.0</td>\n",
       "      <td>ID=exon-XR_003481482.1-2;Parent=rna-XR_0034814...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>343524</td>\n",
       "      <td>NW_020822658.1</td>\n",
       "      <td>3706656</td>\n",
       "      <td>3706806</td>\n",
       "      <td>-</td>\n",
       "      <td>0.325000</td>\n",
       "      <td>CHO_GROCap2_and_CHO_GRO1</td>\n",
       "      <td>exon (ID=exon-XR_003488114.1-1;Parent=rna-XR_0...</td>\n",
       "      <td>-exon (ID=exon-XR_003488114.1-1;Parent=rna-XR_...</td>\n",
       "      <td>554.0</td>\n",
       "      <td>ID=exon-XR_003488114.1-1;Parent=rna-XR_0034881...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>344967</td>\n",
       "      <td>NW_020822419.1</td>\n",
       "      <td>11645255</td>\n",
       "      <td>11645405</td>\n",
       "      <td>+</td>\n",
       "      <td>0.325000</td>\n",
       "      <td>CHO_GROCap2_and_CHO_GRO1</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003480616.1-1;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003480616.1-1;Parent...</td>\n",
       "      <td>-178.0</td>\n",
       "      <td>ID=exon-XR_003480618.1-1;Parent=rna-XR_0034806...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>343309</td>\n",
       "      <td>NW_020822698.1</td>\n",
       "      <td>6390597</td>\n",
       "      <td>6390747</td>\n",
       "      <td>-</td>\n",
       "      <td>0.325000</td>\n",
       "      <td>CHO_GROCap2_and_CHO_GRO1</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003488458.1-2;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003488458.1-2;Parent...</td>\n",
       "      <td>-494.0</td>\n",
       "      <td>ID=exon-XR_003488458.1-2;Parent=rna-XR_0034884...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>429753</td>\n",
       "      <td>NW_020822559.1</td>\n",
       "      <td>1602631</td>\n",
       "      <td>1602781</td>\n",
       "      <td>-</td>\n",
       "      <td>0.301000</td>\n",
       "      <td>CHO_GROCap1_and_CHO_GRO1</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003485612.1-1;Parent=...</td>\n",
       "      <td>-promoter-TSS (ID=exon-XR_003485612.1-1;Parent...</td>\n",
       "      <td>-601.0</td>\n",
       "      <td>ID=exon-XR_003485612.1-1;Parent=rna-XR_0034856...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12270 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               Chr     Start  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                             \n",
       "539323                                              NW_020822460.1   1084630   \n",
       "353477                                              NW_020822618.1    401845   \n",
       "349134                                              NW_020822505.1    689618   \n",
       "509442                                              NW_020822421.1  11363589   \n",
       "418707                                              NW_020822519.1  30535313   \n",
       "...                                                            ...       ...   \n",
       "430755                                              NW_020822439.1  25030902   \n",
       "343524                                              NW_020822658.1   3706656   \n",
       "344967                                              NW_020822419.1  11645255   \n",
       "343309                                              NW_020822698.1   6390597   \n",
       "429753                                              NW_020822559.1   1602631   \n",
       "\n",
       "                                                         End Strand  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                    \n",
       "539323                                               1084782      +   \n",
       "353477                                                402000      -   \n",
       "349134                                                689773      -   \n",
       "509442                                              11363741      +   \n",
       "418707                                              30535519      +   \n",
       "...                                                      ...    ...   \n",
       "430755                                              25031052      +   \n",
       "343524                                               3706806      -   \n",
       "344967                                              11645405      +   \n",
       "343309                                               6390747      -   \n",
       "429753                                               1602781      -   \n",
       "\n",
       "                                                      Peak Score  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                 \n",
       "539323                                              66162.929688   \n",
       "353477                                              21905.132812   \n",
       "349134                                              18150.236328   \n",
       "509442                                              11703.328125   \n",
       "418707                                               6618.855469   \n",
       "...                                                          ...   \n",
       "430755                                                  0.339000   \n",
       "343524                                                  0.325000   \n",
       "344967                                                  0.325000   \n",
       "343309                                                  0.325000   \n",
       "429753                                                  0.301000   \n",
       "\n",
       "                                                                              Focus Ratio/Region Size  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                                                      \n",
       "539323                                              MiscOrgans_START1_and_MiscOrgans_startinput1|L...   \n",
       "353477                                              MiscOrgans_START1_and_MiscOrgans_startinput1|L...   \n",
       "349134                                              Brain_GROCap1_and_Brain_GRO1|Heart_START1_and_...   \n",
       "509442                                              MiscOrgans_START1_and_MiscOrgans_startinput1|L...   \n",
       "418707                                              Brain_GROCap1_and_Brain_GRO1|Liver_START1_and_...   \n",
       "...                                                                                               ...   \n",
       "430755                                                                       CHO_GROCap1_and_CHO_GRO1   \n",
       "343524                                                                       CHO_GROCap2_and_CHO_GRO1   \n",
       "344967                                                                       CHO_GROCap2_and_CHO_GRO1   \n",
       "343309                                                                       CHO_GROCap2_and_CHO_GRO1   \n",
       "429753                                                                       CHO_GROCap1_and_CHO_GRO1   \n",
       "\n",
       "                                                                                           Annotation  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                                                      \n",
       "539323                                              promoter-TSS (ID=exon-XR_003482412.1-1;Parent=...   \n",
       "353477                                              promoter-TSS (ID=exon-XR_003487470.1-1;Parent=...   \n",
       "349134                                              promoter-TSS (ID=exon-XR_003483964.1-10;Parent...   \n",
       "509442                                              promoter-TSS (ID=exon-XR_003480724.1-1;Parent=...   \n",
       "418707                                              promoter-TSS (ID=exon-XR_003484581.1-1;Parent=...   \n",
       "...                                                                                               ...   \n",
       "430755                                              TTS (ID=exon-XR_003481482.1-2;Parent=rna-XR_00...   \n",
       "343524                                              exon (ID=exon-XR_003488114.1-1;Parent=rna-XR_0...   \n",
       "344967                                              promoter-TSS (ID=exon-XR_003480616.1-1;Parent=...   \n",
       "343309                                              promoter-TSS (ID=exon-XR_003488458.1-2;Parent=...   \n",
       "429753                                              promoter-TSS (ID=exon-XR_003485612.1-1;Parent=...   \n",
       "\n",
       "                                                                                  Detailed Annotation  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                                                      \n",
       "539323                                              -promoter-TSS (ID=exon-XR_003482412.1-1;Parent...   \n",
       "353477                                              -promoter-TSS (ID=exon-XR_003487470.1-1;Parent...   \n",
       "349134                                              -promoter-TSS (ID=exon-XR_003483964.1-10;Paren...   \n",
       "509442                                              -promoter-TSS (ID=exon-XR_003480724.1-1;Parent...   \n",
       "418707                                              -promoter-TSS (ID=exon-XR_003484581.1-1;Parent...   \n",
       "...                                                                                               ...   \n",
       "430755                                              -TTS (ID=exon-XR_003481482.1-2;Parent=rna-XR_0...   \n",
       "343524                                              -exon (ID=exon-XR_003488114.1-1;Parent=rna-XR_...   \n",
       "344967                                              -promoter-TSS (ID=exon-XR_003480616.1-1;Parent...   \n",
       "343309                                              -promoter-TSS (ID=exon-XR_003488458.1-2;Parent...   \n",
       "429753                                              -promoter-TSS (ID=exon-XR_003485612.1-1;Parent...   \n",
       "\n",
       "                                                    Distance to TSS  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                    \n",
       "539323                                                         -1.0   \n",
       "353477                                                          4.0   \n",
       "349134                                                          3.0   \n",
       "509442                                                         -2.0   \n",
       "418707                                                         28.0   \n",
       "...                                                             ...   \n",
       "430755                                                        892.0   \n",
       "343524                                                        554.0   \n",
       "344967                                                       -178.0   \n",
       "343309                                                       -494.0   \n",
       "429753                                                       -601.0   \n",
       "\n",
       "                                                                                   Nearest PromoterID  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                                                      \n",
       "539323                                              ID=exon-XR_003482412.1-1;Parent=rna-XR_0034824...   \n",
       "353477                                              ID=exon-XR_003487470.1-1;Parent=rna-XR_0034874...   \n",
       "349134                                              ID=exon-XR_003484049.1-1;Parent=rna-XR_0034840...   \n",
       "509442                                              ID=exon-XR_003480724.1-1;Parent=rna-XR_0034807...   \n",
       "418707                                              ID=exon-XR_003484611.1-1;Parent=rna-XR_0034846...   \n",
       "...                                                                                               ...   \n",
       "430755                                              ID=exon-XR_003481482.1-2;Parent=rna-XR_0034814...   \n",
       "343524                                              ID=exon-XR_003488114.1-1;Parent=rna-XR_0034881...   \n",
       "344967                                              ID=exon-XR_003480618.1-1;Parent=rna-XR_0034806...   \n",
       "343309                                              ID=exon-XR_003488458.1-2;Parent=rna-XR_0034884...   \n",
       "429753                                              ID=exon-XR_003485612.1-1;Parent=rna-XR_0034856...   \n",
       "\n",
       "                                                    Entrez ID  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...              \n",
       "539323                                                    NaN   \n",
       "353477                                                    NaN   \n",
       "349134                                                    NaN   \n",
       "509442                                                    NaN   \n",
       "418707                                                    NaN   \n",
       "...                                                       ...   \n",
       "430755                                                    NaN   \n",
       "343524                                                    NaN   \n",
       "344967                                                    NaN   \n",
       "343309                                                    NaN   \n",
       "429753                                                    NaN   \n",
       "\n",
       "                                                    Nearest Unigene  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                    \n",
       "539323                                                          NaN   \n",
       "353477                                                          NaN   \n",
       "349134                                                          NaN   \n",
       "509442                                                          NaN   \n",
       "418707                                                          NaN   \n",
       "...                                                             ...   \n",
       "430755                                                          NaN   \n",
       "343524                                                          NaN   \n",
       "344967                                                          NaN   \n",
       "343309                                                          NaN   \n",
       "429753                                                          NaN   \n",
       "\n",
       "                                                    Nearest Refseq  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                   \n",
       "539323                                                         NaN   \n",
       "353477                                                         NaN   \n",
       "349134                                                         NaN   \n",
       "509442                                                         NaN   \n",
       "418707                                                         NaN   \n",
       "...                                                            ...   \n",
       "430755                                                         NaN   \n",
       "343524                                                         NaN   \n",
       "344967                                                         NaN   \n",
       "343309                                                         NaN   \n",
       "429753                                                         NaN   \n",
       "\n",
       "                                                    Nearest Ensembl  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                    \n",
       "539323                                                          NaN   \n",
       "353477                                                          NaN   \n",
       "349134                                                          NaN   \n",
       "509442                                                          NaN   \n",
       "418707                                                          NaN   \n",
       "...                                                             ...   \n",
       "430755                                                          NaN   \n",
       "343524                                                          NaN   \n",
       "344967                                                          NaN   \n",
       "343309                                                          NaN   \n",
       "429753                                                          NaN   \n",
       "\n",
       "                                                    Gene Name  Gene Alias  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                          \n",
       "539323                                                    NaN         NaN   \n",
       "353477                                                    NaN         NaN   \n",
       "349134                                                    NaN         NaN   \n",
       "509442                                                    NaN         NaN   \n",
       "418707                                                    NaN         NaN   \n",
       "...                                                       ...         ...   \n",
       "430755                                                    NaN         NaN   \n",
       "343524                                                    NaN         NaN   \n",
       "344967                                                    NaN         NaN   \n",
       "343309                                                    NaN         NaN   \n",
       "429753                                                    NaN         NaN   \n",
       "\n",
       "                                                    Gene Description  \\\n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...                     \n",
       "539323                                                           NaN   \n",
       "353477                                                           NaN   \n",
       "349134                                                           NaN   \n",
       "509442                                                           NaN   \n",
       "418707                                                           NaN   \n",
       "...                                                              ...   \n",
       "430755                                                           NaN   \n",
       "343524                                                           NaN   \n",
       "344967                                                           NaN   \n",
       "343309                                                           NaN   \n",
       "429753                                                           NaN   \n",
       "\n",
       "                                                    Gene Type  \n",
       "PeakID (cmd=annotatePeaks.pl data/processed/GCF...             \n",
       "539323                                                    NaN  \n",
       "353477                                                    NaN  \n",
       "349134                                                    NaN  \n",
       "509442                                                    NaN  \n",
       "418707                                                    NaN  \n",
       "...                                                       ...  \n",
       "430755                                                    NaN  \n",
       "343524                                                    NaN  \n",
       "344967                                                    NaN  \n",
       "343309                                                    NaN  \n",
       "429753                                                    NaN  \n",
       "\n",
       "[12270 rows x 18 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncrnas = all_merge_anno_df[all_merge_anno_df['Annotation'].fillna('').str.contains(\"ncRNA\")]\n",
    "#ncrnas_prom = ncrnas[(ncrnas[\"Annotation\"].fillna('').str.contains(\"promoter\"))]\n",
    "ncrnas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.402164Z",
     "start_time": "2020-09-01T17:25:22.382952Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(12270, 18)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncrnas.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.614423Z",
     "start_time": "2020-09-01T17:25:22.406812Z"
    },
    "code_folding": [],
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'merged_stable_minus_eTSS' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-27-8301bbe83312>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;31m#merge_df.loc[merged_stable_minus_eTSS.values.astype(int)]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0mstable_minus_eTSS_merged_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mall_merge_anno_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mmerged_stable_minus_eTSS\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\",\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mint\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m \u001b[0mstable_minus_eTSS_merged_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'merged_stable_minus_eTSS' is not defined"
     ]
    }
   ],
   "source": [
    "#merged_stable_minus_eTSS = all_csrna_merged[all_csrna_merged[out_csrna_merged].isin(stable_minus_eTSS[out_csrna_merged].values)][all_merged_f]\n",
    "\n",
    "#### If comma, just take the first peak\n",
    "#merge_df.loc[merged_stable_minus_eTSS.values.astype(int)]\n",
    "\n",
    "stable_minus_eTSS_merged_df = all_merge_anno_df.loc[merged_stable_minus_eTSS.apply(lambda x: x.split(\",\")[0]).values.astype(int)]\n",
    "stable_minus_eTSS_merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.617046Z",
     "start_time": "2020-09-01T17:24:12.853Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stable_minus_eTSS_merged_df.loc[stable_minus_eTSS_merged_df.index.isin(ncrnas.index)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.618305Z",
     "start_time": "2020-09-01T17:24:12.858Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "tmp = ncrnas.loc[ncrnas.index.isin(merged_stable_minus_eTSS.apply(lambda x: x.split(\",\")[0]).values.astype(int))]\n",
    "tmp[~(tmp[\"Annotation\"].isnull())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.619781Z",
     "start_time": "2020-09-01T17:24:12.861Z"
    }
   },
   "outputs": [],
   "source": [
    "tmp[(tmp[\"Annotation\"].isnull())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove the ncRNA peakss"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.621132Z",
     "start_time": "2020-09-01T17:24:12.868Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stable_minus_eTSS_merged_df = stable_minus_eTSS_merged_df.loc[~(stable_minus_eTSS_merged_df.index.isin(ncrnas.index))]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get distance to TSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.622459Z",
     "start_time": "2020-09-01T17:24:12.876Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stable_minus_eTSS_merged_df = pardf(stable_minus_eTSS_merged_df,\n",
    "                         annotation.distance_to_landmarks,\n",
    "                         func_args=(mrna_peak, 'transcript_id', ('gene', ),\n",
    "                                    False, True),\n",
    "                         num_processes=24)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.623800Z",
     "start_time": "2020-09-01T17:24:12.878Z"
    }
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "sns.distplot(stable_minus_eTSS_merged_df[\"Distance to TSS\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.625124Z",
     "start_time": "2020-09-01T17:24:12.880Z"
    }
   },
   "outputs": [],
   "source": [
    "sns.distplot(np.clip(np.abs(stable_minus_eTSS_merged_df[\"Distance to TSS\"]), None, 10000))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.626430Z",
     "start_time": "2020-09-01T17:24:12.882Z"
    }
   },
   "outputs": [],
   "source": [
    "sns.distplot(np.log10(np.abs(stable_minus_eTSS_merged_df[\"Distance to TSS\"])+1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.627756Z",
     "start_time": "2020-09-01T17:24:12.884Z"
    }
   },
   "outputs": [],
   "source": [
    "(np.abs(stable_minus_eTSS_merged_df[\"Distance to TSS\"] )<1000).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.629069Z",
     "start_time": "2020-09-01T17:24:12.887Z"
    }
   },
   "outputs": [],
   "source": [
    "stable_minus_eTSS_merged_df[stable_minus_eTSS_merged_df[\"Nearest gene\"] == \"Mgat3\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.630295Z",
     "start_time": "2020-09-01T17:24:12.888Z"
    }
   },
   "outputs": [],
   "source": [
    "sns.distplot(np.log2(stable_minus_eTSS_merged_df[\"Peak Score\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.631742Z",
     "start_time": "2020-09-01T17:24:12.890Z"
    }
   },
   "outputs": [],
   "source": [
    "stable_minus_eTSS_merged_df[\"Peak Score\"].describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove Nearest TSS null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.633049Z",
     "start_time": "2020-09-01T17:24:12.895Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stable_minus_eTSS_merged_df = stable_minus_eTSS_merged_df.dropna(axis=0,subset=[\"Nearest TSS\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Change the name, and Save stable_minus_eTSS_ncRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.634360Z",
     "start_time": "2020-09-01T17:24:12.903Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stable_minus_eTSS_merged_df['rank'] = stable_minus_eTSS_merged_df.groupby([\"Nearest gene\", \"Nearest TSS\"])[\"Peak Score\"].rank(ascending=False,\n",
    "                                                                                                                             method='first')\n",
    "stable_minus_eTSS_merged_df[\"Name\"] = stable_minus_eTSS_merged_df.apply(lambda x: x[\"Nearest gene\"] + \"_\"\n",
    "                                                                      + x[\"Nearest TSS\"] + \"_\"\n",
    "                                                                      + str(int(x[\"rank\"])),axis=1)\n",
    "stable_minus_eTSS_merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-09-01T17:25:22.635675Z",
     "start_time": "2020-09-01T17:24:12.905Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stable_minus_eTSS_merged_df.set_index(\"Name\",drop=False,verify_integrity=True).to_csv(join(save_dir,\"stableTSS_minus_eTSS.tsv\"),\n",
    "                                                                                    sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Convert to bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-20T05:12:04.057927Z",
     "start_time": "2020-08-20T05:12:03.308125Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\tConverted 66022 peaks total\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!pos2bed.pl {out_csrna_merged} > {out_csrna_merged}.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-20T05:15:23.482586Z",
     "start_time": "2020-08-20T05:15:22.964065Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\tConverted 58808 peaks total\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!pos2bed.pl {out_all_csrna_merged} > {out_all_csrna_merged}.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Take all stable RNA and get distance to nearest gene \n",
    "## See which peaks are left out of the final output (from eTSS or gene_df), and add them. \n",
    "## Check if Mgat3 is nearest TSS. \n",
    "### How many added? Distance? \n",
    "### How many have ambigous - distance to nearby other genes too. If this is the case, can directionally help?\n",
    "### Is direction properly accounted for? "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-31T17:38:22.177969Z",
     "start_time": "2020-08-31T17:38:21.973792Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>GCF/merged/samples.merge</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#name (cmd = mergePeaks -d given -strand GCF/merged/samples.merge stable_rna/GCF/merged_stable_peaks.tsv -venn stable_rna/GCF/merged_stable_peaks.venn)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822429.1-6852555-2</td>\n",
       "      <td>NW_020822429.1</td>\n",
       "      <td>6852464</td>\n",
       "      <td>6852668</td>\n",
       "      <td>-</td>\n",
       "      <td>32.527054</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>2799</td>\n",
       "      <td>Merged-NW_020822429.1-6852544-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822429.1-8601620-2</td>\n",
       "      <td>NW_020822429.1</td>\n",
       "      <td>8601541</td>\n",
       "      <td>8601697</td>\n",
       "      <td>+</td>\n",
       "      <td>27.842501</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>537618</td>\n",
       "      <td>Merged-NW_020822429.1-8601622-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822519.1-2588673-2</td>\n",
       "      <td>NW_020822519.1</td>\n",
       "      <td>2588568</td>\n",
       "      <td>2588831</td>\n",
       "      <td>-</td>\n",
       "      <td>41.549290</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>21856</td>\n",
       "      <td>Merged-NW_020822519.1-2588648-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822407.1-5171143-3</td>\n",
       "      <td>NW_020822407.1</td>\n",
       "      <td>5170887</td>\n",
       "      <td>5171328</td>\n",
       "      <td>+</td>\n",
       "      <td>22.681282</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>3</td>\n",
       "      <td>1190</td>\n",
       "      <td>Merged-NW_020822407.1-5171071-2,Merged-NW_0208...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822615.1-22431564-2</td>\n",
       "      <td>NW_020822615.1</td>\n",
       "      <td>22431441</td>\n",
       "      <td>22431672</td>\n",
       "      <td>-</td>\n",
       "      <td>39.881836</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>1192</td>\n",
       "      <td>Merged-NW_020822615.1-22431582-3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               chr     start  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                             \n",
       "Merged-NW_020822429.1-6852555-2                     NW_020822429.1   6852464   \n",
       "Merged-NW_020822429.1-8601620-2                     NW_020822429.1   8601541   \n",
       "Merged-NW_020822519.1-2588673-2                     NW_020822519.1   2588568   \n",
       "Merged-NW_020822407.1-5171143-3                     NW_020822407.1   5170887   \n",
       "Merged-NW_020822615.1-22431564-2                    NW_020822615.1  22431441   \n",
       "\n",
       "                                                         end strand  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                    \n",
       "Merged-NW_020822429.1-6852555-2                      6852668      -   \n",
       "Merged-NW_020822429.1-8601620-2                      8601697      +   \n",
       "Merged-NW_020822519.1-2588673-2                      2588831      -   \n",
       "Merged-NW_020822407.1-5171143-3                      5171328      +   \n",
       "Merged-NW_020822615.1-22431564-2                    22431672      -   \n",
       "\n",
       "                                                         Stat  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...              \n",
       "Merged-NW_020822429.1-6852555-2                     32.527054   \n",
       "Merged-NW_020822429.1-8601620-2                     27.842501   \n",
       "Merged-NW_020822519.1-2588673-2                     41.549290   \n",
       "Merged-NW_020822407.1-5171143-3                     22.681282   \n",
       "Merged-NW_020822615.1-22431564-2                    39.881836   \n",
       "\n",
       "                                                                                         Parent files  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                                                      \n",
       "Merged-NW_020822429.1-6852555-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822429.1-8601620-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822519.1-2588673-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822407.1-5171143-3                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822615.1-22431564-2                    GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "\n",
       "                                                    Total subpeaks  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                   \n",
       "Merged-NW_020822429.1-6852555-2                                  2   \n",
       "Merged-NW_020822429.1-8601620-2                                  2   \n",
       "Merged-NW_020822519.1-2588673-2                                  2   \n",
       "Merged-NW_020822407.1-5171143-3                                  3   \n",
       "Merged-NW_020822615.1-22431564-2                                 2   \n",
       "\n",
       "                                                   GCF/merged/samples.merge  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                            \n",
       "Merged-NW_020822429.1-6852555-2                                        2799   \n",
       "Merged-NW_020822429.1-8601620-2                                      537618   \n",
       "Merged-NW_020822519.1-2588673-2                                       21856   \n",
       "Merged-NW_020822407.1-5171143-3                                        1190   \n",
       "Merged-NW_020822615.1-22431564-2                                       1192   \n",
       "\n",
       "                                                               stable_rna/GCF/merged_stable_peaks.tsv  \n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                                                     \n",
       "Merged-NW_020822429.1-6852555-2                                       Merged-NW_020822429.1-6852544-3  \n",
       "Merged-NW_020822429.1-8601620-2                                       Merged-NW_020822429.1-8601622-1  \n",
       "Merged-NW_020822519.1-2588673-2                                       Merged-NW_020822519.1-2588648-3  \n",
       "Merged-NW_020822407.1-5171143-3                     Merged-NW_020822407.1-5171071-2,Merged-NW_0208...  \n",
       "Merged-NW_020822615.1-22431564-2                                     Merged-NW_020822615.1-22431582-3  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_stable_merge = pd.read_csv(out_all_csrna_merged, sep=\"\\t\", index_col=0)\n",
    "all_stable_merge.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-28T15:38:28.588771Z",
     "start_time": "2020-08-28T15:38:28.039712Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "stable_merge = pd.read_csv(join(save_dir, 'merged_stable_peaks.tsv'), sep='\\t', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-19T22:45:37.698088Z",
     "start_time": "2020-08-19T22:45:37.225136Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New pandarallel memory created - Size: 2000 MB\n",
      "Pandarallel will run on 48 workers\n"
     ]
    }
   ],
   "source": [
    "# from pandarallel import pandarallel\n",
    "# pandarallel.initialize()\n",
    "\n",
    "# def get_nearest_gene(peak_sample_indices, samples_stable):\n",
    "#     curr_p = peak_sample_indices.iloc[7:]\n",
    "#     curr_p_samp = curr_p[~(curr_p.isnull())]\n",
    "#     if curr_p_samp[0].count('TSS') > 1:\n",
    "#         ind = ','.join(curr_p_samp[0].split(',')[:2])\n",
    "#         if ind not in samples_stable[curr_p_samp.index[0]].index:\n",
    "#             print(\"Not here\", curr_p_samp)\n",
    "#             return np.nan\n",
    "#         else:\n",
    "#             return samples_stable[curr_p_samp.index[0]].loc[ind][\"Closest TSS gene\"]\n",
    "#     return samples_stable[curr_p_samp.index[0]].loc[curr_p_samp[0]][\"Closest TSS gene\"]\n",
    "    \n",
    "\n",
    "# stable_merge[\"Nearest txn\"] = stable_merge.parallel_apply(get_nearest_gene,args=(samples_stable,),axis=1)\n",
    "\n",
    "# stable_merge[\"Nearest Gene\"] = stable_merge[\"Nearest txn\"].map(pd.read_csv(\"GCF/genome/start_site_mRNA.tsv\",sep=\"\\t\",index_col=0)[\"gene\"])\n",
    "\n",
    "# stable_merge['Nearest Gene'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get the nearest TSS and Gene for each stable RNA peak. \n",
    "-Use distance_to_landmarks, but add optionn of nearest_same_direction\n",
    "-Need to check if the stable can be bidirectional"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-31T17:41:43.853811Z",
     "start_time": "2020-08-31T17:41:43.831650Z"
    }
   },
   "outputs": [],
   "source": [
    "def capitalize_label(df, labels=(\"chr\", \"start\", \"end\", \"strand\")):\n",
    "    cols = df.columns\n",
    "    new_cols = {}\n",
    "    for i in cols:\n",
    "        if i in labels:\n",
    "            new_cols[i] = i[0].upper() + i[1:]\n",
    "    df = df.rename(new_cols, axis=1)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-31T17:41:41.253235Z",
     "start_time": "2020-08-31T17:41:41.083354Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'capitalize_label' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-14-2baa9163511b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mstable_merge\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcapitalize_label\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstable_merge\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'capitalize_label' is not defined"
     ]
    }
   ],
   "source": [
    "#stable_merge = capitalize_label(stable_merge)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge stable RNA and the samples.merge, and force_same_strand to allow the nearest gene to be the one in the right direction."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-28T15:57:25.574586Z",
     "start_time": "2020-08-28T15:57:05.152395Z"
    }
   },
   "outputs": [],
   "source": [
    "# stable_merge_anno = pardf(stable_merge,\n",
    "#                    annotation.distance_to_landmarks,\n",
    "#                    func_args=(mrna_peak, 'transcript_id', ('gene', ), False,\n",
    "#                               True),\n",
    "#                    num_processes=24)\n",
    "\n",
    "# stable_merge_anno[stable_merge_anno[\"Nearest gene\"] == \"Mgat3\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-31T17:42:10.542833Z",
     "start_time": "2020-08-31T17:41:53.177403Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>GCF/merged/samples.merge</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "      <th>Nearest gene</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#name (cmd = mergePeaks -d given -strand GCF/merged/samples.merge stable_rna/GCF/merged_stable_peaks.tsv -venn stable_rna/GCF/merged_stable_peaks.venn)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822470.1-7464789-2</td>\n",
       "      <td>NW_020822470.1</td>\n",
       "      <td>7464714</td>\n",
       "      <td>7464864</td>\n",
       "      <td>+</td>\n",
       "      <td>16.190001</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>322872</td>\n",
       "      <td>Merged-NW_020822470.1-7464789-1</td>\n",
       "      <td>Mgat3</td>\n",
       "      <td>NM_001244074.1</td>\n",
       "      <td>-36645.0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822470.1-7464518-2</td>\n",
       "      <td>NW_020822470.1</td>\n",
       "      <td>7464424</td>\n",
       "      <td>7464613</td>\n",
       "      <td>+</td>\n",
       "      <td>7.117500</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>296878</td>\n",
       "      <td>Merged-NW_020822470.1-7464525-6</td>\n",
       "      <td>Mgat3</td>\n",
       "      <td>NM_001244074.1</td>\n",
       "      <td>-36916.0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822470.1-7510807-2</td>\n",
       "      <td>NW_020822470.1</td>\n",
       "      <td>7510707</td>\n",
       "      <td>7510959</td>\n",
       "      <td>+</td>\n",
       "      <td>19.467571</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>307626</td>\n",
       "      <td>Merged-NW_020822470.1-7510782-2</td>\n",
       "      <td>Mgat3</td>\n",
       "      <td>NM_001244074.1</td>\n",
       "      <td>9399.0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>Merged-NW_020822470.1-7476017-2</td>\n",
       "      <td>NW_020822470.1</td>\n",
       "      <td>7475931</td>\n",
       "      <td>7476104</td>\n",
       "      <td>+</td>\n",
       "      <td>109.631248</td>\n",
       "      <td>GCF/merged/samples.merge|stable_rna/GCF/merged...</td>\n",
       "      <td>2</td>\n",
       "      <td>114687</td>\n",
       "      <td>Merged-NW_020822470.1-7476017-2</td>\n",
       "      <td>Mgat3</td>\n",
       "      <td>NM_001244074.1</td>\n",
       "      <td>-25417.0</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               Chr    Start  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                            \n",
       "Merged-NW_020822470.1-7464789-2                     NW_020822470.1  7464714   \n",
       "Merged-NW_020822470.1-7464518-2                     NW_020822470.1  7464424   \n",
       "Merged-NW_020822470.1-7510807-2                     NW_020822470.1  7510707   \n",
       "Merged-NW_020822470.1-7476017-2                     NW_020822470.1  7475931   \n",
       "\n",
       "                                                        End Strand  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                   \n",
       "Merged-NW_020822470.1-7464789-2                     7464864      +   \n",
       "Merged-NW_020822470.1-7464518-2                     7464613      +   \n",
       "Merged-NW_020822470.1-7510807-2                     7510959      +   \n",
       "Merged-NW_020822470.1-7476017-2                     7476104      +   \n",
       "\n",
       "                                                          Stat  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...               \n",
       "Merged-NW_020822470.1-7464789-2                      16.190001   \n",
       "Merged-NW_020822470.1-7464518-2                       7.117500   \n",
       "Merged-NW_020822470.1-7510807-2                      19.467571   \n",
       "Merged-NW_020822470.1-7476017-2                     109.631248   \n",
       "\n",
       "                                                                                         Parent files  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                                                      \n",
       "Merged-NW_020822470.1-7464789-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822470.1-7464518-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822470.1-7510807-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "Merged-NW_020822470.1-7476017-2                     GCF/merged/samples.merge|stable_rna/GCF/merged...   \n",
       "\n",
       "                                                    Total subpeaks  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                   \n",
       "Merged-NW_020822470.1-7464789-2                                  2   \n",
       "Merged-NW_020822470.1-7464518-2                                  2   \n",
       "Merged-NW_020822470.1-7510807-2                                  2   \n",
       "Merged-NW_020822470.1-7476017-2                                  2   \n",
       "\n",
       "                                                   GCF/merged/samples.merge  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                            \n",
       "Merged-NW_020822470.1-7464789-2                                      322872   \n",
       "Merged-NW_020822470.1-7464518-2                                      296878   \n",
       "Merged-NW_020822470.1-7510807-2                                      307626   \n",
       "Merged-NW_020822470.1-7476017-2                                      114687   \n",
       "\n",
       "                                                   stable_rna/GCF/merged_stable_peaks.tsv  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                                          \n",
       "Merged-NW_020822470.1-7464789-2                           Merged-NW_020822470.1-7464789-1   \n",
       "Merged-NW_020822470.1-7464518-2                           Merged-NW_020822470.1-7464525-6   \n",
       "Merged-NW_020822470.1-7510807-2                           Merged-NW_020822470.1-7510782-2   \n",
       "Merged-NW_020822470.1-7476017-2                           Merged-NW_020822470.1-7476017-2   \n",
       "\n",
       "                                                   Nearest gene  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                \n",
       "Merged-NW_020822470.1-7464789-2                           Mgat3   \n",
       "Merged-NW_020822470.1-7464518-2                           Mgat3   \n",
       "Merged-NW_020822470.1-7510807-2                           Mgat3   \n",
       "Merged-NW_020822470.1-7476017-2                           Mgat3   \n",
       "\n",
       "                                                       Nearest TSS  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                   \n",
       "Merged-NW_020822470.1-7464789-2                     NM_001244074.1   \n",
       "Merged-NW_020822470.1-7464518-2                     NM_001244074.1   \n",
       "Merged-NW_020822470.1-7510807-2                     NM_001244074.1   \n",
       "Merged-NW_020822470.1-7476017-2                     NM_001244074.1   \n",
       "\n",
       "                                                    Distance to TSS  \\\n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...                    \n",
       "Merged-NW_020822470.1-7464789-2                            -36645.0   \n",
       "Merged-NW_020822470.1-7464518-2                            -36916.0   \n",
       "Merged-NW_020822470.1-7510807-2                              9399.0   \n",
       "Merged-NW_020822470.1-7476017-2                            -25417.0   \n",
       "\n",
       "                                                   isSameStrand  \n",
       "#name (cmd = mergePeaks -d given -strand GCF/me...               \n",
       "Merged-NW_020822470.1-7464789-2                            True  \n",
       "Merged-NW_020822470.1-7464518-2                            True  \n",
       "Merged-NW_020822470.1-7510807-2                            True  \n",
       "Merged-NW_020822470.1-7476017-2                            True  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_stable_merge_anno = pardf(capitalize_label(all_stable_merge),\n",
    "                         annotation.distance_to_landmarks,\n",
    "                         func_args=(mrna_peak, 'transcript_id', ('gene', ),\n",
    "                                    False, True),\n",
    "                         num_processes=24)\n",
    "\n",
    "all_stable_merge_anno[all_stable_merge_anno[\"Nearest gene\"] == \"Mgat3\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-31T17:43:36.959767Z",
     "start_time": "2020-08-31T17:43:36.897539Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>GCF/merged/samples.merge</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "      <th>Nearest gene</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#name (cmd = mergePeaks -d given -strand GCF/merged/samples.merge stable_rna/GCF/merged_stable_peaks.tsv -venn stable_rna/GCF/merged_stable_peaks.venn)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Chr, Start, End, Strand, Stat, Parent files, Total subpeaks, GCF/merged/samples.merge, stable_rna/GCF/merged_stable_peaks.tsv, Nearest gene, Nearest TSS, Distance to TSS, isSameStrand]\n",
       "Index: []"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_stable_merge_anno[~all_stable_merge_anno[\"Parent files\"].str.contains(\"|\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-31T17:44:32.110931Z",
     "start_time": "2020-08-31T17:44:32.068420Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>GCF/merged/samples.merge</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "      <th>Nearest gene</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#name (cmd = mergePeaks -d given -strand GCF/merged/samples.merge stable_rna/GCF/merged_stable_peaks.tsv -venn stable_rna/GCF/merged_stable_peaks.venn)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Chr, Start, End, Strand, Stat, Parent files, Total subpeaks, GCF/merged/samples.merge, stable_rna/GCF/merged_stable_peaks.tsv, Nearest gene, Nearest TSS, Distance to TSS, isSameStrand]\n",
       "Index: []"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_stable_merge_anno[all_stable_merge_anno[\"GCF/merged/samples.merge\"].isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-31T17:43:02.032056Z",
     "start_time": "2020-08-31T17:43:01.991206Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>GCF/merged/samples.merge</th>\n",
       "      <th>stable_rna/GCF/merged_stable_peaks.tsv</th>\n",
       "      <th>Nearest gene</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#name (cmd = mergePeaks -d given -strand GCF/merged/samples.merge stable_rna/GCF/merged_stable_peaks.tsv -venn stable_rna/GCF/merged_stable_peaks.venn)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Chr, Start, End, Strand, Stat, Parent files, Total subpeaks, GCF/merged/samples.merge, stable_rna/GCF/merged_stable_peaks.tsv, Nearest gene, Nearest TSS, Distance to TSS, isSameStrand]\n",
       "Index: []"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_stable_merge_anno[all_stable_merge_anno[\"Parent files\"] == \"stable_rna/GCF/merged_stable_peaks.tsv\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-28T16:13:27.901128Z",
     "start_time": "2020-08-28T16:13:27.226923Z"
    }
   },
   "outputs": [],
   "source": [
    "capitalize_label(all_stable_merge_anno).to_csv(out_all_csrna_merged.replace(\".tsv\",\".anno.tsv\"),sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-28T16:13:28.341194Z",
     "start_time": "2020-08-28T16:13:28.314828Z"
    }
   },
   "outputs": [],
   "source": [
    "anno_out_all_csrna_merged = out_all_csrna_merged.replace(\".tsv\",\".anno.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-08-28T16:13:29.535226Z",
     "start_time": "2020-08-28T16:13:28.993168Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\tConverted 58812 peaks total\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!pos2bed.pl {anno_out_all_csrna_merged} > {anno_out_all_csrna_merged}.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
