{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T17:25:07.549842Z",
     "start_time": "2021-04-21T17:25:07.495146Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/ipykernel_launcher.py:17: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'OUTDIR': '/data/isshamie/TSS/Analysis/TSS_code/data/processed/manuscript',\n",
       " 'genome_fasta': '/data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna',\n",
       " 'annotation': {'synapse': '/data/isshamie/genome/hamster/syn20999279_picr.gff',\n",
       "  'GCF': '/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gff'},\n",
       " 'ncrna_keys': {'synapse': ['lncRNA'],\n",
       "  'GCF': ['snoRNA', 'snRNA', 'miRNA', 'lnc_RNA']},\n",
       " 'MERGE_TISSUES': ['all', 'bmdm'],\n",
       " 'atac_f': '/data/isshamie/TSS/Analysis/Analysis_NCBI_PICR/Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort',\n",
       " 'all_atac': '/data/isshamie/TSS/Analysis/Analysis_NCBI_PICR/Results/ATAC_results/ATAC_merge.bed',\n",
       " 'META_SAMPLES': '/data/isshamie/TSS/Analysis/Analysis_03302020/parameters/samples.tsv',\n",
       " 'peaks_dir': '/data/isshamie/TSS/Analysis/Analysis_NCBI_PICR/f04_peaks/',\n",
       " 'cpm_thresh': [1, 0],\n",
       " 'numsamples_thresh': [2, 1]}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "import yaml\n",
    "import pandas as pd\n",
    "from pandarallel import pandarallel\n",
    "import click\n",
    "import numpy as np\n",
    "from tss.manuscript.utils import expand_anno_id\n",
    "from tss.manuscript.peak_distance import create_anno_tss\n",
    "\n",
    "genome=\"GCF\"\n",
    "thresh=0\n",
    "numSamp=1\n",
    "with open(\"manuscript_params.yaml\") as file:\n",
    "    # The FullLoader parameter handles the conversion from YAML\n",
    "    # scalar values to Python the dictionary format\n",
    "    params=yaml.load(file)\n",
    "params"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### init anno files and parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T17:25:20.880086Z",
     "start_time": "2021-04-21T17:25:20.845051Z"
    }
   },
   "outputs": [],
   "source": [
    "cpu=32\n",
    "gff_f= \"/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gff\"\n",
    "reftss_f=f\"data/{genome}/genome/refTSS.tsv\"\n",
    "cds_f = f\"data/{genome}/genome/CDS.bed\"\n",
    "intron_f = f\"data/{genome}/genome/introns.bed\"\n",
    "pc= \"mRNA\"#.split(',')\n",
    "ncrna= params['ncrna_keys'][genome] #','.join(params['ncrna_keys'][genome])\n",
    "ncrna\n",
    "\n",
    "### Peak file names\n",
    "\n",
    "merge = \"data/peaksInfo/samples.merge\"\n",
    "raw_peaks=\"data/peaksInfo/peaks.merge.tss\"\n",
    "minimal=\"data/peaksInfo/peaks.merge.tss.minimal\"\n",
    "\n",
    "\n",
    "### Peaks info on samples and the average values. Also uses the thresholds for confident peaks here\n",
    "\n",
    "peaks_sourceMeta = \"data/peaksInfo/peaks.sourceMeta.tsv\"\n",
    "peaks_sourceConfident = \"data/peaksInfo/peaks.sourceMeta.confident.index\"\n",
    "expr_thresh = 1 #cpm threshold\n",
    "num_samples_thresh = 1\n",
    "\n",
    "### Using the annotation\n",
    "\n",
    "homeranno= f\"data/{genome}/anno/peaks.merge.tss.anno\"\n",
    "peaks_minimal = f\"data/{genome}/anno/peaks.merge.tss.minimal.anno\"\n",
    "peaks_homeranno = f\"data/{genome}/anno/peaks.merge.tss.minimal.anno\"\n",
    "peaks_distance = f\"data/{genome}/anno/peaks.distance.tsv\"\n",
    "\n",
    "### Filter introns and cds names\n",
    "\n",
    "noncds_peaks = f\"data/{genome}/genebody_filters/peaks.nocds.index\"\n",
    "nonintron_peaks = f\"data/{genome}/genebody_filters/peaks.nointron.index\"\n",
    "non_cds_intron_peaks = f\"data/{genome}/genebody_filters/peaks.nocds.nointron.index\"\n",
    "\n",
    "### revised TSS\n",
    "\n",
    "pc_tss = f\"data/{genome}/refType/pc.index\"\n",
    "ncrna_tss = f\"data/{genome}/refType/ncrna.index\"\n",
    "intergenic_tss = f\"data/{genome}/refType/ncrna.index\"\n",
    "\n",
    "\n",
    "# filtered tss\n",
    "allTSS_f = f\"data/{genome}/allTSS/allTSS_thresh{thresh}_numSamp{numSamp}.tsv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T17:28:40.863554Z",
     "start_time": "2021-04-21T17:28:40.634870Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Distance to Nearest TSS</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to Nearest TSS_Rev</th>\n",
       "      <th>Nearest TSS_Rev</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>homeranno</th>\n",
       "      <th>refType</th>\n",
       "      <th>isCDS</th>\n",
       "      <th>isIntron</th>\n",
       "      <th>isAbove_numsamples</th>\n",
       "      <th>isCHO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>60370.0</td>\n",
       "      <td>XM_027432388.1</td>\n",
       "      <td>-746.0</td>\n",
       "      <td>XM_027432478.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027391402.1-1;Parent=...</td>\n",
       "      <td>divergent</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>-11.0</td>\n",
       "      <td>XM_027421990.1</td>\n",
       "      <td>-53290.0</td>\n",
       "      <td>XM_027421992.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027422995.1-6;Parent=...</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>296.0</td>\n",
       "      <td>XM_027428527.1</td>\n",
       "      <td>24329.0</td>\n",
       "      <td>XM_027428469.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>47.0</td>\n",
       "      <td>XM_027418757.1</td>\n",
       "      <td>377336.0</td>\n",
       "      <td>XM_027418744.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003480724.1-1;Parent=...</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>118811.0</td>\n",
       "      <td>XM_027424703.1</td>\n",
       "      <td>191.0</td>\n",
       "      <td>XM_027424708.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>divergent</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>532240</td>\n",
       "      <td>536617</td>\n",
       "      <td>-406001.0</td>\n",
       "      <td>XM_027407647.1</td>\n",
       "      <td>790639.0</td>\n",
       "      <td>XM_027407642.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>532249</td>\n",
       "      <td>536626</td>\n",
       "      <td>-17151.0</td>\n",
       "      <td>XM_027407847.1</td>\n",
       "      <td>1956.0</td>\n",
       "      <td>XM_027407848.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>532252</td>\n",
       "      <td>536629</td>\n",
       "      <td>-113749.0</td>\n",
       "      <td>XM_027407818.1</td>\n",
       "      <td>65496.0</td>\n",
       "      <td>XR_003483883.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>532253</td>\n",
       "      <td>536630</td>\n",
       "      <td>-80520.0</td>\n",
       "      <td>XM_027409383.1</td>\n",
       "      <td>16925.0</td>\n",
       "      <td>XM_027409587.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>532254</td>\n",
       "      <td>536631</td>\n",
       "      <td>-20882.0</td>\n",
       "      <td>XM_027390077.1</td>\n",
       "      <td>93536.0</td>\n",
       "      <td>XM_027390075.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74022 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            ID  Distance to Nearest TSS     Nearest TSS  \\\n",
       "2            2                  60370.0  XM_027432388.1   \n",
       "4            4                    -11.0  XM_027421990.1   \n",
       "5            5                    296.0  XM_027428527.1   \n",
       "6            6                     47.0  XM_027418757.1   \n",
       "7            7                 118811.0  XM_027424703.1   \n",
       "...        ...                      ...             ...   \n",
       "532240  536617                -406001.0  XM_027407647.1   \n",
       "532249  536626                 -17151.0  XM_027407847.1   \n",
       "532252  536629                -113749.0  XM_027407818.1   \n",
       "532253  536630                 -80520.0  XM_027409383.1   \n",
       "532254  536631                 -20882.0  XM_027390077.1   \n",
       "\n",
       "        Distance to Nearest TSS_Rev Nearest TSS_Rev Annotation  \\\n",
       "2                            -746.0  XM_027432478.1       mRNA   \n",
       "4                          -53290.0  XM_027421992.1       mRNA   \n",
       "5                           24329.0  XM_027428469.1       mRNA   \n",
       "6                          377336.0  XM_027418744.1       mRNA   \n",
       "7                             191.0  XM_027424708.1       mRNA   \n",
       "...                             ...             ...        ...   \n",
       "532240                     790639.0  XM_027407642.1       mRNA   \n",
       "532249                       1956.0  XM_027407848.1       mRNA   \n",
       "532252                      65496.0  XR_003483883.1       mRNA   \n",
       "532253                      16925.0  XM_027409587.1       mRNA   \n",
       "532254                      93536.0  XM_027390075.1       mRNA   \n",
       "\n",
       "                                                homeranno     refType  isCDS  \\\n",
       "2       promoter-TSS (ID=exon-XM_027391402.1-1;Parent=...   divergent  False   \n",
       "4       promoter-TSS (ID=exon-XM_027422995.1-6;Parent=...        mRNA  False   \n",
       "5                                              Intergenic        mRNA  False   \n",
       "6       promoter-TSS (ID=exon-XR_003480724.1-1;Parent=...        mRNA  False   \n",
       "7                                              Intergenic   divergent  False   \n",
       "...                                                   ...         ...    ...   \n",
       "532240                                         Intergenic  Intergenic  False   \n",
       "532249                                         Intergenic  Intergenic  False   \n",
       "532252                                         Intergenic  Intergenic  False   \n",
       "532253                                         Intergenic  Intergenic  False   \n",
       "532254                                         Intergenic  Intergenic  False   \n",
       "\n",
       "        isIntron  isAbove_numsamples    isCHO  \n",
       "2          False                True  Overlap  \n",
       "4          False                True  Overlap  \n",
       "5          False                True  Tissues  \n",
       "6          False                True  Overlap  \n",
       "7          False                True  Overlap  \n",
       "...          ...                 ...      ...  \n",
       "532240     False                True  Tissues  \n",
       "532249     False                True  Tissues  \n",
       "532252     False                True  Tissues  \n",
       "532253     False                True  Overlap  \n",
       "532254     False                True  Overlap  \n",
       "\n",
       "[74022 rows x 12 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(allTSS_f, index_col=0)\n",
    "df\n",
    "\n",
    "pc = df[df['refType'] == 'mRNA']\n",
    "pc\n",
    "\n",
    "refTSS = pd.read_csv(reftss_f, index_col=0, sep='\\t')\n",
    "refTSS\n",
    "\n",
    "## Get number of genes\n",
    "\n",
    "pc['Gene'] = pc['Nearest TSS'].map(refTSS['gene'].to_dict())\n",
    "pc\n",
    "\n",
    "pc.groupby('Gene').plot.bar()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. process genome "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T18:07:09.937437Z",
     "start_time": "2021-04-20T18:06:49.194148Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "col_names [0, 1, 2, 3, 4, 5, 6, 7, 8]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "54410it [00:14, 3832.00it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>How</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>.1</th>\n",
       "      <th>Strand</th>\n",
       "      <th>.2</th>\n",
       "      <th>ID</th>\n",
       "      <th>actual_start</th>\n",
       "      <th>...</th>\n",
       "      <th>gene</th>\n",
       "      <th>model_evidence</th>\n",
       "      <th>product</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>Note</th>\n",
       "      <th>exception</th>\n",
       "      <th>inference</th>\n",
       "      <th>end_range</th>\n",
       "      <th>partial</th>\n",
       "      <th>start_range</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>XR_003479383.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>269294</td>\n",
       "      <td>269294</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XR_003479383.1</td>\n",
       "      <td>269294</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC113832197</td>\n",
       "      <td>Supporting evidence includes similarity to: 10...</td>\n",
       "      <td>uncharacterized LOC113832197%2C transcript var...</td>\n",
       "      <td>XR_003479383.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XR_003479384.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>551299</td>\n",
       "      <td>551299</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XR_003479384.1</td>\n",
       "      <td>551299</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC113832198</td>\n",
       "      <td>Supporting evidence includes similarity to: 10...</td>\n",
       "      <td>uncharacterized LOC113832198</td>\n",
       "      <td>XR_003479384.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XR_003479386.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>696459</td>\n",
       "      <td>696459</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XR_003479386.1</td>\n",
       "      <td>696459</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC103159926</td>\n",
       "      <td>Supporting evidence includes similarity to: 10...</td>\n",
       "      <td>uncharacterized LOC103159926%2C transcript var...</td>\n",
       "      <td>XR_003479386.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XR_003479388.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>696459</td>\n",
       "      <td>696459</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XR_003479388.1</td>\n",
       "      <td>696459</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC103159926</td>\n",
       "      <td>Supporting evidence includes similarity to: 10...</td>\n",
       "      <td>uncharacterized LOC103159926%2C transcript var...</td>\n",
       "      <td>XR_003479388.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XR_003479385.1</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>710127</td>\n",
       "      <td>710127</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XR_003479385.1</td>\n",
       "      <td>710127</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC103159926</td>\n",
       "      <td>Supporting evidence includes similarity to: 10...</td>\n",
       "      <td>uncharacterized LOC103159926%2C transcript var...</td>\n",
       "      <td>XR_003479385.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434593.1</td>\n",
       "      <td>NW_020824186.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>29432</td>\n",
       "      <td>29432</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XM_027434593.1</td>\n",
       "      <td>29432</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC103158952</td>\n",
       "      <td>Supporting evidence includes similarity to: 2 ...</td>\n",
       "      <td>PRAME family member 12-like%2C transcript vari...</td>\n",
       "      <td>XM_027434593.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434595.1</td>\n",
       "      <td>NW_020824188.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>2147</td>\n",
       "      <td>2147</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XM_027434595.1</td>\n",
       "      <td>2147</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC113839149</td>\n",
       "      <td>Supporting evidence includes similarity to: 25...</td>\n",
       "      <td>zinc finger protein 431-like%2C transcript var...</td>\n",
       "      <td>XM_027434595.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434594.1</td>\n",
       "      <td>NW_020824188.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>2147</td>\n",
       "      <td>2147</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XM_027434594.1</td>\n",
       "      <td>2147</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC113839149</td>\n",
       "      <td>Supporting evidence includes similarity to: 30...</td>\n",
       "      <td>zinc finger protein 431-like%2C transcript var...</td>\n",
       "      <td>XM_027434594.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434596.1</td>\n",
       "      <td>NW_020824188.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>2147</td>\n",
       "      <td>2147</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XM_027434596.1</td>\n",
       "      <td>2147</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC113839149</td>\n",
       "      <td>Supporting evidence includes similarity to: 27...</td>\n",
       "      <td>zinc finger protein 431-like%2C transcript var...</td>\n",
       "      <td>XM_027434596.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>XM_027434597.1</td>\n",
       "      <td>NW_020824192.1</td>\n",
       "      <td>Gnomon</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>6650</td>\n",
       "      <td>6650</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>rna-XM_027434597.1</td>\n",
       "      <td>6650</td>\n",
       "      <td>...</td>\n",
       "      <td>LOC113839150</td>\n",
       "      <td>Supporting evidence includes similarity to: 8 ...</td>\n",
       "      <td>mothers against decapentaplegic homolog 1-like</td>\n",
       "      <td>XM_027434597.1</td>\n",
       "      <td>The sequence of the model RefSeq transcript wa...</td>\n",
       "      <td>unclassified transcription discrepancy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54410 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Chr     How Annotation   Start     End .1 Strand  \\\n",
       "XR_003479383.1  NW_020822366.1  Gnomon    lnc_RNA  269294  269294  .      +   \n",
       "XR_003479384.1  NW_020822366.1  Gnomon    lnc_RNA  551299  551299  .      +   \n",
       "XR_003479386.1  NW_020822366.1  Gnomon    lnc_RNA  696459  696459  .      +   \n",
       "XR_003479388.1  NW_020822366.1  Gnomon    lnc_RNA  696459  696459  .      +   \n",
       "XR_003479385.1  NW_020822366.1  Gnomon    lnc_RNA  710127  710127  .      +   \n",
       "...                        ...     ...        ...     ...     ... ..    ...   \n",
       "XM_027434593.1  NW_020824186.1  Gnomon       mRNA   29432   29432  .      +   \n",
       "XM_027434595.1  NW_020824188.1  Gnomon       mRNA    2147    2147  .      +   \n",
       "XM_027434594.1  NW_020824188.1  Gnomon       mRNA    2147    2147  .      +   \n",
       "XM_027434596.1  NW_020824188.1  Gnomon       mRNA    2147    2147  .      +   \n",
       "XM_027434597.1  NW_020824192.1  Gnomon       mRNA    6650    6650  .      -   \n",
       "\n",
       "               .2                  ID  actual_start  ...          gene  \\\n",
       "XR_003479383.1  .  rna-XR_003479383.1        269294  ...  LOC113832197   \n",
       "XR_003479384.1  .  rna-XR_003479384.1        551299  ...  LOC113832198   \n",
       "XR_003479386.1  .  rna-XR_003479386.1        696459  ...  LOC103159926   \n",
       "XR_003479388.1  .  rna-XR_003479388.1        696459  ...  LOC103159926   \n",
       "XR_003479385.1  .  rna-XR_003479385.1        710127  ...  LOC103159926   \n",
       "...            ..                 ...           ...  ...           ...   \n",
       "XM_027434593.1  .  rna-XM_027434593.1         29432  ...  LOC103158952   \n",
       "XM_027434595.1  .  rna-XM_027434595.1          2147  ...  LOC113839149   \n",
       "XM_027434594.1  .  rna-XM_027434594.1          2147  ...  LOC113839149   \n",
       "XM_027434596.1  .  rna-XM_027434596.1          2147  ...  LOC113839149   \n",
       "XM_027434597.1  .  rna-XM_027434597.1          6650  ...  LOC113839150   \n",
       "\n",
       "                                                   model_evidence  \\\n",
       "XR_003479383.1  Supporting evidence includes similarity to: 10...   \n",
       "XR_003479384.1  Supporting evidence includes similarity to: 10...   \n",
       "XR_003479386.1  Supporting evidence includes similarity to: 10...   \n",
       "XR_003479388.1  Supporting evidence includes similarity to: 10...   \n",
       "XR_003479385.1  Supporting evidence includes similarity to: 10...   \n",
       "...                                                           ...   \n",
       "XM_027434593.1  Supporting evidence includes similarity to: 2 ...   \n",
       "XM_027434595.1  Supporting evidence includes similarity to: 25...   \n",
       "XM_027434594.1  Supporting evidence includes similarity to: 30...   \n",
       "XM_027434596.1  Supporting evidence includes similarity to: 27...   \n",
       "XM_027434597.1  Supporting evidence includes similarity to: 8 ...   \n",
       "\n",
       "                                                          product  \\\n",
       "XR_003479383.1  uncharacterized LOC113832197%2C transcript var...   \n",
       "XR_003479384.1                       uncharacterized LOC113832198   \n",
       "XR_003479386.1  uncharacterized LOC103159926%2C transcript var...   \n",
       "XR_003479388.1  uncharacterized LOC103159926%2C transcript var...   \n",
       "XR_003479385.1  uncharacterized LOC103159926%2C transcript var...   \n",
       "...                                                           ...   \n",
       "XM_027434593.1  PRAME family member 12-like%2C transcript vari...   \n",
       "XM_027434595.1  zinc finger protein 431-like%2C transcript var...   \n",
       "XM_027434594.1  zinc finger protein 431-like%2C transcript var...   \n",
       "XM_027434596.1  zinc finger protein 431-like%2C transcript var...   \n",
       "XM_027434597.1     mothers against decapentaplegic homolog 1-like   \n",
       "\n",
       "                 transcript_id  \\\n",
       "XR_003479383.1  XR_003479383.1   \n",
       "XR_003479384.1  XR_003479384.1   \n",
       "XR_003479386.1  XR_003479386.1   \n",
       "XR_003479388.1  XR_003479388.1   \n",
       "XR_003479385.1  XR_003479385.1   \n",
       "...                        ...   \n",
       "XM_027434593.1  XM_027434593.1   \n",
       "XM_027434595.1  XM_027434595.1   \n",
       "XM_027434594.1  XM_027434594.1   \n",
       "XM_027434596.1  XM_027434596.1   \n",
       "XM_027434597.1  XM_027434597.1   \n",
       "\n",
       "                                                             Note  \\\n",
       "XR_003479383.1                                                NaN   \n",
       "XR_003479384.1                                                NaN   \n",
       "XR_003479386.1                                                NaN   \n",
       "XR_003479388.1                                                NaN   \n",
       "XR_003479385.1                                                NaN   \n",
       "...                                                           ...   \n",
       "XM_027434593.1                                                NaN   \n",
       "XM_027434595.1                                                NaN   \n",
       "XM_027434594.1                                                NaN   \n",
       "XM_027434596.1                                                NaN   \n",
       "XM_027434597.1  The sequence of the model RefSeq transcript wa...   \n",
       "\n",
       "                                             exception inference end_range  \\\n",
       "XR_003479383.1                                     NaN       NaN       NaN   \n",
       "XR_003479384.1                                     NaN       NaN       NaN   \n",
       "XR_003479386.1                                     NaN       NaN       NaN   \n",
       "XR_003479388.1                                     NaN       NaN       NaN   \n",
       "XR_003479385.1                                     NaN       NaN       NaN   \n",
       "...                                                ...       ...       ...   \n",
       "XM_027434593.1                                     NaN       NaN       NaN   \n",
       "XM_027434595.1                                     NaN       NaN       NaN   \n",
       "XM_027434594.1                                     NaN       NaN       NaN   \n",
       "XM_027434596.1                                     NaN       NaN       NaN   \n",
       "XM_027434597.1  unclassified transcription discrepancy       NaN       NaN   \n",
       "\n",
       "               partial start_range  \n",
       "XR_003479383.1     NaN         NaN  \n",
       "XR_003479384.1     NaN         NaN  \n",
       "XR_003479386.1     NaN         NaN  \n",
       "XR_003479388.1     NaN         NaN  \n",
       "XR_003479385.1     NaN         NaN  \n",
       "...                ...         ...  \n",
       "XM_027434593.1     NaN         NaN  \n",
       "XM_027434595.1     NaN         NaN  \n",
       "XM_027434594.1     NaN         NaN  \n",
       "XM_027434596.1     NaN         NaN  \n",
       "XM_027434597.1     NaN         NaN  \n",
       "\n",
       "[54410 rows x 24 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_anno_tss(reftss_f, gff_f, pc, ncrna, break_char='=')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2a. Create minimal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T18:03:04.937786Z",
     "start_time": "2021-04-20T18:02:52.115211Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "samples merge data/peaksInfo/samples.merge\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3051: DtypeWarning: Columns (8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "print('samples merge', merge)\n",
    "peaks_df = pd.read_csv(merge, sep=\"\\t\", index_col=0)\n",
    "peaks_df[\"Start\"] = (\n",
    "        1.0 * (peaks_df[\"End\"] + peaks_df[\"Start\"]) / 2).astype(int)\n",
    "peaks_df[\"End\"] = peaks_df[\"Start\"]  # length 1\n",
    "peaks_df[\"actual_start\"] = peaks_df[\"End\"]\n",
    "peaks_df.head()\n",
    "peaks_df.to_csv(raw_peaks, sep=\"\\t\", index=False)\n",
    "peaks_df[['Chr', 'Start', 'End', 'Strand', 'Stat']].to_csv(minimal, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2b Create homeranno"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### in snakemake"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Combine minimal and anno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T01:54:01.148835Z",
     "start_time": "2021-04-21T01:54:01.099229Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID\n",
       "1-539324    promoter-TSS (ID=exon-XR_003482412.1-1;Parent=...\n",
       "1-353478    promoter-TSS (ID=exon-XR_003487470.1-1;Parent=...\n",
       "1-146       promoter-TSS (ID=exon-XM_027391402.1-1;Parent=...\n",
       "1-349135    promoter-TSS (ID=exon-XR_003483964.1-10;Parent...\n",
       "1-300006    promoter-TSS (ID=exon-XM_027422995.1-6;Parent=...\n",
       "                                  ...                        \n",
       "1-295553                                           Intergenic\n",
       "1-295682                                           Intergenic\n",
       "1-305616                                           Intergenic\n",
       "1-285023                                           Intergenic\n",
       "1-287900                                           Intergenic\n",
       "Name: Annotation, Length: 540127, dtype: object"
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T01:57:56.147800Z",
     "start_time": "2021-04-21T01:57:52.227332Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "merge_df\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Annotation</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NW_020822452.1</td>\n",
       "      <td>16590385</td>\n",
       "      <td>16590385</td>\n",
       "      <td>+</td>\n",
       "      <td>16.227333</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003482412.1-1;Parent=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NW_020822597.1</td>\n",
       "      <td>23595396</td>\n",
       "      <td>23595396</td>\n",
       "      <td>+</td>\n",
       "      <td>5.740000</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003487470.1-1;Parent=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>NW_020822680.1</td>\n",
       "      <td>728718</td>\n",
       "      <td>728718</td>\n",
       "      <td>+</td>\n",
       "      <td>7.579579</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027391402.1-1;Parent=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NW_020822602.1</td>\n",
       "      <td>10883205</td>\n",
       "      <td>10883205</td>\n",
       "      <td>+</td>\n",
       "      <td>3.992000</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003483964.1-10;Parent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>37427907</td>\n",
       "      <td>37427907</td>\n",
       "      <td>+</td>\n",
       "      <td>129.031738</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027422995.1-6;Parent=...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Chr     Start       End Strand        Stat  \\\n",
       "ID                                                          \n",
       "0   NW_020822452.1  16590385  16590385      +   16.227333   \n",
       "1   NW_020822597.1  23595396  23595396      +    5.740000   \n",
       "2   NW_020822680.1    728718    728718      +    7.579579   \n",
       "3   NW_020822602.1  10883205  10883205      +    3.992000   \n",
       "4   NW_020822601.1  37427907  37427907      +  129.031738   \n",
       "\n",
       "                                           Annotation  \n",
       "ID                                                     \n",
       "0   promoter-TSS (ID=exon-XR_003482412.1-1;Parent=...  \n",
       "1   promoter-TSS (ID=exon-XR_003487470.1-1;Parent=...  \n",
       "2   promoter-TSS (ID=exon-XM_027391402.1-1;Parent=...  \n",
       "3   promoter-TSS (ID=exon-XR_003483964.1-10;Parent...  \n",
       "4   promoter-TSS (ID=exon-XM_027422995.1-6;Parent=...  "
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_df = pd.read_csv(minimal, index_col=0, sep='\\t')\n",
    "anno_df = pd.read_csv(homeranno, sep='\\t', na_filter=False)\n",
    "anno_df.index.name = \"ID\"\n",
    "anno_df = anno_df['Annotation']\n",
    "merge_df = pd.concat((merge_df, anno_df), axis=1)\n",
    "print('merge_df')\n",
    "(merge_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T18:03:12.710378Z",
     "start_time": "2021-04-20T18:03:04.940281Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "merge_df\n",
      "               Chr     Start       End Strand        Stat  \\\n",
      "ID                                                          \n",
      "0   NW_020822452.1  16590385  16590385      +   16.227333   \n",
      "1   NW_020822597.1  23595396  23595396      +    5.740000   \n",
      "2   NW_020822680.1    728718    728718      +    7.579579   \n",
      "3   NW_020822602.1  10883205  10883205      +    3.992000   \n",
      "4   NW_020822601.1  37427907  37427907      +  129.031738   \n",
      "\n",
      "                                           Annotation  \n",
      "ID                                                     \n",
      "0   promoter-TSS (ID=exon-XM_027399668.1-1;Parent=...  \n",
      "1                                          Intergenic  \n",
      "2   promoter-TSS (ID=exon-XM_027432478.1-3;Parent=...  \n",
      "3                                          Intergenic  \n",
      "4   promoter-TSS (ID=exon-XM_027421990.1-1;Parent=...  \n"
     ]
    }
   ],
   "source": [
    "merge_df = pd.read_csv(minimal, index_col=0, sep='\\t')\n",
    "anno_df = pd.read_csv(homeranno, sep='\\t', na_filter=False, index_col=0)\n",
    "anno_df.index.name = \"ID\"\n",
    "anno_df = anno_df['Annotation']\n",
    "merge_df = pd.concat((merge_df, anno_df), axis=1)\n",
    "print('merge_df')\n",
    "print(merge_df.head())\n",
    "merge_df.to_csv(peaks_homeranno, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Get distance to TSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T18:11:08.453039Z",
     "start_time": "2021-04-20T18:11:08.423704Z"
    }
   },
   "outputs": [],
   "source": [
    "def get_distance_anno(peaks_df, in_anno_df, nb_workers=32):\n",
    "    \"\"\" Get nearest distance and if same strand\n",
    "\n",
    "    :param peaks_df:\n",
    "    :return:\n",
    "    \"\"\"\n",
    "    \n",
    "    anno_chr_str = in_anno_df.sort_values(['Chr','Strand']).set_index(['Chr', \"Strand\"])\n",
    "    assert((peaks_df['Start']-peaks_df['End']==0).all())\n",
    "\n",
    "    def get_nearest_info(x, isSame, col_id=\"transcript_id\"):\n",
    "        # peaks of one gene\n",
    "        curr_str = x['Strand']\n",
    "        assert (curr_str == \"+\" or curr_str == \"-\")\n",
    "        if not isSame:\n",
    "            if curr_str == \"-\":\n",
    "                curr_str = \"+\"\n",
    "            else:\n",
    "                curr_str = \"-\"\n",
    "            name='TSS_Rev'\n",
    "        else:\n",
    "            name=\"TSS\"\n",
    "\n",
    "        if (x['Chr'], curr_str) not in anno_chr_str.index:\n",
    "            return pd.Series([np.nan, \"\"],\n",
    "                             index=['Distance to Nearest '+name,\n",
    "                                    f'Nearest ' + name])\n",
    "        curr_anno = anno_chr_str.loc[(x['Chr'], curr_str)].reset_index()\n",
    "        dist_to_peak = x['Start']-curr_anno['actual_start']\n",
    "        \n",
    "        min_idx = np.abs(dist_to_peak).idxmin()\n",
    "        min_dist = dist_to_peak.loc[min_idx]\n",
    "        min_id = curr_anno.loc[min_idx, col_id]\n",
    "        return pd.Series([min_dist, min_id], index=[f'Distance to Nearest {name}', f'Nearest {name}'])\n",
    "\n",
    "    nearest_peaksFw = peaks_df.parallel_apply(get_nearest_info, axis=1,\n",
    "                                              args=(True,))\n",
    "    nearest_peaksRev = peaks_df.parallel_apply(get_nearest_info, axis=1,\n",
    "                                               args=(False,))\n",
    "    peaks_anno_dist = pd.merge(nearest_peaksFw, nearest_peaksRev, left_index=True, right_index=True)\n",
    "    return peaks_anno_dist\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T18:23:46.639253Z",
     "start_time": "2021-04-20T18:23:46.165659Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: Progress bar is an experimental feature. This can lead to a considerable performance loss.\n",
      "New pandarallel memory created - Size: 2000 MB\n",
      "Pandarallel will run on 32 workers\n"
     ]
    }
   ],
   "source": [
    "pandarallel.initialize(nb_workers=cpu, verbose=2, progress_bar=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T18:29:22.193959Z",
     "start_time": "2021-04-20T18:23:48.442849Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "7e490ea34db74725a2e39f025911421a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "VBox(children=(HBox(children=(IntProgress(value=0, description='0.00%', max=16879), Label(value='0 / 16879')))…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d9f6d6ed8118455fb9898643dbe709fd",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "VBox(children=(HBox(children=(IntProgress(value=0, description='0.00%', max=16879), Label(value='0 / 16879')))…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "reftss = pd.read_csv(reftss_f, sep='\\t', index_col=0)\n",
    "reftss['actual_start'] = reftss['actual_start'].astype(int)\n",
    "peaks_df = pd.read_csv(peaks_homeranno, sep=\"\\t\", index_col=0)\n",
    "#peaks_df['actual_start'] = peaks_df['actual_start'].astype(int)\n",
    "peaks_anno_dist = get_distance_anno(peaks_df, reftss,\n",
    "                                    nb_workers=cpu)\n",
    "peaks_anno_dist.to_csv(peaks_distance,sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Extract and remove CDS and intron peaks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5a. get intron bed file - uses R GenomeRanges\n",
    "## 5b. get CDS bed file (should already have)\n",
    "## 5c. remove the filters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T18:47:31.897457Z",
     "start_time": "2021-04-20T18:47:31.874976Z"
    }
   },
   "outputs": [],
   "source": [
    "from tss.config import HOMER_PATH\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T19:10:10.912361Z",
     "start_time": "2021-04-20T19:09:55.483081Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data2/resources/software/homer/v4.11_10242019/bin///pos2bed.pl data/GCF/anno/peaks.merge.tss.minimal.anno > data/GCF/anno/peaks.merge.tss.minimal.anno.bed\n",
      "bedtools intersect -s -v -a data/GCF/anno/peaks.merge.tss.minimal.anno.bed -b data/input/CDS.bed > data/GCF/genebody_filters/peaks.nocds.index.bed\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/ipykernel_launcher.py:22: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Function that uses bedtools intersect to not have any peaks that\n",
    "overlap with Coding sequence since that would be harder to call a start site (even if alternative start sites).\n",
    "Will make bed file of this and tsv file.\n",
    "\n",
    "\"\"\"\n",
    "# Create temporary bed file and then convert to peak-like\n",
    "# Create bed\n",
    "cmd = f\"{HOMER_PATH}/pos2bed.pl {peaks_homeranno} > {peaks_homeranno}.bed\"\n",
    "print(cmd)\n",
    "os.system(cmd)\n",
    "\n",
    "#-v says get the non-intersecting peaks\n",
    "cmd = f\"bedtools intersect -s -v -a {peaks_homeranno}.bed -b {cds_f} > {noncds_peaks}.bed\"\n",
    "\n",
    "print(cmd)\n",
    "os.system(cmd)\n",
    "\n",
    "bed_df = pd.read_csv(f\"{noncds_peaks}.bed\", sep='\\t', header=None)\n",
    "bed_df[3].to_csv(noncds_peaks, index=None, header=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T19:55:24.723451Z",
     "start_time": "2021-04-20T19:55:19.708119Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bedtools intersect -s -v -a data/GCF/anno/peaks.merge.tss.minimal.anno.bed -b data/GCF/genome/introns.bed > data/GCF/genebody_filters/peaks.nointron.index.bed\n"
     ]
    }
   ],
   "source": [
    "\"\"\"\n",
    "Function that uses bedtools intersect to not have any peaks that\n",
    "overlap with Coding sequence since that would be harder to call a start site (even if alternative start sites).\n",
    "Will make bed file of this and tsv file.\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "#-v says get the non-intersecting peaks\n",
    "cmd = f\"bedtools intersect -s -v -a {peaks_homeranno}.bed -b {intron_f} > {nonintron_peaks}.bed\"\n",
    "\n",
    "print(cmd)\n",
    "os.system(cmd)\n",
    "\n",
    "bed_df = pd.read_csv(f\"{nonintron_peaks}.bed\", sep='\\t', header=None)\n",
    "bed_df[3].to_csv(nonintron_peaks, index=None, header=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Call pc and ncrna genes by distance and get refType"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T23:28:05.361196Z",
     "start_time": "2021-04-20T23:27:58.649106Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "peaks_distance=f\"data/{genome}/anno/peaks.distance.tsv\"\n",
    "refTSS=f\"data/{genome}/genome/refTSS.tsv\"\n",
    "homeranno=f\"data/{genome}/anno/peaks.merge.tss.anno\"\n",
    "\n",
    "#output\n",
    "refType = f\"data/{genome}/refType/peaks.refType.tsv\"\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T23:29:10.656587Z",
     "start_time": "2021-04-20T23:29:10.635292Z"
    }
   },
   "outputs": [],
   "source": [
    "#params\n",
    "intergenic_dist=1000\n",
    "bins=(-1000,1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T23:29:11.217410Z",
     "start_time": "2021-04-20T23:29:10.658887Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df before dropping no nearest tss (540127, 4)\n",
      "df after dropping no nearest tss (535730, 4)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  if __name__ == '__main__':\n",
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/ipykernel_launcher.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  # This is added back by InteractiveShellApp.init_path()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Distance to Nearest TSS</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to Nearest TSS_Rev</th>\n",
       "      <th>Nearest TSS_Rev</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>homeranno</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>117319.0</td>\n",
       "      <td>XM_027399667.1</td>\n",
       "      <td>462.0</td>\n",
       "      <td>XM_027399668.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027399668.1-1;Parent=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>-23850.0</td>\n",
       "      <td>XM_027420351.1</td>\n",
       "      <td>162428.0</td>\n",
       "      <td>XM_027420364.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>60370.0</td>\n",
       "      <td>XM_027432388.1</td>\n",
       "      <td>-746.0</td>\n",
       "      <td>XM_027432478.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027432478.1-3;Parent=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>9079.0</td>\n",
       "      <td>XM_027423054.1</td>\n",
       "      <td>14247.0</td>\n",
       "      <td>XM_027422736.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>-11.0</td>\n",
       "      <td>XM_027421990.1</td>\n",
       "      <td>-53290.0</td>\n",
       "      <td>XM_027421992.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027421990.1-1;Parent=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540122</td>\n",
       "      <td>54872.0</td>\n",
       "      <td>XM_027415965.1</td>\n",
       "      <td>58.0</td>\n",
       "      <td>XM_027415967.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027415967.1-1;Parent=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540123</td>\n",
       "      <td>-7856.0</td>\n",
       "      <td>XM_027405626.1</td>\n",
       "      <td>149995.0</td>\n",
       "      <td>XR_003483403.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540124</td>\n",
       "      <td>107085.0</td>\n",
       "      <td>XR_003480196.1</td>\n",
       "      <td>-6829.0</td>\n",
       "      <td>XR_003480188.1</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>Intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540125</td>\n",
       "      <td>-1383.0</td>\n",
       "      <td>XM_027416357.1</td>\n",
       "      <td>-30066.0</td>\n",
       "      <td>XM_027416362.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>TTS (ID=exon-XM_027416357.1-1;Parent=rna-XM_02...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540126</td>\n",
       "      <td>-87955.0</td>\n",
       "      <td>XM_027421500.1</td>\n",
       "      <td>-21927.0</td>\n",
       "      <td>XM_027421498.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>535730 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Distance to Nearest TSS     Nearest TSS  Distance to Nearest TSS_Rev  \\\n",
       "ID                                                                             \n",
       "0                      117319.0  XM_027399667.1                        462.0   \n",
       "1                      -23850.0  XM_027420351.1                     162428.0   \n",
       "2                       60370.0  XM_027432388.1                       -746.0   \n",
       "3                        9079.0  XM_027423054.1                      14247.0   \n",
       "4                         -11.0  XM_027421990.1                     -53290.0   \n",
       "...                         ...             ...                          ...   \n",
       "540122                  54872.0  XM_027415965.1                         58.0   \n",
       "540123                  -7856.0  XM_027405626.1                     149995.0   \n",
       "540124                 107085.0  XR_003480196.1                      -6829.0   \n",
       "540125                  -1383.0  XM_027416357.1                     -30066.0   \n",
       "540126                 -87955.0  XM_027421500.1                     -21927.0   \n",
       "\n",
       "       Nearest TSS_Rev Annotation  \\\n",
       "ID                                  \n",
       "0       XM_027399668.1       mRNA   \n",
       "1       XM_027420364.1       mRNA   \n",
       "2       XM_027432478.1       mRNA   \n",
       "3       XM_027422736.1       mRNA   \n",
       "4       XM_027421992.1       mRNA   \n",
       "...                ...        ...   \n",
       "540122  XM_027415967.1       mRNA   \n",
       "540123  XR_003483403.1       mRNA   \n",
       "540124  XR_003480188.1    lnc_RNA   \n",
       "540125  XM_027416362.1       mRNA   \n",
       "540126  XM_027421498.1       mRNA   \n",
       "\n",
       "                                                homeranno  \n",
       "ID                                                         \n",
       "0       promoter-TSS (ID=exon-XM_027399668.1-1;Parent=...  \n",
       "1                                              Intergenic  \n",
       "2       promoter-TSS (ID=exon-XM_027432478.1-3;Parent=...  \n",
       "3                                              Intergenic  \n",
       "4       promoter-TSS (ID=exon-XM_027421990.1-1;Parent=...  \n",
       "...                                                   ...  \n",
       "540122  promoter-TSS (ID=exon-XM_027415967.1-1;Parent=...  \n",
       "540123                                         Intergenic  \n",
       "540124                                         Intergenic  \n",
       "540125  TTS (ID=exon-XM_027416357.1-1;Parent=rna-XM_02...  \n",
       "540126                                         Intergenic  \n",
       "\n",
       "[535730 rows x 6 columns]"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(peaks_distance,sep='\\t', index_col=0)\n",
    "ref_df=pd.read_csv(refTSS,sep='\\t', index_col=0)\n",
    "homeranno_df = pd.read_csv(homeranno,sep='\\t', index_col=0)\n",
    "\n",
    "print('df before dropping no nearest tss', df.shape)\n",
    "df = df.dropna(subset=[\"Nearest TSS\"])\n",
    "print('df after dropping no nearest tss', df.shape)\n",
    "df['Annotation'] = df[\"Nearest TSS\"].map(ref_df[\"Annotation\"].to_dict())\n",
    "\n",
    "df['homeranno'] = homeranno_df.loc[df.index, 'Annotation']\n",
    "df\n",
    "\n",
    "df.head()\n",
    "\n",
    "# get intergenic\n",
    "intergenic = df[(df['homeranno'].str.upper() == \"INTERGENIC\") &\n",
    "                (df['Distance to Nearest TSS']<intergenic_dist)]\n",
    "\n",
    "intergenic\n",
    "\n",
    "promoter = df[(df['Distance to Nearest TSS']>bins[0]) & (df['Distance to Nearest TSS']<bins[1])]\n",
    "promoter\n",
    "\n",
    "ncrna\n",
    "\n",
    "\n",
    "pc_promoters = promoter[promoter['Annotation']==pc]\n",
    "ncrna_promoters = promoter[promoter['Annotation'].isin(ncrna)]\n",
    "print(pc_promoters.shape)\n",
    "pc_promoters.head()\n",
    "\n",
    "print('Total', pc_promoters.shape[0])\n",
    "print('annotated as promoter by homer', pc_promoters['homeranno'].str.contains(\"promoter\").sum())\n",
    "\n",
    "print(ncrna_promoters.shape)\n",
    "ncrna_promoters.head()\n",
    "\n",
    "nonpromoters = df.drop(promoter.index)\n",
    "divergent = nonpromoters[(nonpromoters['Distance to Nearest TSS_Rev']>bins[0]) &\n",
    "            (nonpromoters['Distance to Nearest TSS_Rev']<bins[1])]\n",
    "print('divergent', divergent.shape)\n",
    "\n",
    "df['refType'] = None\n",
    "df.loc[intergenic.index, 'refType'] = \"Intergenic\"\n",
    "df.loc[pc_promoters.index, 'refType'] = \"mRNA\"\n",
    "df.loc[ncrna_promoters.index, 'refType'] = \"ncRNA\"\n",
    "df.loc[divergent.index, 'refType'] = \"divergent\"\n",
    "df\n",
    "\n",
    "df.to_csv(refType,sep='\\t')\n",
    "\n",
    "%matplotlib inline\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "sns.countplot(df['refType'])\n",
    "plt.savefig(refType+\".png\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T03:06:22.128300Z",
     "start_time": "2021-04-21T03:06:22.072306Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MiscOrgans</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Brain</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Heart</th>\n",
       "      <th>CHO</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>FemaleReproductive</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Kidney</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>BMDM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540122</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540123</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540124</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540125</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540126</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>540127 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        MiscOrgans  Liver  Brain   Lung  Heart    CHO  Pancreas  \\\n",
       "0            False  False  False   True  False   True     False   \n",
       "1             True   True   True   True  False  False     False   \n",
       "2            False  False   True   True  False   True      True   \n",
       "3             True  False   True   True  False   True      True   \n",
       "4             True   True   True   True   True   True      True   \n",
       "...            ...    ...    ...    ...    ...    ...       ...   \n",
       "540122       False  False  False  False  False  False      True   \n",
       "540123       False  False  False   True  False  False     False   \n",
       "540124       False  False  False   True  False  False     False   \n",
       "540125        True  False  False  False  False  False     False   \n",
       "540126        True  False  False   True  False  False     False   \n",
       "\n",
       "        FemaleReproductive  Muscle  Kidney  Spleen   BMDM  \n",
       "0                    False   False    True   False   True  \n",
       "1                     True   False    True    True  False  \n",
       "2                     True    True    True    True   True  \n",
       "3                     True    True    True    True   True  \n",
       "4                     True    True    True    True   True  \n",
       "...                    ...     ...     ...     ...    ...  \n",
       "540122               False   False   False    True   True  \n",
       "540123               False   False   False    True   True  \n",
       "540124               False   False   False    True   True  \n",
       "540125               False   False   False   False   True  \n",
       "540126                True    True    True   False  False  \n",
       "\n",
       "[540127 rows x 12 columns]"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "above_thresh = tissues>1\n",
    "above_thresh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T03:12:55.423236Z",
     "start_time": "2021-04-21T03:12:55.377848Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "a=above_thresh.loc[above_thresh[\"CHO\"]].index\n",
    "b=above_thresh.loc[above_thresh.drop(\"CHO\", axis=1).any(axis=1)].index\n",
    "tot=above_thresh.loc[above_thresh.any(axis=1)].index\n",
    "\n",
    "tissues.loc[set(a).intersection(set(b)), \"isCHO\"] = \"Overlap\"\n",
    "tissues.loc[set(a)-(set(b)), \"isCHO\"] = \"CHO\"\n",
    "tissues.loc[set(b)-set(a), \"isCHO\"] = \"Tissues\"\n",
    "tissues.loc[set(tot)-(set(a).union(set(b))), \"isCHO\"] = \"None\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T03:13:20.687534Z",
     "start_time": "2021-04-21T03:13:19.893479Z"
    }
   },
   "outputs": [],
   "source": [
    "import tss.utils.sankey_helper as sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T04:24:28.406824Z",
     "start_time": "2021-04-21T04:24:26.627422Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Distance to Nearest TSS</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to Nearest TSS_Rev</th>\n",
       "      <th>Nearest TSS_Rev</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>homeranno</th>\n",
       "      <th>refType</th>\n",
       "      <th>isCDS</th>\n",
       "      <th>isIntron</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>117319.0</td>\n",
       "      <td>XM_027399667.1</td>\n",
       "      <td>462.0</td>\n",
       "      <td>XM_027399668.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003482412.1-1;Parent=...</td>\n",
       "      <td>divergent</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>-23850.0</td>\n",
       "      <td>XM_027420351.1</td>\n",
       "      <td>162428.0</td>\n",
       "      <td>XM_027420364.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003487470.1-1;Parent=...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>60370.0</td>\n",
       "      <td>XM_027432388.1</td>\n",
       "      <td>-746.0</td>\n",
       "      <td>XM_027432478.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027391402.1-1;Parent=...</td>\n",
       "      <td>divergent</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>9079.0</td>\n",
       "      <td>XM_027423054.1</td>\n",
       "      <td>14247.0</td>\n",
       "      <td>XM_027422736.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XR_003483964.1-10;Parent...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>-11.0</td>\n",
       "      <td>XM_027421990.1</td>\n",
       "      <td>-53290.0</td>\n",
       "      <td>XM_027421992.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>promoter-TSS (ID=exon-XM_027422995.1-6;Parent=...</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540122</td>\n",
       "      <td>54872.0</td>\n",
       "      <td>XM_027415965.1</td>\n",
       "      <td>58.0</td>\n",
       "      <td>XM_027415967.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>divergent</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540123</td>\n",
       "      <td>-7856.0</td>\n",
       "      <td>XM_027405626.1</td>\n",
       "      <td>149995.0</td>\n",
       "      <td>XR_003483403.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540124</td>\n",
       "      <td>107085.0</td>\n",
       "      <td>XR_003480196.1</td>\n",
       "      <td>-6829.0</td>\n",
       "      <td>XR_003480188.1</td>\n",
       "      <td>lnc_RNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540125</td>\n",
       "      <td>-1383.0</td>\n",
       "      <td>XM_027416357.1</td>\n",
       "      <td>-30066.0</td>\n",
       "      <td>XM_027416362.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540126</td>\n",
       "      <td>-87955.0</td>\n",
       "      <td>XM_027421500.1</td>\n",
       "      <td>-21927.0</td>\n",
       "      <td>XM_027421498.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>535730 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Distance to Nearest TSS     Nearest TSS  Distance to Nearest TSS_Rev  \\\n",
       "ID                                                                             \n",
       "0                      117319.0  XM_027399667.1                        462.0   \n",
       "1                      -23850.0  XM_027420351.1                     162428.0   \n",
       "2                       60370.0  XM_027432388.1                       -746.0   \n",
       "3                        9079.0  XM_027423054.1                      14247.0   \n",
       "4                         -11.0  XM_027421990.1                     -53290.0   \n",
       "...                         ...             ...                          ...   \n",
       "540122                  54872.0  XM_027415965.1                         58.0   \n",
       "540123                  -7856.0  XM_027405626.1                     149995.0   \n",
       "540124                 107085.0  XR_003480196.1                      -6829.0   \n",
       "540125                  -1383.0  XM_027416357.1                     -30066.0   \n",
       "540126                 -87955.0  XM_027421500.1                     -21927.0   \n",
       "\n",
       "       Nearest TSS_Rev Annotation  \\\n",
       "ID                                  \n",
       "0       XM_027399668.1       mRNA   \n",
       "1       XM_027420364.1       mRNA   \n",
       "2       XM_027432478.1       mRNA   \n",
       "3       XM_027422736.1       mRNA   \n",
       "4       XM_027421992.1       mRNA   \n",
       "...                ...        ...   \n",
       "540122  XM_027415967.1       mRNA   \n",
       "540123  XR_003483403.1       mRNA   \n",
       "540124  XR_003480188.1    lnc_RNA   \n",
       "540125  XM_027416362.1       mRNA   \n",
       "540126  XM_027421498.1       mRNA   \n",
       "\n",
       "                                                homeranno     refType  isCDS  \\\n",
       "ID                                                                             \n",
       "0       promoter-TSS (ID=exon-XR_003482412.1-1;Parent=...   divergent   True   \n",
       "1       promoter-TSS (ID=exon-XR_003487470.1-1;Parent=...         NaN  False   \n",
       "2       promoter-TSS (ID=exon-XM_027391402.1-1;Parent=...   divergent  False   \n",
       "3       promoter-TSS (ID=exon-XR_003483964.1-10;Parent...         NaN  False   \n",
       "4       promoter-TSS (ID=exon-XM_027422995.1-6;Parent=...        mRNA  False   \n",
       "...                                                   ...         ...    ...   \n",
       "540122                                         Intergenic   divergent  False   \n",
       "540123                                         Intergenic  Intergenic  False   \n",
       "540124                                         Intergenic         NaN  False   \n",
       "540125                                         Intergenic  Intergenic  False   \n",
       "540126                                         Intergenic  Intergenic  False   \n",
       "\n",
       "        isIntron  \n",
       "ID                \n",
       "0           True  \n",
       "1           True  \n",
       "2          False  \n",
       "3          False  \n",
       "4          False  \n",
       "...          ...  \n",
       "540122     False  \n",
       "540123     False  \n",
       "540124     False  \n",
       "540125      True  \n",
       "540126     False  \n",
       "\n",
       "[535730 rows x 9 columns]"
      ]
     },
     "execution_count": 267,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genebody = pd.read_csv(f\"data/{genome}/refType/peaks.refType.geneBody.tsv\", sep='\\t', index_col=0)\n",
    "genebody"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-21T05:45:30.302749Z",
     "start_time": "2021-04-21T05:45:30.084858Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>num_above_cpm</th>\n",
       "      <th>isAbove_numsamples</th>\n",
       "      <th>isCHO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>True</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "      <td>Overlap</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540122</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540123</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540124</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540125</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540126</td>\n",
       "      <td>5</td>\n",
       "      <td>True</td>\n",
       "      <td>Tissues</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>540127 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        num_above_cpm  isAbove_numsamples    isCHO\n",
       "0                   4                True  Overlap\n",
       "1                   7                True  Tissues\n",
       "2                   9                True  Overlap\n",
       "3                  10                True  Overlap\n",
       "4                  12                True  Overlap\n",
       "...               ...                 ...      ...\n",
       "540122              3                True  Tissues\n",
       "540123              3                True  Tissues\n",
       "540124              3                True  Tissues\n",
       "540125              2               False  Tissues\n",
       "540126              5                True  Tissues\n",
       "\n",
       "[540127 rows x 3 columns]"
      ]
     },
     "execution_count": 270,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thresh=1\n",
    "tissues_score = pd.read_csv(f'data/peaksInfo/tissues_scoring.thresh{thresh}.tsv',sep='\\t',index_col=0)\n",
    "tissues_score"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
