{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:20.706371Z",
     "start_time": "2019-03-07T06:33:20.607072Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Parameters specific to where your folders are and your data\n",
    "parameter_file = 'params/params.yaml'\n",
    "import yaml\n",
    "import sys\n",
    "\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "#p = dic2obj(**doc)\n",
    "\n",
    "data_folder = doc['data_folder']\n",
    "tissues = doc['tissues'].split(',')\n",
    "sys.path.append(doc['pipeline_path'])\n",
    "ref_fa = doc['ref_fa']\n",
    "annotation=doc['annotation']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:20.713389Z",
     "start_time": "2019-03-07T06:33:20.709670Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss_annotation = doc['tss_annotation']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:21.392840Z",
     "start_time": "2019-03-07T06:33:20.716442Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Number of tissues: ', 13)\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import pickle\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "from matplotlib import pyplot as plt\n",
    "from itertools import product\n",
    "import glob\n",
    "import re\n",
    "from matplotlib_venn import venn2\n",
    "from matplotlib import rcParams\n",
    "import inspect\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "sys.setrecursionlimit(3000)\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "rcParams['figure.figsize'] = 8, 6\n",
    "import tqdm\n",
    "### notebook specific configuration ###\n",
    "from os.path import basename\n",
    "#mpl.style.use('ggplot')\n",
    "mpl.style.use('fivethirtyeight')\n",
    "from cycler import cycler\n",
    "mpl.rcParams['axes.prop_cycle'] = cycler(color='bgrcmyk')\n",
    "\n",
    "from Homer import *\n",
    "\n",
    "print('Number of tissues: ',len(tissues))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:21.515708Z",
     "start_time": "2019-03-07T06:33:21.394925Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Name</th>\n",
       "      <th>Full</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDM1hKLA_ATAC1</td>\n",
       "      <td>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDM1hKLA_ATAC2</td>\n",
       "      <td>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>GRO</td>\n",
       "      <td>BMDM1hKLA_GRO1</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>BMDM1hKLA_GROCap1</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDMwt</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDMwt_ATAC1</td>\n",
       "      <td>CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Tissue Experiment  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  BMDM1hKLA       ATAC   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  BMDM1hKLA       ATAC   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  BMDM1hKLA        GRO   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA     GROCap   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...     BMDMwt       ATAC   \n",
       "\n",
       "                                                                 Name  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...    BMDM1hKLA_ATAC1   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...    BMDM1hKLA_ATAC2   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...     BMDM1hKLA_GRO1   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA_GROCap1   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...       BMDMwt_ATAC1   \n",
       "\n",
       "                                                                                                 Full  \n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...  \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...  \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...  \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...  \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...  CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_samples = pd.read_csv('../raw_metaSamples.tsv',sep=\"\\t\",index_col=0)\n",
    "meta_samples.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load anno_peaks. \n",
    "Later will filter based on Distance to TSS\n",
    "\n",
    "## Load peaks_tissues matrix\n",
    "Later will be used to merge two data types in a tissue and for more confident peaks if they have more than 1 tissues present\n",
    "\n",
    "## Load tss\n",
    "\n",
    "## Load txn_df\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:23.090740Z",
     "start_time": "2019-03-07T06:33:21.517689Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "      <th>Nearest gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_020822452.1</td>\n",
       "      <td>16590187</td>\n",
       "      <td>16590584</td>\n",
       "      <td>+</td>\n",
       "      <td>16.227333</td>\n",
       "      <td>promoter-TSS (ID=id181373;Parent=rna15526;Dbxr...</td>\n",
       "      <td>XM_027399668.1</td>\n",
       "      <td>-462.0</td>\n",
       "      <td>False</td>\n",
       "      <td>Clic4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_020822597.1</td>\n",
       "      <td>23595304</td>\n",
       "      <td>23595489</td>\n",
       "      <td>+</td>\n",
       "      <td>5.740000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027420351.1</td>\n",
       "      <td>-23850.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Baz2b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_020822680.1</td>\n",
       "      <td>728607</td>\n",
       "      <td>728829</td>\n",
       "      <td>+</td>\n",
       "      <td>7.579579</td>\n",
       "      <td>promoter-TSS (ID=id609942;Parent=rna53731;Dbxr...</td>\n",
       "      <td>XM_027432478.1</td>\n",
       "      <td>746.0</td>\n",
       "      <td>False</td>\n",
       "      <td>Irak1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_020822602.1</td>\n",
       "      <td>10883074</td>\n",
       "      <td>10883337</td>\n",
       "      <td>+</td>\n",
       "      <td>3.992000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027423054.1</td>\n",
       "      <td>9079.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Gbgt1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>37427816</td>\n",
       "      <td>37427999</td>\n",
       "      <td>+</td>\n",
       "      <td>129.031738</td>\n",
       "      <td>promoter-TSS (ID=id474643;Parent=rna41513;Dbxr...</td>\n",
       "      <td>XM_027421990.1</td>\n",
       "      <td>-11.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Shf</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Chr     Start       End Strand        Stat  \\\n",
       "ID                                                          \n",
       "0   NW_020822452.1  16590187  16590584      +   16.227333   \n",
       "1   NW_020822597.1  23595304  23595489      +    5.740000   \n",
       "2   NW_020822680.1    728607    728829      +    7.579579   \n",
       "3   NW_020822602.1  10883074  10883337      +    3.992000   \n",
       "4   NW_020822601.1  37427816  37427999      +  129.031738   \n",
       "\n",
       "                                           Annotation     Nearest TSS  \\\n",
       "ID                                                                      \n",
       "0   promoter-TSS (ID=id181373;Parent=rna15526;Dbxr...  XM_027399668.1   \n",
       "1                                          Intergenic  XM_027420351.1   \n",
       "2   promoter-TSS (ID=id609942;Parent=rna53731;Dbxr...  XM_027432478.1   \n",
       "3                                          Intergenic  XM_027423054.1   \n",
       "4   promoter-TSS (ID=id474643;Parent=rna41513;Dbxr...  XM_027421990.1   \n",
       "\n",
       "    Distance to TSS  isSameStrand Nearest gene  \n",
       "ID                                              \n",
       "0            -462.0         False        Clic4  \n",
       "1          -23850.0          True        Baz2b  \n",
       "2             746.0         False        Irak1  \n",
       "3            9079.0          True        Gbgt1  \n",
       "4             -11.0          True          Shf  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno_peaks = pd.read_csv('Results/merged/peaks_with_tss_distances.tsv',sep='\\t',index_col=0)\n",
    "anno_peaks.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:24.909761Z",
     "start_time": "2019-03-07T06:33:23.092594Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f04_peaks/MiscOrgans_START1_and_MiscOrgans_startinput1</th>\n",
       "      <th>f04_peaks/Liver_START2_and_Liver_startinput1</th>\n",
       "      <th>f04_peaks/Brain_GROCap1_and_Brain_GRO1</th>\n",
       "      <th>f04_peaks/Liver_START1_and_Liver_startinput1</th>\n",
       "      <th>f04_peaks/Lung_GROCap1_and_Lung_GRO1</th>\n",
       "      <th>f04_peaks/Brain_START3_and_Brain_startinput1</th>\n",
       "      <th>f04_peaks/Heart_START1_and_Heart_startinput1</th>\n",
       "      <th>f04_peaks/CHO_GROCap2_and_CHO_GRO1</th>\n",
       "      <th>f04_peaks/Heart_START2_and_Heart_startinput1</th>\n",
       "      <th>f04_peaks/Liver_GROCap1_and_Liver_GRO1</th>\n",
       "      <th>...</th>\n",
       "      <th>f04_peaks/MiscOrgans_START2_and_MiscOrgans_startinput1</th>\n",
       "      <th>f04_peaks/Spleen_START1_and_Spleen_startinput1</th>\n",
       "      <th>f04_peaks/Lung_START2_and_Lung_startinput1</th>\n",
       "      <th>f04_peaks/CHO_START1_and_CHO_startinput1</th>\n",
       "      <th>f04_peaks/FemaleReproductive_START1_and_FemaleReproductive_startinput1</th>\n",
       "      <th>f04_peaks/Kidney_START2_and_Kidney_startinput1</th>\n",
       "      <th>f04_peaks/Kidney_GROCap1_and_Kidney_GRO1</th>\n",
       "      <th>f04_peaks/CHO_GROCap1_and_CHO_GRO1</th>\n",
       "      <th>f04_peaks/BMDMwt_GROCap1_and_BMDMwt_GRO1</th>\n",
       "      <th>f04_peaks/Spleen_START2_and_Spleen_startinput1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.41</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.270</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.54</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.67</td>\n",
       "      <td>31.8</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.38</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.39</td>\n",
       "      <td>33.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.97</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.060</td>\n",
       "      <td>3.3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5.98</td>\n",
       "      <td>1.05</td>\n",
       "      <td>40.6</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5.69</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.41</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.952</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.79</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>22.6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.82</td>\n",
       "      <td>1.73</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.83</td>\n",
       "      <td>21.60</td>\n",
       "      <td>5.66</td>\n",
       "      <td>2.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3.95</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.24</td>\n",
       "      <td>1.67</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.10</td>\n",
       "      <td>3.46</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.63</td>\n",
       "      <td>2.45</td>\n",
       "      <td>2.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>165.30</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.860</td>\n",
       "      <td>81.8</td>\n",
       "      <td>0.00</td>\n",
       "      <td>43.50</td>\n",
       "      <td>51.20</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.3</td>\n",
       "      <td>...</td>\n",
       "      <td>58.80</td>\n",
       "      <td>18.20</td>\n",
       "      <td>0.0</td>\n",
       "      <td>91.90</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.52</td>\n",
       "      <td>4.78</td>\n",
       "      <td>3.39</td>\n",
       "      <td>158.90</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   f04_peaks/MiscOrgans_START1_and_MiscOrgans_startinput1  \\\n",
       "0                                               1.41        \n",
       "1                                               1.97        \n",
       "2                                               1.41        \n",
       "3                                               3.95        \n",
       "4                                             165.30        \n",
       "\n",
       "   f04_peaks/Liver_START2_and_Liver_startinput1  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           0.0   \n",
       "\n",
       "   f04_peaks/Brain_GROCap1_and_Brain_GRO1  \\\n",
       "0                                   1.270   \n",
       "1                                   2.060   \n",
       "2                                   0.952   \n",
       "3                                   0.000   \n",
       "4                                   1.860   \n",
       "\n",
       "   f04_peaks/Liver_START1_and_Liver_startinput1  \\\n",
       "0                                           0.0   \n",
       "1                                           3.3   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                          81.8   \n",
       "\n",
       "   f04_peaks/Lung_GROCap1_and_Lung_GRO1  \\\n",
       "0                                  6.54   \n",
       "1                                  0.00   \n",
       "2                                  3.79   \n",
       "3                                  0.00   \n",
       "4                                  0.00   \n",
       "\n",
       "   f04_peaks/Brain_START3_and_Brain_startinput1  \\\n",
       "0                                          0.00   \n",
       "1                                          0.00   \n",
       "2                                          0.00   \n",
       "3                                          3.24   \n",
       "4                                         43.50   \n",
       "\n",
       "   f04_peaks/Heart_START1_and_Heart_startinput1  \\\n",
       "0                                          1.67   \n",
       "1                                          0.00   \n",
       "2                                          0.00   \n",
       "3                                          1.67   \n",
       "4                                         51.20   \n",
       "\n",
       "   f04_peaks/CHO_GROCap2_and_CHO_GRO1  \\\n",
       "0                                31.8   \n",
       "1                                 0.0   \n",
       "2                                22.6   \n",
       "3                                 0.0   \n",
       "4                                 0.0   \n",
       "\n",
       "   f04_peaks/Heart_START2_and_Heart_startinput1  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           0.0   \n",
       "\n",
       "   f04_peaks/Liver_GROCap1_and_Liver_GRO1  ...  \\\n",
       "0                                     0.0  ...   \n",
       "1                                     0.0  ...   \n",
       "2                                     0.0  ...   \n",
       "3                                     0.0  ...   \n",
       "4                                     7.3  ...   \n",
       "\n",
       "   f04_peaks/MiscOrgans_START2_and_MiscOrgans_startinput1  \\\n",
       "0                                               0.00        \n",
       "1                                               5.98        \n",
       "2                                               0.00        \n",
       "3                                               2.10        \n",
       "4                                              58.80        \n",
       "\n",
       "   f04_peaks/Spleen_START1_and_Spleen_startinput1  \\\n",
       "0                                            0.00   \n",
       "1                                            1.05   \n",
       "2                                            0.00   \n",
       "3                                            3.46   \n",
       "4                                           18.20   \n",
       "\n",
       "   f04_peaks/Lung_START2_and_Lung_startinput1  \\\n",
       "0                                         0.0   \n",
       "1                                        40.6   \n",
       "2                                         0.0   \n",
       "3                                         0.0   \n",
       "4                                         0.0   \n",
       "\n",
       "   f04_peaks/CHO_START1_and_CHO_startinput1  \\\n",
       "0                                      5.38   \n",
       "1                                      0.00   \n",
       "2                                      6.82   \n",
       "3                                      3.14   \n",
       "4                                     91.90   \n",
       "\n",
       "   f04_peaks/FemaleReproductive_START1_and_FemaleReproductive_startinput1  \\\n",
       "0                                               0.00                        \n",
       "1                                               5.69                        \n",
       "2                                               1.73                        \n",
       "3                                               0.00                        \n",
       "4                                               0.00                        \n",
       "\n",
       "   f04_peaks/Kidney_START2_and_Kidney_startinput1  \\\n",
       "0                                             0.0   \n",
       "1                                             0.0   \n",
       "2                                             0.0   \n",
       "3                                             0.0   \n",
       "4                                             0.0   \n",
       "\n",
       "   f04_peaks/Kidney_GROCap1_and_Kidney_GRO1  \\\n",
       "0                                      6.39   \n",
       "1                                      0.00   \n",
       "2                                      1.83   \n",
       "3                                      0.00   \n",
       "4                                      1.52   \n",
       "\n",
       "   f04_peaks/CHO_GROCap1_and_CHO_GRO1  \\\n",
       "0                               33.00   \n",
       "1                                0.00   \n",
       "2                               21.60   \n",
       "3                                2.63   \n",
       "4                                4.78   \n",
       "\n",
       "   f04_peaks/BMDMwt_GROCap1_and_BMDMwt_GRO1  \\\n",
       "0                                      0.00   \n",
       "1                                      0.00   \n",
       "2                                      5.66   \n",
       "3                                      2.45   \n",
       "4                                      3.39   \n",
       "\n",
       "   f04_peaks/Spleen_START2_and_Spleen_startinput1  \n",
       "0                                            0.00  \n",
       "1                                            5.72  \n",
       "2                                            2.51  \n",
       "3                                            2.29  \n",
       "4                                          158.90  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peaks_tissue = pd.read_csv(\"Results/merged/samples.merge.peaksexpression\",index_col=0,sep=\"\\t\")\n",
    "peaks_tissue.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Read in tss and txn_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:25.391547Z",
     "start_time": "2019-03-07T06:33:24.911657Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Dbxref</th>\n",
       "      <th>Name</th>\n",
       "      <th>gbkey</th>\n",
       "      <th>gene</th>\n",
       "      <th>...</th>\n",
       "      <th>product</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>Note</th>\n",
       "      <th>exception</th>\n",
       "      <th>end_range</th>\n",
       "      <th>partial</th>\n",
       "      <th>start_range</th>\n",
       "      <th>inference</th>\n",
       "      <th>Length</th>\n",
       "      <th>actual_start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Genbank:XM_027386529.1;Name=XM_027386529.1;gbkey=mRNA;gene=Bmp4;model_evidence=Supporting evidence includes similarity to: 29 Proteins%2C and 100%25 coverage of the annotated genomic feature by RNAseq alignments%2C including 15 samples with support for all annotated introns;product=bone morphogenetic protein 4%2C transcript variant X3;transcript_id=XM_027386529.1</th>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>717871</td>\n",
       "      <td>722308</td>\n",
       "      <td>-</td>\n",
       "      <td>rna7</td>\n",
       "      <td>gene4</td>\n",
       "      <td>GeneID:100754303,Genbank:XM_027386529.1</td>\n",
       "      <td>XM_027386529.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Bmp4</td>\n",
       "      <td>...</td>\n",
       "      <td>bone morphogenetic protein 4%2C transcript var...</td>\n",
       "      <td>XM_027386529.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4437</td>\n",
       "      <td>722308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Genbank:XM_027386528.1;Name=XM_027386528.1;gbkey=mRNA;gene=Bmp4;model_evidence=Supporting evidence includes similarity to: 29 Proteins%2C and 100%25 coverage of the annotated genomic feature by RNAseq alignments%2C including 18 samples with support for all annotated introns;product=bone morphogenetic protein 4%2C transcript variant X2;transcript_id=XM_027386528.1</th>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>717871</td>\n",
       "      <td>724610</td>\n",
       "      <td>-</td>\n",
       "      <td>rna6</td>\n",
       "      <td>gene4</td>\n",
       "      <td>GeneID:100754303,Genbank:XM_027386528.1</td>\n",
       "      <td>XM_027386528.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Bmp4</td>\n",
       "      <td>...</td>\n",
       "      <td>bone morphogenetic protein 4%2C transcript var...</td>\n",
       "      <td>XM_027386528.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6739</td>\n",
       "      <td>724610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,Genbank:XM_027386531.1;Name=XM_027386531.1;gbkey=mRNA;gene=Cdkn3;model_evidence=Supporting evidence includes similarity to: 5 Proteins%2C and 100%25 coverage of the annotated genomic feature by RNAseq alignments%2C including 98 samples with support for all annotated introns;product=cyclin dependent kinase inhibitor 3%2C transcript variant X1;transcript_id=XM_027386531.1</th>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>1085435</td>\n",
       "      <td>1097417</td>\n",
       "      <td>+</td>\n",
       "      <td>rna10</td>\n",
       "      <td>gene6</td>\n",
       "      <td>GeneID:100752216,Genbank:XM_027386531.1</td>\n",
       "      <td>XM_027386531.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Cdkn3</td>\n",
       "      <td>...</td>\n",
       "      <td>cyclin dependent kinase inhibitor 3%2C transcr...</td>\n",
       "      <td>XM_027386531.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11982</td>\n",
       "      <td>1085435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,Genbank:XM_027386533.1;Name=XM_027386533.1;gbkey=mRNA;gene=Cdkn3;model_evidence=Supporting evidence includes similarity to: 1 Protein%2C and 100%25 coverage of the annotated genomic feature by RNAseq alignments%2C including 84 samples with support for all annotated introns;product=cyclin dependent kinase inhibitor 3%2C transcript variant X3;transcript_id=XM_027386533.1</th>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>1085744</td>\n",
       "      <td>1097417</td>\n",
       "      <td>+</td>\n",
       "      <td>rna12</td>\n",
       "      <td>gene6</td>\n",
       "      <td>GeneID:100752216,Genbank:XM_027386533.1</td>\n",
       "      <td>XM_027386533.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Cdkn3</td>\n",
       "      <td>...</td>\n",
       "      <td>cyclin dependent kinase inhibitor 3%2C transcr...</td>\n",
       "      <td>XM_027386533.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11673</td>\n",
       "      <td>1085744</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,Genbank:XM_027386534.1;Name=XM_027386534.1;gbkey=mRNA;gene=Cnih1;model_evidence=Supporting evidence includes similarity to: 10 Proteins%2C and 100%25 coverage of the annotated genomic feature by RNAseq alignments%2C including 99 samples with support for all annotated introns;product=cornichon family AMPA receptor auxiliary protein 1%2C transcript variant X2;transcript_id=XM_027386534.1</th>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>1101897</td>\n",
       "      <td>1114413</td>\n",
       "      <td>-</td>\n",
       "      <td>rna13</td>\n",
       "      <td>gene7</td>\n",
       "      <td>GeneID:103160710,Genbank:XM_027386534.1</td>\n",
       "      <td>XM_027386534.1</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>Cnih1</td>\n",
       "      <td>...</td>\n",
       "      <td>cornichon family AMPA receptor auxiliary prote...</td>\n",
       "      <td>XM_027386534.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12516</td>\n",
       "      <td>1114413</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                               Chr    Start  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...  NW_020822366.1   717871   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...  NW_020822366.1   717871   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...  NW_020822366.1  1085435   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...  NW_020822366.1  1085744   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...  NW_020822366.1  1101897   \n",
       "\n",
       "                                                        End Strand     ID  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...   722308      -   rna7   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...   724610      -   rna6   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...  1097417      +  rna10   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...  1097417      +  rna12   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...  1114413      -  rna13   \n",
       "\n",
       "                                                   Parent  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...  gene4   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...  gene4   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...  gene6   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...  gene6   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...  gene7   \n",
       "\n",
       "                                                                                     Dbxref  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...  GeneID:100754303,Genbank:XM_027386529.1   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...  GeneID:100754303,Genbank:XM_027386528.1   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...  GeneID:100752216,Genbank:XM_027386531.1   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...  GeneID:100752216,Genbank:XM_027386533.1   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...  GeneID:103160710,Genbank:XM_027386534.1   \n",
       "\n",
       "                                                              Name gbkey  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...  XM_027386529.1  mRNA   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...  XM_027386528.1  mRNA   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...  XM_027386531.1  mRNA   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...  XM_027386533.1  mRNA   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...  XM_027386534.1  mRNA   \n",
       "\n",
       "                                                     gene  ...  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...   Bmp4  ...   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...   Bmp4  ...   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...  Cdkn3  ...   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...  Cdkn3  ...   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...  Cnih1  ...   \n",
       "\n",
       "                                                                                              product  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...  bone morphogenetic protein 4%2C transcript var...   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...  bone morphogenetic protein 4%2C transcript var...   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...  cyclin dependent kinase inhibitor 3%2C transcr...   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...  cyclin dependent kinase inhibitor 3%2C transcr...   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...  cornichon family AMPA receptor auxiliary prote...   \n",
       "\n",
       "                                                     transcript_id Note  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...  XM_027386529.1  NaN   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...  XM_027386528.1  NaN   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...  XM_027386531.1  NaN   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...  XM_027386533.1  NaN   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...  XM_027386534.1  NaN   \n",
       "\n",
       "                                                   exception end_range  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...       NaN       NaN   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...       NaN       NaN   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...       NaN       NaN   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...       NaN       NaN   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...       NaN       NaN   \n",
       "\n",
       "                                                   partial start_range  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...     NaN         NaN   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...     NaN         NaN   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...     NaN         NaN   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...     NaN         NaN   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...     NaN         NaN   \n",
       "\n",
       "                                                   inference Length  \\\n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...       NaN   4437   \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...       NaN   6739   \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...       NaN  11982   \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...       NaN  11673   \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...       NaN  12516   \n",
       "\n",
       "                                                    actual_start  \n",
       "ID=rna7;Parent=gene4;Dbxref=GeneID:100754303,Ge...        722308  \n",
       "ID=rna6;Parent=gene4;Dbxref=GeneID:100754303,Ge...        724610  \n",
       "ID=rna10;Parent=gene6;Dbxref=GeneID:100752216,G...       1085435  \n",
       "ID=rna12;Parent=gene6;Dbxref=GeneID:100752216,G...       1085744  \n",
       "ID=rna13;Parent=gene7;Dbxref=GeneID:103160710,G...       1114413  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tss = pd.read_csv(tss_annotation,sep='\\t',index_col=0)\n",
    "\n",
    "tss.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:29.497503Z",
     "start_time": "2019-03-07T06:33:25.393288Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>peaks</th>\n",
       "      <th>closest_to_X_numberOfPeaks</th>\n",
       "      <th>bin_-1000_1000</th>\n",
       "      <th>sameStrand_bin_-1000_1000</th>\n",
       "      <th>samples</th>\n",
       "      <th>minDistance</th>\n",
       "      <th>minDistancePeak</th>\n",
       "      <th>maxValue</th>\n",
       "      <th>maxPeakId</th>\n",
       "      <th>maxSample</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>maxSamples</th>\n",
       "      <th>hasGene</th>\n",
       "      <th>Number of SS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>XM_027386529.1</th>\n",
       "      <td>[1394, 3038, 17726, 21888, 24414, 35067, 69088...</td>\n",
       "      <td>101</td>\n",
       "      <td>[90264, 97971, 124234, 181215, 317724, 348823]</td>\n",
       "      <td>[90264, 181215, 348823]</td>\n",
       "      <td>[MiscOrgans_START1_and_MiscOrgans_startinput1,...</td>\n",
       "      <td>63.0</td>\n",
       "      <td>348823</td>\n",
       "      <td>59.2</td>\n",
       "      <td>348823</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>722245</td>\n",
       "      <td>722246</td>\n",
       "      <td>-</td>\n",
       "      <td>Bmp4</td>\n",
       "      <td>XM_027386529.1</td>\n",
       "      <td>[MiscOrgans_START1_and_MiscOrgans_startinput1,...</td>\n",
       "      <td>True</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XM_027386528.1</th>\n",
       "      <td>[17698, 163408, 224556, 226028, 268860, 275685...</td>\n",
       "      <td>28</td>\n",
       "      <td>[163408, 308710]</td>\n",
       "      <td>[308710]</td>\n",
       "      <td>[MiscOrgans_START1_and_MiscOrgans_startinput1,...</td>\n",
       "      <td>27.0</td>\n",
       "      <td>308710</td>\n",
       "      <td>11864.6</td>\n",
       "      <td>308710</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>724583</td>\n",
       "      <td>724584</td>\n",
       "      <td>-</td>\n",
       "      <td>Bmp4</td>\n",
       "      <td>XM_027386528.1</td>\n",
       "      <td>[MiscOrgans_START1_and_MiscOrgans_startinput1,...</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XM_027386531.1</th>\n",
       "      <td>[17679, 17707, 83014, 89129, 99005, 108459, 10...</td>\n",
       "      <td>29</td>\n",
       "      <td>[138244, 237253]</td>\n",
       "      <td>[237253]</td>\n",
       "      <td>[MiscOrgans_START1_and_MiscOrgans_startinput1,...</td>\n",
       "      <td>74.0</td>\n",
       "      <td>237253</td>\n",
       "      <td>15.3</td>\n",
       "      <td>237253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>1085509</td>\n",
       "      <td>1085510</td>\n",
       "      <td>+</td>\n",
       "      <td>Cdkn3</td>\n",
       "      <td>XM_027386531.1</td>\n",
       "      <td>[MiscOrgans_START1_and_MiscOrgans_startinput1,...</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XM_027386533.1</th>\n",
       "      <td>[68904, 137124, 249586, 270147, 335001, 400304...</td>\n",
       "      <td>7</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>inf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XM_027386534.1</th>\n",
       "      <td>[14157, 68988, 69200, 116089, 132667, 167473, ...</td>\n",
       "      <td>17</td>\n",
       "      <td>[132667, 298307, 350581, 353645]</td>\n",
       "      <td>[298307]</td>\n",
       "      <td>[MiscOrgans_START1_and_MiscOrgans_startinput1,...</td>\n",
       "      <td>88.0</td>\n",
       "      <td>298307</td>\n",
       "      <td>57.2</td>\n",
       "      <td>298307</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822366.1</td>\n",
       "      <td>1114325</td>\n",
       "      <td>1114326</td>\n",
       "      <td>-</td>\n",
       "      <td>Cnih1</td>\n",
       "      <td>XM_027386534.1</td>\n",
       "      <td>[MiscOrgans_START1_and_MiscOrgans_startinput1,...</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            peaks  \\\n",
       "XM_027386529.1  [1394, 3038, 17726, 21888, 24414, 35067, 69088...   \n",
       "XM_027386528.1  [17698, 163408, 224556, 226028, 268860, 275685...   \n",
       "XM_027386531.1  [17679, 17707, 83014, 89129, 99005, 108459, 10...   \n",
       "XM_027386533.1  [68904, 137124, 249586, 270147, 335001, 400304...   \n",
       "XM_027386534.1  [14157, 68988, 69200, 116089, 132667, 167473, ...   \n",
       "\n",
       "                closest_to_X_numberOfPeaks  \\\n",
       "XM_027386529.1                         101   \n",
       "XM_027386528.1                          28   \n",
       "XM_027386531.1                          29   \n",
       "XM_027386533.1                           7   \n",
       "XM_027386534.1                          17   \n",
       "\n",
       "                                                bin_-1000_1000  \\\n",
       "XM_027386529.1  [90264, 97971, 124234, 181215, 317724, 348823]   \n",
       "XM_027386528.1                                [163408, 308710]   \n",
       "XM_027386531.1                                [138244, 237253]   \n",
       "XM_027386533.1                                              []   \n",
       "XM_027386534.1                [132667, 298307, 350581, 353645]   \n",
       "\n",
       "               sameStrand_bin_-1000_1000  \\\n",
       "XM_027386529.1   [90264, 181215, 348823]   \n",
       "XM_027386528.1                  [308710]   \n",
       "XM_027386531.1                  [237253]   \n",
       "XM_027386533.1                        []   \n",
       "XM_027386534.1                  [298307]   \n",
       "\n",
       "                                                          samples  \\\n",
       "XM_027386529.1  [MiscOrgans_START1_and_MiscOrgans_startinput1,...   \n",
       "XM_027386528.1  [MiscOrgans_START1_and_MiscOrgans_startinput1,...   \n",
       "XM_027386531.1  [MiscOrgans_START1_and_MiscOrgans_startinput1,...   \n",
       "XM_027386533.1                                                 []   \n",
       "XM_027386534.1  [MiscOrgans_START1_and_MiscOrgans_startinput1,...   \n",
       "\n",
       "                minDistance minDistancePeak maxValue maxPeakId maxSample  \\\n",
       "XM_027386529.1         63.0          348823     59.2    348823       NaN   \n",
       "XM_027386528.1         27.0          308710  11864.6    308710       NaN   \n",
       "XM_027386531.1         74.0          237253     15.3    237253       NaN   \n",
       "XM_027386533.1          inf             NaN      NaN       NaN       NaN   \n",
       "XM_027386534.1         88.0          298307     57.2    298307       NaN   \n",
       "\n",
       "                           Chr    Start      End Strand   gene  \\\n",
       "XM_027386529.1  NW_020822366.1   722245   722246      -   Bmp4   \n",
       "XM_027386528.1  NW_020822366.1   724583   724584      -   Bmp4   \n",
       "XM_027386531.1  NW_020822366.1  1085509  1085510      +  Cdkn3   \n",
       "XM_027386533.1             NaN      NaN      NaN    NaN    NaN   \n",
       "XM_027386534.1  NW_020822366.1  1114325  1114326      -  Cnih1   \n",
       "\n",
       "                    transcript  \\\n",
       "XM_027386529.1  XM_027386529.1   \n",
       "XM_027386528.1  XM_027386528.1   \n",
       "XM_027386531.1  XM_027386531.1   \n",
       "XM_027386533.1             NaN   \n",
       "XM_027386534.1  XM_027386534.1   \n",
       "\n",
       "                                                       maxSamples  hasGene  \\\n",
       "XM_027386529.1  [MiscOrgans_START1_and_MiscOrgans_startinput1,...     True   \n",
       "XM_027386528.1  [MiscOrgans_START1_and_MiscOrgans_startinput1,...     True   \n",
       "XM_027386531.1  [MiscOrgans_START1_and_MiscOrgans_startinput1,...     True   \n",
       "XM_027386533.1                                                 []    False   \n",
       "XM_027386534.1  [MiscOrgans_START1_and_MiscOrgans_startinput1,...     True   \n",
       "\n",
       "                Number of SS  \n",
       "XM_027386529.1             3  \n",
       "XM_027386528.1             1  \n",
       "XM_027386531.1             1  \n",
       "XM_027386533.1             0  \n",
       "XM_027386534.1             1  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "txn_df = pickle.load(open('Results/tss_annotation/txn_df_02.p','rb'))# pd.read_csv('Results/txn_df.csv',index_col=0)\n",
    "txn_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge tissue columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:29.556636Z",
     "start_time": "2019-03-07T06:33:29.502273Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def merge_two_columns(df, mapping_dict):\n",
    "    '''Function that merges columns by taking the mean of them.\n",
    "    Merges based on if they have the same element in the meta_samples meta_column.\n",
    "    Returns:\n",
    "        new_df: Dataframe but with the columns of interest merged. Also column names are now based\n",
    "                on the unique meta_samples[meta_column].\n",
    "    '''\n",
    "\n",
    "    vals = mapping_dict.keys()#np.unique(meta_samples[meta_column].values)\n",
    "    new_df = pd.DataFrame(index=df.index, columns=vals)\n",
    "    for i in vals:\n",
    "        if not mapping_dict[i] == []:#(meta_samples[meta_column] == i).any():\n",
    "            new_col = (df.loc[:, mapping_dict[i]])\n",
    "            new_col = new_col.mean(axis=1)\n",
    "            new_df.loc[:, i] = new_col\n",
    "    new_df = new_df.loc[:, ~(new_df.isnull().all())]\n",
    "    return new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:29.594414Z",
     "start_time": "2019-03-07T06:33:29.559984Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tissue_to_peaks = defaultdict(list)\n",
    "for val in peaks_tissue.columns.values:\n",
    "    count = 0\n",
    "    for t in np.unique(meta_samples[\"Tissue\"].values):\n",
    "        if t in val:\n",
    "            count += 1\n",
    "            tissue_to_peaks[t].append(val)\n",
    "            if count > 1:\n",
    "                print(\"This sample has multiple tissues associated with it\")\n",
    "                print(\"Sample: \",val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:32.261720Z",
     "start_time": "2019-03-07T06:33:29.597006Z"
    },
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "peaks_tissue_merged = merge_two_columns(peaks_tissue,mapping_dict=tissue_to_peaks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:32.329732Z",
     "start_time": "2019-03-07T06:33:32.264297Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Heart</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>MiscOrgans</th>\n",
       "      <th>FemaleReproductive</th>\n",
       "      <th>CHO</th>\n",
       "      <th>BMDM1hKLA</th>\n",
       "      <th>Brain</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Liver</th>\n",
       "      <th>BMDMwt</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Kidney</th>\n",
       "      <th>Pancreas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.835</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.630</td>\n",
       "      <td>26.996</td>\n",
       "      <td>84.2</td>\n",
       "      <td>0.790000</td>\n",
       "      <td>2.180000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.715</td>\n",
       "      <td>2.130000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.000</td>\n",
       "      <td>3.385</td>\n",
       "      <td>3.975</td>\n",
       "      <td>4.265</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.608571</td>\n",
       "      <td>16.596667</td>\n",
       "      <td>1.1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.545</td>\n",
       "      <td>1.050000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.255</td>\n",
       "      <td>0.705</td>\n",
       "      <td>4.025</td>\n",
       "      <td>19.044</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.887429</td>\n",
       "      <td>2.386667</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.66</td>\n",
       "      <td>1.100</td>\n",
       "      <td>1.870000</td>\n",
       "      <td>1.490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.835</td>\n",
       "      <td>2.875</td>\n",
       "      <td>3.025</td>\n",
       "      <td>6.000</td>\n",
       "      <td>1.680</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.914286</td>\n",
       "      <td>1.633333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.45</td>\n",
       "      <td>1.495</td>\n",
       "      <td>1.153333</td>\n",
       "      <td>2.385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>25.600</td>\n",
       "      <td>88.550</td>\n",
       "      <td>112.050</td>\n",
       "      <td>136.550</td>\n",
       "      <td>20.292</td>\n",
       "      <td>0.0</td>\n",
       "      <td>189.308571</td>\n",
       "      <td>50.966667</td>\n",
       "      <td>29.7</td>\n",
       "      <td>3.39</td>\n",
       "      <td>124.150</td>\n",
       "      <td>43.640000</td>\n",
       "      <td>95.500</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Heart  Spleen  MiscOrgans  FemaleReproductive     CHO  BMDM1hKLA  \\\n",
       "0   0.835   0.000       0.705               0.630  26.996       84.2   \n",
       "1   0.000   3.385       3.975               4.265   0.000        0.0   \n",
       "2   0.000   1.255       0.705               4.025  19.044        0.0   \n",
       "3   0.835   2.875       3.025               6.000   1.680        0.0   \n",
       "4  25.600  88.550     112.050             136.550  20.292        0.0   \n",
       "\n",
       "        Brain       Lung  Liver  BMDMwt   Muscle     Kidney  Pancreas  \n",
       "0    0.790000   2.180000    0.0    0.00    0.715   2.130000     0.000  \n",
       "1    1.608571  16.596667    1.1    0.00    0.545   1.050000     0.000  \n",
       "2    1.887429   2.386667    0.0    5.66    1.100   1.870000     1.490  \n",
       "3    3.914286   1.633333    0.0    2.45    1.495   1.153333     2.385  \n",
       "4  189.308571  50.966667   29.7    3.39  124.150  43.640000    95.500  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peaks_tissue_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create directory of different peak groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:32.359236Z",
     "start_time": "2019-03-07T06:33:32.331614Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "if not os.path.exists(\"Results/peak_groups\"):\n",
    "    os.mkdir(\"Results/peak_groups\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:32.388118Z",
     "start_time": "2019-03-07T06:33:32.361013Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def save_peak_as_bed(peaks_df,f_save, name='index',score_col = 'Peak Score',is_sort = True):\n",
    "    ## Assumes there is Chr,Start,End,Name is index\n",
    "    ## Peak file is 1-index, bed files are 0, so need to shift everything by one\n",
    "    \n",
    "    if name == 'index':\n",
    "        peaks_df.loc[:,'Name'] = peaks_df.index\n",
    "    else:\n",
    "        peaks_df.loc[:,'Name'] = peaks_df[name]\n",
    "    \n",
    "    if score_col == '': #If no scoring col, set to 0\n",
    "        peaks_df.loc[:,'Score'] = 0\n",
    "        score_col = 'Score'\n",
    "        \n",
    "    peaks_df.loc[:,'Start'] = peaks_df['Start'] - 1\n",
    "    peaks_df.loc[:,'End'] = peaks_df['End'] - 1\n",
    "    \n",
    "    peaks_df = peaks_df[['Chr','Start','End','Name',score_col,'Strand']]\n",
    "    \n",
    "    if is_sort:\n",
    "        peaks_df.sort_values(['Chr','Start'],inplace=True)\n",
    "        \n",
    "    if not f_save.endswith('.bed'):\n",
    "        f_save = f_save + '.bed'\n",
    "    peaks_df.to_csv(f_save ,sep='\\t',index=False,header=None)\n",
    "    return\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## a) Separate confident promoters and confident ncRNA \n",
    "Define by distance to RefSeq being 0 and ncRNA far >40kb from a gene and in at least two tissues\n",
    "\n",
    "### c) Predicted promoters vs RefSeq promoters\n",
    "\n",
    "### d) Promoters organized based on distance to refseq (so heatmap is not sorted by heat but by distance)\n",
    "\n",
    "### e) Only look at peaks specific for each tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:51.450899Z",
     "start_time": "2019-03-07T06:33:51.390734Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def filter_only_with_divergent(df,divergent_threshold = 500 ):\n",
    "    inds_to_keep = []\n",
    "    for ind,val in df[df['isSameStrand']].iterrows():\n",
    "        curr = df[(val['Chr'] == df['Chr']) & (df['isSameStrand'] == False)]\n",
    "        if val['Strand'] == '-':\n",
    "            curr = curr[(val['End'] - curr['Start'] < divergent_threshold) & \n",
    "                       (val['End'] - curr['Start'] > 0)]\n",
    "        else: \n",
    "            curr = curr[(val['Start'] - curr['End'] < divergent_threshold) &\n",
    "                       (val['Start'] - curr['End'] > 0 ) ]\n",
    "\n",
    "        ## If there are divergent peaks, add this peak\n",
    "        if len(curr) > 0: \n",
    "            inds_to_keep.append(ind)\n",
    "    \n",
    "    df = df.loc[inds_to_keep]\n",
    "    return df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T22:36:31.003762Z",
     "start_time": "2019-03-07T22:36:30.875557Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def tissue_specific_promoter(t,\n",
    "                             anno_peaks,\n",
    "                             peaks_tissue_merged,\n",
    "                             txn_df,tss,\n",
    "                             peak_count_thresh = 1, \n",
    "                             prom_bin=[-1000, 1000],\n",
    "                             enhancer_d=40000,\n",
    "                             divergent_thr=500,\n",
    "                             outfolder='Results/peak_groups/', \n",
    "                             has_cs=True):\n",
    "\n",
    "    ''' Function to create bed files relevant for a specific tissue\n",
    "        Will save:\n",
    "        a) All peaks that were found in tissue\n",
    "        b) Peaks within prom_bin distance from reference & has divergent peak nearby (divergent_thr distance)\n",
    "        c) Same as b, but no divergent restriction\n",
    "        d) Same as c, but the peak center is now the annnotated start site instead of experimental\n",
    "        e) ncRNA, possibly eRNA that are greater than enhancer_d from any annotated start sites.\n",
    "        f) no_peaks: genes with no peaks nearby, even divergent peaks'''\n",
    "    \n",
    "    \n",
    "    \n",
    "    t_peaks = peaks_tissue_merged[peaks_tissue_merged.loc[:, t] > peak_count_thresh].index #Peaks in that tissue\n",
    "    t_peaks = anno_peaks.loc[t_peaks]\n",
    "    t_peaks = t_peaks[(~(t_peaks['Distance to TSS'] == np.infty))] ## Remove peaks that are on a scaffold with no gene\n",
    "\n",
    "    ### Absence of peaks for that tissue.\n",
    "    no_t_peaks = txn_df[txn_df['samples'].apply(lambda x: np.array(map(lambda y: t not in y,x)).any())]\n",
    "    no_t_peaks = tss[tss['transcript_id'].apply(lambda x: x in no_t_peaks.index)]\n",
    "    #no_t_peaks = peaks_tissue_merged[peaks_tissue_merged.loc[:, t] <= peak_count_thresh].index #Peaks not in the tissue\n",
    "    #no_t_peaks = anno_peaks.loc[no_t_peaks]\n",
    "    \n",
    "    t_prom = t_peaks[(t_peaks['Distance to TSS'] > prom_bin[0])\n",
    "                     & (t_peaks['Distance to TSS'] < prom_bin[1])]\n",
    "    t_prom_isSame = t_prom[t_prom['isSameStrand']]\n",
    "    \n",
    "    \n",
    "    t_enh = t_peaks[np.abs(t_peaks['Distance to TSS']) > enhancer_d]\n",
    "    t_prom_div = filter_only_with_divergent(\n",
    "        t_prom, divergent_threshold=divergent_thr)\n",
    "    t_enh_div = filter_only_with_divergent(\n",
    "        t_enh, divergent_threshold=divergent_thr)\n",
    "\n",
    "    t_prom_div = t_prom_div.sort_values(['Distance to TSS'])\n",
    "    \n",
    "    #Look at RefSeq vs experimental TSS\n",
    "    tss_withPeaks = t_prom_isSame['Nearest TSS'].unique()\n",
    "    refSeq = tss[tss['transcript_id'].apply(lambda x: x in tss_withPeaks)] \n",
    "    # Dont want the whole gene length, just the start of the peak \n",
    "    refSeq['Start'] = refSeq['actual_start'] #actual_start takes care of +/- strand\n",
    "    refSeq['End'] = refSeq['actual_start']\n",
    "    \n",
    "    ## Saving bed files\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    if not os.path.exists(out_dir):\n",
    "        os.mkdir(out_dir)\n",
    "\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_peaks,\n",
    "        os.path.join(out_dir, 'all_peaks'),\n",
    "        name='index',\n",
    "        score_col='Stat')\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_prom_div,\n",
    "        os.path.join(out_dir, 'confident_promoters'),\n",
    "        name='index',\n",
    "        score_col='Stat')\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_prom_isSame,\n",
    "        os.path.join(out_dir, 'confident_promoters_noDiv'),\n",
    "        name='index',\n",
    "        score_col='Stat')  \n",
    "    \n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_enh_div,\n",
    "        os.path.join(out_dir, 'confident_enhancers'),\n",
    "        name='index',\n",
    "        score_col='Stat')\n",
    "    \n",
    "\n",
    "    save_peak_as_bed(\n",
    "        refSeq,\n",
    "        os.path.join(out_dir, 'confident_promoters_noDiv_PICR'),\n",
    "        name='index')\n",
    "    save_peak_as_bed(\n",
    "        no_t_peaks,\n",
    "        os.path.join(out_dir, 'no_peaks'),\n",
    "        name='index')\n",
    "\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T06:33:57.886125Z",
     "start_time": "2019-03-07T06:33:57.852981Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tissues_with_atac = ['CHO','Brain','Liver','Lung','Kidney','Spleen','BMDMwt']#,'BMDM1hKLA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T22:32:04.922167Z",
     "start_time": "2019-03-07T22:32:04.831240Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "      <th>Nearest gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NW_020822452.1</td>\n",
       "      <td>16590187</td>\n",
       "      <td>16590584</td>\n",
       "      <td>+</td>\n",
       "      <td>16.227333</td>\n",
       "      <td>promoter-TSS (ID=id181373;Parent=rna15526;Dbxr...</td>\n",
       "      <td>XM_027399668.1</td>\n",
       "      <td>-462.0</td>\n",
       "      <td>False</td>\n",
       "      <td>Clic4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NW_020822597.1</td>\n",
       "      <td>23595304</td>\n",
       "      <td>23595489</td>\n",
       "      <td>+</td>\n",
       "      <td>5.740000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027420351.1</td>\n",
       "      <td>-23850.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Baz2b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NW_020822680.1</td>\n",
       "      <td>728607</td>\n",
       "      <td>728829</td>\n",
       "      <td>+</td>\n",
       "      <td>7.579579</td>\n",
       "      <td>promoter-TSS (ID=id609942;Parent=rna53731;Dbxr...</td>\n",
       "      <td>XM_027432478.1</td>\n",
       "      <td>746.0</td>\n",
       "      <td>False</td>\n",
       "      <td>Irak1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NW_020822602.1</td>\n",
       "      <td>10883074</td>\n",
       "      <td>10883337</td>\n",
       "      <td>+</td>\n",
       "      <td>3.992000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027423054.1</td>\n",
       "      <td>9079.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Gbgt1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>37427816</td>\n",
       "      <td>37427999</td>\n",
       "      <td>+</td>\n",
       "      <td>129.031738</td>\n",
       "      <td>promoter-TSS (ID=id474643;Parent=rna41513;Dbxr...</td>\n",
       "      <td>XM_027421990.1</td>\n",
       "      <td>-11.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Shf</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Chr     Start       End Strand        Stat  \\\n",
       "ID                                                          \n",
       "0   NW_020822452.1  16590187  16590584      +   16.227333   \n",
       "1   NW_020822597.1  23595304  23595489      +    5.740000   \n",
       "2   NW_020822680.1    728607    728829      +    7.579579   \n",
       "3   NW_020822602.1  10883074  10883337      +    3.992000   \n",
       "4   NW_020822601.1  37427816  37427999      +  129.031738   \n",
       "\n",
       "                                           Annotation     Nearest TSS  \\\n",
       "ID                                                                      \n",
       "0   promoter-TSS (ID=id181373;Parent=rna15526;Dbxr...  XM_027399668.1   \n",
       "1                                          Intergenic  XM_027420351.1   \n",
       "2   promoter-TSS (ID=id609942;Parent=rna53731;Dbxr...  XM_027432478.1   \n",
       "3                                          Intergenic  XM_027423054.1   \n",
       "4   promoter-TSS (ID=id474643;Parent=rna41513;Dbxr...  XM_027421990.1   \n",
       "\n",
       "    Distance to TSS  isSameStrand Nearest gene  \n",
       "ID                                              \n",
       "0            -462.0         False        Clic4  \n",
       "1          -23850.0          True        Baz2b  \n",
       "2             746.0         False        Irak1  \n",
       "3            9079.0          True        Gbgt1  \n",
       "4             -11.0          True          Shf  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno_peaks.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-03-07T22:38:32.875535Z",
     "start_time": "2019-03-07T22:36:33.461505Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:52: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:53: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "\"['Peak Score'] not in index\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-21-0caf01e6c2de>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mu'time'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34mu''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34mu'for t in tissues_with_atac:\\n    print(t)\\n    tissue_specific_promoter(\\n        t=t,\\n        anno_peaks=anno_peaks,\\n        peaks_tissue_merged=peaks_tissue_merged,\\n        txn_df=txn_df,\\n        tss=tss, has_cs = False)'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.pyc\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2113\u001b[0m             \u001b[0mmagic_arg_s\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvar_expand\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstack_depth\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2114\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2115\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2116\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2117\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<decorator-gen-59>\u001b[0m in \u001b[0;36mtime\u001b[0;34m(self, line, cell, local_ns)\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/IPython/core/magic.pyc\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    187\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 188\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    189\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    190\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/IPython/core/magics/execution.pyc\u001b[0m in \u001b[0;36mtime\u001b[0;34m(self, line, cell, local_ns)\u001b[0m\n\u001b[1;32m   1183\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1184\u001b[0m             \u001b[0mst\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclock2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1185\u001b[0;31m             \u001b[0;32mexec\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mglob\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlocal_ns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1186\u001b[0m             \u001b[0mend\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclock2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1187\u001b[0m             \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m<ipython-input-20-579f5f43c267>\u001b[0m in \u001b[0;36mtissue_specific_promoter\u001b[0;34m(t, anno_peaks, peaks_tissue_merged, txn_df, tss, peak_count_thresh, prom_bin, enhancer_d, divergent_thr, outfolder, has_cs)\u001b[0m\n\u001b[1;32m     88\u001b[0m         \u001b[0mrefSeq\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     89\u001b[0m         \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mout_dir\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'confident_promoters_noDiv_PICR'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 90\u001b[0;31m         name='index')\n\u001b[0m\u001b[1;32m     91\u001b[0m     save_peak_as_bed(\n\u001b[1;32m     92\u001b[0m         \u001b[0mno_t_peaks\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-14-d1ca41aa5cd1>\u001b[0m in \u001b[0;36msave_peak_as_bed\u001b[0;34m(peaks_df, f_save, name, score_col, is_sort)\u001b[0m\n\u001b[1;32m     15\u001b[0m     \u001b[0mpeaks_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'End'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpeaks_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'End'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     16\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 17\u001b[0;31m     \u001b[0mpeaks_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpeaks_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Chr'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'Start'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'End'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'Name'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mscore_col\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'Strand'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     18\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     19\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mis_sort\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/frame.pyc\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2932\u001b[0m                 \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2933\u001b[0m             indexer = self.loc._convert_to_indexer(key, axis=1,\n\u001b[0;32m-> 2934\u001b[0;31m                                                    raise_missing=True)\n\u001b[0m\u001b[1;32m   2935\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2936\u001b[0m         \u001b[0;31m# take() does not accept boolean indexers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/indexing.pyc\u001b[0m in \u001b[0;36m_convert_to_indexer\u001b[0;34m(self, obj, axis, is_setter, raise_missing)\u001b[0m\n\u001b[1;32m   1352\u001b[0m                 kwargs = {'raise_missing': True if is_setter else\n\u001b[1;32m   1353\u001b[0m                           raise_missing}\n\u001b[0;32m-> 1354\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_listlike_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1355\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1356\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/indexing.pyc\u001b[0m in \u001b[0;36m_get_listlike_indexer\u001b[0;34m(self, key, axis, raise_missing)\u001b[0m\n\u001b[1;32m   1159\u001b[0m         self._validate_read_indexer(keyarr, indexer,\n\u001b[1;32m   1160\u001b[0m                                     \u001b[0mo\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_axis_number\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1161\u001b[0;31m                                     raise_missing=raise_missing)\n\u001b[0m\u001b[1;32m   1162\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mkeyarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1163\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/indexing.pyc\u001b[0m in \u001b[0;36m_validate_read_indexer\u001b[0;34m(self, key, indexer, axis, raise_missing)\u001b[0m\n\u001b[1;32m   1250\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'loc'\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mraise_missing\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1251\u001b[0m                 \u001b[0mnot_found\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0max\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1252\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"{} not in index\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnot_found\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1253\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1254\u001b[0m             \u001b[0;31m# we skip the warning on Categorical/Interval\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: \"['Peak Score'] not in index\""
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    tissue_specific_promoter(\n",
    "        t=t,\n",
    "        anno_peaks=anno_peaks,\n",
    "        peaks_tissue_merged=peaks_tissue_merged,\n",
    "        txn_df=txn_df,\n",
    "        tss=tss, has_cs = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TSS peaks near ATAC peaks. \n",
    "Using Bedtools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sort ATAC bed files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.628Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Sort ATAC bed files\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    atac_f = glob.glob(os.path.join(data_folder, t, 'ATAC/out/peak/macs2/overlap/optimal_set/*narrowPeak.gz'))[0]\n",
    "    atac_f_out = atac_f.replace('.gz','') + '.sort'\n",
    "    atac_f_out = os.path.join(\"Results/ATAC_results/\",t,os.path.basename(atac_f).replace('.gz','.sort'))\n",
    "    if not os.path.exists(os.path.join(\"Results/ATAC_results/\",t)):\n",
    "        os.mkdir(os.path.join(\"Results/ATAC_results/\",t))\n",
    "    \n",
    "    cmd = 'bedtools sort -i %s > %s' % (atac_f,atac_f_out)   \n",
    "    print(cmd)\n",
    "    os.system(cmd)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get peaks close to TSS within distance of 500 bp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.632Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def peaks_closest(f_a, f_b, out_f, dist_thresh):\n",
    "    '''Function that calls bedtools closest to find the closest peak of two bed files, and awk to filter those within distance dist_thresh'''\n",
    "\n",
    "    cmd = \"bedtools closest -d -t first -a %s -b %s | awk '{if ($NF < %d) print }' > %s\" % (\n",
    "       f_a, f_b, dist_thresh,out_f) #t is to only report one of the atac peaks\n",
    "    \n",
    "#     cmd = \"bedtools closest -d -t first -a %s -b %s  > %s\" % (\n",
    "#         f_a, f_b,out_f)\n",
    "    \n",
    "    print(cmd)\n",
    "    os.system(cmd)\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.633Z"
    },
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Intersect ATAC and TSS\n",
    "dist_thresh = 500 #500 basepairs left or right\n",
    "outfolder = 'Results/peak_groups/'\n",
    "f_names = [\n",
    "    'all_peaks','confident_promoters', 'confident_enhancers', 'confident_promoters_noDiv',\n",
    "    'confident_promoters_noDiv_PICR', 'no_peaks'\n",
    "]\n",
    "\n",
    "\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    ## Outdirectory to save in\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    \n",
    "    #ATAC peak file\n",
    "    atac_out = glob.glob(os.path.join(\"Results/ATAC_results\", t , '*sort'))[0]\n",
    "    \n",
    "    ## The different TSS files\n",
    "    for f in f_names:\n",
    "        f_in = os.path.join(out_dir, f) + '.bed'\n",
    "        out_f = '%s_atac_overlap_%dbp.bed' % (f_in,dist_thresh)\n",
    "        peaks_closest(f_in,atac_out,out_f,dist_thresh=dist_thresh)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.634Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ls Results/peak_groups/Brain/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.635Z"
    },
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Intersect ATAC and TSS\n",
    "dist_thresh = 100 #500 basepairs left or right\n",
    "outfolder = 'Results/peak_groups/'\n",
    "f_names = [\n",
    "    'all_peaks','confident_promoters', 'confident_enhancers', 'confident_promoters_noDiv',\n",
    "    'confident_promoters_noDiv_PICR', 'no_peaks'\n",
    "]\n",
    "\n",
    "\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    ## Outdirectory to save in\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    \n",
    "    #ATAC peak file\n",
    "    atac_out = glob.glob(os.path.join(\"Results/ATAC_results\", t , '*sort'))[0]\n",
    "    \n",
    "    ## The different TSS files\n",
    "    for f in f_names:\n",
    "        f_in = os.path.join(out_dir, f) + '.bed'\n",
    "        out_f = '%s_atac_overlap_%dbp.bed' % (f_in,dist_thresh)\n",
    "        peaks_closest(f_in,atac_out,out_f,dist_thresh=dist_thresh)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Use wc -l to determine the number of peaks that are near ATAC, compared to total\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.638Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from subprocess import check_output\n",
    "\n",
    "def wc(filename):\n",
    "    return int(check_output([\"wc\", \"-l\", filename]).split()[0])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.640Z"
    },
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "dist_thresholds = [100,500]\n",
    "atac_tss_df = pd.DataFrame()\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    ## The different TSS files\n",
    "    for f in f_names:\n",
    "        f_in = os.path.join(out_dir, f) + '.bed'\n",
    "        \n",
    "        for d in dist_thresholds:\n",
    "            out_f = '%s_atac_overlap_%dbp.bed' % (f_in,d)\n",
    "            print(f_in)\n",
    "            pre_filt = wc(f_in)\n",
    "            print(out_f)\n",
    "            post_filt = wc(out_f)\n",
    "            name = os.path.basename(f_in)\n",
    "            frac = 1.0*post_filt/pre_filt\n",
    "            atac_tss_df.at[t,name] = pre_filt\n",
    "            name_atac = '_with_ATAC_%d' % d\n",
    "            ## Add to dataframe\n",
    "            \n",
    "            atac_tss_df.at[t,name + name_atac] = post_filt\n",
    "            atac_tss_df.at[t,name + name_atac + '_fraction'] = frac\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.641Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "atac_tss_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.642Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "col_names = np.array(atac_tss_df.columns.values)\n",
    "#to_drop = col_names[map(lambda x: 'fraction' not in x or '100' not in x,atac_tss_df.columns.values)]\n",
    "#output = atac_tss_df.drop(to_drop,axis=1) \n",
    "output = atac_tss_df.copy()\n",
    "col_names = np.array(output.columns.values)\n",
    "for i,name in enumerate((col_names)):\n",
    "    name = name.replace('.bed','')\n",
    "    #name = name.replace('fraction','')\n",
    "    name = name.replace('ATAC','ATAC peak within')\n",
    "    #name = 'Fraction of ' + name\n",
    "    #name = name + 'bp'\n",
    "    col_names[i] = name.replace('_',' ')\n",
    "output.columns = col_names\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-02-07T02:24:12.643Z"
    },
    "collapsed": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "output.transpose()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---------"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
