{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1. Environment Setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-14T10:27:13.500701Z",
     "start_time": "2019-02-14T10:27:12.758457Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Number of tissues: ', 13)\n"
     ]
    }
   ],
   "source": [
    "## Parameters specific to where your folders are and your data\n",
    "parameter_file = 'params/params.yaml'\n",
    "import yaml\n",
    "import sys\n",
    "\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "#p = dic2obj(**doc)\n",
    "\n",
    "data_folder = doc['data_folder']\n",
    "tissues = doc['tissues'].split(',')\n",
    "sys.path.append(doc['pipeline_path'])\n",
    "ref_fa = doc['ref_fa']\n",
    "anno_gff=doc['annotation']\n",
    "mRNA_peak_file = doc[\"mRNA_peak_file\"]\n",
    "mRNA_peak_150_file = doc[\"mRNA_peak_150_file\"]\n",
    "supplemental = doc[\"supplemental\"]\n",
    "tss_annotation = doc['tss_annotation']\n",
    "\n",
    "import os\n",
    "os.chdir\n",
    "import sys\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import pickle\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "from matplotlib import pyplot as plt\n",
    "from itertools import product\n",
    "import glob\n",
    "import re\n",
    "from matplotlib_venn import venn2\n",
    "from matplotlib import rcParams\n",
    "import inspect\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "sys.setrecursionlimit(3000)\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "rcParams['figure.figsize'] = 8, 6\n",
    "import tqdm\n",
    "import time\n",
    "\n",
    "from os.path import basename\n",
    "##mpl.use('Agg')\n",
    "#mpl.style.use('ggplot')\n",
    "#mpl.style.use('fivethirtyeight')\n",
    "from Homer import *\n",
    "\n",
    "print('Number of tissues: ',len(tissues))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-14T10:27:16.041772Z",
     "start_time": "2019-02-14T10:27:16.018226Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from create_output import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-14T10:27:16.062850Z",
     "start_time": "2019-02-14T10:27:16.043811Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sys.path.append(\"/home/isshamie/software/homebrew/parallel_functions/\")\n",
    "import parallel_functions as pf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-14T10:27:16.274711Z",
     "start_time": "2019-02-14T10:27:16.064774Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import motif_analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-14T10:27:16.316574Z",
     "start_time": "2019-02-14T10:27:16.276649Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['scl',\n",
       " 'cdx2',\n",
       " 'fli',\n",
       " 'pu1',\n",
       " 'elf3',\n",
       " 'klf10',\n",
       " 'sp5',\n",
       " 'oct2',\n",
       " 'ebna1',\n",
       " 'phox2a',\n",
       " 'hinfp',\n",
       " 'bcl6',\n",
       " 'smad4',\n",
       " 'pknox1',\n",
       " 'gata3',\n",
       " 'myf5',\n",
       " 'staf',\n",
       " 'pbx2',\n",
       " 'tbet',\n",
       " 'znf528',\n",
       " 'pax7-longest',\n",
       " 'spdef',\n",
       " 'tra',\n",
       " 'klf14',\n",
       " 'nrf',\n",
       " 'p53-myc',\n",
       " 'gata3',\n",
       " 'sp2',\n",
       " 'foxa1',\n",
       " 'hif2a',\n",
       " 'vdr',\n",
       " 'znf264',\n",
       " 'gabp',\n",
       " 'tead',\n",
       " 'irf1',\n",
       " 'sf1',\n",
       " 'znf136',\n",
       " 'p65',\n",
       " 'klf6',\n",
       " 'stat6',\n",
       " 'tcf7l2',\n",
       " 'zscan22',\n",
       " 'six4',\n",
       " 'gfi1b',\n",
       " 'foxl2',\n",
       " 'tr4',\n",
       " 'gata3',\n",
       " 'mef2d',\n",
       " 'grhl2',\n",
       " 'ptf1a',\n",
       " 'znf322',\n",
       " 'lxre',\n",
       " 'ar-half',\n",
       " 'npas',\n",
       " 'fosl2',\n",
       " 'arnt',\n",
       " 'gata3',\n",
       " 'lhx2',\n",
       " 'nfat-ap1',\n",
       " 'ets-runx',\n",
       " 'erg',\n",
       " 'meis1',\n",
       " 'tbx20',\n",
       " 'znf675',\n",
       " 'hoxd12',\n",
       " 'nfatc1',\n",
       " 'rfx2',\n",
       " 'znf415',\n",
       " 'hnf1b',\n",
       " 'e2f6',\n",
       " 'usf2',\n",
       " 'foxo3',\n",
       " 'pit1-long',\n",
       " 'ews-fli1',\n",
       " 'oct',\n",
       " 'ehf',\n",
       " 'are-fox',\n",
       " 'pax3-fkhr',\n",
       " 'jun-cre',\n",
       " 'sox3',\n",
       " 'limb',\n",
       " 'runx1m',\n",
       " 'tfe3',\n",
       " 'stat1',\n",
       " 'couptf2',\n",
       " 'heb',\n",
       " 'zfp809',\n",
       " 'nfy',\n",
       " 'gata2',\n",
       " 'p53',\n",
       " 'zfp281',\n",
       " 'brn2',\n",
       " 'jun-ap1',\n",
       " 'elf1',\n",
       " 'rxr',\n",
       " 'atf7',\n",
       " 'zic3',\n",
       " 'foxa2',\n",
       " 'pax5',\n",
       " 'pax7-long',\n",
       " 'znf416',\n",
       " 'boris',\n",
       " 'isre',\n",
       " 'reverb',\n",
       " 'cebp-cebp',\n",
       " 'hoxb13',\n",
       " 'dmrt1',\n",
       " 'zbtb12',\n",
       " 'pax8',\n",
       " 'max',\n",
       " 'hnf6',\n",
       " 'rfx6',\n",
       " 'nrf2',\n",
       " 'bapx1',\n",
       " 'pbx1',\n",
       " 'e47',\n",
       " 'zkscan1',\n",
       " 'tata',\n",
       " 'znf519',\n",
       " 'tcf4',\n",
       " 'dmc1',\n",
       " 'znf16',\n",
       " 'lrh1',\n",
       " 'pparg',\n",
       " 'cebp-ap1',\n",
       " 'bzip-irf',\n",
       " 'dr5',\n",
       " 'srebp1a',\n",
       " 'egr2',\n",
       " 'jund',\n",
       " 'dux4',\n",
       " 'six2',\n",
       " 'cmyc',\n",
       " 'chr',\n",
       " 'sp1',\n",
       " 'lef1',\n",
       " 'mef2a',\n",
       " 'oct-oct',\n",
       " 'gre-raw',\n",
       " 'bach1',\n",
       " 'ets-ebox',\n",
       " 'gre',\n",
       " 'znf263',\n",
       " 'hoxd10',\n",
       " 'irf3',\n",
       " 'bhlhe40',\n",
       " 'znf669',\n",
       " 'klf9',\n",
       " 'oct4-sox17',\n",
       " 'mafk',\n",
       " 'ets',\n",
       " 'tcfcp2l1',\n",
       " 'rbpj1-ebox',\n",
       " 'pbx3',\n",
       " 'gfy',\n",
       " 'pit1',\n",
       " 'mef2b',\n",
       " 'nf1',\n",
       " 'tbx21',\n",
       " 'gata3',\n",
       " 'pax5-short',\n",
       " 'tcf12',\n",
       " 'elk4',\n",
       " 'nanog',\n",
       " 'runx1',\n",
       " 'etv2',\n",
       " 'klf3',\n",
       " 'sox17',\n",
       " 'znf711',\n",
       " 'hsf',\n",
       " 'foxk2',\n",
       " 'erra',\n",
       " 'tgif2',\n",
       " 'nkx2',\n",
       " 'crx',\n",
       " 'ascl1',\n",
       " 'atoh1',\n",
       " 'nmyc',\n",
       " 'ap2',\n",
       " 'bmal',\n",
       " 'tgif1',\n",
       " 'ews-erg',\n",
       " 'stat3',\n",
       " 'prdm1',\n",
       " 'myba',\n",
       " 'couptf1',\n",
       " 'lhx3',\n",
       " 'oct6',\n",
       " 'etv1',\n",
       " 'smad3',\n",
       " 'nkx2',\n",
       " 'isl1',\n",
       " 'e2f1',\n",
       " 'foxa3',\n",
       " 'ets-distal',\n",
       " 'rfx5',\n",
       " 'gata4',\n",
       " 'tead4',\n",
       " 'lhx1',\n",
       " 'znf692',\n",
       " 'oct4-sox2',\n",
       " 'atf2',\n",
       " 'foxp1',\n",
       " 'tbr1',\n",
       " 'rbpj1',\n",
       " 'trb',\n",
       " 'rfx1',\n",
       " 'thrb',\n",
       " 'nf1-half',\n",
       " 'ebf-old',\n",
       " 'fra1',\n",
       " 'tlx',\n",
       " 'sox15',\n",
       " 'foxk1',\n",
       " 'hif1a',\n",
       " 'tead2',\n",
       " 'ere',\n",
       " 'irf8',\n",
       " 'pr',\n",
       " 'npas2',\n",
       " 'pu1-irf',\n",
       " 'batf',\n",
       " 'oct',\n",
       " 'atf4',\n",
       " 'hoxa13',\n",
       " 'e2f',\n",
       " 'nfe2l2',\n",
       " 'egr1',\n",
       " 'gfy-staf',\n",
       " 'gli',\n",
       " 'ap4',\n",
       " 'sox2',\n",
       " 'gfx',\n",
       " 'sox10',\n",
       " 'tead3',\n",
       " 'sox4',\n",
       " 'xbox',\n",
       " 'hif1b',\n",
       " 'znf165',\n",
       " 'nrf1',\n",
       " 'myog',\n",
       " 'snai2',\n",
       " 'zic',\n",
       " 'rorc',\n",
       " 'gata',\n",
       " 'srf',\n",
       " 'gata-scl',\n",
       " 'barx1',\n",
       " 'otx2',\n",
       " 'p63',\n",
       " 'brachyury',\n",
       " 'e2f4',\n",
       " 'rara',\n",
       " 'atf1',\n",
       " 'stat5',\n",
       " 'zbtb18',\n",
       " 'prop1',\n",
       " 'znf317',\n",
       " 'pax6',\n",
       " 'pitx1-ebox',\n",
       " 'ronin',\n",
       " 'nkx2',\n",
       " 'hand2',\n",
       " 'cux2',\n",
       " 'elk1',\n",
       " 'hoxd11',\n",
       " 'cMyc',\n",
       " 'chop',\n",
       " 'neurog2',\n",
       " 'ebf',\n",
       " 'esrrb',\n",
       " 'cebp',\n",
       " 'clock',\n",
       " 'bach2',\n",
       " 'stat4',\n",
       " 'mafa',\n",
       " 'atf3',\n",
       " 'sox9',\n",
       " 'hoxc9',\n",
       " 'twist1',\n",
       " 'pitx1',\n",
       " 'dr0',\n",
       " 'tcf3',\n",
       " 'gata6',\n",
       " 'zfp3',\n",
       " 'smad2',\n",
       " 'p73',\n",
       " 'es',\n",
       " 'ebox',\n",
       " 'foxo1',\n",
       " 'nfkb',\n",
       " 'gsc',\n",
       " 'nr5a2',\n",
       " 'nfil3',\n",
       " 'spib',\n",
       " 'e2f3',\n",
       " 'elf5',\n",
       " 'nkx6',\n",
       " 'hlf',\n",
       " 'rfx3',\n",
       " 'stat3',\n",
       " 'nfe2',\n",
       " 'runx2',\n",
       " 'mafF',\n",
       " 'tcf21',\n",
       " 'fra2',\n",
       " 'cre',\n",
       " 'myb',\n",
       " 'prdm14',\n",
       " 'e2f7',\n",
       " 'mitf',\n",
       " 'ets1',\n",
       " 'znf41',\n",
       " 'yy1',\n",
       " 'zfx',\n",
       " 'prdm9',\n",
       " 'batf3-irf8',\n",
       " 'hnf4',\n",
       " 'prdm10',\n",
       " 'irf2',\n",
       " 'nr2f6',\n",
       " 'maz',\n",
       " 'brn1',\n",
       " 'klf4',\n",
       " 'lrf',\n",
       " 'hsf1',\n",
       " 'oct-oct',\n",
       " 'irf4',\n",
       " 'eklf',\n",
       " 'klf5',\n",
       " 'hnf1',\n",
       " 't1isre',\n",
       " 'sox6',\n",
       " 'mafb',\n",
       " 'znf652',\n",
       " 'znf467',\n",
       " 'pgr',\n",
       " 'znf7',\n",
       " 'dmrt6',\n",
       " 'neurod1',\n",
       " 'hoxa9',\n",
       " 'pu1-irf8',\n",
       " 'usf1',\n",
       " 'stat6',\n",
       " 'p50',\n",
       " 'tbox-smad',\n",
       " 'pdx1',\n",
       " 'ctcf',\n",
       " 'foxm1',\n",
       " 'nf1-fox',\n",
       " 'fxr',\n",
       " 'tbx5',\n",
       " 'nur77',\n",
       " 'fox-ebox',\n",
       " 'pax7',\n",
       " 'hoxb4',\n",
       " 'junb',\n",
       " 'six1',\n",
       " 'bmyb',\n",
       " 'tcf7',\n",
       " 'hoxa2',\n",
       " 'rest',\n",
       " 'zeb1',\n",
       " 'cdx4',\n",
       " 'tead1',\n",
       " 'znf189',\n",
       " 'hoxa11',\n",
       " 'mef2c',\n",
       " 'runx',\n",
       " 'ctcf-mys',\n",
       " 'ap2gamma',\n",
       " 'hoxd13',\n",
       " 'srebp2',\n",
       " 'eomes',\n",
       " 'oct4',\n",
       " 'nkx3',\n",
       " 'foxa1',\n",
       " 'foxh1',\n",
       " 'myod',\n",
       " 'scrt1',\n",
       " 'e2a',\n",
       " 'etv4',\n",
       " 'are',\n",
       " 'ap1',\n",
       " 'znf382',\n",
       " 'rarg',\n",
       " 'zbtb33',\n",
       " 'hoxa10',\n",
       " 'mynn',\n",
       " 'mnt',\n",
       " 'olig2',\n",
       " 'foxf1']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "motif_folder = doc['motif_folder']\n",
    "motifs_of_interest = map(lambda x: os.path.basename(x).split('.')[0],glob.glob(motif_folder + '/*motif'))\n",
    "motifs_of_interest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-14T10:27:19.036573Z",
     "start_time": "2019-02-14T10:27:19.015650Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "out_dir = \"Results/Figures/SuppFigure3/\"\n",
    "if not os.path.exists(out_dir):\n",
    "    os.mkdir(out_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Preprocessing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create Fasta file for the output bed sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-14T10:27:24.808439Z",
     "start_time": "2019-02-14T10:27:21.719293Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tExtracting sequences from file: /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna\n",
      "\tLooking for peak sequences in a single file (/data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna)\n",
      "\tExtracting 86 sequences from NW_020822366.1\n",
      "\tExtracting 462 sequences from NW_020822370.1\n",
      "\tExtracting 82 sequences from NW_020822372.1\n",
      "\tExtracting 82 sequences from NW_020822373.1\n",
      "\tExtracting 42 sequences from NW_020822374.1\n",
      "\tExtracting 363 sequences from NW_020822375.1\n",
      "\tExtracting 76 sequences from NW_020822376.1\n",
      "\tExtracting 69 sequences from NW_020822377.1\n",
      "\tExtracting 56 sequences from NW_020822379.1\n",
      "\tExtracting 239 sequences from NW_020822381.1\n",
      "\tExtracting 32 sequences from NW_020822382.1\n",
      "\tExtracting 15 sequences from NW_020822385.1\n",
      "\tExtracting 98 sequences from NW_020822388.1\n",
      "\tExtracting 20 sequences from NW_020822390.1\n",
      "\tExtracting 7 sequences from NW_020822391.1\n",
      "\tExtracting 89 sequences from NW_020822393.1\n",
      "\tExtracting 20 sequences from NW_020822394.1\n",
      "\tExtracting 15 sequences from NW_020822396.1\n",
      "\tExtracting 231 sequences from NW_020822398.1\n",
      "\tExtracting 10 sequences from NW_020822399.1\n",
      "\tExtracting 10 sequences from NW_020822402.1\n",
      "\tExtracting 213 sequences from NW_020822403.1\n",
      "\tExtracting 2 sequences from NW_020822404.1\n",
      "\tExtracting 4 sequences from NW_020822405.1\n",
      "\tExtracting 467 sequences from NW_020822406.1\n",
      "\tExtracting 210 sequences from NW_020822407.1\n",
      "\tExtracting 174 sequences from NW_020822409.1\n",
      "\tExtracting 234 sequences from NW_020822410.1\n",
      "\tExtracting 147 sequences from NW_020822412.1\n",
      "\tExtracting 2 sequences from NW_020822413.1\n",
      "\tExtracting 666 sequences from NW_020822415.1\n",
      "\tExtracting 4 sequences from NW_020822416.1\n",
      "\tExtracting 42 sequences from NW_020822419.1\n",
      "\tExtracting 334 sequences from NW_020822420.1\n",
      "\tExtracting 83 sequences from NW_020822421.1\n",
      "\tExtracting 296 sequences from NW_020822423.1\n",
      "\tExtracting 119 sequences from NW_020822425.1\n",
      "\tExtracting 54 sequences from NW_020822426.1\n",
      "\tExtracting 48 sequences from NW_020822427.1\n",
      "\tExtracting 183 sequences from NW_020822428.1\n",
      "\tExtracting 281 sequences from NW_020822429.1\n",
      "\tExtracting 58 sequences from NW_020822432.1\n",
      "\tExtracting 87 sequences from NW_020822434.1\n",
      "\tExtracting 61 sequences from NW_020822435.1\n",
      "\tExtracting 111 sequences from NW_020822436.1\n",
      "\tExtracting 431 sequences from NW_020822438.1\n",
      "\tExtracting 1197 sequences from NW_020822439.1\n",
      "\tExtracting 64 sequences from NW_020822440.1\n",
      "\tExtracting 110 sequences from NW_020822442.1\n",
      "\tExtracting 145 sequences from NW_020822443.1\n",
      "\tExtracting 46 sequences from NW_020822444.1\n",
      "\tExtracting 119 sequences from NW_020822446.1\n",
      "\tExtracting 24 sequences from NW_020822449.1\n",
      "\tExtracting 175 sequences from NW_020822450.1\n",
      "\tExtracting 6 sequences from NW_020822451.1\n",
      "\tExtracting 714 sequences from NW_020822452.1\n",
      "\tExtracting 1 sequences from NW_020822453.1\n",
      "\tExtracting 126 sequences from NW_020822454.1\n",
      "\tExtracting 90 sequences from NW_020822455.1\n",
      "\tExtracting 114 sequences from NW_020822457.1\n",
      "\tExtracting 181 sequences from NW_020822458.1\n",
      "\tExtracting 110 sequences from NW_020822459.1\n",
      "\tExtracting 186 sequences from NW_020822460.1\n",
      "\tExtracting 456 sequences from NW_020822461.1\n",
      "\tExtracting 166 sequences from NW_020822462.1\n",
      "\tExtracting 156 sequences from NW_020822464.1\n",
      "\tExtracting 220 sequences from NW_020822465.1\n",
      "\tExtracting 156 sequences from NW_020822466.1\n",
      "\tExtracting 66 sequences from NW_020822467.1\n",
      "\tExtracting 495 sequences from NW_020822468.1\n",
      "\tExtracting 73 sequences from NW_020822469.1\n",
      "\tExtracting 363 sequences from NW_020822470.1\n",
      "\tExtracting 64 sequences from NW_020822471.1\n",
      "\tExtracting 264 sequences from NW_020822474.1\n",
      "\tExtracting 18 sequences from NW_020822475.1\n",
      "\tExtracting 67 sequences from NW_020822476.1\n",
      "\tExtracting 25 sequences from NW_020822483.1\n",
      "\tExtracting 45 sequences from NW_020822484.1\n",
      "\tExtracting 270 sequences from NW_020822486.1\n",
      "\tExtracting 396 sequences from NW_020822487.1\n",
      "\tExtracting 21 sequences from NW_020822488.1\n",
      "\tExtracting 108 sequences from NW_020822490.1\n",
      "\tExtracting 17 sequences from NW_020822493.1\n",
      "\tExtracting 337 sequences from NW_020822499.1\n",
      "\tExtracting 417 sequences from NW_020822501.1\n",
      "\tExtracting 296 sequences from NW_020822503.1\n",
      "\tExtracting 181 sequences from NW_020822504.1\n",
      "\tExtracting 600 sequences from NW_020822505.1\n",
      "\tExtracting 246 sequences from NW_020822506.1\n",
      "\tExtracting 302 sequences from NW_020822507.1\n",
      "\tExtracting 196 sequences from NW_020822508.1\n",
      "\tExtracting 110 sequences from NW_020822511.1\n",
      "\tExtracting 66 sequences from NW_020822512.1\n",
      "\tExtracting 157 sequences from NW_020822513.1\n",
      "\tExtracting 39 sequences from NW_020822518.1\n",
      "\tExtracting 383 sequences from NW_020822519.1\n",
      "\tExtracting 116 sequences from NW_020822520.1\n",
      "\tExtracting 389 sequences from NW_020822526.1\n",
      "\tExtracting 8 sequences from NW_020822528.1\n",
      "\tExtracting 658 sequences from NW_020822529.1\n",
      "\tExtracting 248 sequences from NW_020822530.1\n",
      "\tExtracting 275 sequences from NW_020822531.1\n",
      "\tExtracting 360 sequences from NW_020822532.1\n",
      "\tExtracting 235 sequences from NW_020822533.1\n",
      "\tExtracting 82 sequences from NW_020822534.1\n",
      "\tExtracting 187 sequences from NW_020822535.1\n",
      "\tExtracting 141 sequences from NW_020822537.1\n",
      "\tExtracting 180 sequences from NW_020822538.1\n",
      "\tExtracting 164 sequences from NW_020822539.1\n",
      "\tExtracting 26 sequences from NW_020822544.1\n",
      "\tExtracting 44 sequences from NW_020822545.1\n",
      "\tExtracting 35 sequences from NW_020822550.1\n",
      "\tExtracting 11 sequences from NW_020822552.1\n",
      "\tExtracting 34 sequences from NW_020822555.1\n",
      "\tExtracting 122 sequences from NW_020822557.1\n",
      "\tExtracting 119 sequences from NW_020822558.1\n",
      "\tExtracting 25 sequences from NW_020822559.1\n",
      "\tExtracting 9 sequences from NW_020822563.1\n",
      "\tExtracting 713 sequences from NW_020822567.1\n",
      "\tExtracting 361 sequences from NW_020822570.1\n",
      "\tExtracting 332 sequences from NW_020822577.1\n",
      "\tExtracting 1 sequences from NW_020822585.1\n",
      "\tExtracting 353 sequences from NW_020822586.1\n",
      "\tExtracting 56 sequences from NW_020822591.1\n",
      "\tExtracting 45 sequences from NW_020822592.1\n",
      "\tExtracting 15 sequences from NW_020822595.1\n",
      "\tExtracting 126 sequences from NW_020822596.1\n",
      "\tExtracting 306 sequences from NW_020822597.1\n",
      "\tExtracting 217 sequences from NW_020822598.1\n",
      "\tExtracting 1013 sequences from NW_020822601.1\n",
      "\tExtracting 266 sequences from NW_020822602.1\n",
      "\tExtracting 208 sequences from NW_020822603.1\n",
      "\tExtracting 205 sequences from NW_020822604.1\n",
      "\tExtracting 104 sequences from NW_020822605.1\n",
      "\tExtracting 213 sequences from NW_020822606.1\n",
      "\tExtracting 43 sequences from NW_020822607.1\n",
      "\tExtracting 319 sequences from NW_020822608.1\n",
      "\tExtracting 1 sequences from NW_020822609.1\n",
      "\tExtracting 299 sequences from NW_020822610.1\n",
      "\tExtracting 430 sequences from NW_020822611.1\n",
      "\tExtracting 154 sequences from NW_020822614.1\n",
      "\tExtracting 1078 sequences from NW_020822615.1\n",
      "\tExtracting 325 sequences from NW_020822616.1\n",
      "\tExtracting 165 sequences from NW_020822618.1\n",
      "\tExtracting 98 sequences from NW_020822619.1\n",
      "\tExtracting 44 sequences from NW_020822620.1\n",
      "\tExtracting 24 sequences from NW_020822621.1\n",
      "\tExtracting 57 sequences from NW_020822623.1\n",
      "\tExtracting 24 sequences from NW_020822626.1\n",
      "\tExtracting 15 sequences from NW_020822627.1\n",
      "\tExtracting 21 sequences from NW_020822628.1\n",
      "\tExtracting 115 sequences from NW_020822629.1\n",
      "\tExtracting 229 sequences from NW_020822630.1\n",
      "\tExtracting 140 sequences from NW_020822632.1\n",
      "\tExtracting 157 sequences from NW_020822634.1\n",
      "\tExtracting 124 sequences from NW_020822636.1\n",
      "\tExtracting 108 sequences from NW_020822638.1\n",
      "\tExtracting 25 sequences from NW_020822641.1\n",
      "\tExtracting 57 sequences from NW_020822642.1\n",
      "\tExtracting 251 sequences from NW_020822643.1\n",
      "\tExtracting 129 sequences from NW_020822644.1\n",
      "\tExtracting 20 sequences from NW_020822645.1\n",
      "\tExtracting 61 sequences from NW_020822646.1\n",
      "\tExtracting 29 sequences from NW_020822647.1\n",
      "\tExtracting 35 sequences from NW_020822648.1\n",
      "\tExtracting 46 sequences from NW_020822649.1\n",
      "\tExtracting 27 sequences from NW_020822650.1\n",
      "\tExtracting 28 sequences from NW_020822651.1\n",
      "\tExtracting 53 sequences from NW_020822652.1\n",
      "\tExtracting 8 sequences from NW_020822653.1\n",
      "\tExtracting 28 sequences from NW_020822654.1\n",
      "\tExtracting 4 sequences from NW_020822656.1\n",
      "\tExtracting 150 sequences from NW_020822657.1\n",
      "\tExtracting 110 sequences from NW_020822658.1\n",
      "\tExtracting 76 sequences from NW_020822663.1\n",
      "\tExtracting 20 sequences from NW_020822664.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 43 sequences from NW_020822665.1\n",
      "\tExtracting 41 sequences from NW_020822666.1\n",
      "\tExtracting 33 sequences from NW_020822668.1\n",
      "\tExtracting 23 sequences from NW_020822670.1\n",
      "\tExtracting 28 sequences from NW_020822671.1\n",
      "\tExtracting 37 sequences from NW_020822672.1\n",
      "\tExtracting 24 sequences from NW_020822674.1\n",
      "\tExtracting 41 sequences from NW_020822676.1\n",
      "\tExtracting 59 sequences from NW_020822680.1\n",
      "\tExtracting 20 sequences from NW_020822681.1\n",
      "\tExtracting 9 sequences from NW_020822686.1\n",
      "\tExtracting 113 sequences from NW_020822687.1\n",
      "\tExtracting 138 sequences from NW_020822688.1\n",
      "\tExtracting 1 sequences from NW_020822690.1\n",
      "\tExtracting 145 sequences from NW_020822698.1\n",
      "\tExtracting 57 sequences from NW_020822699.1\n",
      "\tExtracting 63 sequences from NW_020822701.1\n",
      "\tExtracting 1 sequences from NW_020822702.1\n",
      "\tExtracting 53 sequences from NW_020822703.1\n",
      "\tExtracting 1 sequences from NW_020822734.1\n",
      "\tExtracting 1 sequences from NW_020822867.1\n",
      "\tExtracting 1 sequences from NW_020822982.1\n",
      "\tExtracting 3 sequences from NW_020823085.1\n",
      "\tExtracting 1 sequences from NW_020823094.1\n",
      "\tExtracting 1 sequences from NW_020823142.1\n",
      "\tExtracting 4 sequences from NW_020823156.1\n",
      "\tExtracting 2 sequences from NW_020823167.1\n",
      "\tExtracting 4 sequences from NW_020823398.1\n",
      "\tExtracting 17 sequences from NW_020823422.1\n",
      "\tExtracting 1 sequences from NW_020823442.1\n",
      "\tExtracting 19 sequences from NW_020823468.1\n",
      "\tExtracting 3 sequences from NW_020823473.1\n",
      "\tExtracting 1 sequences from NW_020823474.1\n",
      "\tExtracting 1 sequences from NW_020823490.1\n",
      "\tExtracting 11 sequences from NW_020823503.1\n",
      "\tExtracting 1 sequences from NW_020823510.1\n",
      "\tExtracting 2 sequences from NW_020823518.1\n",
      "\tExtracting 4 sequences from NW_020823529.1\n",
      "\tExtracting 2 sequences from NW_020823533.1\n",
      "\tExtracting 4 sequences from NW_020823558.1\n",
      "\tExtracting 1 sequences from NW_020823586.1\n",
      "\tExtracting 1 sequences from NW_020823587.1\n",
      "\tExtracting 1 sequences from NW_020823602.1\n",
      "\tExtracting 7 sequences from NW_020823627.1\n",
      "\tExtracting 2 sequences from NW_020823629.1\n",
      "\tExtracting 1 sequences from NW_020823639.1\n",
      "\tExtracting 1 sequences from NW_020823676.1\n",
      "\tExtracting 1 sequences from NW_020823792.1\n",
      "\tExtracting 4 sequences from NW_020823819.1\n",
      "\tExtracting 1 sequences from NW_020823856.1\n",
      "\tExtracting 1 sequences from NW_020823876.1\n",
      "\tExtracting 1 sequences from NW_020823898.1\n",
      "\tExtracting 1 sequences from NW_020823908.1\n",
      "\tExtracting 2 sequences from NW_020823928.1\n",
      "\tExtracting 1 sequences from NW_020823970.1\n",
      "\tExtracting 1 sequences from NW_020824001.1\n",
      "\tExtracting 1 sequences from NW_020824030.1\n",
      "\tExtracting 1 sequences from NW_020824056.1\n",
      "\tExtracting 2 sequences from NW_020824066.1\n",
      "\tExtracting 2 sequences from NW_020824120.1\n",
      "\tExtracting 1 sequences from NW_020824123.1\n"
     ]
    }
   ],
   "source": [
    "bed_file = \"Results/output/TSS1.exp.bed\"\n",
    "out_f = os.path.join(out_dir,\"TSS1.exp.fa\")\n",
    "cmd = \"homerTools extract {bed_f} {ref_fa} -fa > {out_f}\".format(bed_f=bed_file,ref_fa=ref_fa,out_f=out_f)\n",
    "!{cmd}\n",
    "#Homer.extract_peak_sequences(bed_file,peak_list,genome,f_save,upstream=1000,downstream=100)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Do the same for reference. Also need to create the bed file beforehand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-02-14T10:27:29.120142Z",
     "start_time": "2019-02-14T10:27:24.810591Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "homerTools extract Results/Figures/SuppFigure3/TSS1.exp_refseq_centered.bed /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna -fa > Results/Figures/SuppFigure3/TSS1.exp_refseq_centered.fa\n",
      "\n",
      "\tExtracting sequences from file: /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna\n",
      "\tLooking for peak sequences in a single file (/data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna)\n",
      "\tExtracting 86 sequences from NW_020822366.1\n",
      "\tExtracting 462 sequences from NW_020822370.1\n",
      "\tExtracting 82 sequences from NW_020822372.1\n",
      "\tExtracting 82 sequences from NW_020822373.1\n",
      "\tExtracting 42 sequences from NW_020822374.1\n",
      "\tExtracting 363 sequences from NW_020822375.1\n",
      "\tExtracting 76 sequences from NW_020822376.1\n",
      "\tExtracting 69 sequences from NW_020822377.1\n",
      "\tExtracting 56 sequences from NW_020822379.1\n",
      "\tExtracting 239 sequences from NW_020822381.1\n",
      "\tExtracting 32 sequences from NW_020822382.1\n",
      "\tExtracting 15 sequences from NW_020822385.1\n",
      "\tExtracting 98 sequences from NW_020822388.1\n",
      "\tExtracting 20 sequences from NW_020822390.1\n",
      "\tExtracting 7 sequences from NW_020822391.1\n",
      "\tExtracting 89 sequences from NW_020822393.1\n",
      "\tExtracting 20 sequences from NW_020822394.1\n",
      "\tExtracting 15 sequences from NW_020822396.1\n",
      "\tExtracting 231 sequences from NW_020822398.1\n",
      "\tExtracting 10 sequences from NW_020822399.1\n",
      "\tExtracting 10 sequences from NW_020822402.1\n",
      "\tExtracting 213 sequences from NW_020822403.1\n",
      "\tExtracting 2 sequences from NW_020822404.1\n",
      "\tExtracting 4 sequences from NW_020822405.1\n",
      "\tExtracting 467 sequences from NW_020822406.1\n",
      "\tExtracting 210 sequences from NW_020822407.1\n",
      "\tExtracting 174 sequences from NW_020822409.1\n",
      "\tExtracting 234 sequences from NW_020822410.1\n",
      "\tExtracting 147 sequences from NW_020822412.1\n",
      "\tExtracting 2 sequences from NW_020822413.1\n",
      "\tExtracting 666 sequences from NW_020822415.1\n",
      "\tExtracting 4 sequences from NW_020822416.1\n",
      "\tExtracting 42 sequences from NW_020822419.1\n",
      "\tExtracting 334 sequences from NW_020822420.1\n",
      "\tExtracting 83 sequences from NW_020822421.1\n",
      "\tExtracting 296 sequences from NW_020822423.1\n",
      "\tExtracting 119 sequences from NW_020822425.1\n",
      "\tExtracting 54 sequences from NW_020822426.1\n",
      "\tExtracting 48 sequences from NW_020822427.1\n",
      "\tExtracting 183 sequences from NW_020822428.1\n",
      "\tExtracting 281 sequences from NW_020822429.1\n",
      "\tExtracting 58 sequences from NW_020822432.1\n",
      "\tExtracting 87 sequences from NW_020822434.1\n",
      "\tExtracting 61 sequences from NW_020822435.1\n",
      "\tExtracting 111 sequences from NW_020822436.1\n",
      "\tExtracting 431 sequences from NW_020822438.1\n",
      "\tExtracting 1197 sequences from NW_020822439.1\n",
      "\tExtracting 64 sequences from NW_020822440.1\n",
      "\tExtracting 110 sequences from NW_020822442.1\n",
      "\tExtracting 145 sequences from NW_020822443.1\n",
      "\tExtracting 46 sequences from NW_020822444.1\n",
      "\tExtracting 119 sequences from NW_020822446.1\n",
      "\tExtracting 24 sequences from NW_020822449.1\n",
      "\tExtracting 175 sequences from NW_020822450.1\n",
      "\tExtracting 6 sequences from NW_020822451.1\n",
      "\tExtracting 714 sequences from NW_020822452.1\n",
      "\tExtracting 1 sequences from NW_020822453.1\n",
      "\tExtracting 126 sequences from NW_020822454.1\n",
      "\tExtracting 90 sequences from NW_020822455.1\n",
      "\tExtracting 114 sequences from NW_020822457.1\n",
      "\tExtracting 181 sequences from NW_020822458.1\n",
      "\tExtracting 110 sequences from NW_020822459.1\n",
      "\tExtracting 186 sequences from NW_020822460.1\n",
      "\tExtracting 456 sequences from NW_020822461.1\n",
      "\tExtracting 166 sequences from NW_020822462.1\n",
      "\tExtracting 156 sequences from NW_020822464.1\n",
      "\tExtracting 220 sequences from NW_020822465.1\n",
      "\tExtracting 156 sequences from NW_020822466.1\n",
      "\tExtracting 66 sequences from NW_020822467.1\n",
      "\tExtracting 495 sequences from NW_020822468.1\n",
      "\tExtracting 73 sequences from NW_020822469.1\n",
      "\tExtracting 363 sequences from NW_020822470.1\n",
      "\tExtracting 64 sequences from NW_020822471.1\n",
      "\tExtracting 264 sequences from NW_020822474.1\n",
      "\tExtracting 18 sequences from NW_020822475.1\n",
      "\tExtracting 67 sequences from NW_020822476.1\n",
      "\tExtracting 25 sequences from NW_020822483.1\n",
      "\tExtracting 45 sequences from NW_020822484.1\n",
      "\tExtracting 270 sequences from NW_020822486.1\n",
      "\tExtracting 396 sequences from NW_020822487.1\n",
      "\tExtracting 21 sequences from NW_020822488.1\n",
      "\tExtracting 108 sequences from NW_020822490.1\n",
      "\tExtracting 17 sequences from NW_020822493.1\n",
      "\tExtracting 337 sequences from NW_020822499.1\n",
      "\tExtracting 417 sequences from NW_020822501.1\n",
      "\tExtracting 296 sequences from NW_020822503.1\n",
      "\tExtracting 181 sequences from NW_020822504.1\n",
      "\tExtracting 600 sequences from NW_020822505.1\n",
      "\tExtracting 246 sequences from NW_020822506.1\n",
      "\tExtracting 302 sequences from NW_020822507.1\n",
      "\tExtracting 196 sequences from NW_020822508.1\n",
      "\tExtracting 110 sequences from NW_020822511.1\n",
      "\tExtracting 66 sequences from NW_020822512.1\n",
      "\tExtracting 157 sequences from NW_020822513.1\n",
      "\tExtracting 39 sequences from NW_020822518.1\n",
      "\tExtracting 383 sequences from NW_020822519.1\n",
      "\tExtracting 116 sequences from NW_020822520.1\n",
      "\tExtracting 389 sequences from NW_020822526.1\n",
      "\tExtracting 8 sequences from NW_020822528.1\n",
      "\tExtracting 658 sequences from NW_020822529.1\n",
      "\tExtracting 248 sequences from NW_020822530.1\n",
      "\tExtracting 275 sequences from NW_020822531.1\n",
      "\tExtracting 360 sequences from NW_020822532.1\n",
      "\tExtracting 235 sequences from NW_020822533.1\n",
      "\tExtracting 82 sequences from NW_020822534.1\n",
      "\tExtracting 187 sequences from NW_020822535.1\n",
      "\tExtracting 143 sequences from NW_020822537.1\n",
      "\tExtracting 180 sequences from NW_020822538.1\n",
      "\tExtracting 164 sequences from NW_020822539.1\n",
      "\tExtracting 26 sequences from NW_020822544.1\n",
      "\tExtracting 44 sequences from NW_020822545.1\n",
      "\tExtracting 35 sequences from NW_020822550.1\n",
      "\tExtracting 11 sequences from NW_020822552.1\n",
      "\tExtracting 34 sequences from NW_020822555.1\n",
      "\tExtracting 122 sequences from NW_020822557.1\n",
      "\tExtracting 119 sequences from NW_020822558.1\n",
      "\tExtracting 25 sequences from NW_020822559.1\n",
      "\tExtracting 9 sequences from NW_020822563.1\n",
      "\tExtracting 713 sequences from NW_020822567.1\n",
      "\tExtracting 361 sequences from NW_020822570.1\n",
      "\tExtracting 332 sequences from NW_020822577.1\n",
      "\tExtracting 1 sequences from NW_020822585.1\n",
      "\tExtracting 353 sequences from NW_020822586.1\n",
      "\tExtracting 56 sequences from NW_020822591.1\n",
      "\tExtracting 45 sequences from NW_020822592.1\n",
      "\tExtracting 15 sequences from NW_020822595.1\n",
      "\tExtracting 126 sequences from NW_020822596.1\n",
      "\tExtracting 306 sequences from NW_020822597.1\n",
      "\tExtracting 217 sequences from NW_020822598.1\n",
      "\tExtracting 1013 sequences from NW_020822601.1\n",
      "\tExtracting 266 sequences from NW_020822602.1\n",
      "\tExtracting 208 sequences from NW_020822603.1\n",
      "\tExtracting 205 sequences from NW_020822604.1\n",
      "\tExtracting 104 sequences from NW_020822605.1\n",
      "\tExtracting 213 sequences from NW_020822606.1\n",
      "\tExtracting 43 sequences from NW_020822607.1\n",
      "\tExtracting 319 sequences from NW_020822608.1\n",
      "\tExtracting 1 sequences from NW_020822609.1\n",
      "\tExtracting 299 sequences from NW_020822610.1\n",
      "\tExtracting 430 sequences from NW_020822611.1\n",
      "\tExtracting 154 sequences from NW_020822614.1\n",
      "\tExtracting 1078 sequences from NW_020822615.1\n",
      "\tExtracting 325 sequences from NW_020822616.1\n",
      "\tExtracting 165 sequences from NW_020822618.1\n",
      "\tExtracting 98 sequences from NW_020822619.1\n",
      "\tExtracting 44 sequences from NW_020822620.1\n",
      "\tExtracting 24 sequences from NW_020822621.1\n",
      "\tExtracting 57 sequences from NW_020822623.1\n",
      "\tExtracting 24 sequences from NW_020822626.1\n",
      "\tExtracting 15 sequences from NW_020822627.1\n",
      "\tExtracting 21 sequences from NW_020822628.1\n",
      "\tExtracting 115 sequences from NW_020822629.1\n",
      "\tExtracting 229 sequences from NW_020822630.1\n",
      "\tExtracting 140 sequences from NW_020822632.1\n",
      "\tExtracting 157 sequences from NW_020822634.1\n",
      "\tExtracting 124 sequences from NW_020822636.1\n",
      "\tExtracting 108 sequences from NW_020822638.1\n",
      "\tExtracting 25 sequences from NW_020822641.1\n",
      "\tExtracting 57 sequences from NW_020822642.1\n",
      "\tExtracting 251 sequences from NW_020822643.1\n",
      "\tExtracting 129 sequences from NW_020822644.1\n",
      "\tExtracting 20 sequences from NW_020822645.1\n",
      "\tExtracting 61 sequences from NW_020822646.1\n",
      "\tExtracting 29 sequences from NW_020822647.1\n",
      "\tExtracting 35 sequences from NW_020822648.1\n",
      "\tExtracting 46 sequences from NW_020822649.1\n",
      "\tExtracting 27 sequences from NW_020822650.1\n",
      "\tExtracting 28 sequences from NW_020822651.1\n",
      "\tExtracting 53 sequences from NW_020822652.1\n",
      "\tExtracting 8 sequences from NW_020822653.1\n",
      "\tExtracting 28 sequences from NW_020822654.1\n",
      "\tExtracting 4 sequences from NW_020822656.1\n",
      "\tExtracting 148 sequences from NW_020822657.1\n",
      "\tExtracting 110 sequences from NW_020822658.1\n",
      "\tExtracting 76 sequences from NW_020822663.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 20 sequences from NW_020822664.1\n",
      "\tExtracting 43 sequences from NW_020822665.1\n",
      "\tExtracting 41 sequences from NW_020822666.1\n",
      "\tExtracting 33 sequences from NW_020822668.1\n",
      "\tExtracting 23 sequences from NW_020822670.1\n",
      "\tExtracting 28 sequences from NW_020822671.1\n",
      "\tExtracting 37 sequences from NW_020822672.1\n",
      "\tExtracting 24 sequences from NW_020822674.1\n",
      "\tExtracting 41 sequences from NW_020822676.1\n",
      "\tExtracting 59 sequences from NW_020822680.1\n",
      "\tExtracting 20 sequences from NW_020822681.1\n",
      "\tExtracting 9 sequences from NW_020822686.1\n",
      "\tExtracting 113 sequences from NW_020822687.1\n",
      "\tExtracting 138 sequences from NW_020822688.1\n",
      "\tExtracting 1 sequences from NW_020822690.1\n",
      "\tExtracting 145 sequences from NW_020822698.1\n",
      "\tExtracting 57 sequences from NW_020822699.1\n",
      "\tExtracting 63 sequences from NW_020822701.1\n",
      "\tExtracting 1 sequences from NW_020822702.1\n",
      "\tExtracting 53 sequences from NW_020822703.1\n",
      "\tExtracting 1 sequences from NW_020822734.1\n",
      "\tExtracting 1 sequences from NW_020822867.1\n",
      "\tExtracting 1 sequences from NW_020822982.1\n",
      "\tExtracting 3 sequences from NW_020823085.1\n",
      "\tExtracting 1 sequences from NW_020823094.1\n",
      "\tExtracting 1 sequences from NW_020823142.1\n",
      "\tExtracting 4 sequences from NW_020823156.1\n",
      "\tExtracting 2 sequences from NW_020823167.1\n",
      "\tExtracting 4 sequences from NW_020823398.1\n",
      "\tExtracting 17 sequences from NW_020823422.1\n",
      "\tExtracting 1 sequences from NW_020823442.1\n",
      "\tExtracting 19 sequences from NW_020823468.1\n",
      "\tExtracting 3 sequences from NW_020823473.1\n",
      "\tExtracting 1 sequences from NW_020823474.1\n",
      "\tExtracting 1 sequences from NW_020823490.1\n",
      "\tExtracting 11 sequences from NW_020823503.1\n",
      "\tExtracting 1 sequences from NW_020823510.1\n",
      "\tExtracting 2 sequences from NW_020823518.1\n",
      "\tExtracting 4 sequences from NW_020823529.1\n",
      "\tExtracting 2 sequences from NW_020823533.1\n",
      "\tExtracting 4 sequences from NW_020823558.1\n",
      "\tExtracting 1 sequences from NW_020823586.1\n",
      "\tExtracting 1 sequences from NW_020823587.1\n",
      "\tExtracting 1 sequences from NW_020823602.1\n",
      "\tExtracting 7 sequences from NW_020823627.1\n",
      "\tExtracting 2 sequences from NW_020823629.1\n",
      "\tExtracting 1 sequences from NW_020823639.1\n",
      "\tExtracting 1 sequences from NW_020823676.1\n",
      "\tExtracting 1 sequences from NW_020823792.1\n",
      "\tExtracting 4 sequences from NW_020823819.1\n",
      "\tExtracting 1 sequences from NW_020823856.1\n",
      "\tExtracting 1 sequences from NW_020823876.1\n",
      "\tExtracting 1 sequences from NW_020823898.1\n",
      "\tExtracting 1 sequences from NW_020823908.1\n",
      "\tExtracting 2 sequences from NW_020823928.1\n",
      "\tExtracting 1 sequences from NW_020823970.1\n",
      "\tExtracting 1 sequences from NW_020824001.1\n",
      "\tExtracting 1 sequences from NW_020824030.1\n",
      "\tExtracting 1 sequences from NW_020824056.1\n",
      "\tExtracting 2 sequences from NW_020824066.1\n",
      "\tExtracting 2 sequences from NW_020824120.1\n",
      "\tExtracting 1 sequences from NW_020824123.1\n"
     ]
    }
   ],
   "source": [
    "bed_f = \"Results/output/TSS1.exp.bed\"\n",
    "meta_f = \"Results/output/TSS1.exp.meta\"\n",
    "save_f = os.path.join(out_dir,\"TSS1.exp_refseq_centered.bed\")\n",
    "refseq_f = mRNA_peak_150_file\n",
    "exp_bed_to_refseq(bed_f, meta_f, refseq_f, save_f = save_f)\n",
    "out_f= os.path.join(out_dir,os.path.basename(save_f).replace(\".bed\",\".fa\"))\n",
    "\n",
    "cmd = \"homerTools extract {bed_f} {ref_fa} -fa > {out_f}\".format(bed_f=save_f,ref_fa=ref_fa,out_f=out_f)\n",
    "print(cmd)\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-01-25T02:42:20.016Z"
    },
    "deletable": false,
    "editable": false,
    "run_control": {
     "frozen": true
    }
   },
   "outputs": [],
   "source": [
    "motifs_of_interest = ['inr','tata', 'sp1', 'ctcf']\n",
    "motifs_of_interest = ['inr']\n",
    "fa_file = os.path.join(out_dir,\"TSS1.exp.fa\")\n",
    "f_save = 'Results/Figures/SuppFigure3/TSS_exp_instance_nonbin'\n",
    "motif_analysis.all_motifs_of_interest_instance(motifs_of_interest,fa_file,\n",
    "                                    motif_folder, f_save,\n",
    "                                    lowmem=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Motif Generation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Experimental"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T21:13:39.627923Z",
     "start_time": "2019-01-25T19:56:50.848308Z"
    },
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "t = time.time()\n",
    "motifs_of_interest = ['inr']\n",
    "fa_file = os.path.join(out_dir,\"TSS1.exp.fa\")\n",
    "\n",
    "f_save = 'Results/Figures/SuppFigure3/TSS_exp_instance_bin'\n",
    "motif_analysis.all_motifs_of_interest_instance(motifs_of_interest,fa_file,\n",
    "                                    motif_folder, f_save,\n",
    "                                    lowmem=True)\n",
    "print(t - time.time())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T21:31:47.921916Z",
     "start_time": "2019-01-25T21:13:39.630328Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "t = time.time()\n",
    "motifs_of_interest = ['inr']\n",
    "fa_file = os.path.join(out_dir,\"TSS1.exp.fa\")\n",
    "\n",
    "f_save = 'Results/Figures/SuppFigure3/TSS_exp_instance_nonbin'\n",
    "motif_analysis.all_motifs_of_interest_instance(motifs_of_interest,fa_file,\n",
    "                                    motif_folder, f_save,\n",
    "                                    lowmem=False)\n",
    "print(t - time.time())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-01-26T00:51:30.067Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "t = time.time()\n",
    "motifs_of_interest = ['inr']\n",
    "fa_file = os.path.join(out_dir,\"TSS1.exp.fa\")\n",
    "\n",
    "f_save = 'Results/Figures/SuppFigure3/TSS_exp_instance_nonbinV2'\n",
    "motif_analysis.all_motifs_of_interest_instance(motifs_of_interest,fa_file,\n",
    "                                    motif_folder, f_save,\n",
    "                                    lowmem=False)\n",
    "print(t - time.time())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Refseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-28T19:53:44.633354Z",
     "start_time": "2019-01-28T19:34:28.457004Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "t = time.time()\n",
    "motifs_of_interest = ['inr']\n",
    "fa_file = os.path.join(out_dir,\"TSS1.exp_refseq_centered.fa\")\n",
    "\n",
    "f_save = 'Results/Figures/SuppFigure3/TSS1.exp_refseq_centered_instance_nonbin'\n",
    "motif_analysis.all_motifs_of_interest_instance(motifs_of_interest,fa_file,\n",
    "                                    motif_folder, f_save,\n",
    "                                    lowmem=False)\n",
    "print(time.time()-t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-25T19:10:51.483653Z",
     "start_time": "2019-01-25T19:10:51.118302Z"
    },
    "deletable": false,
    "editable": false,
    "run_control": {
     "frozen": true
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "inr_df = pd.read_csv(f_save+\"_inr_instance.tsv\",sep=\"\\t\",index_col=0)\n",
    "inr_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4. Plotting"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-28T20:57:10.245116Z",
     "start_time": "2019-01-28T20:57:10.198897Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from motif_analysis import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-28T21:43:30.487226Z",
     "start_time": "2019-01-28T21:43:30.245118Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "peaks_df = read_bed_file(\"Results/output/TSS1.exp.bed\")\n",
    "peaks_df[\"Length\"] = peaks_df[\"End\"] - peaks_df[\"Start\"]\n",
    "#peaks_df.head()\n",
    "peaks_df = peaks_df[\"Length\"].to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-28T21:44:15.657207Z",
     "start_time": "2019-01-28T21:44:05.249522Z"
    },
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#motifs_list = glob.glob(\"Results/Figures/SuppFigure3/*instance.p\")\n",
    "\n",
    "#pfm_list = glob.glob(os.path.join(supp_f, \"homer_motifs/*.pfm\"))\n",
    "pfm_path = os.path.join(supplemental, \"homer_motifs\")\n",
    "\n",
    "#names_list = list(map(lambda x: x.split(\"instance\")[1].replace(\"_\",\"\"), motifs_list))\n",
    "\n",
    "motifs_list = ['Results/Figures/SuppFigure3/TSS_exp_instance_nonbinV2_inr_instance.p','Results/Figures/SuppFigure3/TSS1.exp_refseq_centered_instance_nonbin_inr_instance.p']\n",
    "names_list = [\"inr\",\"inr\"]\n",
    "print(names_list)\n",
    "print(motifs_list)\n",
    "\n",
    "#peaks_df = pd.read_csv(\"Results/output/TSS1.exp.bed\",sep=\"\\t\",index_col=0)\n",
    "\n",
    "wrap_motifs(motifs_list,names_list,peaks_df,pfm_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5. Run for more motifs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-28T22:44:13.501009Z",
     "start_time": "2019-01-28T22:44:13.416891Z"
    }
   },
   "outputs": [],
   "source": [
    "glob.glob(os.path.join(motif_folder,\"*pfm\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-01-28T22:46:03.495Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "t = time.time()\n",
    "motifs_of_interest = ['tata', 'sp1', 'ctcf']\n",
    "fa_file = os.path.join(out_dir,\"TSS1.exp.fa\")\n",
    "\n",
    "f_save = 'Results/Figures/SuppFigure3/TSS_exp_instance_nonbin'\n",
    "motif_analysis.all_motifs_of_interest_instance(motifs_of_interest,fa_file,\n",
    "                                    motif_folder, f_save,\n",
    "                                    lowmem=False)\n",
    "print(time.time()-t)\n",
    "\n",
    "### Refseq\n",
    "\n",
    "t = time.time()\n",
    "\n",
    "fa_file = os.path.join(out_dir,\"TSS1.exp_refseq_centered.fa\")\n",
    "\n",
    "f_save = 'Results/Figures/SuppFigure3/TSS1.exp_refseq_centered_instance_nonbin'\n",
    "motif_analysis.all_motifs_of_interest_instance(motifs_of_interest,fa_file,\n",
    "                                    motif_folder, f_save,\n",
    "                                    lowmem=False)\n",
    "print(time.time()-t)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-31T02:43:56.258867Z",
     "start_time": "2019-01-31T02:41:51.229661Z"
    },
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#motifs_list = glob.glob(\"Results/Figures/SuppFigure3/*instance.p\")\n",
    "\n",
    "#pfm_list = glob.glob(os.path.join(supp_f, \"homer_motifs/*.pfm\"))\n",
    "pfm_path = os.path.join(supplemental, \"homer_motifs\")\n",
    "\n",
    "#names_list = list(map(lambda x: x.split(\"instance\")[1].replace(\"_\",\"\"), motifs_list))\n",
    "\n",
    "motifs_list = ['Results/Figures/SuppFigure3/TSS_exp_instance_nonbinV2_inr_instance.p',\n",
    "               'Results/Figures/SuppFigure3/TSS1.exp_refseq_centered_instance_nonbin_inr_instance.p',\n",
    "               'Results/Figures/SuppFigure3/TSS_exp_instance_nonbin_tata_instance.p',\n",
    "               'Results/Figures/SuppFigure3/TSS1.exp_refseq_centered_instance_nonbin_tata_instance.p',\n",
    "               'Results/Figures/SuppFigure3/TSS_exp_instance_nonbin_sp1_instance.p',\n",
    "               'Results/Figures/SuppFigure3/TSS1.exp_refseq_centered_instance_nonbin_sp1_instance.p',\n",
    "               'Results/Figures/SuppFigure3/TSS_exp_instance_nonbin_ctcf_instance.p',\n",
    "               'Results/Figures/SuppFigure3/TSS1.exp_refseq_centered_instance_nonbin_ctcf_instance.p',\n",
    "              ]\n",
    "\n",
    "names_list = [\"inr exp\",\"inr ref\", 'tata exp', 'tata ref', 'sp1 exp', 'sp1 ref', 'ctcf exp', 'ctcf ref']\n",
    "motif_pfm = [os.path.join(pfm_path, \"inr.motif.pfm\"),os.path.join(pfm_path, \"inr.motif.pfm\"),\n",
    "            os.path.join(pfm_path, \"tata.motif.pfm\"),os.path.join(pfm_path, \"tata.motif.pfm\"),\n",
    "            os.path.join(pfm_path, \"sp1.motif.pfm\"),os.path.join(pfm_path, \"sp1.motif.pfm\"),\n",
    "            os.path.join(pfm_path, \"ctcf.motif.pfm\"),os.path.join(pfm_path, \"ctcf.motif.pfm\")\n",
    "            ]\n",
    "\n",
    "print(names_list)\n",
    "print(motifs_list)\n",
    "\n",
    "#peaks_df = pd.read_csv(\"Results/output/TSS1.exp.bed\",sep=\"\\t\",index_col=0)\n",
    "\n",
    "wrap_motifs(motifs_list,names_list,peaks_df,motif_pfm,out_dir=\"Results/Figures/SuppFigure3/\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
