{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load metasamples and take the Brain one."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T23:43:33.978141Z",
     "start_time": "2020-04-19T23:43:33.407830Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Full</th>\n",
       "      <th>Name</th>\n",
       "      <th>Tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_R1_001.fastq.gz</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...</td>\n",
       "      <td>BMDM1hKLA_ATAC1</td>\n",
       "      <td>BMDM1hKLA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1_001.fastq.gz</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...</td>\n",
       "      <td>BMDM1hKLA_ATAC2</td>\n",
       "      <td>BMDM1hKLA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_001.fastq.gz</td>\n",
       "      <td>GRO</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...</td>\n",
       "      <td>BMDM1hKLA_GRO1</td>\n",
       "      <td>BMDM1hKLA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_001.fastq.gz</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...</td>\n",
       "      <td>BMDM1hKLA_GROCap1</td>\n",
       "      <td>BMDM1hKLA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R1_001.fastq.gz</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_...</td>\n",
       "      <td>BMDMwt_ATAC1</td>\n",
       "      <td>BMDMwt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL5_Liver_S5_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL5_Liver_S5_L006_R1_001.fastq.gz</td>\n",
       "      <td>Liver_RNASeq1</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL6_Heart_S6_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL6_Heart_S6_L006_R1_001.fastq.gz</td>\n",
       "      <td>Heart_RNASeq1</td>\n",
       "      <td>Heart</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL7_Brain2_S7_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL7_Brain2_S7_L006_R1_001.fastq.gz</td>\n",
       "      <td>Brain_RNASeq2</td>\n",
       "      <td>Brain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL8_Muscle_S8_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL8_Muscle_S8_L006_R1_001.fastq.gz</td>\n",
       "      <td>Muscle_RNASeq1</td>\n",
       "      <td>Muscle</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL9_Kidney_S9_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL9_Kidney_S9_L006_R1_001.fastq.gz</td>\n",
       "      <td>Kidney_RNASeq1</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Experiment  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...       ATAC   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...       ATAC   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...        GRO   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...     GROCap   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...       ATAC   \n",
       "...                                                       ...   \n",
       "NL5_Liver_S5_L006_R1_001.fastq.gz                      RNASeq   \n",
       "NL6_Heart_S6_L006_R1_001.fastq.gz                      RNASeq   \n",
       "NL7_Brain2_S7_L006_R1_001.fastq.gz                     RNASeq   \n",
       "NL8_Muscle_S8_L006_R1_001.fastq.gz                     RNASeq   \n",
       "NL9_Kidney_S9_L006_R1_001.fastq.gz                     RNASeq   \n",
       "\n",
       "                                                                                                 Full  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...  CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_...   \n",
       "...                                                                                               ...   \n",
       "NL5_Liver_S5_L006_R1_001.fastq.gz                                   NL5_Liver_S5_L006_R1_001.fastq.gz   \n",
       "NL6_Heart_S6_L006_R1_001.fastq.gz                                   NL6_Heart_S6_L006_R1_001.fastq.gz   \n",
       "NL7_Brain2_S7_L006_R1_001.fastq.gz                                 NL7_Brain2_S7_L006_R1_001.fastq.gz   \n",
       "NL8_Muscle_S8_L006_R1_001.fastq.gz                                 NL8_Muscle_S8_L006_R1_001.fastq.gz   \n",
       "NL9_Kidney_S9_L006_R1_001.fastq.gz                                 NL9_Kidney_S9_L006_R1_001.fastq.gz   \n",
       "\n",
       "                                                                 Name  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...    BMDM1hKLA_ATAC1   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...    BMDM1hKLA_ATAC2   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...     BMDM1hKLA_GRO1   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA_GROCap1   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...       BMDMwt_ATAC1   \n",
       "...                                                               ...   \n",
       "NL5_Liver_S5_L006_R1_001.fastq.gz                       Liver_RNASeq1   \n",
       "NL6_Heart_S6_L006_R1_001.fastq.gz                       Heart_RNASeq1   \n",
       "NL7_Brain2_S7_L006_R1_001.fastq.gz                      Brain_RNASeq2   \n",
       "NL8_Muscle_S8_L006_R1_001.fastq.gz                     Muscle_RNASeq1   \n",
       "NL9_Kidney_S9_L006_R1_001.fastq.gz                     Kidney_RNASeq1   \n",
       "\n",
       "                                                       Tissue  \n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  BMDM1hKLA  \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  BMDM1hKLA  \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  BMDM1hKLA  \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA  \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...     BMDMwt  \n",
       "...                                                       ...  \n",
       "NL5_Liver_S5_L006_R1_001.fastq.gz                       Liver  \n",
       "NL6_Heart_S6_L006_R1_001.fastq.gz                       Heart  \n",
       "NL7_Brain2_S7_L006_R1_001.fastq.gz                      Brain  \n",
       "NL8_Muscle_S8_L006_R1_001.fastq.gz                     Muscle  \n",
       "NL9_Kidney_S9_L006_R1_001.fastq.gz                     Kidney  \n",
       "\n",
       "[78 rows x 4 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "from os.path import join, basename\n",
    "import glob\n",
    "import shutil\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import sys\n",
    "import sarge\n",
    "%matplotlib inline\n",
    "samples = pd.read_csv(\"../parameters/samples.tsv\", sep=\"\\t\", index_col=0)\n",
    "samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Directory names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T23:43:33.985306Z",
     "start_time": "2020-04-19T23:43:33.980279Z"
    }
   },
   "outputs": [],
   "source": [
    "raw_head_dir = \"/media/lewislab/Dropbox (UCSD SBRG)/LewisPub/Data/TSS/\"\n",
    "params_dir = \"../parameters\"\n",
    "out_dir = \"../Results/csRNA_pipeline/\"\n",
    "bam_dir = join(out_dir, \"bam\")\n",
    "sortBam_dir = join(out_dir, \"sortBam\")\n",
    "tag_dir = join(out_dir, \"tags\")\n",
    "csRNA_dir = join(out_dir, 'csRNA_peaks')\n",
    "\n",
    "bedgraph_dir = join(out_dir, \"bedgraph\")\n",
    "\n",
    "if not os.path.exists(bedgraph_dir):\n",
    "    os.mkdir(bedgraph_dir)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T23:43:34.148053Z",
     "start_time": "2020-04-19T23:43:33.987522Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'data_folder': '/data/isshamie/TSS/NCBI_PICR_103_Processed/', 'results': '/data/isshamie/TSS/Analysis/Analysis_03302020/Results/', 'figures': '/data/isshamie/TSS/Analysis/Analysis_03302020/Results/', 'RNAseq_QC': '/data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/2017_12_15_hamster_Seq/fastqc/', 'RNAseq_folder': '/data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/2017_12_15_hamster_Seq/', 'pipeline_path': '/data/isshamie/TSS/Analysis/src/', 'tissues': 'BMDM1hKLA,BMDMwt,Brain,FemaleReproductive,Heart,Kidney,Liver,Lung,MiscOrgans,Muscle,Pancreas,Spleen,CHO', 'ref_fa': '/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna', 'annotation': '/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gff', 'annotation_gtf': '/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gtf', 'alt_annotation': '/data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3', 'alt_annotation_gtf': '/data/isshamie/genome/ncbi_anno_103/alt_CriGri-PICR_top_level.gtf', 'motif_folder': '/data/isshamie/TSS/Analysis/supplemental/homer_motifs/', 'supplemental': '/data/isshamie/TSS/Analysis/supplemental', 'cho_rna_file': '/data/shangzhong/DE/tpm_using_picr.txt', 'htseq_anno_source': 'ncbi', 'strand_specific': 'yes', 'id_name': 'id'}\n",
      "/data2/resources/software/homer/v4.11_10242019/bin//homer\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/ipykernel_launcher.py:4: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "import yaml\n",
    "parameter_file = \"../parameters/params.yaml\"\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "print(doc)\n",
    "gtf = doc[\"annotation_gtf\"]\n",
    "ref_fa = doc[\"ref_fa\"]\n",
    "db_path = \"/data/isshamie/genome/ncbi_anno_103/CriGri-PICR_STAR_db\"\n",
    "results = doc[\"results\"]\n",
    "\n",
    "\n",
    "homer_path =  \"/data2/resources/software/homer/v4.11_10242019/bin/\"\n",
    "os.environ['PATH'] = f\"{homer_path}:{os.environ['PATH']}\"\n",
    "!which homer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T23:43:34.162140Z",
     "start_time": "2020-04-19T23:43:34.152762Z"
    }
   },
   "outputs": [],
   "source": [
    "def make_bedgraph_file(in_dir,out_f):\n",
    "    ''' Create bedgraph file for experiment.'''\n",
    "    cmd = ('makeUCSCfile {in_dir} -o {out_file} -style rnaseq').format(in_dir=in_dir,out_file=out_f)\n",
    "    print(cmd);sys.stdout.flush()\n",
    "    sarge.run(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-19T23:54:18.096611Z",
     "start_time": "2020-04-19T23:43:34.164652Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NL10_Misc_S10_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Misc_S10\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Misc_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Misc_S10 -o ../Results/csRNA_pipeline/bedgraph/Misc_RNASeq1.bedgraph.gz -style rnaseq\n",
      "NL11_Brain7-1_S11_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Brain7\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Brain_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Brain7 -o ../Results/csRNA_pipeline/bedgraph/Brain_RNASeq1.bedgraph.gz -style rnaseq\n",
      "NL1_Pancreas_S1_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Pancreas_S1\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Pancreas_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Pancreas_S1 -o ../Results/csRNA_pipeline/bedgraph/Pancreas_RNASeq1.bedgraph.gz -style rnaseq\n",
      "NL2_ReproductiveTract_S2_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/ReproductiveTract_S2\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/ReproductiveTract_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/ReproductiveTract_S2 -o ../Results/csRNA_pipeline/bedgraph/ReproductiveTract_RNASeq1.bedgraph.gz -style rnaseq\n",
      "NL3_Spleen_S3_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Spleen_S3\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Spleen_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Spleen_S3 -o ../Results/csRNA_pipeline/bedgraph/Spleen_RNASeq1.bedgraph.gz -style rnaseq\n",
      "NL4_Lung_S4_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Lung_S4\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Lung_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Lung_S4 -o ../Results/csRNA_pipeline/bedgraph/Lung_RNASeq1.bedgraph.gz -style rnaseq\n",
      "NL5_Liver_S5_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Liver_S5\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Liver_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Liver_S5 -o ../Results/csRNA_pipeline/bedgraph/Liver_RNASeq1.bedgraph.gz -style rnaseq\n",
      "NL6_Heart_S6_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Heart_S6\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Heart_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Heart_S6 -o ../Results/csRNA_pipeline/bedgraph/Heart_RNASeq1.bedgraph.gz -style rnaseq\n",
      "NL7_Brain2_S7_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Brain2_S7\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Brain_RNASeq2.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Brain2_S7 -o ../Results/csRNA_pipeline/bedgraph/Brain_RNASeq2.bedgraph.gz -style rnaseq\n",
      "NL8_Muscle_S8_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Muscle_S8\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Muscle_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Muscle_S8 -o ../Results/csRNA_pipeline/bedgraph/Muscle_RNASeq1.bedgraph.gz -style rnaseq\n",
      "NL9_Kidney_S9_L006_R1_001.fastq.gz\n",
      "RNA_tag /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Kidney_S9\n",
      "Bedgraph ../Results/csRNA_pipeline/bedgraph/Kidney_RNASeq1.bedgraph.gz\n",
      "makeUCSCfile /data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/tags/Kidney_S9 -o ../Results/csRNA_pipeline/bedgraph/Kidney_RNASeq1.bedgraph.gz -style rnaseq\n"
     ]
    }
   ],
   "source": [
    "for ind, rnaSeq in samples[(samples[\"Experiment\"]==\"RNASeq\")].iterrows():    \n",
    "    ## findCSRNA\n",
    "    name = rnaSeq.name #samples[(samples[\"Experiment\"] == \"RNASeq\") & (samples[\"Tissue\"] == tis)][\"Full\"]\n",
    "    print(name)\n",
    "    name = \"_\".join(name.replace(\".fastq.gz\",\"\").split(\"_\")[1:3]).split(\"-\")[0]\n",
    "    RNA_tag = join(doc[\"data_folder\"],\"RNAseq\",'tags',name)#name))[0]\n",
    "    print('RNA_tag', RNA_tag)\n",
    "\n",
    "    \n",
    "    out_f = join(bedgraph_dir, rnaSeq.Name+'.bedgraph')\n",
    "    print('Bedgraph', out_f)\n",
    "    #out_f = '%s_%s_%s_%s' % (tis,csRNA[\"Name\"].split(\"_\")[1],csInput[\"Name\"].split(\"_\")[1],rnaSeq[\"Name\"].split(\"_\")[1])\n",
    "\n",
    "    make_bedgraph_file(RNA_tag,out_f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-13T23:52:32.288452Z",
     "start_time": "2020-04-13T23:52:32.255952Z"
    }
   },
   "outputs": [],
   "source": [
    "def run_tissue(tis, samples, out_dir):\n",
    "    samples = samples[(samples[\"Tissue\"]==tis)]\n",
    "    if not ((samples[\"Experiment\"] == \"START\").any() & (samples[\"Experiment\"] == \"STARTinput\").any() & \n",
    "        (samples[\"Experiment\"] == \"RNASeq\").any()):\n",
    "            print(\"Not csRNA, csInput, RNASeq\")\n",
    "            return\n",
    "        \n",
    "    for ind, csRNA in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"START\")].iterrows():\n",
    "        copy_to_outdir(raw_dir, csRNA.name, out_d=out_dir, out_name=csRNA[\"Name\"])\n",
    "        ## csRNA trim, align index and tags\n",
    "        trim(join(out_dir,  csRNA[\"Name\"]+\".fastq.gz\"))\n",
    "        trim_f = join(out_dir,  csRNA[\"Name\"] + \".fastq.gz.trimmed\")    \n",
    "        bam_out = join(bam_dir, csRNA[\"Name\"])\n",
    "        cmd = f\"STAR --genomeDir {db_path} --runThreadN 24 --readFilesIn {trim_f} --outFileNamePrefix {bam_out} --outSAMstrandField intronMotif --outMultimapperOrder Random --outSAMmultNmax 1 --outSAMtype BAM SortedByCoordinate --outFilterMultimapNmax 10000 --limitOutSAMoneReadBytes 10000000\"\n",
    "        print(cmd)\n",
    "        os.system(cmd)      \n",
    "        cmd = f\"samtools index {join(bam_dir,csRNA['Name'])}Aligned.sortedByCoord.out.bam\"\n",
    "        print(cmd)\n",
    "        os.system(cmd) \n",
    "        ## Tags\n",
    "        csRNA_tag = join(tag_dir, csRNA[\"Name\"])\n",
    "        csRNA_bam = join(bam_dir, csRNA[\"Name\"])\n",
    "        cmd = f\"makeTagDirectory {csRNA_tag} {csRNA_bam}Aligned.sortedByCoord.out.bam -genome {ref_fa} -single -checkGC -fragLength 150\"\n",
    "        print(cmd)\n",
    "        os.system(cmd) \n",
    "    \n",
    "    for ind2, csInput in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"STARTinput\")].iterrows():\n",
    "        copy_to_outdir(raw_dir, csInput.name, out_d=out_dir, out_name=csInput[\"Name\"])\n",
    "        ## csInput trim, align, index and tags\n",
    "        trim(join(out_dir,  csInput[\"Name\"]+\".fastq.gz\"))\n",
    "        trim_f = join(out_dir,  csInput[\"Name\"] + \".fastq.gz.trimmed\")\n",
    "        bam_out = join(bam_dir, csInput[\"Name\"])\n",
    "        cmd = f\"STAR --genomeDir {db_path} --runThreadN 24 --readFilesIn {trim_f} --outFileNamePrefix {bam_out} --outSAMstrandField intronMotif --outMultimapperOrder Random --outSAMmultNmax 1 --outSAMtype BAM SortedByCoordinate --outFilterMultimapNmax 10000 --limitOutSAMoneReadBytes 10000000\"\n",
    "        print(cmd)\n",
    "        os.system(cmd) \n",
    "        cmd = f\"samtools index {join(bam_dir,csInput['Name'])}Aligned.sortedByCoord.out.bam\"\n",
    "        print(cmd)\n",
    "        os.system(cmd)\n",
    "        csInput_tag = join(tag_dir, csInput[\"Name\"])\n",
    "        csInput_bam = join(bam_dir, csInput[\"Name\"])\n",
    "        cmd = f\"makeTagDirectory {csInput_tag} {csInput_bam}Aligned.sortedByCoord.out.bam -genome {ref_fa}  -single -checkGC -fragLength 150\"\n",
    "        print(cmd)\n",
    "        os.system(cmd)\n",
    "\n",
    "    \n",
    "    ## Tag QC\n",
    "    f = plt.figure()\n",
    "    for ind, csRNA in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"START\")].iterrows():\n",
    "        csrna_count = pd.read_csv(join(csRNA_tag, \"tagCountDistribution.txt\"),sep=\"\\t\", index_col=0, dtype=float)\n",
    "        csrna_count[\"Tag per position\"] = csrna_count.index.astype(int)\n",
    "        plt.scatter(np.log10(csrna_count.index),np.log10(csrna_count[\"Fraction of Positions\"]), color=\"blue\")#, loglog=True)\n",
    "        for ind2, csInput in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"STARTinput\")].iterrows():\n",
    "            input_count = pd.read_csv(join(csInput_tag, \"tagCountDistribution.txt\"),sep=\"\\t\", index_col=0)\n",
    "            input_count[\"Tag per position\"] = input_count.index.astype(int)\n",
    "            plt.scatter(np.log10(input_count.index), np.log10(input_count[\"Fraction of Positions\"]), color=\"orange\")#, loglog=True)    \n",
    "\n",
    "    f = plt.figure()\n",
    "    for ind, csRNA in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"START\")].iterrows():         \n",
    "        csrna_length = pd.read_csv(join(csRNA_tag, \"tagLengthDistribution.txt\"),sep=\"\\t\", index_col=0, dtype=float)\n",
    "        csrna_length[\"Read Length\"] = csrna_length.index.astype(int)\n",
    "        plt.scatter(csrna_length.index,csrna_length[\"Fraction of Tags\"], color=\"blue\")\n",
    "        for ind2, csInput in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"STARTinput\")].iterrows():\n",
    "            input_length = pd.read_csv(join(csInput_tag, \"tagLengthDistribution.txt\"),sep=\"\\t\", index_col=0)\n",
    "            input_length[\"Read Length\"] = input_length.index.astype(int)\n",
    "            plt.scatter(input_length.index,input_length[\"Fraction of Tags\"], color=\"orange\")\n",
    "    plt.xlabel(\"Read length\")\n",
    "    plt.ylabel(\"Fraction of Tags\")\n",
    "            \n",
    "    ## findcsRNATSS peaks\n",
    "    for ind, csRNA in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"START\")].iterrows():         \n",
    "        csRNA_tag = join(tag_dir, csRNA[\"Name\"])\n",
    "        for ind2, csInput in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"STARTinput\")].iterrows():\n",
    "            csInput_tag = join(tag_dir, csInput[\"Name\"])\n",
    "            for ind3, rnaSeq in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"RNASeq\")].iterrows():    \n",
    "                print(csRNA.name, \"\\n\", csInput.name,\"\\n\", rnaSeq.name)   \n",
    "                ## findCSRNA\n",
    "                name = samples[(samples[\"Experiment\"] == \"RNASeq\") & (samples[\"Tissue\"] == tis)][\"Full\"]\n",
    "                name = name[0].replace(\".fastq.gz\",\"\").split(\"_\")[1].split(\"-\")[0]\n",
    "                RNA_tag = glob.glob(join(doc[\"data_folder\"],\"RNAseq\",'tags',\"*\"+name+\"*\"))[0]#name))[0]\n",
    "\n",
    "                rnaSeq[\"Name\"].split(\"_\")[1]\n",
    "                out_name = '%s_%s_%s_%s' % (tis,csRNA[\"Name\"].split(\"_\")[1],csInput[\"Name\"].split(\"_\")[1],rnaSeq[\"Name\"].split(\"_\")[1])\n",
    "\n",
    "                out_f = os.path.join(csRNA_dir,out_name)\n",
    "                cmd = f'findcsRNATSS.pl {csRNA_tag} -i {csInput_tag} -rna {RNA_tag} -gtf {gtf} -genome {ref_fa} -cpu 30 -o {out_f}'\n",
    "                print(cmd)\n",
    "                os.system(cmd)\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-13T23:52:32.301650Z",
     "start_time": "2020-04-13T23:52:32.290159Z"
    },
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "def findcsRNA(tis, samples, out_dir):\n",
    "    samples = samples[(samples[\"Tissue\"]==tis)]\n",
    "    if not ((samples[\"Experiment\"] == \"START\").any() & (samples[\"Experiment\"] == \"STARTinput\").any() & \n",
    "        (samples[\"Experiment\"] == \"RNASeq\").any()):\n",
    "            print(\"Not csRNA, csInput, RNASeq\")\n",
    "            return\n",
    "\n",
    "    ## findcsRNATSS peaks\n",
    "    for ind, csRNA in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"START\")].iterrows():         \n",
    "        csRNA_tag = join(tag_dir, csRNA[\"Name\"])\n",
    "        for ind2, csInput in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"STARTinput\")].iterrows():\n",
    "            csInput_tag = join(tag_dir, csInput[\"Name\"])\n",
    "            for ind3, rnaSeq in samples[(samples[\"Tissue\"]==tis) & (samples[\"Experiment\"]==\"RNASeq\")].iterrows():    \n",
    "                print(csRNA.name, \"\\n\", csInput.name,\"\\n\", rnaSeq.name)   \n",
    "                ## findCSRNA\n",
    "                name = rnaSeq.name #samples[(samples[\"Experiment\"] == \"RNASeq\") & (samples[\"Tissue\"] == tis)][\"Full\"]\n",
    "                print(name)\n",
    "                name = \"_\".join(name.replace(\".fastq.gz\",\"\").split(\"_\")[1:3]).split(\"-\")[0]\n",
    "                RNA_tag = join(doc[\"data_folder\"],\"RNAseq\",'tags',name)#name))[0]\n",
    "                print('RNA_tag', RNA_tag)\n",
    "\n",
    "                out_name = '%s_%s_%s_%s' % (tis,csRNA[\"Name\"].split(\"_\")[1],csInput[\"Name\"].split(\"_\")[1],rnaSeq[\"Name\"].split(\"_\")[1])\n",
    "\n",
    "                out_f = os.path.join(csRNA_dir,out_name)\n",
    "                cmd = f'findcsRNATSS.pl {csRNA_tag} -i {csInput_tag} -rna {RNA_tag} -gtf {gtf} -genome {ref_fa} -cpu 30 -o {out_f}'\n",
    "                print(cmd)\n",
    "                os.system(cmd)\n",
    "    return"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
