{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## First merge with tissues, then separate by unique and overlapping sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-15T21:35:13.602730Z",
     "start_time": "2020-04-15T21:35:12.814710Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "params {'EMAIL': 'isshamie@ucsd.edu', 'parameters': 'parameters/params.yaml'}\n",
      "/data2/resources/software/homer/v4.11_10242019/bin//\n"
     ]
    }
   ],
   "source": [
    "from tss.utils.Homer import merge_peaks, annotate_peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-15T21:35:13.627814Z",
     "start_time": "2020-04-15T21:35:13.605206Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Full</th>\n",
       "      <th>Name</th>\n",
       "      <th>Tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_R1_001.fastq.gz</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...</td>\n",
       "      <td>BMDM1hKLA_ATAC1</td>\n",
       "      <td>BMDM1hKLA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1_001.fastq.gz</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...</td>\n",
       "      <td>BMDM1hKLA_ATAC2</td>\n",
       "      <td>BMDM1hKLA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_001.fastq.gz</td>\n",
       "      <td>GRO</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...</td>\n",
       "      <td>BMDM1hKLA_GRO1</td>\n",
       "      <td>BMDM1hKLA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_001.fastq.gz</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...</td>\n",
       "      <td>BMDM1hKLA_GROCap1</td>\n",
       "      <td>BMDM1hKLA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R1_001.fastq.gz</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_...</td>\n",
       "      <td>BMDMwt_ATAC1</td>\n",
       "      <td>BMDMwt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL5_Liver_S5_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL5_Liver_S5_L006_R1_001.fastq.gz</td>\n",
       "      <td>Liver_RNASeq1</td>\n",
       "      <td>Liver</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL6_Heart_S6_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL6_Heart_S6_L006_R1_001.fastq.gz</td>\n",
       "      <td>Heart_RNASeq1</td>\n",
       "      <td>Heart</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL7_Brain2_S7_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL7_Brain2_S7_L006_R1_001.fastq.gz</td>\n",
       "      <td>Brain_RNASeq2</td>\n",
       "      <td>Brain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL8_Muscle_S8_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL8_Muscle_S8_L006_R1_001.fastq.gz</td>\n",
       "      <td>Muscle_RNASeq1</td>\n",
       "      <td>Muscle</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NL9_Kidney_S9_L006_R1_001.fastq.gz</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>NL9_Kidney_S9_L006_R1_001.fastq.gz</td>\n",
       "      <td>Kidney_RNASeq1</td>\n",
       "      <td>Kidney</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Experiment  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...       ATAC   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...       ATAC   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...        GRO   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...     GROCap   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...       ATAC   \n",
       "...                                                       ...   \n",
       "NL5_Liver_S5_L006_R1_001.fastq.gz                      RNASeq   \n",
       "NL6_Heart_S6_L006_R1_001.fastq.gz                      RNASeq   \n",
       "NL7_Brain2_S7_L006_R1_001.fastq.gz                     RNASeq   \n",
       "NL8_Muscle_S8_L006_R1_001.fastq.gz                     RNASeq   \n",
       "NL9_Kidney_S9_L006_R1_001.fastq.gz                     RNASeq   \n",
       "\n",
       "                                                                                                 Full  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...  CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_...   \n",
       "...                                                                                               ...   \n",
       "NL5_Liver_S5_L006_R1_001.fastq.gz                                   NL5_Liver_S5_L006_R1_001.fastq.gz   \n",
       "NL6_Heart_S6_L006_R1_001.fastq.gz                                   NL6_Heart_S6_L006_R1_001.fastq.gz   \n",
       "NL7_Brain2_S7_L006_R1_001.fastq.gz                                 NL7_Brain2_S7_L006_R1_001.fastq.gz   \n",
       "NL8_Muscle_S8_L006_R1_001.fastq.gz                                 NL8_Muscle_S8_L006_R1_001.fastq.gz   \n",
       "NL9_Kidney_S9_L006_R1_001.fastq.gz                                 NL9_Kidney_S9_L006_R1_001.fastq.gz   \n",
       "\n",
       "                                                                 Name  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...    BMDM1hKLA_ATAC1   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...    BMDM1hKLA_ATAC2   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...     BMDM1hKLA_GRO1   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA_GROCap1   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...       BMDMwt_ATAC1   \n",
       "...                                                               ...   \n",
       "NL5_Liver_S5_L006_R1_001.fastq.gz                       Liver_RNASeq1   \n",
       "NL6_Heart_S6_L006_R1_001.fastq.gz                       Heart_RNASeq1   \n",
       "NL7_Brain2_S7_L006_R1_001.fastq.gz                      Brain_RNASeq2   \n",
       "NL8_Muscle_S8_L006_R1_001.fastq.gz                     Muscle_RNASeq1   \n",
       "NL9_Kidney_S9_L006_R1_001.fastq.gz                     Kidney_RNASeq1   \n",
       "\n",
       "                                                       Tissue  \n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  BMDM1hKLA  \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  BMDM1hKLA  \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  BMDM1hKLA  \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA  \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...     BMDMwt  \n",
       "...                                                       ...  \n",
       "NL5_Liver_S5_L006_R1_001.fastq.gz                       Liver  \n",
       "NL6_Heart_S6_L006_R1_001.fastq.gz                       Heart  \n",
       "NL7_Brain2_S7_L006_R1_001.fastq.gz                      Brain  \n",
       "NL8_Muscle_S8_L006_R1_001.fastq.gz                     Muscle  \n",
       "NL9_Kidney_S9_L006_R1_001.fastq.gz                     Kidney  \n",
       "\n",
       "[78 rows x 4 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "from os.path import join, basename\n",
    "import glob\n",
    "import shutil\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "%matplotlib inline\n",
    "samples = pd.read_csv(\"../parameters/samples.tsv\", sep=\"\\t\", index_col=0)\n",
    "samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Directory names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-15T21:35:13.639790Z",
     "start_time": "2020-04-15T21:35:13.629958Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'data_folder': '/data/isshamie/TSS/NCBI_PICR_103_Processed/', 'results': '/data/isshamie/TSS/Analysis/Analysis_03302020/Results/', 'figures': '/data/isshamie/TSS/Analysis/Analysis_03302020/Results/', 'RNAseq_QC': '/data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/2017_12_15_hamster_Seq/fastqc/', 'RNAseq_folder': '/data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/2017_12_15_hamster_Seq/', 'pipeline_path': '/data/isshamie/TSS/Analysis/src/', 'tissues': 'BMDM1hKLA,BMDMwt,Brain,FemaleReproductive,Heart,Kidney,Liver,Lung,MiscOrgans,Muscle,Pancreas,Spleen,CHO', 'ref_fa': '/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna', 'annotation': '/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gff', 'annotation_gtf': '/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gtf', 'alt_annotation': '/data/isshamie/genome/hamster/ncbi_anno_103/hamster/alt_CriGri-PICR_top_level.gff3', 'alt_annotation_gtf': '/data/isshamie/genome/ncbi_anno_103/alt_CriGri-PICR_top_level.gtf', 'motif_folder': '/data/isshamie/TSS/Analysis/supplemental/homer_motifs/', 'supplemental': '/data/isshamie/TSS/Analysis/supplemental', 'cho_rna_file': '/data/shangzhong/DE/tpm_using_picr.txt', 'htseq_anno_source': 'ncbi', 'strand_specific': 'yes', 'id_name': 'id'}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/ipykernel_launcher.py:4: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "import yaml\n",
    "parameter_file = \"../parameters/params.yaml\"\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "print(doc)\n",
    "gtf = doc[\"annotation_gtf\"]\n",
    "ref_fa = doc[\"ref_fa\"]\n",
    "\n",
    "results = doc[\"results\"]\n",
    "os.chdir(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-15T21:35:13.648953Z",
     "start_time": "2020-04-15T21:35:13.641813Z"
    }
   },
   "outputs": [],
   "source": [
    "out_dir = \"csRNA_pipeline/\"\n",
    "csRNA_dir = join(out_dir, 'csRNA_peaks')\n",
    "\n",
    "\n",
    "homer_path =  \"/data2/resources/software/homer/v4.11_10242019/bin/\"\n",
    "os.environ['PATH'] = f\"{homer_path}:{os.environ['PATH']}\"\n",
    "!which homer"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Outdirs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-15T21:35:13.802449Z",
     "start_time": "2020-04-15T21:35:13.795866Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "tissues_csRNA_dir = join(out_dir,\"csRNA_peaks_tissues\")\n",
    "unique_csRNA_dir =  join(out_dir,\"csRNA_peaks_tissues.merged\")\n",
    "\n",
    "if not os.path.exists(tissues_csRNA_dir):\n",
    "    os.mkdir(tissues_csRNA_dir)\n",
    "    \n",
    "if not os.path.exists(unique_csRNA_dir):\n",
    "    os.mkdir(unique_csRNA_dir)\n",
    "    \n",
    "merged_f = join(unique_csRNA_dir, 'csRNA_peaks.tissues.merge')\n",
    "anno_f = merged_f + '.anno'\n",
    "seq_f = anno_f + '.fa'\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge tissues"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-15T21:35:37.746490Z",
     "start_time": "2020-04-15T21:35:13.804316Z"
    },
    "code_folding": [],
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BMDM1hKLA\n",
      "BMDMwt\n",
      "Brain\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/Brain.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/Brain.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/Brain_START1_STARTinput2_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START2_STARTinput1_RNASeq2.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START3_STARTinput2_RNASeq2.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START1_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START1_STARTinput2_RNASeq2.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START1_STARTinput1_RNASeq2.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START2_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START2_STARTinput2_RNASeq2.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START3_STARTinput2_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START2_STARTinput2_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START3_STARTinput1_RNASeq2.tss.txt csRNA_pipeline/csRNA_peaks/Brain_START3_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/Brain.tss.txt.merge\n",
      "CHO\n",
      "FemaleReproductive\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/FemaleReproductive.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/FemaleReproductive.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/FemaleReproductive_START2_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/FemaleReproductive_START1_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/FemaleReproductive.tss.txt.merge\n",
      "Heart\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/Heart_START2_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Heart_START1_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.txt.merge\n",
      "Intestine\n",
      "Kidney\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/Kidney.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/Kidney.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/Kidney_START2_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Kidney_START1_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/Kidney.tss.txt.merge\n",
      "Liver\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/Liver.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/Liver.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/Liver_START1_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Liver_START2_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/Liver.tss.txt.merge\n",
      "Lung\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/Lung.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/Lung.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/Lung_START1_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Lung_START2_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/Lung.tss.txt.merge\n",
      "MiscOrgans\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/MiscOrgans_START2_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/MiscOrgans_START1_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans.tss.txt.merge\n",
      "Muscle\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/Muscle.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/Muscle.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/Muscle_START2_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Muscle_START1_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/Muscle.tss.txt.merge\n",
      "Pancreas\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/Pancreas.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/Pancreas.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/Pancreas_START1_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Pancreas_START2_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/Pancreas.tss.txt.merge\n",
      "Skin\n",
      "Spleen\n",
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues/Spleen.tss.txt.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues/Spleen.tss.txt.merge_mat -strand csRNA_pipeline/csRNA_peaks/Spleen_START1_STARTinput1_RNASeq1.tss.txt csRNA_pipeline/csRNA_peaks/Spleen_START2_STARTinput1_RNASeq1.tss.txt > csRNA_pipeline/csRNA_peaks_tissues/Spleen.tss.txt.merge\n",
      "Total\n"
     ]
    }
   ],
   "source": [
    "for tis, df in samples.groupby(\"Tissue\"):#samples.loc[samples[\"Tissue\"].isin([\"FemaleReproductive\", \"MiscOrgans\"])].groupby(\"Tissue\"):\n",
    "    print(tis)\n",
    "    files = glob.glob(join(csRNA_dir,\"*\"+tis+\"*.tss.txt\" ))\n",
    "    if len(files) != 0:\n",
    "        out_f = join(tissues_csRNA_dir, tis+'.tss.txt.merge')\n",
    "#         mat_f = out_f + \".matrix\"\n",
    "#         venn_f = out_f + \".venn\"\n",
    "#         cmd = f\"mergePeaks -d given {' '.join(files)} -matrix {mat_f} -venn {venn_f} > {out_f}\"\n",
    "#         #print(files)\n",
    "#         print(cmd)\n",
    "#         os.system(cmd)\n",
    "        \n",
    "        merge_peaks(input_files=files, output_file=out_f, dist='given', type_merge='')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Merge over all tissues "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-15T21:35:51.333299Z",
     "start_time": "2020-04-15T21:35:37.749239Z"
    },
    "code_folding": [],
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mergePeaks -d given -venn csRNA_pipeline/csRNA_peaks_tissues.merged/csRNA_peaks.tissues.merge_venn -matrix csRNA_pipeline/csRNA_peaks_tissues.merged/csRNA_peaks.tissues.merge_mat -strand csRNA_pipeline/csRNA_peaks_tissues/FemaleReproductive.tss.txt.merge csRNA_pipeline/csRNA_peaks_tissues/Heart.tss.txt.merge csRNA_pipeline/csRNA_peaks_tissues/Muscle.tss.txt.merge csRNA_pipeline/csRNA_peaks_tissues/Spleen.tss.txt.merge csRNA_pipeline/csRNA_peaks_tissues/Lung.tss.txt.merge csRNA_pipeline/csRNA_peaks_tissues/Kidney.tss.txt.merge csRNA_pipeline/csRNA_peaks_tissues/Brain.tss.txt.merge csRNA_pipeline/csRNA_peaks_tissues/Liver.tss.txt.merge csRNA_pipeline/csRNA_peaks_tissues/MiscOrgans.tss.txt.merge csRNA_pipeline/csRNA_peaks_tissues/Pancreas.tss.txt.merge > csRNA_pipeline/csRNA_peaks_tissues.merged/csRNA_peaks.tissues.merge\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3242: DtypeWarning: Columns (8,9,10,11,12,13,14,15,16,17) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    }
   ],
   "source": [
    "# for tis, df in samples.groupby(\"Tissue\"):#samples.loc[samples[\"Tissue\"].isin([\"FemaleReproductive\", \"MiscOrgans\"])].groupby(\"Tissue\"):\n",
    "#     print(tis)\n",
    "files = glob.glob(join(tissues_csRNA_dir,\"*.tss.txt.merge\" ))\n",
    "if len(files) != 0:   \n",
    "    #out_f = join(unique_csRNA_dir, 'csRNA_peaks.tissues.merge')\n",
    "#     mat_f = out_f + \".matrix\"\n",
    "#     venn_f = out_f + \".venn\"\n",
    "#     cmd = f\"mergePeaks -d given {' '.join(files)} -matrix {mat_f} -venn {venn_f} > {out_f}\"\n",
    "#     #print(files)\n",
    "#     print(cmd)\n",
    "#     os.system(cmd)\n",
    "\n",
    "    merge_peaks(input_files=files, output_file=merged_f, dist='given', type_merge='')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Annotate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-15T21:39:55.759028Z",
     "start_time": "2020-04-15T21:36:57.012557Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl csRNA_pipeline/csRNA_peaks_tissues.merged/csRNA_peaks.tissues.merge /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna -gtf /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.gtf > csRNA_pipeline/csRNA_peaks_tissues.merged/csRNA_peaks.tissues.merge.anno\n"
     ]
    }
   ],
   "source": [
    "annotate_peaks(peak_file=merged_f,output_file=anno_f,ref_fa=ref_fa,annotation=gtf )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extract sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-04-15T21:39:59.621552Z",
     "start_time": "2020-04-15T21:39:55.763513Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tCustom genome sequence file: /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna\n",
      "\n",
      "\tExtracting sequences from file: /data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna\n",
      "\tLooking for peak sequences in a single file (/data/isshamie/genome/hamster/ncbi_anno_103/GCF_003668045.1_CriGri-PICR/GCF_003668045.1_CriGri-PICR_genomic.fna)\n",
      "\tExtracting 700 sequences from NW_020822366.1\n",
      "\tExtracting 3247 sequences from NW_020822370.1\n",
      "\tExtracting 618 sequences from NW_020822372.1\n",
      "\tExtracting 606 sequences from NW_020822373.1\n",
      "\tExtracting 341 sequences from NW_020822374.1\n",
      "\tExtracting 3004 sequences from NW_020822375.1\n",
      "\tExtracting 659 sequences from NW_020822376.1\n",
      "\tExtracting 465 sequences from NW_020822377.1\n",
      "\tExtracting 470 sequences from NW_020822379.1\n",
      "\tExtracting 897 sequences from NW_020822381.1\n",
      "\tExtracting 289 sequences from NW_020822382.1\n",
      "\tExtracting 214 sequences from NW_020822385.1\n",
      "\tExtracting 602 sequences from NW_020822388.1\n",
      "\tExtracting 231 sequences from NW_020822390.1\n",
      "\tExtracting 63 sequences from NW_020822391.1\n",
      "\tExtracting 341 sequences from NW_020822393.1\n",
      "\tExtracting 135 sequences from NW_020822394.1\n",
      "\tExtracting 11 sequences from NW_020822395.1\n",
      "\tExtracting 108 sequences from NW_020822396.1\n",
      "\tExtracting 1691 sequences from NW_020822398.1\n",
      "\tExtracting 65 sequences from NW_020822399.1\n",
      "\tExtracting 111 sequences from NW_020822402.1\n",
      "\tExtracting 1857 sequences from NW_020822403.1\n",
      "\tExtracting 11 sequences from NW_020822404.1\n",
      "\tExtracting 19 sequences from NW_020822405.1\n",
      "\tExtracting 3257 sequences from NW_020822406.1\n",
      "\tExtracting 1510 sequences from NW_020822407.1\n",
      "\tExtracting 3 sequences from NW_020822408.1\n",
      "\tExtracting 1510 sequences from NW_020822409.1\n",
      "\tExtracting 1912 sequences from NW_020822410.1\n",
      "\tExtracting 6 sequences from NW_020822411.1\n",
      "\tExtracting 1049 sequences from NW_020822412.1\n",
      "\tExtracting 3 sequences from NW_020822413.1\n",
      "\tExtracting 1 sequences from NW_020822414.1\n",
      "\tExtracting 4933 sequences from NW_020822415.1\n",
      "\tExtracting 29 sequences from NW_020822416.1\n",
      "\tExtracting 2 sequences from NW_020822417.1\n",
      "\tExtracting 1 sequences from NW_020822418.1\n",
      "\tExtracting 512 sequences from NW_020822419.1\n",
      "\tExtracting 1724 sequences from NW_020822420.1\n",
      "\tExtracting 640 sequences from NW_020822421.1\n",
      "\tExtracting 1891 sequences from NW_020822423.1\n",
      "\tExtracting 1 sequences from NW_020822424.1\n",
      "\tExtracting 1607 sequences from NW_020822425.1\n",
      "\tExtracting 473 sequences from NW_020822426.1\n",
      "\tExtracting 693 sequences from NW_020822427.1\n",
      "\tExtracting 1397 sequences from NW_020822428.1\n",
      "\tExtracting 1286 sequences from NW_020822429.1\n",
      "\tExtracting 1 sequences from NW_020822430.1\n",
      "\tExtracting 3 sequences from NW_020822431.1\n",
      "\tExtracting 605 sequences from NW_020822432.1\n",
      "\tExtracting 945 sequences from NW_020822434.1\n",
      "\tExtracting 508 sequences from NW_020822435.1\n",
      "\tExtracting 680 sequences from NW_020822436.1\n",
      "\tExtracting 3643 sequences from NW_020822438.1\n",
      "\tExtracting 8893 sequences from NW_020822439.1\n",
      "\tExtracting 582 sequences from NW_020822440.1\n",
      "\tExtracting 634 sequences from NW_020822442.1\n",
      "\tExtracting 1295 sequences from NW_020822443.1\n",
      "\tExtracting 393 sequences from NW_020822444.1\n",
      "\tExtracting 567 sequences from NW_020822446.1\n",
      "\tExtracting 104 sequences from NW_020822449.1\n",
      "\tExtracting 1582 sequences from NW_020822450.1\n",
      "\tExtracting 48 sequences from NW_020822451.1\n",
      "\tExtracting 4343 sequences from NW_020822452.1\n",
      "\tExtracting 2 sequences from NW_020822453.1\n",
      "\tExtracting 1074 sequences from NW_020822454.1\n",
      "\tExtracting 849 sequences from NW_020822455.1\n",
      "\tExtracting 1225 sequences from NW_020822457.1\n",
      "\tExtracting 1318 sequences from NW_020822458.1\n",
      "\tExtracting 1086 sequences from NW_020822459.1\n",
      "\tExtracting 1832 sequences from NW_020822460.1\n",
      "\tExtracting 4027 sequences from NW_020822461.1\n",
      "\tExtracting 1368 sequences from NW_020822462.1\n",
      "\tExtracting 1265 sequences from NW_020822464.1\n",
      "\tExtracting 1588 sequences from NW_020822465.1\n",
      "\tExtracting 1317 sequences from NW_020822466.1\n",
      "\tExtracting 617 sequences from NW_020822467.1\n",
      "\tExtracting 3448 sequences from NW_020822468.1\n",
      "\tExtracting 594 sequences from NW_020822469.1\n",
      "\tExtracting 2002 sequences from NW_020822470.1\n",
      "\tExtracting 562 sequences from NW_020822471.1\n",
      "\tExtracting 1191 sequences from NW_020822474.1\n",
      "\tExtracting 171 sequences from NW_020822475.1\n",
      "\tExtracting 800 sequences from NW_020822476.1\n",
      "\tExtracting 1 sequences from NW_020822479.1\n",
      "\tExtracting 262 sequences from NW_020822483.1\n",
      "\tExtracting 185 sequences from NW_020822484.1\n",
      "\tExtracting 2589 sequences from NW_020822486.1\n",
      "\tExtracting 2442 sequences from NW_020822487.1\n",
      "\tExtracting 155 sequences from NW_020822488.1\n",
      "\tExtracting 509 sequences from NW_020822490.1\n",
      "\tExtracting 91 sequences from NW_020822493.1\n",
      "\tExtracting 56 sequences from NW_020822495.1\n",
      "\tExtracting 36 sequences from NW_020822498.1\n",
      "\tExtracting 2721 sequences from NW_020822499.1\n",
      "\tExtracting 2 sequences from NW_020822500.1\n",
      "\tExtracting 3122 sequences from NW_020822501.1\n",
      "\tExtracting 3 sequences from NW_020822502.1\n",
      "\tExtracting 1955 sequences from NW_020822503.1\n",
      "\tExtracting 2035 sequences from NW_020822504.1\n",
      "\tExtracting 3045 sequences from NW_020822505.1\n",
      "\tExtracting 1750 sequences from NW_020822506.1\n",
      "\tExtracting 2571 sequences from NW_020822507.1\n",
      "\tExtracting 1611 sequences from NW_020822508.1\n",
      "\tExtracting 1 sequences from NW_020822510.1\n",
      "\tExtracting 1081 sequences from NW_020822511.1\n",
      "\tExtracting 524 sequences from NW_020822512.1\n",
      "\tExtracting 1234 sequences from NW_020822513.1\n",
      "\tExtracting 203 sequences from NW_020822518.1\n",
      "\tExtracting 3198 sequences from NW_020822519.1\n",
      "\tExtracting 498 sequences from NW_020822520.1\n",
      "\tExtracting 2902 sequences from NW_020822526.1\n",
      "\tExtracting 19 sequences from NW_020822528.1\n",
      "\tExtracting 5448 sequences from NW_020822529.1\n",
      "\tExtracting 1815 sequences from NW_020822530.1\n",
      "\tExtracting 1948 sequences from NW_020822531.1\n",
      "\tExtracting 1897 sequences from NW_020822532.1\n",
      "\tExtracting 1561 sequences from NW_020822533.1\n",
      "\tExtracting 642 sequences from NW_020822534.1\n",
      "\tExtracting 1453 sequences from NW_020822535.1\n",
      "\tExtracting 3 sequences from NW_020822536.1\n",
      "\tExtracting 1312 sequences from NW_020822537.1\n",
      "\tExtracting 1114 sequences from NW_020822538.1\n",
      "\tExtracting 983 sequences from NW_020822539.1\n",
      "\tExtracting 1 sequences from NW_020822543.1\n",
      "\tExtracting 385 sequences from NW_020822544.1\n",
      "\tExtracting 489 sequences from NW_020822545.1\n",
      "\tExtracting 414 sequences from NW_020822550.1\n",
      "\tExtracting 98 sequences from NW_020822552.1\n",
      "\tExtracting 540 sequences from NW_020822555.1\n",
      "\tExtracting 666 sequences from NW_020822557.1\n",
      "\tExtracting 439 sequences from NW_020822558.1\n",
      "\tExtracting 274 sequences from NW_020822559.1\n",
      "\tExtracting 1 sequences from NW_020822561.1\n",
      "\tExtracting 115 sequences from NW_020822563.1\n",
      "\tExtracting 1 sequences from NW_020822564.1\n",
      "\tExtracting 5 sequences from NW_020822566.1\n",
      "\tExtracting 6682 sequences from NW_020822567.1\n",
      "\tExtracting 3118 sequences from NW_020822570.1\n",
      "\tExtracting 3 sequences from NW_020822572.1\n",
      "\tExtracting 2 sequences from NW_020822573.1\n",
      "\tExtracting 2 sequences from NW_020822574.1\n",
      "\tExtracting 3 sequences from NW_020822575.1\n",
      "\tExtracting 2 sequences from NW_020822576.1\n",
      "\tExtracting 2605 sequences from NW_020822577.1\n",
      "\tExtracting 2 sequences from NW_020822578.1\n",
      "\tExtracting 2 sequences from NW_020822579.1\n",
      "\tExtracting 4 sequences from NW_020822580.1\n",
      "\tExtracting 4 sequences from NW_020822581.1\n",
      "\tExtracting 1 sequences from NW_020822585.1\n",
      "\tExtracting 1839 sequences from NW_020822586.1\n",
      "\tExtracting 1 sequences from NW_020822588.1\n",
      "\tExtracting 1 sequences from NW_020822590.1\n",
      "\tExtracting 409 sequences from NW_020822591.1\n",
      "\tExtracting 476 sequences from NW_020822592.1\n",
      "\tExtracting 1 sequences from NW_020822593.1\n",
      "\tExtracting 1 sequences from NW_020822594.1\n",
      "\tExtracting 191 sequences from NW_020822595.1\n",
      "\tExtracting 622 sequences from NW_020822596.1\n",
      "\tExtracting 2870 sequences from NW_020822597.1\n",
      "\tExtracting 864 sequences from NW_020822598.1\n",
      "\tExtracting 7114 sequences from NW_020822601.1\n",
      "\tExtracting 1534 sequences from NW_020822602.1\n",
      "\tExtracting 1823 sequences from NW_020822603.1\n",
      "\tExtracting 1343 sequences from NW_020822604.1\n",
      "\tExtracting 583 sequences from NW_020822605.1\n",
      "\tExtracting 1230 sequences from NW_020822606.1\n",
      "\tExtracting 241 sequences from NW_020822607.1\n",
      "\tExtracting 2500 sequences from NW_020822608.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 3 sequences from NW_020822609.1\n",
      "\tExtracting 1867 sequences from NW_020822610.1\n",
      "\tExtracting 2703 sequences from NW_020822611.1\n",
      "\tExtracting 1098 sequences from NW_020822614.1\n",
      "\tExtracting 6674 sequences from NW_020822615.1\n",
      "\tExtracting 1680 sequences from NW_020822616.1\n",
      "\tExtracting 903 sequences from NW_020822618.1\n",
      "\tExtracting 567 sequences from NW_020822619.1\n",
      "\tExtracting 348 sequences from NW_020822620.1\n",
      "\tExtracting 160 sequences from NW_020822621.1\n",
      "\tExtracting 389 sequences from NW_020822623.1\n",
      "\tExtracting 141 sequences from NW_020822626.1\n",
      "\tExtracting 71 sequences from NW_020822627.1\n",
      "\tExtracting 96 sequences from NW_020822628.1\n",
      "\tExtracting 1375 sequences from NW_020822629.1\n",
      "\tExtracting 1787 sequences from NW_020822630.1\n",
      "\tExtracting 876 sequences from NW_020822632.1\n",
      "\tExtracting 1271 sequences from NW_020822634.1\n",
      "\tExtracting 1167 sequences from NW_020822636.1\n",
      "\tExtracting 1005 sequences from NW_020822638.1\n",
      "\tExtracting 6 sequences from NW_020822640.1\n",
      "\tExtracting 298 sequences from NW_020822641.1\n",
      "\tExtracting 524 sequences from NW_020822642.1\n",
      "\tExtracting 1104 sequences from NW_020822643.1\n",
      "\tExtracting 433 sequences from NW_020822644.1\n",
      "\tExtracting 148 sequences from NW_020822645.1\n",
      "\tExtracting 360 sequences from NW_020822646.1\n",
      "\tExtracting 123 sequences from NW_020822647.1\n",
      "\tExtracting 242 sequences from NW_020822648.1\n",
      "\tExtracting 206 sequences from NW_020822649.1\n",
      "\tExtracting 122 sequences from NW_020822650.1\n",
      "\tExtracting 79 sequences from NW_020822651.1\n",
      "\tExtracting 138 sequences from NW_020822652.1\n",
      "\tExtracting 42 sequences from NW_020822653.1\n",
      "\tExtracting 143 sequences from NW_020822654.1\n",
      "\tExtracting 5 sequences from NW_020822655.1\n",
      "\tExtracting 12 sequences from NW_020822656.1\n",
      "\tExtracting 1863 sequences from NW_020822657.1\n",
      "\tExtracting 1767 sequences from NW_020822658.1\n",
      "\tExtracting 367 sequences from NW_020822663.1\n",
      "\tExtracting 218 sequences from NW_020822664.1\n",
      "\tExtracting 253 sequences from NW_020822665.1\n",
      "\tExtracting 194 sequences from NW_020822666.1\n",
      "\tExtracting 154 sequences from NW_020822668.1\n",
      "\tExtracting 116 sequences from NW_020822670.1\n",
      "\tExtracting 141 sequences from NW_020822671.1\n",
      "\tExtracting 175 sequences from NW_020822672.1\n",
      "\tExtracting 14 sequences from NW_020822673.1\n",
      "\tExtracting 93 sequences from NW_020822674.1\n",
      "\tExtracting 16 sequences from NW_020822675.1\n",
      "\tExtracting 135 sequences from NW_020822676.1\n",
      "\tExtracting 8 sequences from NW_020822677.1\n",
      "\tExtracting 202 sequences from NW_020822680.1\n",
      "\tExtracting 60 sequences from NW_020822681.1\n",
      "\tExtracting 1 sequences from NW_020822683.1\n",
      "\tExtracting 24 sequences from NW_020822685.1\n",
      "\tExtracting 48 sequences from NW_020822686.1\n",
      "\tExtracting 843 sequences from NW_020822687.1\n",
      "\tExtracting 752 sequences from NW_020822688.1\n",
      "\tExtracting 1 sequences from NW_020822689.1\n",
      "\tExtracting 9 sequences from NW_020822690.1\n",
      "\tExtracting 1 sequences from NW_020822693.1\n",
      "\tExtracting 1 sequences from NW_020822694.1\n",
      "\tExtracting 814 sequences from NW_020822698.1\n",
      "\tExtracting 325 sequences from NW_020822699.1\n",
      "\tExtracting 398 sequences from NW_020822701.1\n",
      "\tExtracting 426 sequences from NW_020822703.1\n",
      "\tExtracting 3 sequences from NW_020822714.1\n",
      "\tExtracting 2 sequences from NW_020822721.1\n",
      "\tExtracting 1 sequences from NW_020822730.1\n",
      "\tExtracting 1 sequences from NW_020822732.1\n",
      "\tExtracting 2 sequences from NW_020822733.1\n",
      "\tExtracting 3 sequences from NW_020822734.1\n",
      "\tExtracting 1 sequences from NW_020822743.1\n",
      "\tExtracting 2 sequences from NW_020822744.1\n",
      "\tExtracting 1 sequences from NW_020822745.1\n",
      "\tExtracting 1 sequences from NW_020822746.1\n",
      "\tExtracting 1 sequences from NW_020822774.1\n",
      "\tExtracting 3 sequences from NW_020822775.1\n",
      "\tExtracting 1 sequences from NW_020822785.1\n",
      "\tExtracting 1 sequences from NW_020822789.1\n",
      "\tExtracting 2 sequences from NW_020822790.1\n",
      "\tExtracting 1 sequences from NW_020822795.1\n",
      "\tExtracting 1 sequences from NW_020822801.1\n",
      "\tExtracting 1 sequences from NW_020822803.1\n",
      "\tExtracting 2 sequences from NW_020822807.1\n",
      "\tExtracting 1 sequences from NW_020822809.1\n",
      "\tExtracting 1 sequences from NW_020822823.1\n",
      "\tExtracting 1 sequences from NW_020822825.1\n",
      "\tExtracting 6 sequences from NW_020822828.1\n",
      "\tExtracting 3 sequences from NW_020822830.1\n",
      "\tExtracting 7 sequences from NW_020822839.1\n",
      "\tExtracting 1 sequences from NW_020822854.1\n",
      "\tExtracting 1 sequences from NW_020822856.1\n",
      "\tExtracting 1 sequences from NW_020822862.1\n",
      "\tExtracting 1 sequences from NW_020822870.1\n",
      "\tExtracting 1 sequences from NW_020822873.1\n",
      "\tExtracting 1 sequences from NW_020822876.1\n",
      "\tExtracting 1 sequences from NW_020822879.1\n",
      "\tExtracting 1 sequences from NW_020822883.1\n",
      "\tExtracting 2 sequences from NW_020822887.1\n",
      "\tExtracting 6 sequences from NW_020822897.1\n",
      "\tExtracting 1 sequences from NW_020822902.1\n",
      "\tExtracting 1 sequences from NW_020822923.1\n",
      "\tExtracting 1 sequences from NW_020822930.1\n",
      "\tExtracting 1 sequences from NW_020822932.1\n",
      "\tExtracting 1 sequences from NW_020822941.1\n",
      "\tExtracting 2 sequences from NW_020822947.1\n",
      "\tExtracting 1 sequences from NW_020822960.1\n",
      "\tExtracting 1 sequences from NW_020822963.1\n",
      "\tExtracting 3 sequences from NW_020822970.1\n",
      "\tExtracting 1 sequences from NW_020822971.1\n",
      "\tExtracting 1 sequences from NW_020822976.1\n",
      "\tExtracting 2 sequences from NW_020822991.1\n",
      "\tExtracting 1 sequences from NW_020822994.1\n",
      "\tExtracting 1 sequences from NW_020823003.1\n",
      "\tExtracting 4 sequences from NW_020823005.1\n",
      "\tExtracting 4 sequences from NW_020823008.1\n",
      "\tExtracting 1 sequences from NW_020823030.1\n",
      "\tExtracting 1 sequences from NW_020823033.1\n",
      "\tExtracting 1 sequences from NW_020823063.1\n",
      "\tExtracting 1 sequences from NW_020823070.1\n",
      "\tExtracting 1 sequences from NW_020823071.1\n",
      "\tExtracting 1 sequences from NW_020823079.1\n",
      "\tExtracting 1 sequences from NW_020823080.1\n",
      "\tExtracting 1 sequences from NW_020823081.1\n",
      "\tExtracting 1 sequences from NW_020823094.1\n",
      "\tExtracting 1 sequences from NW_020823142.1\n",
      "\tExtracting 4 sequences from NW_020823156.1\n",
      "\tExtracting 2 sequences from NW_020823160.1\n",
      "\tExtracting 1 sequences from NW_020823161.1\n",
      "\tExtracting 7 sequences from NW_020823165.1\n",
      "\tExtracting 3 sequences from NW_020823167.1\n",
      "\tExtracting 3 sequences from NW_020823172.1\n",
      "\tExtracting 1 sequences from NW_020823192.1\n",
      "\tExtracting 1 sequences from NW_020823193.1\n",
      "\tExtracting 2 sequences from NW_020823212.1\n",
      "\tExtracting 1 sequences from NW_020823221.1\n",
      "\tExtracting 4 sequences from NW_020823226.1\n",
      "\tExtracting 2 sequences from NW_020823246.1\n",
      "\tExtracting 2 sequences from NW_020823257.1\n",
      "\tExtracting 2 sequences from NW_020823258.1\n",
      "\tExtracting 1 sequences from NW_020823280.1\n",
      "\tExtracting 2 sequences from NW_020823286.1\n",
      "\tExtracting 1 sequences from NW_020823296.1\n",
      "\tExtracting 1 sequences from NW_020823299.1\n",
      "\tExtracting 1 sequences from NW_020823304.1\n",
      "\tExtracting 1 sequences from NW_020823310.1\n",
      "\tExtracting 8 sequences from NW_020823318.1\n",
      "\tExtracting 1 sequences from NW_020823326.1\n",
      "\tExtracting 1 sequences from NW_020823346.1\n",
      "\tExtracting 1 sequences from NW_020823357.1\n",
      "\tExtracting 6 sequences from NW_020823398.1\n",
      "\tExtracting 2 sequences from NW_020823400.1\n",
      "\tExtracting 3 sequences from NW_020823411.1\n",
      "\tExtracting 1 sequences from NW_020823413.1\n",
      "\tExtracting 58 sequences from NW_020823422.1\n",
      "\tExtracting 1 sequences from NW_020823423.1\n",
      "\tExtracting 1 sequences from NW_020823432.1\n",
      "\tExtracting 1 sequences from NW_020823436.1\n",
      "\tExtracting 7 sequences from NW_020823443.1\n",
      "\tExtracting 2 sequences from NW_020823445.1\n",
      "\tExtracting 4 sequences from NW_020823453.1\n",
      "\tExtracting 48 sequences from NW_020823468.1\n",
      "\tExtracting 37 sequences from NW_020823469.1\n",
      "\tExtracting 6 sequences from NW_020823470.1\n",
      "\tExtracting 14 sequences from NW_020823473.1\n",
      "\tExtracting 13 sequences from NW_020823474.1\n",
      "\tExtracting 2 sequences from NW_020823475.1\n",
      "\tExtracting 1 sequences from NW_020823476.1\n",
      "\tExtracting 8 sequences from NW_020823477.1\n",
      "\tExtracting 1 sequences from NW_020823480.1\n",
      "\tExtracting 2 sequences from NW_020823481.1\n",
      "\tExtracting 4 sequences from NW_020823482.1\n",
      "\tExtracting 1 sequences from NW_020823483.1\n",
      "\tExtracting 2 sequences from NW_020823484.1\n",
      "\tExtracting 2 sequences from NW_020823485.1\n",
      "\tExtracting 3 sequences from NW_020823486.1\n",
      "\tExtracting 1 sequences from NW_020823488.1\n",
      "\tExtracting 25 sequences from NW_020823490.1\n",
      "\tExtracting 4 sequences from NW_020823491.1\n",
      "\tExtracting 3 sequences from NW_020823492.1\n",
      "\tExtracting 2 sequences from NW_020823493.1\n",
      "\tExtracting 2 sequences from NW_020823494.1\n",
      "\tExtracting 2 sequences from NW_020823495.1\n",
      "\tExtracting 1 sequences from NW_020823498.1\n",
      "\tExtracting 5 sequences from NW_020823499.1\n",
      "\tExtracting 9 sequences from NW_020823500.1\n",
      "\tExtracting 2 sequences from NW_020823502.1\n",
      "\tExtracting 21 sequences from NW_020823503.1\n",
      "\tExtracting 3 sequences from NW_020823505.1\n",
      "\tExtracting 2 sequences from NW_020823510.1\n",
      "\tExtracting 1 sequences from NW_020823511.1\n",
      "\tExtracting 11 sequences from NW_020823512.1\n",
      "\tExtracting 9 sequences from NW_020823513.1\n",
      "\tExtracting 4 sequences from NW_020823514.1\n",
      "\tExtracting 5 sequences from NW_020823516.1\n",
      "\tExtracting 1 sequences from NW_020823517.1\n",
      "\tExtracting 4 sequences from NW_020823518.1\n",
      "\tExtracting 2 sequences from NW_020823519.1\n",
      "\tExtracting 1 sequences from NW_020823521.1\n",
      "\tExtracting 2 sequences from NW_020823523.1\n",
      "\tExtracting 2 sequences from NW_020823524.1\n",
      "\tExtracting 1 sequences from NW_020823525.1\n",
      "\tExtracting 1 sequences from NW_020823527.1\n",
      "\tExtracting 17 sequences from NW_020823529.1\n",
      "\tExtracting 1 sequences from NW_020823531.1\n",
      "\tExtracting 7 sequences from NW_020823532.1\n",
      "\tExtracting 13 sequences from NW_020823533.1\n",
      "\tExtracting 1 sequences from NW_020823535.1\n",
      "\tExtracting 1 sequences from NW_020823536.1\n",
      "\tExtracting 2 sequences from NW_020823537.1\n",
      "\tExtracting 5 sequences from NW_020823538.1\n",
      "\tExtracting 1 sequences from NW_020823539.1\n",
      "\tExtracting 3 sequences from NW_020823541.1\n",
      "\tExtracting 2 sequences from NW_020823543.1\n",
      "\tExtracting 1 sequences from NW_020823544.1\n",
      "\tExtracting 7 sequences from NW_020823545.1\n",
      "\tExtracting 3 sequences from NW_020823546.1\n",
      "\tExtracting 1 sequences from NW_020823549.1\n",
      "\tExtracting 7 sequences from NW_020823550.1\n",
      "\tExtracting 2 sequences from NW_020823555.1\n",
      "\tExtracting 7 sequences from NW_020823556.1\n",
      "\tExtracting 2 sequences from NW_020823557.1\n",
      "\tExtracting 25 sequences from NW_020823558.1\n",
      "\tExtracting 2 sequences from NW_020823559.1\n",
      "\tExtracting 1 sequences from NW_020823561.1\n",
      "\tExtracting 9 sequences from NW_020823563.1\n",
      "\tExtracting 5 sequences from NW_020823564.1\n",
      "\tExtracting 1 sequences from NW_020823567.1\n",
      "\tExtracting 2 sequences from NW_020823569.1\n",
      "\tExtracting 3 sequences from NW_020823570.1\n",
      "\tExtracting 4 sequences from NW_020823571.1\n",
      "\tExtracting 3 sequences from NW_020823574.1\n",
      "\tExtracting 3 sequences from NW_020823575.1\n",
      "\tExtracting 1 sequences from NW_020823578.1\n",
      "\tExtracting 1 sequences from NW_020823579.1\n",
      "\tExtracting 1 sequences from NW_020823580.1\n",
      "\tExtracting 3 sequences from NW_020823581.1\n",
      "\tExtracting 1 sequences from NW_020823584.1\n",
      "\tExtracting 5 sequences from NW_020823585.1\n",
      "\tExtracting 7 sequences from NW_020823586.1\n",
      "\tExtracting 3 sequences from NW_020823587.1\n",
      "\tExtracting 3 sequences from NW_020823588.1\n",
      "\tExtracting 1 sequences from NW_020823591.1\n",
      "\tExtracting 1 sequences from NW_020823592.1\n",
      "\tExtracting 1 sequences from NW_020823593.1\n",
      "\tExtracting 3 sequences from NW_020823596.1\n",
      "\tExtracting 3 sequences from NW_020823597.1\n",
      "\tExtracting 2 sequences from NW_020823601.1\n",
      "\tExtracting 2 sequences from NW_020823603.1\n",
      "\tExtracting 1 sequences from NW_020823607.1\n",
      "\tExtracting 1 sequences from NW_020823609.1\n",
      "\tExtracting 2 sequences from NW_020823611.1\n",
      "\tExtracting 4 sequences from NW_020823612.1\n",
      "\tExtracting 4 sequences from NW_020823613.1\n",
      "\tExtracting 2 sequences from NW_020823614.1\n",
      "\tExtracting 8 sequences from NW_020823615.1\n",
      "\tExtracting 2 sequences from NW_020823616.1\n",
      "\tExtracting 1 sequences from NW_020823620.1\n",
      "\tExtracting 1 sequences from NW_020823623.1\n",
      "\tExtracting 2 sequences from NW_020823624.1\n",
      "\tExtracting 1 sequences from NW_020823625.1\n",
      "\tExtracting 29 sequences from NW_020823627.1\n",
      "\tExtracting 8 sequences from NW_020823629.1\n",
      "\tExtracting 2 sequences from NW_020823631.1\n",
      "\tExtracting 1 sequences from NW_020823635.1\n",
      "\tExtracting 3 sequences from NW_020823637.1\n",
      "\tExtracting 4 sequences from NW_020823639.1\n",
      "\tExtracting 1 sequences from NW_020823648.1\n",
      "\tExtracting 1 sequences from NW_020823651.1\n",
      "\tExtracting 3 sequences from NW_020823655.1\n",
      "\tExtracting 3 sequences from NW_020823656.1\n",
      "\tExtracting 2 sequences from NW_020823658.1\n",
      "\tExtracting 1 sequences from NW_020823659.1\n",
      "\tExtracting 1 sequences from NW_020823662.1\n",
      "\tExtracting 3 sequences from NW_020823663.1\n",
      "\tExtracting 1 sequences from NW_020823666.1\n",
      "\tExtracting 1 sequences from NW_020823669.1\n",
      "\tExtracting 2 sequences from NW_020823674.1\n",
      "\tExtracting 1 sequences from NW_020823675.1\n",
      "\tExtracting 6 sequences from NW_020823676.1\n",
      "\tExtracting 2 sequences from NW_020823677.1\n",
      "\tExtracting 1 sequences from NW_020823678.1\n",
      "\tExtracting 4 sequences from NW_020823679.1\n",
      "\tExtracting 1 sequences from NW_020823682.1\n",
      "\tExtracting 1 sequences from NW_020823689.1\n",
      "\tExtracting 1 sequences from NW_020823690.1\n",
      "\tExtracting 1 sequences from NW_020823691.1\n",
      "\tExtracting 1 sequences from NW_020823693.1\n",
      "\tExtracting 4 sequences from NW_020823695.1\n",
      "\tExtracting 4 sequences from NW_020823697.1\n",
      "\tExtracting 1 sequences from NW_020823698.1\n",
      "\tExtracting 3 sequences from NW_020823699.1\n",
      "\tExtracting 13 sequences from NW_020823701.1\n",
      "\tExtracting 6 sequences from NW_020823702.1\n",
      "\tExtracting 3 sequences from NW_020823704.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 2 sequences from NW_020823707.1\r\n",
      "\tExtracting 1 sequences from NW_020823709.1\r\n",
      "\tExtracting 1 sequences from NW_020823713.1\r\n",
      "\tExtracting 5 sequences from NW_020823717.1\r\n",
      "\tExtracting 1 sequences from NW_020823718.1\r\n",
      "\tExtracting 2 sequences from NW_020823720.1\r\n",
      "\tExtracting 1 sequences from NW_020823721.1\r\n",
      "\tExtracting 1 sequences from NW_020823722.1\r\n",
      "\tExtracting 2 sequences from NW_020823723.1\r\n",
      "\tExtracting 4 sequences from NW_020823728.1\r\n",
      "\tExtracting 1 sequences from NW_020823733.1\r\n",
      "\tExtracting 2 sequences from NW_020823736.1\r\n",
      "\tExtracting 5 sequences from NW_020823741.1\r\n",
      "\tExtracting 3 sequences from NW_020823744.1\r\n",
      "\tExtracting 1 sequences from NW_020823748.1\r\n",
      "\tExtracting 5 sequences from NW_020823749.1\r\n",
      "\tExtracting 1 sequences from NW_020823752.1\r\n",
      "\tExtracting 4 sequences from NW_020823754.1\r\n",
      "\tExtracting 2 sequences from NW_020823763.1\r\n",
      "\tExtracting 1 sequences from NW_020823765.1\r\n",
      "\tExtracting 1 sequences from NW_020823766.1\r\n",
      "\tExtracting 9 sequences from NW_020823767.1\r\n",
      "\tExtracting 2 sequences from NW_020823768.1\r\n",
      "\tExtracting 12 sequences from NW_020823770.1\r\n",
      "\tExtracting 1 sequences from NW_020823772.1\r\n",
      "\tExtracting 1 sequences from NW_020823791.1\r\n",
      "\tExtracting 5 sequences from NW_020823792.1\r\n",
      "\tExtracting 1 sequences from NW_020823798.1\r\n",
      "\tExtracting 1 sequences from NW_020823799.1\r\n",
      "\tExtracting 1 sequences from NW_020823806.1\r\n",
      "\tExtracting 6 sequences from NW_020823812.1\r\n",
      "\tExtracting 2 sequences from NW_020823817.1\r\n",
      "\tExtracting 23 sequences from NW_020823819.1\r\n",
      "\tExtracting 1 sequences from NW_020823823.1\r\n",
      "\tExtracting 1 sequences from NW_020823830.1\r\n",
      "\tExtracting 3 sequences from NW_020823831.1\r\n",
      "\tExtracting 2 sequences from NW_020823832.1\r\n",
      "\tExtracting 1 sequences from NW_020823834.1\r\n",
      "\tExtracting 1 sequences from NW_020823847.1\r\n",
      "\tExtracting 1 sequences from NW_020823848.1\r\n",
      "\tExtracting 1 sequences from NW_020823849.1\r\n",
      "\tExtracting 1 sequences from NW_020823855.1\r\n",
      "\tExtracting 2 sequences from NW_020823860.1\r\n",
      "\tExtracting 1 sequences from NW_020823868.1\r\n",
      "\tExtracting 7 sequences from NW_020823871.1\r\n",
      "\tExtracting 1 sequences from NW_020823873.1\r\n",
      "\tExtracting 3 sequences from NW_020823876.1\r\n",
      "\tExtracting 1 sequences from NW_020823880.1\r\n",
      "\tExtracting 1 sequences from NW_020823882.1\r\n",
      "\tExtracting 3 sequences from NW_020823887.1\r\n",
      "\tExtracting 1 sequences from NW_020823894.1\r\n",
      "\tExtracting 2 sequences from NW_020823896.1\r\n",
      "\tExtracting 1 sequences from NW_020823899.1\r\n",
      "\tExtracting 1 sequences from NW_020823905.1\r\n",
      "\tExtracting 3 sequences from NW_020823906.1\r\n",
      "\tExtracting 1 sequences from NW_020823910.1\r\n",
      "\tExtracting 4 sequences from NW_020823912.1\r\n",
      "\tExtracting 1 sequences from NW_020823913.1\r\n",
      "\tExtracting 1 sequences from NW_020823917.1\r\n",
      "\tExtracting 1 sequences from NW_020823924.1\r\n",
      "\tExtracting 6 sequences from NW_020823928.1\r\n",
      "\tExtracting 1 sequences from NW_020823929.1\r\n",
      "\tExtracting 1 sequences from NW_020823930.1\r\n",
      "\tExtracting 1 sequences from NW_020823931.1\r\n",
      "\tExtracting 1 sequences from NW_020823938.1\r\n",
      "\tExtracting 1 sequences from NW_020823939.1\r\n",
      "\tExtracting 1 sequences from NW_020823945.1\r\n",
      "\tExtracting 1 sequences from NW_020823946.1\r\n",
      "\tExtracting 3 sequences from NW_020823948.1\r\n",
      "\tExtracting 2 sequences from NW_020823954.1\r\n",
      "\tExtracting 16 sequences from NW_020823957.1\r\n",
      "\tExtracting 1 sequences from NW_020823959.1\r\n",
      "\tExtracting 1 sequences from NW_020823961.1\r\n",
      "\tExtracting 1 sequences from NW_020823970.1\r\n",
      "\tExtracting 1 sequences from NW_020823976.1\r\n",
      "\tExtracting 1 sequences from NW_020823977.1\r\n",
      "\tExtracting 1 sequences from NW_020823980.1\r\n",
      "\tExtracting 1 sequences from NW_020823984.1\r\n",
      "\tExtracting 6 sequences from NW_020823989.1\r\n",
      "\tExtracting 2 sequences from NW_020823991.1\r\n",
      "\tExtracting 1 sequences from NW_020824001.1\r\n",
      "\tExtracting 1 sequences from NW_020824002.1\r\n",
      "\tExtracting 1 sequences from NW_020824003.1\r\n",
      "\tExtracting 1 sequences from NW_020824010.1\r\n",
      "\tExtracting 1 sequences from NW_020824012.1\r\n",
      "\tExtracting 2 sequences from NW_020824013.1\r\n",
      "\tExtracting 1 sequences from NW_020824015.1\r\n",
      "\tExtracting 1 sequences from NW_020824017.1\r\n",
      "\tExtracting 1 sequences from NW_020824020.1\r\n",
      "\tExtracting 1 sequences from NW_020824021.1\r\n",
      "\tExtracting 1 sequences from NW_020824024.1\r\n",
      "\tExtracting 3 sequences from NW_020824028.1\r\n",
      "\tExtracting 1 sequences from NW_020824030.1\r\n",
      "\tExtracting 2 sequences from NW_020824034.1\r\n",
      "\tExtracting 1 sequences from NW_020824036.1\r\n",
      "\tExtracting 2 sequences from NW_020824037.1\r\n",
      "\tExtracting 1 sequences from NW_020824045.1\r\n",
      "\tExtracting 1 sequences from NW_020824049.1\r\n",
      "\tExtracting 1 sequences from NW_020824050.1\r\n",
      "\tExtracting 1 sequences from NW_020824053.1\r\n",
      "\tExtracting 4 sequences from NW_020824056.1\r\n",
      "\tExtracting 1 sequences from NW_020824059.1\r\n",
      "\tExtracting 1 sequences from NW_020824064.1\r\n",
      "\tExtracting 4 sequences from NW_020824065.1\r\n",
      "\tExtracting 3 sequences from NW_020824066.1\r\n",
      "\tExtracting 3 sequences from NW_020824067.1\r\n",
      "\tExtracting 2 sequences from NW_020824069.1\r\n",
      "\tExtracting 1 sequences from NW_020824071.1\r\n",
      "\tExtracting 1 sequences from NW_020824072.1\r\n",
      "\tExtracting 1 sequences from NW_020824073.1\r\n",
      "\tExtracting 1 sequences from NW_020824074.1\r\n",
      "\tExtracting 1 sequences from NW_020824075.1\r\n",
      "\tExtracting 1 sequences from NW_020824082.1\r\n",
      "\tExtracting 1 sequences from NW_020824083.1\r\n",
      "\tExtracting 1 sequences from NW_020824091.1\r\n",
      "\tExtracting 1 sequences from NW_020824093.1\r\n",
      "\tExtracting 1 sequences from NW_020824096.1\r\n",
      "\tExtracting 1 sequences from NW_020824098.1\r\n",
      "\tExtracting 2 sequences from NW_020824100.1\r\n",
      "\tExtracting 2 sequences from NW_020824101.1\r\n",
      "\tExtracting 5 sequences from NW_020824107.1\r\n",
      "\tExtracting 4 sequences from NW_020824120.1\r\n",
      "\tExtracting 3 sequences from NW_020824125.1\r\n",
      "\tExtracting 2 sequences from NW_020824128.1\r\n",
      "\tExtracting 1 sequences from NW_020824140.1\r\n",
      "\tExtracting 1 sequences from NW_020824143.1\r\n",
      "\tExtracting 1 sequences from NW_020824149.1\r\n",
      "\tExtracting 1 sequences from NW_020824152.1\r\n",
      "\tExtracting 3 sequences from NW_020824156.1\r\n",
      "\tExtracting 1 sequences from NW_020824158.1\r\n",
      "\tExtracting 5 sequences from NW_020824159.1\r\n",
      "\tExtracting 2 sequences from NW_020824162.1\r\n",
      "\tExtracting 1 sequences from NW_020824166.1\r\n",
      "\tExtracting 1 sequences from NW_020824174.1\r\n",
      "\tExtracting 1 sequences from NW_020824175.1\r\n",
      "\tExtracting 1 sequences from NW_020824187.1\r\n",
      "\tExtracting 1 sequences from NW_020824191.1\r\n",
      "\tExtracting 3 sequences from NW_020824193.1\r\n",
      "\tExtracting 1 sequences from NW_020824194.1\r\n",
      "\tExtracting 1 sequences from NW_020824195.1\r\n"
     ]
    }
   ],
   "source": [
    "cmd = 'homerTools extract {f_in} {ref_fa} -fa > {seq_f}'.format(f_in=anno_f,ref_fa=ref_fa,seq_f=seq_f)\n",
    "!{cmd}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
