{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run de novo motif analysis on different subsets of TSS.\n",
    "## 1. Extract the TSS of the regions of interest from the bed file using the Tissues in the meta file. \n",
    "### Subsets:\n",
    "## Unique:\n",
    "### 1. TSS unique to a tissue. \n",
    "### 2. TSS in a tissue with more than 2SD more than the median. (Mean absolute deviance)\n",
    "## Houekeeping\n",
    "### 3. TSS in housekeeping genes (max/median < 2) \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:14.279773Z",
     "start_time": "2020-05-16T02:21:13.144032Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/ipykernel_launcher.py:7: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  import sys\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data2/resources/software/homer/v4.11_10242019/bin//homer\r\n"
     ]
    }
   ],
   "source": [
    "## Parameters specific to where your folders are and your data\n",
    "parameter_file = '../../parameters/params.yaml'\n",
    "import yaml\n",
    "import sys\n",
    "import os\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "#p = dic2obj(**doc)\n",
    "\n",
    "#data_folder = doc['data_folder']\n",
    "#tissues = doc['tissues'].split(',')\n",
    "#sys.path.append(doc['pipeline_path'])\n",
    "ref_fa = doc['GENOME_FA']\n",
    "#anno_gff=doc['annotation']\n",
    "#mRNA_peak_file = doc[\"mRNA_peak_file\"]\n",
    "#tss_annotation = doc['tss_annotation']\n",
    "\n",
    "from scipy.stats import zscore\n",
    "\n",
    "import sys\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import pickle\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "from matplotlib import pyplot as plt\n",
    "from itertools import product\n",
    "import glob\n",
    "import re\n",
    "from os.path import join\n",
    "from matplotlib_venn import venn2\n",
    "from matplotlib import rcParams\n",
    "import inspect\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "sys.setrecursionlimit(3000)\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "rcParams['figure.figsize'] = 8, 6\n",
    "import tqdm\n",
    "from collections import defaultdict\n",
    "from os.path import basename\n",
    "##mpl.use('Agg')\n",
    "#mpl.style.use('ggplot')\n",
    "#mpl.style.use('fivethirtyeight')\n",
    "#from Homer import *\n",
    "\n",
    "homer_path =  \"/data2/resources/software/homer/v4.11_10242019/bin/\"\n",
    "os.environ['PATH'] = f\"{homer_path}:{os.environ['PATH']}\"\n",
    "!which homer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:14.307033Z",
     "start_time": "2020-05-16T02:21:14.282592Z"
    }
   },
   "outputs": [],
   "source": [
    "import click"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:14.353830Z",
     "start_time": "2020-05-16T02:21:14.309278Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "params {'PREFIX': 'ncbi_pice_103', 'ROOT_DIR': '/data/isshamie/TSS/', 'DATA_DIR': '/data/isshamie/TSS/NCBI_PICR_103_Processed/', 'DATA_PROCESSED_DIR': '/data/isshamie/TSS/TSS/data/processed/', 'SUPPLEMENTAL_DIR': '/data/isshamie/TSS/Analysis/supplemental/', 'PIPELINE': '/data/isshamie/TSS/tss', 'GENOME_DIR': '/data/isshamie/genome', 'GENOME_FA': '/data/isshamie/genome/picr.fa', 'GENOME_GFF3': '/data/isshamie/genome/updated_final_sort.gff3', 'GENOME_GTF': '/data/isshamie/genome/updated_final_sort.gtf', 'TISSUES': 'BMDM1hKLA,BMDMwt,Brain,FemaleReproductive,Heart,Intestine,', 'META_FILE': '/data/isshamie/TSS/TSS/parameters/raw_metaSamples.tsv'}\n"
     ]
    }
   ],
   "source": [
    "from tss.utils import wrap_findMotifs\n",
    "from tss.data.data_io import read_bed_file\n",
    "from tss.data.data_io import write_bed_file\n",
    "from tss.utils import wrap_findMotifs\n",
    "from tss.data import data_io\n",
    "from tss.visualize.fig_utils import helper_save\n",
    "from tss.data.tissues_collapse import merge_columns\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:14.372320Z",
     "start_time": "2020-05-16T02:21:14.355838Z"
    }
   },
   "outputs": [],
   "source": [
    "os.chdir(doc[\"results\"])\n",
    "genome=\"alt\"\n",
    "outdir= join(\"de_novo_motifs\",genome,'enhancers_unique')\n",
    "\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Outdir:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:14.389123Z",
     "start_time": "2020-05-16T02:21:14.374002Z"
    }
   },
   "outputs": [],
   "source": [
    "unique_gene_peaks_dir = \"unique_genes/\"\n",
    "unique_dir = \"/data/isshamie/TSS/Analysis/TSS_code/figures/alt/barplots_eTSS/bmdm/\" #gene_tissues_expressedLiver.uniqueGenes.txt\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Define unique tissues"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:14.408066Z",
     "start_time": "2020-05-16T02:21:14.390900Z"
    }
   },
   "outputs": [],
   "source": [
    "os.chdir(doc[\"results\"])\n",
    "genome=\"alt\"\n",
    "outdir= join(\"de_novo_motifs\",genome)\n",
    "data_dir = join(doc[\"results\"], genome)\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "    \n",
    "enh_dir = join(outdir,'enhancer_unique')\n",
    "if not os.path.exists(enh_dir):\n",
    "    os.mkdir(enh_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:14.534763Z",
     "start_time": "2020-05-16T02:21:14.409688Z"
    }
   },
   "outputs": [],
   "source": [
    "os.chdir(doc[\"results\"])\n",
    "TSS1_bed = read_bed_file('alt/eTSS/TSS.exp.bed')\n",
    "TSS1_meta = pd.read_csv('alt/eTSS/TSS.exp.meta.tsv',sep=\"\\t\",index_col=0)\n",
    "unique_genes_dir = \"enhancer_unique_genes\"\n",
    "silent_bed_f = 'alt/eTSS/TSS1.ref.bed'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:14.555967Z",
     "start_time": "2020-05-16T02:21:14.537580Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "tissue_expr_f = join(data_dir,\"tissues_collapse/tissues.merge.peaksexpression\")\n",
    "peak_anno_f = join(data_dir, \"merged\",\"peaks_with_tss_distances_size1.peak\")\n",
    "merged_f = join(data_dir, \"merged\",\"samples.merge\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load peak and expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:15.152056Z",
     "start_time": "2020-05-16T02:21:14.557754Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MiscOrgans</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Brain</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Heart</th>\n",
       "      <th>CHO</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>BMDM1hKLA</th>\n",
       "      <th>FemaleReproductive</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Kidney</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>BMDMwt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.790000</td>\n",
       "      <td>2.180000</td>\n",
       "      <td>0.835</td>\n",
       "      <td>26.996</td>\n",
       "      <td>0.000</td>\n",
       "      <td>84.2</td>\n",
       "      <td>0.630</td>\n",
       "      <td>0.715</td>\n",
       "      <td>2.130000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>3.975</td>\n",
       "      <td>1.1</td>\n",
       "      <td>1.608571</td>\n",
       "      <td>16.596667</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.265</td>\n",
       "      <td>0.545</td>\n",
       "      <td>1.050000</td>\n",
       "      <td>3.385</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.887429</td>\n",
       "      <td>2.386667</td>\n",
       "      <td>0.000</td>\n",
       "      <td>19.044</td>\n",
       "      <td>1.490</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.025</td>\n",
       "      <td>1.100</td>\n",
       "      <td>1.870000</td>\n",
       "      <td>1.255</td>\n",
       "      <td>5.66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>3.025</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.914286</td>\n",
       "      <td>1.633333</td>\n",
       "      <td>0.835</td>\n",
       "      <td>1.680</td>\n",
       "      <td>2.385</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.000</td>\n",
       "      <td>1.495</td>\n",
       "      <td>1.153333</td>\n",
       "      <td>2.875</td>\n",
       "      <td>2.45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>112.050</td>\n",
       "      <td>29.7</td>\n",
       "      <td>189.308571</td>\n",
       "      <td>50.966667</td>\n",
       "      <td>25.600</td>\n",
       "      <td>20.292</td>\n",
       "      <td>95.500</td>\n",
       "      <td>0.0</td>\n",
       "      <td>136.550</td>\n",
       "      <td>124.150</td>\n",
       "      <td>43.640000</td>\n",
       "      <td>88.550</td>\n",
       "      <td>3.39</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   MiscOrgans  Liver       Brain       Lung   Heart     CHO  Pancreas  \\\n",
       "0       0.705    0.0    0.790000   2.180000   0.835  26.996     0.000   \n",
       "1       3.975    1.1    1.608571  16.596667   0.000   0.000     0.000   \n",
       "2       0.705    0.0    1.887429   2.386667   0.000  19.044     1.490   \n",
       "3       3.025    0.0    3.914286   1.633333   0.835   1.680     2.385   \n",
       "4     112.050   29.7  189.308571  50.966667  25.600  20.292    95.500   \n",
       "\n",
       "   BMDM1hKLA  FemaleReproductive   Muscle     Kidney  Spleen  BMDMwt  \n",
       "0       84.2               0.630    0.715   2.130000   0.000    0.00  \n",
       "1        0.0               4.265    0.545   1.050000   3.385    0.00  \n",
       "2        0.0               4.025    1.100   1.870000   1.255    5.66  \n",
       "3        0.0               6.000    1.495   1.153333   2.875    2.45  \n",
       "4        0.0             136.550  124.150  43.640000  88.550    3.39  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tissue_expr = pd.read_csv(tissue_expr_f,sep='\\t', index_col=0)\n",
    "#tissue_expr = np.log2(tissue_expr+1)\n",
    "tissue_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:16.459113Z",
     "start_time": "2020-05-16T02:21:15.154309Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "      <th>Nearest gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NW_020822452.1</td>\n",
       "      <td>16590385</td>\n",
       "      <td>16590385</td>\n",
       "      <td>+</td>\n",
       "      <td>16.227333</td>\n",
       "      <td>promoter-TSS (ID=id181373;Parent=rna15526;Dbxr...</td>\n",
       "      <td>XM_027399668.1</td>\n",
       "      <td>-462.0</td>\n",
       "      <td>False</td>\n",
       "      <td>Clic4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NW_020822597.1</td>\n",
       "      <td>23595396</td>\n",
       "      <td>23595396</td>\n",
       "      <td>+</td>\n",
       "      <td>5.740000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027420351.1</td>\n",
       "      <td>-23850.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Baz2b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>NW_020822680.1</td>\n",
       "      <td>728718</td>\n",
       "      <td>728718</td>\n",
       "      <td>+</td>\n",
       "      <td>7.579579</td>\n",
       "      <td>promoter-TSS (ID=id609942;Parent=rna53731;Dbxr...</td>\n",
       "      <td>XM_027432478.1</td>\n",
       "      <td>746.0</td>\n",
       "      <td>False</td>\n",
       "      <td>Irak1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NW_020822602.1</td>\n",
       "      <td>10883205</td>\n",
       "      <td>10883205</td>\n",
       "      <td>+</td>\n",
       "      <td>3.992000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027423054.1</td>\n",
       "      <td>9079.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Gbgt1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>37427907</td>\n",
       "      <td>37427907</td>\n",
       "      <td>+</td>\n",
       "      <td>129.031738</td>\n",
       "      <td>promoter-TSS (ID=id474636;Parent=rna41512;Dbxr...</td>\n",
       "      <td>XM_027421990.1</td>\n",
       "      <td>-11.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Shf</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540122</td>\n",
       "      <td>NW_020822538.1</td>\n",
       "      <td>6114468</td>\n",
       "      <td>6114468</td>\n",
       "      <td>+</td>\n",
       "      <td>7.122000</td>\n",
       "      <td>promoter-TSS (ID=id398770;Parent=rna34727;Dbxr...</td>\n",
       "      <td>XM_027415967.1</td>\n",
       "      <td>-58.0</td>\n",
       "      <td>False</td>\n",
       "      <td>Selplg</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540123</td>\n",
       "      <td>NW_020822487.1</td>\n",
       "      <td>8747242</td>\n",
       "      <td>8747242</td>\n",
       "      <td>+</td>\n",
       "      <td>7.508000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027405626.1</td>\n",
       "      <td>-7856.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Herpud1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540124</td>\n",
       "      <td>NW_020822407.1</td>\n",
       "      <td>2824603</td>\n",
       "      <td>2824603</td>\n",
       "      <td>+</td>\n",
       "      <td>3.612167</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027390727.1</td>\n",
       "      <td>-132495.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Klf3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540125</td>\n",
       "      <td>NW_020822545.1</td>\n",
       "      <td>3178880</td>\n",
       "      <td>3178880</td>\n",
       "      <td>-</td>\n",
       "      <td>2.400286</td>\n",
       "      <td>TTS (ID=id403905;Parent=rna35141;Dbxref=GeneID...</td>\n",
       "      <td>XM_027416357.1</td>\n",
       "      <td>1383.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Ccdc85c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540126</td>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>8668670</td>\n",
       "      <td>8668670</td>\n",
       "      <td>+</td>\n",
       "      <td>3.498750</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027421498.1</td>\n",
       "      <td>21927.0</td>\n",
       "      <td>False</td>\n",
       "      <td>LOC100757597</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>540127 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Chr     Start       End Strand        Stat  \\\n",
       "ID                                                              \n",
       "0       NW_020822452.1  16590385  16590385      +   16.227333   \n",
       "1       NW_020822597.1  23595396  23595396      +    5.740000   \n",
       "2       NW_020822680.1    728718    728718      +    7.579579   \n",
       "3       NW_020822602.1  10883205  10883205      +    3.992000   \n",
       "4       NW_020822601.1  37427907  37427907      +  129.031738   \n",
       "...                ...       ...       ...    ...         ...   \n",
       "540122  NW_020822538.1   6114468   6114468      +    7.122000   \n",
       "540123  NW_020822487.1   8747242   8747242      +    7.508000   \n",
       "540124  NW_020822407.1   2824603   2824603      +    3.612167   \n",
       "540125  NW_020822545.1   3178880   3178880      -    2.400286   \n",
       "540126  NW_020822601.1   8668670   8668670      +    3.498750   \n",
       "\n",
       "                                               Annotation     Nearest TSS  \\\n",
       "ID                                                                          \n",
       "0       promoter-TSS (ID=id181373;Parent=rna15526;Dbxr...  XM_027399668.1   \n",
       "1                                              Intergenic  XM_027420351.1   \n",
       "2       promoter-TSS (ID=id609942;Parent=rna53731;Dbxr...  XM_027432478.1   \n",
       "3                                              Intergenic  XM_027423054.1   \n",
       "4       promoter-TSS (ID=id474636;Parent=rna41512;Dbxr...  XM_027421990.1   \n",
       "...                                                   ...             ...   \n",
       "540122  promoter-TSS (ID=id398770;Parent=rna34727;Dbxr...  XM_027415967.1   \n",
       "540123                                         Intergenic  XM_027405626.1   \n",
       "540124                                         Intergenic  XM_027390727.1   \n",
       "540125  TTS (ID=id403905;Parent=rna35141;Dbxref=GeneID...  XM_027416357.1   \n",
       "540126                                         Intergenic  XM_027421498.1   \n",
       "\n",
       "        Distance to TSS  isSameStrand  Nearest gene  \n",
       "ID                                                   \n",
       "0                -462.0         False         Clic4  \n",
       "1              -23850.0          True         Baz2b  \n",
       "2                 746.0         False         Irak1  \n",
       "3                9079.0          True         Gbgt1  \n",
       "4                 -11.0          True           Shf  \n",
       "...                 ...           ...           ...  \n",
       "540122            -58.0         False        Selplg  \n",
       "540123          -7856.0          True       Herpud1  \n",
       "540124        -132495.0          True          Klf3  \n",
       "540125           1383.0          True       Ccdc85c  \n",
       "540126          21927.0         False  LOC100757597  \n",
       "\n",
       "[540127 rows x 10 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peak_anno = pd.read_csv(peak_anno_f,sep='\\t', index_col=0)\n",
    "peak_anno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:19.599353Z",
     "start_time": "2020-05-16T02:21:16.461251Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/envs/TSS/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3051: DtypeWarning: Columns (8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>MiscOrgans_START1_and_MiscOrgans_startinput1</th>\n",
       "      <th>Liver_START2_and_Liver_startinput1</th>\n",
       "      <th>Brain_GROCap1_and_Brain_GRO1</th>\n",
       "      <th>...</th>\n",
       "      <th>MiscOrgans_START2_and_MiscOrgans_startinput1</th>\n",
       "      <th>Spleen_START1_and_Spleen_startinput1</th>\n",
       "      <th>Lung_START2_and_Lung_startinput1</th>\n",
       "      <th>CHO_START1_and_CHO_startinput1</th>\n",
       "      <th>FemaleReproductive_START1_and_FemaleReproductive_startinput1</th>\n",
       "      <th>Kidney_START2_and_Kidney_startinput1</th>\n",
       "      <th>Kidney_GROCap1_and_Kidney_GRO1</th>\n",
       "      <th>CHO_GROCap1_and_CHO_GRO1</th>\n",
       "      <th>BMDMwt_GROCap1_and_BMDMwt_GRO1</th>\n",
       "      <th>Spleen_START2_and_Spleen_startinput1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NW_020822452.1</td>\n",
       "      <td>16590187</td>\n",
       "      <td>16590584</td>\n",
       "      <td>+</td>\n",
       "      <td>16.227333</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>15</td>\n",
       "      <td>NW_020822452.1-1136</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822452.1-1139</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822452.1-450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822452.1-419</td>\n",
       "      <td>NW_020822452.1-145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NW_020822597.1</td>\n",
       "      <td>23595304</td>\n",
       "      <td>23595489</td>\n",
       "      <td>+</td>\n",
       "      <td>5.740000</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>16</td>\n",
       "      <td>NW_020822597.1-387</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822597.1-324</td>\n",
       "      <td>...</td>\n",
       "      <td>NW_020822597.1-103</td>\n",
       "      <td>NW_020822597.1-248</td>\n",
       "      <td>NW_020822597.1-10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822597.1-101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822597.1-126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>NW_020822680.1</td>\n",
       "      <td>728607</td>\n",
       "      <td>728829</td>\n",
       "      <td>+</td>\n",
       "      <td>7.579579</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>19</td>\n",
       "      <td>NW_020822680.1-92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822680.1-114</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822680.1-37</td>\n",
       "      <td>NW_020822680.1-56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822680.1-67</td>\n",
       "      <td>NW_020822680.1-19</td>\n",
       "      <td>NW_020822680.1-44</td>\n",
       "      <td>NW_020822680.1-72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NW_020822602.1</td>\n",
       "      <td>10883074</td>\n",
       "      <td>10883337</td>\n",
       "      <td>+</td>\n",
       "      <td>3.992000</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>20</td>\n",
       "      <td>NW_020822602.1-204</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NW_020822602.1-254</td>\n",
       "      <td>NW_020822602.1-122</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822602.1-226</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822602.1-303</td>\n",
       "      <td>NW_020822602.1-594</td>\n",
       "      <td>NW_020822602.1-266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>37427816</td>\n",
       "      <td>37427999</td>\n",
       "      <td>+</td>\n",
       "      <td>129.031738</td>\n",
       "      <td>MiscOrgans_START1_and_MiscOrgans_startinput1|B...</td>\n",
       "      <td>23</td>\n",
       "      <td>NW_020822601.1-18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822601.1-1445</td>\n",
       "      <td>...</td>\n",
       "      <td>NW_020822601.1-52</td>\n",
       "      <td>NW_020822601.1-123</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822601.1-41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NW_020822601.1-1949</td>\n",
       "      <td>NW_020822601.1-992</td>\n",
       "      <td>NW_020822601.1-2287</td>\n",
       "      <td>NW_020822601.1-12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Chr     Start       End Strand        Stat  \\\n",
       "ID                                                          \n",
       "0   NW_020822452.1  16590187  16590584      +   16.227333   \n",
       "1   NW_020822597.1  23595304  23595489      +    5.740000   \n",
       "2   NW_020822680.1    728607    728829      +    7.579579   \n",
       "3   NW_020822602.1  10883074  10883337      +    3.992000   \n",
       "4   NW_020822601.1  37427816  37427999      +  129.031738   \n",
       "\n",
       "                                         Parent files  Total subpeaks  \\\n",
       "ID                                                                      \n",
       "0   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              15   \n",
       "1   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              16   \n",
       "2   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              19   \n",
       "3   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              20   \n",
       "4   MiscOrgans_START1_and_MiscOrgans_startinput1|B...              23   \n",
       "\n",
       "   MiscOrgans_START1_and_MiscOrgans_startinput1  \\\n",
       "ID                                                \n",
       "0                           NW_020822452.1-1136   \n",
       "1                            NW_020822597.1-387   \n",
       "2                             NW_020822680.1-92   \n",
       "3                            NW_020822602.1-204   \n",
       "4                             NW_020822601.1-18   \n",
       "\n",
       "   Liver_START2_and_Liver_startinput1 Brain_GROCap1_and_Brain_GRO1  ...  \\\n",
       "ID                                                                  ...   \n",
       "0                                 NaN          NW_020822452.1-1139  ...   \n",
       "1                                 NaN           NW_020822597.1-324  ...   \n",
       "2                                 NaN           NW_020822680.1-114  ...   \n",
       "3                                 NaN                          NaN  ...   \n",
       "4                                 NaN          NW_020822601.1-1445  ...   \n",
       "\n",
       "   MiscOrgans_START2_and_MiscOrgans_startinput1  \\\n",
       "ID                                                \n",
       "0                                           NaN   \n",
       "1                            NW_020822597.1-103   \n",
       "2                                           NaN   \n",
       "3                            NW_020822602.1-254   \n",
       "4                             NW_020822601.1-52   \n",
       "\n",
       "   Spleen_START1_and_Spleen_startinput1 Lung_START2_and_Lung_startinput1  \\\n",
       "ID                                                                         \n",
       "0                                   NaN                              NaN   \n",
       "1                    NW_020822597.1-248                NW_020822597.1-10   \n",
       "2                                   NaN                              NaN   \n",
       "3                    NW_020822602.1-122                              NaN   \n",
       "4                    NW_020822601.1-123                              NaN   \n",
       "\n",
       "   CHO_START1_and_CHO_startinput1  \\\n",
       "ID                                  \n",
       "0              NW_020822452.1-450   \n",
       "1                             NaN   \n",
       "2               NW_020822680.1-37   \n",
       "3              NW_020822602.1-226   \n",
       "4               NW_020822601.1-41   \n",
       "\n",
       "   FemaleReproductive_START1_and_FemaleReproductive_startinput1  \\\n",
       "ID                                                                \n",
       "0                                                 NaN             \n",
       "1                                  NW_020822597.1-101             \n",
       "2                                   NW_020822680.1-56             \n",
       "3                                                 NaN             \n",
       "4                                                 NaN             \n",
       "\n",
       "   Kidney_START2_and_Kidney_startinput1 Kidney_GROCap1_and_Kidney_GRO1  \\\n",
       "ID                                                                       \n",
       "0                                   NaN             NW_020822452.1-419   \n",
       "1                                   NaN                            NaN   \n",
       "2                                   NaN              NW_020822680.1-67   \n",
       "3                                   NaN                            NaN   \n",
       "4                                   NaN            NW_020822601.1-1949   \n",
       "\n",
       "   CHO_GROCap1_and_CHO_GRO1 BMDMwt_GROCap1_and_BMDMwt_GRO1  \\\n",
       "ID                                                           \n",
       "0        NW_020822452.1-145                            NaN   \n",
       "1                       NaN                            NaN   \n",
       "2         NW_020822680.1-19              NW_020822680.1-44   \n",
       "3        NW_020822602.1-303             NW_020822602.1-594   \n",
       "4        NW_020822601.1-992            NW_020822601.1-2287   \n",
       "\n",
       "   Spleen_START2_and_Spleen_startinput1  \n",
       "ID                                       \n",
       "0                                   NaN  \n",
       "1                    NW_020822597.1-126  \n",
       "2                     NW_020822680.1-72  \n",
       "3                    NW_020822602.1-266  \n",
       "4                     NW_020822601.1-12  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.read_csv(merged_f,sep='\\t', index_col=0)\n",
    "merged_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Average the BMDM samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:19.625428Z",
     "start_time": "2020-05-16T02:21:19.602403Z"
    }
   },
   "outputs": [],
   "source": [
    "BMDM = tissue_expr.columns.str.contains(\"BMDM\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:20.950198Z",
     "start_time": "2020-05-16T02:21:19.627574Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BMDM</th>\n",
       "      <th>MiscOrgans</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Brain</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Heart</th>\n",
       "      <th>CHO</th>\n",
       "      <th>Pancreas</th>\n",
       "      <th>FemaleReproductive</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Kidney</th>\n",
       "      <th>Spleen</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>42.100</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.790000</td>\n",
       "      <td>2.180000</td>\n",
       "      <td>0.835</td>\n",
       "      <td>26.996</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.630</td>\n",
       "      <td>0.715</td>\n",
       "      <td>2.130000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>0.000</td>\n",
       "      <td>3.975</td>\n",
       "      <td>1.1</td>\n",
       "      <td>1.608571</td>\n",
       "      <td>16.596667</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>4.265</td>\n",
       "      <td>0.545</td>\n",
       "      <td>1.050000</td>\n",
       "      <td>3.385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>2.830</td>\n",
       "      <td>0.705</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.887429</td>\n",
       "      <td>2.386667</td>\n",
       "      <td>0.000</td>\n",
       "      <td>19.044</td>\n",
       "      <td>1.490</td>\n",
       "      <td>4.025</td>\n",
       "      <td>1.100</td>\n",
       "      <td>1.870000</td>\n",
       "      <td>1.255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1.225</td>\n",
       "      <td>3.025</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.914286</td>\n",
       "      <td>1.633333</td>\n",
       "      <td>0.835</td>\n",
       "      <td>1.680</td>\n",
       "      <td>2.385</td>\n",
       "      <td>6.000</td>\n",
       "      <td>1.495</td>\n",
       "      <td>1.153333</td>\n",
       "      <td>2.875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1.695</td>\n",
       "      <td>112.050</td>\n",
       "      <td>29.7</td>\n",
       "      <td>189.308571</td>\n",
       "      <td>50.966667</td>\n",
       "      <td>25.600</td>\n",
       "      <td>20.292</td>\n",
       "      <td>95.500</td>\n",
       "      <td>136.550</td>\n",
       "      <td>124.150</td>\n",
       "      <td>43.640000</td>\n",
       "      <td>88.550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540122</td>\n",
       "      <td>1.520</td>\n",
       "      <td>0.890</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.920000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.565</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>12.450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540123</td>\n",
       "      <td>11.500</td>\n",
       "      <td>0.645</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.123333</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.940</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540124</td>\n",
       "      <td>2.440</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.093333</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.314333</td>\n",
       "      <td>4.785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540125</td>\n",
       "      <td>1.320</td>\n",
       "      <td>3.475</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.461714</td>\n",
       "      <td>0.893333</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.260</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540126</td>\n",
       "      <td>0.000</td>\n",
       "      <td>2.580</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.221429</td>\n",
       "      <td>2.043333</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.660</td>\n",
       "      <td>1.795</td>\n",
       "      <td>2.746667</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>540127 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          BMDM  MiscOrgans  Liver       Brain       Lung   Heart     CHO  \\\n",
       "0       42.100       0.705    0.0    0.790000   2.180000   0.835  26.996   \n",
       "1        0.000       3.975    1.1    1.608571  16.596667   0.000   0.000   \n",
       "2        2.830       0.705    0.0    1.887429   2.386667   0.000  19.044   \n",
       "3        1.225       3.025    0.0    3.914286   1.633333   0.835   1.680   \n",
       "4        1.695     112.050   29.7  189.308571  50.966667  25.600  20.292   \n",
       "...        ...         ...    ...         ...        ...     ...     ...   \n",
       "540122   1.520       0.890    0.0    0.000000   0.920000   0.000   0.000   \n",
       "540123  11.500       0.645    0.0    0.000000   1.123333   0.000   0.000   \n",
       "540124   2.440       0.000    0.0    0.000000   2.093333   0.000   0.000   \n",
       "540125   1.320       3.475    0.0    0.461714   0.893333   0.000   0.260   \n",
       "540126   0.000       2.580    0.0    0.221429   2.043333   0.000   0.000   \n",
       "\n",
       "        Pancreas  FemaleReproductive   Muscle     Kidney  Spleen  \n",
       "0          0.000               0.630    0.715   2.130000   0.000  \n",
       "1          0.000               4.265    0.545   1.050000   3.385  \n",
       "2          1.490               4.025    1.100   1.870000   1.255  \n",
       "3          2.385               6.000    1.495   1.153333   2.875  \n",
       "4         95.500             136.550  124.150  43.640000  88.550  \n",
       "...          ...                 ...      ...        ...     ...  \n",
       "540122     1.565               0.000    0.000   0.000000  12.450  \n",
       "540123     0.940               0.000    0.000   0.000000   4.000  \n",
       "540124     0.000               0.000    0.000   0.314333   4.785  \n",
       "540125     0.000               0.000    0.000   0.000000   0.000  \n",
       "540126     0.000               1.660    1.795   2.746667   0.000  \n",
       "\n",
       "[540127 rows x 12 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bmdm_map = defaultdict(list)\n",
    "bmdm_map['BMDM'] = ['BMDMwt','BMDM1hKLA']\n",
    "for i in tissue_expr.columns:\n",
    "    if i not in bmdm_map['BMDM']:\n",
    "        bmdm_map[i] = [i]\n",
    "tissue_expr = merge_columns(tissue_expr,bmdm_map)\n",
    "tissue_expr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create distal peaks and get zscore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:20.975419Z",
     "start_time": "2020-05-16T02:21:20.952954Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(540127, 10)\n",
      "(540127, 12)\n"
     ]
    }
   ],
   "source": [
    "print(peak_anno.shape)\n",
    "print(tissue_expr.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:21.343333Z",
     "start_time": "2020-05-16T02:21:20.977634Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "184940\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "ID\n",
       "0         False\n",
       "1         False\n",
       "2         False\n",
       "3          True\n",
       "4         False\n",
       "          ...  \n",
       "540122    False\n",
       "540123    False\n",
       "540124    False\n",
       "540125    False\n",
       "540126     True\n",
       "Length: 540127, dtype: bool"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#(peak_anno[\"Distance to TSS\"] < -2000) &\n",
    "distal_index =  (peak_anno[\"Annotation\"].fillna('').str.contains(\"Intergenic\")) & (tissue_expr.max(axis=1) >= 1) & (np.abs((peak_anno[\"Distance to TSS\"] > 2000)))\n",
    "#distal_index =  (peak_anno[\"Annotation\"].fillna('').str.contains(\"Intergenic\")) & (np.abs((peak_anno[\"Distance to TSS\"] > 2000)))\n",
    "\n",
    "print(distal_index.sum())\n",
    "distal_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T02:21:25.493144Z",
     "start_time": "2020-05-16T02:21:25.273471Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "      <th>Nearest gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>NW_020822602.1</td>\n",
       "      <td>10883205</td>\n",
       "      <td>10883205</td>\n",
       "      <td>+</td>\n",
       "      <td>3.992000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027423054.1</td>\n",
       "      <td>9079.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Gbgt1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>NW_020822567.1</td>\n",
       "      <td>46712264</td>\n",
       "      <td>46712264</td>\n",
       "      <td>-</td>\n",
       "      <td>3.665000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027418086.1</td>\n",
       "      <td>29793.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Sctr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>NW_020822505.1</td>\n",
       "      <td>5449423</td>\n",
       "      <td>5449423</td>\n",
       "      <td>-</td>\n",
       "      <td>2.476000</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027408794.1</td>\n",
       "      <td>86859.0</td>\n",
       "      <td>False</td>\n",
       "      <td>Tpcn2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>NW_020822438.1</td>\n",
       "      <td>19361255</td>\n",
       "      <td>19361255</td>\n",
       "      <td>-</td>\n",
       "      <td>37.450500</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027395681.1</td>\n",
       "      <td>137002.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Larp7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>NW_020822529.1</td>\n",
       "      <td>28068900</td>\n",
       "      <td>28068900</td>\n",
       "      <td>+</td>\n",
       "      <td>3.591714</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027412515.1</td>\n",
       "      <td>59919.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Nfkbiz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540087</td>\n",
       "      <td>NW_020822526.1</td>\n",
       "      <td>7767101</td>\n",
       "      <td>7767101</td>\n",
       "      <td>+</td>\n",
       "      <td>11.712222</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027411731.1</td>\n",
       "      <td>18004.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Snx19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540089</td>\n",
       "      <td>NW_020822529.1</td>\n",
       "      <td>50722923</td>\n",
       "      <td>50722923</td>\n",
       "      <td>+</td>\n",
       "      <td>2.578333</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027412895.1</td>\n",
       "      <td>34275.0</td>\n",
       "      <td>True</td>\n",
       "      <td>Tmem44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540111</td>\n",
       "      <td>NW_020822420.1</td>\n",
       "      <td>8626257</td>\n",
       "      <td>8626257</td>\n",
       "      <td>+</td>\n",
       "      <td>6.926063</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027393079.1</td>\n",
       "      <td>14018.0</td>\n",
       "      <td>False</td>\n",
       "      <td>Bcan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540120</td>\n",
       "      <td>NW_020822507.1</td>\n",
       "      <td>17223168</td>\n",
       "      <td>17223168</td>\n",
       "      <td>+</td>\n",
       "      <td>1.719750</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027410017.1</td>\n",
       "      <td>62472.0</td>\n",
       "      <td>True</td>\n",
       "      <td>LOC107978826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>540126</td>\n",
       "      <td>NW_020822601.1</td>\n",
       "      <td>8668670</td>\n",
       "      <td>8668670</td>\n",
       "      <td>+</td>\n",
       "      <td>3.498750</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>XM_027421498.1</td>\n",
       "      <td>21927.0</td>\n",
       "      <td>False</td>\n",
       "      <td>LOC100757597</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>184940 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Chr     Start       End Strand       Stat  Annotation  \\\n",
       "ID                                                                         \n",
       "3       NW_020822602.1  10883205  10883205      +   3.992000  Intergenic   \n",
       "10      NW_020822567.1  46712264  46712264      -   3.665000  Intergenic   \n",
       "15      NW_020822505.1   5449423   5449423      -   2.476000  Intergenic   \n",
       "20      NW_020822438.1  19361255  19361255      -  37.450500  Intergenic   \n",
       "27      NW_020822529.1  28068900  28068900      +   3.591714  Intergenic   \n",
       "...                ...       ...       ...    ...        ...         ...   \n",
       "540087  NW_020822526.1   7767101   7767101      +  11.712222  Intergenic   \n",
       "540089  NW_020822529.1  50722923  50722923      +   2.578333  Intergenic   \n",
       "540111  NW_020822420.1   8626257   8626257      +   6.926063  Intergenic   \n",
       "540120  NW_020822507.1  17223168  17223168      +   1.719750  Intergenic   \n",
       "540126  NW_020822601.1   8668670   8668670      +   3.498750  Intergenic   \n",
       "\n",
       "           Nearest TSS  Distance to TSS  isSameStrand  Nearest gene  \n",
       "ID                                                                   \n",
       "3       XM_027423054.1           9079.0          True         Gbgt1  \n",
       "10      XM_027418086.1          29793.0          True          Sctr  \n",
       "15      XM_027408794.1          86859.0         False         Tpcn2  \n",
       "20      XM_027395681.1         137002.0          True         Larp7  \n",
       "27      XM_027412515.1          59919.0          True        Nfkbiz  \n",
       "...                ...              ...           ...           ...  \n",
       "540087  XM_027411731.1          18004.0          True         Snx19  \n",
       "540089  XM_027412895.1          34275.0          True        Tmem44  \n",
       "540111  XM_027393079.1          14018.0         False          Bcan  \n",
       "540120  XM_027410017.1          62472.0          True  LOC107978826  \n",
       "540126  XM_027421498.1          21927.0         False  LOC100757597  \n",
       "\n",
       "[184940 rows x 10 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "distal_anno = peak_anno[distal_index]\n",
    "distal_expr = tissue_expr[distal_index]\n",
    "distal_merged = merged_df[distal_index]\n",
    "distal_anno\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2020-05-16T05:22:51.970800Z",
     "start_time": "2020-05-16T02:34:21.147434Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "de_novo_motifs/alt/enhancer_unique/BMDM.uniqueGenes\n",
      "(44128, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/BMDM.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/BMDM.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/BMDM.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/MiscOrgans.uniqueGenes\n",
      "(2353, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/MiscOrgans.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/MiscOrgans.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/MiscOrgans.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/Liver.uniqueGenes\n",
      "(10498, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/Liver.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/Liver.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/Liver.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/Brain.uniqueGenes\n",
      "(16847, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/Brain.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/Brain.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/Brain.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/Lung.uniqueGenes\n",
      "(13012, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/Lung.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/Lung.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/Lung.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/Heart.uniqueGenes\n",
      "(9067, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/Heart.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/Heart.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/Heart.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/CHO.uniqueGenes\n",
      "(20410, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/CHO.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/CHO.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/CHO.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/Pancreas.uniqueGenes\n",
      "(2660, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/Pancreas.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/Pancreas.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/Pancreas.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/FemaleReproductive.uniqueGenes\n",
      "(8267, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/FemaleReproductive.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/FemaleReproductive.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/FemaleReproductive.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/Muscle.uniqueGenes\n",
      "(4931, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/Muscle.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/Muscle.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/Muscle.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/Kidney.uniqueGenes\n",
      "(15054, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/Kidney.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/Kidney.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/Kidney.uniqueGenes.log\n",
      "de_novo_motifs/alt/enhancer_unique/Spleen.uniqueGenes\n",
      "(4955, 10)\n",
      "nohup findMotifsGenome.pl de_novo_motifs/alt/enhancer_unique/Spleen.uniqueGenes.tsv /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna de_novo_motifs/alt/enhancer_unique/Spleen.uniqueGenes -size 200 -len 6,8,10 -p 4  > de_novo_motifs/alt/enhancer_unique/Spleen.uniqueGenes.log\n"
     ]
    }
   ],
   "source": [
    "all_unique_distal = distal_expr[(distal_expr==0).sum(axis=1) == distal_expr.shape[1]-1]\n",
    "for tis in all_unique_distal.columns.values:\n",
    "    tis_out = join(enh_dir, tis + '.uniqueGenes')\n",
    "    tis_peak_out = tis_out+'.tsv'\n",
    "    \n",
    "    print(tis_out)\n",
    "    curr_distal_index = all_unique_distal[all_unique_distal[tis] > 0].index\n",
    "    curr_anno = peak_anno.loc[curr_distal_index]\n",
    "    print(curr_anno.shape)\n",
    "    \n",
    "    curr_anno.to_csv(tis_peak_out,sep='\\t')\n",
    "    wrap_findMotifs.run_findMotifs(tis_peak_out, tis_out, ref_fa, args=None,  bg=None)\n",
    "\n",
    "    #(all_unique_distal>0).sum()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
