{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Pipeline:\n",
    "1. merge_samples\n",
    "\t-merge_peaks\n",
    "\t-annotate_peaks\n",
    "\t-bash: homerTools extract\n",
    "\t-bash: makeTagDirectory\n",
    "\t-make_bedgraph_file\n",
    "2. peaks_break_into_files\n",
    "\n",
    "3. peaks_to_annotation.ipynb\n",
    "\n",
    "4. Merge_into_tissues\n",
    "\n",
    "5. retrieve_tss_peaks\n",
    "\n",
    "6. Rerun Figures2\n",
    "\n",
    "7. ATAC: Create species \n",
    "\n",
    "8. ATAC: run pipeline\n",
    "\n",
    "9. ATAC: Copy folders\n",
    "\n",
    "10. Separate peaks based on expression  (needed?)\n",
    "\n",
    "11. create_output\n",
    "\n",
    "12. Rerun Figure3\n",
    "\n",
    "13. Tag Histograms Compare Experimental to RefSeq\n",
    "\n",
    "14. Nucleotide Frequency\n",
    "\n",
    "15. RerunÂ  CDF of TSS with Samples.ipynb\n",
    "\n",
    "16. Compare RNA_seq\n",
    "\n",
    "17. Igv snapshot\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Wrapper on merge_peaks and anno_peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:09:15.289569Z",
     "start_time": "2019-01-21T23:09:14.427559Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "parameter_file = 'params/params.yaml'\n",
    "import yaml\n",
    "import sys\n",
    "import glob\n",
    "import os\n",
    "import pandas as pd\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "#p = dic2obj(**doc)\n",
    "\n",
    "data_folder = doc['data_folder']\n",
    "tissues = doc['tissues'].split(',')\n",
    "sys.path.append(doc['pipeline_path'])\n",
    "ref_fa = doc['ref_fa']\n",
    "#annotation=doc['annotation_gtf']\n",
    "annotation=doc['annotation']\n",
    "from Homer import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:09:15.373479Z",
     "start_time": "2019-01-21T23:09:15.292117Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Name</th>\n",
       "      <th>Full</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDM1hKLA_ATAC1</td>\n",
       "      <td>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDM1hKLA_ATAC2</td>\n",
       "      <td>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>GRO</td>\n",
       "      <td>BMDM1hKLA_GRO1</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>BMDM1hKLA_GROCap1</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDMwt</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDMwt_ATAC1</td>\n",
       "      <td>CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDMVehic1_ATAC_JHS997_SD_GTGTGGTG_S8_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDMwt</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDMwt_ATAC2</td>\n",
       "      <td>CHBMDMVehic1_ATAC_JHS997_SD_GTGTGGTG_S8_L001_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDMwt</td>\n",
       "      <td>GRO</td>\n",
       "      <td>BMDMwt_GRO1</td>\n",
       "      <td>CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDMwt_GRO_JHS1033_SD_TAGCTT_S48_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDMwt</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>BMDMwt_GROCap1</td>\n",
       "      <td>CHBMDMwt_GRO_JHS1033_SD_TAGCTT_S48_L003_R1_001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBrain_ATAC_JHS992_SD_ACCACTGT_S4_L001_R1_001.fastq.gz</th>\n",
       "      <td>Brain</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>Brain_ATAC1</td>\n",
       "      <td>CHBrain_ATAC_JHS992_SD_ACCACTGT_S4_L001_R1_001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBrain_GRO_JHS1029_SD_CACCGG_S44_L003_R1_001.fastq.gz</th>\n",
       "      <td>Brain</td>\n",
       "      <td>GRO</td>\n",
       "      <td>Brain_GRO1</td>\n",
       "      <td>CHBrain_GRO_JHS1029_SD_CACCGG_S44_L003_R1_001....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBrain_GRO_JHS1036_SD_ATGTCA_S51_L003_R1_001.fastq.gz</th>\n",
       "      <td>Brain</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>Brain_GROCap1</td>\n",
       "      <td>CHBrain_GRO_JHS1036_SD_ATGTCA_S51_L003_R1_001....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBrain_mSTART_JHS1057_SD_ATTCCT_S79_L004_R1_001.fastq.gz</th>\n",
       "      <td>Brain</td>\n",
       "      <td>START</td>\n",
       "      <td>Brain_START1</td>\n",
       "      <td>CHBrain_mSTART_JHS1057_SD_ATTCCT_S79_L004_R1_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBrain7neg1_mSTART_JHS1061_SD_CACGAT_S83_L004_R1_001.fastq.gz</th>\n",
       "      <td>Brain</td>\n",
       "      <td>START</td>\n",
       "      <td>Brain_START2</td>\n",
       "      <td>CHBrain7neg1_mSTART_JHS1061_SD_CACGAT_S83_L004...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBrain_mSTART_JHS1016_SD_GTAGAG_S28_L002_R1_001.fastq.gz</th>\n",
       "      <td>Brain</td>\n",
       "      <td>START</td>\n",
       "      <td>Brain_START3</td>\n",
       "      <td>CHBrain_mSTART_JHS1016_SD_GTAGAG_S28_L002_R1_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBrainnegmaybe7neg1_mSTARTinput_JHS1082_SD_CACGAT_S104_L005_R1_001.fastq.gz</th>\n",
       "      <td>Brain</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>Brain_STARTinput1</td>\n",
       "      <td>CHBrainnegmaybe7neg1_mSTARTinput_JHS1082_SD_CA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBrain_mSTARTinput_JHS1078_SD_ATTCCT_S100_L005_R1_001.fastq.gz</th>\n",
       "      <td>Brain</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>Brain_STARTinput2</td>\n",
       "      <td>CHBrain_mSTARTinput_JHS1078_SD_ATTCCT_S100_L00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHO-ATAC-SD378_S17_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHO_ATAC1</td>\n",
       "      <td>CHO-ATAC-SD378_S17_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hamster-CHO-ATAC-SD332_S22_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHO_ATAC2</td>\n",
       "      <td>Hamster-CHO-ATAC-SD332_S22_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHO-ATAC-SD380_S18_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHO_ATAC3</td>\n",
       "      <td>CHO-ATAC-SD380_S18_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hamster-CHO-ATAC-SD335_S23_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CHO_ATAC4</td>\n",
       "      <td>Hamster-CHO-ATAC-SD335_S23_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHO-GRO-SD370_S9_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>GRO</td>\n",
       "      <td>CHO_GRO1</td>\n",
       "      <td>CHO-GRO-SD370_S9_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHO-GRO-SD369_S8_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>GRO</td>\n",
       "      <td>CHO_GRO2</td>\n",
       "      <td>CHO-GRO-SD369_S8_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHO-5GRO-SD367_S6_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>CHO_GROCap1</td>\n",
       "      <td>CHO-5GRO-SD367_S6_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHO-5GRO-SD371_S10_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>CHO_GROCap2</td>\n",
       "      <td>CHO-5GRO-SD371_S10_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHO--mSTART-JHS823_S21_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>START</td>\n",
       "      <td>CHO_START1</td>\n",
       "      <td>CHO--mSTART-JHS823_S21_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHO--mSTART_input-JHS802_S11_R1_001.fastq.gz</th>\n",
       "      <td>CHO</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>CHO_STARTinput1</td>\n",
       "      <td>CHO--mSTART_input-JHS802_S11_R1_001.fastq.gz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHFemaleReproductive_mSTART_JHS1011_SD_TAGCTT_S23_L002_R1_001.fastq.gz</th>\n",
       "      <td>FemaleReproductive</td>\n",
       "      <td>START</td>\n",
       "      <td>FemaleReproductive_START1</td>\n",
       "      <td>CHFemaleReproductive_mSTART_JHS1011_SD_TAGCTT_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHFemaleReproductive_mSTART_JHS1052_SD_CGTACG_S74_L004_R1_001.fastq.gz</th>\n",
       "      <td>FemaleReproductive</td>\n",
       "      <td>START</td>\n",
       "      <td>FemaleReproductive_START2</td>\n",
       "      <td>CHFemaleReproductive_mSTART_JHS1052_SD_CGTACG_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHFemaleReproductive_mSTARTinput_JHS1073_SD_CGTACG_S95_L005_R1_001.fastq.gz</th>\n",
       "      <td>FemaleReproductive</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>FemaleReproductive_STARTinput1</td>\n",
       "      <td>CHFemaleReproductive_mSTARTinput_JHS1073_SD_CG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHHeart_mSTART_JHS1056_SD_ATGAGC_S78_L004_R1_001.fastq.gz</th>\n",
       "      <td>Heart</td>\n",
       "      <td>START</td>\n",
       "      <td>Heart_START1</td>\n",
       "      <td>CHHeart_mSTART_JHS1056_SD_ATGAGC_S78_L004_R1_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHKidney_mSTART_JHS1022_SD_GAGTGG_S34_L002_R1_001.fastq.gz</th>\n",
       "      <td>Kidney</td>\n",
       "      <td>START</td>\n",
       "      <td>Kidney_START2</td>\n",
       "      <td>CHKidney_mSTART_JHS1022_SD_GAGTGG_S34_L002_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHKidney_mSTARTinput_JHS1080_SD_CAACTA_S102_L005_R1_001.fastq.gz</th>\n",
       "      <td>Kidney</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>Kidney_STARTinput1</td>\n",
       "      <td>CHKidney_mSTARTinput_JHS1080_SD_CAACTA_S102_L0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLiver_ATAC_JHS994_SD_CCGTTTGT_S6_L001_R1_001.fastq.gz</th>\n",
       "      <td>Liver</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>Liver_ATAC1</td>\n",
       "      <td>CHLiver_ATAC_JHS994_SD_CCGTTTGT_S6_L001_R1_001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLiver_GRO_JHS1031_SD_TAATCG_S46_L003_R1_001.fastq.gz</th>\n",
       "      <td>Liver</td>\n",
       "      <td>GRO</td>\n",
       "      <td>Liver_GRO1</td>\n",
       "      <td>CHLiver_GRO_JHS1031_SD_TAATCG_S46_L003_R1_001....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLiver_GRO_JHS1038_SD_GTAGAG_S53_L003_R1_001.fastq.gz</th>\n",
       "      <td>Liver</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>Liver_GROCap1</td>\n",
       "      <td>CHLiver_GRO_JHS1038_SD_GTAGAG_S53_L003_R1_001....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLiver_mSTART_JHS1055_SD_ACTGAT_S77_L004_R1_001.fastq.gz</th>\n",
       "      <td>Liver</td>\n",
       "      <td>START</td>\n",
       "      <td>Liver_START1</td>\n",
       "      <td>CHLiver_mSTART_JHS1055_SD_ACTGAT_S77_L004_R1_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLiver_mSTART_JHS1015_SD_CCGTCC_S27_L002_R1_001.fastq.gz</th>\n",
       "      <td>Liver</td>\n",
       "      <td>START</td>\n",
       "      <td>Liver_START2</td>\n",
       "      <td>CHLiver_mSTART_JHS1015_SD_CCGTCC_S27_L002_R1_0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLiver_mSTARTinput_JHS1076_SD_ACTGAT_S98_L005_R1_001.fastq.gz</th>\n",
       "      <td>Liver</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>Liver_STARTinput1</td>\n",
       "      <td>CHLiver_mSTARTinput_JHS1076_SD_ACTGAT_S98_L005...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLung_ATAC_JHS993_SD_TGGATCTG_S5_L001_R1_001.fastq.gz</th>\n",
       "      <td>Lung</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>Lung_ATAC1</td>\n",
       "      <td>CHLung_ATAC_JHS993_SD_TGGATCTG_S5_L001_R1_001....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLung_GRO_JHS1030_SD_CACGAT_S45_L003_R1_001.fastq.gz</th>\n",
       "      <td>Lung</td>\n",
       "      <td>GRO</td>\n",
       "      <td>Lung_GRO1</td>\n",
       "      <td>CHLung_GRO_JHS1030_SD_CACGAT_S45_L003_R1_001.f...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLung_GRO_JHS1037_SD_CCGTCC_S52_L003_R1_001.fastq.gz</th>\n",
       "      <td>Lung</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>Lung_GROCap1</td>\n",
       "      <td>CHLung_GRO_JHS1037_SD_CCGTCC_S52_L003_R1_001.f...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLung_mSTART_JHS1054_SD_GGTAGC_S76_L004_R1_001.fastq.gz</th>\n",
       "      <td>Lung</td>\n",
       "      <td>START</td>\n",
       "      <td>Lung_START1</td>\n",
       "      <td>CHLung_mSTART_JHS1054_SD_GGTAGC_S76_L004_R1_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLung_mSTART_JHS1020_SD_GTTTCG_S32_L002_R1_001.fastq.gz</th>\n",
       "      <td>Lung</td>\n",
       "      <td>START</td>\n",
       "      <td>Lung_START2</td>\n",
       "      <td>CHLung_mSTART_JHS1020_SD_GTTTCG_S32_L002_R1_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHLung_mSTARTinput_JHS1075_SD_GGTAGC_S97_L005_R1_001.fastq.gz</th>\n",
       "      <td>Lung</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>Lung_STARTinput1</td>\n",
       "      <td>CHLung_mSTARTinput_JHS1075_SD_GGTAGC_S97_L005_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHMiscOrgans_mSTART_JHS1060_SD_CACCGG_S82_L004_R1_001.fastq.gz</th>\n",
       "      <td>MiscOrgans</td>\n",
       "      <td>START</td>\n",
       "      <td>MiscOrgans_START1</td>\n",
       "      <td>CHMiscOrgans_mSTART_JHS1060_SD_CACCGG_S82_L004...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHMiscOrgans_mSTART_JHS1018_SD_GTGAAA_S30_L002_R1_001.fastq.gz</th>\n",
       "      <td>MiscOrgans</td>\n",
       "      <td>START</td>\n",
       "      <td>MiscOrgans_START2</td>\n",
       "      <td>CHMiscOrgans_mSTART_JHS1018_SD_GTGAAA_S30_L002...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHMiscOrgans_mSTARTinput_JHS1081_SD_CACCGG_S103_L005_R1_001.fastq.gz</th>\n",
       "      <td>MiscOrgans</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>MiscOrgans_STARTinput1</td>\n",
       "      <td>CHMiscOrgans_mSTARTinput_JHS1081_SD_CACCGG_S10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHMuscle_mSTART_JHS1019_SD_GTGGCC_S31_L002_R1_001.fastq.gz</th>\n",
       "      <td>Muscle</td>\n",
       "      <td>START</td>\n",
       "      <td>Muscle_START1</td>\n",
       "      <td>CHMuscle_mSTART_JHS1019_SD_GTGGCC_S31_L002_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHMuscle_mSTART_JHS1058_SD_CAAAAG_S80_L004_R1_001.fastq.gz</th>\n",
       "      <td>Muscle</td>\n",
       "      <td>START</td>\n",
       "      <td>Muscle_START2</td>\n",
       "      <td>CHMuscle_mSTART_JHS1058_SD_CAAAAG_S80_L004_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHMuscle_mSTARTinput_JHS1079_SD_CAAAAG_S101_L005_R1_001.fastq.gz</th>\n",
       "      <td>Muscle</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>Muscle_STARTinput1</td>\n",
       "      <td>CHMuscle_mSTARTinput_JHS1079_SD_CAAAAG_S101_L0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHPancreas_mSTART_JHS1014_SD_ATGTCA_S26_L002_R1_001.fastq.gz</th>\n",
       "      <td>Pancreas</td>\n",
       "      <td>START</td>\n",
       "      <td>Pancreas_START1</td>\n",
       "      <td>CHPancreas_mSTART_JHS1014_SD_ATGTCA_S26_L002_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHPancreas_mSTART_JHS1051_SD_GTTTCG_S73_L004_R1_001.fastq.gz</th>\n",
       "      <td>Pancreas</td>\n",
       "      <td>START</td>\n",
       "      <td>Pancreas_START2</td>\n",
       "      <td>CHPancreas_mSTART_JHS1051_SD_GTTTCG_S73_L004_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHPancreas_mSTARTinput_JHS1072_SD_GTTTCG_S94_L005_R1_001.fastq.gz</th>\n",
       "      <td>Pancreas</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>Pancreas_STARTinput1</td>\n",
       "      <td>CHPancreas_mSTARTinput_JHS1072_SD_GTTTCG_S94_L...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHSkin_mSTART_JHS1021_SD_CGTACG_S33_L002_R1_001.fastq.gz</th>\n",
       "      <td>Skin</td>\n",
       "      <td>START</td>\n",
       "      <td>Skin_START1</td>\n",
       "      <td>CHSkin_mSTART_JHS1021_SD_CGTACG_S33_L002_R1_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHSpleenBlood_ATAC_JHS995_SD_TGCTGGGT_S7_L001_R1_001.fastq.gz</th>\n",
       "      <td>Spleen</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>Spleen_ATAC1</td>\n",
       "      <td>CHSpleenBlood_ATAC_JHS995_SD_TGCTGGGT_S7_L001_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBloondnegSpleen_GRO_JHS1032_SD_TCCCGA_S47_L003_R1_001.fastq (3).gz</th>\n",
       "      <td>Spleen</td>\n",
       "      <td>GRO</td>\n",
       "      <td>Spleen_GRO1</td>\n",
       "      <td>CHBloondnegSpleen_GRO_JHS1032_SD_TCCCGA_S47_L0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHSpleednegBlood_mSTART_JHS1017_SD_GTCCGC_S29_L002_R1_001.fastq.gz</th>\n",
       "      <td>Spleen</td>\n",
       "      <td>START</td>\n",
       "      <td>Spleen_START1</td>\n",
       "      <td>CHSpleednegBlood_mSTART_JHS1017_SD_GTCCGC_S29_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHSpleen_mSTART_JHS1053_SD_GAGTGG_S75_L004_R1_001.fastq.gz</th>\n",
       "      <td>Spleen</td>\n",
       "      <td>START</td>\n",
       "      <td>Spleen_START2</td>\n",
       "      <td>CHSpleen_mSTART_JHS1053_SD_GAGTGG_S75_L004_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHSpleen_mSTARTinput_JHS1074_SD_GAGTGG_S96_L005_R1_001.fastq.gz</th>\n",
       "      <td>Spleen</td>\n",
       "      <td>STARTinput</td>\n",
       "      <td>Spleen_STARTinput1</td>\n",
       "      <td>CHSpleen_mSTARTinput_JHS1074_SD_GAGTGG_S96_L00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHTotal_mSTART_JHS1023_SD_GGTAGC_S35_L002_R1_001.fastq.gz</th>\n",
       "      <td>Total</td>\n",
       "      <td>START</td>\n",
       "      <td>Total_START1</td>\n",
       "      <td>CHTotal_mSTART_JHS1023_SD_GGTAGC_S35_L002_R1_0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67 rows Ã 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                Tissue  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...           BMDM1hKLA   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...           BMDM1hKLA   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...           BMDM1hKLA   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...           BMDM1hKLA   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...              BMDMwt   \n",
       "CHBMDMVehic1_ATAC_JHS997_SD_GTGTGGTG_S8_L001_R1...              BMDMwt   \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....              BMDMwt   \n",
       "CHBMDMwt_GRO_JHS1033_SD_TAGCTT_S48_L003_R1_001....              BMDMwt   \n",
       "CHBrain_ATAC_JHS992_SD_ACCACTGT_S4_L001_R1_001....               Brain   \n",
       "CHBrain_GRO_JHS1029_SD_CACCGG_S44_L003_R1_001.f...               Brain   \n",
       "CHBrain_GRO_JHS1036_SD_ATGTCA_S51_L003_R1_001.f...               Brain   \n",
       "CHBrain_mSTART_JHS1057_SD_ATTCCT_S79_L004_R1_00...               Brain   \n",
       "CHBrain7neg1_mSTART_JHS1061_SD_CACGAT_S83_L004_...               Brain   \n",
       "CHBrain_mSTART_JHS1016_SD_GTAGAG_S28_L002_R1_00...               Brain   \n",
       "CHBrainnegmaybe7neg1_mSTARTinput_JHS1082_SD_CAC...               Brain   \n",
       "CHBrain_mSTARTinput_JHS1078_SD_ATTCCT_S100_L005...               Brain   \n",
       "CHO-ATAC-SD378_S17_R1_001.fastq.gz                                 CHO   \n",
       "Hamster-CHO-ATAC-SD332_S22_R1_001.fastq.gz                         CHO   \n",
       "CHO-ATAC-SD380_S18_R1_001.fastq.gz                                 CHO   \n",
       "Hamster-CHO-ATAC-SD335_S23_R1_001.fastq.gz                         CHO   \n",
       "CHO-GRO-SD370_S9_R1_001.fastq.gz                                   CHO   \n",
       "CHO-GRO-SD369_S8_R1_001.fastq.gz                                   CHO   \n",
       "CHO-5GRO-SD367_S6_R1_001.fastq.gz                                  CHO   \n",
       "CHO-5GRO-SD371_S10_R1_001.fastq.gz                                 CHO   \n",
       "CHO--mSTART-JHS823_S21_R1_001.fastq.gz                             CHO   \n",
       "CHO--mSTART_input-JHS802_S11_R1_001.fastq.gz                       CHO   \n",
       "CHFemaleReproductive_mSTART_JHS1011_SD_TAGCTT_S...  FemaleReproductive   \n",
       "CHFemaleReproductive_mSTART_JHS1052_SD_CGTACG_S...  FemaleReproductive   \n",
       "CHFemaleReproductive_mSTARTinput_JHS1073_SD_CGT...  FemaleReproductive   \n",
       "CHHeart_mSTART_JHS1056_SD_ATGAGC_S78_L004_R1_00...               Heart   \n",
       "...                                                                ...   \n",
       "CHKidney_mSTART_JHS1022_SD_GAGTGG_S34_L002_R1_0...              Kidney   \n",
       "CHKidney_mSTARTinput_JHS1080_SD_CAACTA_S102_L00...              Kidney   \n",
       "CHLiver_ATAC_JHS994_SD_CCGTTTGT_S6_L001_R1_001....               Liver   \n",
       "CHLiver_GRO_JHS1031_SD_TAATCG_S46_L003_R1_001.f...               Liver   \n",
       "CHLiver_GRO_JHS1038_SD_GTAGAG_S53_L003_R1_001.f...               Liver   \n",
       "CHLiver_mSTART_JHS1055_SD_ACTGAT_S77_L004_R1_00...               Liver   \n",
       "CHLiver_mSTART_JHS1015_SD_CCGTCC_S27_L002_R1_00...               Liver   \n",
       "CHLiver_mSTARTinput_JHS1076_SD_ACTGAT_S98_L005_...               Liver   \n",
       "CHLung_ATAC_JHS993_SD_TGGATCTG_S5_L001_R1_001.f...                Lung   \n",
       "CHLung_GRO_JHS1030_SD_CACGAT_S45_L003_R1_001.fa...                Lung   \n",
       "CHLung_GRO_JHS1037_SD_CCGTCC_S52_L003_R1_001.fa...                Lung   \n",
       "CHLung_mSTART_JHS1054_SD_GGTAGC_S76_L004_R1_001...                Lung   \n",
       "CHLung_mSTART_JHS1020_SD_GTTTCG_S32_L002_R1_001...                Lung   \n",
       "CHLung_mSTARTinput_JHS1075_SD_GGTAGC_S97_L005_R...                Lung   \n",
       "CHMiscOrgans_mSTART_JHS1060_SD_CACCGG_S82_L004_...          MiscOrgans   \n",
       "CHMiscOrgans_mSTART_JHS1018_SD_GTGAAA_S30_L002_...          MiscOrgans   \n",
       "CHMiscOrgans_mSTARTinput_JHS1081_SD_CACCGG_S103...          MiscOrgans   \n",
       "CHMuscle_mSTART_JHS1019_SD_GTGGCC_S31_L002_R1_0...              Muscle   \n",
       "CHMuscle_mSTART_JHS1058_SD_CAAAAG_S80_L004_R1_0...              Muscle   \n",
       "CHMuscle_mSTARTinput_JHS1079_SD_CAAAAG_S101_L00...              Muscle   \n",
       "CHPancreas_mSTART_JHS1014_SD_ATGTCA_S26_L002_R1...            Pancreas   \n",
       "CHPancreas_mSTART_JHS1051_SD_GTTTCG_S73_L004_R1...            Pancreas   \n",
       "CHPancreas_mSTARTinput_JHS1072_SD_GTTTCG_S94_L0...            Pancreas   \n",
       "CHSkin_mSTART_JHS1021_SD_CGTACG_S33_L002_R1_001...                Skin   \n",
       "CHSpleenBlood_ATAC_JHS995_SD_TGCTGGGT_S7_L001_R...              Spleen   \n",
       "CHBloondnegSpleen_GRO_JHS1032_SD_TCCCGA_S47_L00...              Spleen   \n",
       "CHSpleednegBlood_mSTART_JHS1017_SD_GTCCGC_S29_L...              Spleen   \n",
       "CHSpleen_mSTART_JHS1053_SD_GAGTGG_S75_L004_R1_0...              Spleen   \n",
       "CHSpleen_mSTARTinput_JHS1074_SD_GAGTGG_S96_L005...              Spleen   \n",
       "CHTotal_mSTART_JHS1023_SD_GGTAGC_S35_L002_R1_00...               Total   \n",
       "\n",
       "                                                    Experiment  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...        ATAC   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...        ATAC   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...         GRO   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...      GROCap   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...        ATAC   \n",
       "CHBMDMVehic1_ATAC_JHS997_SD_GTGTGGTG_S8_L001_R1...        ATAC   \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....         GRO   \n",
       "CHBMDMwt_GRO_JHS1033_SD_TAGCTT_S48_L003_R1_001....      GROCap   \n",
       "CHBrain_ATAC_JHS992_SD_ACCACTGT_S4_L001_R1_001....        ATAC   \n",
       "CHBrain_GRO_JHS1029_SD_CACCGG_S44_L003_R1_001.f...         GRO   \n",
       "CHBrain_GRO_JHS1036_SD_ATGTCA_S51_L003_R1_001.f...      GROCap   \n",
       "CHBrain_mSTART_JHS1057_SD_ATTCCT_S79_L004_R1_00...       START   \n",
       "CHBrain7neg1_mSTART_JHS1061_SD_CACGAT_S83_L004_...       START   \n",
       "CHBrain_mSTART_JHS1016_SD_GTAGAG_S28_L002_R1_00...       START   \n",
       "CHBrainnegmaybe7neg1_mSTARTinput_JHS1082_SD_CAC...  STARTinput   \n",
       "CHBrain_mSTARTinput_JHS1078_SD_ATTCCT_S100_L005...  STARTinput   \n",
       "CHO-ATAC-SD378_S17_R1_001.fastq.gz                        ATAC   \n",
       "Hamster-CHO-ATAC-SD332_S22_R1_001.fastq.gz                ATAC   \n",
       "CHO-ATAC-SD380_S18_R1_001.fastq.gz                        ATAC   \n",
       "Hamster-CHO-ATAC-SD335_S23_R1_001.fastq.gz                ATAC   \n",
       "CHO-GRO-SD370_S9_R1_001.fastq.gz                           GRO   \n",
       "CHO-GRO-SD369_S8_R1_001.fastq.gz                           GRO   \n",
       "CHO-5GRO-SD367_S6_R1_001.fastq.gz                       GROCap   \n",
       "CHO-5GRO-SD371_S10_R1_001.fastq.gz                      GROCap   \n",
       "CHO--mSTART-JHS823_S21_R1_001.fastq.gz                   START   \n",
       "CHO--mSTART_input-JHS802_S11_R1_001.fastq.gz        STARTinput   \n",
       "CHFemaleReproductive_mSTART_JHS1011_SD_TAGCTT_S...       START   \n",
       "CHFemaleReproductive_mSTART_JHS1052_SD_CGTACG_S...       START   \n",
       "CHFemaleReproductive_mSTARTinput_JHS1073_SD_CGT...  STARTinput   \n",
       "CHHeart_mSTART_JHS1056_SD_ATGAGC_S78_L004_R1_00...       START   \n",
       "...                                                        ...   \n",
       "CHKidney_mSTART_JHS1022_SD_GAGTGG_S34_L002_R1_0...       START   \n",
       "CHKidney_mSTARTinput_JHS1080_SD_CAACTA_S102_L00...  STARTinput   \n",
       "CHLiver_ATAC_JHS994_SD_CCGTTTGT_S6_L001_R1_001....        ATAC   \n",
       "CHLiver_GRO_JHS1031_SD_TAATCG_S46_L003_R1_001.f...         GRO   \n",
       "CHLiver_GRO_JHS1038_SD_GTAGAG_S53_L003_R1_001.f...      GROCap   \n",
       "CHLiver_mSTART_JHS1055_SD_ACTGAT_S77_L004_R1_00...       START   \n",
       "CHLiver_mSTART_JHS1015_SD_CCGTCC_S27_L002_R1_00...       START   \n",
       "CHLiver_mSTARTinput_JHS1076_SD_ACTGAT_S98_L005_...  STARTinput   \n",
       "CHLung_ATAC_JHS993_SD_TGGATCTG_S5_L001_R1_001.f...        ATAC   \n",
       "CHLung_GRO_JHS1030_SD_CACGAT_S45_L003_R1_001.fa...         GRO   \n",
       "CHLung_GRO_JHS1037_SD_CCGTCC_S52_L003_R1_001.fa...      GROCap   \n",
       "CHLung_mSTART_JHS1054_SD_GGTAGC_S76_L004_R1_001...       START   \n",
       "CHLung_mSTART_JHS1020_SD_GTTTCG_S32_L002_R1_001...       START   \n",
       "CHLung_mSTARTinput_JHS1075_SD_GGTAGC_S97_L005_R...  STARTinput   \n",
       "CHMiscOrgans_mSTART_JHS1060_SD_CACCGG_S82_L004_...       START   \n",
       "CHMiscOrgans_mSTART_JHS1018_SD_GTGAAA_S30_L002_...       START   \n",
       "CHMiscOrgans_mSTARTinput_JHS1081_SD_CACCGG_S103...  STARTinput   \n",
       "CHMuscle_mSTART_JHS1019_SD_GTGGCC_S31_L002_R1_0...       START   \n",
       "CHMuscle_mSTART_JHS1058_SD_CAAAAG_S80_L004_R1_0...       START   \n",
       "CHMuscle_mSTARTinput_JHS1079_SD_CAAAAG_S101_L00...  STARTinput   \n",
       "CHPancreas_mSTART_JHS1014_SD_ATGTCA_S26_L002_R1...       START   \n",
       "CHPancreas_mSTART_JHS1051_SD_GTTTCG_S73_L004_R1...       START   \n",
       "CHPancreas_mSTARTinput_JHS1072_SD_GTTTCG_S94_L0...  STARTinput   \n",
       "CHSkin_mSTART_JHS1021_SD_CGTACG_S33_L002_R1_001...       START   \n",
       "CHSpleenBlood_ATAC_JHS995_SD_TGCTGGGT_S7_L001_R...        ATAC   \n",
       "CHBloondnegSpleen_GRO_JHS1032_SD_TCCCGA_S47_L00...         GRO   \n",
       "CHSpleednegBlood_mSTART_JHS1017_SD_GTCCGC_S29_L...       START   \n",
       "CHSpleen_mSTART_JHS1053_SD_GAGTGG_S75_L004_R1_0...       START   \n",
       "CHSpleen_mSTARTinput_JHS1074_SD_GAGTGG_S96_L005...  STARTinput   \n",
       "CHTotal_mSTART_JHS1023_SD_GGTAGC_S35_L002_R1_00...       START   \n",
       "\n",
       "                                                                              Name  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...                 BMDM1hKLA_ATAC1   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...                 BMDM1hKLA_ATAC2   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...                  BMDM1hKLA_GRO1   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...               BMDM1hKLA_GROCap1   \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...                    BMDMwt_ATAC1   \n",
       "CHBMDMVehic1_ATAC_JHS997_SD_GTGTGGTG_S8_L001_R1...                    BMDMwt_ATAC2   \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....                     BMDMwt_GRO1   \n",
       "CHBMDMwt_GRO_JHS1033_SD_TAGCTT_S48_L003_R1_001....                  BMDMwt_GROCap1   \n",
       "CHBrain_ATAC_JHS992_SD_ACCACTGT_S4_L001_R1_001....                     Brain_ATAC1   \n",
       "CHBrain_GRO_JHS1029_SD_CACCGG_S44_L003_R1_001.f...                      Brain_GRO1   \n",
       "CHBrain_GRO_JHS1036_SD_ATGTCA_S51_L003_R1_001.f...                   Brain_GROCap1   \n",
       "CHBrain_mSTART_JHS1057_SD_ATTCCT_S79_L004_R1_00...                    Brain_START1   \n",
       "CHBrain7neg1_mSTART_JHS1061_SD_CACGAT_S83_L004_...                    Brain_START2   \n",
       "CHBrain_mSTART_JHS1016_SD_GTAGAG_S28_L002_R1_00...                    Brain_START3   \n",
       "CHBrainnegmaybe7neg1_mSTARTinput_JHS1082_SD_CAC...               Brain_STARTinput1   \n",
       "CHBrain_mSTARTinput_JHS1078_SD_ATTCCT_S100_L005...               Brain_STARTinput2   \n",
       "CHO-ATAC-SD378_S17_R1_001.fastq.gz                                       CHO_ATAC1   \n",
       "Hamster-CHO-ATAC-SD332_S22_R1_001.fastq.gz                               CHO_ATAC2   \n",
       "CHO-ATAC-SD380_S18_R1_001.fastq.gz                                       CHO_ATAC3   \n",
       "Hamster-CHO-ATAC-SD335_S23_R1_001.fastq.gz                               CHO_ATAC4   \n",
       "CHO-GRO-SD370_S9_R1_001.fastq.gz                                          CHO_GRO1   \n",
       "CHO-GRO-SD369_S8_R1_001.fastq.gz                                          CHO_GRO2   \n",
       "CHO-5GRO-SD367_S6_R1_001.fastq.gz                                      CHO_GROCap1   \n",
       "CHO-5GRO-SD371_S10_R1_001.fastq.gz                                     CHO_GROCap2   \n",
       "CHO--mSTART-JHS823_S21_R1_001.fastq.gz                                  CHO_START1   \n",
       "CHO--mSTART_input-JHS802_S11_R1_001.fastq.gz                       CHO_STARTinput1   \n",
       "CHFemaleReproductive_mSTART_JHS1011_SD_TAGCTT_S...       FemaleReproductive_START1   \n",
       "CHFemaleReproductive_mSTART_JHS1052_SD_CGTACG_S...       FemaleReproductive_START2   \n",
       "CHFemaleReproductive_mSTARTinput_JHS1073_SD_CGT...  FemaleReproductive_STARTinput1   \n",
       "CHHeart_mSTART_JHS1056_SD_ATGAGC_S78_L004_R1_00...                    Heart_START1   \n",
       "...                                                                            ...   \n",
       "CHKidney_mSTART_JHS1022_SD_GAGTGG_S34_L002_R1_0...                   Kidney_START2   \n",
       "CHKidney_mSTARTinput_JHS1080_SD_CAACTA_S102_L00...              Kidney_STARTinput1   \n",
       "CHLiver_ATAC_JHS994_SD_CCGTTTGT_S6_L001_R1_001....                     Liver_ATAC1   \n",
       "CHLiver_GRO_JHS1031_SD_TAATCG_S46_L003_R1_001.f...                      Liver_GRO1   \n",
       "CHLiver_GRO_JHS1038_SD_GTAGAG_S53_L003_R1_001.f...                   Liver_GROCap1   \n",
       "CHLiver_mSTART_JHS1055_SD_ACTGAT_S77_L004_R1_00...                    Liver_START1   \n",
       "CHLiver_mSTART_JHS1015_SD_CCGTCC_S27_L002_R1_00...                    Liver_START2   \n",
       "CHLiver_mSTARTinput_JHS1076_SD_ACTGAT_S98_L005_...               Liver_STARTinput1   \n",
       "CHLung_ATAC_JHS993_SD_TGGATCTG_S5_L001_R1_001.f...                      Lung_ATAC1   \n",
       "CHLung_GRO_JHS1030_SD_CACGAT_S45_L003_R1_001.fa...                       Lung_GRO1   \n",
       "CHLung_GRO_JHS1037_SD_CCGTCC_S52_L003_R1_001.fa...                    Lung_GROCap1   \n",
       "CHLung_mSTART_JHS1054_SD_GGTAGC_S76_L004_R1_001...                     Lung_START1   \n",
       "CHLung_mSTART_JHS1020_SD_GTTTCG_S32_L002_R1_001...                     Lung_START2   \n",
       "CHLung_mSTARTinput_JHS1075_SD_GGTAGC_S97_L005_R...                Lung_STARTinput1   \n",
       "CHMiscOrgans_mSTART_JHS1060_SD_CACCGG_S82_L004_...               MiscOrgans_START1   \n",
       "CHMiscOrgans_mSTART_JHS1018_SD_GTGAAA_S30_L002_...               MiscOrgans_START2   \n",
       "CHMiscOrgans_mSTARTinput_JHS1081_SD_CACCGG_S103...          MiscOrgans_STARTinput1   \n",
       "CHMuscle_mSTART_JHS1019_SD_GTGGCC_S31_L002_R1_0...                   Muscle_START1   \n",
       "CHMuscle_mSTART_JHS1058_SD_CAAAAG_S80_L004_R1_0...                   Muscle_START2   \n",
       "CHMuscle_mSTARTinput_JHS1079_SD_CAAAAG_S101_L00...              Muscle_STARTinput1   \n",
       "CHPancreas_mSTART_JHS1014_SD_ATGTCA_S26_L002_R1...                 Pancreas_START1   \n",
       "CHPancreas_mSTART_JHS1051_SD_GTTTCG_S73_L004_R1...                 Pancreas_START2   \n",
       "CHPancreas_mSTARTinput_JHS1072_SD_GTTTCG_S94_L0...            Pancreas_STARTinput1   \n",
       "CHSkin_mSTART_JHS1021_SD_CGTACG_S33_L002_R1_001...                     Skin_START1   \n",
       "CHSpleenBlood_ATAC_JHS995_SD_TGCTGGGT_S7_L001_R...                    Spleen_ATAC1   \n",
       "CHBloondnegSpleen_GRO_JHS1032_SD_TCCCGA_S47_L00...                     Spleen_GRO1   \n",
       "CHSpleednegBlood_mSTART_JHS1017_SD_GTCCGC_S29_L...                   Spleen_START1   \n",
       "CHSpleen_mSTART_JHS1053_SD_GAGTGG_S75_L004_R1_0...                   Spleen_START2   \n",
       "CHSpleen_mSTARTinput_JHS1074_SD_GAGTGG_S96_L005...              Spleen_STARTinput1   \n",
       "CHTotal_mSTART_JHS1023_SD_GGTAGC_S35_L002_R1_00...                    Total_START1   \n",
       "\n",
       "                                                                                                 Full  \n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...  \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...  \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...  \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...  \n",
       "CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_R...  CHBMDMVehic2_ATAC_JHS999_SD_TGGTCACA_S10_L001_...  \n",
       "CHBMDMVehic1_ATAC_JHS997_SD_GTGTGGTG_S8_L001_R1...  CHBMDMVehic1_ATAC_JHS997_SD_GTGTGGTG_S8_L001_R...  \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....  CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001...  \n",
       "CHBMDMwt_GRO_JHS1033_SD_TAGCTT_S48_L003_R1_001....  CHBMDMwt_GRO_JHS1033_SD_TAGCTT_S48_L003_R1_001...  \n",
       "CHBrain_ATAC_JHS992_SD_ACCACTGT_S4_L001_R1_001....  CHBrain_ATAC_JHS992_SD_ACCACTGT_S4_L001_R1_001...  \n",
       "CHBrain_GRO_JHS1029_SD_CACCGG_S44_L003_R1_001.f...  CHBrain_GRO_JHS1029_SD_CACCGG_S44_L003_R1_001....  \n",
       "CHBrain_GRO_JHS1036_SD_ATGTCA_S51_L003_R1_001.f...  CHBrain_GRO_JHS1036_SD_ATGTCA_S51_L003_R1_001....  \n",
       "CHBrain_mSTART_JHS1057_SD_ATTCCT_S79_L004_R1_00...  CHBrain_mSTART_JHS1057_SD_ATTCCT_S79_L004_R1_0...  \n",
       "CHBrain7neg1_mSTART_JHS1061_SD_CACGAT_S83_L004_...  CHBrain7neg1_mSTART_JHS1061_SD_CACGAT_S83_L004...  \n",
       "CHBrain_mSTART_JHS1016_SD_GTAGAG_S28_L002_R1_00...  CHBrain_mSTART_JHS1016_SD_GTAGAG_S28_L002_R1_0...  \n",
       "CHBrainnegmaybe7neg1_mSTARTinput_JHS1082_SD_CAC...  CHBrainnegmaybe7neg1_mSTARTinput_JHS1082_SD_CA...  \n",
       "CHBrain_mSTARTinput_JHS1078_SD_ATTCCT_S100_L005...  CHBrain_mSTARTinput_JHS1078_SD_ATTCCT_S100_L00...  \n",
       "CHO-ATAC-SD378_S17_R1_001.fastq.gz                                 CHO-ATAC-SD378_S17_R1_001.fastq.gz  \n",
       "Hamster-CHO-ATAC-SD332_S22_R1_001.fastq.gz                 Hamster-CHO-ATAC-SD332_S22_R1_001.fastq.gz  \n",
       "CHO-ATAC-SD380_S18_R1_001.fastq.gz                                 CHO-ATAC-SD380_S18_R1_001.fastq.gz  \n",
       "Hamster-CHO-ATAC-SD335_S23_R1_001.fastq.gz                 Hamster-CHO-ATAC-SD335_S23_R1_001.fastq.gz  \n",
       "CHO-GRO-SD370_S9_R1_001.fastq.gz                                     CHO-GRO-SD370_S9_R1_001.fastq.gz  \n",
       "CHO-GRO-SD369_S8_R1_001.fastq.gz                                     CHO-GRO-SD369_S8_R1_001.fastq.gz  \n",
       "CHO-5GRO-SD367_S6_R1_001.fastq.gz                                   CHO-5GRO-SD367_S6_R1_001.fastq.gz  \n",
       "CHO-5GRO-SD371_S10_R1_001.fastq.gz                                 CHO-5GRO-SD371_S10_R1_001.fastq.gz  \n",
       "CHO--mSTART-JHS823_S21_R1_001.fastq.gz                         CHO--mSTART-JHS823_S21_R1_001.fastq.gz  \n",
       "CHO--mSTART_input-JHS802_S11_R1_001.fastq.gz             CHO--mSTART_input-JHS802_S11_R1_001.fastq.gz  \n",
       "CHFemaleReproductive_mSTART_JHS1011_SD_TAGCTT_S...  CHFemaleReproductive_mSTART_JHS1011_SD_TAGCTT_...  \n",
       "CHFemaleReproductive_mSTART_JHS1052_SD_CGTACG_S...  CHFemaleReproductive_mSTART_JHS1052_SD_CGTACG_...  \n",
       "CHFemaleReproductive_mSTARTinput_JHS1073_SD_CGT...  CHFemaleReproductive_mSTARTinput_JHS1073_SD_CG...  \n",
       "CHHeart_mSTART_JHS1056_SD_ATGAGC_S78_L004_R1_00...  CHHeart_mSTART_JHS1056_SD_ATGAGC_S78_L004_R1_0...  \n",
       "...                                                                                               ...  \n",
       "CHKidney_mSTART_JHS1022_SD_GAGTGG_S34_L002_R1_0...  CHKidney_mSTART_JHS1022_SD_GAGTGG_S34_L002_R1_...  \n",
       "CHKidney_mSTARTinput_JHS1080_SD_CAACTA_S102_L00...  CHKidney_mSTARTinput_JHS1080_SD_CAACTA_S102_L0...  \n",
       "CHLiver_ATAC_JHS994_SD_CCGTTTGT_S6_L001_R1_001....  CHLiver_ATAC_JHS994_SD_CCGTTTGT_S6_L001_R1_001...  \n",
       "CHLiver_GRO_JHS1031_SD_TAATCG_S46_L003_R1_001.f...  CHLiver_GRO_JHS1031_SD_TAATCG_S46_L003_R1_001....  \n",
       "CHLiver_GRO_JHS1038_SD_GTAGAG_S53_L003_R1_001.f...  CHLiver_GRO_JHS1038_SD_GTAGAG_S53_L003_R1_001....  \n",
       "CHLiver_mSTART_JHS1055_SD_ACTGAT_S77_L004_R1_00...  CHLiver_mSTART_JHS1055_SD_ACTGAT_S77_L004_R1_0...  \n",
       "CHLiver_mSTART_JHS1015_SD_CCGTCC_S27_L002_R1_00...  CHLiver_mSTART_JHS1015_SD_CCGTCC_S27_L002_R1_0...  \n",
       "CHLiver_mSTARTinput_JHS1076_SD_ACTGAT_S98_L005_...  CHLiver_mSTARTinput_JHS1076_SD_ACTGAT_S98_L005...  \n",
       "CHLung_ATAC_JHS993_SD_TGGATCTG_S5_L001_R1_001.f...  CHLung_ATAC_JHS993_SD_TGGATCTG_S5_L001_R1_001....  \n",
       "CHLung_GRO_JHS1030_SD_CACGAT_S45_L003_R1_001.fa...  CHLung_GRO_JHS1030_SD_CACGAT_S45_L003_R1_001.f...  \n",
       "CHLung_GRO_JHS1037_SD_CCGTCC_S52_L003_R1_001.fa...  CHLung_GRO_JHS1037_SD_CCGTCC_S52_L003_R1_001.f...  \n",
       "CHLung_mSTART_JHS1054_SD_GGTAGC_S76_L004_R1_001...  CHLung_mSTART_JHS1054_SD_GGTAGC_S76_L004_R1_00...  \n",
       "CHLung_mSTART_JHS1020_SD_GTTTCG_S32_L002_R1_001...  CHLung_mSTART_JHS1020_SD_GTTTCG_S32_L002_R1_00...  \n",
       "CHLung_mSTARTinput_JHS1075_SD_GGTAGC_S97_L005_R...  CHLung_mSTARTinput_JHS1075_SD_GGTAGC_S97_L005_...  \n",
       "CHMiscOrgans_mSTART_JHS1060_SD_CACCGG_S82_L004_...  CHMiscOrgans_mSTART_JHS1060_SD_CACCGG_S82_L004...  \n",
       "CHMiscOrgans_mSTART_JHS1018_SD_GTGAAA_S30_L002_...  CHMiscOrgans_mSTART_JHS1018_SD_GTGAAA_S30_L002...  \n",
       "CHMiscOrgans_mSTARTinput_JHS1081_SD_CACCGG_S103...  CHMiscOrgans_mSTARTinput_JHS1081_SD_CACCGG_S10...  \n",
       "CHMuscle_mSTART_JHS1019_SD_GTGGCC_S31_L002_R1_0...  CHMuscle_mSTART_JHS1019_SD_GTGGCC_S31_L002_R1_...  \n",
       "CHMuscle_mSTART_JHS1058_SD_CAAAAG_S80_L004_R1_0...  CHMuscle_mSTART_JHS1058_SD_CAAAAG_S80_L004_R1_...  \n",
       "CHMuscle_mSTARTinput_JHS1079_SD_CAAAAG_S101_L00...  CHMuscle_mSTARTinput_JHS1079_SD_CAAAAG_S101_L0...  \n",
       "CHPancreas_mSTART_JHS1014_SD_ATGTCA_S26_L002_R1...  CHPancreas_mSTART_JHS1014_SD_ATGTCA_S26_L002_R...  \n",
       "CHPancreas_mSTART_JHS1051_SD_GTTTCG_S73_L004_R1...  CHPancreas_mSTART_JHS1051_SD_GTTTCG_S73_L004_R...  \n",
       "CHPancreas_mSTARTinput_JHS1072_SD_GTTTCG_S94_L0...  CHPancreas_mSTARTinput_JHS1072_SD_GTTTCG_S94_L...  \n",
       "CHSkin_mSTART_JHS1021_SD_CGTACG_S33_L002_R1_001...  CHSkin_mSTART_JHS1021_SD_CGTACG_S33_L002_R1_00...  \n",
       "CHSpleenBlood_ATAC_JHS995_SD_TGCTGGGT_S7_L001_R...  CHSpleenBlood_ATAC_JHS995_SD_TGCTGGGT_S7_L001_...  \n",
       "CHBloondnegSpleen_GRO_JHS1032_SD_TCCCGA_S47_L00...  CHBloondnegSpleen_GRO_JHS1032_SD_TCCCGA_S47_L0...  \n",
       "CHSpleednegBlood_mSTART_JHS1017_SD_GTCCGC_S29_L...  CHSpleednegBlood_mSTART_JHS1017_SD_GTCCGC_S29_...  \n",
       "CHSpleen_mSTART_JHS1053_SD_GAGTGG_S75_L004_R1_0...  CHSpleen_mSTART_JHS1053_SD_GAGTGG_S75_L004_R1_...  \n",
       "CHSpleen_mSTARTinput_JHS1074_SD_GAGTGG_S96_L005...  CHSpleen_mSTARTinput_JHS1074_SD_GAGTGG_S96_L00...  \n",
       "CHTotal_mSTART_JHS1023_SD_GGTAGC_S35_L002_R1_00...  CHTotal_mSTART_JHS1023_SD_GGTAGC_S35_L002_R1_0...  \n",
       "\n",
       "[67 rows x 4 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_files = pd.read_csv(\"../raw_metaSamples.tsv\",sep='\\t',index_col=0)\n",
    "meta_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:09:15.380992Z",
     "start_time": "2019-01-21T23:09:15.375649Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['f04_peaks/MiscOrgans_START1_and_MiscOrgans_startinput1', 'f04_peaks/Liver_START2_and_Liver_startinput1', 'f04_peaks/Brain_GROCap1_and_Brain_GRO1', 'f04_peaks/Liver_START1_and_Liver_startinput1', 'f04_peaks/Lung_GROCap1_and_Lung_GRO1', 'f04_peaks/Brain_START3_and_Brain_startinput1', 'f04_peaks/Heart_START1_and_Heart_startinput1', 'f04_peaks/CHO_GROCap2_and_CHO_GRO1', 'f04_peaks/Heart_START2_and_Heart_startinput1', 'f04_peaks/Liver_GROCap1_and_Liver_GRO1', 'f04_peaks/Pancreas_START2_and_Pancreas_startinput1', 'f04_peaks/BMDM1hKLA_GROCap1_and_BMDM1hKLA_GRO1', 'f04_peaks/Pancreas_START1_and_Pancreas_startinput1', 'f04_peaks/FemaleReproductive_START2_and_FemaleReproductive_startinput1', 'f04_peaks/Muscle_START1_and_Muscle_startinput1', 'f04_peaks/Brain_START1_and_Brain_startinput1', 'f04_peaks/Muscle_START2_and_Muscle_startinput1', 'f04_peaks/Brain_START2_and_Brain_startinput2', 'f04_peaks/Brain_START1_and_Brain_startinput2', 'f04_peaks/CHO_GROCap1_and_CHO_GRO2', 'f04_peaks/Brain_START2_and_Brain_startinput1', 'f04_peaks/Lung_START1_and_Lung_startinput1', 'f04_peaks/Brain_START3_and_Brain_startinput2', 'f04_peaks/CHO_GROCap2_and_CHO_GRO2', 'f04_peaks/Kidney_START1_and_Kidney_startinput1', 'f04_peaks/MiscOrgans_START2_and_MiscOrgans_startinput1', 'f04_peaks/Spleen_START1_and_Spleen_startinput1', 'f04_peaks/Lung_START2_and_Lung_startinput1', 'f04_peaks/CHO_START1_and_CHO_startinput1', 'f04_peaks/FemaleReproductive_START1_and_FemaleReproductive_startinput1', 'f04_peaks/Kidney_START2_and_Kidney_startinput1', 'f04_peaks/Kidney_GROCap1_and_Kidney_GRO1', 'f04_peaks/CHO_GROCap1_and_CHO_GRO1', 'f04_peaks/BMDMwt_GROCap1_and_BMDMwt_GRO1', 'f04_peaks/Spleen_START2_and_Spleen_startinput1']\n"
     ]
    }
   ],
   "source": [
    "all_peak_merge_files = glob.glob(\"f04_peaks/*\")\n",
    "print(all_peak_merge_files)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:09:15.387268Z",
     "start_time": "2019-01-21T23:09:15.383000Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "if not os.path.exists(\"Results\"):\n",
    "    os.mkdir(\"Results/\")\n",
    "if not os.path.exists(\"Results/merged/\"):\n",
    "    os.mkdir(\"Results/merged\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:09:15.392312Z",
     "start_time": "2019-01-21T23:09:15.389420Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "output_file = 'Results/merged/samples.merge'                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:12:04.849578Z",
     "start_time": "2019-01-21T23:09:15.394785Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mergePeaks -d given -venn Results/merged/samples.merge_venn -matrix Results/merged/samples.merge_mat -strand f04_peaks/MiscOrgans_START1_and_MiscOrgans_startinput1 f04_peaks/Liver_START2_and_Liver_startinput1 f04_peaks/Brain_GROCap1_and_Brain_GRO1 f04_peaks/Liver_START1_and_Liver_startinput1 f04_peaks/Lung_GROCap1_and_Lung_GRO1 f04_peaks/Brain_START3_and_Brain_startinput1 f04_peaks/Heart_START1_and_Heart_startinput1 f04_peaks/CHO_GROCap2_and_CHO_GRO1 f04_peaks/Heart_START2_and_Heart_startinput1 f04_peaks/Liver_GROCap1_and_Liver_GRO1 f04_peaks/Pancreas_START2_and_Pancreas_startinput1 f04_peaks/BMDM1hKLA_GROCap1_and_BMDM1hKLA_GRO1 f04_peaks/Pancreas_START1_and_Pancreas_startinput1 f04_peaks/FemaleReproductive_START2_and_FemaleReproductive_startinput1 f04_peaks/Muscle_START1_and_Muscle_startinput1 f04_peaks/Brain_START1_and_Brain_startinput1 f04_peaks/Muscle_START2_and_Muscle_startinput1 f04_peaks/Brain_START2_and_Brain_startinput2 f04_peaks/Brain_START1_and_Brain_startinput2 f04_peaks/CHO_GROCap1_and_CHO_GRO2 f04_peaks/Brain_START2_and_Brain_startinput1 f04_peaks/Lung_START1_and_Lung_startinput1 f04_peaks/Brain_START3_and_Brain_startinput2 f04_peaks/CHO_GROCap2_and_CHO_GRO2 f04_peaks/Kidney_START1_and_Kidney_startinput1 f04_peaks/MiscOrgans_START2_and_MiscOrgans_startinput1 f04_peaks/Spleen_START1_and_Spleen_startinput1 f04_peaks/Lung_START2_and_Lung_startinput1 f04_peaks/CHO_START1_and_CHO_startinput1 f04_peaks/FemaleReproductive_START1_and_FemaleReproductive_startinput1 f04_peaks/Kidney_START2_and_Kidney_startinput1 f04_peaks/Kidney_GROCap1_and_Kidney_GRO1 f04_peaks/CHO_GROCap1_and_CHO_GRO1 f04_peaks/BMDMwt_GROCap1_and_BMDMwt_GRO1 f04_peaks/Spleen_START2_and_Spleen_startinput1 > Results/merged/samples.merge\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2827: DtypeWarning: Columns (8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  if self.run_code(code, result):\n"
     ]
    }
   ],
   "source": [
    "merge_peaks(all_peak_merge_files,output_file,dist='given')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Annotate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:15:52.746979Z",
     "start_time": "2019-01-21T23:12:04.852894Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotatePeaks.pl Results/merged/samples.merge /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna -gff /data/isshamie/genome/ncbi_anno_103/alt_CriGri-PICR_top_level.gff3 > Results/merged/samples.merge.anno\n"
     ]
    }
   ],
   "source": [
    "output_anno_file = 'Results/merged/samples.merge.anno' \n",
    "annotate_peaks(peak_file=output_file,output_file=output_anno_file,ref_fa=ref_fa,annotation=annotation)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extract sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:16:01.041071Z",
     "start_time": "2019-01-21T23:15:52.751704Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tExtracting sequences from file: /data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna\n",
      "\tLooking for peak sequences in a single file (/data/isshamie/genome/ncbi_anno_103/GCF_003668045.1_CriGri-PICR_genomic.fna)\n",
      "\tExtracting 1547 sequences from NW_020822366.1\n",
      "\tExtracting 8450 sequences from NW_020822370.1\n",
      "\tExtracting 1435 sequences from NW_020822372.1\n",
      "\tExtracting 1697 sequences from NW_020822373.1\n",
      "\tExtracting 936 sequences from NW_020822374.1\n",
      "\tExtracting 7343 sequences from NW_020822375.1\n",
      "\tExtracting 1590 sequences from NW_020822376.1\n",
      "\tExtracting 1204 sequences from NW_020822377.1\n",
      "\tExtracting 1257 sequences from NW_020822379.1\n",
      "\tExtracting 1747 sequences from NW_020822381.1\n",
      "\tExtracting 739 sequences from NW_020822382.1\n",
      "\tExtracting 547 sequences from NW_020822385.1\n",
      "\tExtracting 1252 sequences from NW_020822388.1\n",
      "\tExtracting 513 sequences from NW_020822390.1\n",
      "\tExtracting 140 sequences from NW_020822391.1\n",
      "\tExtracting 711 sequences from NW_020822393.1\n",
      "\tExtracting 284 sequences from NW_020822394.1\n",
      "\tExtracting 28 sequences from NW_020822395.1\n",
      "\tExtracting 289 sequences from NW_020822396.1\n",
      "\tExtracting 4426 sequences from NW_020822398.1\n",
      "\tExtracting 133 sequences from NW_020822399.1\n",
      "\tExtracting 1 sequences from NW_020822401.1\n",
      "\tExtracting 242 sequences from NW_020822402.1\n",
      "\tExtracting 4368 sequences from NW_020822403.1\n",
      "\tExtracting 23 sequences from NW_020822404.1\n",
      "\tExtracting 28 sequences from NW_020822405.1\n",
      "\tExtracting 7874 sequences from NW_020822406.1\n",
      "\tExtracting 4035 sequences from NW_020822407.1\n",
      "\tExtracting 5 sequences from NW_020822408.1\n",
      "\tExtracting 3657 sequences from NW_020822409.1\n",
      "\tExtracting 5086 sequences from NW_020822410.1\n",
      "\tExtracting 12 sequences from NW_020822411.1\n",
      "\tExtracting 3046 sequences from NW_020822412.1\n",
      "\tExtracting 13 sequences from NW_020822413.1\n",
      "\tExtracting 2 sequences from NW_020822414.1\n",
      "\tExtracting 12552 sequences from NW_020822415.1\n",
      "\tExtracting 42 sequences from NW_020822416.1\n",
      "\tExtracting 3 sequences from NW_020822417.1\n",
      "\tExtracting 1273 sequences from NW_020822419.1\n",
      "\tExtracting 4038 sequences from NW_020822420.1\n",
      "\tExtracting 1968 sequences from NW_020822421.1\n",
      "\tExtracting 4302 sequences from NW_020822423.1\n",
      "\tExtracting 4126 sequences from NW_020822425.1\n",
      "\tExtracting 1225 sequences from NW_020822426.1\n",
      "\tExtracting 1652 sequences from NW_020822427.1\n",
      "\tExtracting 3068 sequences from NW_020822428.1\n",
      "\tExtracting 2778 sequences from NW_020822429.1\n",
      "\tExtracting 3 sequences from NW_020822430.1\n",
      "\tExtracting 6 sequences from NW_020822431.1\n",
      "\tExtracting 1618 sequences from NW_020822432.1\n",
      "\tExtracting 2157 sequences from NW_020822434.1\n",
      "\tExtracting 1363 sequences from NW_020822435.1\n",
      "\tExtracting 1677 sequences from NW_020822436.1\n",
      "\tExtracting 9098 sequences from NW_020822438.1\n",
      "\tExtracting 21219 sequences from NW_020822439.1\n",
      "\tExtracting 1387 sequences from NW_020822440.1\n",
      "\tExtracting 1496 sequences from NW_020822442.1\n",
      "\tExtracting 2644 sequences from NW_020822443.1\n",
      "\tExtracting 990 sequences from NW_020822444.1\n",
      "\tExtracting 1252 sequences from NW_020822446.1\n",
      "\tExtracting 252 sequences from NW_020822449.1\n",
      "\tExtracting 3935 sequences from NW_020822450.1\n",
      "\tExtracting 95 sequences from NW_020822451.1\n",
      "\tExtracting 9838 sequences from NW_020822452.1\n",
      "\tExtracting 13 sequences from NW_020822453.1\n",
      "\tExtracting 2646 sequences from NW_020822454.1\n",
      "\tExtracting 2118 sequences from NW_020822455.1\n",
      "\tExtracting 3 sequences from NW_020822456.1\n",
      "\tExtracting 3219 sequences from NW_020822457.1\n",
      "\tExtracting 3186 sequences from NW_020822458.1\n",
      "\tExtracting 2626 sequences from NW_020822459.1\n",
      "\tExtracting 4394 sequences from NW_020822460.1\n",
      "\tExtracting 11332 sequences from NW_020822461.1\n",
      "\tExtracting 3653 sequences from NW_020822462.1\n",
      "\tExtracting 3106 sequences from NW_020822464.1\n",
      "\tExtracting 3625 sequences from NW_020822465.1\n",
      "\tExtracting 3143 sequences from NW_020822466.1\n",
      "\tExtracting 1778 sequences from NW_020822467.1\n",
      "\tExtracting 8518 sequences from NW_020822468.1\n",
      "\tExtracting 1585 sequences from NW_020822469.1\n",
      "\tExtracting 4155 sequences from NW_020822470.1\n",
      "\tExtracting 1455 sequences from NW_020822471.1\n",
      "\tExtracting 2490 sequences from NW_020822474.1\n",
      "\tExtracting 591 sequences from NW_020822475.1\n",
      "\tExtracting 1837 sequences from NW_020822476.1\n",
      "\tExtracting 3 sequences from NW_020822479.1\n",
      "\tExtracting 1 sequences from NW_020822482.1\n",
      "\tExtracting 676 sequences from NW_020822483.1\n",
      "\tExtracting 422 sequences from NW_020822484.1\n",
      "\tExtracting 6960 sequences from NW_020822486.1\n",
      "\tExtracting 5887 sequences from NW_020822487.1\n",
      "\tExtracting 426 sequences from NW_020822488.1\n",
      "\tExtracting 991 sequences from NW_020822490.1\n",
      "\tExtracting 261 sequences from NW_020822493.1\n",
      "\tExtracting 171 sequences from NW_020822495.1\n",
      "\tExtracting 1 sequences from NW_020822497.1\n",
      "\tExtracting 142 sequences from NW_020822498.1\n",
      "\tExtracting 6933 sequences from NW_020822499.1\n",
      "\tExtracting 19 sequences from NW_020822500.1\n",
      "\tExtracting 7180 sequences from NW_020822501.1\n",
      "\tExtracting 16 sequences from NW_020822502.1\n",
      "\tExtracting 5057 sequences from NW_020822503.1\n",
      "\tExtracting 4566 sequences from NW_020822504.1\n",
      "\tExtracting 6512 sequences from NW_020822505.1\n",
      "\tExtracting 4561 sequences from NW_020822506.1\n",
      "\tExtracting 5568 sequences from NW_020822507.1\n",
      "\tExtracting 4103 sequences from NW_020822508.1\n",
      "\tExtracting 1 sequences from NW_020822510.1\n",
      "\tExtracting 2578 sequences from NW_020822511.1\n",
      "\tExtracting 1541 sequences from NW_020822512.1\n",
      "\tExtracting 2812 sequences from NW_020822513.1\n",
      "\tExtracting 554 sequences from NW_020822518.1\n",
      "\tExtracting 8441 sequences from NW_020822519.1\n",
      "\tExtracting 1019 sequences from NW_020822520.1\n",
      "\tExtracting 1 sequences from NW_020822525.1\n",
      "\tExtracting 6937 sequences from NW_020822526.1\n",
      "\tExtracting 26 sequences from NW_020822528.1\n",
      "\tExtracting 13905 sequences from NW_020822529.1\n",
      "\tExtracting 4686 sequences from NW_020822530.1\n",
      "\tExtracting 4458 sequences from NW_020822531.1\n",
      "\tExtracting 4353 sequences from NW_020822532.1\n",
      "\tExtracting 3946 sequences from NW_020822533.1\n",
      "\tExtracting 1587 sequences from NW_020822534.1\n",
      "\tExtracting 3456 sequences from NW_020822535.1\n",
      "\tExtracting 1 sequences from NW_020822536.1\n",
      "\tExtracting 2962 sequences from NW_020822537.1\n",
      "\tExtracting 2406 sequences from NW_020822538.1\n",
      "\tExtracting 2280 sequences from NW_020822539.1\n",
      "\tExtracting 1 sequences from NW_020822543.1\n",
      "\tExtracting 1037 sequences from NW_020822544.1\n",
      "\tExtracting 1117 sequences from NW_020822545.1\n",
      "\tExtracting 889 sequences from NW_020822550.1\n",
      "\tExtracting 240 sequences from NW_020822552.1\n",
      "\tExtracting 1075 sequences from NW_020822555.1\n",
      "\tExtracting 1557 sequences from NW_020822557.1\n",
      "\tExtracting 1025 sequences from NW_020822558.1\n",
      "\tExtracting 789 sequences from NW_020822559.1\n",
      "\tExtracting 1 sequences from NW_020822561.1\n",
      "\tExtracting 272 sequences from NW_020822563.1\n",
      "\tExtracting 9 sequences from NW_020822566.1\n",
      "\tExtracting 16749 sequences from NW_020822567.1\n",
      "\tExtracting 1 sequences from NW_020822568.1\n",
      "\tExtracting 1 sequences from NW_020822569.1\n",
      "\tExtracting 7359 sequences from NW_020822570.1\n",
      "\tExtracting 2 sequences from NW_020822572.1\n",
      "\tExtracting 2 sequences from NW_020822573.1\n",
      "\tExtracting 4 sequences from NW_020822574.1\n",
      "\tExtracting 5 sequences from NW_020822575.1\n",
      "\tExtracting 2 sequences from NW_020822576.1\n",
      "\tExtracting 5879 sequences from NW_020822577.1\n",
      "\tExtracting 3 sequences from NW_020822578.1\n",
      "\tExtracting 2 sequences from NW_020822579.1\n",
      "\tExtracting 6 sequences from NW_020822580.1\n",
      "\tExtracting 7 sequences from NW_020822581.1\n",
      "\tExtracting 1 sequences from NW_020822584.1\n",
      "\tExtracting 6 sequences from NW_020822585.1\n",
      "\tExtracting 3904 sequences from NW_020822586.1\n",
      "\tExtracting 1 sequences from NW_020822588.1\n",
      "\tExtracting 3 sequences from NW_020822590.1\n",
      "\tExtracting 1039 sequences from NW_020822591.1\n",
      "\tExtracting 1442 sequences from NW_020822592.1\n",
      "\tExtracting 2 sequences from NW_020822593.1\n",
      "\tExtracting 551 sequences from NW_020822595.1\n",
      "\tExtracting 1473 sequences from NW_020822596.1\n",
      "\tExtracting 7603 sequences from NW_020822597.1\n",
      "\tExtracting 1534 sequences from NW_020822598.1\n",
      "\tExtracting 18206 sequences from NW_020822601.1\n",
      "\tExtracting 3513 sequences from NW_020822602.1\n",
      "\tExtracting 3980 sequences from NW_020822603.1\n",
      "\tExtracting 3283 sequences from NW_020822604.1\n",
      "\tExtracting 1497 sequences from NW_020822605.1\n",
      "\tExtracting 2608 sequences from NW_020822606.1\n",
      "\tExtracting 632 sequences from NW_020822607.1\n",
      "\tExtracting 6168 sequences from NW_020822608.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 6 sequences from NW_020822609.1\n",
      "\tExtracting 4363 sequences from NW_020822610.1\n",
      "\tExtracting 5987 sequences from NW_020822611.1\n",
      "\tExtracting 2922 sequences from NW_020822614.1\n",
      "\tExtracting 14964 sequences from NW_020822615.1\n",
      "\tExtracting 3776 sequences from NW_020822616.1\n",
      "\tExtracting 1 sequences from NW_020822617.1\n",
      "\tExtracting 1818 sequences from NW_020822618.1\n",
      "\tExtracting 1237 sequences from NW_020822619.1\n",
      "\tExtracting 997 sequences from NW_020822620.1\n",
      "\tExtracting 497 sequences from NW_020822621.1\n",
      "\tExtracting 884 sequences from NW_020822623.1\n",
      "\tExtracting 279 sequences from NW_020822626.1\n",
      "\tExtracting 185 sequences from NW_020822627.1\n",
      "\tExtracting 219 sequences from NW_020822628.1\n",
      "\tExtracting 3435 sequences from NW_020822629.1\n",
      "\tExtracting 4422 sequences from NW_020822630.1\n",
      "\tExtracting 2067 sequences from NW_020822632.1\n",
      "\tExtracting 3080 sequences from NW_020822634.1\n",
      "\tExtracting 1 sequences from NW_020822635.1\n",
      "\tExtracting 2650 sequences from NW_020822636.1\n",
      "\tExtracting 2485 sequences from NW_020822638.1\n",
      "\tExtracting 6 sequences from NW_020822640.1\n",
      "\tExtracting 764 sequences from NW_020822641.1\n",
      "\tExtracting 1062 sequences from NW_020822642.1\n",
      "\tExtracting 2157 sequences from NW_020822643.1\n",
      "\tExtracting 907 sequences from NW_020822644.1\n",
      "\tExtracting 281 sequences from NW_020822645.1\n",
      "\tExtracting 632 sequences from NW_020822646.1\n",
      "\tExtracting 237 sequences from NW_020822647.1\n",
      "\tExtracting 487 sequences from NW_020822648.1\n",
      "\tExtracting 355 sequences from NW_020822649.1\n",
      "\tExtracting 273 sequences from NW_020822650.1\n",
      "\tExtracting 166 sequences from NW_020822651.1\n",
      "\tExtracting 231 sequences from NW_020822652.1\n",
      "\tExtracting 72 sequences from NW_020822653.1\n",
      "\tExtracting 288 sequences from NW_020822654.1\n",
      "\tExtracting 4 sequences from NW_020822655.1\n",
      "\tExtracting 37 sequences from NW_020822656.1\n",
      "\tExtracting 4260 sequences from NW_020822657.1\n",
      "\tExtracting 3992 sequences from NW_020822658.1\n",
      "\tExtracting 951 sequences from NW_020822663.1\n",
      "\tExtracting 678 sequences from NW_020822664.1\n",
      "\tExtracting 668 sequences from NW_020822665.1\n",
      "\tExtracting 551 sequences from NW_020822666.1\n",
      "\tExtracting 359 sequences from NW_020822668.1\n",
      "\tExtracting 273 sequences from NW_020822670.1\n",
      "\tExtracting 355 sequences from NW_020822671.1\n",
      "\tExtracting 479 sequences from NW_020822672.1\n",
      "\tExtracting 25 sequences from NW_020822673.1\n",
      "\tExtracting 270 sequences from NW_020822674.1\n",
      "\tExtracting 25 sequences from NW_020822675.1\n",
      "\tExtracting 337 sequences from NW_020822676.1\n",
      "\tExtracting 16 sequences from NW_020822677.1\n",
      "\tExtracting 429 sequences from NW_020822680.1\n",
      "\tExtracting 167 sequences from NW_020822681.1\n",
      "\tExtracting 10 sequences from NW_020822683.1\n",
      "\tExtracting 125 sequences from NW_020822685.1\n",
      "\tExtracting 109 sequences from NW_020822686.1\n",
      "\tExtracting 2204 sequences from NW_020822687.1\n",
      "\tExtracting 2197 sequences from NW_020822688.1\n",
      "\tExtracting 6 sequences from NW_020822689.1\n",
      "\tExtracting 15 sequences from NW_020822690.1\n",
      "\tExtracting 2 sequences from NW_020822691.1\n",
      "\tExtracting 1 sequences from NW_020822692.1\n",
      "\tExtracting 2 sequences from NW_020822693.1\n",
      "\tExtracting 1 sequences from NW_020822694.1\n",
      "\tExtracting 1 sequences from NW_020822695.1\n",
      "\tExtracting 5 sequences from NW_020822697.1\n",
      "\tExtracting 2012 sequences from NW_020822698.1\n",
      "\tExtracting 870 sequences from NW_020822699.1\n",
      "\tExtracting 1060 sequences from NW_020822701.1\n",
      "\tExtracting 1 sequences from NW_020822702.1\n",
      "\tExtracting 1061 sequences from NW_020822703.1\n",
      "\tExtracting 2 sequences from NW_020822704.1\n",
      "\tExtracting 3 sequences from NW_020822714.1\n",
      "\tExtracting 1 sequences from NW_020822716.1\n",
      "\tExtracting 3 sequences from NW_020822721.1\n",
      "\tExtracting 1 sequences from NW_020822726.1\n",
      "\tExtracting 1 sequences from NW_020822730.1\n",
      "\tExtracting 2 sequences from NW_020822732.1\n",
      "\tExtracting 1 sequences from NW_020822733.1\n",
      "\tExtracting 10 sequences from NW_020822734.1\n",
      "\tExtracting 1 sequences from NW_020822743.1\n",
      "\tExtracting 2 sequences from NW_020822744.1\n",
      "\tExtracting 2 sequences from NW_020822745.1\n",
      "\tExtracting 1 sequences from NW_020822751.1\n",
      "\tExtracting 2 sequences from NW_020822757.1\n",
      "\tExtracting 1 sequences from NW_020822761.1\n",
      "\tExtracting 2 sequences from NW_020822763.1\n",
      "\tExtracting 1 sequences from NW_020822773.1\n",
      "\tExtracting 3 sequences from NW_020822775.1\n",
      "\tExtracting 1 sequences from NW_020822782.1\n",
      "\tExtracting 1 sequences from NW_020822784.1\n",
      "\tExtracting 2 sequences from NW_020822785.1\n",
      "\tExtracting 1 sequences from NW_020822789.1\n",
      "\tExtracting 4 sequences from NW_020822790.1\n",
      "\tExtracting 1 sequences from NW_020822795.1\n",
      "\tExtracting 1 sequences from NW_020822799.1\n",
      "\tExtracting 1 sequences from NW_020822801.1\n",
      "\tExtracting 2 sequences from NW_020822803.1\n",
      "\tExtracting 1 sequences from NW_020822806.1\n",
      "\tExtracting 3 sequences from NW_020822807.1\n",
      "\tExtracting 1 sequences from NW_020822809.1\n",
      "\tExtracting 2 sequences from NW_020822814.1\n",
      "\tExtracting 1 sequences from NW_020822818.1\n",
      "\tExtracting 1 sequences from NW_020822825.1\n",
      "\tExtracting 4 sequences from NW_020822828.1\n",
      "\tExtracting 4 sequences from NW_020822830.1\n",
      "\tExtracting 1 sequences from NW_020822831.1\n",
      "\tExtracting 1 sequences from NW_020822838.1\n",
      "\tExtracting 6 sequences from NW_020822839.1\n",
      "\tExtracting 1 sequences from NW_020822847.1\n",
      "\tExtracting 1 sequences from NW_020822851.1\n",
      "\tExtracting 1 sequences from NW_020822852.1\n",
      "\tExtracting 1 sequences from NW_020822854.1\n",
      "\tExtracting 4 sequences from NW_020822856.1\n",
      "\tExtracting 2 sequences from NW_020822861.1\n",
      "\tExtracting 3 sequences from NW_020822862.1\n",
      "\tExtracting 1 sequences from NW_020822863.1\n",
      "\tExtracting 1 sequences from NW_020822864.1\n",
      "\tExtracting 1 sequences from NW_020822867.1\n",
      "\tExtracting 1 sequences from NW_020822870.1\n",
      "\tExtracting 1 sequences from NW_020822872.1\n",
      "\tExtracting 8 sequences from NW_020822873.1\n",
      "\tExtracting 1 sequences from NW_020822876.1\n",
      "\tExtracting 1 sequences from NW_020822877.1\n",
      "\tExtracting 1 sequences from NW_020822881.1\n",
      "\tExtracting 1 sequences from NW_020822882.1\n",
      "\tExtracting 3 sequences from NW_020822883.1\n",
      "\tExtracting 3 sequences from NW_020822887.1\n",
      "\tExtracting 3 sequences from NW_020822891.1\n",
      "\tExtracting 1 sequences from NW_020822896.1\n",
      "\tExtracting 12 sequences from NW_020822897.1\n",
      "\tExtracting 1 sequences from NW_020822902.1\n",
      "\tExtracting 5 sequences from NW_020822903.1\n",
      "\tExtracting 1 sequences from NW_020822904.1\n",
      "\tExtracting 1 sequences from NW_020822910.1\n",
      "\tExtracting 1 sequences from NW_020822911.1\n",
      "\tExtracting 1 sequences from NW_020822912.1\n",
      "\tExtracting 1 sequences from NW_020822923.1\n",
      "\tExtracting 1 sequences from NW_020822930.1\n",
      "\tExtracting 1 sequences from NW_020822932.1\n",
      "\tExtracting 1 sequences from NW_020822937.1\n",
      "\tExtracting 2 sequences from NW_020822941.1\n",
      "\tExtracting 1 sequences from NW_020822942.1\n",
      "\tExtracting 2 sequences from NW_020822945.1\n",
      "\tExtracting 3 sequences from NW_020822947.1\n",
      "\tExtracting 1 sequences from NW_020822953.1\n",
      "\tExtracting 2 sequences from NW_020822956.1\n",
      "\tExtracting 1 sequences from NW_020822959.1\n",
      "\tExtracting 3 sequences from NW_020822963.1\n",
      "\tExtracting 3 sequences from NW_020822970.1\n",
      "\tExtracting 1 sequences from NW_020822973.1\n",
      "\tExtracting 1 sequences from NW_020822975.1\n",
      "\tExtracting 1 sequences from NW_020822976.1\n",
      "\tExtracting 1 sequences from NW_020822978.1\n",
      "\tExtracting 1 sequences from NW_020822982.1\n",
      "\tExtracting 1 sequences from NW_020822988.1\n",
      "\tExtracting 1 sequences from NW_020822990.1\n",
      "\tExtracting 2 sequences from NW_020822991.1\n",
      "\tExtracting 1 sequences from NW_020823003.1\n",
      "\tExtracting 3 sequences from NW_020823005.1\n",
      "\tExtracting 6 sequences from NW_020823008.1\n",
      "\tExtracting 1 sequences from NW_020823013.1\n",
      "\tExtracting 1 sequences from NW_020823014.1\n",
      "\tExtracting 1 sequences from NW_020823020.1\n",
      "\tExtracting 3 sequences from NW_020823030.1\n",
      "\tExtracting 1 sequences from NW_020823040.1\n",
      "\tExtracting 1 sequences from NW_020823041.1\n",
      "\tExtracting 6 sequences from NW_020823045.1\n",
      "\tExtracting 1 sequences from NW_020823053.1\n",
      "\tExtracting 1 sequences from NW_020823059.1\n",
      "\tExtracting 1 sequences from NW_020823063.1\n",
      "\tExtracting 1 sequences from NW_020823071.1\n",
      "\tExtracting 2 sequences from NW_020823075.1\n",
      "\tExtracting 2 sequences from NW_020823078.1\n",
      "\tExtracting 2 sequences from NW_020823079.1\n",
      "\tExtracting 1 sequences from NW_020823080.1\n",
      "\tExtracting 1 sequences from NW_020823081.1\n",
      "\tExtracting 1 sequences from NW_020823082.1\n",
      "\tExtracting 1 sequences from NW_020823084.1\n",
      "\tExtracting 3 sequences from NW_020823085.1\n",
      "\tExtracting 1 sequences from NW_020823086.1\n",
      "\tExtracting 2 sequences from NW_020823093.1\n",
      "\tExtracting 1 sequences from NW_020823094.1\n",
      "\tExtracting 1 sequences from NW_020823097.1\n",
      "\tExtracting 1 sequences from NW_020823098.1\n",
      "\tExtracting 1 sequences from NW_020823099.1\n",
      "\tExtracting 1 sequences from NW_020823100.1\n",
      "\tExtracting 1 sequences from NW_020823103.1\n",
      "\tExtracting 1 sequences from NW_020823107.1\n",
      "\tExtracting 1 sequences from NW_020823116.1\n",
      "\tExtracting 3 sequences from NW_020823118.1\n",
      "\tExtracting 1 sequences from NW_020823119.1\n",
      "\tExtracting 1 sequences from NW_020823124.1\n",
      "\tExtracting 1 sequences from NW_020823130.1\n",
      "\tExtracting 2 sequences from NW_020823142.1\n",
      "\tExtracting 1 sequences from NW_020823143.1\n",
      "\tExtracting 1 sequences from NW_020823155.1\n",
      "\tExtracting 5 sequences from NW_020823156.1\n",
      "\tExtracting 1 sequences from NW_020823158.1\n",
      "\tExtracting 1 sequences from NW_020823160.1\n",
      "\tExtracting 2 sequences from NW_020823161.1\n",
      "\tExtracting 2 sequences from NW_020823162.1\n",
      "\tExtracting 9 sequences from NW_020823165.1\n",
      "\tExtracting 2 sequences from NW_020823166.1\n",
      "\tExtracting 4 sequences from NW_020823167.1\n",
      "\tExtracting 6 sequences from NW_020823172.1\n",
      "\tExtracting 2 sequences from NW_020823176.1\n",
      "\tExtracting 1 sequences from NW_020823181.1\n",
      "\tExtracting 1 sequences from NW_020823195.1\n",
      "\tExtracting 1 sequences from NW_020823199.1\n",
      "\tExtracting 1 sequences from NW_020823207.1\n",
      "\tExtracting 1 sequences from NW_020823212.1\n",
      "\tExtracting 1 sequences from NW_020823213.1\n",
      "\tExtracting 1 sequences from NW_020823216.1\n",
      "\tExtracting 1 sequences from NW_020823217.1\n",
      "\tExtracting 3 sequences from NW_020823221.1\n",
      "\tExtracting 3 sequences from NW_020823224.1\n",
      "\tExtracting 5 sequences from NW_020823226.1\n",
      "\tExtracting 1 sequences from NW_020823227.1\n",
      "\tExtracting 2 sequences from NW_020823239.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 1 sequences from NW_020823241.1\r\n",
      "\tExtracting 2 sequences from NW_020823243.1\r\n",
      "\tExtracting 1 sequences from NW_020823246.1\r\n",
      "\tExtracting 1 sequences from NW_020823252.1\r\n",
      "\tExtracting 6 sequences from NW_020823257.1\r\n",
      "\tExtracting 5 sequences from NW_020823258.1\r\n",
      "\tExtracting 1 sequences from NW_020823280.1\r\n",
      "\tExtracting 6 sequences from NW_020823286.1\r\n",
      "\tExtracting 1 sequences from NW_020823295.1\r\n",
      "\tExtracting 2 sequences from NW_020823296.1\r\n",
      "\tExtracting 1 sequences from NW_020823298.1\r\n",
      "\tExtracting 2 sequences from NW_020823300.1\r\n",
      "\tExtracting 1 sequences from NW_020823304.1\r\n",
      "\tExtracting 1 sequences from NW_020823306.1\r\n",
      "\tExtracting 1 sequences from NW_020823310.1\r\n",
      "\tExtracting 9 sequences from NW_020823318.1\r\n",
      "\tExtracting 3 sequences from NW_020823326.1\r\n",
      "\tExtracting 2 sequences from NW_020823327.1\r\n",
      "\tExtracting 1 sequences from NW_020823334.1\r\n",
      "\tExtracting 1 sequences from NW_020823361.1\r\n",
      "\tExtracting 1 sequences from NW_020823364.1\r\n",
      "\tExtracting 1 sequences from NW_020823372.1\r\n",
      "\tExtracting 1 sequences from NW_020823391.1\r\n",
      "\tExtracting 2 sequences from NW_020823394.1\r\n",
      "\tExtracting 8 sequences from NW_020823398.1\r\n",
      "\tExtracting 3 sequences from NW_020823400.1\r\n",
      "\tExtracting 1 sequences from NW_020823409.1\r\n",
      "\tExtracting 1 sequences from NW_020823410.1\r\n",
      "\tExtracting 7 sequences from NW_020823411.1\r\n",
      "\tExtracting 3 sequences from NW_020823413.1\r\n",
      "\tExtracting 1 sequences from NW_020823415.1\r\n",
      "\tExtracting 144 sequences from NW_020823422.1\r\n",
      "\tExtracting 1 sequences from NW_020823432.1\r\n",
      "\tExtracting 1 sequences from NW_020823436.1\r\n",
      "\tExtracting 1 sequences from NW_020823437.1\r\n",
      "\tExtracting 1 sequences from NW_020823439.1\r\n",
      "\tExtracting 1 sequences from NW_020823442.1\r\n",
      "\tExtracting 8 sequences from NW_020823443.1\r\n",
      "\tExtracting 1 sequences from NW_020823445.1\r\n",
      "\tExtracting 3 sequences from NW_020823453.1\r\n",
      "\tExtracting 2 sequences from NW_020823464.1\r\n",
      "\tExtracting 133 sequences from NW_020823468.1\r\n",
      "\tExtracting 101 sequences from NW_020823469.1\r\n",
      "\tExtracting 9 sequences from NW_020823470.1\r\n",
      "\tExtracting 5 sequences from NW_020823471.1\r\n",
      "\tExtracting 7 sequences from NW_020823472.1\r\n",
      "\tExtracting 47 sequences from NW_020823473.1\r\n",
      "\tExtracting 31 sequences from NW_020823474.1\r\n",
      "\tExtracting 5 sequences from NW_020823475.1\r\n",
      "\tExtracting 2 sequences from NW_020823476.1\r\n",
      "\tExtracting 6 sequences from NW_020823477.1\r\n",
      "\tExtracting 1 sequences from NW_020823478.1\r\n",
      "\tExtracting 4 sequences from NW_020823479.1\r\n",
      "\tExtracting 1 sequences from NW_020823480.1\r\n",
      "\tExtracting 1 sequences from NW_020823481.1\r\n",
      "\tExtracting 2 sequences from NW_020823482.1\r\n",
      "\tExtracting 3 sequences from NW_020823483.1\r\n",
      "\tExtracting 10 sequences from NW_020823484.1\r\n",
      "\tExtracting 2 sequences from NW_020823485.1\r\n",
      "\tExtracting 3 sequences from NW_020823486.1\r\n",
      "\tExtracting 1 sequences from NW_020823488.1\r\n",
      "\tExtracting 5 sequences from NW_020823489.1\r\n",
      "\tExtracting 46 sequences from NW_020823490.1\r\n",
      "\tExtracting 1 sequences from NW_020823491.1\r\n",
      "\tExtracting 2 sequences from NW_020823492.1\r\n",
      "\tExtracting 2 sequences from NW_020823493.1\r\n",
      "\tExtracting 2 sequences from NW_020823494.1\r\n",
      "\tExtracting 1 sequences from NW_020823495.1\r\n",
      "\tExtracting 1 sequences from NW_020823496.1\r\n",
      "\tExtracting 1 sequences from NW_020823498.1\r\n",
      "\tExtracting 11 sequences from NW_020823499.1\r\n",
      "\tExtracting 9 sequences from NW_020823500.1\r\n",
      "\tExtracting 5 sequences from NW_020823502.1\r\n",
      "\tExtracting 58 sequences from NW_020823503.1\r\n",
      "\tExtracting 6 sequences from NW_020823505.1\r\n",
      "\tExtracting 2 sequences from NW_020823506.1\r\n",
      "\tExtracting 1 sequences from NW_020823507.1\r\n",
      "\tExtracting 1 sequences from NW_020823508.1\r\n",
      "\tExtracting 3 sequences from NW_020823510.1\r\n",
      "\tExtracting 3 sequences from NW_020823511.1\r\n",
      "\tExtracting 13 sequences from NW_020823512.1\r\n",
      "\tExtracting 8 sequences from NW_020823513.1\r\n",
      "\tExtracting 5 sequences from NW_020823514.1\r\n",
      "\tExtracting 4 sequences from NW_020823515.1\r\n",
      "\tExtracting 8 sequences from NW_020823516.1\r\n",
      "\tExtracting 3 sequences from NW_020823517.1\r\n",
      "\tExtracting 12 sequences from NW_020823518.1\r\n",
      "\tExtracting 5 sequences from NW_020823519.1\r\n",
      "\tExtracting 4 sequences from NW_020823521.1\r\n",
      "\tExtracting 1 sequences from NW_020823522.1\r\n",
      "\tExtracting 1 sequences from NW_020823523.1\r\n",
      "\tExtracting 20 sequences from NW_020823524.1\r\n",
      "\tExtracting 3 sequences from NW_020823525.1\r\n",
      "\tExtracting 5 sequences from NW_020823526.1\r\n",
      "\tExtracting 3 sequences from NW_020823527.1\r\n",
      "\tExtracting 2 sequences from NW_020823528.1\r\n",
      "\tExtracting 42 sequences from NW_020823529.1\r\n",
      "\tExtracting 2 sequences from NW_020823530.1\r\n",
      "\tExtracting 4 sequences from NW_020823531.1\r\n",
      "\tExtracting 9 sequences from NW_020823532.1\r\n",
      "\tExtracting 19 sequences from NW_020823533.1\r\n",
      "\tExtracting 1 sequences from NW_020823534.1\r\n",
      "\tExtracting 1 sequences from NW_020823535.1\r\n",
      "\tExtracting 1 sequences from NW_020823536.1\r\n",
      "\tExtracting 2 sequences from NW_020823537.1\r\n",
      "\tExtracting 7 sequences from NW_020823538.1\r\n",
      "\tExtracting 1 sequences from NW_020823539.1\r\n",
      "\tExtracting 8 sequences from NW_020823541.1\r\n",
      "\tExtracting 2 sequences from NW_020823542.1\r\n",
      "\tExtracting 1 sequences from NW_020823543.1\r\n",
      "\tExtracting 1 sequences from NW_020823544.1\r\n",
      "\tExtracting 17 sequences from NW_020823545.1\r\n",
      "\tExtracting 3 sequences from NW_020823546.1\r\n",
      "\tExtracting 3 sequences from NW_020823547.1\r\n",
      "\tExtracting 2 sequences from NW_020823548.1\r\n",
      "\tExtracting 2 sequences from NW_020823549.1\r\n",
      "\tExtracting 8 sequences from NW_020823550.1\r\n",
      "\tExtracting 1 sequences from NW_020823551.1\r\n",
      "\tExtracting 2 sequences from NW_020823554.1\r\n",
      "\tExtracting 6 sequences from NW_020823555.1\r\n",
      "\tExtracting 7 sequences from NW_020823556.1\r\n",
      "\tExtracting 43 sequences from NW_020823558.1\r\n",
      "\tExtracting 11 sequences from NW_020823559.1\r\n",
      "\tExtracting 2 sequences from NW_020823561.1\r\n",
      "\tExtracting 1 sequences from NW_020823562.1\r\n",
      "\tExtracting 15 sequences from NW_020823563.1\r\n",
      "\tExtracting 25 sequences from NW_020823564.1\r\n",
      "\tExtracting 1 sequences from NW_020823565.1\r\n",
      "\tExtracting 2 sequences from NW_020823566.1\r\n",
      "\tExtracting 1 sequences from NW_020823567.1\r\n",
      "\tExtracting 1 sequences from NW_020823568.1\r\n",
      "\tExtracting 2 sequences from NW_020823569.1\r\n",
      "\tExtracting 3 sequences from NW_020823570.1\r\n",
      "\tExtracting 7 sequences from NW_020823571.1\r\n",
      "\tExtracting 1 sequences from NW_020823572.1\r\n",
      "\tExtracting 2 sequences from NW_020823573.1\r\n",
      "\tExtracting 4 sequences from NW_020823574.1\r\n",
      "\tExtracting 5 sequences from NW_020823575.1\r\n",
      "\tExtracting 1 sequences from NW_020823576.1\r\n",
      "\tExtracting 4 sequences from NW_020823578.1\r\n",
      "\tExtracting 1 sequences from NW_020823579.1\r\n",
      "\tExtracting 1 sequences from NW_020823580.1\r\n",
      "\tExtracting 5 sequences from NW_020823581.1\r\n",
      "\tExtracting 1 sequences from NW_020823583.1\r\n",
      "\tExtracting 2 sequences from NW_020823584.1\r\n",
      "\tExtracting 5 sequences from NW_020823585.1\r\n",
      "\tExtracting 14 sequences from NW_020823586.1\r\n",
      "\tExtracting 5 sequences from NW_020823587.1\r\n",
      "\tExtracting 3 sequences from NW_020823588.1\r\n",
      "\tExtracting 2 sequences from NW_020823589.1\r\n",
      "\tExtracting 4 sequences from NW_020823591.1\r\n",
      "\tExtracting 3 sequences from NW_020823592.1\r\n",
      "\tExtracting 1 sequences from NW_020823593.1\r\n",
      "\tExtracting 1 sequences from NW_020823594.1\r\n",
      "\tExtracting 1 sequences from NW_020823595.1\r\n",
      "\tExtracting 3 sequences from NW_020823596.1\r\n",
      "\tExtracting 3 sequences from NW_020823597.1\r\n",
      "\tExtracting 2 sequences from NW_020823598.1\r\n",
      "\tExtracting 1 sequences from NW_020823599.1\r\n",
      "\tExtracting 3 sequences from NW_020823601.1\r\n",
      "\tExtracting 1 sequences from NW_020823602.1\r\n",
      "\tExtracting 2 sequences from NW_020823603.1\r\n",
      "\tExtracting 1 sequences from NW_020823605.1\r\n",
      "\tExtracting 2 sequences from NW_020823606.1\r\n",
      "\tExtracting 2 sequences from NW_020823607.1\r\n",
      "\tExtracting 3 sequences from NW_020823608.1\r\n",
      "\tExtracting 1 sequences from NW_020823609.1\r\n",
      "\tExtracting 3 sequences from NW_020823610.1\r\n",
      "\tExtracting 5 sequences from NW_020823612.1\r\n",
      "\tExtracting 4 sequences from NW_020823613.1\r\n",
      "\tExtracting 3 sequences from NW_020823614.1\r\n",
      "\tExtracting 11 sequences from NW_020823615.1\r\n",
      "\tExtracting 3 sequences from NW_020823616.1\r\n",
      "\tExtracting 1 sequences from NW_020823619.1\r\n",
      "\tExtracting 1 sequences from NW_020823620.1\r\n",
      "\tExtracting 1 sequences from NW_020823621.1\r\n",
      "\tExtracting 1 sequences from NW_020823622.1\r\n",
      "\tExtracting 1 sequences from NW_020823623.1\r\n",
      "\tExtracting 4 sequences from NW_020823624.1\r\n",
      "\tExtracting 1 sequences from NW_020823625.1\r\n",
      "\tExtracting 53 sequences from NW_020823627.1\r\n",
      "\tExtracting 9 sequences from NW_020823629.1\r\n",
      "\tExtracting 1 sequences from NW_020823631.1\r\n",
      "\tExtracting 1 sequences from NW_020823633.1\r\n",
      "\tExtracting 1 sequences from NW_020823636.1\r\n",
      "\tExtracting 3 sequences from NW_020823637.1\r\n",
      "\tExtracting 2 sequences from NW_020823638.1\r\n",
      "\tExtracting 5 sequences from NW_020823639.1\r\n",
      "\tExtracting 3 sequences from NW_020823641.1\r\n",
      "\tExtracting 1 sequences from NW_020823642.1\r\n",
      "\tExtracting 1 sequences from NW_020823646.1\r\n",
      "\tExtracting 1 sequences from NW_020823648.1\r\n",
      "\tExtracting 1 sequences from NW_020823649.1\r\n",
      "\tExtracting 1 sequences from NW_020823652.1\r\n",
      "\tExtracting 4 sequences from NW_020823655.1\r\n",
      "\tExtracting 5 sequences from NW_020823656.1\r\n",
      "\tExtracting 1 sequences from NW_020823657.1\r\n",
      "\tExtracting 2 sequences from NW_020823658.1\r\n",
      "\tExtracting 1 sequences from NW_020823659.1\r\n",
      "\tExtracting 2 sequences from NW_020823662.1\r\n",
      "\tExtracting 14 sequences from NW_020823663.1\r\n",
      "\tExtracting 2 sequences from NW_020823664.1\r\n",
      "\tExtracting 1 sequences from NW_020823665.1\r\n",
      "\tExtracting 4 sequences from NW_020823666.1\r\n",
      "\tExtracting 1 sequences from NW_020823667.1\r\n",
      "\tExtracting 1 sequences from NW_020823668.1\r\n",
      "\tExtracting 3 sequences from NW_020823672.1\r\n",
      "\tExtracting 2 sequences from NW_020823674.1\r\n",
      "\tExtracting 1 sequences from NW_020823675.1\r\n",
      "\tExtracting 16 sequences from NW_020823676.1\r\n",
      "\tExtracting 4 sequences from NW_020823677.1\r\n",
      "\tExtracting 7 sequences from NW_020823679.1\r\n",
      "\tExtracting 1 sequences from NW_020823682.1\r\n",
      "\tExtracting 1 sequences from NW_020823684.1\r\n",
      "\tExtracting 1 sequences from NW_020823686.1\r\n",
      "\tExtracting 1 sequences from NW_020823693.1\r\n",
      "\tExtracting 7 sequences from NW_020823695.1\r\n",
      "\tExtracting 1 sequences from NW_020823696.1\r\n",
      "\tExtracting 8 sequences from NW_020823697.1\r\n",
      "\tExtracting 5 sequences from NW_020823699.1\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 2 sequences from NW_020823700.1\r\n",
      "\tExtracting 27 sequences from NW_020823701.1\r\n",
      "\tExtracting 1 sequences from NW_020823702.1\r\n",
      "\tExtracting 2 sequences from NW_020823703.1\r\n",
      "\tExtracting 9 sequences from NW_020823704.1\r\n",
      "\tExtracting 4 sequences from NW_020823707.1\r\n",
      "\tExtracting 3 sequences from NW_020823710.1\r\n",
      "\tExtracting 1 sequences from NW_020823713.1\r\n",
      "\tExtracting 7 sequences from NW_020823717.1\r\n",
      "\tExtracting 1 sequences from NW_020823718.1\r\n",
      "\tExtracting 2 sequences from NW_020823720.1\r\n",
      "\tExtracting 1 sequences from NW_020823721.1\r\n",
      "\tExtracting 3 sequences from NW_020823723.1\r\n",
      "\tExtracting 2 sequences from NW_020823724.1\r\n",
      "\tExtracting 1 sequences from NW_020823725.1\r\n",
      "\tExtracting 5 sequences from NW_020823728.1\r\n",
      "\tExtracting 1 sequences from NW_020823730.1\r\n",
      "\tExtracting 2 sequences from NW_020823731.1\r\n",
      "\tExtracting 1 sequences from NW_020823732.1\r\n",
      "\tExtracting 4 sequences from NW_020823736.1\r\n",
      "\tExtracting 7 sequences from NW_020823741.1\r\n",
      "\tExtracting 1 sequences from NW_020823743.1\r\n",
      "\tExtracting 4 sequences from NW_020823744.1\r\n",
      "\tExtracting 1 sequences from NW_020823745.1\r\n",
      "\tExtracting 3 sequences from NW_020823746.1\r\n",
      "\tExtracting 1 sequences from NW_020823748.1\r\n",
      "\tExtracting 7 sequences from NW_020823749.1\r\n",
      "\tExtracting 1 sequences from NW_020823750.1\r\n",
      "\tExtracting 1 sequences from NW_020823752.1\r\n",
      "\tExtracting 3 sequences from NW_020823754.1\r\n",
      "\tExtracting 1 sequences from NW_020823761.1\r\n",
      "\tExtracting 1 sequences from NW_020823762.1\r\n",
      "\tExtracting 3 sequences from NW_020823763.1\r\n",
      "\tExtracting 1 sequences from NW_020823764.1\r\n",
      "\tExtracting 1 sequences from NW_020823765.1\r\n",
      "\tExtracting 15 sequences from NW_020823767.1\r\n",
      "\tExtracting 1 sequences from NW_020823768.1\r\n",
      "\tExtracting 2 sequences from NW_020823769.1\r\n",
      "\tExtracting 13 sequences from NW_020823770.1\r\n",
      "\tExtracting 1 sequences from NW_020823772.1\r\n",
      "\tExtracting 1 sequences from NW_020823776.1\r\n",
      "\tExtracting 1 sequences from NW_020823782.1\r\n",
      "\tExtracting 1 sequences from NW_020823785.1\r\n",
      "\tExtracting 1 sequences from NW_020823787.1\r\n",
      "\tExtracting 1 sequences from NW_020823788.1\r\n",
      "\tExtracting 2 sequences from NW_020823789.1\r\n",
      "\tExtracting 2 sequences from NW_020823791.1\r\n",
      "\tExtracting 5 sequences from NW_020823792.1\r\n",
      "\tExtracting 1 sequences from NW_020823797.1\r\n",
      "\tExtracting 1 sequences from NW_020823798.1\r\n",
      "\tExtracting 2 sequences from NW_020823799.1\r\n",
      "\tExtracting 2 sequences from NW_020823805.1\r\n",
      "\tExtracting 2 sequences from NW_020823806.1\r\n",
      "\tExtracting 2 sequences from NW_020823808.1\r\n",
      "\tExtracting 10 sequences from NW_020823812.1\r\n",
      "\tExtracting 3 sequences from NW_020823814.1\r\n",
      "\tExtracting 1 sequences from NW_020823817.1\r\n",
      "\tExtracting 30 sequences from NW_020823819.1\r\n",
      "\tExtracting 1 sequences from NW_020823820.1\r\n",
      "\tExtracting 2 sequences from NW_020823823.1\r\n",
      "\tExtracting 5 sequences from NW_020823831.1\r\n",
      "\tExtracting 1 sequences from NW_020823832.1\r\n",
      "\tExtracting 3 sequences from NW_020823833.1\r\n",
      "\tExtracting 2 sequences from NW_020823834.1\r\n",
      "\tExtracting 1 sequences from NW_020823842.1\r\n",
      "\tExtracting 1 sequences from NW_020823846.1\r\n",
      "\tExtracting 2 sequences from NW_020823847.1\r\n",
      "\tExtracting 2 sequences from NW_020823848.1\r\n",
      "\tExtracting 3 sequences from NW_020823849.1\r\n",
      "\tExtracting 1 sequences from NW_020823850.1\r\n",
      "\tExtracting 1 sequences from NW_020823855.1\r\n",
      "\tExtracting 1 sequences from NW_020823856.1\r\n",
      "\tExtracting 1 sequences from NW_020823857.1\r\n",
      "\tExtracting 1 sequences from NW_020823860.1\r\n",
      "\tExtracting 1 sequences from NW_020823864.1\r\n",
      "\tExtracting 1 sequences from NW_020823866.1\r\n",
      "\tExtracting 1 sequences from NW_020823870.1\r\n",
      "\tExtracting 7 sequences from NW_020823871.1\r\n",
      "\tExtracting 1 sequences from NW_020823873.1\r\n",
      "\tExtracting 3 sequences from NW_020823876.1\r\n",
      "\tExtracting 1 sequences from NW_020823877.1\r\n",
      "\tExtracting 4 sequences from NW_020823879.1\r\n",
      "\tExtracting 1 sequences from NW_020823881.1\r\n",
      "\tExtracting 8 sequences from NW_020823887.1\r\n",
      "\tExtracting 1 sequences from NW_020823891.1\r\n",
      "\tExtracting 5 sequences from NW_020823894.1\r\n",
      "\tExtracting 1 sequences from NW_020823896.1\r\n",
      "\tExtracting 1 sequences from NW_020823898.1\r\n",
      "\tExtracting 6 sequences from NW_020823905.1\r\n",
      "\tExtracting 6 sequences from NW_020823906.1\r\n",
      "\tExtracting 2 sequences from NW_020823908.1\r\n",
      "\tExtracting 1 sequences from NW_020823909.1\r\n",
      "\tExtracting 1 sequences from NW_020823910.1\r\n",
      "\tExtracting 1 sequences from NW_020823911.1\r\n",
      "\tExtracting 3 sequences from NW_020823912.1\r\n",
      "\tExtracting 1 sequences from NW_020823914.1\r\n",
      "\tExtracting 1 sequences from NW_020823915.1\r\n",
      "\tExtracting 2 sequences from NW_020823916.1\r\n",
      "\tExtracting 3 sequences from NW_020823923.1\r\n",
      "\tExtracting 9 sequences from NW_020823928.1\r\n",
      "\tExtracting 1 sequences from NW_020823929.1\r\n",
      "\tExtracting 1 sequences from NW_020823931.1\r\n",
      "\tExtracting 2 sequences from NW_020823932.1\r\n",
      "\tExtracting 1 sequences from NW_020823938.1\r\n",
      "\tExtracting 5 sequences from NW_020823939.1\r\n",
      "\tExtracting 1 sequences from NW_020823942.1\r\n",
      "\tExtracting 2 sequences from NW_020823945.1\r\n",
      "\tExtracting 1 sequences from NW_020823946.1\r\n",
      "\tExtracting 2 sequences from NW_020823948.1\r\n",
      "\tExtracting 2 sequences from NW_020823952.1\r\n",
      "\tExtracting 5 sequences from NW_020823954.1\r\n",
      "\tExtracting 1 sequences from NW_020823956.1\r\n",
      "\tExtracting 20 sequences from NW_020823957.1\r\n",
      "\tExtracting 1 sequences from NW_020823959.1\r\n",
      "\tExtracting 1 sequences from NW_020823961.1\r\n",
      "\tExtracting 1 sequences from NW_020823970.1\r\n",
      "\tExtracting 1 sequences from NW_020823973.1\r\n",
      "\tExtracting 1 sequences from NW_020823975.1\r\n",
      "\tExtracting 2 sequences from NW_020823976.1\r\n",
      "\tExtracting 1 sequences from NW_020823977.1\r\n",
      "\tExtracting 2 sequences from NW_020823980.1\r\n",
      "\tExtracting 1 sequences from NW_020823984.1\r\n",
      "\tExtracting 1 sequences from NW_020823985.1\r\n",
      "\tExtracting 15 sequences from NW_020823989.1\r\n",
      "\tExtracting 2 sequences from NW_020823991.1\r\n",
      "\tExtracting 1 sequences from NW_020823993.1\r\n",
      "\tExtracting 3 sequences from NW_020823996.1\r\n",
      "\tExtracting 3 sequences from NW_020823998.1\r\n",
      "\tExtracting 1 sequences from NW_020824000.1\r\n",
      "\tExtracting 11 sequences from NW_020824001.1\r\n",
      "\tExtracting 1 sequences from NW_020824002.1\r\n",
      "\tExtracting 2 sequences from NW_020824003.1\r\n",
      "\tExtracting 7 sequences from NW_020824006.1\r\n",
      "\tExtracting 1 sequences from NW_020824007.1\r\n",
      "\tExtracting 1 sequences from NW_020824008.1\r\n",
      "\tExtracting 1 sequences from NW_020824011.1\r\n",
      "\tExtracting 1 sequences from NW_020824012.1\r\n",
      "\tExtracting 1 sequences from NW_020824013.1\r\n",
      "\tExtracting 1 sequences from NW_020824015.1\r\n",
      "\tExtracting 1 sequences from NW_020824017.1\r\n",
      "\tExtracting 1 sequences from NW_020824020.1\r\n",
      "\tExtracting 1 sequences from NW_020824021.1\r\n",
      "\tExtracting 6 sequences from NW_020824027.1\r\n",
      "\tExtracting 3 sequences from NW_020824028.1\r\n",
      "\tExtracting 2 sequences from NW_020824029.1\r\n",
      "\tExtracting 2 sequences from NW_020824030.1\r\n",
      "\tExtracting 7 sequences from NW_020824034.1\r\n",
      "\tExtracting 2 sequences from NW_020824037.1\r\n",
      "\tExtracting 3 sequences from NW_020824042.1\r\n",
      "\tExtracting 1 sequences from NW_020824047.1\r\n",
      "\tExtracting 2 sequences from NW_020824049.1\r\n",
      "\tExtracting 1 sequences from NW_020824050.1\r\n",
      "\tExtracting 1 sequences from NW_020824051.1\r\n",
      "\tExtracting 1 sequences from NW_020824052.1\r\n",
      "\tExtracting 2 sequences from NW_020824053.1\r\n",
      "\tExtracting 11 sequences from NW_020824056.1\r\n",
      "\tExtracting 1 sequences from NW_020824061.1\r\n",
      "\tExtracting 1 sequences from NW_020824063.1\r\n",
      "\tExtracting 1 sequences from NW_020824064.1\r\n",
      "\tExtracting 4 sequences from NW_020824065.1\r\n",
      "\tExtracting 5 sequences from NW_020824066.1\r\n",
      "\tExtracting 3 sequences from NW_020824067.1\r\n",
      "\tExtracting 1 sequences from NW_020824069.1\r\n",
      "\tExtracting 1 sequences from NW_020824070.1\r\n",
      "\tExtracting 2 sequences from NW_020824073.1\r\n",
      "\tExtracting 2 sequences from NW_020824074.1\r\n",
      "\tExtracting 1 sequences from NW_020824082.1\r\n",
      "\tExtracting 1 sequences from NW_020824083.1\r\n",
      "\tExtracting 1 sequences from NW_020824084.1\r\n",
      "\tExtracting 2 sequences from NW_020824091.1\r\n",
      "\tExtracting 1 sequences from NW_020824093.1\r\n",
      "\tExtracting 3 sequences from NW_020824096.1\r\n",
      "\tExtracting 1 sequences from NW_020824097.1\r\n",
      "\tExtracting 1 sequences from NW_020824098.1\r\n",
      "\tExtracting 1 sequences from NW_020824100.1\r\n",
      "\tExtracting 1 sequences from NW_020824101.1\r\n",
      "\tExtracting 1 sequences from NW_020824104.1\r\n",
      "\tExtracting 7 sequences from NW_020824107.1\r\n",
      "\tExtracting 1 sequences from NW_020824111.1\r\n",
      "\tExtracting 1 sequences from NW_020824114.1\r\n",
      "\tExtracting 6 sequences from NW_020824120.1\r\n",
      "\tExtracting 2 sequences from NW_020824123.1\r\n",
      "\tExtracting 3 sequences from NW_020824125.1\r\n",
      "\tExtracting 3 sequences from NW_020824128.1\r\n",
      "\tExtracting 1 sequences from NW_020824140.1\r\n",
      "\tExtracting 1 sequences from NW_020824145.1\r\n",
      "\tExtracting 1 sequences from NW_020824149.1\r\n",
      "\tExtracting 1 sequences from NW_020824151.1\r\n",
      "\tExtracting 1 sequences from NW_020824152.1\r\n",
      "\tExtracting 4 sequences from NW_020824156.1\r\n",
      "\tExtracting 3 sequences from NW_020824157.1\r\n",
      "\tExtracting 2 sequences from NW_020824158.1\r\n",
      "\tExtracting 5 sequences from NW_020824159.1\r\n",
      "\tExtracting 3 sequences from NW_020824162.1\r\n",
      "\tExtracting 4 sequences from NW_020824166.1\r\n",
      "\tExtracting 2 sequences from NW_020824169.1\r\n",
      "\tExtracting 1 sequences from NW_020824172.1\r\n",
      "\tExtracting 1 sequences from NW_020824174.1\r\n",
      "\tExtracting 1 sequences from NW_020824175.1\r\n",
      "\tExtracting 1 sequences from NW_020824181.1\r\n",
      "\tExtracting 1 sequences from NW_020824187.1\r\n",
      "\tExtracting 2 sequences from NW_020824188.1\r\n",
      "\tExtracting 3 sequences from NW_020824193.1\r\n"
     ]
    }
   ],
   "source": [
    "f_out = 'Results/merged/samples.merge.anno.fa'\n",
    "cmd = 'homerTools extract {f_in} {ref_fa} -fa > {f_out}'.format(f_in=output_anno_file,ref_fa=ref_fa,f_out=f_out)\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge tags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:24:44.484843Z",
     "start_time": "2019-01-21T23:16:01.043717Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "makeTagDirectory Results/merged/tags_TSS_merged -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2 /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2 /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1 /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1 /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2 /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1\n",
      "\n",
      "\tCreating directory: Results/merged/tags_TSS_merged and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDM1hKLA/GROCap/f03_tags/trim_BMDM1hKLA_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/BMDMwt/GROCap/f03_tags/trim_BMDMwt_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/GROCap/f03_tags/trim_Brain_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START3\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/GROCap/f03_tags/trim_Kidney_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/GROCap/f03_tags/trim_Liver_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/GROCap/f03_tags/trim_Lung_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_START2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/GROCap/f03_tags/trim_CHO_GROCap2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_START1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2352148474\n",
      "\tTotal Tags = 620386830.0\n",
      "\tTotal Positions = 45573561\n",
      "\tAverage tag length = 42.3\n",
      "\tMedian tags per position = 2 (ideal: 1)\n",
      "\tAverage tags per position = 11.393\n",
      "\t\t!! Might have some clonal amplification in this sample if sonication was used\n",
      "\t\tIf using BED files, try using -forceBED to ignore the 5th column\n",
      "\t\tIgnore if analyzing RNA, MNase, etc. data\n",
      "\tFragment Length Estimate: 398\n",
      "\tPeak Width Estimate: 463\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 5.2\n",
      "\t\tDiff strand fold enrichment: 3.6\n",
      "\t\tSame / Diff fold enrichment: 1.3\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "CPU times: user 14.3 s, sys: 2.38 s, total: 16.7 s\n",
      "Wall time: 8min 43s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "all_tags_start = []\n",
    "for t in tissues:\n",
    "    for s in ['GROCap','START']:\n",
    "        #curr = glob.glob(j + '*f03_tags/trim*.peak')\n",
    "        curr_raw = glob.glob(data_folder + t + '/*/')\n",
    "        #print(curr_raw)\n",
    "        for j in curr_raw:\n",
    "            curr_type = j.split('/')[-2]  \n",
    "            if curr_type == 'GROCap':\n",
    "                #print(os.path.join(j,'f03_tags/*5GRO*'))\n",
    "                new = glob.glob(os.path.join(data_folder,s,j,'f03_tags/*GROCap*'))\n",
    "            elif curr_type == 'START':\n",
    "                new = np.array(glob.glob(os.path.join(data_folder,s,j,'f03_tags/*')))\n",
    "                if not len(new) == 0:\n",
    "                    new = (new[(np.array(map(lambda x: 'input' not in x,new)))]) #Remove the input\n",
    "            else:\n",
    "                continue\n",
    "            all_tags_start.extend(new)\n",
    "\n",
    "\n",
    "\n",
    "curr_out = 'Results/merged/tags_TSS_merged'\n",
    "cmd = 'makeTagDirectory {out_f} -single -d {files}'.format(out_f=curr_out,files = ' '.join(all_tags_start))\n",
    "print(cmd)\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:26:10.711467Z",
     "start_time": "2019-01-21T23:24:44.489122Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "makeTagDirectory Results/merged/tags_input_merged -single -d /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_STARTinput2 /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_STARTinput1 /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_STARTinput2\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_STARTinput1\n",
      "\tWill add tag directory: /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_STARTinput1\n",
      "\n",
      "\tCreating directory: Results/merged/tags_input_merged and removing existing *.tags.tsv\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_STARTinput2\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Brain/START/f03_tags/trim_Brain_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/FemaleReproductive/START/f03_tags/trim_FemaleReproductive_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Heart/START/f03_tags/trim_Heart_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Kidney/START/f03_tags/trim_Kidney_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Liver/START/f03_tags/trim_Liver_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Lung/START/f03_tags/trim_Lung_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/MiscOrgans/START/f03_tags/trim_MiscOrgans_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Muscle/START/f03_tags/trim_Muscle_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Pancreas/START/f03_tags/trim_Pancreas_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/Spleen/START/f03_tags/trim_Spleen_STARTinput1\n",
      "\tAdding tag directory /data/isshamie/TSS/NCBI_PICR_103_Processed/CHO/START/f03_tags/trim_CHO_STARTinput1\n",
      "\n",
      "\tOptimizing tag files...\n",
      "\tOptimizing single genome.tags.tsv file...\n",
      "\tEstimated genome size = 2336088320\n",
      "\tTotal Tags = 197527013.0\n",
      "\tTotal Positions = 14297353\n",
      "\tAverage tag length = 46.9\n",
      "\tMedian tags per position = 1 (ideal: 1)\n",
      "\tAverage tags per position = 4.741\n",
      "\tFragment Length Estimate: 169\n",
      "\tPeak Width Estimate: 17\n",
      "\t\t!!! No reliable estimate for peak size\n",
      "\t\tSetting Peak width estimate to be equal to fragment length estimate\n",
      "\tAutocorrelation quality control metrics:\n",
      "\t\tSame strand fold enrichment: 3.1\n",
      "\t\tDiff strand fold enrichment: 2.7\n",
      "\t\tSame / Diff fold enrichment: 5.8\n",
      "\n",
      "\t\tGuessing sample is ChIP-Seq or unstranded RNA-Seq - autocorrelation looks good.\n",
      "\n",
      "CPU times: user 2.34 s, sys: 476 ms, total: 2.82 s\n",
      "Wall time: 1min 26s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "all_tags_start = []\n",
    "for t in tissues:\n",
    "    for s in ['GROCap','START']:\n",
    "        #curr = glob.glob(j + '*f03_tags/trim*.peak')\n",
    "        j = glob.glob(os.path.join(data_folder, t,s))\n",
    "        if len(j) > 0:\n",
    "            j = j[0]\n",
    "        else:\n",
    "            continue\n",
    "        curr_type = s\n",
    "        if curr_type == 'GROCap':\n",
    "            #print(os.path.join(j,'f03_tags/*5GRO*'))\n",
    "            new = np.array(glob.glob(os.path.join(data_folder,s,j,'f03_tags/*GRO*')))\n",
    "            if not len(new) == 0:\n",
    "                new = (new[(np.array(map(lambda x: 'Cap' not in x,new)))]) #Remove the input\n",
    "        elif curr_type == 'START':\n",
    "            new = np.array(glob.glob(os.path.join(data_folder,s,j,'f03_tags/*input*')))\n",
    "        else:\n",
    "            continue\n",
    "        all_tags_start.extend(new)\n",
    "\n",
    "\n",
    "\n",
    "curr_out = 'Results/merged/tags_input_merged'\n",
    "cmd = 'makeTagDirectory {out_f} -single -d {files}'.format(out_f=curr_out,files = ' '.join(all_tags_start))\n",
    "print(cmd)\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:26:10.739446Z",
     "start_time": "2019-01-21T23:26:10.715518Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def make_bedgraph_file(in_dir,out_file):\n",
    "    ''' Create bedgraph file for experiment. - strand will have negative values to be used'''\n",
    "    \n",
    "    out_pos = os.path.join(os.path.dirname(out_file),  \"pos_\" + os.path.basename(out_file))\n",
    "    out_neg = os.path.join(os.path.dirname(out_file),  \"neg_\" + os.path.basename(out_file))\n",
    "    \n",
    "    cmd = ('makeUCSCfile {in_dir} -o {out_file} -strand + -fragLength 1').format(in_dir=in_dir,out_file=out_pos)\n",
    "    print(cmd);sys.stdout.flush()\n",
    "    sarge.run(cmd)\n",
    "\n",
    "    cmd = ('makeUCSCfile {in_dir} -o {out_file} -strand - -fragLength 1').format(in_dir=in_dir,out_file=out_neg)\n",
    "    print(cmd);sys.stdout.flush()\n",
    "    sarge.run(cmd)\n",
    "    print('here')\n",
    "    print(out_file)\n",
    "    cmd = \"gunzip -c %s.gz | awk '{$4=-$4; print}' | gzip > %s_tmp.gz\" % (out_neg,out_neg)# .format(out_f_neg=out_neg) #Change - strand to have neg values\n",
    "    print(cmd)\n",
    "    sarge.run(cmd)\n",
    "    out = out_pos.replace('_pos','') #trim_CHO--mSTART-JHS823_S21_R1_001_pos.bedgraph.gz\n",
    "    \n",
    "    cmd = (\"cat {out_f_0}.gz {out_f_1}_tmp.gz > {final}.gz\").format(out_f_0=out_pos,out_f_1=out_neg,final=out_file) #Concatenate the 2 into new merged file\n",
    "    print(cmd)\n",
    "    sarge.run(cmd)\n",
    "    cmd = (\"rm {out_f_0}.gz {out_f_1}_tmp.gz {out_f_1}.gz\").format(out_f_0=out_pos,out_f_1=out_neg) #replace pos and neg\n",
    "    print(cmd)\n",
    "    sarge.run(cmd)\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:27:25.045326Z",
     "start_time": "2019-01-21T23:26:10.743177Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "makeUCSCfile Results/merged/tags_input_merged -o Results/merged/pos_tags_input_merged.bedgraph -strand + -fragLength 1\n",
      "makeUCSCfile Results/merged/tags_input_merged -o Results/merged/neg_tags_input_merged.bedgraph -strand - -fragLength 1\n",
      "here\n",
      "Results/merged/tags_input_merged.bedgraph\n",
      "gunzip -c Results/merged/neg_tags_input_merged.bedgraph.gz | awk '{$4=-$4; print}' | gzip > Results/merged/neg_tags_input_merged.bedgraph_tmp.gz\n",
      "cat Results/merged/pos_tags_input_merged.bedgraph.gz Results/merged/neg_tags_input_merged.bedgraph_tmp.gz > Results/merged/tags_input_merged.bedgraph.gz\n",
      "rm Results/merged/pos_tags_input_merged.bedgraph.gz Results/merged/neg_tags_input_merged.bedgraph_tmp.gz Results/merged/neg_tags_input_merged.bedgraph.gz\n"
     ]
    }
   ],
   "source": [
    "in_dir = \"Results/merged/tags_input_merged\"\n",
    "out_file = \"Results/merged/tags_input_merged.bedgraph\"\n",
    "\n",
    "make_bedgraph_file(\"Results/merged/tags_input_merged\",out_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:31:07.158282Z",
     "start_time": "2019-01-21T23:27:25.048550Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "makeUCSCfile Results/merged/tags_TSS_merged -o Results/merged/pos_tags_TSS_merged.bedgraph -strand + -fragLength 1\n",
      "makeUCSCfile Results/merged/tags_TSS_merged -o Results/merged/neg_tags_TSS_merged.bedgraph -strand - -fragLength 1\n",
      "here\n",
      "Results/merged/tags_TSS_merged.bedgraph\n",
      "gunzip -c Results/merged/neg_tags_TSS_merged.bedgraph.gz | awk '{$4=-$4; print}' | gzip > Results/merged/neg_tags_TSS_merged.bedgraph_tmp.gz\n",
      "cat Results/merged/pos_tags_TSS_merged.bedgraph.gz Results/merged/neg_tags_TSS_merged.bedgraph_tmp.gz > Results/merged/tags_TSS_merged.bedgraph.gz\n",
      "rm Results/merged/pos_tags_TSS_merged.bedgraph.gz Results/merged/neg_tags_TSS_merged.bedgraph_tmp.gz Results/merged/neg_tags_TSS_merged.bedgraph.gz\n"
     ]
    }
   ],
   "source": [
    "in_dir = \"Results/merged/tags_TSS_merged\"\n",
    "out_file = \"Results/merged/tags_TSS_merged.bedgraph\"\n",
    "\n",
    "make_bedgraph_file(\"Results/merged/tags_TSS_merged\",out_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:31:07.169835Z",
     "start_time": "2019-01-21T23:31:07.163045Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'commit_hash': u'5a894b9',\n",
      " 'commit_source': 'installation',\n",
      " 'default_encoding': 'UTF-8',\n",
      " 'ipython_path': '/home/isshamie/software/anaconda2/lib/python2.7/site-packages/IPython',\n",
      " 'ipython_version': '5.3.0',\n",
      " 'os_name': 'posix',\n",
      " 'platform': 'Linux-4.4.0-130-generic-x86_64-with-debian-stretch-sid',\n",
      " 'sys_executable': '/home/isshamie/software/anaconda2/bin/python',\n",
      " 'sys_platform': 'linux2',\n",
      " 'sys_version': '2.7.13 |Anaconda custom (64-bit)| (default, Dec 20 2016, 23:09:15) \\n[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]'}\n"
     ]
    }
   ],
   "source": [
    "import IPython\n",
    "print IPython.sys_info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-21T23:31:07.295163Z",
     "start_time": "2019-01-21T23:31:07.172093Z"
    }
   },
   "outputs": [
    {
     "data": {
      "application/json": {
       "Software versions": [
        {
         "module": "Python",
         "version": "2.7.13 64bit [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]"
        },
        {
         "module": "IPython",
         "version": "5.3.0"
        },
        {
         "module": "OS",
         "version": "Linux 4.4.0 130 generic x86_64 with debian stretch sid"
        },
        {
         "module": "pandas",
         "version": "0.23.1"
        },
        {
         "module": "numpy",
         "version": "1.13.3"
        },
        {
         "module": "seaborn",
         "version": "0.7.1"
        }
       ]
      },
      "text/html": [
       "<table><tr><th>Software</th><th>Version</th></tr><tr><td>Python</td><td>2.7.13 64bit [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]</td></tr><tr><td>IPython</td><td>5.3.0</td></tr><tr><td>OS</td><td>Linux 4.4.0 130 generic x86_64 with debian stretch sid</td></tr><tr><td>pandas</td><td>0.23.1</td></tr><tr><td>numpy</td><td>1.13.3</td></tr><tr><td>seaborn</td><td>0.7.1</td></tr><tr><td colspan='2'>Mon Jan 21 15:31:07 2019 PST</td></tr></table>"
      ],
      "text/latex": [
       "\\begin{tabular}{|l|l|}\\hline\n",
       "{\\bf Software} & {\\bf Version} \\\\ \\hline\\hline\n",
       "Python & 2.7.13 64bit [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] \\\\ \\hline\n",
       "IPython & 5.3.0 \\\\ \\hline\n",
       "OS & Linux 4.4.0 130 generic x86\\_64 with debian stretch sid \\\\ \\hline\n",
       "pandas & 0.23.1 \\\\ \\hline\n",
       "numpy & 1.13.3 \\\\ \\hline\n",
       "seaborn & 0.7.1 \\\\ \\hline\n",
       "\\hline \\multicolumn{2}{|l|}{Mon Jan 21 15:31:07 2019 PST} \\\\ \\hline\n",
       "\\end{tabular}\n"
      ],
      "text/plain": [
       "Software versions\n",
       "Python 2.7.13 64bit [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]\n",
       "IPython 5.3.0\n",
       "OS Linux 4.4.0 130 generic x86_64 with debian stretch sid\n",
       "pandas 0.23.1\n",
       "numpy 1.13.3\n",
       "seaborn 0.7.1\n",
       "Mon Jan 21 15:31:07 2019 PST"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%load_ext version_information\n",
    "%version_information pandas, numpy, seaborn\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-22T03:42:33.518980Z",
     "start_time": "2019-01-22T03:42:26.635888Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['You are using pip version 9.0.1, however version 18.1 is available.',\n",
       " \"You should consider upgrading via the 'pip install --upgrade pip' command.\",\n",
       " 'alabaster==0.7.10',\n",
       " 'anaconda-client==1.6.3',\n",
       " 'anaconda-navigator==1.6.2',\n",
       " 'anaconda-project==0.6.0',\n",
       " 'asn1crypto==0.22.0',\n",
       " 'astroid==1.4.9',\n",
       " 'astropy==2.0.7',\n",
       " 'Babel==2.4.0',\n",
       " 'backports-abc==0.5',\n",
       " 'backports.functools-lru-cache==1.5',\n",
       " 'backports.shutil-get-terminal-size==1.0.0',\n",
       " 'backports.ssl-match-hostname==3.4.0.2',\n",
       " 'beautifulsoup4==4.6.0',\n",
       " 'biopython==1.72',\n",
       " 'bitarray==0.8.1',\n",
       " 'blaze==0.10.1',\n",
       " 'bleach==1.5.0',\n",
       " 'bokeh==0.12.5',\n",
       " 'boto==2.46.1',\n",
       " 'Bottleneck==1.2.1',\n",
       " 'brewer2mpl==1.4.1',\n",
       " 'bz2file==0.98',\n",
       " 'cdecimal==2.3',\n",
       " 'certifi==2018.8.24',\n",
       " 'cffi==1.10.0',\n",
       " 'chardet==3.0.3',\n",
       " 'Circos==1.3.5',\n",
       " 'click==6.7',\n",
       " 'cloudpickle==0.2.2',\n",
       " 'clyent==1.2.2',\n",
       " 'colorama==0.3.9',\n",
       " 'colormath==2.1.1',\n",
       " 'conda==4.5.11',\n",
       " 'configparser==3.5.0',\n",
       " 'contextlib2==0.5.5',\n",
       " 'cooler==0.7.6',\n",
       " 'cryptography==1.8.1',\n",
       " 'cutadapt==1.16',\n",
       " 'cycler==0.10.0',\n",
       " 'Cython==0.25.2',\n",
       " 'cytoolz==0.8.2',\n",
       " 'dask==0.18.1',\n",
       " 'datashape==0.5.4',\n",
       " 'decorator==4.3.0',\n",
       " 'deepTools==3.1.0',\n",
       " 'dill==0.2.8.2',\n",
       " 'distributed==1.16.3',\n",
       " 'docutils==0.13.1',\n",
       " 'entrypoints==0.2.2',\n",
       " 'enum34==1.1.6',\n",
       " 'et-xmlfile==1.0.1',\n",
       " 'fastcache==1.0.2',\n",
       " 'feather-format==0.4.0',\n",
       " 'Flask==0.12.2',\n",
       " 'Flask-Cors==3.0.2',\n",
       " 'funcsigs==1.0.2',\n",
       " 'functools32==3.2.3.post2',\n",
       " 'future==0.16.0',\n",
       " 'futures==3.1.1',\n",
       " 'gevent==1.2.1',\n",
       " 'greenlet==0.4.12',\n",
       " 'grin==1.2.1',\n",
       " 'h5py==2.7.0',\n",
       " 'HeapDict==1.0.0',\n",
       " 'HiCExplorer==2.1.4',\n",
       " 'html5lib==0.999',\n",
       " 'idna==2.5',\n",
       " 'imageio==2.3.0',\n",
       " 'imagesize==0.7.1',\n",
       " 'intervaltree==2.1.0',\n",
       " 'ipaddress==1.0.18',\n",
       " 'ipykernel==4.6.1',\n",
       " 'ipyparallel==6.0.2',\n",
       " 'ipython==5.3.0',\n",
       " 'ipython-genutils==0.2.0',\n",
       " 'ipywidgets==6.0.0',\n",
       " 'isort==4.2.5',\n",
       " 'itsdangerous==0.24',\n",
       " 'jdcal==1.3',\n",
       " 'jedi==0.10.2',\n",
       " 'Jinja2==2.9.6',\n",
       " 'joblib==0.11',\n",
       " 'jsonschema==2.6.0',\n",
       " 'jupyter==1.0.0',\n",
       " 'jupyter-client==5.0.1',\n",
       " 'jupyter-console==5.1.0',\n",
       " 'jupyter-contrib-core==0.3.3',\n",
       " 'jupyter-contrib-nbextensions==0.5.0',\n",
       " 'jupyter-core==4.3.0',\n",
       " 'jupyter-highlight-selected-word==0.1.1',\n",
       " 'jupyter-latex-envs==1.4.4',\n",
       " 'jupyter-nbextensions-configurator==0.4.0',\n",
       " 'jupyterlab==0.31.12',\n",
       " 'jupyterlab-launcher==0.10.5',\n",
       " 'lazy-object-proxy==1.2.2',\n",
       " 'llvmlite==0.23.0',\n",
       " 'locket==0.2.0',\n",
       " 'lockfile==0.12.2',\n",
       " 'luigi==2.7.8',\n",
       " 'lxml==3.7.3',\n",
       " 'lzstring==1.0.3',\n",
       " 'Markdown==2.6.11',\n",
       " 'MarkupSafe==0.23',\n",
       " 'matplotlib==2.1.1',\n",
       " 'matplotlib-venn==0.11.5',\n",
       " 'mistune==0.7.4',\n",
       " 'mkl-fft==1.0.2',\n",
       " 'mkl-random==1.0.1',\n",
       " 'mpmath==0.19',\n",
       " 'msgpack-python==0.4.8',\n",
       " 'multipledispatch==0.4.9',\n",
       " 'multiprocess==0.70.5',\n",
       " 'multiqc==1.3',\n",
       " 'natsort==5.1.1',\n",
       " 'navigator-updater==0.1.0',\n",
       " 'nbconvert==5.1.1',\n",
       " 'nbformat==4.3.0',\n",
       " 'network-evaluation-tools==1.0.2',\n",
       " 'networkx==2.2',\n",
       " 'nltk==3.2.3',\n",
       " 'nose==1.3.7',\n",
       " 'notebook==5.0.0',\n",
       " 'numba==0.38.1',\n",
       " 'numexpr==2.6.5',\n",
       " 'numpy==1.13.3',\n",
       " 'numpydoc==0.6.0',\n",
       " 'odo==0.5.0',\n",
       " 'olefile==0.44',\n",
       " 'openpyxl==2.4.7',\n",
       " 'packaging==16.8',\n",
       " 'pandas==0.23.1',\n",
       " 'pandocfilters==1.4.1',\n",
       " 'partd==0.3.8',\n",
       " 'pathlib2==2.2.1',\n",
       " 'patsy==0.4.1',\n",
       " 'pep8==1.7.0',\n",
       " 'pexpect==4.2.1',\n",
       " 'pickleshare==0.7.4',\n",
       " 'Pillow==4.3.0',\n",
       " 'plotly==2.7.0',\n",
       " 'ply==3.10',\n",
       " 'prompt-toolkit==1.0.14',\n",
       " 'psutil==5.2.2',\n",
       " 'ptyprocess==0.5.1',\n",
       " 'py==1.4.33',\n",
       " 'py2bit==0.3.0',\n",
       " 'pyarrow==0.9.0',\n",
       " 'pybedtools==0.7.10',\n",
       " 'pyBigWig==0.3.11',\n",
       " 'pycairo==1.10.0',\n",
       " 'pycodestyle==2.4.0',\n",
       " 'pycosat==0.6.3',\n",
       " 'pycparser==2.17',\n",
       " 'pycrypto==2.6.1',\n",
       " 'pycurl==7.43.0',\n",
       " 'pyfaidx==0.5.4.1',\n",
       " 'pyflakes==1.5.0',\n",
       " 'pyGenomeTracks==2.0',\n",
       " 'Pygments==2.2.0',\n",
       " 'pylint==1.6.4',\n",
       " 'pyodbc==4.0.16',\n",
       " 'pyOpenSSL==17.0.0',\n",
       " 'pypairix==0.3.6',\n",
       " 'pyparsing==2.1.4',\n",
       " 'pysam==0.14.1',\n",
       " 'pytest==3.0.7',\n",
       " 'pytest-ipynb==1.1.0',\n",
       " 'python-daemon==2.2.0',\n",
       " 'python-dateutil==2.6.0',\n",
       " 'pytz==2017.2',\n",
       " 'PyUpSet==0.1.1.post7',\n",
       " 'PyWavelets==0.5.2',\n",
       " 'PyYAML==3.12',\n",
       " 'pyzmq==16.0.2',\n",
       " 'QtAwesome==0.4.4',\n",
       " 'qtconsole==4.3.0',\n",
       " 'QtPy==1.2.1',\n",
       " 'requests==2.14.2',\n",
       " 'rope==0.9.4',\n",
       " 'ruffus==2.6.3',\n",
       " 'runipy==0.1.5',\n",
       " 'sarge==0.1.4',\n",
       " 'scandir==1.5',\n",
       " 'scikit-image==0.14.0',\n",
       " 'scikit-learn==0.19.1',\n",
       " 'scipy==1.0.1',\n",
       " 'seaborn==0.7.1',\n",
       " 'simplegeneric==0.8.1',\n",
       " 'simplejson==3.13.2',\n",
       " 'singledispatch==3.4.0.3',\n",
       " 'six==1.10.0',\n",
       " 'snowballstemmer==1.2.1',\n",
       " 'sortedcollections==0.5.3',\n",
       " 'sortedcontainers==1.5.7',\n",
       " 'spectra==0.0.9',\n",
       " 'Sphinx==1.5.6',\n",
       " 'spyder==3.1.4',\n",
       " 'SQLAlchemy==1.1.9',\n",
       " 'statsmodels==0.9.0',\n",
       " 'subprocess32==3.2.7',\n",
       " 'sympy==1.0',\n",
       " 'tables==3.4.2',\n",
       " 'tblib==1.3.2',\n",
       " 'terminado==0.6',\n",
       " 'testpath==0.3',\n",
       " 'toolz==0.8.2',\n",
       " 'tornado==4.5.1',\n",
       " 'tqdm==4.15.0',\n",
       " 'traitlets==4.3.2',\n",
       " 'unicodecsv==0.14.1',\n",
       " 'Unidecode==1.0.22',\n",
       " 'version-information==1.0.3',\n",
       " 'wcwidth==0.1.7',\n",
       " 'Werkzeug==0.12.2',\n",
       " 'widgetsnbextension==2.0.0',\n",
       " 'wrapt==1.10.10',\n",
       " 'xlrd==1.0.0',\n",
       " 'XlsxWriter==0.9.6',\n",
       " 'xlwt==1.2.0',\n",
       " 'xopen==0.3.3',\n",
       " 'yapf==0.22.0',\n",
       " 'zict==0.1.2']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#find the names of the imported modules\n",
    "import types\n",
    "def imports():\n",
    "    for name, val in globals().items():\n",
    "        if isinstance(val, types.ModuleType):\n",
    "            yield val.__name__\n",
    "\n",
    "#exclude all modules not listed by `!pip freeze`\n",
    "excludes = ['__builtin__', 'types', 'IPython.core.shadowns', 'sys', 'os']\n",
    "imported_modules = [module for module in imports() if module not in excludes]\n",
    "pip_modules = !pip freeze #you could also use `!conda list` with anaconda\n",
    "conda_modules = !conda list #you could also use `!conda list` with anaconda\n",
    "\n",
    "\n",
    "pip_modules\n",
    "# #print the names and versions of the imported modules\n",
    "# for module in pip_modules:\n",
    "#     name, version = module.split('==')\n",
    "#     if name in imported_modules:\n",
    "#         print(name + '\\t' + version)\n",
    "\n",
    "        \n",
    "# for module in conda_modules:\n",
    "#     name, version = module.split('==')\n",
    "#     if name in imported_modules:\n",
    "#         print(name + '\\t' + version)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-01-22T03:42:33.542079Z",
     "start_time": "2019-01-22T03:42:33.521164Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['# packages in environment at /home/isshamie/software/anaconda2:',\n",
       " '#',\n",
       " '# Name                    Version                   Build  Channel',\n",
       " '_license                  1.1                      py27_1  ',\n",
       " 'alabaster                 0.7.10                   py27_0  ',\n",
       " 'anaconda                  custom           py27h4a00acb_0  ',\n",
       " 'anaconda-client           1.6.3                    py27_0  ',\n",
       " 'anaconda-navigator        1.6.2                    py27_0  ',\n",
       " 'anaconda-project          0.6.0                    py27_0  ',\n",
       " 'argparse                  1.4.0                     <pip>',\n",
       " 'asn1crypto                0.22.0                   py27_0  ',\n",
       " 'astroid                   1.4.9                    py27_0  ',\n",
       " 'astropy                   2.0.7                    py27_1    conda-forge',\n",
       " 'babel                     2.4.0                    py27_0  ',\n",
       " 'backports                 1.0                      py27_0  ',\n",
       " 'backports.functools_lru_cache 1.5                      py27_0    conda-forge',\n",
       " 'backports_abc             0.5                      py27_0  ',\n",
       " 'bcftools                  1.7                           0    bioconda',\n",
       " 'beautifulsoup4            4.6.0                    py27_0  ',\n",
       " 'bedtools                  2.27.1                        1    bioconda',\n",
       " 'biopython                 1.71                      <pip>',\n",
       " 'biopython                 1.72                     py27_0    conda-forge',\n",
       " 'bitarray                  0.8.1                    py27_0  ',\n",
       " 'blaze                     0.10.1                   py27_0  ',\n",
       " 'bleach                    1.5.0                    py27_0  ',\n",
       " 'bokeh                     0.12.5                   py27_1  ',\n",
       " 'boto                      2.46.1                   py27_0  ',\n",
       " 'bottleneck                1.2.1            py27h7eb728f_1    conda-forge',\n",
       " 'brewer2mpl                1.4.1                     <pip>',\n",
       " 'bzip2                     1.0.6                h9a117a8_4  ',\n",
       " 'ca-certificates           2018.8.24            ha4d7672_0    conda-forge',\n",
       " 'cairo                     1.14.6                        5    conda-forge',\n",
       " 'cdecimal                  2.3                      py27_2  ',\n",
       " 'certifi                   2018.8.24             py27_1001    conda-forge',\n",
       " 'cffi                      1.10.0                   py27_0  ',\n",
       " 'chardet                   3.0.3                    py27_0  ',\n",
       " 'Circos                    1.3.5                     <pip>',\n",
       " 'click                     6.7                      py27_0  ',\n",
       " 'cloudpickle               0.2.2                    py27_0  ',\n",
       " 'clyent                    1.2.2                    py27_0  ',\n",
       " 'colorama                  0.3.9                    py27_0  ',\n",
       " 'colormath                 2.1.1                     <pip>',\n",
       " 'conda                     4.5.11                   py27_0    conda-forge',\n",
       " 'conda-env                 2.6.0                h36134e3_1  ',\n",
       " 'configparser              3.5.0                    py27_0  ',\n",
       " 'contextlib2               0.5.5                    py27_0  ',\n",
       " 'cooler                    0.7.6                    py27_1    bioconda',\n",
       " 'cryptography              1.8.1                    py27_0  ',\n",
       " 'curl                      7.52.1                        0  ',\n",
       " 'cycler                    0.10.0                   py27_0  ',\n",
       " 'cython                    0.25.2                   py27_0  ',\n",
       " 'cytoolz                   0.8.2                    py27_0  ',\n",
       " 'dask                      0.14.3                   py27_1  ',\n",
       " 'dask-core                 0.18.1                     py_0    conda-forge',\n",
       " 'datashape                 0.5.4                    py27_0  ',\n",
       " 'dbus                      1.10.10                       0  ',\n",
       " 'decorator                 4.3.0                     <pip>',\n",
       " 'decorator                 4.3.0                      py_0    conda-forge',\n",
       " 'deeptools                 3.1.0            py27h470a237_0    bioconda',\n",
       " 'dill                      0.2.8.2                  py27_0    conda-forge',\n",
       " 'distributed               1.16.3                   py27_0  ',\n",
       " 'docutils                  0.13.1                   py27_0  ',\n",
       " 'entrypoints               0.2.2                    py27_1  ',\n",
       " 'enum34                    1.1.6                    py27_0  ',\n",
       " 'et_xmlfile                1.0.1                    py27_0  ',\n",
       " 'expat                     2.1.0                         0  ',\n",
       " 'fastcache                 1.0.2                    py27_1  ',\n",
       " 'feather-format            0.4.0                     <pip>',\n",
       " 'flask                     0.12.2                   py27_0  ',\n",
       " 'flask-cors                3.0.2                    py27_0  ',\n",
       " 'fontconfig                2.12.1                        5    conda-forge',\n",
       " 'freetype                  2.7                           2    conda-forge',\n",
       " 'funcsigs                  1.0.2                    py27_0  ',\n",
       " 'functools32               3.2.3.2                  py27_0  ',\n",
       " 'future                    0.16.0                   py27_0    conda-forge',\n",
       " 'future                    0.16.0                    <pip>',\n",
       " 'futures                   3.1.1                    py27_0  ',\n",
       " 'get_terminal_size         1.0.0                    py27_0  ',\n",
       " 'gettext                   0.19.8.1                      0    conda-forge',\n",
       " 'gevent                    1.2.1                    py27_0  ',\n",
       " 'glib                      2.51.4                        0    conda-forge',\n",
       " 'greenlet                  0.4.12                   py27_0  ',\n",
       " 'grin                      1.2.1                    py27_3  ',\n",
       " 'gst-plugins-base          1.8.0                         0  ',\n",
       " 'gstreamer                 1.8.0                         0  ',\n",
       " 'h5py                      2.7.0               np113py27_0  ',\n",
       " 'harfbuzz                  1.3.4                         2    conda-forge',\n",
       " 'hdf5                      1.8.17                        1  ',\n",
       " 'heapdict                  1.0.0                    py27_1  ',\n",
       " 'hicexplorer               2.1.4                    py27_2    bioconda',\n",
       " 'html5lib                  0.999                    py27_0  ',\n",
       " 'htslib                    1.7                           0    bioconda',\n",
       " 'icu                       58.2                 hfc679d8_0    conda-forge',\n",
       " 'idna                      2.5                      py27_0  ',\n",
       " 'imageio                   2.3.0                    py27_0    conda-forge',\n",
       " 'imagesize                 0.7.1                    py27_0  ',\n",
       " 'intel-openmp              2018.0.3                      0  ',\n",
       " 'intervaltree              2.1.0                    py27_0    conda-forge',\n",
       " 'ipaddress                 1.0.18                   py27_0  ',\n",
       " 'ipykernel                 4.6.1                    py27_0  ',\n",
       " 'ipyparallel               6.0.2                    py27_0  ',\n",
       " 'ipython                   5.3.0                    py27_0  ',\n",
       " 'ipython_genutils          0.2.0                    py27_0  ',\n",
       " 'ipywidgets                6.0.0                    py27_0  ',\n",
       " 'isort                     4.2.5                    py27_0  ',\n",
       " 'itsdangerous              0.24                     py27_0  ',\n",
       " 'jbig                      2.1                           0  ',\n",
       " 'jdcal                     1.3                      py27_0  ',\n",
       " 'jedi                      0.10.2                   py27_2  ',\n",
       " 'jinja2                    2.9.6                    py27_0  ',\n",
       " 'joblib                    0.11                      <pip>',\n",
       " 'jpeg                      9b                            0  ',\n",
       " 'jsonschema                2.6.0                    py27_0  ',\n",
       " 'jupyter                   1.0.0                    py27_3  ',\n",
       " 'jupyter_client            5.0.1                    py27_0  ',\n",
       " 'jupyter_console           5.1.0                    py27_0  ',\n",
       " 'jupyter_contrib_core      0.3.3                    py27_1    conda-forge',\n",
       " 'jupyter_contrib_nbextensions 0.5.0                    py27_0    conda-forge',\n",
       " 'jupyter_core              4.3.0                    py27_0  ',\n",
       " 'jupyter_highlight_selected_word 0.1.1                    py27_0    conda-forge',\n",
       " 'jupyter_latex_envs        1.4.4                    py27_0    conda-forge',\n",
       " 'jupyter_nbextensions_configurator 0.4.0                    py27_0    conda-forge',\n",
       " 'jupyterlab                0.31.12                  py27_1    conda-forge',\n",
       " 'jupyterlab_launcher       0.10.5                   py27_0    conda-forge',\n",
       " 'lazy-object-proxy         1.2.2                    py27_0  ',\n",
       " 'libcurl                   7.60.0               h1ad7b7a_0  ',\n",
       " 'libffi                    3.2.1                         1  ',\n",
       " 'libgcc                    4.8.5                         2  ',\n",
       " 'libgcc-ng                 7.2.0                h7cc24e2_2  ',\n",
       " 'libgfortran               3.0.0                         1  ',\n",
       " 'libgfortran-ng            7.2.0                hdf63c60_3  ',\n",
       " 'libiconv                  1.14                          0  ',\n",
       " 'libpng                    1.6.27                        0  ',\n",
       " 'libsodium                 1.0.10                        0  ',\n",
       " 'libssh2                   1.8.0                h5b517e9_2    conda-forge',\n",
       " 'libstdcxx-ng              7.2.0                hdf63c60_3  ',\n",
       " 'libtiff                   4.0.6                         3  ',\n",
       " 'libtool                   2.4.2                         0  ',\n",
       " 'libxcb                    1.12                          1  ',\n",
       " 'libxml2                   2.9.4                         0  ',\n",
       " 'libxslt                   1.1.29                        0  ',\n",
       " 'llvmlite                  0.23.0                   py27_1    conda-forge',\n",
       " 'locket                    0.2.0                    py27_1  ',\n",
       " 'lockfile                  0.12.2                    <pip>',\n",
       " 'luigi                     2.7.8                     <pip>',\n",
       " 'lxml                      3.7.3                    py27_0  ',\n",
       " 'lzstring                  1.0.3                     <pip>',\n",
       " 'Markdown                  2.6.11                    <pip>',\n",
       " 'markupsafe                0.23                     py27_2  ',\n",
       " 'matplotlib                2.1.1                    py27_0    conda-forge',\n",
       " 'matplotlib-venn           0.11.5                    <pip>',\n",
       " 'mistune                   0.7.4                    py27_0  ',\n",
       " 'mkl                       2018.0.3                      1  ',\n",
       " 'mkl-service               1.1.2                    py27_3  ',\n",
       " 'mkl_fft                   1.0.2                    py27_0    conda-forge',\n",
       " 'mkl_random                1.0.1                    py27_0    conda-forge',\n",
       " 'mpmath                    0.19                     py27_1  ',\n",
       " 'msgpack-python            0.4.8                    py27_0  ',\n",
       " 'multipledispatch          0.4.9                    py27_0  ',\n",
       " 'multiprocess              0.70.5                   py27_0    conda-forge',\n",
       " 'multiqc                   1.3                       <pip>',\n",
       " 'natsort                   5.1.1                     <pip>',\n",
       " 'navigator-updater         0.1.0                    py27_0  ',\n",
       " 'nbconvert                 5.1.1                    py27_0  ',\n",
       " 'nbformat                  4.3.0                    py27_0  ',\n",
       " 'networkx                  2.2                        py_1    conda-forge',\n",
       " 'networkx                  2.2                       <pip>',\n",
       " 'nltk                      3.2.3                    py27_0  ',\n",
       " 'nose                      1.3.7                    py27_1  ',\n",
       " 'notebook                  5.0.0                    py27_0  ',\n",
       " 'numba                     0.38.1                   py27_0    conda-forge',\n",
       " 'numexpr                   2.6.5                    py27_0    conda-forge',\n",
       " 'numpy                     1.13.3           py27hdbf6ddf_4  ',\n",
       " 'numpydoc                  0.6.0                    py27_0  ',\n",
       " 'odo                       0.5.0                    py27_1  ',\n",
       " 'olefile                   0.44                     py27_0  ',\n",
       " 'openpyxl                  2.4.7                    py27_0  ',\n",
       " 'openssl                   1.0.2p               h470a237_0    conda-forge',\n",
       " 'packaging                 16.8                     py27_0  ',\n",
       " 'pairix                    0.3.6                    py27_0    bioconda',\n",
       " 'pandas                    0.23.1                   py27_0    conda-forge',\n",
       " 'pandocfilters             1.4.1                    py27_0  ',\n",
       " 'pango                     1.40.3                        2    conda-forge',\n",
       " 'partd                     0.3.8                    py27_0  ',\n",
       " 'path.py                   10.3.1                   py27_0  ',\n",
       " 'pathlib2                  2.2.1                    py27_0  ',\n",
       " 'patsy                     0.4.1                    py27_0  ',\n",
       " 'pcre                      8.39                          1  ',\n",
       " 'pep8                      1.7.0                    py27_0  ',\n",
       " 'pexpect                   4.2.1                    py27_0  ',\n",
       " 'pickleshare               0.7.4                    py27_0  ',\n",
       " 'pillow                    4.3.0                    py27_1    conda-forge',\n",
       " 'pip                       9.0.1                    py27_1  ',\n",
       " 'pixman                    0.34.0                        0  ',\n",
       " 'plotly                    2.7.0                      py_1    conda-forge',\n",
       " 'ply                       3.10                     py27_0  ',\n",
       " 'prompt_toolkit            1.0.14                   py27_0  ',\n",
       " 'psutil                    5.2.2                    py27_0  ',\n",
       " 'ptyprocess                0.5.1                    py27_0  ',\n",
       " 'py                        1.4.33                   py27_0  ',\n",
       " 'py2bit                    0.3.0                    py27_0    bioconda',\n",
       " 'pyarrow                   0.9.0                     <pip>',\n",
       " 'pybedtools                0.7.10                   py27_2    bioconda',\n",
       " 'pybigwig                  0.3.11           py27he7a497f_2    bioconda',\n",
       " 'pycairo                   1.10.0                   py27_0  ',\n",
       " 'pycodestyle               2.4.0                     <pip>',\n",
       " 'pycosat                   0.6.3            py27ha4109ae_0  ',\n",
       " 'pycparser                 2.17                     py27_0  ',\n",
       " 'pycrypto                  2.6.1                    py27_6  ',\n",
       " 'pycurl                    7.43.0                   py27_2  ',\n",
       " 'pyfaidx                   0.5.4.1                    py_0    bioconda',\n",
       " 'pyflakes                  1.5.0                    py27_0  ',\n",
       " 'pygenometracks            2.0                      py27_2    bioconda',\n",
       " 'pygments                  2.2.0                    py27_0  ',\n",
       " 'pylint                    1.6.4                    py27_1  ',\n",
       " 'pyodbc                    4.0.16                   py27_0  ',\n",
       " 'pyopenssl                 17.0.0                   py27_0  ',\n",
       " 'pyparsing                 2.1.4                    py27_0  ',\n",
       " 'pyqt                      5.6.0                    py27_2  ',\n",
       " 'pysam                     0.14.1          py27_htslib1.7_0    bioconda',\n",
       " 'pytables                  3.4.2               np113py27_0  ',\n",
       " 'pytest                    3.0.7                    py27_0  ',\n",
       " 'pytest-ipynb              1.1.0                     <pip>',\n",
       " 'python                    2.7.13                        0  ',\n",
       " 'python-daemon             2.2.0                     <pip>',\n",
       " 'python-dateutil           2.6.0                    py27_0  ',\n",
       " 'pytz                      2017.2                   py27_0  ',\n",
       " 'PyUpSet                   0.1.1.post7               <pip>',\n",
       " 'pywavelets                0.5.2                    py27_1    conda-forge',\n",
       " 'pyyaml                    3.12                     py27_0  ',\n",
       " 'pyzmq                     16.0.2                   py27_0  ',\n",
       " 'qt                        5.6.2                         6    conda-forge',\n",
       " 'qtawesome                 0.4.4                    py27_0  ',\n",
       " 'qtconsole                 4.3.0                    py27_0  ',\n",
       " 'qtpy                      1.2.1                    py27_0  ',\n",
       " 'readline                  6.2                           2  ',\n",
       " 'requests                  2.14.2                   py27_0  ',\n",
       " 'rope                      0.9.4                    py27_1  ',\n",
       " 'ruamel_yaml               0.11.14                  py27_1  ',\n",
       " 'ruffus                    2.6.3                     <pip>',\n",
       " 'runipy                    0.1.5                     <pip>',\n",
       " 'samtools                  1.7                           1    bioconda',\n",
       " 'sarge                     0.1.4                     <pip>',\n",
       " 'scandir                   1.5                      py27_0  ',\n",
       " 'scikit-image              0.14.0           py27hfc679d8_1    conda-forge',\n",
       " 'scikit-learn              0.19.1           py27h445a80a_0  ',\n",
       " 'scipy                     1.0.1            py27hfc37229_0  ',\n",
       " 'seaborn                   0.7.1                    py27_0  ',\n",
       " 'setuptools                39.0.1                   py27_0  ',\n",
       " 'simplegeneric             0.8.1                    py27_1  ',\n",
       " 'simplejson                3.13.2                    <pip>',\n",
       " 'singledispatch            3.4.0.3                  py27_0  ',\n",
       " 'sip                       4.18                     py27_0  ',\n",
       " 'six                       1.10.0                   py27_0  ',\n",
       " 'snowballstemmer           1.2.1                    py27_0  ',\n",
       " 'sortedcollections         0.5.3                    py27_0  ',\n",
       " 'sortedcontainers          1.5.7                    py27_0  ',\n",
       " 'spectra                   0.0.9                     <pip>',\n",
       " 'sphinx                    1.5.6                    py27_0  ',\n",
       " 'spyder                    3.1.4                    py27_0  ',\n",
       " 'sqlalchemy                1.1.9                    py27_0  ',\n",
       " 'sqlite                    3.13.0                        0  ',\n",
       " 'ssl_match_hostname        3.4.0.2                  py27_1  ',\n",
       " 'statsmodels               0.9.0                    py27_0    conda-forge',\n",
       " 'subprocess32              3.2.7                    py27_0  ',\n",
       " 'sympy                     1.0                      py27_0  ',\n",
       " 'tblib                     1.3.2                    py27_0  ',\n",
       " 'terminado                 0.6                      py27_0  ',\n",
       " 'testpath                  0.3                      py27_0  ',\n",
       " 'tk                        8.5.18                        0  ',\n",
       " 'toolz                     0.8.2                    py27_0  ',\n",
       " 'tornado                   4.5.1                    py27_0  ',\n",
       " 'tqdm                      4.15.0                   py27_0  ',\n",
       " 'traitlets                 4.3.2                    py27_0  ',\n",
       " 'unicodecsv                0.14.1                   py27_0  ',\n",
       " 'unidecode                 1.0.22                   py27_0    conda-forge',\n",
       " 'unixodbc                  2.3.4                         0  ',\n",
       " 'version-information       1.0.3                     <pip>',\n",
       " 'wcwidth                   0.1.7                    py27_0  ',\n",
       " 'werkzeug                  0.12.2                   py27_0  ',\n",
       " 'wheel                     0.29.0                   py27_0  ',\n",
       " 'widgetsnbextension        2.0.0                    py27_0  ',\n",
       " 'wrapt                     1.10.10                  py27_0  ',\n",
       " 'xlrd                      1.0.0                    py27_0  ',\n",
       " 'xlsxwriter                0.9.6                    py27_0  ',\n",
       " 'xlwt                      1.2.0                    py27_0  ',\n",
       " 'xz                        5.2.2                         1  ',\n",
       " 'yaml                      0.1.6                         0  ',\n",
       " 'yapf                      0.22.0                    <pip>',\n",
       " 'zeromq                    4.1.5                         0  ',\n",
       " 'zict                      0.1.2                    py27_0  ',\n",
       " 'zlib                      1.2.11               ha838bed_2  ']"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conda_modules"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
