---
title: "DESeq2_Analysis"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
shhh <- suppressPackageStartupMessages # It's a library, so shhh!
shhh(library("tximport"))
shhh(library("readr"))
shhh(library("tximportData"))
shhh(library("dplyr"))
shhh(library(ggfortify))
shhh(library("data.table")  )
shhh(library("ggplot2"))
shhh(library("AnnotationDbi"))
shhh(library("DESeq2"))
shhh(library("EnhancedVolcano"))
shhh(library("dplyr") )
shhh(library("RColorBrewer"))
library("dplyr")
shhh(library("pheatmap"))
#shhh(library("sva"))
shhh(library("tidyr"))
shhh(library("stringr"))
shhh(source("dt.to.df.R"))
source("get_gene_lengths.R")
library(yaml)
library(GenomicFeatures)

source("runDEandSave.R")
```

```{r}
doc <- read_yaml('../parameters/params.yaml')
```

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = setwd(doc$results))

```

## Load sample info and merge with proportion
```{r}
#dir <- "../Data"
dir <- "../parameters/"

samples <- fread(file.path(dir,"rna_samples.csv"), sep=",", header=TRUE, colClasses = 'factor', stringsAsFactors = TRUE)

colnames(samples) <- make.names(colnames(samples))
#samples$Number <- as.integer(as.character(samples$Number))

```

## Generate files_dt , which is samples. 
```{r}
#files <- paste0(doc["results"],samples[,Number],"count.txt")#list.files(file.path(dir,"salmon_noTG"),full.names = TRUE)
ht_dir <- file.path(doc["data_folder"], "RNAseq","htseq_rev")
files <- list.files(ht_dir,pattern="*count.txt")
files_dt <- data.table(samples) 

# Convert Number to character
files_dt[,"File"] <- as.character(files_dt[,File])

#setwd(ht_dir)
#count <- tximport(paste0(files_dt[,File],'.count.txt'), txtIn=FALSE, tx2gene = hamster.tx2gene)
```
```{r}
setwd(file.path(doc["results"],'genome','picr','gcf'))
gene_l <-gene_lengths(doc$annotation_gtf, doc$ref_fa, id=gene)
gene_l
write.csv(gene_l, 'gene_lengths.csv')
```

```{r}
ddsHTSeq <- DESeqDataSetFromHTSeqCount(sampleTable = samples,
                                       directory = "/data/isshamie/TSS/NCBI_PICR_103_Processed/RNAseq/htseq_rev",
                                       design= ~ Tissue)
ddsHTSeq
```


```{r}
genome_dir <- "/data/isshamie/genome/mouse/GCF_000001635.26_GRCm38.p6/"
#mouse.txdb <- GenomicFeatures::makeTxDbFromGFF(file.path(genome_dir,"GCF_000001635.26_GRCm38.p6_genomic.gff"))
hamster.txdb <- makeTxDbFromGFF(file.path(doc["annotation"]))

#hamster.k <- AnnotationDbi::keys(hamster.txdb, keytype = "TXNAME")
#hamster.tx2gene <- AnnotationDbi::select(hamster.txdb, hamster.k, "GENEID", "TXNAME")
```

```{r}
exonic <- exonsBy(hamster.txdb, by="gene")
red.exonic <- reduce(exonic)
exon.lengths <- sum(width(red.exonic))
```

#A1cf3081
## Generate contrasts
```{r}
geno <- levels(samples[,Genotype])
contrast_v = matrix(nrow=0,ncol=3)
for (i in 1:(length(geno)-1)){
  for (j in (i+1):(length(geno))){
    contrast_v <- rbind(contrast_v,c("Genotype",geno[i],geno[j]))
  }
}
```

## Run DE
```{r}
dds <- DESeqDataSetFromTximport(count,colData = dt.to.df(files_dt),design = ~Genotype)
keep <- rowSums(counts(dds)) >= dim(counts(dds))[2]*3 #10
dds <- dds[keep,]

runDEandSave(dds,fsave ="man_genotype",contrast_v = contrast_v)
```


