{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pickle\n",
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ref_fa = '/data/isshamie/genome/picr.fa'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read in tss experimental and all start sites and merge them. Set the actual_start to be tss experimental when there is one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:15: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  from ipykernel import kernelapp as app\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Score</th>\n",
       "      <th>Strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>Tissues</th>\n",
       "      <th>cs</th>\n",
       "      <th>Length</th>\n",
       "      <th>actual_start</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>RRP7A;gene_2;asmbl_4;CHO;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>45849</td>\n",
       "      <td>45999</td>\n",
       "      <td>3.040000</td>\n",
       "      <td>-</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>CHO</td>\n",
       "      <td>cs=1</td>\n",
       "      <td>150</td>\n",
       "      <td>45923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RRP7A;gene_2;asmbl_10;Brain,CHO,Spleen;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>44409</td>\n",
       "      <td>44566</td>\n",
       "      <td>10.105000</td>\n",
       "      <td>-</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>asmbl_10</td>\n",
       "      <td>Brain,CHO,Spleen</td>\n",
       "      <td>cs=1</td>\n",
       "      <td>157</td>\n",
       "      <td>44487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SERHL;gene_1_4;asmbl_14;BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscle,Kidney,Pancreas;cs=2</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>20879</td>\n",
       "      <td>21067</td>\n",
       "      <td>96.550003</td>\n",
       "      <td>+</td>\n",
       "      <td>SERHL</td>\n",
       "      <td>gene_1_4</td>\n",
       "      <td>asmbl_14</td>\n",
       "      <td>BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...</td>\n",
       "      <td>cs=2</td>\n",
       "      <td>188</td>\n",
       "      <td>20972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POLDIP3;gene_3_1;asmbl_17;BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscle,Kidney,Pancreas;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>98719</td>\n",
       "      <td>98916</td>\n",
       "      <td>146.399994</td>\n",
       "      <td>-</td>\n",
       "      <td>POLDIP3</td>\n",
       "      <td>gene_3_1</td>\n",
       "      <td>asmbl_17</td>\n",
       "      <td>BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...</td>\n",
       "      <td>cs=1</td>\n",
       "      <td>197</td>\n",
       "      <td>98817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POLDIP3;gene_3_1;asmbl_19;BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscle,Kidney;cs=3</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>78177</td>\n",
       "      <td>78416</td>\n",
       "      <td>34.799999</td>\n",
       "      <td>-</td>\n",
       "      <td>POLDIP3</td>\n",
       "      <td>gene_3_1</td>\n",
       "      <td>asmbl_19</td>\n",
       "      <td>BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...</td>\n",
       "      <td>cs=3</td>\n",
       "      <td>239</td>\n",
       "      <td>78296</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Chr  Start    End  \\\n",
       "3                                                                          \n",
       "RRP7A;gene_2;asmbl_4;CHO;cs=1                       picr_0  45849  45999   \n",
       "RRP7A;gene_2;asmbl_10;Brain,CHO,Spleen;cs=1         picr_0  44409  44566   \n",
       "SERHL;gene_1_4;asmbl_14;BMDMwt,Heart,Spleen,CHO...  picr_0  20879  21067   \n",
       "POLDIP3;gene_3_1;asmbl_17;BMDMwt,Heart,Spleen,C...  picr_0  98719  98916   \n",
       "POLDIP3;gene_3_1;asmbl_19;BMDMwt,Heart,Spleen,C...  picr_0  78177  78416   \n",
       "\n",
       "                                                         Score Strand  \\\n",
       "3                                                                       \n",
       "RRP7A;gene_2;asmbl_4;CHO;cs=1                         3.040000      -   \n",
       "RRP7A;gene_2;asmbl_10;Brain,CHO,Spleen;cs=1          10.105000      -   \n",
       "SERHL;gene_1_4;asmbl_14;BMDMwt,Heart,Spleen,CHO...   96.550003      +   \n",
       "POLDIP3;gene_3_1;asmbl_17;BMDMwt,Heart,Spleen,C...  146.399994      -   \n",
       "POLDIP3;gene_3_1;asmbl_19;BMDMwt,Heart,Spleen,C...   34.799999      -   \n",
       "\n",
       "                                                       gene   gene_id  \\\n",
       "3                                                                       \n",
       "RRP7A;gene_2;asmbl_4;CHO;cs=1                         RRP7A    gene_2   \n",
       "RRP7A;gene_2;asmbl_10;Brain,CHO,Spleen;cs=1           RRP7A    gene_2   \n",
       "SERHL;gene_1_4;asmbl_14;BMDMwt,Heart,Spleen,CHO...    SERHL  gene_1_4   \n",
       "POLDIP3;gene_3_1;asmbl_17;BMDMwt,Heart,Spleen,C...  POLDIP3  gene_3_1   \n",
       "POLDIP3;gene_3_1;asmbl_19;BMDMwt,Heart,Spleen,C...  POLDIP3  gene_3_1   \n",
       "\n",
       "                                                   transcript_id  \\\n",
       "3                                                                  \n",
       "RRP7A;gene_2;asmbl_4;CHO;cs=1                            asmbl_4   \n",
       "RRP7A;gene_2;asmbl_10;Brain,CHO,Spleen;cs=1             asmbl_10   \n",
       "SERHL;gene_1_4;asmbl_14;BMDMwt,Heart,Spleen,CHO...      asmbl_14   \n",
       "POLDIP3;gene_3_1;asmbl_17;BMDMwt,Heart,Spleen,C...      asmbl_17   \n",
       "POLDIP3;gene_3_1;asmbl_19;BMDMwt,Heart,Spleen,C...      asmbl_19   \n",
       "\n",
       "                                                                                              Tissues  \\\n",
       "3                                                                                                       \n",
       "RRP7A;gene_2;asmbl_4;CHO;cs=1                                                                     CHO   \n",
       "RRP7A;gene_2;asmbl_10;Brain,CHO,Spleen;cs=1                                          Brain,CHO,Spleen   \n",
       "SERHL;gene_1_4;asmbl_14;BMDMwt,Heart,Spleen,CHO...  BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...   \n",
       "POLDIP3;gene_3_1;asmbl_17;BMDMwt,Heart,Spleen,C...  BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...   \n",
       "POLDIP3;gene_3_1;asmbl_19;BMDMwt,Heart,Spleen,C...  BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...   \n",
       "\n",
       "                                                      cs  Length  actual_start  \n",
       "3                                                                               \n",
       "RRP7A;gene_2;asmbl_4;CHO;cs=1                       cs=1     150         45923  \n",
       "RRP7A;gene_2;asmbl_10;Brain,CHO,Spleen;cs=1         cs=1     157         44487  \n",
       "SERHL;gene_1_4;asmbl_14;BMDMwt,Heart,Spleen,CHO...  cs=2     188         20972  \n",
       "POLDIP3;gene_3_1;asmbl_17;BMDMwt,Heart,Spleen,C...  cs=1     197         98817  \n",
       "POLDIP3;gene_3_1;asmbl_19;BMDMwt,Heart,Spleen,C...  cs=3     239         78296  "
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "tss = pd.read_csv('Results/tss_final/tss_peakCenter_1kb_1kb.bed',sep='\\t',header=None, index_col =3)\n",
    " \n",
    "#Munge data for downstream processing\n",
    "cols = tss.columns.values\n",
    "cols = ['Chr','Start','End','Score','Strand']\n",
    "tss.columns= cols\n",
    "\n",
    "name_vals = ['gene','gene_id','transcript_id','Tissues','cs']\n",
    "\n",
    "tmp_tss = tss.copy()\n",
    "for ind in tmp_tss.index.values:\n",
    "    curr = ind.split(';')\n",
    "    for i,val in enumerate(curr):\n",
    "        tmp_tss.set_value(ind,name_vals[i],val)\n",
    "\n",
    "tmp_tss['Length'] = tmp_tss['End'] - tmp_tss['Start']\n",
    "\n",
    "tss = tmp_tss\n",
    "tss['actual_start'] = tss.apply(lambda x: int(np.floor(((x['Start']+x['End'])-1)/2)) \n",
    "                if x['Strand'] == '+' else int(np.ceil(((x['Start']+x['End'])-1)/2)),axis=1)\n",
    "\n",
    "#tss['actual_start'] = tss.apply( lambda x: x['Start'] if x['Strand'] == '+' else x['End'],axis=1)\n",
    "\n",
    "## Just in case make a copy\n",
    "tss['actual_start_copy'] = tss['actual_start']\n",
    "\n",
    "tss.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>cs</th>\n",
       "      <th>Length</th>\n",
       "      <th>actual_start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gene=SERHL;transcript_id=asmbl_2;cs=2</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>176</td>\n",
       "      <td>38694</td>\n",
       "      <td>+</td>\n",
       "      <td>asmbl_2</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>SERHL</td>\n",
       "      <td>asmbl_2</td>\n",
       "      <td>2</td>\n",
       "      <td>38518</td>\n",
       "      <td>176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_4;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7715</td>\n",
       "      <td>46630</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>1</td>\n",
       "      <td>38915</td>\n",
       "      <td>46630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_7;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7759</td>\n",
       "      <td>44597</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_7</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_7</td>\n",
       "      <td>1</td>\n",
       "      <td>36838</td>\n",
       "      <td>44597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_8;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7767</td>\n",
       "      <td>44597</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_8</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_8</td>\n",
       "      <td>1</td>\n",
       "      <td>36830</td>\n",
       "      <td>44597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_9;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7789</td>\n",
       "      <td>44568</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_9</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_9</td>\n",
       "      <td>1</td>\n",
       "      <td>36779</td>\n",
       "      <td>44568</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Chr  Start    End  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  picr_0    176  38694   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7715  46630   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7759  44597   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7767  44597   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7789  44568   \n",
       "\n",
       "                                                   Strand       ID    Parent  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...      +  asmbl_2  gene_1_2   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_4    gene_2   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_7    gene_2   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_8    gene_2   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_9    gene_2   \n",
       "\n",
       "                                                     gene_id   gene  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  gene_1_2  SERHL   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "\n",
       "                                                   transcript_id  cs  Length  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...       asmbl_2   2   38518   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_4   1   38915   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_7   1   36838   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_8   1   36830   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_9   1   36779   \n",
       "\n",
       "                                                    actual_start  \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...           176  \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...         46630  \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...         44597  \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...         44597  \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...         44568  "
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_start_site = pd.read_csv('/data/isshamie/genome/start_site_mRNA_updated_final_sort.tsv',index_col = 0,sep='\\t')\n",
    "annotation_start_site.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(52382, 12)"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_start_site.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge the two "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss['actual_start_copy'] = tss['actual_start']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(52382, 21)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start_x</th>\n",
       "      <th>End_x</th>\n",
       "      <th>Score</th>\n",
       "      <th>Strand</th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>Tissues</th>\n",
       "      <th>cs_x</th>\n",
       "      <th>Length_x</th>\n",
       "      <th>...</th>\n",
       "      <th>Start_y</th>\n",
       "      <th>End_y</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>cs_y</th>\n",
       "      <th>Length_y</th>\n",
       "      <th>actual_start_y</th>\n",
       "      <th>actual_start</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcript_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>asmbl_4</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>45849.0</td>\n",
       "      <td>45999.0</td>\n",
       "      <td>3.040000</td>\n",
       "      <td>-</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>CHO</td>\n",
       "      <td>cs=1</td>\n",
       "      <td>150.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7715</td>\n",
       "      <td>46630</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>1</td>\n",
       "      <td>38915</td>\n",
       "      <td>46630</td>\n",
       "      <td>45923.0</td>\n",
       "      <td>7715</td>\n",
       "      <td>46630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_10</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>44409.0</td>\n",
       "      <td>44566.0</td>\n",
       "      <td>10.105000</td>\n",
       "      <td>-</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>Brain,CHO,Spleen</td>\n",
       "      <td>cs=1</td>\n",
       "      <td>157.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7821</td>\n",
       "      <td>44513</td>\n",
       "      <td>asmbl_10</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>1</td>\n",
       "      <td>36692</td>\n",
       "      <td>44513</td>\n",
       "      <td>44487.0</td>\n",
       "      <td>7821</td>\n",
       "      <td>44513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_14</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>20879.0</td>\n",
       "      <td>21067.0</td>\n",
       "      <td>96.550003</td>\n",
       "      <td>+</td>\n",
       "      <td>SERHL</td>\n",
       "      <td>gene_1_4</td>\n",
       "      <td>BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...</td>\n",
       "      <td>cs=2</td>\n",
       "      <td>188.0</td>\n",
       "      <td>...</td>\n",
       "      <td>20942</td>\n",
       "      <td>38694</td>\n",
       "      <td>asmbl_14</td>\n",
       "      <td>gene_1_4</td>\n",
       "      <td>2</td>\n",
       "      <td>17752</td>\n",
       "      <td>20942</td>\n",
       "      <td>20972.0</td>\n",
       "      <td>20942</td>\n",
       "      <td>38694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_17</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>98719.0</td>\n",
       "      <td>98916.0</td>\n",
       "      <td>146.399994</td>\n",
       "      <td>-</td>\n",
       "      <td>POLDIP3</td>\n",
       "      <td>gene_3_1</td>\n",
       "      <td>BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...</td>\n",
       "      <td>cs=1</td>\n",
       "      <td>197.0</td>\n",
       "      <td>...</td>\n",
       "      <td>49679</td>\n",
       "      <td>98922</td>\n",
       "      <td>asmbl_17</td>\n",
       "      <td>gene_3_1</td>\n",
       "      <td>1</td>\n",
       "      <td>49243</td>\n",
       "      <td>98922</td>\n",
       "      <td>98817.0</td>\n",
       "      <td>49679</td>\n",
       "      <td>98922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_19</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>78177.0</td>\n",
       "      <td>78416.0</td>\n",
       "      <td>34.799999</td>\n",
       "      <td>-</td>\n",
       "      <td>POLDIP3</td>\n",
       "      <td>gene_3_1</td>\n",
       "      <td>BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...</td>\n",
       "      <td>cs=3</td>\n",
       "      <td>239.0</td>\n",
       "      <td>...</td>\n",
       "      <td>64336</td>\n",
       "      <td>78255</td>\n",
       "      <td>asmbl_19</td>\n",
       "      <td>gene_3_1</td>\n",
       "      <td>3</td>\n",
       "      <td>13919</td>\n",
       "      <td>78255</td>\n",
       "      <td>78296.0</td>\n",
       "      <td>64336</td>\n",
       "      <td>78255</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Chr  Start_x    End_x       Score Strand     gene   gene_id  \\\n",
       "transcript_id                                                                   \n",
       "asmbl_4        picr_0  45849.0  45999.0    3.040000      -    RRP7A    gene_2   \n",
       "asmbl_10       picr_0  44409.0  44566.0   10.105000      -    RRP7A    gene_2   \n",
       "asmbl_14       picr_0  20879.0  21067.0   96.550003      +    SERHL  gene_1_4   \n",
       "asmbl_17       picr_0  98719.0  98916.0  146.399994      -  POLDIP3  gene_3_1   \n",
       "asmbl_19       picr_0  78177.0  78416.0   34.799999      -  POLDIP3  gene_3_1   \n",
       "\n",
       "                                                         Tissues  cs_x  \\\n",
       "transcript_id                                                            \n",
       "asmbl_4                                                      CHO  cs=1   \n",
       "asmbl_10                                        Brain,CHO,Spleen  cs=1   \n",
       "asmbl_14       BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...  cs=2   \n",
       "asmbl_17       BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...  cs=1   \n",
       "asmbl_19       BMDMwt,Heart,Spleen,CHO,Brain,Lung,Liver,Muscl...  cs=3   \n",
       "\n",
       "               Length_x  ...    Start_y  End_y        ID    Parent cs_y  \\\n",
       "transcript_id            ...                                              \n",
       "asmbl_4           150.0  ...       7715  46630   asmbl_4    gene_2    1   \n",
       "asmbl_10          157.0  ...       7821  44513  asmbl_10    gene_2    1   \n",
       "asmbl_14          188.0  ...      20942  38694  asmbl_14  gene_1_4    2   \n",
       "asmbl_17          197.0  ...      49679  98922  asmbl_17  gene_3_1    1   \n",
       "asmbl_19          239.0  ...      64336  78255  asmbl_19  gene_3_1    3   \n",
       "\n",
       "              Length_y  actual_start_y  actual_start  Start    End  \n",
       "transcript_id                                                       \n",
       "asmbl_4          38915           46630       45923.0   7715  46630  \n",
       "asmbl_10         36692           44513       44487.0   7821  44513  \n",
       "asmbl_14         17752           20942       20972.0  20942  38694  \n",
       "asmbl_17         49243           98922       98817.0  49679  98922  \n",
       "asmbl_19         13919           78255       78296.0  64336  78255  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = pd.merge(tss,annotation_start_site,on=['transcript_id','gene','gene_id','Strand','Chr'],how='outer')#,left_on='actual_start')\n",
    "\n",
    "## Set actual_start to be from left if its not null otherwise right\n",
    "merged['actual_start'] = merged.apply(lambda x: x['actual_start_x'] if not np.isnan(x['actual_start_x']) \n",
    "                                      else x['actual_start_y'], axis=1)\n",
    "print(merged.shape)\n",
    "\n",
    "merged['Start'] = merged['Start_y']\n",
    "merged['End'] = merged['End_y']\n",
    "\n",
    "merged = merged.set_index('transcript_id')\n",
    "\n",
    "# Sort by chr, start and end\n",
    "merged = merged.sort_values(['Chr','Start','End'])\n",
    "merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load CDS and setup for later merging"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:15: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  from ipykernel import kernelapp as app\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 30s, sys: 4.42 s, total: 1min 35s\n",
      "Wall time: 58.4 s\n"
     ]
    }
   ],
   "source": [
    "%%time \n",
    "cds = pd.read_csv('/data/isshamie/genome/CDS.gff3',sep='\\t',header=None,index_col=8)\n",
    "#Munge data for downstream processing\n",
    "cols = cds.columns.values\n",
    "cols = ['Chr','Merge','Type','Start','End','.','Strand','..']\n",
    "cds.columns= cols\n",
    "#Drop the unncesseary columns\n",
    "cds.drop(['Merge','.','..','Type'],axis=1,inplace=True)\n",
    "\n",
    "## Setup actual_start for cds\n",
    "cds['actual_start'] = cds.apply(lambda x: int(x['Start']) if x['Strand'] == '+' else int(x['End']),axis=1)\n",
    "\n",
    "\n",
    "tmp_cds = cds.copy()\n",
    "for ind in tmp_cds.index.values:\n",
    "    curr = ind.split(';')\n",
    "    for i in curr:\n",
    "        curr_split = i.split('=')\n",
    "        tmp_cds.set_value(ind,curr_split[0],curr_split[1])\n",
    "\n",
    "tmp_cds['Length'] = tmp_cds['End'] - tmp_cds['Start']\n",
    "\n",
    "cds = tmp_cds\n",
    "cds = cds.rename(columns={'protein_id':'transcript_id'})\n",
    "cds.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save as bed file and extract sequences"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Extract sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  import sys\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  if __name__ == '__main__':\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  # Remove the CWD from sys.path while we load stuff.\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  # This is added back by InteractiveShellApp.init_path()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>ID</th>\n",
       "      <th>Score</th>\n",
       "      <th>Strand</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcript_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>asmbl_2</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>-1824</td>\n",
       "      <td>2177</td>\n",
       "      <td>SERHL;gene_1_2;asmbl_2</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_4</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>43923</td>\n",
       "      <td>47924</td>\n",
       "      <td>RRP7A;gene_2;asmbl_4</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_7</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>42597</td>\n",
       "      <td>46598</td>\n",
       "      <td>RRP7A;gene_2;asmbl_7</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_8</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>42597</td>\n",
       "      <td>46598</td>\n",
       "      <td>RRP7A;gene_2;asmbl_8</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_9</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>42568</td>\n",
       "      <td>46569</td>\n",
       "      <td>RRP7A;gene_2;asmbl_9</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_10</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>42487</td>\n",
       "      <td>46488</td>\n",
       "      <td>RRP7A;gene_2;asmbl_10</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_11</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>16149</td>\n",
       "      <td>20150</td>\n",
       "      <td>SERHL;gene_1_4;asmbl_11</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_12</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>18787</td>\n",
       "      <td>22788</td>\n",
       "      <td>SERHL;gene_1_4;asmbl_12</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_14</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>18972</td>\n",
       "      <td>22973</td>\n",
       "      <td>SERHL;gene_1_4;asmbl_14</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_15</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>20504</td>\n",
       "      <td>24505</td>\n",
       "      <td>SERHL;gene_1_4;asmbl_15</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_16</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>42549</td>\n",
       "      <td>46550</td>\n",
       "      <td>RRP7A;gene_2;asmbl_16</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_17</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>96817</td>\n",
       "      <td>100818</td>\n",
       "      <td>POLDIP3;gene_3_1;asmbl_17</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_18</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>63510</td>\n",
       "      <td>67511</td>\n",
       "      <td>POLDIP3;gene_3_1;asmbl_18</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_19</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>76296</td>\n",
       "      <td>80297</td>\n",
       "      <td>POLDIP3;gene_3_1;asmbl_19</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_20</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>90199</td>\n",
       "      <td>94200</td>\n",
       "      <td>CYB5R3;gene_3_2;asmbl_20</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_21</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>183875</td>\n",
       "      <td>187876</td>\n",
       "      <td>A4GALT;gene_4_1;asmbl_21</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_22</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>183760</td>\n",
       "      <td>187761</td>\n",
       "      <td>A4GALT;gene_4_3;asmbl_22</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_24</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>184288</td>\n",
       "      <td>188289</td>\n",
       "      <td>A4GALT;gene_4_3;asmbl_24</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_30</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>183727</td>\n",
       "      <td>187728</td>\n",
       "      <td>A4GALT;gene_4_3;asmbl_30</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_27</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>188544</td>\n",
       "      <td>192545</td>\n",
       "      <td>A4GALT;gene_4_3;asmbl_27</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_31</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>296212</td>\n",
       "      <td>300213</td>\n",
       "      <td>ARFGAP3;gene_5_1;asmbl_31</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_33</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>273669</td>\n",
       "      <td>277670</td>\n",
       "      <td>ARFGAP3;gene_5_1;asmbl_33</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_32</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>296241</td>\n",
       "      <td>300242</td>\n",
       "      <td>ARFGAP3;gene_5_1;asmbl_32</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_39</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>396907</td>\n",
       "      <td>400908</td>\n",
       "      <td>PACSIN2;gene_6;asmbl_39</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_41</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>396870</td>\n",
       "      <td>400871</td>\n",
       "      <td>PACSIN2;gene_6;asmbl_41</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_42</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>396885</td>\n",
       "      <td>400886</td>\n",
       "      <td>PACSIN2;gene_6;asmbl_42</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_43</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>396822</td>\n",
       "      <td>400823</td>\n",
       "      <td>PACSIN2;gene_6;asmbl_43</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_37</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>362375</td>\n",
       "      <td>366376</td>\n",
       "      <td>PACSIN2;gene_6;asmbl_37</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_44</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>396907</td>\n",
       "      <td>400908</td>\n",
       "      <td>PACSIN2;gene_6;asmbl_44</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_46</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>396878</td>\n",
       "      <td>400879</td>\n",
       "      <td>PACSIN2;gene_6;asmbl_46</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86715</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>1659205</td>\n",
       "      <td>1663206</td>\n",
       "      <td>NOC3L;gene_30652;asmbl_86715</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86717</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>1814047</td>\n",
       "      <td>1818048</td>\n",
       "      <td>NOVA1;gene_30653;asmbl_86717</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86718</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>1814550</td>\n",
       "      <td>1818551</td>\n",
       "      <td>NOVA1;gene_30653;asmbl_86718</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86719</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>1815516</td>\n",
       "      <td>1819517</td>\n",
       "      <td>NOVA1;gene_30653;asmbl_86719</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86720</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>1816096</td>\n",
       "      <td>1820097</td>\n",
       "      <td>NOVA1;gene_30653;asmbl_86720</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86722</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>2237032</td>\n",
       "      <td>2241033</td>\n",
       "      <td>ODC1;gene_30655;asmbl_86722</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86745</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>2898175</td>\n",
       "      <td>2902176</td>\n",
       "      <td>ALG12;gene_30659_2;asmbl_86745</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86746</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>3281730</td>\n",
       "      <td>3285731</td>\n",
       "      <td>BCLAF1;gene_30660;asmbl_86746</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86747</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>3426604</td>\n",
       "      <td>3430605</td>\n",
       "      <td>STXBP6;gene_30661;asmbl_86747</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86748</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>3483279</td>\n",
       "      <td>3487280</td>\n",
       "      <td>STXBP6;gene_30661;asmbl_86748</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86749</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>3707553</td>\n",
       "      <td>3711554</td>\n",
       "      <td>LYPLA1;gene_30662;asmbl_86749</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86750</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>3778406</td>\n",
       "      <td>3782407</td>\n",
       "      <td>ERVK-10;gene_30663;asmbl_86750</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86753</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>4060714</td>\n",
       "      <td>4064715</td>\n",
       "      <td>NRCAM;gene_30665;asmbl_86753</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86752</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>4060761</td>\n",
       "      <td>4064762</td>\n",
       "      <td>NRCAM;gene_30665;asmbl_86752</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86760</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>3931613</td>\n",
       "      <td>3935614</td>\n",
       "      <td>NRCAM;gene_30665;asmbl_86760</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86762</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>3931613</td>\n",
       "      <td>3935614</td>\n",
       "      <td>NRCAM;gene_30665;asmbl_86762</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86764</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>4128605</td>\n",
       "      <td>4132606</td>\n",
       "      <td>PNPLA8;gene_30666;asmbl_86764</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86765</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>4138492</td>\n",
       "      <td>4142493</td>\n",
       "      <td>DNAJB9;gene_30667;asmbl_86765</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86766</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>4146701</td>\n",
       "      <td>4150702</td>\n",
       "      <td>DNAJB9;gene_30667;asmbl_86766</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86767</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>4147349</td>\n",
       "      <td>4151350</td>\n",
       "      <td>DNAJB9;gene_30667;asmbl_86767</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86768</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>4148222</td>\n",
       "      <td>4152223</td>\n",
       "      <td>DNAJB9;gene_30667;asmbl_86768</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86773</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>5681256</td>\n",
       "      <td>5685257</td>\n",
       "      <td>AASDHPPT;gene_30671;asmbl_86773</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86774</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>6273077</td>\n",
       "      <td>6277078</td>\n",
       "      <td>ENV;gene_30672;asmbl_86774</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86776</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>6821783</td>\n",
       "      <td>6825784</td>\n",
       "      <td>LRRN3;gene_30674;asmbl_86776</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86777</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>6822757</td>\n",
       "      <td>6826758</td>\n",
       "      <td>LRRN3;gene_30674;asmbl_86777</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86779</th>\n",
       "      <td>picr_99</td>\n",
       "      <td>6822889</td>\n",
       "      <td>6826890</td>\n",
       "      <td>LRRN3;gene_30674;asmbl_86779</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86784</th>\n",
       "      <td>picr_991</td>\n",
       "      <td>12644</td>\n",
       "      <td>16645</td>\n",
       "      <td>IGKV2-30;gene_30675;asmbl_86784</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86785</th>\n",
       "      <td>picr_996</td>\n",
       "      <td>18730</td>\n",
       "      <td>22731</td>\n",
       "      <td>EFNB2;gene_30676;asmbl_86785</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86788</th>\n",
       "      <td>picr_997</td>\n",
       "      <td>10858</td>\n",
       "      <td>14859</td>\n",
       "      <td>KV3AK_MOUSE;gene_30677;asmbl_86788</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_86789</th>\n",
       "      <td>picr_997</td>\n",
       "      <td>28011</td>\n",
       "      <td>32012</td>\n",
       "      <td>KV3A9_MOUSE;gene_30678;asmbl_86789</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52382 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Chr    Start      End                                  ID  \\\n",
       "transcript_id                                                                   \n",
       "asmbl_2          picr_0    -1824     2177              SERHL;gene_1_2;asmbl_2   \n",
       "asmbl_4          picr_0    43923    47924                RRP7A;gene_2;asmbl_4   \n",
       "asmbl_7          picr_0    42597    46598                RRP7A;gene_2;asmbl_7   \n",
       "asmbl_8          picr_0    42597    46598                RRP7A;gene_2;asmbl_8   \n",
       "asmbl_9          picr_0    42568    46569                RRP7A;gene_2;asmbl_9   \n",
       "asmbl_10         picr_0    42487    46488               RRP7A;gene_2;asmbl_10   \n",
       "asmbl_11         picr_0    16149    20150             SERHL;gene_1_4;asmbl_11   \n",
       "asmbl_12         picr_0    18787    22788             SERHL;gene_1_4;asmbl_12   \n",
       "asmbl_14         picr_0    18972    22973             SERHL;gene_1_4;asmbl_14   \n",
       "asmbl_15         picr_0    20504    24505             SERHL;gene_1_4;asmbl_15   \n",
       "asmbl_16         picr_0    42549    46550               RRP7A;gene_2;asmbl_16   \n",
       "asmbl_17         picr_0    96817   100818           POLDIP3;gene_3_1;asmbl_17   \n",
       "asmbl_18         picr_0    63510    67511           POLDIP3;gene_3_1;asmbl_18   \n",
       "asmbl_19         picr_0    76296    80297           POLDIP3;gene_3_1;asmbl_19   \n",
       "asmbl_20         picr_0    90199    94200            CYB5R3;gene_3_2;asmbl_20   \n",
       "asmbl_21         picr_0   183875   187876            A4GALT;gene_4_1;asmbl_21   \n",
       "asmbl_22         picr_0   183760   187761            A4GALT;gene_4_3;asmbl_22   \n",
       "asmbl_24         picr_0   184288   188289            A4GALT;gene_4_3;asmbl_24   \n",
       "asmbl_30         picr_0   183727   187728            A4GALT;gene_4_3;asmbl_30   \n",
       "asmbl_27         picr_0   188544   192545            A4GALT;gene_4_3;asmbl_27   \n",
       "asmbl_31         picr_0   296212   300213           ARFGAP3;gene_5_1;asmbl_31   \n",
       "asmbl_33         picr_0   273669   277670           ARFGAP3;gene_5_1;asmbl_33   \n",
       "asmbl_32         picr_0   296241   300242           ARFGAP3;gene_5_1;asmbl_32   \n",
       "asmbl_39         picr_0   396907   400908             PACSIN2;gene_6;asmbl_39   \n",
       "asmbl_41         picr_0   396870   400871             PACSIN2;gene_6;asmbl_41   \n",
       "asmbl_42         picr_0   396885   400886             PACSIN2;gene_6;asmbl_42   \n",
       "asmbl_43         picr_0   396822   400823             PACSIN2;gene_6;asmbl_43   \n",
       "asmbl_37         picr_0   362375   366376             PACSIN2;gene_6;asmbl_37   \n",
       "asmbl_44         picr_0   396907   400908             PACSIN2;gene_6;asmbl_44   \n",
       "asmbl_46         picr_0   396878   400879             PACSIN2;gene_6;asmbl_46   \n",
       "...                 ...      ...      ...                                 ...   \n",
       "asmbl_86715     picr_99  1659205  1663206        NOC3L;gene_30652;asmbl_86715   \n",
       "asmbl_86717     picr_99  1814047  1818048        NOVA1;gene_30653;asmbl_86717   \n",
       "asmbl_86718     picr_99  1814550  1818551        NOVA1;gene_30653;asmbl_86718   \n",
       "asmbl_86719     picr_99  1815516  1819517        NOVA1;gene_30653;asmbl_86719   \n",
       "asmbl_86720     picr_99  1816096  1820097        NOVA1;gene_30653;asmbl_86720   \n",
       "asmbl_86722     picr_99  2237032  2241033         ODC1;gene_30655;asmbl_86722   \n",
       "asmbl_86745     picr_99  2898175  2902176      ALG12;gene_30659_2;asmbl_86745   \n",
       "asmbl_86746     picr_99  3281730  3285731       BCLAF1;gene_30660;asmbl_86746   \n",
       "asmbl_86747     picr_99  3426604  3430605       STXBP6;gene_30661;asmbl_86747   \n",
       "asmbl_86748     picr_99  3483279  3487280       STXBP6;gene_30661;asmbl_86748   \n",
       "asmbl_86749     picr_99  3707553  3711554       LYPLA1;gene_30662;asmbl_86749   \n",
       "asmbl_86750     picr_99  3778406  3782407      ERVK-10;gene_30663;asmbl_86750   \n",
       "asmbl_86753     picr_99  4060714  4064715        NRCAM;gene_30665;asmbl_86753   \n",
       "asmbl_86752     picr_99  4060761  4064762        NRCAM;gene_30665;asmbl_86752   \n",
       "asmbl_86760     picr_99  3931613  3935614        NRCAM;gene_30665;asmbl_86760   \n",
       "asmbl_86762     picr_99  3931613  3935614        NRCAM;gene_30665;asmbl_86762   \n",
       "asmbl_86764     picr_99  4128605  4132606       PNPLA8;gene_30666;asmbl_86764   \n",
       "asmbl_86765     picr_99  4138492  4142493       DNAJB9;gene_30667;asmbl_86765   \n",
       "asmbl_86766     picr_99  4146701  4150702       DNAJB9;gene_30667;asmbl_86766   \n",
       "asmbl_86767     picr_99  4147349  4151350       DNAJB9;gene_30667;asmbl_86767   \n",
       "asmbl_86768     picr_99  4148222  4152223       DNAJB9;gene_30667;asmbl_86768   \n",
       "asmbl_86773     picr_99  5681256  5685257     AASDHPPT;gene_30671;asmbl_86773   \n",
       "asmbl_86774     picr_99  6273077  6277078          ENV;gene_30672;asmbl_86774   \n",
       "asmbl_86776     picr_99  6821783  6825784        LRRN3;gene_30674;asmbl_86776   \n",
       "asmbl_86777     picr_99  6822757  6826758        LRRN3;gene_30674;asmbl_86777   \n",
       "asmbl_86779     picr_99  6822889  6826890        LRRN3;gene_30674;asmbl_86779   \n",
       "asmbl_86784    picr_991    12644    16645     IGKV2-30;gene_30675;asmbl_86784   \n",
       "asmbl_86785    picr_996    18730    22731        EFNB2;gene_30676;asmbl_86785   \n",
       "asmbl_86788    picr_997    10858    14859  KV3AK_MOUSE;gene_30677;asmbl_86788   \n",
       "asmbl_86789    picr_997    28011    32012  KV3A9_MOUSE;gene_30678;asmbl_86789   \n",
       "\n",
       "               Score Strand  \n",
       "transcript_id                \n",
       "asmbl_2            0      +  \n",
       "asmbl_4            0      -  \n",
       "asmbl_7            0      -  \n",
       "asmbl_8            0      -  \n",
       "asmbl_9            0      -  \n",
       "asmbl_10           0      -  \n",
       "asmbl_11           0      +  \n",
       "asmbl_12           0      +  \n",
       "asmbl_14           0      +  \n",
       "asmbl_15           0      +  \n",
       "asmbl_16           0      -  \n",
       "asmbl_17           0      -  \n",
       "asmbl_18           0      -  \n",
       "asmbl_19           0      -  \n",
       "asmbl_20           0      -  \n",
       "asmbl_21           0      -  \n",
       "asmbl_22           0      -  \n",
       "asmbl_24           0      -  \n",
       "asmbl_30           0      -  \n",
       "asmbl_27           0      -  \n",
       "asmbl_31           0      -  \n",
       "asmbl_33           0      -  \n",
       "asmbl_32           0      -  \n",
       "asmbl_39           0      -  \n",
       "asmbl_41           0      -  \n",
       "asmbl_42           0      -  \n",
       "asmbl_43           0      -  \n",
       "asmbl_37           0      -  \n",
       "asmbl_44           0      -  \n",
       "asmbl_46           0      -  \n",
       "...              ...    ...  \n",
       "asmbl_86715        0      -  \n",
       "asmbl_86717        0      +  \n",
       "asmbl_86718        0      +  \n",
       "asmbl_86719        0      +  \n",
       "asmbl_86720        0      +  \n",
       "asmbl_86722        0      +  \n",
       "asmbl_86745        0      -  \n",
       "asmbl_86746        0      -  \n",
       "asmbl_86747        0      +  \n",
       "asmbl_86748        0      +  \n",
       "asmbl_86749        0      -  \n",
       "asmbl_86750        0      -  \n",
       "asmbl_86753        0      -  \n",
       "asmbl_86752        0      -  \n",
       "asmbl_86760        0      -  \n",
       "asmbl_86762        0      -  \n",
       "asmbl_86764        0      -  \n",
       "asmbl_86765        0      +  \n",
       "asmbl_86766        0      +  \n",
       "asmbl_86767        0      +  \n",
       "asmbl_86768        0      +  \n",
       "asmbl_86773        0      -  \n",
       "asmbl_86774        0      -  \n",
       "asmbl_86776        0      +  \n",
       "asmbl_86777        0      +  \n",
       "asmbl_86779        0      +  \n",
       "asmbl_86784        0      -  \n",
       "asmbl_86785        0      +  \n",
       "asmbl_86788        0      +  \n",
       "asmbl_86789        0      +  \n",
       "\n",
       "[52382 rows x 6 columns]"
      ]
     },
     "execution_count": 242,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "bins = [-2000,2000]\n",
    "\n",
    "alltss_f = 'Results/tss_final/tss_peakCenter_1kb_1kb_ALLGENES_FLANK_2kb.bed'\n",
    "\n",
    "merged_bed = merged[['Chr','actual_start','Strand','gene_id','gene']]\n",
    "merged_bed['transcript_id'] = merged_bed.index\n",
    "merged_bed['Start']  = merged_bed['actual_start'] + bins[0]\n",
    "merged_bed['Start'] = merged_bed['Start'].astype(int)\n",
    "merged_bed['End'] = merged_bed['actual_start'] +  bins[1] + 1\n",
    "merged_bed['End'] = merged_bed['End'].astype(int)\n",
    "merged_bed['ID'] = merged_bed.apply(lambda x: x['gene'] + ';' + x['gene_id'] + ';' + x['transcript_id'],axis=1)\n",
    "merged_bed['Score'] = 0\n",
    "merged_bed = merged_bed[['Chr','Start','End','ID','Score','Strand']]\n",
    "\n",
    "merged_bed.to_csv(alltss_f,header=None,index=None,sep='\\t')\n",
    "\n",
    "merged_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {
    "deletable": false,
    "editable": false,
    "run_control": {
     "frozen": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "homerTools extract Results/tss_final/tss_peakCenter_1kb_1kb_ALLGENES_FLANK_2kb.bed /data/isshamie/genome/picr.fa -fa > Results/tss_final/tss_peakCenter_1kb_1kb_ALLGENES_FLANK_2kb.fa\n",
      "\n",
      "\tExtracting sequences from file: /data/isshamie/genome/picr.fa\n",
      "\tLooking for peak sequences in a single file (/data/isshamie/genome/picr.fa)\n",
      "\tExtracting 2093 sequences from picr_0\n",
      "\tExtracting 1147 sequences from picr_1\n",
      "\tExtracting 1913 sequences from picr_2\n",
      "\tExtracting 1117 sequences from picr_3\n",
      "\tExtracting 1063 sequences from picr_4\n",
      "\tExtracting 747 sequences from picr_5\n",
      "\tExtracting 719 sequences from picr_6\n",
      "\tExtracting 1684 sequences from picr_8\n",
      "\tExtracting 654 sequences from picr_9\n",
      "\tExtracting 767 sequences from picr_10\n",
      "\tExtracting 516 sequences from picr_11\n",
      "\tExtracting 190 sequences from picr_7b\n",
      "\tExtracting 599 sequences from picr_12\n",
      "\tExtracting 584 sequences from picr_13\n",
      "\tExtracting 506 sequences from picr_14\n",
      "\tExtracting 609 sequences from picr_15\n",
      "\tExtracting 676 sequences from picr_16\n",
      "\tExtracting 285 sequences from picr_17\n",
      "\tExtracting 381 sequences from picr_18\n",
      "\tExtracting 465 sequences from picr_19\n",
      "\tExtracting 563 sequences from picr_20\n",
      "\tExtracting 375 sequences from picr_21\n",
      "\tExtracting 1124 sequences from picr_22\n",
      "\tExtracting 688 sequences from picr_23\n",
      "\tExtracting 709 sequences from picr_24\n",
      "\tExtracting 232 sequences from picr_25\n",
      "\tExtracting 269 sequences from picr_26\n",
      "\tExtracting 594 sequences from picr_27\n",
      "\tExtracting 323 sequences from picr_28\n",
      "\tExtracting 639 sequences from picr_29\n",
      "\tExtracting 230 sequences from picr_30\n",
      "\tExtracting 265 sequences from picr_31\n",
      "\tExtracting 176 sequences from picr_32\n",
      "\tExtracting 347 sequences from picr_33\n",
      "\tExtracting 315 sequences from picr_34\n",
      "\tExtracting 958 sequences from picr_35\n",
      "\tExtracting 411 sequences from picr_36\n",
      "\tExtracting 230 sequences from picr_37\n",
      "\tExtracting 466 sequences from picr_38\n",
      "\tExtracting 473 sequences from picr_39\n",
      "\tExtracting 536 sequences from picr_40\n",
      "\tExtracting 197 sequences from picr_41\n",
      "\tExtracting 189 sequences from picr_42\n",
      "\tExtracting 349 sequences from picr_43\n",
      "\tExtracting 305 sequences from picr_44\n",
      "\tExtracting 171 sequences from picr_45\n",
      "\tExtracting 178 sequences from picr_46\n",
      "\tExtracting 449 sequences from picr_47\n",
      "\tExtracting 301 sequences from picr_48\n",
      "\tExtracting 201 sequences from picr_49\n",
      "\tExtracting 412 sequences from picr_50\n",
      "\tExtracting 197 sequences from picr_51\n",
      "\tExtracting 625 sequences from picr_52\n",
      "\tExtracting 564 sequences from picr_53\n",
      "\tExtracting 486 sequences from picr_54\n",
      "\tExtracting 221 sequences from picr_55\n",
      "\tExtracting 410 sequences from picr_56\n",
      "\tExtracting 347 sequences from picr_57\n",
      "\tExtracting 556 sequences from picr_58\n",
      "\tExtracting 560 sequences from picr_59\n",
      "\tExtracting 92 sequences from picr_60\n",
      "\tExtracting 548 sequences from picr_61\n",
      "\tExtracting 405 sequences from picr_62\n",
      "\tExtracting 217 sequences from picr_63\n",
      "\tExtracting 125 sequences from picr_64\n",
      "\tExtracting 219 sequences from picr_65\n",
      "\tExtracting 259 sequences from picr_66\n",
      "\tExtracting 112 sequences from picr_67\n",
      "\tExtracting 114 sequences from picr_68\n",
      "\tExtracting 165 sequences from picr_69\n",
      "\tExtracting 521 sequences from picr_70\n",
      "\tExtracting 306 sequences from picr_71\n",
      "\tExtracting 131 sequences from picr_72\n",
      "\tExtracting 221 sequences from picr_73\n",
      "\tExtracting 588 sequences from picr_74\n",
      "\tExtracting 280 sequences from picr_75\n",
      "\tExtracting 134 sequences from picr_76\n",
      "\tExtracting 93 sequences from picr_77\n",
      "\tExtracting 347 sequences from picr_7a\n",
      "\tExtracting 117 sequences from picr_78\n",
      "\tExtracting 225 sequences from picr_79\n",
      "\tExtracting 64 sequences from picr_80\n",
      "\tExtracting 67 sequences from picr_81\n",
      "\tExtracting 126 sequences from picr_82\n",
      "\tExtracting 243 sequences from picr_83\n",
      "\tExtracting 302 sequences from picr_84\n",
      "\tExtracting 434 sequences from picr_85\n",
      "\tExtracting 100 sequences from picr_86\n",
      "\tExtracting 276 sequences from picr_87\n",
      "\tExtracting 192 sequences from picr_88\n",
      "\tExtracting 340 sequences from picr_89\n",
      "\tExtracting 94 sequences from picr_90\n",
      "\tExtracting 117 sequences from picr_91\n",
      "\tExtracting 72 sequences from picr_92\n",
      "\tExtracting 506 sequences from picr_93\n",
      "\tExtracting 136 sequences from picr_94\n",
      "\tExtracting 117 sequences from picr_95\n",
      "\tExtracting 151 sequences from picr_96\n",
      "\tExtracting 50 sequences from picr_97\n",
      "\tExtracting 241 sequences from picr_98\n",
      "\tExtracting 27 sequences from picr_99\n",
      "\tExtracting 228 sequences from picr_100\n",
      "\tExtracting 148 sequences from picr_101\n",
      "\tExtracting 116 sequences from picr_102\n",
      "\tExtracting 391 sequences from picr_103\n",
      "\tExtracting 43 sequences from picr_104\n",
      "\tExtracting 149 sequences from picr_105\n",
      "\tExtracting 267 sequences from picr_106\n",
      "\tExtracting 40 sequences from picr_107\n",
      "\tExtracting 236 sequences from picr_108\n",
      "\tExtracting 101 sequences from picr_109\n",
      "\tExtracting 80 sequences from picr_110\n",
      "\tExtracting 136 sequences from picr_111\n",
      "\tExtracting 55 sequences from picr_112\n",
      "\tExtracting 121 sequences from picr_113\n",
      "\tExtracting 41 sequences from picr_114\n",
      "\tExtracting 125 sequences from picr_115\n",
      "\tExtracting 80 sequences from picr_116\n",
      "\tExtracting 238 sequences from picr_117\n",
      "\tExtracting 76 sequences from picr_118\n",
      "\tExtracting 402 sequences from picr_119\n",
      "\tExtracting 287 sequences from picr_120\n",
      "\tExtracting 76 sequences from picr_121\n",
      "\tExtracting 129 sequences from picr_122\n",
      "\tExtracting 116 sequences from picr_123\n",
      "\tExtracting 111 sequences from picr_124\n",
      "\tExtracting 77 sequences from picr_125\n",
      "\tExtracting 64 sequences from picr_126\n",
      "\tExtracting 220 sequences from picr_127\n",
      "\tExtracting 59 sequences from picr_128\n",
      "\tExtracting 28 sequences from picr_129\n",
      "\tExtracting 417 sequences from picr_130\n",
      "\tExtracting 79 sequences from picr_131\n",
      "\tExtracting 406 sequences from picr_132\n",
      "\tExtracting 255 sequences from picr_133\n",
      "\tExtracting 58 sequences from picr_134\n",
      "\tExtracting 80 sequences from picr_135\n",
      "\tExtracting 180 sequences from picr_136\n",
      "\tExtracting 99 sequences from picr_137\n",
      "\tExtracting 201 sequences from picr_138\n",
      "\tExtracting 228 sequences from picr_139\n",
      "\tExtracting 57 sequences from picr_140\n",
      "\tExtracting 42 sequences from picr_141\n",
      "\tExtracting 35 sequences from picr_142\n",
      "\tExtracting 30 sequences from picr_143\n",
      "\tExtracting 45 sequences from picr_144\n",
      "\tExtracting 29 sequences from picr_145\n",
      "\tExtracting 66 sequences from picr_146\n",
      "\tExtracting 7 sequences from picr_147\n",
      "\tExtracting 128 sequences from picr_148\n",
      "\tExtracting 72 sequences from picr_149\n",
      "\tExtracting 3 sequences from picr_150\n",
      "\tExtracting 44 sequences from picr_151\n",
      "\tExtracting 67 sequences from picr_152\n",
      "\tExtracting 79 sequences from picr_153\n",
      "\tExtracting 185 sequences from picr_154\n",
      "\tExtracting 65 sequences from picr_155\n",
      "\tExtracting 1 sequences from picr_156\n",
      "\tExtracting 159 sequences from picr_157\n",
      "\tExtracting 187 sequences from picr_158\n",
      "\tExtracting 29 sequences from picr_159\n",
      "\tExtracting 15 sequences from picr_160\n",
      "\tExtracting 122 sequences from picr_161\n",
      "\tExtracting 52 sequences from picr_162\n",
      "\tExtracting 55 sequences from picr_163\n",
      "\tExtracting 21 sequences from picr_164\n",
      "\tExtracting 130 sequences from picr_166\n",
      "\tExtracting 148 sequences from picr_167\n",
      "\tExtracting 27 sequences from picr_168\n",
      "\tExtracting 65 sequences from picr_169\n",
      "\tExtracting 17 sequences from picr_170\n",
      "\tExtracting 31 sequences from picr_171\n",
      "\tExtracting 19 sequences from picr_172\n",
      "\tExtracting 49 sequences from picr_173\n",
      "\tExtracting 63 sequences from picr_174\n",
      "\tExtracting 3 sequences from picr_175\n",
      "\tExtracting 2 sequences from picr_176\n",
      "\tExtracting 1 sequences from picr_177\n",
      "\tExtracting 66 sequences from picr_178\n",
      "\tExtracting 2 sequences from picr_179\n",
      "\tExtracting 7 sequences from picr_180\n",
      "\tExtracting 3 sequences from picr_181\n",
      "\tExtracting 11 sequences from picr_182\n",
      "\tExtracting 48 sequences from picr_183\n",
      "\tExtracting 5 sequences from picr_184\n",
      "\tExtracting 61 sequences from picr_185\n",
      "\tExtracting 60 sequences from picr_186\n",
      "\tExtracting 38 sequences from picr_187\n",
      "\tExtracting 26 sequences from picr_188\n",
      "\tExtracting 1 sequences from picr_190\n",
      "\tExtracting 63 sequences from picr_191\n",
      "\tExtracting 50 sequences from picr_192\n",
      "\tExtracting 19 sequences from picr_193\n",
      "\tExtracting 57 sequences from picr_195\n",
      "\tExtracting 1 sequences from picr_197\n",
      "\tExtracting 7 sequences from picr_198\n",
      "\tExtracting 29 sequences from picr_199\n",
      "\tExtracting 32 sequences from picr_200\n",
      "\tExtracting 4 sequences from picr_201\n",
      "\tExtracting 2 sequences from picr_203\n",
      "\tExtracting 3 sequences from picr_204\n",
      "\tExtracting 5 sequences from picr_205\n",
      "\tExtracting 2 sequences from picr_207\n",
      "\tExtracting 1 sequences from picr_208\n",
      "\tExtracting 41 sequences from picr_209\n",
      "\tExtracting 2 sequences from picr_210\n",
      "\tExtracting 6 sequences from picr_211\n",
      "\tExtracting 5 sequences from picr_213\n",
      "\tExtracting 1 sequences from picr_216\n",
      "\tExtracting 3 sequences from picr_220\n",
      "\tExtracting 1 sequences from picr_221\n",
      "\tExtracting 6 sequences from picr_222\n",
      "\tExtracting 1 sequences from picr_223\n",
      "\tExtracting 2 sequences from picr_226\n",
      "\tExtracting 1 sequences from picr_227\n",
      "\tExtracting 1 sequences from picr_229\n",
      "\tExtracting 8 sequences from picr_231\n",
      "\tExtracting 18 sequences from picr_232\n",
      "\tExtracting 2 sequences from picr_233\n",
      "\tExtracting 2 sequences from picr_234\n",
      "\tExtracting 20 sequences from picr_235\n",
      "\tExtracting 3 sequences from picr_237\n",
      "\tExtracting 4 sequences from picr_241\n",
      "\tExtracting 2 sequences from picr_242\n",
      "\tExtracting 20 sequences from picr_244\n",
      "\tExtracting 15 sequences from picr_245\n",
      "\tExtracting 5 sequences from picr_248\n",
      "\tExtracting 2 sequences from picr_249\n",
      "\tExtracting 5 sequences from picr_250\n",
      "\tExtracting 3 sequences from picr_251\n",
      "\tExtracting 17 sequences from picr_253\n",
      "\tExtracting 1 sequences from picr_255\n",
      "\tExtracting 2 sequences from picr_256\n",
      "\tExtracting 1 sequences from picr_260\n",
      "\tExtracting 2 sequences from picr_261\n",
      "\tExtracting 6 sequences from picr_263\n",
      "\tExtracting 2 sequences from picr_264\n",
      "\tExtracting 6 sequences from picr_265\n",
      "\tExtracting 10 sequences from picr_268\n",
      "\tExtracting 4 sequences from picr_269\n",
      "\tExtracting 3 sequences from picr_271\n",
      "\tExtracting 3 sequences from picr_272\n",
      "\tExtracting 5 sequences from picr_274\n",
      "\tExtracting 2 sequences from picr_275\n",
      "\tExtracting 4 sequences from picr_276\n",
      "\tExtracting 5 sequences from picr_279\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 1 sequences from picr_284\n",
      "\tExtracting 1 sequences from picr_285\n",
      "\tExtracting 4 sequences from picr_289\n",
      "\tExtracting 3 sequences from picr_294\n",
      "\tExtracting 1 sequences from picr_295\n",
      "\tExtracting 6 sequences from picr_296\n",
      "\tExtracting 5 sequences from picr_298\n",
      "\tExtracting 5 sequences from picr_303\n",
      "\tExtracting 1 sequences from picr_308\n",
      "\tExtracting 5 sequences from picr_309\n",
      "\tExtracting 1 sequences from picr_310\n",
      "\tExtracting 2 sequences from picr_311\n",
      "\tExtracting 15 sequences from picr_312\n",
      "\tExtracting 1 sequences from picr_314\n",
      "\tExtracting 7 sequences from picr_316\n",
      "\tExtracting 13 sequences from picr_317\n",
      "\tExtracting 3 sequences from picr_318\n",
      "\tExtracting 1 sequences from picr_319\n",
      "\tExtracting 4 sequences from picr_321\n",
      "\tExtracting 6 sequences from picr_323\n",
      "\tExtracting 3 sequences from picr_325\n",
      "\tExtracting 11 sequences from picr_328\n",
      "\tExtracting 3 sequences from picr_329\n",
      "\tExtracting 3 sequences from picr_330\n",
      "\tExtracting 7 sequences from picr_331\n",
      "\tExtracting 6 sequences from picr_332\n",
      "\tExtracting 1 sequences from picr_333\n",
      "\tExtracting 7 sequences from picr_334\n",
      "\tExtracting 1 sequences from picr_336\n",
      "\tExtracting 1 sequences from picr_338\n",
      "\tExtracting 28 sequences from picr_341\n",
      "\tExtracting 2 sequences from picr_342\n",
      "\tExtracting 4 sequences from picr_345\n",
      "\tExtracting 1 sequences from picr_347\n",
      "\tExtracting 4 sequences from picr_348\n",
      "\tExtracting 4 sequences from picr_349\n",
      "\tExtracting 2 sequences from picr_351\n",
      "\tExtracting 6 sequences from picr_353\n",
      "\tExtracting 6 sequences from picr_355\n",
      "\tExtracting 10 sequences from picr_356\n",
      "\tExtracting 6 sequences from picr_358\n",
      "\tExtracting 5 sequences from picr_360\n",
      "\tExtracting 3 sequences from picr_361\n",
      "\tExtracting 5 sequences from picr_362\n",
      "\tExtracting 3 sequences from picr_363\n",
      "\tExtracting 1 sequences from picr_364\n",
      "\tExtracting 2 sequences from picr_365\n",
      "\tExtracting 7 sequences from picr_367\n",
      "\tExtracting 1 sequences from picr_370\n",
      "\tExtracting 1 sequences from picr_372\n",
      "\tExtracting 1 sequences from picr_374\n",
      "\tExtracting 4 sequences from picr_375\n",
      "\tExtracting 10 sequences from picr_378\n",
      "\tExtracting 6 sequences from picr_380\n",
      "\tExtracting 1 sequences from picr_383\n",
      "\tExtracting 5 sequences from picr_386\n",
      "\tExtracting 4 sequences from picr_389\n",
      "\tExtracting 2 sequences from picr_390\n",
      "\tExtracting 8 sequences from picr_391\n",
      "\tExtracting 6 sequences from picr_393\n",
      "\tExtracting 1 sequences from picr_395\n",
      "\tExtracting 1 sequences from picr_397\n",
      "\tExtracting 4 sequences from picr_400\n",
      "\tExtracting 2 sequences from picr_404\n",
      "\tExtracting 5 sequences from picr_410\n",
      "\tExtracting 3 sequences from picr_412\n",
      "\tExtracting 8 sequences from picr_413\n",
      "\tExtracting 5 sequences from picr_415\n",
      "\tExtracting 2 sequences from picr_416\n",
      "\tExtracting 1 sequences from picr_417\n",
      "\tExtracting 1 sequences from picr_425\n",
      "\tExtracting 1 sequences from picr_426\n",
      "\tExtracting 3 sequences from picr_432\n",
      "\tExtracting 9 sequences from picr_433\n",
      "\tExtracting 2 sequences from picr_434\n",
      "\tExtracting 5 sequences from picr_435\n",
      "\tExtracting 1 sequences from picr_437\n",
      "\tExtracting 6 sequences from picr_438\n",
      "\tExtracting 1 sequences from picr_440\n",
      "\tExtracting 1 sequences from picr_442\n",
      "\tExtracting 2 sequences from picr_446\n",
      "\tExtracting 1 sequences from picr_450\n",
      "\tExtracting 1 sequences from picr_452\n",
      "\tExtracting 2 sequences from picr_454\n",
      "\tExtracting 1 sequences from picr_455\n",
      "\tExtracting 4 sequences from picr_456\n",
      "\tExtracting 2 sequences from picr_457\n",
      "\tExtracting 5 sequences from picr_458\n",
      "\tExtracting 1 sequences from picr_462\n",
      "\tExtracting 10 sequences from picr_463\n",
      "\tExtracting 2 sequences from picr_472\n",
      "\tExtracting 8 sequences from picr_474\n",
      "\tExtracting 1 sequences from picr_476\n",
      "\tExtracting 3 sequences from picr_477\n",
      "\tExtracting 2 sequences from picr_478\n",
      "\tExtracting 1 sequences from picr_479\n",
      "\tExtracting 3 sequences from picr_480\n",
      "\tExtracting 1 sequences from picr_481\n",
      "\tExtracting 2 sequences from picr_485\n",
      "\tExtracting 1 sequences from picr_487\n",
      "\tExtracting 2 sequences from picr_488\n",
      "\tExtracting 2 sequences from picr_489\n",
      "\tExtracting 1 sequences from picr_491\n",
      "\tExtracting 6 sequences from picr_492\n",
      "\tExtracting 1 sequences from picr_493\n",
      "\tExtracting 1 sequences from picr_494\n",
      "\tExtracting 1 sequences from picr_495\n",
      "\tExtracting 7 sequences from picr_498\n",
      "\tExtracting 2 sequences from picr_500\n",
      "\tExtracting 1 sequences from picr_502\n",
      "\tExtracting 1 sequences from picr_504\n",
      "\tExtracting 2 sequences from picr_506\n",
      "\tExtracting 1 sequences from picr_507\n",
      "\tExtracting 3 sequences from picr_509\n",
      "\tExtracting 4 sequences from picr_512\n",
      "\tExtracting 4 sequences from picr_517\n",
      "\tExtracting 1 sequences from picr_524\n",
      "\tExtracting 3 sequences from picr_527\n",
      "\tExtracting 2 sequences from picr_529\n",
      "\tExtracting 1 sequences from picr_531\n",
      "\tExtracting 3 sequences from picr_533\n",
      "\tExtracting 1 sequences from picr_534\n",
      "\tExtracting 1 sequences from picr_538\n",
      "\tExtracting 1 sequences from picr_539\n",
      "\tExtracting 1 sequences from picr_540\n",
      "\tExtracting 1 sequences from picr_542\n",
      "\tExtracting 2 sequences from picr_546\n",
      "\tExtracting 1 sequences from picr_550\n",
      "\tExtracting 1 sequences from picr_556\n",
      "\tExtracting 1 sequences from picr_557\n",
      "\tExtracting 3 sequences from picr_559\n",
      "\tExtracting 4 sequences from picr_561\n",
      "\tExtracting 1 sequences from picr_566\n",
      "\tExtracting 1 sequences from picr_568\n",
      "\tExtracting 4 sequences from picr_570\n",
      "\tExtracting 2 sequences from picr_571\n",
      "\tExtracting 3 sequences from picr_572\n",
      "\tExtracting 1 sequences from picr_576\n",
      "\tExtracting 15 sequences from picr_579\n",
      "\tExtracting 1 sequences from picr_580\n",
      "\tExtracting 1 sequences from picr_581\n",
      "\tExtracting 1 sequences from picr_583\n",
      "\tExtracting 11 sequences from picr_591\n",
      "\tExtracting 2 sequences from picr_595\n",
      "\tExtracting 2 sequences from picr_596\n",
      "\tExtracting 1 sequences from picr_599\n",
      "\tExtracting 1 sequences from picr_601\n",
      "\tExtracting 2 sequences from picr_603\n",
      "\tExtracting 1 sequences from picr_604\n",
      "\tExtracting 2 sequences from picr_606\n",
      "\tExtracting 2 sequences from picr_608\n",
      "\tExtracting 1 sequences from picr_618\n",
      "\tExtracting 2 sequences from picr_619\n",
      "\tExtracting 1 sequences from picr_620\n",
      "\tExtracting 2 sequences from picr_622\n",
      "\tExtracting 3 sequences from picr_624\n",
      "\tExtracting 1 sequences from picr_627\n",
      "\tExtracting 2 sequences from picr_638\n",
      "\tExtracting 1 sequences from picr_641\n",
      "\tExtracting 2 sequences from picr_642\n",
      "\tExtracting 16 sequences from picr_653\n",
      "\tExtracting 1 sequences from picr_654\n",
      "\tExtracting 1 sequences from picr_655\n",
      "\tExtracting 5 sequences from picr_656\n",
      "\tExtracting 8 sequences from picr_665\n",
      "\tExtracting 2 sequences from picr_668\n",
      "\tExtracting 4 sequences from picr_673\n",
      "\tExtracting 1 sequences from picr_674\n",
      "\tExtracting 2 sequences from picr_676\n",
      "\tExtracting 1 sequences from picr_680\n",
      "\tExtracting 3 sequences from picr_684\n",
      "\tExtracting 2 sequences from picr_685\n",
      "\tExtracting 1 sequences from picr_686\n",
      "\tExtracting 2 sequences from picr_688\n",
      "\tExtracting 2 sequences from picr_690\n",
      "\tExtracting 2 sequences from picr_692\n",
      "\tExtracting 1 sequences from picr_698\n",
      "\tExtracting 11 sequences from picr_709\n",
      "\tExtracting 1 sequences from picr_715\n",
      "\tExtracting 1 sequences from picr_716\n",
      "\tExtracting 1 sequences from picr_721\n",
      "\tExtracting 4 sequences from picr_729\n",
      "\tExtracting 8 sequences from picr_731\n",
      "\tExtracting 5 sequences from picr_738\n",
      "\tExtracting 1 sequences from picr_744\n",
      "\tExtracting 1 sequences from picr_746\n",
      "\tExtracting 2 sequences from picr_761\n",
      "\tExtracting 1 sequences from picr_764\n",
      "\tExtracting 3 sequences from picr_768\n",
      "\tExtracting 3 sequences from picr_769\n",
      "\tExtracting 1 sequences from picr_772\n",
      "\tExtracting 13 sequences from picr_776\n",
      "\tExtracting 2 sequences from picr_778\n",
      "\tExtracting 1 sequences from picr_780\n",
      "\tExtracting 7 sequences from picr_788\n",
      "\tExtracting 1 sequences from picr_793\n",
      "\tExtracting 3 sequences from picr_799\n",
      "\tExtracting 1 sequences from picr_806\n",
      "\tExtracting 2 sequences from picr_817\n",
      "\tExtracting 2 sequences from picr_820\n",
      "\tExtracting 2 sequences from picr_824\n",
      "\tExtracting 1 sequences from picr_833\n",
      "\tExtracting 5 sequences from picr_843\n",
      "\tExtracting 2 sequences from picr_845\n",
      "\tExtracting 1 sequences from picr_851\n",
      "\tExtracting 4 sequences from picr_853\n",
      "\tExtracting 4 sequences from picr_854\n",
      "\tExtracting 1 sequences from picr_857\n",
      "\tExtracting 1 sequences from picr_858\n",
      "\tExtracting 2 sequences from picr_859\n",
      "\tExtracting 1 sequences from picr_862\n",
      "\tExtracting 1 sequences from picr_864\n",
      "\tExtracting 1 sequences from picr_866\n",
      "\tExtracting 1 sequences from picr_872\n",
      "\tExtracting 3 sequences from picr_875\n",
      "\tExtracting 1 sequences from picr_884\n",
      "\tExtracting 3 sequences from picr_885\n",
      "\tExtracting 6 sequences from picr_886\n",
      "\tExtracting 1 sequences from picr_889\n",
      "\tExtracting 3 sequences from picr_896\n",
      "\tExtracting 1 sequences from picr_897\n",
      "\tExtracting 1 sequences from picr_902\n",
      "\tExtracting 3 sequences from picr_905\n",
      "\tExtracting 1 sequences from picr_911\n",
      "\tExtracting 1 sequences from picr_914\n",
      "\tExtracting 1 sequences from picr_922\n",
      "\tExtracting 1 sequences from picr_928\n",
      "\tExtracting 2 sequences from picr_932\n",
      "\tExtracting 4 sequences from picr_934\n",
      "\tExtracting 2 sequences from picr_936\n",
      "\tExtracting 1 sequences from picr_938\n",
      "\tExtracting 1 sequences from picr_943\n",
      "\tExtracting 1 sequences from picr_954\n",
      "\tExtracting 1 sequences from picr_955\n",
      "\tExtracting 1 sequences from picr_956\n",
      "\tExtracting 2 sequences from picr_964\n",
      "\tExtracting 1 sequences from picr_967\n",
      "\tExtracting 2 sequences from picr_978\n",
      "\tExtracting 3 sequences from picr_980\n",
      "\tExtracting 3 sequences from picr_985\n",
      "\tExtracting 1 sequences from picr_991\n",
      "\tExtracting 1 sequences from picr_996\n",
      "\tExtracting 2 sequences from picr_997\n",
      "\tExtracting 1 sequences from picr_1009\n",
      "\tExtracting 1 sequences from picr_1010\n",
      "\tExtracting 5 sequences from picr_1018\n",
      "\tExtracting 1 sequences from picr_1019\n",
      "\tExtracting 3 sequences from picr_1021\n",
      "\tExtracting 2 sequences from picr_1022\n",
      "\tExtracting 2 sequences from picr_1029\n",
      "\tExtracting 3 sequences from picr_1030\n",
      "\tExtracting 1 sequences from picr_1034\n",
      "\tExtracting 2 sequences from picr_1046\n",
      "\tExtracting 1 sequences from picr_1059\n",
      "\tExtracting 1 sequences from picr_1064\n",
      "\tExtracting 1 sequences from picr_1065\n",
      "\tExtracting 1 sequences from picr_1067\n",
      "\tExtracting 1 sequences from picr_1069\n",
      "\tExtracting 2 sequences from picr_1074\n",
      "\tExtracting 3 sequences from picr_1078\n",
      "\tExtracting 1 sequences from picr_1079\n",
      "\tExtracting 1 sequences from picr_1085\n",
      "\tExtracting 4 sequences from picr_1093\n",
      "\tExtracting 1 sequences from picr_1096\n",
      "\tExtracting 1 sequences from picr_1097\n",
      "\tExtracting 1 sequences from picr_1100\n",
      "\tExtracting 2 sequences from picr_1108\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 1 sequences from picr_1122\r\n",
      "\tExtracting 1 sequences from picr_1125\r\n",
      "\tExtracting 1 sequences from picr_1127\r\n",
      "\tExtracting 2 sequences from picr_1135\r\n",
      "\tExtracting 3 sequences from picr_1137\r\n",
      "\tExtracting 1 sequences from picr_1140\r\n",
      "\tExtracting 1 sequences from picr_1142\r\n",
      "\tExtracting 1 sequences from picr_1149\r\n",
      "\tExtracting 1 sequences from picr_1154\r\n",
      "\tExtracting 4 sequences from picr_1165\r\n",
      "\tExtracting 5 sequences from picr_1173\r\n",
      "\tExtracting 1 sequences from picr_1174\r\n",
      "\tExtracting 1 sequences from picr_1181\r\n",
      "\tExtracting 1 sequences from picr_1194\r\n",
      "\tExtracting 1 sequences from picr_1200\r\n",
      "\tExtracting 1 sequences from picr_1202\r\n",
      "\tExtracting 3 sequences from picr_1209\r\n",
      "\tExtracting 1 sequences from picr_1216\r\n",
      "\tExtracting 1 sequences from picr_1223\r\n",
      "\tExtracting 4 sequences from picr_1232\r\n",
      "\tExtracting 1 sequences from picr_1233\r\n",
      "\tExtracting 1 sequences from picr_1241\r\n",
      "\tExtracting 1 sequences from picr_1251\r\n",
      "\tExtracting 2 sequences from picr_1258\r\n",
      "\tExtracting 1 sequences from picr_1262\r\n",
      "\tExtracting 1 sequences from picr_1270\r\n",
      "\tExtracting 1 sequences from picr_1282\r\n",
      "\tExtracting 1 sequences from picr_1288\r\n",
      "\tExtracting 1 sequences from picr_1289\r\n",
      "\tExtracting 2 sequences from picr_1298\r\n",
      "\tExtracting 1 sequences from picr_1300\r\n",
      "\tExtracting 1 sequences from picr_1302\r\n",
      "\tExtracting 1 sequences from picr_1303\r\n",
      "\tExtracting 2 sequences from picr_1312\r\n",
      "\tExtracting 1 sequences from picr_1314\r\n",
      "\tExtracting 1 sequences from picr_1322\r\n",
      "\tExtracting 1 sequences from picr_1333\r\n",
      "\tExtracting 1 sequences from picr_1338\r\n",
      "\tExtracting 1 sequences from picr_1340\r\n",
      "\tExtracting 1 sequences from picr_1356\r\n",
      "\tExtracting 1 sequences from picr_1366\r\n",
      "\tExtracting 1 sequences from picr_1371\r\n",
      "\tExtracting 1 sequences from picr_1374\r\n",
      "\tExtracting 1 sequences from picr_1386\r\n",
      "\tExtracting 2 sequences from picr_1389\r\n",
      "\tExtracting 1 sequences from picr_1392\r\n",
      "\tExtracting 1 sequences from picr_1401\r\n",
      "\tExtracting 3 sequences from picr_1410\r\n",
      "\tExtracting 1 sequences from picr_1412\r\n",
      "\tExtracting 1 sequences from picr_1416\r\n",
      "\tExtracting 1 sequences from picr_1417\r\n",
      "\tExtracting 1 sequences from picr_1420\r\n",
      "\tExtracting 2 sequences from picr_1425\r\n",
      "\tExtracting 1 sequences from picr_1434\r\n",
      "\tExtracting 2 sequences from picr_1455\r\n",
      "\tExtracting 1 sequences from picr_1456\r\n",
      "\tExtracting 1 sequences from picr_1458\r\n",
      "\tExtracting 1 sequences from picr_1471\r\n",
      "\tExtracting 2 sequences from picr_1490\r\n",
      "\tExtracting 3 sequences from picr_1497\r\n",
      "\tExtracting 2 sequences from picr_1501\r\n",
      "\tExtracting 1 sequences from picr_1503\r\n",
      "\tExtracting 1 sequences from picr_1530\r\n",
      "\tExtracting 1 sequences from picr_1532\r\n",
      "\tExtracting 1 sequences from picr_1557\r\n",
      "\tExtracting 1 sequences from picr_1573\r\n",
      "\tExtracting 2 sequences from picr_1583\r\n",
      "\tExtracting 1 sequences from picr_1590\r\n",
      "\tExtracting 1 sequences from picr_1598\r\n",
      "\tExtracting 2 sequences from picr_1608\r\n",
      "\tExtracting 2 sequences from picr_1628\r\n",
      "\tExtracting 1 sequences from picr_1631\r\n",
      "\tExtracting 1 sequences from picr_1637\r\n",
      "\tExtracting 1 sequences from picr_1645\r\n",
      "\tExtracting 1 sequences from picr_1647\r\n",
      "\tExtracting 5 sequences from picr_1669\r\n",
      "\tExtracting 3 sequences from picr_1677\r\n",
      "\tExtracting 1 sequences from picr_1678\r\n",
      "\tExtracting 1 sequences from picr_1700\r\n",
      "\tExtracting 1 sequences from picr_1725\r\n",
      "\tExtracting 3 sequences from picr_1757\r\n",
      "\tExtracting 1 sequences from picr_1759\r\n",
      "\tExtracting 1 sequences from picr_1763\r\n",
      "\tExtracting 1 sequences from picr_1767\r\n",
      "\tExtracting 1 sequences from picr_1789\r\n",
      "\tExtracting 2 sequences from picr_1794\r\n",
      "\tExtracting 1 sequences from picr_1795\r\n"
     ]
    }
   ],
   "source": [
    "out_fa = alltss_f.replace('bed','fa')\n",
    "cmd = 'homerTools extract {bed_f} {ref_fa} -fa > {out}'.format(bed_f=alltss_f,ref_fa=ref_fa,out=out_fa)\n",
    "print(cmd)\n",
    "!{cmd}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### get the center, do +/- 2kb for all, save to proper bed file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loop through each peak and\n",
    "NOTE: Note which strand you are working with  \n",
    "1) Add sequence -2k:+2k  \n",
    "2) Determine if there is a nearby transcript id within 2k bp upstream and if it is part of the same gene_id or different. If it is the same, write where the end of it is.   \n",
    "3) Determine where the coding region of this associated gene is.   \n",
    "4) Determine if any genes downstream within 2kb (Does not matter which strand its in)  \n",
    "\n",
    "Next, loop through each non-experimental transcript, pick the first isoform and extract their sequence and do the same as before . \n",
    "\n",
    "INPUT: \n",
    "1) Annotation file: List of each mRNA, exon, and coding region  \n",
    "2) txn peak file  \n",
    "3) picr.fa genome for extracting sequences  \n",
    "4) txn_df to get transcripts that do not have a peak (May not be as necessary)  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create output dataframe tss_full"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  import sys\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  if __name__ == '__main__':\n"
     ]
    }
   ],
   "source": [
    "tss_full = merged[['gene_id','gene']] #merged[['transcript_id','gene_id','gene']]\n",
    "#tss_full = tss_full.set_index('transcript_id')\n",
    "\n",
    "## Add columns\n",
    "tss_full['Gene Upstream Position'] = np.nan\n",
    "tss_full['Coding Region Start'] = np.nan\n",
    "tss_full['Gene Downstream Position'] = np.nan\n",
    "tss_full['Experimental'] = 0\n",
    "tss_full['ATAC'] = np.nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>Gene Upstream Position</th>\n",
       "      <th>Coding Region Start</th>\n",
       "      <th>Gene Downstream Position</th>\n",
       "      <th>Experimental</th>\n",
       "      <th>ATAC</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transcript_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>asmbl_2</th>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>SERHL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_4</th>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_7</th>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_8</th>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_9</th>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                gene_id   gene  Gene Upstream Position  Coding Region Start  \\\n",
       "transcript_id                                                                 \n",
       "asmbl_2        gene_1_2  SERHL                     NaN                  NaN   \n",
       "asmbl_4          gene_2  RRP7A                     NaN                  NaN   \n",
       "asmbl_7          gene_2  RRP7A                     NaN                  NaN   \n",
       "asmbl_8          gene_2  RRP7A                     NaN                  NaN   \n",
       "asmbl_9          gene_2  RRP7A                     NaN                  NaN   \n",
       "\n",
       "               Gene Downstream Position  Experimental  ATAC  \n",
       "transcript_id                                                \n",
       "asmbl_2                             NaN             0   NaN  \n",
       "asmbl_4                             NaN             0   NaN  \n",
       "asmbl_7                             NaN             0   NaN  \n",
       "asmbl_8                             NaN             0   NaN  \n",
       "asmbl_9                             NaN             0   NaN  "
      ]
     },
     "execution_count": 270,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tss_full.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "efd0cb077a634cca980e0edf738b4a0f"
      }
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:50: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:6: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "  \n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:41: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:42: FutureWarning: set_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "## Loop through each transcript\n",
    "bin_val = 2000\n",
    "for ind,val in tqdm.tqdm_notebook(merged.iterrows()):\n",
    "    # First determine if experimental:\n",
    "    if ~(np.isnan(merged.loc[ind,'actual_start_x'])):\n",
    "        tss_full.set_value(ind,'Experimental',1)\n",
    "    \n",
    "    ## For each, get all transcripts of the same Chromosome\n",
    "    filt = foo[foo['Chr'] == val['Chr']]\n",
    "\n",
    "    ## Keep transcripts that are within bin_val away from the current transcript\n",
    "    filt = filt[np.abs(val['actual_start'] - filt['actual_start']) < bin_val]\n",
    "    \n",
    "    ## Keep transcripts from a different gene_id, so no isoforms\n",
    "    filt = filt[~(filt['gene_id'] == val['gene_id'])]\n",
    "    \n",
    "    \n",
    "    if not len(filt) == 0:\n",
    "#         print(val['gene'])\n",
    "#         print(val['gene_id'])\n",
    "#         print(filt)\n",
    "        \n",
    "        ## Determine nearest upstream genes distance and nearest downstream genes distance\n",
    "        \n",
    "        ## Multiplier is to reverse the up- and downstream genes when the transcript is on the negative strand\n",
    "        if val['Strand'] == '+':\n",
    "            multiplier = 1\n",
    "        elif val['Strand'] == '-':\n",
    "            multiplier = -1\n",
    "        else:\n",
    "            print('Strand info is wrong',ind)\n",
    "            continue\n",
    "        # For upstream, the upstream genes will have smaller start, so the value will be negative relative to the center\n",
    "        up = filt[multiplier*(filt['actual_start'] - val['actual_start']) < 0]\n",
    "        upstream = np.max(multiplier*(up['actual_start'] - val['actual_start']))\n",
    "        # For downstream, the distance will be positive\n",
    "        down = filt[multiplier*(filt['actual_start'] - val['actual_start']) > 0]\n",
    "        downstream = np.min(multiplier*(down['actual_start'] - val['actual_start']))\n",
    "        \n",
    "        #Set values in the matrix\n",
    "        tss_full.set_value(ind,'Gene Upstream Position',upstream) \n",
    "        tss_full.set_value(ind,'Gene Downstream Position',downstream)\n",
    "\n",
    "    \n",
    "    ## Get nearest CDS \n",
    "    cds_filt = cds[cds['transcript_id'] == ind]\n",
    "    \n",
    "    ##Assuming that cds will always be downstream of start site (by definition)\n",
    "    if not len(cds_filt) == 0:\n",
    "        tss_full.set_value(ind,'Coding Region Start',np.min(np.abs(val['actual_start'] - cds_filt['actual_start'])))\n",
    "                           "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss_full.to_csv('Results/tss_final/s_final/tss_peakCenter_1kb_1kb_ALLGENES_FLANK_2kb.META.tsv',sep='tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss_full.to_csv('Results/tss_final/tss_peakCenter_1kb_1kb_ALLGENES_FLANK_2kb.META.tsv',sep='tsv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.distplot(tss_full['Gene Upstream Position'].dropna())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.distplot(tss_full['Gene Downstream Position'].dropna())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ignore below this.\n",
    "Misc. Scrap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### First make flanking -2kb to +2kb for each peak and save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss_experiment = 'Results/tss_final/tss_peakCenter_1kb_1kb.bed'\n",
    "tss_flank_out = 'Results/tss_final/tss_peakCenter_1kb_1kb_FLANK_2kb.bed'\n",
    "\n",
    "out_fa = 'Results/tss_final/tss_peakCenter_1kb_1kb_FLANK_2kb.fa'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def extend_bed_file(bed_f, out_f, bins=[-2000,2000]):\n",
    "    df = pd.read_csv(bed_f,sep='\\t',header=None)\n",
    "    center = df.apply(lambda x: int(np.floor(((x[1]+x[2])-1)/2)) \n",
    "                    if x[5] == '+' else int(np.ceil(((x[1]+x[2])-1)/2)),axis=1)\n",
    "    df[1] = center + bins[0]\n",
    "    df[2] = center + bins[1] + 1\n",
    "    df.to_csv(out_f,sep='\\t',header= None,index=False)\n",
    "    return "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "deletable": false,
    "editable": false,
    "run_control": {
     "frozen": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 3.14 s, sys: 40 ms, total: 3.18 s\n",
      "Wall time: 3.19 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "extend_bed_file(tss_experiment,tss_flank_out,bins=[-2000,2000])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "deletable": false,
    "editable": false,
    "run_control": {
     "frozen": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tExtracting sequences from file: /data/isshamie/genome/picr.fa\n",
      "\tLooking for peak sequences in a single file (/data/isshamie/genome/picr.fa)\n",
      "\tExtracting 916 sequences from picr_0\n",
      "\tExtracting 540 sequences from picr_1\n",
      "\tExtracting 744 sequences from picr_2\n",
      "\tExtracting 532 sequences from picr_3\n",
      "\tExtracting 478 sequences from picr_4\n",
      "\tExtracting 362 sequences from picr_5\n",
      "\tExtracting 346 sequences from picr_6\n",
      "\tExtracting 764 sequences from picr_8\n",
      "\tExtracting 306 sequences from picr_9\n",
      "\tExtracting 359 sequences from picr_10\n",
      "\tExtracting 249 sequences from picr_11\n",
      "\tExtracting 82 sequences from picr_7b\n",
      "\tExtracting 275 sequences from picr_12\n",
      "\tExtracting 285 sequences from picr_13\n",
      "\tExtracting 212 sequences from picr_14\n",
      "\tExtracting 285 sequences from picr_15\n",
      "\tExtracting 295 sequences from picr_16\n",
      "\tExtracting 145 sequences from picr_17\n",
      "\tExtracting 174 sequences from picr_18\n",
      "\tExtracting 229 sequences from picr_19\n",
      "\tExtracting 258 sequences from picr_20\n",
      "\tExtracting 174 sequences from picr_21\n",
      "\tExtracting 513 sequences from picr_22\n",
      "\tExtracting 343 sequences from picr_23\n",
      "\tExtracting 322 sequences from picr_24\n",
      "\tExtracting 80 sequences from picr_25\n",
      "\tExtracting 97 sequences from picr_26\n",
      "\tExtracting 251 sequences from picr_27\n",
      "\tExtracting 153 sequences from picr_28\n",
      "\tExtracting 249 sequences from picr_29\n",
      "\tExtracting 96 sequences from picr_30\n",
      "\tExtracting 121 sequences from picr_31\n",
      "\tExtracting 68 sequences from picr_32\n",
      "\tExtracting 161 sequences from picr_33\n",
      "\tExtracting 138 sequences from picr_34\n",
      "\tExtracting 464 sequences from picr_35\n",
      "\tExtracting 187 sequences from picr_36\n",
      "\tExtracting 99 sequences from picr_37\n",
      "\tExtracting 216 sequences from picr_38\n",
      "\tExtracting 225 sequences from picr_39\n",
      "\tExtracting 233 sequences from picr_40\n",
      "\tExtracting 112 sequences from picr_41\n",
      "\tExtracting 82 sequences from picr_42\n",
      "\tExtracting 163 sequences from picr_43\n",
      "\tExtracting 149 sequences from picr_44\n",
      "\tExtracting 96 sequences from picr_45\n",
      "\tExtracting 94 sequences from picr_46\n",
      "\tExtracting 189 sequences from picr_47\n",
      "\tExtracting 129 sequences from picr_48\n",
      "\tExtracting 90 sequences from picr_49\n",
      "\tExtracting 199 sequences from picr_50\n",
      "\tExtracting 101 sequences from picr_51\n",
      "\tExtracting 307 sequences from picr_52\n",
      "\tExtracting 250 sequences from picr_53\n",
      "\tExtracting 215 sequences from picr_54\n",
      "\tExtracting 105 sequences from picr_55\n",
      "\tExtracting 177 sequences from picr_56\n",
      "\tExtracting 160 sequences from picr_57\n",
      "\tExtracting 254 sequences from picr_58\n",
      "\tExtracting 239 sequences from picr_59\n",
      "\tExtracting 36 sequences from picr_60\n",
      "\tExtracting 263 sequences from picr_61\n",
      "\tExtracting 180 sequences from picr_62\n",
      "\tExtracting 92 sequences from picr_63\n",
      "\tExtracting 70 sequences from picr_64\n",
      "\tExtracting 109 sequences from picr_65\n",
      "\tExtracting 108 sequences from picr_66\n",
      "\tExtracting 48 sequences from picr_67\n",
      "\tExtracting 44 sequences from picr_68\n",
      "\tExtracting 64 sequences from picr_69\n",
      "\tExtracting 222 sequences from picr_70\n",
      "\tExtracting 131 sequences from picr_71\n",
      "\tExtracting 56 sequences from picr_72\n",
      "\tExtracting 107 sequences from picr_73\n",
      "\tExtracting 290 sequences from picr_74\n",
      "\tExtracting 121 sequences from picr_75\n",
      "\tExtracting 49 sequences from picr_76\n",
      "\tExtracting 43 sequences from picr_77\n",
      "\tExtracting 165 sequences from picr_7a\n",
      "\tExtracting 67 sequences from picr_78\n",
      "\tExtracting 95 sequences from picr_79\n",
      "\tExtracting 32 sequences from picr_80\n",
      "\tExtracting 29 sequences from picr_81\n",
      "\tExtracting 55 sequences from picr_82\n",
      "\tExtracting 119 sequences from picr_83\n",
      "\tExtracting 150 sequences from picr_84\n",
      "\tExtracting 188 sequences from picr_85\n",
      "\tExtracting 37 sequences from picr_86\n",
      "\tExtracting 122 sequences from picr_87\n",
      "\tExtracting 82 sequences from picr_88\n",
      "\tExtracting 146 sequences from picr_89\n",
      "\tExtracting 43 sequences from picr_90\n",
      "\tExtracting 49 sequences from picr_91\n",
      "\tExtracting 34 sequences from picr_92\n",
      "\tExtracting 223 sequences from picr_93\n",
      "\tExtracting 64 sequences from picr_94\n",
      "\tExtracting 48 sequences from picr_95\n",
      "\tExtracting 80 sequences from picr_96\n",
      "\tExtracting 19 sequences from picr_97\n",
      "\tExtracting 133 sequences from picr_98\n",
      "\tExtracting 12 sequences from picr_99\n",
      "\tExtracting 83 sequences from picr_100\n",
      "\tExtracting 73 sequences from picr_101\n",
      "\tExtracting 44 sequences from picr_102\n",
      "\tExtracting 172 sequences from picr_103\n",
      "\tExtracting 21 sequences from picr_104\n",
      "\tExtracting 80 sequences from picr_105\n",
      "\tExtracting 131 sequences from picr_106\n",
      "\tExtracting 20 sequences from picr_107\n",
      "\tExtracting 94 sequences from picr_108\n",
      "\tExtracting 48 sequences from picr_109\n",
      "\tExtracting 34 sequences from picr_110\n",
      "\tExtracting 64 sequences from picr_111\n",
      "\tExtracting 31 sequences from picr_112\n",
      "\tExtracting 52 sequences from picr_113\n",
      "\tExtracting 20 sequences from picr_114\n",
      "\tExtracting 58 sequences from picr_115\n",
      "\tExtracting 33 sequences from picr_116\n",
      "\tExtracting 111 sequences from picr_117\n",
      "\tExtracting 44 sequences from picr_118\n",
      "\tExtracting 161 sequences from picr_119\n",
      "\tExtracting 151 sequences from picr_120\n",
      "\tExtracting 32 sequences from picr_121\n",
      "\tExtracting 60 sequences from picr_122\n",
      "\tExtracting 46 sequences from picr_123\n",
      "\tExtracting 46 sequences from picr_124\n",
      "\tExtracting 31 sequences from picr_125\n",
      "\tExtracting 27 sequences from picr_126\n",
      "\tExtracting 96 sequences from picr_127\n",
      "\tExtracting 25 sequences from picr_128\n",
      "\tExtracting 9 sequences from picr_129\n",
      "\tExtracting 159 sequences from picr_130\n",
      "\tExtracting 37 sequences from picr_131\n",
      "\tExtracting 168 sequences from picr_132\n",
      "\tExtracting 92 sequences from picr_133\n",
      "\tExtracting 18 sequences from picr_134\n",
      "\tExtracting 38 sequences from picr_135\n",
      "\tExtracting 83 sequences from picr_136\n",
      "\tExtracting 40 sequences from picr_137\n",
      "\tExtracting 84 sequences from picr_138\n",
      "\tExtracting 99 sequences from picr_139\n",
      "\tExtracting 25 sequences from picr_140\n",
      "\tExtracting 20 sequences from picr_141\n",
      "\tExtracting 14 sequences from picr_142\n",
      "\tExtracting 14 sequences from picr_143\n",
      "\tExtracting 22 sequences from picr_144\n",
      "\tExtracting 17 sequences from picr_145\n",
      "\tExtracting 22 sequences from picr_146\n",
      "\tExtracting 32 sequences from picr_148\n",
      "\tExtracting 22 sequences from picr_149\n",
      "\tExtracting 19 sequences from picr_151\n",
      "\tExtracting 28 sequences from picr_152\n",
      "\tExtracting 33 sequences from picr_153\n",
      "\tExtracting 94 sequences from picr_154\n",
      "\tExtracting 31 sequences from picr_155\n",
      "\tExtracting 77 sequences from picr_157\n",
      "\tExtracting 90 sequences from picr_158\n",
      "\tExtracting 15 sequences from picr_159\n",
      "\tExtracting 6 sequences from picr_160\n",
      "\tExtracting 55 sequences from picr_161\n",
      "\tExtracting 22 sequences from picr_162\n",
      "\tExtracting 26 sequences from picr_163\n",
      "\tExtracting 9 sequences from picr_164\n",
      "\tExtracting 49 sequences from picr_166\n",
      "\tExtracting 55 sequences from picr_167\n",
      "\tExtracting 11 sequences from picr_168\n",
      "\tExtracting 19 sequences from picr_169\n",
      "\tExtracting 6 sequences from picr_170\n",
      "\tExtracting 15 sequences from picr_171\n",
      "\tExtracting 12 sequences from picr_172\n",
      "\tExtracting 26 sequences from picr_173\n",
      "\tExtracting 30 sequences from picr_174\n",
      "\tExtracting 2 sequences from picr_176\n",
      "\tExtracting 15 sequences from picr_178\n",
      "\tExtracting 2 sequences from picr_182\n",
      "\tExtracting 21 sequences from picr_183\n",
      "\tExtracting 23 sequences from picr_185\n",
      "\tExtracting 25 sequences from picr_186\n",
      "\tExtracting 13 sequences from picr_187\n",
      "\tExtracting 13 sequences from picr_188\n",
      "\tExtracting 20 sequences from picr_191\n",
      "\tExtracting 19 sequences from picr_192\n",
      "\tExtracting 8 sequences from picr_193\n",
      "\tExtracting 27 sequences from picr_195\n",
      "\tExtracting 3 sequences from picr_198\n",
      "\tExtracting 8 sequences from picr_199\n",
      "\tExtracting 18 sequences from picr_200\n",
      "\tExtracting 1 sequences from picr_201\n",
      "\tExtracting 13 sequences from picr_209\n",
      "\tExtracting 2 sequences from picr_211\n",
      "\tExtracting 1 sequences from picr_222\n",
      "\tExtracting 4 sequences from picr_231\n",
      "\tExtracting 3 sequences from picr_232\n",
      "\tExtracting 1 sequences from picr_234\n",
      "\tExtracting 8 sequences from picr_235\n",
      "\tExtracting 2 sequences from picr_244\n",
      "\tExtracting 2 sequences from picr_245\n",
      "\tExtracting 1 sequences from picr_248\n",
      "\tExtracting 2 sequences from picr_250\n",
      "\tExtracting 6 sequences from picr_253\n",
      "\tExtracting 1 sequences from picr_263\n",
      "\tExtracting 3 sequences from picr_265\n",
      "\tExtracting 1 sequences from picr_268\n",
      "\tExtracting 2 sequences from picr_269\n",
      "\tExtracting 1 sequences from picr_274\n",
      "\tExtracting 3 sequences from picr_276\n",
      "\tExtracting 1 sequences from picr_289\n",
      "\tExtracting 2 sequences from picr_296\n",
      "\tExtracting 2 sequences from picr_298\n",
      "\tExtracting 1 sequences from picr_310\n",
      "\tExtracting 8 sequences from picr_312\n",
      "\tExtracting 2 sequences from picr_316\n",
      "\tExtracting 1 sequences from picr_317\n",
      "\tExtracting 3 sequences from picr_323\n",
      "\tExtracting 2 sequences from picr_328\n",
      "\tExtracting 1 sequences from picr_329\n",
      "\tExtracting 1 sequences from picr_331\n",
      "\tExtracting 1 sequences from picr_332\n",
      "\tExtracting 1 sequences from picr_342\n",
      "\tExtracting 1 sequences from picr_348\n",
      "\tExtracting 1 sequences from picr_355\n",
      "\tExtracting 1 sequences from picr_358\n",
      "\tExtracting 1 sequences from picr_374\n",
      "\tExtracting 6 sequences from picr_378\n",
      "\tExtracting 2 sequences from picr_380\n",
      "\tExtracting 2 sequences from picr_400\n",
      "\tExtracting 1 sequences from picr_415\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tExtracting 1 sequences from picr_432\n",
      "\tExtracting 1 sequences from picr_433\n",
      "\tExtracting 1 sequences from picr_434\n",
      "\tExtracting 2 sequences from picr_435\n",
      "\tExtracting 1 sequences from picr_456\n",
      "\tExtracting 2 sequences from picr_477\n",
      "\tExtracting 1 sequences from picr_478\n",
      "\tExtracting 1 sequences from picr_493\n",
      "\tExtracting 3 sequences from picr_512\n",
      "\tExtracting 1 sequences from picr_517\n",
      "\tExtracting 1 sequences from picr_534\n",
      "\tExtracting 1 sequences from picr_566\n",
      "\tExtracting 7 sequences from picr_591\n",
      "\tExtracting 1 sequences from picr_603\n",
      "\tExtracting 1 sequences from picr_606\n",
      "\tExtracting 1 sequences from picr_653\n",
      "\tExtracting 1 sequences from picr_654\n",
      "\tExtracting 3 sequences from picr_709\n",
      "\tExtracting 1 sequences from picr_721\n",
      "\tExtracting 1 sequences from picr_738\n",
      "\tExtracting 2 sequences from picr_776\n",
      "\tExtracting 1 sequences from picr_778\n",
      "\tExtracting 1 sequences from picr_817\n",
      "\tExtracting 1 sequences from picr_824\n",
      "\tExtracting 1 sequences from picr_843\n",
      "\tExtracting 1 sequences from picr_853\n",
      "\tExtracting 1 sequences from picr_911\n",
      "\tExtracting 1 sequences from picr_954\n",
      "\tExtracting 1 sequences from picr_964\n",
      "\tExtracting 1 sequences from picr_1034\n",
      "\tExtracting 1 sequences from picr_1059\n",
      "\tExtracting 1 sequences from picr_1097\n",
      "\tExtracting 2 sequences from picr_1137\n",
      "\tExtracting 2 sequences from picr_1165\n",
      "\tExtracting 1 sequences from picr_1410\n",
      "\tExtracting 1 sequences from picr_1503\n",
      "\tExtracting 1 sequences from picr_1628\n",
      "\tExtracting 1 sequences from picr_1669\n",
      "\tExtracting 2 sequences from picr_1757\n"
     ]
    }
   ],
   "source": [
    "cmd = 'homerTools extract {bed_f} {ref_fa} -fa > {out}'.format(bed_f=tss_flank_out,ref_fa=ref_fa,out=out_fa)\n",
    "!{cmd}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
