{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:24:51.672960Z",
     "start_time": "2018-11-26T20:24:51.374631Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Parameters specific to where your folders are and your data\n",
    "parameter_file = 'params/params.yaml'\n",
    "import yaml\n",
    "import sys\n",
    "\n",
    "with open(parameter_file,'r') as f:\n",
    "    doc = yaml.load(f)\n",
    "\n",
    "#p = dic2obj(**doc)\n",
    "\n",
    "data_folder = doc['data_folder']\n",
    "tissues = doc['tissues'].split(',')\n",
    "sys.path.append(doc['pipeline_path'])\n",
    "ref_fa = doc['ref_fa']\n",
    "annotation=doc['annotation']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:24:51.679157Z",
     "start_time": "2018-11-26T20:24:51.675822Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tss_annotation = doc['tss_annotation']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:05.846176Z",
     "start_time": "2018-11-26T20:24:51.681931Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('Number of tissues: ', 15)\n"
     ]
    }
   ],
   "source": [
    "from collections import defaultdict\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "import seaborn as sns\n",
    "import pickle\n",
    "import numpy as np\n",
    "import matplotlib as mpl\n",
    "from matplotlib import pyplot as plt\n",
    "from itertools import product\n",
    "import glob\n",
    "import re\n",
    "from matplotlib_venn import venn2\n",
    "from matplotlib import rcParams\n",
    "import inspect\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "sys.setrecursionlimit(3000)\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "rcParams['figure.figsize'] = 8, 6\n",
    "import tqdm\n",
    "### notebook specific configuration ###\n",
    "from os.path import basename\n",
    "#mpl.style.use('ggplot')\n",
    "mpl.style.use('fivethirtyeight')\n",
    "from cycler import cycler\n",
    "mpl.rcParams['axes.prop_cycle'] = cycler(color='bgrcmyk')\n",
    "\n",
    "from Homer import *\n",
    "\n",
    "print('Number of tissues: ',len(tissues))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:06.196373Z",
     "start_time": "2018-11-26T20:25:05.848480Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Name</th>\n",
       "      <th>Full</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDM1hKLA_ATAC1</td>\n",
       "      <td>CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>BMDM1hKLA_ATAC2</td>\n",
       "      <td>CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>GRO</td>\n",
       "      <td>BMDM1hKLA_GRO1</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDM1hKLA</td>\n",
       "      <td>GROCap</td>\n",
       "      <td>BMDM1hKLA_GROCap1</td>\n",
       "      <td>CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001.fastq.gz</th>\n",
       "      <td>BMDMwt</td>\n",
       "      <td>GRO</td>\n",
       "      <td>BMDMwt_GRO1</td>\n",
       "      <td>CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Tissue Experiment  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  BMDM1hKLA       ATAC   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  BMDM1hKLA       ATAC   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  BMDM1hKLA        GRO   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA     GROCap   \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....     BMDMwt        GRO   \n",
       "\n",
       "                                                                 Name  \\\n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...    BMDM1hKLA_ATAC1   \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...    BMDM1hKLA_ATAC2   \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...     BMDM1hKLA_GRO1   \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  BMDM1hKLA_GROCap1   \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....        BMDMwt_GRO1   \n",
       "\n",
       "                                                                                                 Full  \n",
       "CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001_...  CHBMDM1hKLA2_ATAC_JHS1000_SD_AGGTTGGG_S11_L001...  \n",
       "CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R1...  CHBMDM1hKLA1_ATAC_JHS998_SD_TGGGTTTC_S9_L001_R...  \n",
       "CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1027_SD_CAAAAG_S42_L003_R1_...  \n",
       "CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_0...  CHBMDM1hKLA_GRO_JHS1034_SD_AGTCAA_S49_L003_R1_...  \n",
       "CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001....  CHBMDMwt_GRO_JHS1026_SD_ATTCCT_S41_L003_R1_001...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_samples = pd.read_csv('../raw_metaSamples.tsv',sep=\"\\t\",index_col=0)\n",
    "meta_samples.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load anno_peaks. \n",
    "Later will filter based on Distance to TSS\n",
    "\n",
    "## Load peaks_tissues matrix\n",
    "Later will be used to merge two data types in a tissue and for more confident peaks if they have more than 1 tissues present\n",
    "\n",
    "## Load tss\n",
    "\n",
    "## Load txn_df\n",
    "\n",
    "## Load geneID_df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:27.308761Z",
     "start_time": "2018-11-26T20:25:06.198390Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Nearest TSS</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>isSameStrand</th>\n",
       "      <th>Nearest gene</th>\n",
       "      <th>Nearest gene_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>picr_9</td>\n",
       "      <td>30423275</td>\n",
       "      <td>30423564</td>\n",
       "      <td>+</td>\n",
       "      <td>10.506897</td>\n",
       "      <td>exon (asmbl_83977, exon 1 of 1)</td>\n",
       "      <td>asmbl_83977</td>\n",
       "      <td>4098.0</td>\n",
       "      <td>False</td>\n",
       "      <td>SALL1</td>\n",
       "      <td>gene_29627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>picr_27</td>\n",
       "      <td>14937806</td>\n",
       "      <td>14938107</td>\n",
       "      <td>-</td>\n",
       "      <td>22.268621</td>\n",
       "      <td>promoter-TSS (asmbl_40109)</td>\n",
       "      <td>asmbl_40105</td>\n",
       "      <td>-159.0</td>\n",
       "      <td>False</td>\n",
       "      <td>RBM25</td>\n",
       "      <td>gene_13991_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>picr_2</td>\n",
       "      <td>5127066</td>\n",
       "      <td>5127335</td>\n",
       "      <td>+</td>\n",
       "      <td>1.732333</td>\n",
       "      <td>intron (asmbl_29716, intron 9 of 9)</td>\n",
       "      <td>asmbl_29719</td>\n",
       "      <td>-8640.0</td>\n",
       "      <td>True</td>\n",
       "      <td>FAM83C</td>\n",
       "      <td>gene_10160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>picr_15</td>\n",
       "      <td>26110487</td>\n",
       "      <td>26110670</td>\n",
       "      <td>+</td>\n",
       "      <td>1.659667</td>\n",
       "      <td>intron (asmbl_22566, intron 1 of 4)</td>\n",
       "      <td>asmbl_22566</td>\n",
       "      <td>24071.0</td>\n",
       "      <td>True</td>\n",
       "      <td>ZNRF1</td>\n",
       "      <td>gene_7657_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>73670402</td>\n",
       "      <td>73670628</td>\n",
       "      <td>+</td>\n",
       "      <td>1.694250</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>asmbl_3271</td>\n",
       "      <td>-74096.0</td>\n",
       "      <td>True</td>\n",
       "      <td>MAN2A1</td>\n",
       "      <td>gene_975</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr     Start       End Strand       Stat  \\\n",
       "ID                                                  \n",
       "0    picr_9  30423275  30423564      +  10.506897   \n",
       "1   picr_27  14937806  14938107      -  22.268621   \n",
       "2    picr_2   5127066   5127335      +   1.732333   \n",
       "3   picr_15  26110487  26110670      +   1.659667   \n",
       "4    picr_0  73670402  73670628      +   1.694250   \n",
       "\n",
       "                             Annotation  Nearest TSS  Distance to TSS  \\\n",
       "ID                                                                      \n",
       "0       exon (asmbl_83977, exon 1 of 1)  asmbl_83977           4098.0   \n",
       "1            promoter-TSS (asmbl_40109)  asmbl_40105           -159.0   \n",
       "2   intron (asmbl_29716, intron 9 of 9)  asmbl_29719          -8640.0   \n",
       "3   intron (asmbl_22566, intron 1 of 4)  asmbl_22566          24071.0   \n",
       "4                            Intergenic   asmbl_3271         -74096.0   \n",
       "\n",
       "    isSameStrand Nearest gene Nearest gene_id  \n",
       "ID                                             \n",
       "0          False        SALL1      gene_29627  \n",
       "1          False        RBM25    gene_13991_3  \n",
       "2           True       FAM83C      gene_10160  \n",
       "3           True        ZNRF1     gene_7657_3  \n",
       "4           True       MAN2A1        gene_975  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno_peaks = pd.read_csv('Results/merged/peaks_with_tss_distances.tsv',sep='\\t',index_col=0)\n",
    "anno_peaks.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:29.329061Z",
     "start_time": "2018-11-26T20:25:27.311011Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f04_peaks/Brain_START2_and_Brain_STARTinput1</th>\n",
       "      <th>f04_peaks/Brain_GROCap1_and_Brain_GRO1</th>\n",
       "      <th>f04_peaks/FemaleReproductive_START1_and_FemaleReproductive_STARTinput1</th>\n",
       "      <th>f04_peaks/Kidney_START1_and_Kidney_STARTinput1</th>\n",
       "      <th>f04_peaks/Muscle_START2_and_Muscle_STARTinput1</th>\n",
       "      <th>f04_peaks/Heart_START1_and_Heart_STARTinput1</th>\n",
       "      <th>f04_peaks/Lung_GROCap1_and_Lung_GRO1</th>\n",
       "      <th>f04_peaks/Liver_START2_and_Liver_STARTinput1</th>\n",
       "      <th>f04_peaks/MiscOrgans_START2_and_MiscOrgans_STARTinput1</th>\n",
       "      <th>f04_peaks/Muscle_START1_and_Muscle_STARTinput1</th>\n",
       "      <th>...</th>\n",
       "      <th>f04_peaks/Brain_START1_and_Brain_STARTinput2</th>\n",
       "      <th>f04_peaks/FemaleReproductive_START2_and_FemaleReproductive_STARTinput1</th>\n",
       "      <th>f04_peaks/Pancreas_START2_and_Pancreas_STARTinput1</th>\n",
       "      <th>f04_peaks/Brain_START3_and_Brain_STARTinput1</th>\n",
       "      <th>f04_peaks/Kidney_GROCap1_and_Kidney_GRO1</th>\n",
       "      <th>f04_peaks/Brain_START1_and_Brain_STARTinput1</th>\n",
       "      <th>f04_peaks/CHO_GROCap1_and_CHO_GRO1</th>\n",
       "      <th>f04_peaks/BMDMwt_GROCap1_and_BMDMwt_GRO1</th>\n",
       "      <th>f04_peaks/Heart_START2_and_Heart_STARTinput1</th>\n",
       "      <th>f04_peaks/Lung_START1_and_Lung_STARTinput1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15.5</td>\n",
       "      <td>5.12</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.50</td>\n",
       "      <td>22.50</td>\n",
       "      <td>2.34</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.37</td>\n",
       "      <td>1.93</td>\n",
       "      <td>...</td>\n",
       "      <td>23.00</td>\n",
       "      <td>16.10</td>\n",
       "      <td>10.10</td>\n",
       "      <td>11.60</td>\n",
       "      <td>1.05</td>\n",
       "      <td>23.00</td>\n",
       "      <td>3.12</td>\n",
       "      <td>4.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9.6</td>\n",
       "      <td>50.80</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.09</td>\n",
       "      <td>7.78</td>\n",
       "      <td>3.68</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.30</td>\n",
       "      <td>5.78</td>\n",
       "      <td>...</td>\n",
       "      <td>5.25</td>\n",
       "      <td>5.85</td>\n",
       "      <td>4.77</td>\n",
       "      <td>2.48</td>\n",
       "      <td>103.00</td>\n",
       "      <td>5.25</td>\n",
       "      <td>46.00</td>\n",
       "      <td>65.6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.15</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.94</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.5</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   f04_peaks/Brain_START2_and_Brain_STARTinput1  \\\n",
       "0                                          15.5   \n",
       "1                                           9.6   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           0.0   \n",
       "\n",
       "   f04_peaks/Brain_GROCap1_and_Brain_GRO1  \\\n",
       "0                                    5.12   \n",
       "1                                   50.80   \n",
       "2                                    0.00   \n",
       "3                                    0.00   \n",
       "4                                    0.00   \n",
       "\n",
       "   f04_peaks/FemaleReproductive_START1_and_FemaleReproductive_STARTinput1  \\\n",
       "0                                                0.0                        \n",
       "1                                                0.0                        \n",
       "2                                                0.0                        \n",
       "3                                                0.0                        \n",
       "4                                                0.0                        \n",
       "\n",
       "   f04_peaks/Kidney_START1_and_Kidney_STARTinput1  \\\n",
       "0                                           26.50   \n",
       "1                                            4.09   \n",
       "2                                            0.00   \n",
       "3                                            0.00   \n",
       "4                                            0.00   \n",
       "\n",
       "   f04_peaks/Muscle_START2_and_Muscle_STARTinput1  \\\n",
       "0                                           22.50   \n",
       "1                                            7.78   \n",
       "2                                            0.00   \n",
       "3                                            0.00   \n",
       "4                                            0.00   \n",
       "\n",
       "   f04_peaks/Heart_START1_and_Heart_STARTinput1  \\\n",
       "0                                          2.34   \n",
       "1                                          3.68   \n",
       "2                                          0.00   \n",
       "3                                          0.00   \n",
       "4                                          0.00   \n",
       "\n",
       "   f04_peaks/Lung_GROCap1_and_Lung_GRO1  \\\n",
       "0                                   0.0   \n",
       "1                                   0.0   \n",
       "2                                   0.0   \n",
       "3                                   0.0   \n",
       "4                                   0.0   \n",
       "\n",
       "   f04_peaks/Liver_START2_and_Liver_STARTinput1  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           2.5   \n",
       "\n",
       "   f04_peaks/MiscOrgans_START2_and_MiscOrgans_STARTinput1  \\\n",
       "0                                               9.37        \n",
       "1                                               6.30        \n",
       "2                                               0.00        \n",
       "3                                               0.00        \n",
       "4                                               0.00        \n",
       "\n",
       "   f04_peaks/Muscle_START1_and_Muscle_STARTinput1  \\\n",
       "0                                            1.93   \n",
       "1                                            5.78   \n",
       "2                                            0.00   \n",
       "3                                            0.00   \n",
       "4                                            0.00   \n",
       "\n",
       "                      ...                      \\\n",
       "0                     ...                       \n",
       "1                     ...                       \n",
       "2                     ...                       \n",
       "3                     ...                       \n",
       "4                     ...                       \n",
       "\n",
       "   f04_peaks/Brain_START1_and_Brain_STARTinput2  \\\n",
       "0                                         23.00   \n",
       "1                                          5.25   \n",
       "2                                          0.00   \n",
       "3                                          0.00   \n",
       "4                                          0.00   \n",
       "\n",
       "   f04_peaks/FemaleReproductive_START2_and_FemaleReproductive_STARTinput1  \\\n",
       "0                                              16.10                        \n",
       "1                                               5.85                        \n",
       "2                                               0.00                        \n",
       "3                                               0.00                        \n",
       "4                                               0.00                        \n",
       "\n",
       "   f04_peaks/Pancreas_START2_and_Pancreas_STARTinput1  \\\n",
       "0                                              10.10    \n",
       "1                                               4.77    \n",
       "2                                               0.00    \n",
       "3                                               0.00    \n",
       "4                                               0.00    \n",
       "\n",
       "   f04_peaks/Brain_START3_and_Brain_STARTinput1  \\\n",
       "0                                         11.60   \n",
       "1                                          2.48   \n",
       "2                                          0.00   \n",
       "3                                          0.00   \n",
       "4                                          0.00   \n",
       "\n",
       "   f04_peaks/Kidney_GROCap1_and_Kidney_GRO1  \\\n",
       "0                                      1.05   \n",
       "1                                    103.00   \n",
       "2                                      1.15   \n",
       "3                                      1.94   \n",
       "4                                      0.00   \n",
       "\n",
       "   f04_peaks/Brain_START1_and_Brain_STARTinput1  \\\n",
       "0                                         23.00   \n",
       "1                                          5.25   \n",
       "2                                          0.00   \n",
       "3                                          0.00   \n",
       "4                                          0.00   \n",
       "\n",
       "   f04_peaks/CHO_GROCap1_and_CHO_GRO1  \\\n",
       "0                                3.12   \n",
       "1                               46.00   \n",
       "2                                0.00   \n",
       "3                                0.00   \n",
       "4                                0.79   \n",
       "\n",
       "   f04_peaks/BMDMwt_GROCap1_and_BMDMwt_GRO1  \\\n",
       "0                                       4.7   \n",
       "1                                      65.6   \n",
       "2                                       0.0   \n",
       "3                                       0.0   \n",
       "4                                       0.0   \n",
       "\n",
       "   f04_peaks/Heart_START2_and_Heart_STARTinput1  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           0.0   \n",
       "\n",
       "   f04_peaks/Lung_START1_and_Lung_STARTinput1  \n",
       "0                                        14.4  \n",
       "1                                        10.7  \n",
       "2                                         0.0  \n",
       "3                                         0.0  \n",
       "4                                         0.0  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peaks_tissue = pd.read_csv(\"Results/merged/samples.merge.peaksexpression\",index_col=0,sep=\"\\t\")\n",
    "peaks_tissue.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Read in tss and txn_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:29.761882Z",
     "start_time": "2018-11-26T20:25:29.332423Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>ID</th>\n",
       "      <th>Parent</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>cs</th>\n",
       "      <th>Length</th>\n",
       "      <th>actual_start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gene=SERHL;transcript_id=asmbl_2;cs=2</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>176</td>\n",
       "      <td>38694</td>\n",
       "      <td>+</td>\n",
       "      <td>asmbl_2</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>gene_1_2</td>\n",
       "      <td>SERHL</td>\n",
       "      <td>asmbl_2</td>\n",
       "      <td>2</td>\n",
       "      <td>38518</td>\n",
       "      <td>176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_4;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7715</td>\n",
       "      <td>46630</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>1</td>\n",
       "      <td>38915</td>\n",
       "      <td>46630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_7;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7759</td>\n",
       "      <td>44597</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_7</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_7</td>\n",
       "      <td>1</td>\n",
       "      <td>36838</td>\n",
       "      <td>44597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_8;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7767</td>\n",
       "      <td>44597</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_8</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_8</td>\n",
       "      <td>1</td>\n",
       "      <td>36830</td>\n",
       "      <td>44597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RRP7A;transcript_id=asmbl_9;cs=1</th>\n",
       "      <td>picr_0</td>\n",
       "      <td>7789</td>\n",
       "      <td>44568</td>\n",
       "      <td>-</td>\n",
       "      <td>asmbl_9</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_9</td>\n",
       "      <td>1</td>\n",
       "      <td>36779</td>\n",
       "      <td>44568</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Chr  Start    End  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  picr_0    176  38694   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7715  46630   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7759  44597   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7767  44597   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...  picr_0   7789  44568   \n",
       "\n",
       "                                                   Strand       ID    Parent  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...      +  asmbl_2  gene_1_2   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_4    gene_2   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_7    gene_2   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_8    gene_2   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...      -  asmbl_9    gene_2   \n",
       "\n",
       "                                                     gene_id   gene  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...  gene_1_2  SERHL   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...    gene_2  RRP7A   \n",
       "\n",
       "                                                   transcript_id  cs  Length  \\\n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...       asmbl_2   2   38518   \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_4   1   38915   \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_7   1   36838   \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_8   1   36830   \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...       asmbl_9   1   36779   \n",
       "\n",
       "                                                    actual_start  \n",
       "ID=asmbl_2;Parent=gene_1_2;gene_id=gene_1_2;gen...           176  \n",
       "ID=asmbl_4;Parent=gene_2;gene_id=gene_2;gene=RR...         46630  \n",
       "ID=asmbl_7;Parent=gene_2;gene_id=gene_2;gene=RR...         44597  \n",
       "ID=asmbl_8;Parent=gene_2;gene_id=gene_2;gene=RR...         44597  \n",
       "ID=asmbl_9;Parent=gene_2;gene_id=gene_2;gene=RR...         44568  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tss = pd.read_csv(tss_annotation,sep='\\t',index_col=0)\n",
    "\n",
    "tss.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:35.400215Z",
     "start_time": "2018-11-26T20:25:29.764838Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>peaks</th>\n",
       "      <th>closest_to_X_numberOfPeaks</th>\n",
       "      <th>bin_-1000_1000</th>\n",
       "      <th>sameStrand_bin_-1000_1000</th>\n",
       "      <th>samples</th>\n",
       "      <th>minDistance</th>\n",
       "      <th>minDistancePeak</th>\n",
       "      <th>maxValue</th>\n",
       "      <th>maxPeakId</th>\n",
       "      <th>maxSample</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript</th>\n",
       "      <th>maxSamples</th>\n",
       "      <th>hasGene</th>\n",
       "      <th>Number of SS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>asmbl_2</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>inf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_4</th>\n",
       "      <td>[215685, 249167, 398989, 483229]</td>\n",
       "      <td>4</td>\n",
       "      <td>[215685]</td>\n",
       "      <td>[215685]</td>\n",
       "      <td>[CHO_START1_and_CHO_STARTinput1]</td>\n",
       "      <td>706.000000</td>\n",
       "      <td>215685</td>\n",
       "      <td>3.04</td>\n",
       "      <td>215685</td>\n",
       "      <td>NaN</td>\n",
       "      <td>picr_0</td>\n",
       "      <td>45849</td>\n",
       "      <td>45999</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_2</td>\n",
       "      <td>RRP7A</td>\n",
       "      <td>asmbl_4</td>\n",
       "      <td>[CHO_START1_and_CHO_STARTinput1]</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_7</th>\n",
       "      <td>[104402]</td>\n",
       "      <td>1</td>\n",
       "      <td>[104402]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>inf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_8</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>inf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>asmbl_9</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>inf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    peaks  closest_to_X_numberOfPeaks  \\\n",
       "asmbl_2                                []                           0   \n",
       "asmbl_4  [215685, 249167, 398989, 483229]                           4   \n",
       "asmbl_7                          [104402]                           1   \n",
       "asmbl_8                                []                           0   \n",
       "asmbl_9                                []                           0   \n",
       "\n",
       "        bin_-1000_1000 sameStrand_bin_-1000_1000  \\\n",
       "asmbl_2             []                        []   \n",
       "asmbl_4       [215685]                  [215685]   \n",
       "asmbl_7       [104402]                        []   \n",
       "asmbl_8             []                        []   \n",
       "asmbl_9             []                        []   \n",
       "\n",
       "                                  samples  minDistance minDistancePeak  \\\n",
       "asmbl_2                                []          inf             NaN   \n",
       "asmbl_4  [CHO_START1_and_CHO_STARTinput1]   706.000000          215685   \n",
       "asmbl_7                                []          inf             NaN   \n",
       "asmbl_8                                []          inf             NaN   \n",
       "asmbl_9                                []          inf             NaN   \n",
       "\n",
       "        maxValue maxPeakId maxSample     Chr  Start    End Strand gene_id  \\\n",
       "asmbl_2      NaN       NaN       NaN     NaN    NaN    NaN    NaN     NaN   \n",
       "asmbl_4     3.04    215685       NaN  picr_0  45849  45999      -  gene_2   \n",
       "asmbl_7      NaN       NaN       NaN     NaN    NaN    NaN    NaN     NaN   \n",
       "asmbl_8      NaN       NaN       NaN     NaN    NaN    NaN    NaN     NaN   \n",
       "asmbl_9      NaN       NaN       NaN     NaN    NaN    NaN    NaN     NaN   \n",
       "\n",
       "          gene transcript                        maxSamples  hasGene  \\\n",
       "asmbl_2    NaN        NaN                                []    False   \n",
       "asmbl_4  RRP7A    asmbl_4  [CHO_START1_and_CHO_STARTinput1]     True   \n",
       "asmbl_7    NaN        NaN                                []    False   \n",
       "asmbl_8    NaN        NaN                                []    False   \n",
       "asmbl_9    NaN        NaN                                []    False   \n",
       "\n",
       "         Number of SS  \n",
       "asmbl_2             0  \n",
       "asmbl_4             1  \n",
       "asmbl_7             0  \n",
       "asmbl_8             0  \n",
       "asmbl_9             0  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "txn_df = pickle.load(open('Results/tss_annotation/txn_df_02.p','rb'))# pd.read_csv('Results/txn_df.csv',index_col=0)\n",
    "txn_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:39.608807Z",
     "start_time": "2018-11-26T20:25:35.409826Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "geneID_df = pickle.load(open('Results/tss_annotation/geneID_df_02.p','rb'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge tissue columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:39.642741Z",
     "start_time": "2018-11-26T20:25:39.611525Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def merge_two_columns(df, mapping_dict):\n",
    "    '''Function that merges columns by taking the mean of them.\n",
    "    Merges based on if they have the same element in the meta_samples meta_column.\n",
    "    Returns:\n",
    "        new_df: Dataframe but with the columns of interest merged. Also column names are now based\n",
    "                on the unique meta_samples[meta_column].\n",
    "    '''\n",
    "\n",
    "    vals = mapping_dict.keys()#np.unique(meta_samples[meta_column].values)\n",
    "    new_df = pd.DataFrame(index=df.index, columns=vals)\n",
    "    for i in vals:\n",
    "        if not mapping_dict[i] == []:#(meta_samples[meta_column] == i).any():\n",
    "            new_col = (df.loc[:, mapping_dict[i]])\n",
    "            new_col = new_col.mean(axis=1)\n",
    "            new_df.loc[:, i] = new_col\n",
    "    new_df = new_df.loc[:, ~(new_df.isnull().all())]\n",
    "    return new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:39.695314Z",
     "start_time": "2018-11-26T20:25:39.645863Z"
    }
   },
   "outputs": [],
   "source": [
    "tissue_to_peaks = defaultdict(list)\n",
    "for val in peaks_tissue.columns.values:\n",
    "    count = 0\n",
    "    for t in np.unique(meta_samples[\"Tissue\"].values):\n",
    "        if t in val:\n",
    "            count += 1\n",
    "            tissue_to_peaks[t].append(val)\n",
    "            if count > 1:\n",
    "                print(\"This sample has multiple tissues associated with it\")\n",
    "                print(\"Sample: \",val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:41.850448Z",
     "start_time": "2018-11-26T20:25:39.698890Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "peaks_tissue_merged = merge_two_columns(peaks_tissue,mapping_dict=tissue_to_peaks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:41.943979Z",
     "start_time": "2018-11-26T20:25:41.857314Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Heart</th>\n",
       "      <th>BMDMwt</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>FemaleReproductive</th>\n",
       "      <th>CHO</th>\n",
       "      <th>Brain</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Liver</th>\n",
       "      <th>MiscOrgans</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Kidney</th>\n",
       "      <th>Pancreas</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.17</td>\n",
       "      <td>4.7</td>\n",
       "      <td>3.455</td>\n",
       "      <td>8.050</td>\n",
       "      <td>4.1820</td>\n",
       "      <td>15.045714</td>\n",
       "      <td>4.800000</td>\n",
       "      <td>14.380000</td>\n",
       "      <td>10.485</td>\n",
       "      <td>12.215</td>\n",
       "      <td>10.576667</td>\n",
       "      <td>6.875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.84</td>\n",
       "      <td>65.6</td>\n",
       "      <td>5.195</td>\n",
       "      <td>2.925</td>\n",
       "      <td>48.2000</td>\n",
       "      <td>12.208571</td>\n",
       "      <td>3.566667</td>\n",
       "      <td>25.080000</td>\n",
       "      <td>7.100</td>\n",
       "      <td>6.780</td>\n",
       "      <td>37.786667</td>\n",
       "      <td>3.375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.1494</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.100000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.383333</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0798</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.880000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.646667</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.3274</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.713333</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Heart  BMDMwt  Spleen  FemaleReproductive      CHO      Brain      Lung  \\\n",
       "0   1.17     4.7   3.455               8.050   4.1820  15.045714  4.800000   \n",
       "1   1.84    65.6   5.195               2.925  48.2000  12.208571  3.566667   \n",
       "2   0.00     0.0   0.000               0.000   0.1494   0.000000  0.000000   \n",
       "3   0.00     0.0   0.000               0.000   0.0798   0.000000  0.000000   \n",
       "4   0.00     0.0   0.000               0.000   0.3274   0.000000  0.000000   \n",
       "\n",
       "       Liver  MiscOrgans  Muscle     Kidney  Pancreas  \n",
       "0  14.380000      10.485  12.215  10.576667     6.875  \n",
       "1  25.080000       7.100   6.780  37.786667     3.375  \n",
       "2   1.100000       0.000   0.000   0.383333     0.000  \n",
       "3   0.880000       0.000   0.000   0.646667     0.000  \n",
       "4   1.713333       0.000   0.000   0.000000     0.000  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peaks_tissue_merged.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create directory of different peak groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:35:53.590351Z",
     "start_time": "2018-11-26T20:35:53.515052Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "if not os.path.exists(\"Results/peak_groups\"):\n",
    "    os.mkdir(\"Results/peak_groups\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:41.999462Z",
     "start_time": "2018-11-26T20:25:41.967788Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def save_peak_as_bed(peaks_df,f_save, name='index',score_col = 'Peak Score',is_sort = True):\n",
    "    ## Assumes there is Chr,Start,End,Name is index\n",
    "    ## Peak file is 1-index, bed files are 0, so need to shift everything by one\n",
    "    \n",
    "    if name == 'index':\n",
    "        peaks_df.loc[:,'Name'] = peaks_df.index\n",
    "    else:\n",
    "        peaks_df.loc[:,'Name'] = peaks_df[name]\n",
    "    \n",
    "    if score_col == '': #If no scoring col, set to 0\n",
    "        peaks_df.loc[:,'Score'] = 0\n",
    "        score_col = 'Score'\n",
    "        \n",
    "    peaks_df.loc[:,'Start'] = peaks_df['Start'] - 1\n",
    "    peaks_df.loc[:,'End'] = peaks_df['End'] - 1\n",
    "    \n",
    "    peaks_df = peaks_df[['Chr','Start','End','Name',score_col,'Strand']]\n",
    "    \n",
    "    if is_sort:\n",
    "        peaks_df.sort_values(['Chr','Start'],inplace=True)\n",
    "        \n",
    "    if not f_save.endswith('.bed'):\n",
    "        f_save = f_save + '.bed'\n",
    "    peaks_df.to_csv(f_save ,sep='\\t',index=False,header=None)\n",
    "    return\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## a) Separate confident promoters and confident ncRNA \n",
    "Define by distance to RefSeq being 0 and ncRNA far >40kb from a gene and in at least two tissues\n",
    "\n",
    "### c) Predicted promoters vs RefSeq promoters\n",
    "\n",
    "### d) Promoters organized based on distance to refseq (so heatmap is not sorted by heat but by distance)\n",
    "\n",
    "### e) Only look at peaks specific for each tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:42.029291Z",
     "start_time": "2018-11-26T20:25:42.001522Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def filter_only_with_divergent(df,divergent_threshold = 500 ):\n",
    "    inds_to_keep = []\n",
    "    for ind,val in df[df['isSameStrand']].iterrows():\n",
    "        curr = df[(val['Chr'] == df['Chr']) & (df['isSameStrand'] == False)]\n",
    "        if val['Strand'] == '-':\n",
    "            curr = curr[(val['End'] - curr['Start'] < divergent_threshold) & \n",
    "                       (val['End'] - curr['Start'] > 0)]\n",
    "        else: \n",
    "            curr = curr[(val['Start'] - curr['End'] < divergent_threshold) &\n",
    "                       (val['Start'] - curr['End'] > 0 ) ]\n",
    "\n",
    "        ## If there are divergent peaks, add this peak\n",
    "        if len(curr) > 0: \n",
    "            inds_to_keep.append(ind)\n",
    "    \n",
    "    df = df.loc[inds_to_keep]\n",
    "    return df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:39:46.132121Z",
     "start_time": "2018-11-26T20:39:46.039082Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def tissue_specific_promoter(t,\n",
    "                             anno_peaks,\n",
    "                             peaks_tissue_merged,\n",
    "                             txn_df,tss,\n",
    "                             peak_count_thresh = 1, \n",
    "                             prom_bin=[-1000, 1000],\n",
    "                             enhancer_d=40000,\n",
    "                             divergent_thr=500,\n",
    "                             outfolder='Results/peak_groups/'):\n",
    "\n",
    "    ''' Function to create bed files relevant for a specific tissue\n",
    "        Will save:\n",
    "        a) All peaks that were found in tissue\n",
    "        b) Peaks within prom_bin distance from reference & has divergent peak nearby (divergent_thr distance)\n",
    "        c) Same as b, but no divergent restriction\n",
    "        d) Same as c, but the peak center is now the annnotated start site instead of experimental\n",
    "        e) ncRNA, possibly eRNA that are greater than enhancer_d from any annotated start sites.\n",
    "        f) no_peaks: genes with no peaks nearby, even divergent peaks'''\n",
    "    \n",
    "    \n",
    "    \n",
    "    t_peaks = peaks_tissue_merged[peaks_tissue_merged.loc[:, t] > peak_count_thresh].index #Peaks in that tissue\n",
    "    t_peaks = anno_peaks.loc[t_peaks]\n",
    "    t_peaks = t_peaks[(~(t_peaks['Distance to TSS'] == np.infty))] ## Remove peaks that are on a scaffold with no gene\n",
    "\n",
    "    ### Absence of peaks for that tissue.\n",
    "    no_t_peaks = txn_df[txn_df['samples'].apply(lambda x: np.array(map(lambda y: t not in y,x)).any())]\n",
    "    no_t_peaks = tss[tss['transcript_id'].apply(lambda x: x in no_t_peaks.index)]\n",
    "    #no_t_peaks = peaks_tissue_merged[peaks_tissue_merged.loc[:, t] <= peak_count_thresh].index #Peaks not in the tissue\n",
    "    #no_t_peaks = anno_peaks.loc[no_t_peaks]\n",
    "    \n",
    "\n",
    "\n",
    "    t_prom = t_peaks[(t_peaks['Distance to TSS'] > prom_bin[0])\n",
    "                     & (t_peaks['Distance to TSS'] < prom_bin[1])]\n",
    "    t_prom_isSame = t_prom[t_prom['isSameStrand']]\n",
    "    \n",
    "    \n",
    "    t_enh = t_peaks[np.abs(t_peaks['Distance to TSS']) > enhancer_d]\n",
    "    t_prom_div = filter_only_with_divergent(\n",
    "        t_prom, divergent_threshold=divergent_thr)\n",
    "    t_enh_div = filter_only_with_divergent(\n",
    "        t_enh, divergent_threshold=divergent_thr)\n",
    "\n",
    "    t_prom_div = t_prom_div.sort_values(['Distance to TSS'])\n",
    "    \n",
    "    #Look at RefSeq vs experimental TSS\n",
    "    tss_withPeaks = t_prom_isSame['Nearest TSS'].unique()\n",
    "    refSeq = tss[tss['transcript_id'].apply(lambda x: x in tss_withPeaks)] \n",
    "    # Dont want the whole gene length, just the start of the peak \n",
    "    refSeq['Start'] = refSeq['actual_start'] #actual_start takes care of +/- strand\n",
    "    refSeq['End'] = refSeq['actual_start']\n",
    "    \n",
    "    ## Saving bed files\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    if not os.path.exists(out_dir):\n",
    "        os.mkdir(out_dir)\n",
    "\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_peaks,\n",
    "        os.path.join(out_dir, 'all_peaks'),\n",
    "        name='index',\n",
    "        score_col='Stat')    \n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_prom_div,\n",
    "        os.path.join(out_dir, 'confident_promoters'),\n",
    "        name='index',\n",
    "        score_col='Stat')\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_prom_isSame,\n",
    "        os.path.join(out_dir, 'confident_promoters_noDiv'),\n",
    "        name='index',\n",
    "        score_col='Stat')  \n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        refSeq,\n",
    "        os.path.join(out_dir, 'confident_promoters_noDiv_PICR'),\n",
    "        name='index',\n",
    "        score_col='cs')\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        t_enh_div,\n",
    "        os.path.join(out_dir, 'confident_enhancers'),\n",
    "        name='index',\n",
    "        score_col='Stat')\n",
    "    \n",
    "    save_peak_as_bed(\n",
    "        no_t_peaks,\n",
    "        os.path.join(out_dir, 'no_peaks'),\n",
    "        name='index',\n",
    "        score_col='cs')\n",
    "    \n",
    "    \n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:25:42.128263Z",
     "start_time": "2018-11-26T20:25:42.108731Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "tissues_with_atac = ['CHO','Brain','Liver','Lung','Kidney','Spleen','BMDMwt']#,'BMDM1hKLA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T20:53:55.342260Z",
     "start_time": "2018-11-26T20:39:50.297480Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:51: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:52: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/ipykernel_launcher.py:20: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/indexing.py:362: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self.obj[key] = _infer_fill_value(value)\n",
      "/home/isshamie/software/anaconda2/lib/python2.7/site-packages/pandas/core/indexing.py:543: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self.obj[item] = s\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Brain\n",
      "Liver\n",
      "Lung\n",
      "Kidney\n",
      "Spleen\n",
      "BMDMwt\n",
      "CPU times: user 14min 5s, sys: 4.23 s, total: 14min 9s\n",
      "Wall time: 14min 5s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    tissue_specific_promoter(\n",
    "        t=t,\n",
    "        anno_peaks=anno_peaks,\n",
    "        peaks_tissue_merged=peaks_tissue_merged,\n",
    "        txn_df=txn_df,\n",
    "        tss=tss)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TSS peaks near ATAC peaks. \n",
    "Using Bedtools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sort ATAC bed files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T21:05:09.963648Z",
     "start_time": "2018-11-26T21:05:09.594265Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/CHO/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.gz > Results/ATAC_results/CHO/ATAC_ppr.naive_overlap.narrowPeak.sort\n",
      "Brain\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Brain/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Brain/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "Liver\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Liver/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Liver/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "Lung\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Lung/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Lung/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "Kidney\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Kidney/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Kidney/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "Spleen\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/Spleen/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.gz > Results/ATAC_results/Spleen/ATAC_rep1-pr.naive_overlap.narrowPeak.sort\n",
      "BMDMwt\n",
      "bedtools sort -i /data/isshamie/TSS/1015_Processed/BMDMwt/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.gz > Results/ATAC_results/BMDMwt/ATAC_ppr.naive_overlap.narrowPeak.sort\n"
     ]
    }
   ],
   "source": [
    "## Sort ATAC bed files\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    atac_f = glob.glob(os.path.join(data_folder, t, 'ATAC/out/peak/macs2/overlap/optimal_set/*narrowPeak.gz'))[0]\n",
    "    atac_f_out = atac_f.replace('.gz','') + '.sort'\n",
    "    atac_f_out = os.path.join(\"Results/ATAC_results/\",t,os.path.basename(atac_f).replace('.gz','.sort'))\n",
    "    \n",
    "    cmd = 'bedtools sort -i %s > %s' % (atac_f,atac_f_out)   \n",
    "    print(cmd)\n",
    "    os.system(cmd)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get peaks close to TSS within distance of 500 bp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:20:11.904803Z",
     "start_time": "2018-11-26T23:20:11.843946Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def peaks_closest(f_a, f_b, out_f, dist_thresh):\n",
    "    '''Function that calls bedtools closest to find the closest peak of two bed files, and awk to filter those within distance dist_thresh'''\n",
    "\n",
    "    cmd = \"bedtools closest -d -t first -a %s -b %s | awk '{if ($NF < %d) print }' > %s\" % (\n",
    "       f_a, f_b, dist_thresh,out_f) #t is to only report one of the atac peaks\n",
    "    \n",
    "#     cmd = \"bedtools closest -d -t first -a %s -b %s  > %s\" % (\n",
    "#         f_a, f_b,out_f)\n",
    "    \n",
    "    print(cmd)\n",
    "    os.system(cmd)\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-11-26T23:25:05.353Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ls Results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-11-26T23:22:55.319734Z",
     "start_time": "2018-11-26T23:22:55.224569Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n"
     ]
    },
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-34-896d3fac858b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m     \u001b[0;31m#ATAC peak file\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 16\u001b[0;31m     \u001b[0matac_out\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mglob\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mglob\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Results/ATAC_results\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mt\u001b[0m \u001b[0;34m,\u001b[0m \u001b[0;34m'*sort'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     17\u001b[0m     \u001b[0;32mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0matac_out\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m \u001b[0;31m#     ## The different TSS files\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "# Intersect ATAC and TSS\n",
    "dist_thresh = 500 #500 basepairs left or right\n",
    "outfolder = 'Results/peak_groups/'\n",
    "f_names = [\n",
    "    'all_peaks','confident_promoters', 'confident_enhancers', 'confident_promoters_noDiv',\n",
    "    'confident_promoters_noDiv_PICR', 'no_peaks'\n",
    "]\n",
    "\n",
    "\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    ## Outdirectory to save in\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    \n",
    "    #ATAC peak file\n",
    "    atac_out = glob.glob(os.path.join(\"Results/ATAC_results\", t , '*sort'))[0]\n",
    "    print(atac_out)\n",
    "#     ## The different TSS files\n",
    "#     for f in f_names:\n",
    "#         f_in = os.path.join(out_dir, f) + '.bed'\n",
    "#         out_f = '%s_atac_overlap_%dbp.bed' % (f_in,dist_thresh)\n",
    "#         peaks_closest(f_in,atac_out,out_f,dist_thresh=dist_thresh)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "deletable": false,
    "editable": false,
    "run_control": {
     "frozen": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/all_peaks.bed -b ../Processed/CHO/ATAC/out/peak/macs2/overlap/optimal_set/encode_pipeline_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_promoters.bed -b ../Processed/CHO/ATAC/out/peak/macs2/overlap/optimal_set/encode_pipeline_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_enhancers.bed -b ../Processed/CHO/ATAC/out/peak/macs2/overlap/optimal_set/encode_pipeline_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_promoters_noDiv.bed -b ../Processed/CHO/ATAC/out/peak/macs2/overlap/optimal_set/encode_pipeline_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed -b ../Processed/CHO/ATAC/out/peak/macs2/overlap/optimal_set/encode_pipeline_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/CHO/no_peaks.bed -b ../Processed/CHO/ATAC/out/peak/macs2/overlap/optimal_set/encode_pipeline_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/CHO/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Brain\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/all_peaks.bed -b ../Processed/Brain/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_promoters.bed -b ../Processed/Brain/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_enhancers.bed -b ../Processed/Brain/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_promoters_noDiv.bed -b ../Processed/Brain/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed -b ../Processed/Brain/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Brain/no_peaks.bed -b ../Processed/Brain/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Brain/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Liver\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/all_peaks.bed -b ../Processed/Liver/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_promoters.bed -b ../Processed/Liver/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_enhancers.bed -b ../Processed/Liver/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_promoters_noDiv.bed -b ../Processed/Liver/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed -b ../Processed/Liver/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Liver/no_peaks.bed -b ../Processed/Liver/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Liver/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Lung\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/all_peaks.bed -b ../Processed/Lung/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_promoters.bed -b ../Processed/Lung/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_enhancers.bed -b ../Processed/Lung/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_promoters_noDiv.bed -b ../Processed/Lung/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed -b ../Processed/Lung/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Lung/no_peaks.bed -b ../Processed/Lung/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Lung/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Kidney\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/all_peaks.bed -b ../Processed/Kidney/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_promoters.bed -b ../Processed/Kidney/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_enhancers.bed -b ../Processed/Kidney/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_promoters_noDiv.bed -b ../Processed/Kidney/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed -b ../Processed/Kidney/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Kidney/no_peaks.bed -b ../Processed/Kidney/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Kidney/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Spleen\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/all_peaks.bed -b ../Processed/Spleen/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_promoters.bed -b ../Processed/Spleen/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_enhancers.bed -b ../Processed/Spleen/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_promoters_noDiv.bed -b ../Processed/Spleen/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed -b ../Processed/Spleen/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/Spleen/no_peaks.bed -b ../Processed/Spleen/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_rep1-pr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/Spleen/no_peaks.bed_atac_overlap_100bp.bed\n",
      "BMDMwt\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/all_peaks.bed -b ../Processed/BMDMwt/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/all_peaks.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_promoters.bed -b ../Processed/BMDMwt/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_enhancers.bed -b ../Processed/BMDMwt/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed -b ../Processed/BMDMwt/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed -b ../Processed/BMDMwt/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "bedtools closest -d -t first -a Results/peak_groups/BMDMwt/no_peaks.bed -b ../Processed/BMDMwt/ATAC/out/peak/macs2/overlap/optimal_set/ATAC_ppr.naive_overlap.narrowPeak.sort | awk '{if ($NF < 100) print }' > Results/peak_groups/BMDMwt/no_peaks.bed_atac_overlap_100bp.bed\n"
     ]
    }
   ],
   "source": [
    "# Intersect ATAC and TSS\n",
    "dist_thresh = 100 #500 basepairs left or right\n",
    "outfolder = 'Results/peak_groups/'\n",
    "f_names = [\n",
    "    'all_peaks','confident_promoters', 'confident_enhancers', 'confident_promoters_noDiv',\n",
    "    'confident_promoters_noDiv_PICR', 'no_peaks'\n",
    "]\n",
    "\n",
    "\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    ## Outdirectory to save in\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    \n",
    "    #ATAC peak file\n",
    "    atac_out = glob.glob('../Processed/' + t + '/ATAC/out/peak/macs2/overlap/optimal_set/*sort')[0]\n",
    "    \n",
    "    ## The different TSS files\n",
    "    for f in f_names:\n",
    "        f_in = os.path.join(out_dir, f) + '.bed'\n",
    "        out_f = '%s_atac_overlap_%dbp.bed' % (f_in,dist_thresh)\n",
    "        peaks_closest(f_in,atac_out,out_f,dist_thresh=dist_thresh)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Use wc -l to determine the number of peaks that are near ATAC, compared to total\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHO\n",
      "Results/peak_groups/CHO/all_peaks.bed\n",
      "Results/peak_groups/CHO/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/all_peaks.bed\n",
      "Results/peak_groups/CHO/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters.bed\n",
      "Results/peak_groups/CHO/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters.bed\n",
      "Results/peak_groups/CHO/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/confident_enhancers.bed\n",
      "Results/peak_groups/CHO/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/confident_enhancers.bed\n",
      "Results/peak_groups/CHO/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/CHO/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/CHO/no_peaks.bed\n",
      "Results/peak_groups/CHO/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/CHO/no_peaks.bed\n",
      "Results/peak_groups/CHO/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Brain\n",
      "Results/peak_groups/Brain/all_peaks.bed\n",
      "Results/peak_groups/Brain/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/all_peaks.bed\n",
      "Results/peak_groups/Brain/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters.bed\n",
      "Results/peak_groups/Brain/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters.bed\n",
      "Results/peak_groups/Brain/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/confident_enhancers.bed\n",
      "Results/peak_groups/Brain/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/confident_enhancers.bed\n",
      "Results/peak_groups/Brain/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Brain/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Brain/no_peaks.bed\n",
      "Results/peak_groups/Brain/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Brain/no_peaks.bed\n",
      "Results/peak_groups/Brain/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Liver\n",
      "Results/peak_groups/Liver/all_peaks.bed\n",
      "Results/peak_groups/Liver/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/all_peaks.bed\n",
      "Results/peak_groups/Liver/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters.bed\n",
      "Results/peak_groups/Liver/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters.bed\n",
      "Results/peak_groups/Liver/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/confident_enhancers.bed\n",
      "Results/peak_groups/Liver/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/confident_enhancers.bed\n",
      "Results/peak_groups/Liver/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Liver/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Liver/no_peaks.bed\n",
      "Results/peak_groups/Liver/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Liver/no_peaks.bed\n",
      "Results/peak_groups/Liver/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Lung\n",
      "Results/peak_groups/Lung/all_peaks.bed\n",
      "Results/peak_groups/Lung/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/all_peaks.bed\n",
      "Results/peak_groups/Lung/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters.bed\n",
      "Results/peak_groups/Lung/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters.bed\n",
      "Results/peak_groups/Lung/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/confident_enhancers.bed\n",
      "Results/peak_groups/Lung/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/confident_enhancers.bed\n",
      "Results/peak_groups/Lung/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Lung/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Lung/no_peaks.bed\n",
      "Results/peak_groups/Lung/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Lung/no_peaks.bed\n",
      "Results/peak_groups/Lung/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Kidney\n",
      "Results/peak_groups/Kidney/all_peaks.bed\n",
      "Results/peak_groups/Kidney/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/all_peaks.bed\n",
      "Results/peak_groups/Kidney/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters.bed\n",
      "Results/peak_groups/Kidney/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters.bed\n",
      "Results/peak_groups/Kidney/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/confident_enhancers.bed\n",
      "Results/peak_groups/Kidney/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/confident_enhancers.bed\n",
      "Results/peak_groups/Kidney/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Kidney/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Kidney/no_peaks.bed\n",
      "Results/peak_groups/Kidney/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Kidney/no_peaks.bed\n",
      "Results/peak_groups/Kidney/no_peaks.bed_atac_overlap_500bp.bed\n",
      "Spleen\n",
      "Results/peak_groups/Spleen/all_peaks.bed\n",
      "Results/peak_groups/Spleen/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/all_peaks.bed\n",
      "Results/peak_groups/Spleen/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters.bed\n",
      "Results/peak_groups/Spleen/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters.bed\n",
      "Results/peak_groups/Spleen/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Spleen/confident_enhancers.bed\n",
      "Results/peak_groups/Spleen/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/confident_enhancers.bed\n",
      "Results/peak_groups/Spleen/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/Spleen/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/Spleen/no_peaks.bed\n",
      "Results/peak_groups/Spleen/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/Spleen/no_peaks.bed\n",
      "Results/peak_groups/Spleen/no_peaks.bed_atac_overlap_500bp.bed\n",
      "BMDMwt\n",
      "Results/peak_groups/BMDMwt/all_peaks.bed\n",
      "Results/peak_groups/BMDMwt/all_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/all_peaks.bed\n",
      "Results/peak_groups/BMDMwt/all_peaks.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_enhancers.bed\n",
      "Results/peak_groups/BMDMwt/confident_enhancers.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_enhancers.bed\n",
      "Results/peak_groups/BMDMwt/confident_enhancers.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed\n",
      "Results/peak_groups/BMDMwt/confident_promoters_noDiv_PICR.bed_atac_overlap_500bp.bed\n",
      "Results/peak_groups/BMDMwt/no_peaks.bed\n",
      "Results/peak_groups/BMDMwt/no_peaks.bed_atac_overlap_100bp.bed\n",
      "Results/peak_groups/BMDMwt/no_peaks.bed\n",
      "Results/peak_groups/BMDMwt/no_peaks.bed_atac_overlap_500bp.bed\n"
     ]
    }
   ],
   "source": [
    "dist_thresholds = [100,500]\n",
    "atac_tss_df = pd.DataFrame()\n",
    "for t in tissues_with_atac:\n",
    "    print(t)\n",
    "    out_dir = os.path.join(outfolder, t)\n",
    "    ## The different TSS files\n",
    "    for f in f_names:\n",
    "        f_in = os.path.join(out_dir, f) + '.bed'\n",
    "        atac_tss_df.at[t,name] = pre_filt\n",
    "        for d in dist_thresholds:\n",
    "            out_f = '%s_atac_overlap_%dbp.bed' % (f_in,d)\n",
    "            print(f_in)\n",
    "            pre_filt = wc(f_in)\n",
    "            print(out_f)\n",
    "            post_filt = wc(out_f)\n",
    "            name = os.path.basename(f_in)\n",
    "            frac = 1.0*post_filt/pre_filt\n",
    "            \n",
    "            name_atac = '_with_ATAC_%d' % d\n",
    "            ## Add to dataframe\n",
    "            \n",
    "            atac_tss_df.at[t,name + name_atac] = post_filt\n",
    "            atac_tss_df.at[t,name + name_atac + '_fraction'] = frac\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHO</th>\n",
       "      <th>Brain</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Kidney</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>BMDMwt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>all_peaks.bed</th>\n",
       "      <td>50637.000000</td>\n",
       "      <td>39482.000000</td>\n",
       "      <td>52775.000000</td>\n",
       "      <td>36256.000000</td>\n",
       "      <td>43959.000000</td>\n",
       "      <td>46161.000000</td>\n",
       "      <td>85555.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all_peaks.bed_with_ATAC_100</th>\n",
       "      <td>42905.000000</td>\n",
       "      <td>30715.000000</td>\n",
       "      <td>7692.000000</td>\n",
       "      <td>22781.000000</td>\n",
       "      <td>34629.000000</td>\n",
       "      <td>25370.000000</td>\n",
       "      <td>38523.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all_peaks.bed_with_ATAC_100_fraction</th>\n",
       "      <td>0.847305</td>\n",
       "      <td>0.777949</td>\n",
       "      <td>0.145751</td>\n",
       "      <td>0.628337</td>\n",
       "      <td>0.787757</td>\n",
       "      <td>0.549598</td>\n",
       "      <td>0.450272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all_peaks.bed_with_ATAC_500</th>\n",
       "      <td>45820.000000</td>\n",
       "      <td>33064.000000</td>\n",
       "      <td>9495.000000</td>\n",
       "      <td>25590.000000</td>\n",
       "      <td>37135.000000</td>\n",
       "      <td>29273.000000</td>\n",
       "      <td>44334.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>all_peaks.bed_with_ATAC_500_fraction</th>\n",
       "      <td>0.904872</td>\n",
       "      <td>0.837445</td>\n",
       "      <td>0.179915</td>\n",
       "      <td>0.705814</td>\n",
       "      <td>0.844764</td>\n",
       "      <td>0.634150</td>\n",
       "      <td>0.518193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters.bed</th>\n",
       "      <td>5040.000000</td>\n",
       "      <td>3933.000000</td>\n",
       "      <td>2592.000000</td>\n",
       "      <td>3273.000000</td>\n",
       "      <td>3316.000000</td>\n",
       "      <td>3759.000000</td>\n",
       "      <td>3164.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters.bed_with_ATAC_100</th>\n",
       "      <td>4682.000000</td>\n",
       "      <td>3783.000000</td>\n",
       "      <td>938.000000</td>\n",
       "      <td>2873.000000</td>\n",
       "      <td>3124.000000</td>\n",
       "      <td>3127.000000</td>\n",
       "      <td>3018.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters.bed_with_ATAC_100_fraction</th>\n",
       "      <td>0.928968</td>\n",
       "      <td>0.961861</td>\n",
       "      <td>0.361883</td>\n",
       "      <td>0.877788</td>\n",
       "      <td>0.942099</td>\n",
       "      <td>0.831870</td>\n",
       "      <td>0.953856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters.bed_with_ATAC_500</th>\n",
       "      <td>4907.000000</td>\n",
       "      <td>3882.000000</td>\n",
       "      <td>1100.000000</td>\n",
       "      <td>3035.000000</td>\n",
       "      <td>3225.000000</td>\n",
       "      <td>3405.000000</td>\n",
       "      <td>3135.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters.bed_with_ATAC_500_fraction</th>\n",
       "      <td>0.973611</td>\n",
       "      <td>0.987033</td>\n",
       "      <td>0.424383</td>\n",
       "      <td>0.927284</td>\n",
       "      <td>0.972557</td>\n",
       "      <td>0.905826</td>\n",
       "      <td>0.990834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_enhancers.bed</th>\n",
       "      <td>1018.000000</td>\n",
       "      <td>280.000000</td>\n",
       "      <td>481.000000</td>\n",
       "      <td>191.000000</td>\n",
       "      <td>546.000000</td>\n",
       "      <td>416.000000</td>\n",
       "      <td>579.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_enhancers.bed_with_ATAC_100</th>\n",
       "      <td>945.000000</td>\n",
       "      <td>250.000000</td>\n",
       "      <td>51.000000</td>\n",
       "      <td>125.000000</td>\n",
       "      <td>475.000000</td>\n",
       "      <td>238.000000</td>\n",
       "      <td>520.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_enhancers.bed_with_ATAC_100_fraction</th>\n",
       "      <td>0.928291</td>\n",
       "      <td>0.892857</td>\n",
       "      <td>0.106029</td>\n",
       "      <td>0.654450</td>\n",
       "      <td>0.869963</td>\n",
       "      <td>0.572115</td>\n",
       "      <td>0.898100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_enhancers.bed_with_ATAC_500</th>\n",
       "      <td>978.000000</td>\n",
       "      <td>261.000000</td>\n",
       "      <td>60.000000</td>\n",
       "      <td>136.000000</td>\n",
       "      <td>497.000000</td>\n",
       "      <td>265.000000</td>\n",
       "      <td>547.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_enhancers.bed_with_ATAC_500_fraction</th>\n",
       "      <td>0.960707</td>\n",
       "      <td>0.932143</td>\n",
       "      <td>0.124740</td>\n",
       "      <td>0.712042</td>\n",
       "      <td>0.910256</td>\n",
       "      <td>0.637019</td>\n",
       "      <td>0.944732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv.bed</th>\n",
       "      <td>12730.000000</td>\n",
       "      <td>13927.000000</td>\n",
       "      <td>10777.000000</td>\n",
       "      <td>12424.000000</td>\n",
       "      <td>11846.000000</td>\n",
       "      <td>13026.000000</td>\n",
       "      <td>12302.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv.bed_with_ATAC_100</th>\n",
       "      <td>11289.000000</td>\n",
       "      <td>12235.000000</td>\n",
       "      <td>3262.000000</td>\n",
       "      <td>9592.000000</td>\n",
       "      <td>10407.000000</td>\n",
       "      <td>9304.000000</td>\n",
       "      <td>10031.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv.bed_with_ATAC_100_fraction</th>\n",
       "      <td>0.886803</td>\n",
       "      <td>0.878509</td>\n",
       "      <td>0.302682</td>\n",
       "      <td>0.772054</td>\n",
       "      <td>0.878524</td>\n",
       "      <td>0.714264</td>\n",
       "      <td>0.815396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv.bed_with_ATAC_500</th>\n",
       "      <td>12061.000000</td>\n",
       "      <td>13000.000000</td>\n",
       "      <td>3940.000000</td>\n",
       "      <td>10476.000000</td>\n",
       "      <td>11035.000000</td>\n",
       "      <td>10504.000000</td>\n",
       "      <td>11035.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv.bed_with_ATAC_500_fraction</th>\n",
       "      <td>0.947447</td>\n",
       "      <td>0.933439</td>\n",
       "      <td>0.365593</td>\n",
       "      <td>0.843207</td>\n",
       "      <td>0.931538</td>\n",
       "      <td>0.806387</td>\n",
       "      <td>0.897009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv_PICR.bed</th>\n",
       "      <td>11035.000000</td>\n",
       "      <td>12304.000000</td>\n",
       "      <td>9771.000000</td>\n",
       "      <td>11162.000000</td>\n",
       "      <td>10697.000000</td>\n",
       "      <td>11392.000000</td>\n",
       "      <td>10760.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv_PICR.bed_with_ATAC_100</th>\n",
       "      <td>9097.000000</td>\n",
       "      <td>10147.000000</td>\n",
       "      <td>2846.000000</td>\n",
       "      <td>8103.000000</td>\n",
       "      <td>8821.000000</td>\n",
       "      <td>7903.000000</td>\n",
       "      <td>8531.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv_PICR.bed_with_ATAC_100_fraction</th>\n",
       "      <td>0.824377</td>\n",
       "      <td>0.824691</td>\n",
       "      <td>0.291270</td>\n",
       "      <td>0.725945</td>\n",
       "      <td>0.824624</td>\n",
       "      <td>0.693732</td>\n",
       "      <td>0.792844</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv_PICR.bed_with_ATAC_500</th>\n",
       "      <td>10246.000000</td>\n",
       "      <td>11286.000000</td>\n",
       "      <td>3548.000000</td>\n",
       "      <td>9221.000000</td>\n",
       "      <td>9792.000000</td>\n",
       "      <td>9041.000000</td>\n",
       "      <td>9539.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>confident_promoters_noDiv_PICR.bed_with_ATAC_500_fraction</th>\n",
       "      <td>0.928500</td>\n",
       "      <td>0.917263</td>\n",
       "      <td>0.363115</td>\n",
       "      <td>0.826106</td>\n",
       "      <td>0.915397</td>\n",
       "      <td>0.793627</td>\n",
       "      <td>0.886524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no_peaks.bed</th>\n",
       "      <td>16368.000000</td>\n",
       "      <td>16085.000000</td>\n",
       "      <td>17004.000000</td>\n",
       "      <td>16987.000000</td>\n",
       "      <td>16730.000000</td>\n",
       "      <td>16754.000000</td>\n",
       "      <td>16775.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no_peaks.bed_with_ATAC_100</th>\n",
       "      <td>13638.000000</td>\n",
       "      <td>14424.000000</td>\n",
       "      <td>7460.000000</td>\n",
       "      <td>13536.000000</td>\n",
       "      <td>15208.000000</td>\n",
       "      <td>12943.000000</td>\n",
       "      <td>15260.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no_peaks.bed_with_ATAC_100_fraction</th>\n",
       "      <td>0.833211</td>\n",
       "      <td>0.896736</td>\n",
       "      <td>0.438720</td>\n",
       "      <td>0.796845</td>\n",
       "      <td>0.909026</td>\n",
       "      <td>0.772532</td>\n",
       "      <td>0.909687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no_peaks.bed_with_ATAC_500</th>\n",
       "      <td>13837.000000</td>\n",
       "      <td>14629.000000</td>\n",
       "      <td>7834.000000</td>\n",
       "      <td>13893.000000</td>\n",
       "      <td>15389.000000</td>\n",
       "      <td>13256.000000</td>\n",
       "      <td>15429.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>no_peaks.bed_with_ATAC_500_fraction</th>\n",
       "      <td>0.845369</td>\n",
       "      <td>0.909481</td>\n",
       "      <td>0.460715</td>\n",
       "      <td>0.817861</td>\n",
       "      <td>0.919845</td>\n",
       "      <td>0.791214</td>\n",
       "      <td>0.919762</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             CHO  \\\n",
       "all_peaks.bed                                       50637.000000   \n",
       "all_peaks.bed_with_ATAC_100                         42905.000000   \n",
       "all_peaks.bed_with_ATAC_100_fraction                    0.847305   \n",
       "all_peaks.bed_with_ATAC_500                         45820.000000   \n",
       "all_peaks.bed_with_ATAC_500_fraction                    0.904872   \n",
       "confident_promoters.bed                              5040.000000   \n",
       "confident_promoters.bed_with_ATAC_100                4682.000000   \n",
       "confident_promoters.bed_with_ATAC_100_fraction          0.928968   \n",
       "confident_promoters.bed_with_ATAC_500                4907.000000   \n",
       "confident_promoters.bed_with_ATAC_500_fraction          0.973611   \n",
       "confident_enhancers.bed                              1018.000000   \n",
       "confident_enhancers.bed_with_ATAC_100                 945.000000   \n",
       "confident_enhancers.bed_with_ATAC_100_fraction          0.928291   \n",
       "confident_enhancers.bed_with_ATAC_500                 978.000000   \n",
       "confident_enhancers.bed_with_ATAC_500_fraction          0.960707   \n",
       "confident_promoters_noDiv.bed                       12730.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100         11289.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100_fra...      0.886803   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500         12061.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500_fra...      0.947447   \n",
       "confident_promoters_noDiv_PICR.bed                  11035.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_100     9097.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_10...      0.824377   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_500    10246.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_50...      0.928500   \n",
       "no_peaks.bed                                        16368.000000   \n",
       "no_peaks.bed_with_ATAC_100                          13638.000000   \n",
       "no_peaks.bed_with_ATAC_100_fraction                     0.833211   \n",
       "no_peaks.bed_with_ATAC_500                          13837.000000   \n",
       "no_peaks.bed_with_ATAC_500_fraction                     0.845369   \n",
       "\n",
       "                                                           Brain  \\\n",
       "all_peaks.bed                                       39482.000000   \n",
       "all_peaks.bed_with_ATAC_100                         30715.000000   \n",
       "all_peaks.bed_with_ATAC_100_fraction                    0.777949   \n",
       "all_peaks.bed_with_ATAC_500                         33064.000000   \n",
       "all_peaks.bed_with_ATAC_500_fraction                    0.837445   \n",
       "confident_promoters.bed                              3933.000000   \n",
       "confident_promoters.bed_with_ATAC_100                3783.000000   \n",
       "confident_promoters.bed_with_ATAC_100_fraction          0.961861   \n",
       "confident_promoters.bed_with_ATAC_500                3882.000000   \n",
       "confident_promoters.bed_with_ATAC_500_fraction          0.987033   \n",
       "confident_enhancers.bed                               280.000000   \n",
       "confident_enhancers.bed_with_ATAC_100                 250.000000   \n",
       "confident_enhancers.bed_with_ATAC_100_fraction          0.892857   \n",
       "confident_enhancers.bed_with_ATAC_500                 261.000000   \n",
       "confident_enhancers.bed_with_ATAC_500_fraction          0.932143   \n",
       "confident_promoters_noDiv.bed                       13927.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100         12235.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100_fra...      0.878509   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500         13000.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500_fra...      0.933439   \n",
       "confident_promoters_noDiv_PICR.bed                  12304.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_100    10147.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_10...      0.824691   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_500    11286.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_50...      0.917263   \n",
       "no_peaks.bed                                        16085.000000   \n",
       "no_peaks.bed_with_ATAC_100                          14424.000000   \n",
       "no_peaks.bed_with_ATAC_100_fraction                     0.896736   \n",
       "no_peaks.bed_with_ATAC_500                          14629.000000   \n",
       "no_peaks.bed_with_ATAC_500_fraction                     0.909481   \n",
       "\n",
       "                                                           Liver  \\\n",
       "all_peaks.bed                                       52775.000000   \n",
       "all_peaks.bed_with_ATAC_100                          7692.000000   \n",
       "all_peaks.bed_with_ATAC_100_fraction                    0.145751   \n",
       "all_peaks.bed_with_ATAC_500                          9495.000000   \n",
       "all_peaks.bed_with_ATAC_500_fraction                    0.179915   \n",
       "confident_promoters.bed                              2592.000000   \n",
       "confident_promoters.bed_with_ATAC_100                 938.000000   \n",
       "confident_promoters.bed_with_ATAC_100_fraction          0.361883   \n",
       "confident_promoters.bed_with_ATAC_500                1100.000000   \n",
       "confident_promoters.bed_with_ATAC_500_fraction          0.424383   \n",
       "confident_enhancers.bed                               481.000000   \n",
       "confident_enhancers.bed_with_ATAC_100                  51.000000   \n",
       "confident_enhancers.bed_with_ATAC_100_fraction          0.106029   \n",
       "confident_enhancers.bed_with_ATAC_500                  60.000000   \n",
       "confident_enhancers.bed_with_ATAC_500_fraction          0.124740   \n",
       "confident_promoters_noDiv.bed                       10777.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100          3262.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100_fra...      0.302682   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500          3940.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500_fra...      0.365593   \n",
       "confident_promoters_noDiv_PICR.bed                   9771.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_100     2846.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_10...      0.291270   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_500     3548.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_50...      0.363115   \n",
       "no_peaks.bed                                        17004.000000   \n",
       "no_peaks.bed_with_ATAC_100                           7460.000000   \n",
       "no_peaks.bed_with_ATAC_100_fraction                     0.438720   \n",
       "no_peaks.bed_with_ATAC_500                           7834.000000   \n",
       "no_peaks.bed_with_ATAC_500_fraction                     0.460715   \n",
       "\n",
       "                                                            Lung  \\\n",
       "all_peaks.bed                                       36256.000000   \n",
       "all_peaks.bed_with_ATAC_100                         22781.000000   \n",
       "all_peaks.bed_with_ATAC_100_fraction                    0.628337   \n",
       "all_peaks.bed_with_ATAC_500                         25590.000000   \n",
       "all_peaks.bed_with_ATAC_500_fraction                    0.705814   \n",
       "confident_promoters.bed                              3273.000000   \n",
       "confident_promoters.bed_with_ATAC_100                2873.000000   \n",
       "confident_promoters.bed_with_ATAC_100_fraction          0.877788   \n",
       "confident_promoters.bed_with_ATAC_500                3035.000000   \n",
       "confident_promoters.bed_with_ATAC_500_fraction          0.927284   \n",
       "confident_enhancers.bed                               191.000000   \n",
       "confident_enhancers.bed_with_ATAC_100                 125.000000   \n",
       "confident_enhancers.bed_with_ATAC_100_fraction          0.654450   \n",
       "confident_enhancers.bed_with_ATAC_500                 136.000000   \n",
       "confident_enhancers.bed_with_ATAC_500_fraction          0.712042   \n",
       "confident_promoters_noDiv.bed                       12424.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100          9592.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100_fra...      0.772054   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500         10476.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500_fra...      0.843207   \n",
       "confident_promoters_noDiv_PICR.bed                  11162.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_100     8103.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_10...      0.725945   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_500     9221.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_50...      0.826106   \n",
       "no_peaks.bed                                        16987.000000   \n",
       "no_peaks.bed_with_ATAC_100                          13536.000000   \n",
       "no_peaks.bed_with_ATAC_100_fraction                     0.796845   \n",
       "no_peaks.bed_with_ATAC_500                          13893.000000   \n",
       "no_peaks.bed_with_ATAC_500_fraction                     0.817861   \n",
       "\n",
       "                                                          Kidney  \\\n",
       "all_peaks.bed                                       43959.000000   \n",
       "all_peaks.bed_with_ATAC_100                         34629.000000   \n",
       "all_peaks.bed_with_ATAC_100_fraction                    0.787757   \n",
       "all_peaks.bed_with_ATAC_500                         37135.000000   \n",
       "all_peaks.bed_with_ATAC_500_fraction                    0.844764   \n",
       "confident_promoters.bed                              3316.000000   \n",
       "confident_promoters.bed_with_ATAC_100                3124.000000   \n",
       "confident_promoters.bed_with_ATAC_100_fraction          0.942099   \n",
       "confident_promoters.bed_with_ATAC_500                3225.000000   \n",
       "confident_promoters.bed_with_ATAC_500_fraction          0.972557   \n",
       "confident_enhancers.bed                               546.000000   \n",
       "confident_enhancers.bed_with_ATAC_100                 475.000000   \n",
       "confident_enhancers.bed_with_ATAC_100_fraction          0.869963   \n",
       "confident_enhancers.bed_with_ATAC_500                 497.000000   \n",
       "confident_enhancers.bed_with_ATAC_500_fraction          0.910256   \n",
       "confident_promoters_noDiv.bed                       11846.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100         10407.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_100_fra...      0.878524   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500         11035.000000   \n",
       "confident_promoters_noDiv.bed_with_ATAC_500_fra...      0.931538   \n",
       "confident_promoters_noDiv_PICR.bed                  10697.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_100     8821.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_10...      0.824624   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_500     9792.000000   \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_50...      0.915397   \n",
       "no_peaks.bed                                        16730.000000   \n",
       "no_peaks.bed_with_ATAC_100                          15208.000000   \n",
       "no_peaks.bed_with_ATAC_100_fraction                     0.909026   \n",
       "no_peaks.bed_with_ATAC_500                          15389.000000   \n",
       "no_peaks.bed_with_ATAC_500_fraction                     0.919845   \n",
       "\n",
       "                                                          Spleen        BMDMwt  \n",
       "all_peaks.bed                                       46161.000000  85555.000000  \n",
       "all_peaks.bed_with_ATAC_100                         25370.000000  38523.000000  \n",
       "all_peaks.bed_with_ATAC_100_fraction                    0.549598      0.450272  \n",
       "all_peaks.bed_with_ATAC_500                         29273.000000  44334.000000  \n",
       "all_peaks.bed_with_ATAC_500_fraction                    0.634150      0.518193  \n",
       "confident_promoters.bed                              3759.000000   3164.000000  \n",
       "confident_promoters.bed_with_ATAC_100                3127.000000   3018.000000  \n",
       "confident_promoters.bed_with_ATAC_100_fraction          0.831870      0.953856  \n",
       "confident_promoters.bed_with_ATAC_500                3405.000000   3135.000000  \n",
       "confident_promoters.bed_with_ATAC_500_fraction          0.905826      0.990834  \n",
       "confident_enhancers.bed                               416.000000    579.000000  \n",
       "confident_enhancers.bed_with_ATAC_100                 238.000000    520.000000  \n",
       "confident_enhancers.bed_with_ATAC_100_fraction          0.572115      0.898100  \n",
       "confident_enhancers.bed_with_ATAC_500                 265.000000    547.000000  \n",
       "confident_enhancers.bed_with_ATAC_500_fraction          0.637019      0.944732  \n",
       "confident_promoters_noDiv.bed                       13026.000000  12302.000000  \n",
       "confident_promoters_noDiv.bed_with_ATAC_100          9304.000000  10031.000000  \n",
       "confident_promoters_noDiv.bed_with_ATAC_100_fra...      0.714264      0.815396  \n",
       "confident_promoters_noDiv.bed_with_ATAC_500         10504.000000  11035.000000  \n",
       "confident_promoters_noDiv.bed_with_ATAC_500_fra...      0.806387      0.897009  \n",
       "confident_promoters_noDiv_PICR.bed                  11392.000000  10760.000000  \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_100     7903.000000   8531.000000  \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_10...      0.693732      0.792844  \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_500     9041.000000   9539.000000  \n",
       "confident_promoters_noDiv_PICR.bed_with_ATAC_50...      0.793627      0.886524  \n",
       "no_peaks.bed                                        16754.000000  16775.000000  \n",
       "no_peaks.bed_with_ATAC_100                          12943.000000  15260.000000  \n",
       "no_peaks.bed_with_ATAC_100_fraction                     0.772532      0.909687  \n",
       "no_peaks.bed_with_ATAC_500                          13256.000000  15429.000000  \n",
       "no_peaks.bed_with_ATAC_500_fraction                     0.791214      0.919762  "
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "atac_tss_df.transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction of all peaks with ATAC peak within 100 bp</th>\n",
       "      <th>Fraction of confident promoters with ATAC peak within 100 bp</th>\n",
       "      <th>Fraction of confident enhancers with ATAC peak within 100 bp</th>\n",
       "      <th>Fraction of confident promoters noDiv with ATAC peak within 100 bp</th>\n",
       "      <th>Fraction of confident promoters noDiv PICR with ATAC peak within 100 bp</th>\n",
       "      <th>Fraction of no peaks with ATAC peak within 100 bp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CHO</th>\n",
       "      <td>0.847305</td>\n",
       "      <td>0.928968</td>\n",
       "      <td>0.928291</td>\n",
       "      <td>0.886803</td>\n",
       "      <td>0.824377</td>\n",
       "      <td>0.833211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Brain</th>\n",
       "      <td>0.777949</td>\n",
       "      <td>0.961861</td>\n",
       "      <td>0.892857</td>\n",
       "      <td>0.878509</td>\n",
       "      <td>0.824691</td>\n",
       "      <td>0.896736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Liver</th>\n",
       "      <td>0.145751</td>\n",
       "      <td>0.361883</td>\n",
       "      <td>0.106029</td>\n",
       "      <td>0.302682</td>\n",
       "      <td>0.291270</td>\n",
       "      <td>0.438720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lung</th>\n",
       "      <td>0.628337</td>\n",
       "      <td>0.877788</td>\n",
       "      <td>0.654450</td>\n",
       "      <td>0.772054</td>\n",
       "      <td>0.725945</td>\n",
       "      <td>0.796845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kidney</th>\n",
       "      <td>0.787757</td>\n",
       "      <td>0.942099</td>\n",
       "      <td>0.869963</td>\n",
       "      <td>0.878524</td>\n",
       "      <td>0.824624</td>\n",
       "      <td>0.909026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Spleen</th>\n",
       "      <td>0.549598</td>\n",
       "      <td>0.831870</td>\n",
       "      <td>0.572115</td>\n",
       "      <td>0.714264</td>\n",
       "      <td>0.693732</td>\n",
       "      <td>0.772532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BMDMwt</th>\n",
       "      <td>0.450272</td>\n",
       "      <td>0.953856</td>\n",
       "      <td>0.898100</td>\n",
       "      <td>0.815396</td>\n",
       "      <td>0.792844</td>\n",
       "      <td>0.909687</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Fraction of all peaks with ATAC peak within 100 bp  \\\n",
       "CHO                                              0.847305    \n",
       "Brain                                            0.777949    \n",
       "Liver                                            0.145751    \n",
       "Lung                                             0.628337    \n",
       "Kidney                                           0.787757    \n",
       "Spleen                                           0.549598    \n",
       "BMDMwt                                           0.450272    \n",
       "\n",
       "        Fraction of confident promoters with ATAC peak within 100 bp  \\\n",
       "CHO                                              0.928968              \n",
       "Brain                                            0.961861              \n",
       "Liver                                            0.361883              \n",
       "Lung                                             0.877788              \n",
       "Kidney                                           0.942099              \n",
       "Spleen                                           0.831870              \n",
       "BMDMwt                                           0.953856              \n",
       "\n",
       "        Fraction of confident enhancers with ATAC peak within 100 bp  \\\n",
       "CHO                                              0.928291              \n",
       "Brain                                            0.892857              \n",
       "Liver                                            0.106029              \n",
       "Lung                                             0.654450              \n",
       "Kidney                                           0.869963              \n",
       "Spleen                                           0.572115              \n",
       "BMDMwt                                           0.898100              \n",
       "\n",
       "        Fraction of confident promoters noDiv with ATAC peak within 100 bp  \\\n",
       "CHO                                              0.886803                    \n",
       "Brain                                            0.878509                    \n",
       "Liver                                            0.302682                    \n",
       "Lung                                             0.772054                    \n",
       "Kidney                                           0.878524                    \n",
       "Spleen                                           0.714264                    \n",
       "BMDMwt                                           0.815396                    \n",
       "\n",
       "        Fraction of confident promoters noDiv PICR with ATAC peak within 100 bp  \\\n",
       "CHO                                              0.824377                         \n",
       "Brain                                            0.824691                         \n",
       "Liver                                            0.291270                         \n",
       "Lung                                             0.725945                         \n",
       "Kidney                                           0.824624                         \n",
       "Spleen                                           0.693732                         \n",
       "BMDMwt                                           0.792844                         \n",
       "\n",
       "        Fraction of no peaks with ATAC peak within 100 bp  \n",
       "CHO                                              0.833211  \n",
       "Brain                                            0.896736  \n",
       "Liver                                            0.438720  \n",
       "Lung                                             0.796845  \n",
       "Kidney                                           0.909026  \n",
       "Spleen                                           0.772532  \n",
       "BMDMwt                                           0.909687  "
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_names = np.array(atac_tss_df.columns.values)\n",
    "to_drop = col_names[map(lambda x: 'fraction' not in x or '100' not in x,atac_tss_df.columns.values)]\n",
    "output = atac_tss_df.drop(to_drop,axis=1) \n",
    "\n",
    "col_names = np.array(output.columns.values)\n",
    "for i,name in enumerate((col_names)):\n",
    "    name = name.replace('.bed','')\n",
    "    name = name.replace('fraction','')\n",
    "    name = name.replace('ATAC','ATAC peak within')\n",
    "    name = 'Fraction of ' + name\n",
    "    name = name + 'bp'\n",
    "    col_names[i] = name.replace('_',' ')\n",
    "output.columns = col_names\n",
    "\n",
    "output"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---------"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
